{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Author: Ronny F. Pacheco Date: Sep 2024\n",
    "Copyright: © 2024 Ronny Pacheco License: MIT License\n",
    "\n",
    "---\n",
    "\n",
    "MIT License\n",
    "\n",
    "Copyright (c) 2024 Ronny Pacheco\n",
    "\n",
    "Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the \"Software\"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:\n",
    "\n",
    "The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.\n",
    "\n",
    "THE SOFTWARE IS PROVIDED \"AS IS\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Needed modules"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:20:13.296633Z",
     "start_time": "2024-10-03T08:20:12.944186Z"
    }
   },
   "source": [
    "# Load the needed libraries\n",
    "import pickle\n",
    "import os\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import json"
   ],
   "outputs": [],
   "execution_count": 1
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:20:15.198883Z",
     "start_time": "2024-10-03T08:20:15.196266Z"
    }
   },
   "source": [
    "# https://kioku-space.com/en/jupyter-skip-execution/\n",
    "from IPython.core.magic import register_cell_magic # type: ignore\n",
    "\n",
    "\n",
    "@register_cell_magic\n",
    "def skip(line, cell):\n",
    "    print('Skipping cell')\n",
    "    if line and cell:\n",
    "        pass\n",
    "    return"
   ],
   "outputs": [],
   "execution_count": 2
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pickle save"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:20:24.009179Z",
     "start_time": "2024-10-03T08:20:24.004106Z"
    }
   },
   "source": [
    "%%skip\n",
    "# =============================================================================\n",
    "# main function\n",
    "# =============================================================================\n",
    "def data_save_load(option):\n",
    "    \"\"\"\n",
    "    This function is used to save or load data for the jupyter notebook\n",
    "    \"\"\"\n",
    "    path_folder = \"ipynb_db\"  # Folder to save variables\n",
    "    os.makedirs(path_folder, exist_ok=True)  # Create folder if not exist\n",
    "    notebook_name = os.path.basename(os.path.abspath(''))\n",
    "    path_file = os.path.join(path_folder, f\"{notebook_name}.variables.pkl\") # Path to save the variables\n",
    "\n",
    "    if option == \"save\":\n",
    "        with open(path_file, \"wb\") as pickle_file:\n",
    "            dict_variables = {\n",
    "                \"neg_to_filter\" : neg_to_filter\n",
    "            }\n",
    "            pickle.dump(dict_variables, pickle_file)\n",
    "    elif option == \"load\":\n",
    "        with open(path_file, \"rb\") as pickle_file:\n",
    "            variables = pickle.load(pickle_file)\n",
    "        # Now load the variables\n",
    "        for pickle_key, pickle_value in variables.items():\n",
    "            print(f\"* Loading variable: {pickle_key}\")\n",
    "            globals()[pickle_key] = pickle_value\n",
    "# =============================================================================\n",
    "# Call the function\n",
    "# =============================================================================\n",
    "data_save_load(option=\"load\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "* Loading variable: neg_to_filter\n"
     ]
    }
   ],
   "execution_count": 3
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Prepare Data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load negative (or rejected) elements"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:28.129767Z",
     "start_time": "2024-10-03T08:01:28.109940Z"
    }
   },
   "source": [
    "# Let's load the negative nada\n",
    "neg_df = pd.read_csv(\"./data/negative_database_nomatch_corrected_named.csv\", sep=\",\", header=0)\n",
    "print(neg_df.shape)\n",
    "print(neg_df.dtypes)\n",
    "neg_df.head()"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(682, 6)\n",
      "sseqid     object\n",
      "sstart      int64\n",
      "send        int64\n",
      "sstrand    object\n",
      "sseq       object\n",
      "name       object\n",
      "dtype: object\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "    sseqid  sstart    send sstrand  \\\n",
       "0  LinJ.01   36103   36242    plus   \n",
       "1  LinJ.01  113760  114388    plus   \n",
       "2  LinJ.01  146412  146530    plus   \n",
       "3  LinJ.01  261866  262439    plus   \n",
       "4  LinJ.01  271363  271650    plus   \n",
       "\n",
       "                                                sseq                   name  \n",
       "0  AGACAGACCGACACACGCAGCCGTGTGATGCCGCCGCCGAGGGCAG...  rejected_noCDS_c01.10  \n",
       "1  CAGCGCCATGCACGACATGGCCGCTGACGTCCGTAGCCCTAACTCG...  rejected_noCDS_c01.20  \n",
       "2  GCGAATTGTGTTCTGCGCATGCCTCTTCTCTGCCGTGCAGCATGCG...  rejected_noCDS_c01.30  \n",
       "3  CGGACTTGGCAAGTGGCCGCCATCGATGAAAACGCACCATGCCTTT...  rejected_noCDS_c01.40  \n",
       "4  CGAACGCCGCCCTCAATCGCGCGCTGAACTTCACGCGGCGGTCGAC...  rejected_noCDS_c01.50  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sseqid</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>sstrand</th>\n",
       "      <th>sseq</th>\n",
       "      <th>name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>36103</td>\n",
       "      <td>36242</td>\n",
       "      <td>plus</td>\n",
       "      <td>AGACAGACCGACACACGCAGCCGTGTGATGCCGCCGCCGAGGGCAG...</td>\n",
       "      <td>rejected_noCDS_c01.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>113760</td>\n",
       "      <td>114388</td>\n",
       "      <td>plus</td>\n",
       "      <td>CAGCGCCATGCACGACATGGCCGCTGACGTCCGTAGCCCTAACTCG...</td>\n",
       "      <td>rejected_noCDS_c01.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>146412</td>\n",
       "      <td>146530</td>\n",
       "      <td>plus</td>\n",
       "      <td>GCGAATTGTGTTCTGCGCATGCCTCTTCTCTGCCGTGCAGCATGCG...</td>\n",
       "      <td>rejected_noCDS_c01.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>261866</td>\n",
       "      <td>262439</td>\n",
       "      <td>plus</td>\n",
       "      <td>CGGACTTGGCAAGTGGCCGCCATCGATGAAAACGCACCATGCCTTT...</td>\n",
       "      <td>rejected_noCDS_c01.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>271363</td>\n",
       "      <td>271650</td>\n",
       "      <td>plus</td>\n",
       "      <td>CGAACGCCGCCCTCAATCGCGCGCTGAACTTCACGCGGCGGTCGAC...</td>\n",
       "      <td>rejected_noCDS_c01.50</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 81
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### GTF data\n",
    "This one wil be **harder** to prepare"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:28.979660Z",
     "start_time": "2024-10-03T08:01:28.876903Z"
    }
   },
   "source": [
    "# Load data\n",
    "gtf_df = pd.read_csv(\"./data/20240703111001_LINF-Tabla_maestra_v3-20244_RP_v0.8.gtf\", sep=\"\\t\", header=None) \n",
    "print(gtf_df.shape)\n",
    "print(gtf_df.dtypes)\n",
    "gtf_df.head()"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(45368, 9)\n",
      "0    object\n",
      "1    object\n",
      "2    object\n",
      "3     int64\n",
      "4     int64\n",
      "5    object\n",
      "6    object\n",
      "7    object\n",
      "8    object\n",
      "dtype: object\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "         0    1           2     3     4  5  6  7  \\\n",
       "0  LinJ.01  CBM        gene  1520  5066  .  -  .   \n",
       "1  LinJ.01  CBM  transcript  1520  5066  .  -  .   \n",
       "2  LinJ.01  CBM         CDS  3710  4711  .  -  .   \n",
       "3  LinJ.01  CBM        3utr  1520  3709  .  -  .   \n",
       "4  LinJ.01  CBM        5utr  4712  5066  .  -  .   \n",
       "\n",
       "                                                   8  \n",
       "0  gene_id \"LINF_010005000\"; gene_name \"LINF_0100...  \n",
       "1  parent_id \"LINF_010005000\"; transcript_id \"LIN...  \n",
       "2  parent_id \"LINF_01T0005000\"; transcript_id \"LI...  \n",
       "3                      parent_id \"LINF_01T0005000\";   \n",
       "4                      parent_id \"LINF_01T0005000\";   "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>1520</td>\n",
       "      <td>5066</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_010005000\"; gene_name \"LINF_0100...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>CBM</td>\n",
       "      <td>transcript</td>\n",
       "      <td>1520</td>\n",
       "      <td>5066</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>parent_id \"LINF_010005000\"; transcript_id \"LIN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>CBM</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3710</td>\n",
       "      <td>4711</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>parent_id \"LINF_01T0005000\"; transcript_id \"LI...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>CBM</td>\n",
       "      <td>3utr</td>\n",
       "      <td>1520</td>\n",
       "      <td>3709</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>parent_id \"LINF_01T0005000\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>CBM</td>\n",
       "      <td>5utr</td>\n",
       "      <td>4712</td>\n",
       "      <td>5066</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>parent_id \"LINF_01T0005000\";</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 82
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "From `gtf_df`I only need columns 0, 2, 3, 4, 6 and 8"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:30.322163Z",
     "start_time": "2024-10-03T08:01:30.307458Z"
    }
   },
   "source": [
    "# Get from `gtf_df` the needed columns [0, 3, 4, 6, 8]\n",
    "gtf_df = gtf_df[[0, 2, 3, 4, 6, 8]]\n",
    "gtf_df.columns = [\"chrom\", \"feature\", \"start\", \"end\", \"strand\", \"attributes\"]\n",
    "print(gtf_df.shape)\n",
    "print(gtf_df.dtypes)\n",
    "gtf_df.head()"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(45368, 6)\n",
      "chrom         object\n",
      "feature       object\n",
      "start          int64\n",
      "end            int64\n",
      "strand        object\n",
      "attributes    object\n",
      "dtype: object\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "     chrom     feature  start   end strand  \\\n",
       "0  LinJ.01        gene   1520  5066      -   \n",
       "1  LinJ.01  transcript   1520  5066      -   \n",
       "2  LinJ.01         CDS   3710  4711      -   \n",
       "3  LinJ.01        3utr   1520  3709      -   \n",
       "4  LinJ.01        5utr   4712  5066      -   \n",
       "\n",
       "                                          attributes  \n",
       "0  gene_id \"LINF_010005000\"; gene_name \"LINF_0100...  \n",
       "1  parent_id \"LINF_010005000\"; transcript_id \"LIN...  \n",
       "2  parent_id \"LINF_01T0005000\"; transcript_id \"LI...  \n",
       "3                      parent_id \"LINF_01T0005000\";   \n",
       "4                      parent_id \"LINF_01T0005000\";   "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>feature</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>attributes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>gene</td>\n",
       "      <td>1520</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"LINF_010005000\"; gene_name \"LINF_0100...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>transcript</td>\n",
       "      <td>1520</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>parent_id \"LINF_010005000\"; transcript_id \"LIN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3710</td>\n",
       "      <td>4711</td>\n",
       "      <td>-</td>\n",
       "      <td>parent_id \"LINF_01T0005000\"; transcript_id \"LI...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>3utr</td>\n",
       "      <td>1520</td>\n",
       "      <td>3709</td>\n",
       "      <td>-</td>\n",
       "      <td>parent_id \"LINF_01T0005000\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>5utr</td>\n",
       "      <td>4712</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>parent_id \"LINF_01T0005000\";</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 83
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now the field `attributes` it's separated by \";\" and the header it's in a format like `header \"data\"`. We are going to transform the \"attributes\" column in multiple columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Transforming columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get first all the elements that appear in the attributes columns"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:35.595362Z",
     "start_time": "2024-10-03T08:01:32.591667Z"
    }
   },
   "source": [
    "# Let's count first the number of elements in the `attributes` column\n",
    "atr_dict = {}\n",
    "for index, row in gtf_df.iterrows():\n",
    "    # print(index, \":\", sep=\"\")\n",
    "    for atr in row[\"attributes\"].split(\";\"):\n",
    "        atr = atr.strip()  # Remove leading and trailing whitespaces\n",
    "        if len(atr.strip()) == 0:  # Skip empty attribute \"\"\n",
    "            continue\n",
    "        # print(f\"\\t{'-'*50}\")\n",
    "        # print(f\"\\attribute: {atr.strip()}\")  \n",
    "        key = atr.split(\" \")[0] \n",
    "        if key not in atr_dict:\n",
    "            atr_dict[key] = 1\n",
    "\n",
    "        else:\n",
    "            atr_dict[key] += 1\n",
    "        # print(f\"\\t{atr_dict}\")\n",
    "print(atr_dict)"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'gene_id': 9861, 'gene_name': 9861, 'biotype': 17295, 'notes': 17319, 'parent_id': 35507, 'transcript_id': 18215, 'transcript_name': 9660, 'pseudogen': 49}\n"
     ]
    }
   ],
   "execution_count": 84
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:35.856587Z",
     "start_time": "2024-10-03T08:01:35.851340Z"
    }
   },
   "source": [
    "# get a list with the keys of atr_dict\n",
    "atr_keys = list(atr_dict.keys())\n",
    "print(atr_keys)"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['gene_id', 'gene_name', 'biotype', 'notes', 'parent_id', 'transcript_id', 'transcript_name', 'pseudogen']\n"
     ]
    }
   ],
   "execution_count": 85
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we'll have a list with all the elements. When indexing each row in the next steps, we can check if one of these items appear, and if not, we can add a \"None\" value to the attribute"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:40.309452Z",
     "start_time": "2024-10-03T08:01:36.551736Z"
    }
   },
   "source": [
    "# Now that we have the attributes count, let's create a dict for each element in \"test_df\" with the attributes separated\n",
    "new_col_df = []\n",
    "for index, row in gtf_df.iterrows():\n",
    "    # print(index, \":\", sep=\"\")\n",
    "    pre_data = []\n",
    "    for atr in row[\"attributes\"].split(\";\"):\n",
    "        atr = atr.strip()  # Remove leading and trailing whitespaces\n",
    "        if len(atr.strip()) == 0:  # Skip empty attribute \"\"\n",
    "            continue\n",
    "        key = atr.split(\" \")[0]\n",
    "        value = atr.split(\" \")[1].replace('\"', \"\")\n",
    "        pre_data.append({key: value})\n",
    "    \n",
    "    for elem in atr_keys: # type: ignore  # Checking if the elements from atr_keys\n",
    "        if elem not in [list(elem.keys())[0] for elem in pre_data]:  # If the element is not in pre_data, add it with value None\n",
    "            # noinspection PyUnresolvedReferences\n",
    "            pre_data.append({elem: None})\n",
    "\n",
    "    flattened_data = {key: value for sublist in pre_data for key, value in sublist.items()}\n",
    "    new_col_df.append(flattened_data)"
   ],
   "outputs": [],
   "execution_count": 86
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:40.525451Z",
     "start_time": "2024-10-03T08:01:40.346607Z"
    }
   },
   "source": [
    "# Checking how it worked\n",
    "new_col_df  "
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'gene_id': 'LINF_010005000',\n",
       "  'gene_name': 'LINF_010005000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Protein_of_unknown_function_(DUF2946)',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010005000',\n",
       "  'transcript_id': 'LINF_01T0005000',\n",
       "  'transcript_name': 'LINF_01T0005000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Protein_of_unknown_function_(DUF2946)',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005000',\n",
       "  'transcript_id': 'LINF_01T0005000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010005100',\n",
       "  'gene_name': 'LINF_010005100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Endonuclease/Exonuclease/phosphatase_family',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010005100',\n",
       "  'transcript_id': 'LINF_01T0005100',\n",
       "  'transcript_name': 'LINF_01T0005100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Endonuclease/Exonuclease/phosphatase_family',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005100',\n",
       "  'transcript_id': 'LINF_01T0005100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010005200',\n",
       "  'gene_name': 'LINF_010005200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Kinesin-13',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010005200',\n",
       "  'transcript_id': 'LINF_01T0005200',\n",
       "  'transcript_name': 'LINF_01T0005200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Kinesin-13',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005200',\n",
       "  'transcript_id': 'LINF_01T0005200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010005300',\n",
       "  'gene_name': 'LINF_010005300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010005300',\n",
       "  'transcript_id': 'LINF_01T0005300',\n",
       "  'transcript_name': 'LINF_01T0005300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005300',\n",
       "  'transcript_id': 'LINF_01T0005300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010005400',\n",
       "  'gene_name': 'LINF_010005400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Propionyl-CoA_carboxylase_-_α-subunit',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010005400',\n",
       "  'transcript_id': 'LINF_01T0005400',\n",
       "  'transcript_name': 'LINF_01T0005400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Propionyl-CoA_carboxylase_-_α-subunit',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005400',\n",
       "  'transcript_id': 'LINF_01T0005400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010005500',\n",
       "  'gene_name': 'LINF_010005500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'mitoribosomal_protein_mL85',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010005500',\n",
       "  'transcript_id': 'LINF_01T0005500',\n",
       "  'transcript_name': 'LINF_01T0005500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'mitoribosomal_protein_mL85',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005500',\n",
       "  'transcript_id': 'LINF_01T0005500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010005600',\n",
       "  'gene_name': 'LINF_010005600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'BSD-_domain_containing_protein_',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010005600',\n",
       "  'transcript_id': 'LINF_01T0005600',\n",
       "  'transcript_name': 'LINF_01T0005600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'BSD-_domain_containing_protein_',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005600',\n",
       "  'transcript_id': 'LINF_01T0005600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010005700',\n",
       "  'gene_name': 'LINF_010005700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010005700',\n",
       "  'transcript_id': 'LINF_01T0005700',\n",
       "  'transcript_name': 'LINF_01T0005700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005700',\n",
       "  'transcript_id': 'LINF_01T0005700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010005800',\n",
       "  'gene_name': 'LINF_010005800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'mitochondrial_UMP-CMP_kinase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010005800',\n",
       "  'transcript_id': 'LINF_01T0005800',\n",
       "  'transcript_name': 'LINF_01T0005800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'mitochondrial_UMP-CMP_kinase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005800',\n",
       "  'transcript_id': 'LINF_01T0005800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010005900',\n",
       "  'gene_name': 'LINF_010005900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Fucosyltransferase|FUT2',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010005900',\n",
       "  'transcript_id': 'LINF_01T0005900',\n",
       "  'transcript_name': 'LINF_01T0005900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Fucosyltransferase|FUT2',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005900',\n",
       "  'transcript_id': 'LINF_01T0005900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0005900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010006000',\n",
       "  'gene_name': 'LINF_010006000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Monothiol_glutaredoxin_',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010006000',\n",
       "  'transcript_id': 'LINF_01T0006000',\n",
       "  'transcript_name': 'LINF_01T0006000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Monothiol_glutaredoxin_',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006000',\n",
       "  'transcript_id': 'LINF_01T0006000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010006100',\n",
       "  'gene_name': 'LINF_010006100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010006100',\n",
       "  'transcript_id': 'LINF_01T0006100',\n",
       "  'transcript_name': 'LINF_01T0006100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006100',\n",
       "  'transcript_id': 'LINF_01T0006100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010006200',\n",
       "  'gene_name': 'LINF_010006200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010006200',\n",
       "  'transcript_id': 'LINF_01T0006200',\n",
       "  'transcript_name': 'LINF_01T0006200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006200',\n",
       "  'transcript_id': 'LINF_01T0006200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010006300',\n",
       "  'gene_name': 'LINF_010006300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved__',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010006300',\n",
       "  'transcript_id': 'LINF_01T0006300',\n",
       "  'transcript_name': 'LINF_01T0006300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved__',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006300',\n",
       "  'transcript_id': 'LINF_01T0006300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010006400',\n",
       "  'gene_name': 'LINF_010006400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved_',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010006400',\n",
       "  'transcript_id': 'LINF_01T0006400',\n",
       "  'transcript_name': 'LINF_01T0006400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved_',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006400',\n",
       "  'transcript_id': 'LINF_01T0006400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010006500',\n",
       "  'gene_name': 'LINF_010006500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'LSM_domain_containing_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010006500',\n",
       "  'transcript_id': 'LINF_01T0006500',\n",
       "  'transcript_name': 'LINF_01T0006500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'LSM_domain_containing_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006500',\n",
       "  'transcript_id': 'LINF_01T0006500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010006600',\n",
       "  'gene_name': 'LINF_010006600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010006600',\n",
       "  'transcript_id': 'LINF_01T0006600',\n",
       "  'transcript_name': 'LINF_01T0006600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006600',\n",
       "  'transcript_id': 'LINF_01T0006600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010006700',\n",
       "  'gene_name': 'LINF_010006700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'CLC-_type_chloride_channel',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010006700',\n",
       "  'transcript_id': 'LINF_01T0006700',\n",
       "  'transcript_name': 'LINF_01T0006700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'CLC-_type_chloride_channel',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006700',\n",
       "  'transcript_id': 'LINF_01T0006700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010006800',\n",
       "  'gene_name': 'LINF_010006800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'CW21',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010006800',\n",
       "  'transcript_id': 'LINF_01T0006800',\n",
       "  'transcript_name': 'LINF_01T0006800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'CW21',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006800',\n",
       "  'transcript_id': 'LINF_01T0006800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010006900',\n",
       "  'gene_name': 'LINF_010006900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'mt-SAF26',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010006900',\n",
       "  'transcript_id': 'LINF_01T0006900',\n",
       "  'transcript_name': 'LINF_01T0006900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'mt-SAF26',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006900',\n",
       "  'transcript_id': 'LINF_01T0006900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0006900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010007000',\n",
       "  'gene_name': 'LINF_010007000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'ZC3H27|CSBP',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010007000',\n",
       "  'transcript_id': 'LINF_01T0007000',\n",
       "  'transcript_name': 'LINF_01T0007000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'ZC3H27|CSBP',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007000',\n",
       "  'transcript_id': 'LINF_01T0007000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010007050',\n",
       "  'gene_name': 'LINF_010007050',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010007050',\n",
       "  'transcript_id': 'LINF_01T0007050',\n",
       "  'transcript_name': 'LINF_01T0007050',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010007100',\n",
       "  'gene_name': 'LINF_010007100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'cyclophilin_12|CyP12',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010007100',\n",
       "  'transcript_id': 'LINF_01T0007100',\n",
       "  'transcript_name': 'LINF_01T0007100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'cyclophilin_12|CyP12',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007100',\n",
       "  'transcript_id': 'LINF_01T0007100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010007150',\n",
       "  'gene_name': 'LINF_010007150',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010007150',\n",
       "  'transcript_id': 'LINF_01T0007150',\n",
       "  'transcript_name': 'LINF_01T0007150',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010007200',\n",
       "  'gene_name': 'LINF_010007200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved_',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010007200',\n",
       "  'transcript_id': 'LINF_01T0007200',\n",
       "  'transcript_name': 'LINF_01T0007200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved_',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007200',\n",
       "  'transcript_id': 'LINF_01T0007200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010007300',\n",
       "  'gene_name': 'LINF_010007300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'DNA_excision/_repair_protein_SNF2',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010007300',\n",
       "  'transcript_id': 'LINF_01T0007300',\n",
       "  'transcript_name': 'LINF_01T0007300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'DNA_excision/_repair_protein_SNF2',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007300',\n",
       "  'transcript_id': 'LINF_01T0007300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010007400',\n",
       "  'gene_name': 'LINF_010007400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010007400',\n",
       "  'transcript_id': 'LINF_01T0007400',\n",
       "  'transcript_name': 'LINF_01T0007400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007400',\n",
       "  'transcript_id': 'LINF_01T0007400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010007500',\n",
       "  'gene_name': 'LINF_010007500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Calmodulin_binding_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010007500',\n",
       "  'transcript_id': 'LINF_01T0007500',\n",
       "  'transcript_name': 'LINF_01T0007500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Calmodulin_binding_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007500',\n",
       "  'transcript_id': 'LINF_01T0007500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010007600',\n",
       "  'gene_name': 'LINF_010007600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'thioredoxin',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010007600',\n",
       "  'transcript_id': 'LINF_01T0007600',\n",
       "  'transcript_name': 'LINF_01T0007600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'thioredoxin',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007600',\n",
       "  'transcript_id': 'LINF_01T0007600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010007700',\n",
       "  'gene_name': 'LINF_010007700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'mt-LAF3|mitoribosomal_LSU_assembly_factor_3|pseudouridylate_synthase|PUS5',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010007700',\n",
       "  'transcript_id': 'LINF_01T0007700',\n",
       "  'transcript_name': 'LINF_01T0007700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'mt-LAF3|mitoribosomal_LSU_assembly_factor_3|pseudouridylate_synthase|PUS5',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007700',\n",
       "  'transcript_id': 'LINF_01T0007700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010007800',\n",
       "  'gene_name': 'LINF_010007800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Cytochrome_c-oxidase_biogenesis_protein_Cmc1_like_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010007800',\n",
       "  'transcript_id': 'LINF_01T0007800',\n",
       "  'transcript_name': 'LINF_01T0007800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Cytochrome_c-oxidase_biogenesis_protein_Cmc1_like_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007800',\n",
       "  'transcript_id': 'LINF_01T0007800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010007900',\n",
       "  'gene_name': 'LINF_010007900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010007900',\n",
       "  'transcript_id': 'LINF_01T0007900',\n",
       "  'transcript_name': 'LINF_01T0007900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007900',\n",
       "  'transcript_id': 'LINF_01T0007900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0007900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010008000',\n",
       "  'gene_name': 'LINF_010008000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'acidocalcisomal_exopolyphosphatase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010008000',\n",
       "  'transcript_id': 'LINF_01T0008000',\n",
       "  'transcript_name': 'LINF_01T0008000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'acidocalcisomal_exopolyphosphatase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008000',\n",
       "  'transcript_id': 'LINF_01T0008000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010008100',\n",
       "  'gene_name': 'LINF_010008100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010008100',\n",
       "  'transcript_id': 'LINF_01T0008100',\n",
       "  'transcript_name': 'LINF_01T0008100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008100',\n",
       "  'transcript_id': 'LINF_01T0008100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010008200',\n",
       "  'gene_name': 'LINF_010008200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'poly(A)_export_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010008200',\n",
       "  'transcript_id': 'LINF_01T0008200',\n",
       "  'transcript_name': 'LINF_01T0008200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'poly(A)_export_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008200',\n",
       "  'transcript_id': 'LINF_01T0008200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010008300',\n",
       "  'gene_name': 'LINF_010008300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010008300',\n",
       "  'transcript_id': 'LINF_01T0008300',\n",
       "  'transcript_name': 'LINF_01T0008300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008300',\n",
       "  'transcript_id': 'LINF_01T0008300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010008400',\n",
       "  'gene_name': 'LINF_010008400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Mitochondrial_pyruvate_carrier|MPC1',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010008400',\n",
       "  'transcript_id': 'LINF_01T0008400',\n",
       "  'transcript_name': 'LINF_01T0008400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Mitochondrial_pyruvate_carrier|MPC1',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008400',\n",
       "  'transcript_id': 'LINF_01T0008400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010008500',\n",
       "  'gene_name': 'LINF_010008500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010008500',\n",
       "  'transcript_id': 'LINF_01T0008500',\n",
       "  'transcript_name': 'LINF_01T0008500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008500',\n",
       "  'transcript_id': 'LINF_01T0008500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010008600',\n",
       "  'gene_name': 'LINF_010008600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010008600',\n",
       "  'transcript_id': 'LINF_01T0008600',\n",
       "  'transcript_name': 'LINF_01T0008600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008600',\n",
       "  'transcript_id': 'LINF_01T0008600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010008700',\n",
       "  'gene_name': 'LINF_010008700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010008700',\n",
       "  'transcript_id': 'LINF_01T0008700',\n",
       "  'transcript_name': 'LINF_01T0008700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008700',\n",
       "  'transcript_id': 'LINF_01T0008700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010008800',\n",
       "  'gene_name': 'LINF_010008800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Protein_of_unknown_function_(DUF788)',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010008800',\n",
       "  'transcript_id': 'LINF_01T0008800',\n",
       "  'transcript_name': 'LINF_01T0008800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Protein_of_unknown_function_(DUF788)',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008800',\n",
       "  'transcript_id': 'LINF_01T0008800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010008900',\n",
       "  'gene_name': 'LINF_010008900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010008900',\n",
       "  'transcript_id': 'LINF_01T0008900',\n",
       "  'transcript_name': 'LINF_01T0008900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008900',\n",
       "  'transcript_id': 'LINF_01T0008900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0008900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010009000',\n",
       "  'gene_name': 'LINF_010009000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010009000',\n",
       "  'transcript_id': 'LINF_01T0009000',\n",
       "  'transcript_name': 'LINF_01T0009000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009000',\n",
       "  'transcript_id': 'LINF_01T0009000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010009100',\n",
       "  'gene_name': 'LINF_010009100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010009100',\n",
       "  'transcript_id': 'LINF_01T0009100',\n",
       "  'transcript_name': 'LINF_01T0009100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009100',\n",
       "  'transcript_id': 'LINF_01T0009100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010009200',\n",
       "  'gene_name': 'LINF_010009200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'ribosomal_protein_S7_-_putative',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010009200',\n",
       "  'transcript_id': 'LINF_01T0009200',\n",
       "  'transcript_name': 'LINF_01T0009200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'ribosomal_protein_S7_-_putative',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009200',\n",
       "  'transcript_id': 'LINF_01T0009200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010009300',\n",
       "  'gene_name': 'LINF_010009300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'ribosomal_protein_S7|eS7',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010009300',\n",
       "  'transcript_id': 'LINF_01T0009300',\n",
       "  'transcript_name': 'LINF_01T0009300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'ribosomal_protein_S7|eS7',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009300',\n",
       "  'transcript_id': 'LINF_01T0009300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010009400',\n",
       "  'gene_name': 'LINF_010009400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Ring_finger_domain/Zn_finger-containing_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010009400',\n",
       "  'transcript_id': 'LINF_01T0009400',\n",
       "  'transcript_name': 'LINF_01T0009400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Ring_finger_domain/Zn_finger-containing_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009400',\n",
       "  'transcript_id': 'LINF_01T0009400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010009500',\n",
       "  'gene_name': 'LINF_010009500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Protein_of_unknown_function_(DUF791)',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010009500',\n",
       "  'transcript_id': 'LINF_01T0009500',\n",
       "  'transcript_name': 'LINF_01T0009500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Protein_of_unknown_function_(DUF791)',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009500',\n",
       "  'transcript_id': 'LINF_01T0009500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010009600',\n",
       "  'gene_name': 'LINF_010009600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'nicotinamidase|PNC1',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010009600',\n",
       "  'transcript_id': 'LINF_01T0009600',\n",
       "  'transcript_name': 'LINF_01T0009600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'nicotinamidase|PNC1',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009600',\n",
       "  'transcript_id': 'LINF_01T0009600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010009700',\n",
       "  'gene_name': 'LINF_010009700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Protein_of_unknown_function_-_conserved|HYP2',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010009700',\n",
       "  'transcript_id': 'LINF_01T0009700',\n",
       "  'transcript_name': 'LINF_01T0009700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Protein_of_unknown_function_-_conserved|HYP2',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009700',\n",
       "  'transcript_id': 'LINF_01T0009700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010009800',\n",
       "  'gene_name': 'LINF_010009800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'long_chain_fatty_acid-CoA_ligase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010009800',\n",
       "  'transcript_id': 'LINF_01T0009800',\n",
       "  'transcript_name': 'LINF_01T0009800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'long_chain_fatty_acid-CoA_ligase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009800',\n",
       "  'transcript_id': 'LINF_01T0009800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010009900',\n",
       "  'gene_name': 'LINF_010009900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'beta_eliminating_lyase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010009900',\n",
       "  'transcript_id': 'LINF_01T0009900',\n",
       "  'transcript_name': 'LINF_01T0009900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'beta_eliminating_lyase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009900',\n",
       "  'transcript_id': 'LINF_01T0009900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0009900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010010000',\n",
       "  'gene_name': 'LINF_010010000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'long_chain_fatty_acid_CoA_ligase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010010000',\n",
       "  'transcript_id': 'LINF_01T0010000',\n",
       "  'transcript_name': 'LINF_01T0010000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'long_chain_fatty_acid_CoA_ligase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010000',\n",
       "  'transcript_id': 'LINF_01T0010000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010010100',\n",
       "  'gene_name': 'LINF_010010100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'long_chain_fatty_acid_CoA_ligase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010010100',\n",
       "  'transcript_id': 'LINF_01T0010100',\n",
       "  'transcript_name': 'LINF_01T0010100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'long_chain_fatty_acid_CoA_ligase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010100',\n",
       "  'transcript_id': 'LINF_01T0010100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010010200',\n",
       "  'gene_name': 'LINF_010010200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'fatty_acyl_CoA_synthetase_2',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010010200',\n",
       "  'transcript_id': 'LINF_01T0010200',\n",
       "  'transcript_name': 'LINF_01T0010200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'fatty_acyl_CoA_synthetase_2',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010200',\n",
       "  'transcript_id': 'LINF_01T0010200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010010300',\n",
       "  'gene_name': 'LINF_010010300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'long_chain_fatty_acid-CoA_ligase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010010300',\n",
       "  'transcript_id': 'LINF_01T0010300',\n",
       "  'transcript_name': 'LINF_01T0010300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'long_chain_fatty_acid-CoA_ligase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010300',\n",
       "  'transcript_id': 'LINF_01T0010300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010010350',\n",
       "  'gene_name': 'LINF_010010350',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010010350',\n",
       "  'transcript_id': 'LINF_01T0010350',\n",
       "  'transcript_name': 'LINF_01T0010350',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010010400',\n",
       "  'gene_name': 'LINF_010010400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'long_chain_fatty_acid-CoA_ligase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010010400',\n",
       "  'transcript_id': 'LINF_01T0010400',\n",
       "  'transcript_name': 'LINF_01T0010400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'long_chain_fatty_acid-CoA_ligase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010400',\n",
       "  'transcript_id': 'LINF_01T0010400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010010500',\n",
       "  'gene_name': 'LINF_010010500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010010500',\n",
       "  'transcript_id': 'LINF_01T0010500',\n",
       "  'transcript_name': 'LINF_01T0010500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010500',\n",
       "  'transcript_id': 'LINF_01T0010500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010010600',\n",
       "  'gene_name': 'LINF_010010600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010010600',\n",
       "  'transcript_id': 'LINF_01T0010600',\n",
       "  'transcript_name': 'LINF_01T0010600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010600',\n",
       "  'transcript_id': 'LINF_01T0010600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010010700',\n",
       "  'gene_name': 'LINF_010010700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Pre-mRNA_processing_factor_3_(PRP3)',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010010700',\n",
       "  'transcript_id': 'LINF_01T0010700',\n",
       "  'transcript_name': 'LINF_01T0010700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Pre-mRNA_processing_factor_3_(PRP3)',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010700',\n",
       "  'transcript_id': 'LINF_01T0010700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010010800',\n",
       "  'gene_name': 'LINF_010010800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'tricarboxylate_carrier',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010010800',\n",
       "  'transcript_id': 'LINF_01T0010800',\n",
       "  'transcript_name': 'LINF_01T0010800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'tricarboxylate_carrier',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010800',\n",
       "  'transcript_id': 'LINF_01T0010800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010010900',\n",
       "  'gene_name': 'LINF_010010900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010010900',\n",
       "  'transcript_id': 'LINF_01T0010900',\n",
       "  'transcript_name': 'LINF_01T0010900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010900',\n",
       "  'transcript_id': 'LINF_01T0010900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0010900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010011000',\n",
       "  'gene_name': 'LINF_010011000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'RNA_ligase_KREL1',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010011000',\n",
       "  'transcript_id': 'LINF_01T0011000',\n",
       "  'transcript_name': 'LINF_01T0011000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'RNA_ligase_KREL1',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011000',\n",
       "  'transcript_id': 'LINF_01T0011000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010011100',\n",
       "  'gene_name': 'LINF_010011100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Rab-GTPase-TBC_domain_containing_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010011100',\n",
       "  'transcript_id': 'LINF_01T0011100',\n",
       "  'transcript_name': 'LINF_01T0011100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Rab-GTPase-TBC_domain_containing_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011100',\n",
       "  'transcript_id': 'LINF_01T0011100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010011200',\n",
       "  'gene_name': 'LINF_010011200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'DNA-damage_inducible_protein_DDI1-like_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010011200',\n",
       "  'transcript_id': 'LINF_01T0011200',\n",
       "  'transcript_name': 'LINF_01T0011200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'DNA-damage_inducible_protein_DDI1-like_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011200',\n",
       "  'transcript_id': 'LINF_01T0011200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010011300',\n",
       "  'gene_name': 'LINF_010011300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010011300',\n",
       "  'transcript_id': 'LINF_01T0011300',\n",
       "  'transcript_name': 'LINF_01T0011300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011300',\n",
       "  'transcript_id': 'LINF_01T0011300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010011400',\n",
       "  'gene_name': 'LINF_010011400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010011400',\n",
       "  'transcript_id': 'LINF_01T0011400',\n",
       "  'transcript_name': 'LINF_01T0011400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011400',\n",
       "  'transcript_id': 'LINF_01T0011400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010011500',\n",
       "  'gene_name': 'LINF_010011500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'HSP70.a',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010011500',\n",
       "  'transcript_id': 'LINF_01T0011500',\n",
       "  'transcript_name': 'LINF_01T0011500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'HSP70.a',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011500',\n",
       "  'transcript_id': 'LINF_01T0011500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010011600',\n",
       "  'gene_name': 'LINF_010011600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'metallo-peptidase_-_Clan_ME_-_Family_M16',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010011600',\n",
       "  'transcript_id': 'LINF_01T0011600',\n",
       "  'transcript_name': 'LINF_01T0011600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'metallo-peptidase_-_Clan_ME_-_Family_M16',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011600',\n",
       "  'transcript_id': 'LINF_01T0011600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010011700',\n",
       "  'gene_name': 'LINF_010011700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010011700',\n",
       "  'transcript_id': 'LINF_01T0011700',\n",
       "  'transcript_name': 'LINF_01T0011700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011700',\n",
       "  'transcript_id': 'LINF_01T0011700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010011800',\n",
       "  'gene_name': 'LINF_010011800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010011800',\n",
       "  'transcript_id': 'LINF_01T0011800',\n",
       "  'transcript_name': 'LINF_01T0011800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011800',\n",
       "  'transcript_id': 'LINF_01T0011800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010011850',\n",
       "  'gene_name': 'LINF_010011850',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010011850',\n",
       "  'transcript_id': 'LINF_01T0011850',\n",
       "  'transcript_name': 'LINF_01T0011850',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010011900',\n",
       "  'gene_name': 'LINF_010011900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010011900',\n",
       "  'transcript_id': 'LINF_01T0011900',\n",
       "  'transcript_name': 'LINF_01T0011900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011900',\n",
       "  'transcript_id': 'LINF_01T0011900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0011900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010012000',\n",
       "  'gene_name': 'LINF_010012000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'PLAC8_family',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010012000',\n",
       "  'transcript_id': 'LINF_01T0012000',\n",
       "  'transcript_name': 'LINF_01T0012000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'PLAC8_family',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012000',\n",
       "  'transcript_id': 'LINF_01T0012000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010012050',\n",
       "  'gene_name': 'LINF_010012050',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010012050',\n",
       "  'transcript_id': 'LINF_01T0012050',\n",
       "  'transcript_name': 'LINF_01T0012050',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010012100',\n",
       "  'gene_name': 'LINF_010012100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'PLAC8_family',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010012100',\n",
       "  'transcript_id': 'LINF_01T0012100',\n",
       "  'transcript_name': 'LINF_01T0012100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'PLAC8_family',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012100',\n",
       "  'transcript_id': 'LINF_01T0012100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010012200',\n",
       "  'gene_name': 'LINF_010012200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'ubiquitin_activating_enzyme|UBA3',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010012200',\n",
       "  'transcript_id': 'LINF_01T0012200',\n",
       "  'transcript_name': 'LINF_01T0012200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'ubiquitin_activating_enzyme|UBA3',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012200',\n",
       "  'transcript_id': 'LINF_01T0012200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010012300',\n",
       "  'gene_name': 'LINF_010012300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010012300',\n",
       "  'transcript_id': 'LINF_01T0012300',\n",
       "  'transcript_name': 'LINF_01T0012300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012300',\n",
       "  'transcript_id': 'LINF_01T0012300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010012400',\n",
       "  'gene_name': 'LINF_010012400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Meckel_syndrome_type_6_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010012400',\n",
       "  'transcript_id': 'LINF_01T0012400',\n",
       "  'transcript_name': 'LINF_01T0012400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Meckel_syndrome_type_6_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012400',\n",
       "  'transcript_id': 'LINF_01T0012400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010012500',\n",
       "  'gene_name': 'LINF_010012500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010012500',\n",
       "  'transcript_id': 'LINF_01T0012500',\n",
       "  'transcript_name': 'LINF_01T0012500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012500',\n",
       "  'transcript_id': 'LINF_01T0012500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010012600',\n",
       "  'gene_name': 'LINF_010012600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'CMGC_family_protein_kinase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010012600',\n",
       "  'transcript_id': 'LINF_01T0012600',\n",
       "  'transcript_name': 'LINF_01T0012600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'CMGC_family_protein_kinase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012600',\n",
       "  'transcript_id': 'LINF_01T0012600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010012700',\n",
       "  'gene_name': 'LINF_010012700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010012700',\n",
       "  'transcript_id': 'LINF_01T0012700',\n",
       "  'transcript_name': 'LINF_01T0012700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012700',\n",
       "  'transcript_id': 'LINF_01T0012700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010012800',\n",
       "  'gene_name': 'LINF_010012800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Eukaryotic_initiation_factor_4A-1|eIF4A1',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010012800',\n",
       "  'transcript_id': 'LINF_01T0012800',\n",
       "  'transcript_name': 'LINF_01T0012800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Eukaryotic_initiation_factor_4A-1|eIF4A1',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012800',\n",
       "  'transcript_id': 'LINF_01T0012800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010012900',\n",
       "  'gene_name': 'LINF_010012900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Eukaryotic_initiation_factor_4A-1|eIF4A1',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010012900',\n",
       "  'transcript_id': 'LINF_01T0012900',\n",
       "  'transcript_name': 'LINF_01T0012900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Eukaryotic_initiation_factor_4A-1|eIF4A1',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012900',\n",
       "  'transcript_id': 'LINF_01T0012900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0012900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010013000',\n",
       "  'gene_name': 'LINF_010013000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010013000',\n",
       "  'transcript_id': 'LINF_01T0013000',\n",
       "  'transcript_name': 'LINF_01T0013000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0013000',\n",
       "  'transcript_id': 'LINF_01T0013000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0013000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0013000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010013050',\n",
       "  'gene_name': 'LINF_010013050',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010013050',\n",
       "  'transcript_id': 'LINF_01T0013050',\n",
       "  'transcript_name': 'LINF_01T0013050',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010013100',\n",
       "  'gene_name': 'LINF_010013100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Mitochondrial_processome_subunit_5|MPSS5',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010013100',\n",
       "  'transcript_id': 'LINF_01T0013100',\n",
       "  'transcript_name': 'LINF_01T0013100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Mitochondrial_processome_subunit_5|MPSS5',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0013100',\n",
       "  'transcript_id': 'LINF_01T0013100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0013100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0013100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010013200',\n",
       "  'gene_name': 'LINF_010013200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'calcium/potassium_channel_(CAKC)',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010013200',\n",
       "  'transcript_id': 'LINF_01T0013200',\n",
       "  'transcript_name': 'LINF_01T0013200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'calcium/potassium_channel_(CAKC)',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0013200',\n",
       "  'transcript_id': 'LINF_01T0013200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0013200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0013200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010013300',\n",
       "  'gene_name': 'LINF_010013300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'potassium_channel_subunit-like_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010013300',\n",
       "  'transcript_id': 'LINF_01T0013300',\n",
       "  'transcript_name': 'LINF_01T0013300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'potassium_channel_subunit-like_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0013300',\n",
       "  'transcript_id': 'LINF_01T0013300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0013300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0013300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010013350',\n",
       "  'gene_name': 'LINF_010013350',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010013350',\n",
       "  'transcript_id': 'LINF_01T0013350',\n",
       "  'transcript_name': 'LINF_01T0013350',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010013400',\n",
       "  'gene_name': 'LINF_010013400',\n",
       "  'pseudogen': 'unknown',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'peptidyl_dipeptidase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None},\n",
       " {'parent_id': 'LINF_010013400',\n",
       "  'transcript_id': 'LINF_01T0013400',\n",
       "  'transcript_name': 'LINF_01T0013400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'peptidyl_dipeptidase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0013400',\n",
       "  'transcript_id': 'LINF_01T0013400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0013400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_01T0013400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_010013450',\n",
       "  'gene_name': 'LINF_010013450',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_010013450',\n",
       "  'transcript_id': 'LINF_01T0013450',\n",
       "  'transcript_name': 'LINF_01T0013450',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020004900',\n",
       "  'gene_name': 'LINF_020004900',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020004900',\n",
       "  'transcript_id': 'LINF_02T0004900',\n",
       "  'transcript_name': 'LINF_02T0004900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020005000',\n",
       "  'gene_name': 'LINF_020005000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Side_chain_galactosyltransferase|SCG3',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020005000',\n",
       "  'transcript_id': 'LINF_02T0005000',\n",
       "  'transcript_name': 'LINF_02T0005000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Side_chain_galactosyltransferase|SCG3',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005000',\n",
       "  'transcript_id': 'LINF_02T0005000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020005050',\n",
       "  'gene_name': 'LINF_020005050',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020005050',\n",
       "  'transcript_id': 'LINF_02T0005050',\n",
       "  'transcript_name': 'LINF_02T0005050',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020005100',\n",
       "  'gene_name': 'LINF_020005100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'histone_H4',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020005100',\n",
       "  'transcript_id': 'LINF_02T0005100',\n",
       "  'transcript_name': 'LINF_02T0005100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'histone_H4',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005100',\n",
       "  'transcript_id': 'LINF_02T0005100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020005200',\n",
       "  'gene_name': 'LINF_020005200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'tagatose-6-phosphate_kinase-like_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020005200',\n",
       "  'transcript_id': 'LINF_02T0005200',\n",
       "  'transcript_name': 'LINF_02T0005200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'tagatose-6-phosphate_kinase-like_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005200',\n",
       "  'transcript_id': 'LINF_02T0005200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020005300',\n",
       "  'gene_name': 'LINF_020005300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'aminopeptidase_P1',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020005300',\n",
       "  'transcript_id': 'LINF_02T0005300',\n",
       "  'transcript_name': 'LINF_02T0005300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'aminopeptidase_P1',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005300',\n",
       "  'transcript_id': 'LINF_02T0005300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020005330',\n",
       "  'gene_name': 'LINF_020005330',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020005330',\n",
       "  'transcript_id': 'LINF_02T0005330',\n",
       "  'transcript_name': 'LINF_02T0005330',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020005360',\n",
       "  'gene_name': 'LINF_020005360',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020005360',\n",
       "  'transcript_id': 'LINF_02T0005360',\n",
       "  'transcript_name': 'LINF_02T0005360',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020005400',\n",
       "  'gene_name': 'LINF_020005400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'cytochrome_b-domain_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020005400',\n",
       "  'transcript_id': 'LINF_02T0005400',\n",
       "  'transcript_name': 'LINF_02T0005400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'cytochrome_b-domain_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005400',\n",
       "  'transcript_id': 'LINF_02T0005400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020005500',\n",
       "  'gene_name': 'LINF_020005500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020005500',\n",
       "  'transcript_id': 'LINF_02T0005500',\n",
       "  'transcript_name': 'LINF_02T0005500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005500',\n",
       "  'transcript_id': 'LINF_02T0005500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020005600',\n",
       "  'gene_name': 'LINF_020005600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020005600',\n",
       "  'transcript_id': 'LINF_02T0005600',\n",
       "  'transcript_name': 'LINF_02T0005600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005600',\n",
       "  'transcript_id': 'LINF_02T0005600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020005650',\n",
       "  'gene_name': 'LINF_020005650',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020005650',\n",
       "  'transcript_id': 'LINF_02T0005650',\n",
       "  'transcript_name': 'LINF_02T0005650',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020005700',\n",
       "  'gene_name': 'LINF_020005700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020005700',\n",
       "  'transcript_id': 'LINF_02T0005700',\n",
       "  'transcript_name': 'LINF_02T0005700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005700',\n",
       "  'transcript_id': 'LINF_02T0005700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020005800',\n",
       "  'gene_name': 'LINF_020005800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020005800',\n",
       "  'transcript_id': 'LINF_02T0005800',\n",
       "  'transcript_name': 'LINF_02T0005800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005800',\n",
       "  'transcript_id': 'LINF_02T0005800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020005900',\n",
       "  'gene_name': 'LINF_020005900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020005900',\n",
       "  'transcript_id': 'LINF_02T0005900',\n",
       "  'transcript_name': 'LINF_02T0005900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005900',\n",
       "  'transcript_id': 'LINF_02T0005900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0005900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020006000',\n",
       "  'gene_name': 'LINF_020006000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'zinc-finger_containing_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020006000',\n",
       "  'transcript_id': 'LINF_02T0006000',\n",
       "  'transcript_name': 'LINF_02T0006000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'zinc-finger_containing_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006000',\n",
       "  'transcript_id': 'LINF_02T0006000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020006100',\n",
       "  'gene_name': 'LINF_020006100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Exportin-T',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020006100',\n",
       "  'transcript_id': 'LINF_02T0006100',\n",
       "  'transcript_name': 'LINF_02T0006100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Exportin-T',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006100',\n",
       "  'transcript_id': 'LINF_02T0006100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020006200',\n",
       "  'gene_name': 'LINF_020006200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'ATM_Kinase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020006200',\n",
       "  'transcript_id': 'LINF_02T0006200',\n",
       "  'transcript_name': 'LINF_02T0006200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'ATM_Kinase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006200',\n",
       "  'transcript_id': 'LINF_02T0006200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020006300',\n",
       "  'gene_name': 'LINF_020006300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'SF3A2_domain-containing_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020006300',\n",
       "  'transcript_id': 'LINF_02T0006300',\n",
       "  'transcript_name': 'LINF_02T0006300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'SF3A2_domain-containing_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006300',\n",
       "  'transcript_id': 'LINF_02T0006300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020006400',\n",
       "  'gene_name': 'LINF_020006400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Ankyrin_repeat-containing_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020006400',\n",
       "  'transcript_id': 'LINF_02T0006400',\n",
       "  'transcript_name': 'LINF_02T0006400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Ankyrin_repeat-containing_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006400',\n",
       "  'transcript_id': 'LINF_02T0006400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020006500',\n",
       "  'gene_name': 'LINF_020006500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Conserved_oligomeric_Golgi_complex_subunit_1',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020006500',\n",
       "  'transcript_id': 'LINF_02T0006500',\n",
       "  'transcript_name': 'LINF_02T0006500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Conserved_oligomeric_Golgi_complex_subunit_1',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006500',\n",
       "  'transcript_id': 'LINF_02T0006500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020006700',\n",
       "  'gene_name': 'LINF_020006700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'phosphoglycan_beta_1-3_galactosyltransferase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020006700',\n",
       "  'transcript_id': 'LINF_02T0006700',\n",
       "  'transcript_name': 'LINF_02T0006700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'phosphoglycan_beta_1-3_galactosyltransferase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006700',\n",
       "  'transcript_id': 'LINF_02T0006700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020006750',\n",
       "  'gene_name': 'LINF_020006750',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020006750',\n",
       "  'transcript_id': 'LINF_02T0006750',\n",
       "  'transcript_name': 'LINF_02T0006750',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020006800',\n",
       "  'gene_name': 'LINF_020006800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'phosphoglycan_beta_1-3_galactosyltransferase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020006800',\n",
       "  'transcript_id': 'LINF_02T0006800',\n",
       "  'transcript_name': 'LINF_02T0006800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'phosphoglycan_beta_1-3_galactosyltransferase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006800',\n",
       "  'transcript_id': 'LINF_02T0006800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020006900',\n",
       "  'gene_name': 'LINF_020006900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'phosphoglycan_beta_1-3_galactosyltransferase|SCA2',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020006900',\n",
       "  'transcript_id': 'LINF_02T0006900',\n",
       "  'transcript_name': 'LINF_02T0006900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'phosphoglycan_beta_1-3_galactosyltransferase|SCA2',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006900',\n",
       "  'transcript_id': 'LINF_02T0006900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0006900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020007000',\n",
       "  'gene_name': 'LINF_020007000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'phosphoglycan_beta_1-3_galactosyltransferase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020007000',\n",
       "  'transcript_id': 'LINF_02T0007000',\n",
       "  'transcript_name': 'LINF_02T0007000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'phosphoglycan_beta_1-3_galactosyltransferase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007000',\n",
       "  'transcript_id': 'LINF_02T0007000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020007100',\n",
       "  'gene_name': 'LINF_020007100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'phosphoglycan_beta_1-3_galactosyltransferase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020007100',\n",
       "  'transcript_id': 'LINF_02T0007100',\n",
       "  'transcript_name': 'LINF_02T0007100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'phosphoglycan_beta_1-3_galactosyltransferase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007100',\n",
       "  'transcript_id': 'LINF_02T0007100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020007200',\n",
       "  'gene_name': 'LINF_020007200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'phosphoglycan_beta_1-2_arabinosyltransferase|SCA1',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020007200',\n",
       "  'transcript_id': 'LINF_02T0007200',\n",
       "  'transcript_name': 'LINF_02T0007200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'phosphoglycan_beta_1-2_arabinosyltransferase|SCA1',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007200',\n",
       "  'transcript_id': 'LINF_02T0007200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020007300',\n",
       "  'gene_name': 'LINF_020007300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'phosphoglycan_beta_1-3_galactosyltransferase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020007300',\n",
       "  'transcript_id': 'LINF_02T0007300',\n",
       "  'transcript_name': 'LINF_02T0007300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'phosphoglycan_beta_1-3_galactosyltransferase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007300',\n",
       "  'transcript_id': 'LINF_02T0007300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020007400',\n",
       "  'gene_name': 'LINF_020007400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Glycosyltransferase_(GlcNAc)',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020007400',\n",
       "  'transcript_id': 'LINF_02T0007400',\n",
       "  'transcript_name': 'LINF_02T0007400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Glycosyltransferase_(GlcNAc)',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007400',\n",
       "  'transcript_id': 'LINF_02T0007400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020007500',\n",
       "  'gene_name': 'LINF_020007500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Glycosyltransferase_(GlcNAc)',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020007500',\n",
       "  'transcript_id': 'LINF_02T0007500',\n",
       "  'transcript_name': 'LINF_02T0007500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Glycosyltransferase_(GlcNAc)',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007500',\n",
       "  'transcript_id': 'LINF_02T0007500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020007600',\n",
       "  'gene_name': 'LINF_020007600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Rab6_GTPase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020007600',\n",
       "  'transcript_id': 'LINF_02T0007600',\n",
       "  'transcript_name': 'LINF_02T0007600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Rab6_GTPase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007600',\n",
       "  'transcript_id': 'LINF_02T0007600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020007700',\n",
       "  'gene_name': 'LINF_020007700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020007700',\n",
       "  'transcript_id': 'LINF_02T0007700',\n",
       "  'transcript_name': 'LINF_02T0007700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007700',\n",
       "  'transcript_id': 'LINF_02T0007700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020007800',\n",
       "  'gene_name': 'LINF_020007800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020007800',\n",
       "  'transcript_id': 'LINF_02T0007800',\n",
       "  'transcript_name': 'LINF_02T0007800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007800',\n",
       "  'transcript_id': 'LINF_02T0007800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020007900',\n",
       "  'gene_name': 'LINF_020007900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Serine/threonine-protein_kinase_(NEK_family)',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020007900',\n",
       "  'transcript_id': 'LINF_02T0007900',\n",
       "  'transcript_name': 'LINF_02T0007900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Serine/threonine-protein_kinase_(NEK_family)',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007900',\n",
       "  'transcript_id': 'LINF_02T0007900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0007900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020007950',\n",
       "  'gene_name': 'LINF_020007950',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020007950',\n",
       "  'transcript_id': 'LINF_02T0007950',\n",
       "  'transcript_name': 'LINF_02T0007950',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020008000',\n",
       "  'gene_name': 'LINF_020008000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'ATP-binding_cassette_subfamily_A_-_member_1|ABCA1',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020008000',\n",
       "  'transcript_id': 'LINF_02T0008000',\n",
       "  'transcript_name': 'LINF_02T0008000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'ATP-binding_cassette_subfamily_A_-_member_1|ABCA1',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008000',\n",
       "  'transcript_id': 'LINF_02T0008000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020008100',\n",
       "  'gene_name': 'LINF_020008100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'paraflagellar_rod_component_11|PFC11',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020008100',\n",
       "  'transcript_id': 'LINF_02T0008100',\n",
       "  'transcript_name': 'LINF_02T0008100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'paraflagellar_rod_component_11|PFC11',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008100',\n",
       "  'transcript_id': 'LINF_02T0008100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020008200',\n",
       "  'gene_name': 'LINF_020008200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'RRP41B',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020008200',\n",
       "  'transcript_id': 'LINF_02T0008200',\n",
       "  'transcript_name': 'LINF_02T0008200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'RRP41B',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008200',\n",
       "  'transcript_id': 'LINF_02T0008200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020008300',\n",
       "  'gene_name': 'LINF_020008300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'fucosyltransferase|FUT2',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020008300',\n",
       "  'transcript_id': 'LINF_02T0008300',\n",
       "  'transcript_name': 'LINF_02T0008300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'fucosyltransferase|FUT2',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008300',\n",
       "  'transcript_id': 'LINF_02T0008300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020008400',\n",
       "  'gene_name': 'LINF_020008400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'UDP-N-acetylglucosamine_transporter',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020008400',\n",
       "  'transcript_id': 'LINF_02T0008400',\n",
       "  'transcript_name': 'LINF_02T0008400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'UDP-N-acetylglucosamine_transporter',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008400',\n",
       "  'transcript_id': 'LINF_02T0008400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020008500',\n",
       "  'gene_name': 'LINF_020008500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020008500',\n",
       "  'transcript_id': 'LINF_02T0008500',\n",
       "  'transcript_name': 'LINF_02T0008500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008500',\n",
       "  'transcript_id': 'LINF_02T0008500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020008600',\n",
       "  'gene_name': 'LINF_020008600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'casein_kinase_II_-_alpha_chain|CKA1',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020008600',\n",
       "  'transcript_id': 'LINF_02T0008600',\n",
       "  'transcript_name': 'LINF_02T0008600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'casein_kinase_II_-_alpha_chain|CKA1',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008600',\n",
       "  'transcript_id': 'LINF_02T0008600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020008700',\n",
       "  'gene_name': 'LINF_020008700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'proteasome_regulatory_non-ATPase_subunit_6',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020008700',\n",
       "  'transcript_id': 'LINF_02T0008700',\n",
       "  'transcript_name': 'LINF_02T0008700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'proteasome_regulatory_non-ATPase_subunit_6',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008700',\n",
       "  'transcript_id': 'LINF_02T0008700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020008800',\n",
       "  'gene_name': 'LINF_020008800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': \"Putative_tRNA_(cytidine(32)/guanosine(34)-2'-O)-methyltransferase\\xa0\",\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020008800',\n",
       "  'transcript_id': 'LINF_02T0008800',\n",
       "  'transcript_name': 'LINF_02T0008800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': \"Putative_tRNA_(cytidine(32)/guanosine(34)-2'-O)-methyltransferase\\xa0\",\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008800',\n",
       "  'transcript_id': 'LINF_02T0008800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020008900',\n",
       "  'gene_name': 'LINF_020008900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'ubiquitin-conjugating_enzyme_E2|UBC9',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020008900',\n",
       "  'transcript_id': 'LINF_02T0008900',\n",
       "  'transcript_name': 'LINF_02T0008900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'ubiquitin-conjugating_enzyme_E2|UBC9',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008900',\n",
       "  'transcript_id': 'LINF_02T0008900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0008900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020009100',\n",
       "  'gene_name': 'LINF_020009100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Mitochondrial_protein_81|KREPA1',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020009100',\n",
       "  'transcript_id': 'LINF_02T0009100',\n",
       "  'transcript_name': 'LINF_02T0009100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Mitochondrial_protein_81|KREPA1',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009100',\n",
       "  'transcript_id': 'LINF_02T0009100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020009200',\n",
       "  'gene_name': 'LINF_020009200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020009200',\n",
       "  'transcript_id': 'LINF_02T0009200',\n",
       "  'transcript_name': 'LINF_02T0009200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009200',\n",
       "  'transcript_id': 'LINF_02T0009200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020009250',\n",
       "  'gene_name': 'LINF_020009250',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020009250',\n",
       "  'transcript_id': 'LINF_02T0009250',\n",
       "  'transcript_name': 'LINF_02T0009250',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020009300',\n",
       "  'gene_name': 'LINF_020009300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'serine_peptidase_-_Clan_S_-_family_S54',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020009300',\n",
       "  'transcript_id': 'LINF_02T0009300',\n",
       "  'transcript_name': 'LINF_02T0009300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'serine_peptidase_-_Clan_S_-_family_S54',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009300',\n",
       "  'transcript_id': 'LINF_02T0009300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020009400',\n",
       "  'gene_name': 'LINF_020009400',\n",
       "  'pseudogen': 'unknown',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Transmembrane_amino_acid_transporter_protein_-_putative',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None},\n",
       " {'parent_id': 'LINF_020009400',\n",
       "  'transcript_id': 'LINF_02T0009400',\n",
       "  'transcript_name': 'LINF_02T0009400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Transmembrane_amino_acid_transporter_protein_-_putative',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009400',\n",
       "  'transcript_id': 'LINF_02T0009400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020009700',\n",
       "  'gene_name': 'LINF_020009700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'voltage-dependent_anion-selective_channel',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020009700',\n",
       "  'transcript_id': 'LINF_02T0009700',\n",
       "  'transcript_name': 'LINF_02T0009700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'voltage-dependent_anion-selective_channel',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009700',\n",
       "  'transcript_id': 'LINF_02T0009700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020009800',\n",
       "  'gene_name': 'LINF_020009800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'voltage-dependent_anion-selective_channel|VDAC',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020009800',\n",
       "  'transcript_id': 'LINF_02T0009800',\n",
       "  'transcript_name': 'LINF_02T0009800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'voltage-dependent_anion-selective_channel|VDAC',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009800',\n",
       "  'transcript_id': 'LINF_02T0009800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020009900',\n",
       "  'gene_name': 'LINF_020009900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020009900',\n",
       "  'transcript_id': 'LINF_02T0009900',\n",
       "  'transcript_name': 'LINF_02T0009900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009900',\n",
       "  'transcript_id': 'LINF_02T0009900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0009900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020010000',\n",
       "  'gene_name': 'LINF_020010000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020010000',\n",
       "  'transcript_id': 'LINF_02T0010000',\n",
       "  'transcript_name': 'LINF_02T0010000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010000',\n",
       "  'transcript_id': 'LINF_02T0010000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020010100',\n",
       "  'gene_name': 'LINF_020010100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020010100',\n",
       "  'transcript_id': 'LINF_02T0010100',\n",
       "  'transcript_name': 'LINF_02T0010100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010100',\n",
       "  'transcript_id': 'LINF_02T0010100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020010200',\n",
       "  'gene_name': 'LINF_020010200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'glycerate_kinase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020010200',\n",
       "  'transcript_id': 'LINF_02T0010200',\n",
       "  'transcript_name': 'LINF_02T0010200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'glycerate_kinase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010200',\n",
       "  'transcript_id': 'LINF_02T0010200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020010300',\n",
       "  'gene_name': 'LINF_020010300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020010300',\n",
       "  'transcript_id': 'LINF_02T0010300',\n",
       "  'transcript_name': 'LINF_02T0010300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010300',\n",
       "  'transcript_id': 'LINF_02T0010300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020010400',\n",
       "  'gene_name': 'LINF_020010400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020010400',\n",
       "  'transcript_id': 'LINF_02T0010400',\n",
       "  'transcript_name': 'LINF_02T0010400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010400',\n",
       "  'transcript_id': 'LINF_02T0010400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020010500',\n",
       "  'gene_name': 'LINF_020010500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'tRNA-val|Anticodon_gac',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020010500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020010550',\n",
       "  'gene_name': 'LINF_020010550',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020010550',\n",
       "  'transcript_id': 'LINF_02T0010550',\n",
       "  'transcript_name': 'LINF_02T0010550',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020010600',\n",
       "  'gene_name': 'LINF_020010600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020010600',\n",
       "  'transcript_id': 'LINF_02T0010600',\n",
       "  'transcript_name': 'LINF_02T0010600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010600',\n",
       "  'transcript_id': 'LINF_02T0010600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020010700',\n",
       "  'gene_name': 'LINF_020010700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020010700',\n",
       "  'transcript_id': 'LINF_02T0010700',\n",
       "  'transcript_name': 'LINF_02T0010700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010700',\n",
       "  'transcript_id': 'LINF_02T0010700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020010800',\n",
       "  'gene_name': 'LINF_020010800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020010800',\n",
       "  'transcript_id': 'LINF_02T0010800',\n",
       "  'transcript_name': 'LINF_02T0010800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010800',\n",
       "  'transcript_id': 'LINF_02T0010800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020010900',\n",
       "  'gene_name': 'LINF_020010900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020010900',\n",
       "  'transcript_id': 'LINF_02T0010900',\n",
       "  'transcript_name': 'LINF_02T0010900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010900',\n",
       "  'transcript_id': 'LINF_02T0010900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0010900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020011000',\n",
       "  'gene_name': 'LINF_020011000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_kinase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020011000',\n",
       "  'transcript_id': 'LINF_02T0011000',\n",
       "  'transcript_name': 'LINF_02T0011000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_kinase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011000',\n",
       "  'transcript_id': 'LINF_02T0011000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020011030',\n",
       "  'gene_name': 'LINF_020011030',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020011030',\n",
       "  'transcript_id': 'LINF_02T0011030',\n",
       "  'transcript_name': 'LINF_02T0011030',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0010',\n",
       "  'gene_name': 'LINF_02.SLRNA.0010',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0010',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0020',\n",
       "  'gene_name': 'LINF_02.SLRNA.0020',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0020',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0030',\n",
       "  'gene_name': 'LINF_02.SLRNA.0030',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0030',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0040',\n",
       "  'gene_name': 'LINF_02.SLRNA.0040',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0040',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0050',\n",
       "  'gene_name': 'LINF_02.SLRNA.0050',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0050',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0060',\n",
       "  'gene_name': 'LINF_02.SLRNA.0060',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader-like_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0060',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0070',\n",
       "  'gene_name': 'LINF_02.SLRNA.0070',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0070',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0080',\n",
       "  'gene_name': 'LINF_02.SLRNA.0080',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader-like_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0080',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0090',\n",
       "  'gene_name': 'LINF_02.SLRNA.0090',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0090',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0100',\n",
       "  'gene_name': 'LINF_02.SLRNA.0100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader-like_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0110',\n",
       "  'gene_name': 'LINF_02.SLRNA.0110',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0110',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0120',\n",
       "  'gene_name': 'LINF_02.SLRNA.0120',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0120',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0130',\n",
       "  'gene_name': 'LINF_02.SLRNA.0130',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader-like_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0130',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0140',\n",
       "  'gene_name': 'LINF_02.SLRNA.0140',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0140',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0150',\n",
       "  'gene_name': 'LINF_02.SLRNA.0150',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0150',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0160',\n",
       "  'gene_name': 'LINF_02.SLRNA.0160',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0160',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0170',\n",
       "  'gene_name': 'LINF_02.SLRNA.0170',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0170',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0180',\n",
       "  'gene_name': 'LINF_02.SLRNA.0180',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0180',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0190',\n",
       "  'gene_name': 'LINF_02.SLRNA.0190',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0190',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0200',\n",
       "  'gene_name': 'LINF_02.SLRNA.0200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0210',\n",
       "  'gene_name': 'LINF_02.SLRNA.0210',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0210',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0220',\n",
       "  'gene_name': 'LINF_02.SLRNA.0220',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0220',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0230',\n",
       "  'gene_name': 'LINF_02.SLRNA.0230',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0230',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0240',\n",
       "  'gene_name': 'LINF_02.SLRNA.0240',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0240',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0250',\n",
       "  'gene_name': 'LINF_02.SLRNA.0250',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0250',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0260',\n",
       "  'gene_name': 'LINF_02.SLRNA.0260',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0260',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0270',\n",
       "  'gene_name': 'LINF_02.SLRNA.0270',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0270',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0280',\n",
       "  'gene_name': 'LINF_02.SLRNA.0280',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0280',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0290',\n",
       "  'gene_name': 'LINF_02.SLRNA.0290',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0290',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0300',\n",
       "  'gene_name': 'LINF_02.SLRNA.0300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0310',\n",
       "  'gene_name': 'LINF_02.SLRNA.0310',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0310',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0320',\n",
       "  'gene_name': 'LINF_02.SLRNA.0320',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0320',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0330',\n",
       "  'gene_name': 'LINF_02.SLRNA.0330',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0330',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0340',\n",
       "  'gene_name': 'LINF_02.SLRNA.0340',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0340',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0350',\n",
       "  'gene_name': 'LINF_02.SLRNA.0350',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0350',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0360',\n",
       "  'gene_name': 'LINF_02.SLRNA.0360',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0360',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0370',\n",
       "  'gene_name': 'LINF_02.SLRNA.0370',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0370',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0380',\n",
       "  'gene_name': 'LINF_02.SLRNA.0380',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0380',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0390',\n",
       "  'gene_name': 'LINF_02.SLRNA.0390',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0390',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0400',\n",
       "  'gene_name': 'LINF_02.SLRNA.0400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0410',\n",
       "  'gene_name': 'LINF_02.SLRNA.0410',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0410',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0420',\n",
       "  'gene_name': 'LINF_02.SLRNA.0420',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0420',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0430',\n",
       "  'gene_name': 'LINF_02.SLRNA.0430',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0430',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0440',\n",
       "  'gene_name': 'LINF_02.SLRNA.0440',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0440',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0450',\n",
       "  'gene_name': 'LINF_02.SLRNA.0450',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0450',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0460',\n",
       "  'gene_name': 'LINF_02.SLRNA.0460',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0460',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0470',\n",
       "  'gene_name': 'LINF_02.SLRNA.0470',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader-like_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0470',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0480',\n",
       "  'gene_name': 'LINF_02.SLRNA.0480',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0480',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0490',\n",
       "  'gene_name': 'LINF_02.SLRNA.0490',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader-like_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0490',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0500',\n",
       "  'gene_name': 'LINF_02.SLRNA.0500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0510',\n",
       "  'gene_name': 'LINF_02.SLRNA.0510',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader-like_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0510',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0520',\n",
       "  'gene_name': 'LINF_02.SLRNA.0520',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader-like_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0520',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0530',\n",
       "  'gene_name': 'LINF_02.SLRNA.0530',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader-like_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0530',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0540',\n",
       "  'gene_name': 'LINF_02.SLRNA.0540',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0540',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020011060',\n",
       "  'gene_name': 'LINF_020011060',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020011060',\n",
       "  'transcript_id': 'LINF_02T0011060',\n",
       "  'transcript_name': 'LINF_02T0011060',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_02.SLRNA.0550',\n",
       "  'gene_name': 'LINF_02.SLRNA.0550',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'spliced_leader-like_RNA',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02.SLRNA.0550',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020011100',\n",
       "  'gene_name': 'LINF_020011100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020011100',\n",
       "  'transcript_id': 'LINF_02T0011100',\n",
       "  'transcript_name': 'LINF_02T0011100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011100',\n",
       "  'transcript_id': 'LINF_02T0011100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020011200',\n",
       "  'gene_name': 'LINF_020011200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'TPR_(tetratricopeptide)_domain-containing_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020011200',\n",
       "  'transcript_id': 'LINF_02T0011200',\n",
       "  'transcript_name': 'LINF_02T0011200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'TPR_(tetratricopeptide)_domain-containing_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011200',\n",
       "  'transcript_id': 'LINF_02T0011200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020011300',\n",
       "  'gene_name': 'LINF_020011300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Arp2/3_complex_subunit_4|ARPC4',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020011300',\n",
       "  'transcript_id': 'LINF_02T0011300',\n",
       "  'transcript_name': 'LINF_02T0011300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Arp2/3_complex_subunit_4|ARPC4',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011300',\n",
       "  'transcript_id': 'LINF_02T0011300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020011400',\n",
       "  'gene_name': 'LINF_020011400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020011400',\n",
       "  'transcript_id': 'LINF_02T0011400',\n",
       "  'transcript_name': 'LINF_02T0011400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011400',\n",
       "  'transcript_id': 'LINF_02T0011400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020011500',\n",
       "  'gene_name': 'LINF_020011500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Diphthine_methyltransferase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020011500',\n",
       "  'transcript_id': 'LINF_02T0011500',\n",
       "  'transcript_name': 'LINF_02T0011500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Diphthine_methyltransferase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011500',\n",
       "  'transcript_id': 'LINF_02T0011500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020011600',\n",
       "  'gene_name': 'LINF_020011600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Glutamate_semialdehyde_dehydrogenase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020011600',\n",
       "  'transcript_id': 'LINF_02T0011600',\n",
       "  'transcript_name': 'LINF_02T0011600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Glutamate_semialdehyde_dehydrogenase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011600',\n",
       "  'transcript_id': 'LINF_02T0011600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020011700',\n",
       "  'gene_name': 'LINF_020011700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'leucine-rich_repeat_(LRR)-containing_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020011700',\n",
       "  'transcript_id': 'LINF_02T0011700',\n",
       "  'transcript_name': 'LINF_02T0011700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'leucine-rich_repeat_(LRR)-containing_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011700',\n",
       "  'transcript_id': 'LINF_02T0011700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020011800',\n",
       "  'gene_name': 'LINF_020011800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020011800',\n",
       "  'transcript_id': 'LINF_02T0011800',\n",
       "  'transcript_name': 'LINF_02T0011800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011800',\n",
       "  'transcript_id': 'LINF_02T0011800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020011900',\n",
       "  'gene_name': 'LINF_020011900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'UTP2',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020011900',\n",
       "  'transcript_id': 'LINF_02T0011900',\n",
       "  'transcript_name': 'LINF_02T0011900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'UTP2',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011900',\n",
       "  'transcript_id': 'LINF_02T0011900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0011900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020012000',\n",
       "  'gene_name': 'LINF_020012000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'mitochondrial_carrier_protein|MCP13',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020012000',\n",
       "  'transcript_id': 'LINF_02T0012000',\n",
       "  'transcript_name': 'LINF_02T0012000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'mitochondrial_carrier_protein|MCP13',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012000',\n",
       "  'transcript_id': 'LINF_02T0012000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020012100',\n",
       "  'gene_name': 'LINF_020012100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'DNA-directed_RNA_polymerase_III_subunit_RPC3',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020012100',\n",
       "  'transcript_id': 'LINF_02T0012100',\n",
       "  'transcript_name': 'LINF_02T0012100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'DNA-directed_RNA_polymerase_III_subunit_RPC3',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012100',\n",
       "  'transcript_id': 'LINF_02T0012100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020012200',\n",
       "  'gene_name': 'LINF_020012200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020012200',\n",
       "  'transcript_id': 'LINF_02T0012200',\n",
       "  'transcript_name': 'LINF_02T0012200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012200',\n",
       "  'transcript_id': 'LINF_02T0012200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020012300',\n",
       "  'gene_name': 'LINF_020012300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020012300',\n",
       "  'transcript_id': 'LINF_02T0012300',\n",
       "  'transcript_name': 'LINF_02T0012300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'protein_of_unknown_function_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012300',\n",
       "  'transcript_id': 'LINF_02T0012300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020012400',\n",
       "  'gene_name': 'LINF_020012400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'ATP-dependent_Clp_protease_subunit|heat_shock_protein_78_(HSP78)',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020012400',\n",
       "  'transcript_id': 'LINF_02T0012400',\n",
       "  'transcript_name': 'LINF_02T0012400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'ATP-dependent_Clp_protease_subunit|heat_shock_protein_78_(HSP78)',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012400',\n",
       "  'transcript_id': 'LINF_02T0012400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020012430',\n",
       "  'gene_name': 'LINF_020012430',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Hypothetical_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020012430',\n",
       "  'transcript_id': 'LINF_02T0012430',\n",
       "  'transcript_name': 'LINF_02T0012430',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Hypothetical_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012430',\n",
       "  'transcript_id': 'LINF_02T0012430',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012430',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012430',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020012460',\n",
       "  'gene_name': 'LINF_020012460',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'NAD(P)-binding_domain-containing_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020012460',\n",
       "  'transcript_id': 'LINF_02T0012460',\n",
       "  'transcript_name': 'LINF_02T0012460',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'NAD(P)-binding_domain-containing_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012460',\n",
       "  'transcript_id': 'LINF_02T0012460',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012460',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012460',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020012480',\n",
       "  'gene_name': 'LINF_020012480',\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020012480',\n",
       "  'transcript_id': 'LINF_02T0012480',\n",
       "  'transcript_name': 'LINF_02T0012480',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_020012500',\n",
       "  'gene_name': 'LINF_020012500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'dipeptylcarboxypeptidase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_020012500',\n",
       "  'transcript_id': 'LINF_02T0012500',\n",
       "  'transcript_name': 'LINF_02T0012500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'dipeptylcarboxypeptidase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012500',\n",
       "  'transcript_id': 'LINF_02T0012500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_02T0012500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_030005000',\n",
       "  'gene_name': 'LINF_030005000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Leucine_rich_repeat_domain_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_030005000',\n",
       "  'transcript_id': 'LINF_03T0005000',\n",
       "  'transcript_name': 'LINF_03T0005000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Leucine_rich_repeat_domain_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005000',\n",
       "  'transcript_id': 'LINF_03T0005000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_030005100',\n",
       "  'gene_name': 'LINF_030005100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_030005100',\n",
       "  'transcript_id': 'LINF_03T0005100',\n",
       "  'transcript_name': 'LINF_03T0005100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005100',\n",
       "  'transcript_id': 'LINF_03T0005100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_030005200',\n",
       "  'gene_name': 'LINF_030005200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'D-3-phosphoglycerate_dehydrogenase-like_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_030005200',\n",
       "  'transcript_id': 'LINF_03T0005200',\n",
       "  'transcript_name': 'LINF_03T0005200',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'D-3-phosphoglycerate_dehydrogenase-like_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005200',\n",
       "  'transcript_id': 'LINF_03T0005200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005200',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_030005300',\n",
       "  'gene_name': 'LINF_030005300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': '2-aminoethylphosphonate:pyruvate_aminotransferase-like_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_030005300',\n",
       "  'transcript_id': 'LINF_03T0005300',\n",
       "  'transcript_name': 'LINF_03T0005300',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': '2-aminoethylphosphonate:pyruvate_aminotransferase-like_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005300',\n",
       "  'transcript_id': 'LINF_03T0005300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005300',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_030005400',\n",
       "  'gene_name': 'LINF_030005400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'tRNA(Phe)_(4-demethylwyosine(37)-C(7))_aminocarboxypropyltransferase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_030005400',\n",
       "  'transcript_id': 'LINF_03T0005400',\n",
       "  'transcript_name': 'LINF_03T0005400',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'tRNA(Phe)_(4-demethylwyosine(37)-C(7))_aminocarboxypropyltransferase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005400',\n",
       "  'transcript_id': 'LINF_03T0005400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005400',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_030005500',\n",
       "  'gene_name': 'LINF_030005500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_030005500',\n",
       "  'transcript_id': 'LINF_03T0005500',\n",
       "  'transcript_name': 'LINF_03T0005500',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005500',\n",
       "  'transcript_id': 'LINF_03T0005500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005500',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_030005600',\n",
       "  'gene_name': 'LINF_030005600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'glycerol-3-phosphate_acyl_transferase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_030005600',\n",
       "  'transcript_id': 'LINF_03T0005600',\n",
       "  'transcript_name': 'LINF_03T0005600',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'glycerol-3-phosphate_acyl_transferase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005600',\n",
       "  'transcript_id': 'LINF_03T0005600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005600',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_030005700',\n",
       "  'gene_name': 'LINF_030005700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'DNA_primase_large_subunit',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_030005700',\n",
       "  'transcript_id': 'LINF_03T0005700',\n",
       "  'transcript_name': 'LINF_03T0005700',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'DNA_primase_large_subunit',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005700',\n",
       "  'transcript_id': 'LINF_03T0005700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005700',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_030005800',\n",
       "  'gene_name': 'LINF_030005800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Cytochrome_c_oxidase_assembly_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_030005800',\n",
       "  'transcript_id': 'LINF_03T0005800',\n",
       "  'transcript_name': 'LINF_03T0005800',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Cytochrome_c_oxidase_assembly_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005800',\n",
       "  'transcript_id': 'LINF_03T0005800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005800',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_030005900',\n",
       "  'gene_name': 'LINF_030005900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_030005900',\n",
       "  'transcript_id': 'LINF_03T0005900',\n",
       "  'transcript_name': 'LINF_03T0005900',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'hypothetical_protein_-_conserved',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005900',\n",
       "  'transcript_id': 'LINF_03T0005900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0005900',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_030006000',\n",
       "  'gene_name': 'LINF_030006000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Leucine-rich_repeat_(LRR)_domain-containing_protein',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_030006000',\n",
       "  'transcript_id': 'LINF_03T0006000',\n",
       "  'transcript_name': 'LINF_03T0006000',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'Leucine-rich_repeat_(LRR)_domain-containing_protein',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0006000',\n",
       "  'transcript_id': 'LINF_03T0006000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0006000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0006000',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'gene_id': 'LINF_030006100',\n",
       "  'gene_name': 'LINF_030006100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'acetyltransferase',\n",
       "  'parent_id': None,\n",
       "  'transcript_id': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_030006100',\n",
       "  'transcript_id': 'LINF_03T0006100',\n",
       "  'transcript_name': 'LINF_03T0006100',\n",
       "  'biotype': 'protein_coding',\n",
       "  'notes': 'acetyltransferase',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'pseudogen': None},\n",
       " {'parent_id': 'LINF_03T0006100',\n",
       "  'transcript_id': 'LINF_03T0006100',\n",
       "  'gene_id': None,\n",
       "  'gene_name': None,\n",
       "  'biotype': None,\n",
       "  'notes': None,\n",
       "  'transcript_name': None,\n",
       "  'pseudogen': None},\n",
       " ...]"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 87
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:40.910612Z",
     "start_time": "2024-10-03T08:01:40.852835Z"
    }
   },
   "source": [
    "# Transforming the list of dicts into a DataFrame\n",
    "new_col_df = pd.DataFrame(new_col_df)\n",
    "new_col_df"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "              gene_id       gene_name         biotype  \\\n",
       "0      LINF_010005000  LINF_010005000  protein_coding   \n",
       "1                None            None  protein_coding   \n",
       "2                None            None            None   \n",
       "3                None            None            None   \n",
       "4                None            None            None   \n",
       "...               ...             ...             ...   \n",
       "45363            None            None            None   \n",
       "45364            None            None            None   \n",
       "45365            None            None            None   \n",
       "45366  LINF_360082500  LINF_360082500            None   \n",
       "45367            None            None            None   \n",
       "\n",
       "                                       notes        parent_id  \\\n",
       "0      Protein_of_unknown_function_(DUF2946)             None   \n",
       "1      Protein_of_unknown_function_(DUF2946)   LINF_010005000   \n",
       "2                                       None  LINF_01T0005000   \n",
       "3                                       None  LINF_01T0005000   \n",
       "4                                       None  LINF_01T0005000   \n",
       "...                                      ...              ...   \n",
       "45363                                   None  LINF_36T0082400   \n",
       "45364                                   None  LINF_36T0082400   \n",
       "45365                                   None  LINF_36T0082400   \n",
       "45366                                   None             None   \n",
       "45367                                   None   LINF_360082500   \n",
       "\n",
       "         transcript_id  transcript_name pseudogen  \n",
       "0                 None             None      None  \n",
       "1      LINF_01T0005000  LINF_01T0005000      None  \n",
       "2      LINF_01T0005000             None      None  \n",
       "3                 None             None      None  \n",
       "4                 None             None      None  \n",
       "...                ...              ...       ...  \n",
       "45363  LINF_36T0082400             None      None  \n",
       "45364             None             None      None  \n",
       "45365             None             None      None  \n",
       "45366             None             None      None  \n",
       "45367  LINF_36T0082500  LINF_36T0082500      None  \n",
       "\n",
       "[45368 rows x 8 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>biotype</th>\n",
       "      <th>notes</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>pseudogen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>Protein_of_unknown_function_(DUF2946)</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>Protein_of_unknown_function_(DUF2946)</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45363</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0082400</td>\n",
       "      <td>LINF_36T0082400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45364</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0082400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45365</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0082400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45366</th>\n",
       "      <td>LINF_360082500</td>\n",
       "      <td>LINF_360082500</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45367</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_360082500</td>\n",
       "      <td>LINF_36T0082500</td>\n",
       "      <td>LINF_36T0082500</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>45368 rows × 8 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 88
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:41.188708Z",
     "start_time": "2024-10-03T08:01:41.162794Z"
    }
   },
   "source": [
    "# Le's re-order the columns\n",
    "new_col_df = new_col_df[[\"gene_id\", \"gene_name\", \"transcript_id\", \"transcript_name\", \"biotype\", \"parent_id\", \"pseudogen\", \"notes\"]]\n",
    "new_col_df"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "              gene_id       gene_name    transcript_id  transcript_name  \\\n",
       "0      LINF_010005000  LINF_010005000             None             None   \n",
       "1                None            None  LINF_01T0005000  LINF_01T0005000   \n",
       "2                None            None  LINF_01T0005000             None   \n",
       "3                None            None             None             None   \n",
       "4                None            None             None             None   \n",
       "...               ...             ...              ...              ...   \n",
       "45363            None            None  LINF_36T0082400             None   \n",
       "45364            None            None             None             None   \n",
       "45365            None            None             None             None   \n",
       "45366  LINF_360082500  LINF_360082500             None             None   \n",
       "45367            None            None  LINF_36T0082500  LINF_36T0082500   \n",
       "\n",
       "              biotype        parent_id pseudogen  \\\n",
       "0      protein_coding             None      None   \n",
       "1      protein_coding   LINF_010005000      None   \n",
       "2                None  LINF_01T0005000      None   \n",
       "3                None  LINF_01T0005000      None   \n",
       "4                None  LINF_01T0005000      None   \n",
       "...               ...              ...       ...   \n",
       "45363            None  LINF_36T0082400      None   \n",
       "45364            None  LINF_36T0082400      None   \n",
       "45365            None  LINF_36T0082400      None   \n",
       "45366            None             None      None   \n",
       "45367            None   LINF_360082500      None   \n",
       "\n",
       "                                       notes  \n",
       "0      Protein_of_unknown_function_(DUF2946)  \n",
       "1      Protein_of_unknown_function_(DUF2946)  \n",
       "2                                       None  \n",
       "3                                       None  \n",
       "4                                       None  \n",
       "...                                      ...  \n",
       "45363                                   None  \n",
       "45364                                   None  \n",
       "45365                                   None  \n",
       "45366                                   None  \n",
       "45367                                   None  \n",
       "\n",
       "[45368 rows x 8 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>biotype</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>pseudogen</th>\n",
       "      <th>notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF2946)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF2946)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45363</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0082400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0082400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45364</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0082400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45365</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0082400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45366</th>\n",
       "      <td>LINF_360082500</td>\n",
       "      <td>LINF_360082500</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45367</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0082500</td>\n",
       "      <td>LINF_36T0082500</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_360082500</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>45368 rows × 8 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 89
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:41.684395Z",
     "start_time": "2024-10-03T08:01:41.653005Z"
    }
   },
   "source": [
    "# Concatenating the new DataFrame with the original `gtf_df` and dropping the `attributes` column\n",
    "gtf_df = pd.concat([gtf_df, new_col_df], axis=1)\n",
    "gtf_df.drop(columns=\"attributes\", inplace=True)\n",
    "gtf_df"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "         chrom     feature    start      end strand         gene_id  \\\n",
       "0      LinJ.01        gene     1520     5066      -  LINF_010005000   \n",
       "1      LinJ.01  transcript     1520     5066      -            None   \n",
       "2      LinJ.01         CDS     3710     4711      -            None   \n",
       "3      LinJ.01        3utr     1520     3709      -            None   \n",
       "4      LinJ.01        5utr     4712     5066      -            None   \n",
       "...        ...         ...      ...      ...    ...             ...   \n",
       "45363  LinJ.36         CDS  2739458  2740183      -            None   \n",
       "45364  LinJ.36        3utr  2738595  2739457      -            None   \n",
       "45365  LinJ.36        5utr  2740184  2740374      -            None   \n",
       "45366  LinJ.36        gene  2740760  2742268      -  LINF_360082500   \n",
       "45367  LinJ.36  transcript  2740760  2742268      -            None   \n",
       "\n",
       "            gene_name    transcript_id  transcript_name         biotype  \\\n",
       "0      LINF_010005000             None             None  protein_coding   \n",
       "1                None  LINF_01T0005000  LINF_01T0005000  protein_coding   \n",
       "2                None  LINF_01T0005000             None            None   \n",
       "3                None             None             None            None   \n",
       "4                None             None             None            None   \n",
       "...               ...              ...              ...             ...   \n",
       "45363            None  LINF_36T0082400             None            None   \n",
       "45364            None             None             None            None   \n",
       "45365            None             None             None            None   \n",
       "45366  LINF_360082500             None             None            None   \n",
       "45367            None  LINF_36T0082500  LINF_36T0082500            None   \n",
       "\n",
       "             parent_id pseudogen                                  notes  \n",
       "0                 None      None  Protein_of_unknown_function_(DUF2946)  \n",
       "1       LINF_010005000      None  Protein_of_unknown_function_(DUF2946)  \n",
       "2      LINF_01T0005000      None                                   None  \n",
       "3      LINF_01T0005000      None                                   None  \n",
       "4      LINF_01T0005000      None                                   None  \n",
       "...                ...       ...                                    ...  \n",
       "45363  LINF_36T0082400      None                                   None  \n",
       "45364  LINF_36T0082400      None                                   None  \n",
       "45365  LINF_36T0082400      None                                   None  \n",
       "45366             None      None                                   None  \n",
       "45367   LINF_360082500      None                                   None  \n",
       "\n",
       "[45368 rows x 13 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>feature</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>biotype</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>pseudogen</th>\n",
       "      <th>notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>gene</td>\n",
       "      <td>1520</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF2946)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>transcript</td>\n",
       "      <td>1520</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF2946)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3710</td>\n",
       "      <td>4711</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>3utr</td>\n",
       "      <td>1520</td>\n",
       "      <td>3709</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>5utr</td>\n",
       "      <td>4712</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45363</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2739458</td>\n",
       "      <td>2740183</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0082400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0082400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45364</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>3utr</td>\n",
       "      <td>2738595</td>\n",
       "      <td>2739457</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0082400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45365</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>5utr</td>\n",
       "      <td>2740184</td>\n",
       "      <td>2740374</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0082400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45366</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>gene</td>\n",
       "      <td>2740760</td>\n",
       "      <td>2742268</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_360082500</td>\n",
       "      <td>LINF_360082500</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45367</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>transcript</td>\n",
       "      <td>2740760</td>\n",
       "      <td>2742268</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0082500</td>\n",
       "      <td>LINF_36T0082500</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_360082500</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>45368 rows × 13 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 90
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Compare coordinates"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this next part we are going to check for coordinates. To search which elements in the **neg_df** is inside which element in the **gtf_df**."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fail proof the data"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:43.354152Z",
     "start_time": "2024-10-03T08:01:43.317486Z"
    }
   },
   "source": [
    "# Copy data to make some fail-proof analysis\n",
    "neg_df_test = neg_df.copy()\n",
    "gtf_df_test = gtf_df.copy()\n",
    "\n",
    "# Lets shape shapes\n",
    "print(f\"Shape of neg_df: {neg_df.shape}\")\n",
    "print(f\"Shape of gtf_df: {gtf_df.shape}\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of neg_df: (682, 6)\n",
      "Shape of gtf_df: (45368, 13)\n"
     ]
    }
   ],
   "execution_count": 91
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:44.856448Z",
     "start_time": "2024-10-03T08:01:44.838437Z"
    }
   },
   "source": [
    "# Let's start with the shape in gtf_df_test:\n",
    "gtf_df.head()"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "     chrom     feature  start   end strand         gene_id       gene_name  \\\n",
       "0  LinJ.01        gene   1520  5066      -  LINF_010005000  LINF_010005000   \n",
       "1  LinJ.01  transcript   1520  5066      -            None            None   \n",
       "2  LinJ.01         CDS   3710  4711      -            None            None   \n",
       "3  LinJ.01        3utr   1520  3709      -            None            None   \n",
       "4  LinJ.01        5utr   4712  5066      -            None            None   \n",
       "\n",
       "     transcript_id  transcript_name         biotype        parent_id  \\\n",
       "0             None             None  protein_coding             None   \n",
       "1  LINF_01T0005000  LINF_01T0005000  protein_coding   LINF_010005000   \n",
       "2  LINF_01T0005000             None            None  LINF_01T0005000   \n",
       "3             None             None            None  LINF_01T0005000   \n",
       "4             None             None            None  LINF_01T0005000   \n",
       "\n",
       "  pseudogen                                  notes  \n",
       "0      None  Protein_of_unknown_function_(DUF2946)  \n",
       "1      None  Protein_of_unknown_function_(DUF2946)  \n",
       "2      None                                   None  \n",
       "3      None                                   None  \n",
       "4      None                                   None  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>feature</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>biotype</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>pseudogen</th>\n",
       "      <th>notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>gene</td>\n",
       "      <td>1520</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF2946)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>transcript</td>\n",
       "      <td>1520</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF2946)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3710</td>\n",
       "      <td>4711</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>3utr</td>\n",
       "      <td>1520</td>\n",
       "      <td>3709</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>5utr</td>\n",
       "      <td>4712</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 92
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:45.989793Z",
     "start_time": "2024-10-03T08:01:45.980687Z"
    }
   },
   "source": [
    "# Check elements where start < end\n",
    "num_elements_start_less_end = (gtf_df_test['start'] < gtf_df_test['end']).sum()  # type: ignore\n",
    "print(f\"There are {num_elements_start_less_end} elements where start < end.\")\n",
    "\n",
    "# Check elements where start > end\n",
    "num_elements_start_greater_end = (gtf_df_test['start'] > gtf_df_test['end']).sum()  # type: ignore\n",
    "print(f\"There are {num_elements_start_greater_end} elements where start > end.\")\n",
    "\n",
    "# Check elements where start == end\n",
    "num_elements_start_equal_end = (gtf_df_test['start'] == gtf_df_test['end']).sum()  # type: ignore\n",
    "print(f\"There are {num_elements_start_equal_end} elements where start == end.\")\n"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 45366 elements where start < end.\n",
      "There are 0 elements where start > end.\n",
      "There are 2 elements where start == end.\n"
     ]
    }
   ],
   "execution_count": 93
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:47.086585Z",
     "start_time": "2024-10-03T08:01:47.071948Z"
    }
   },
   "source": [
    "# Let's check the rows where start == end\n",
    "gtf_df_test[gtf_df_test['start'] == gtf_df_test['end']]\n"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "         chrom feature    start      end strand gene_id gene_name  \\\n",
       "495    LinJ.02    5utr    28840    28840      -    None      None   \n",
       "33221  LinJ.32    5utr  1041948  1041948      -    None      None   \n",
       "\n",
       "      transcript_id transcript_name biotype        parent_id pseudogen notes  \n",
       "495            None            None    None  LINF_02T0005800      None  None  \n",
       "33221          None            None    None  LINF_32T0033400      None  None  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>feature</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>biotype</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>pseudogen</th>\n",
       "      <th>notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>LinJ.02</td>\n",
       "      <td>5utr</td>\n",
       "      <td>28840</td>\n",
       "      <td>28840</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_02T0005800</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33221</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>5utr</td>\n",
       "      <td>1041948</td>\n",
       "      <td>1041948</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_32T0033400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 94
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:47.949693Z",
     "start_time": "2024-10-03T08:01:47.918532Z"
    }
   },
   "source": [
    "# Let's check the elements where transcript_id, transcript_name, parent_id can be \"LINF_02T0005800\" and \"gene_id\" the same but without \"T\"\n",
    "gtf_df_test[(gtf_df_test['transcript_id'] == \"LINF_02T0005800\") | \n",
    "            (gtf_df_test['transcript_name'] == \"LINF_02T0005800\") | \n",
    "            (gtf_df_test['parent_id'] == \"LINF_02T0005800\") |\n",
    "            (gtf_df_test['gene_id'] == \"LINF_020005800\")]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "       chrom     feature  start    end strand         gene_id       gene_name  \\\n",
       "491  LinJ.02        gene  27302  28840      -  LINF_020005800  LINF_020005800   \n",
       "492  LinJ.02  transcript  27302  28840      -            None            None   \n",
       "493  LinJ.02         CDS  27895  28839      -            None            None   \n",
       "494  LinJ.02        3utr  27302  27894      -            None            None   \n",
       "495  LinJ.02        5utr  28840  28840      -            None            None   \n",
       "\n",
       "       transcript_id  transcript_name         biotype        parent_id  \\\n",
       "491             None             None  protein_coding             None   \n",
       "492  LINF_02T0005800  LINF_02T0005800  protein_coding   LINF_020005800   \n",
       "493  LINF_02T0005800             None            None  LINF_02T0005800   \n",
       "494             None             None            None  LINF_02T0005800   \n",
       "495             None             None            None  LINF_02T0005800   \n",
       "\n",
       "    pseudogen                             notes  \n",
       "491      None  hypothetical_protein_-_conserved  \n",
       "492      None  hypothetical_protein_-_conserved  \n",
       "493      None                              None  \n",
       "494      None                              None  \n",
       "495      None                              None  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>feature</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>biotype</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>pseudogen</th>\n",
       "      <th>notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>491</th>\n",
       "      <td>LinJ.02</td>\n",
       "      <td>gene</td>\n",
       "      <td>27302</td>\n",
       "      <td>28840</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_020005800</td>\n",
       "      <td>LINF_020005800</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>492</th>\n",
       "      <td>LinJ.02</td>\n",
       "      <td>transcript</td>\n",
       "      <td>27302</td>\n",
       "      <td>28840</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_02T0005800</td>\n",
       "      <td>LINF_02T0005800</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>LINF_020005800</td>\n",
       "      <td>None</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493</th>\n",
       "      <td>LinJ.02</td>\n",
       "      <td>CDS</td>\n",
       "      <td>27895</td>\n",
       "      <td>28839</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_02T0005800</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_02T0005800</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>LinJ.02</td>\n",
       "      <td>3utr</td>\n",
       "      <td>27302</td>\n",
       "      <td>27894</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_02T0005800</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>LinJ.02</td>\n",
       "      <td>5utr</td>\n",
       "      <td>28840</td>\n",
       "      <td>28840</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_02T0005800</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 95
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can see that the CDS goes all the way till the final of the transcript except for one base. This base is will be taken by the 5'UTR."
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:48.735433Z",
     "start_time": "2024-10-03T08:01:48.716950Z"
    }
   },
   "source": [
    "# Check the \"feature\" elements:\n",
    "condition =(gtf_df_test['start'] < gtf_df_test['end'])\n",
    "gtf_df_test[condition]['feature'].value_counts()"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "feature\n",
       "gene          9861\n",
       "transcript    9660\n",
       "CDS           8744\n",
       "3utr          8554\n",
       "5utr          8547\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 96
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Interesting, there should be the same numbers of 5utr as 3utr"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:49.803939Z",
     "start_time": "2024-10-03T08:01:49.792645Z"
    }
   },
   "source": [
    "# Checking without condition\n",
    "gtf_df_test['feature'].value_counts()"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "feature\n",
       "gene          9861\n",
       "transcript    9660\n",
       "CDS           8744\n",
       "3utr          8554\n",
       "5utr          8549\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 97
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:51.178291Z",
     "start_time": "2024-10-03T08:01:50.807694Z"
    }
   },
   "source": [
    "# Checking where those elements where there is not a 3utr or 5utr\n",
    "parent_feature_dict = gtf_df_test.groupby('parent_id')['feature'].apply(list).to_dict()\n",
    "filtered_dict = {k: v for k, v in parent_feature_dict.items() if v not in (['transcript'], \n",
    "                                                                           ['CDS'], \n",
    "                                                                           ['CDS', '5utr', '3utr'], \n",
    "                                                                           ['CDS', '3utr', '5utr'],\n",
    "                                                                           ['CDS', '3utr', '5utr', 'CDS', '3utr', '5utr'],\n",
    "                                                                           ['transcript', 'transcript'],\n",
    "                                                                           ['CDS', '5utr', '3utr', 'CDS', '5utr', '3utr'])}\n",
    "filtered_dict"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'LINF_27T0013600': ['CDS', '5utr'],\n",
       " 'LINF_30T0006850': ['CDS', '3utr'],\n",
       " 'LINF_31T0037100': ['CDS', '3utr'],\n",
       " 'LINF_31T0039200': ['CDS', '3utr'],\n",
       " 'LINF_36T0017400': ['CDS', '3utr'],\n",
       " 'LINF_36T0036000': ['CDS', '3utr'],\n",
       " 'LINF_36T0071100': ['CDS', '3utr']}"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 98
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:red\">These are the elements without a 3utr or 5utr</span>\n",
    "\n",
    "Should be careful with LINF_270013600\n"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:52.235764Z",
     "start_time": "2024-10-03T08:01:52.196777Z"
    }
   },
   "source": [
    "gtf_df[\n",
    "    (\n",
    "            (gtf_df[['gene_id', 'transcript_id', 'parent_id']].isin(filtered_dict.keys()).any(axis=1)) | \n",
    "            (gtf_df['gene_id'].isin([elem.replace(\"T\",\"\") for elem in list(filtered_dict.keys())]))\n",
    "     )\n",
    "]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "         chrom     feature    start      end strand         gene_id  \\\n",
       "23181  LinJ.27        gene   327990   328645      +  LINF_270013600   \n",
       "23182  LinJ.27  transcript   327990   328645      +            None   \n",
       "23183  LinJ.27         CDS   328114   328645      +            None   \n",
       "23184  LinJ.27        5utr   327990   328113      +            None   \n",
       "27827  LinJ.30        gene    56144    57262      -  LINF_300006850   \n",
       "27828  LinJ.30  transcript    56144    57262      -            None   \n",
       "27829  LinJ.30         CDS    56787    57262      -            None   \n",
       "27830  LinJ.30        3utr    56144    56786      -            None   \n",
       "31523  LinJ.31        gene  1404369  1405546      -  LINF_310037100   \n",
       "31524  LinJ.31  transcript  1404369  1405546      -            None   \n",
       "31525  LinJ.31         CDS  1404572  1405546      -            None   \n",
       "31526  LinJ.31        3utr  1404369  1404571      -            None   \n",
       "31629  LinJ.31        gene  1473743  1475202      -  LINF_310039200   \n",
       "31630  LinJ.31  transcript  1473743  1475202      -            None   \n",
       "31631  LinJ.31         CDS  1473788  1475202      -            None   \n",
       "31632  LinJ.31        3utr  1473743  1473787      -            None   \n",
       "42001  LinJ.36        gene   447893   450114      +  LINF_360017400   \n",
       "42002  LinJ.36  transcript   447893   450114      +            None   \n",
       "42003  LinJ.36         CDS   447893   449005      +            None   \n",
       "42004  LinJ.36        3utr   449006   450114      +            None   \n",
       "42963  LinJ.36        gene  1180267  1182741      -  LINF_360036000   \n",
       "42964  LinJ.36  transcript  1180267  1182741      -            None   \n",
       "42965  LinJ.36         CDS  1180804  1182741      -            None   \n",
       "42966  LinJ.36        3utr  1180267  1180803      -            None   \n",
       "44784  LinJ.36        gene  2392103  2394184      -  LINF_360071100   \n",
       "44785  LinJ.36  transcript  2392103  2394184      -            None   \n",
       "44786  LinJ.36         CDS  2393831  2394184      -            None   \n",
       "44787  LinJ.36        3utr  2392103  2393830      -            None   \n",
       "\n",
       "            gene_name    transcript_id  transcript_name         biotype  \\\n",
       "23181  LINF_270013600             None             None  protein_coding   \n",
       "23182            None  LINF_27T0013600  LINF_27T0013600  protein_coding   \n",
       "23183            None  LINF_27T0013600             None            None   \n",
       "23184            None             None             None            None   \n",
       "27827  LINF_300006850             None             None  protein_coding   \n",
       "27828            None  LINF_30T0006850  LINF_30T0006850  protein_coding   \n",
       "27829            None  LINF_30T0006850             None            None   \n",
       "27830            None             None             None            None   \n",
       "31523  LINF_310037100             None             None  protein_coding   \n",
       "31524            None  LINF_31T0037100  LINF_31T0037100  protein_coding   \n",
       "31525            None  LINF_31T0037100             None            None   \n",
       "31526            None             None             None            None   \n",
       "31629  LINF_310039200             None             None  protein_coding   \n",
       "31630            None  LINF_31T0039200  LINF_31T0039200  protein_coding   \n",
       "31631            None  LINF_31T0039200             None            None   \n",
       "31632            None             None             None            None   \n",
       "42001  LINF_360017400             None             None  protein_coding   \n",
       "42002            None  LINF_36T0017400  LINF_36T0017400  protein_coding   \n",
       "42003            None  LINF_36T0017400             None            None   \n",
       "42004            None             None             None            None   \n",
       "42963  LINF_360036000             None             None  protein_coding   \n",
       "42964            None  LINF_36T0036000  LINF_36T0036000  protein_coding   \n",
       "42965            None  LINF_36T0036000             None            None   \n",
       "42966            None             None             None            None   \n",
       "44784  LINF_360071100             None             None  protein_coding   \n",
       "44785            None  LINF_36T0071100  LINF_36T0071100  protein_coding   \n",
       "44786            None  LINF_36T0071100             None            None   \n",
       "44787            None             None             None            None   \n",
       "\n",
       "             parent_id pseudogen  \\\n",
       "23181             None   unknown   \n",
       "23182   LINF_270013600      None   \n",
       "23183  LINF_27T0013600      None   \n",
       "23184  LINF_27T0013600      None   \n",
       "27827             None   unknown   \n",
       "27828   LINF_300006850      None   \n",
       "27829  LINF_30T0006850      None   \n",
       "27830  LINF_30T0006850      None   \n",
       "31523             None      None   \n",
       "31524   LINF_310037100      None   \n",
       "31525  LINF_31T0037100      None   \n",
       "31526  LINF_31T0037100      None   \n",
       "31629             None   unknown   \n",
       "31630   LINF_310039200      None   \n",
       "31631  LINF_31T0039200      None   \n",
       "31632  LINF_31T0039200      None   \n",
       "42001             None      None   \n",
       "42002   LINF_360017400      None   \n",
       "42003  LINF_36T0017400      None   \n",
       "42004  LINF_36T0017400      None   \n",
       "42963             None      None   \n",
       "42964   LINF_360036000      None   \n",
       "42965  LINF_36T0036000      None   \n",
       "42966  LINF_36T0036000      None   \n",
       "44784             None      None   \n",
       "44785   LINF_360071100      None   \n",
       "44786  LINF_36T0071100      None   \n",
       "44787  LINF_36T0071100      None   \n",
       "\n",
       "                                                   notes  \n",
       "23181  Stress_responsive_A/B_Barrel_domain-containing...  \n",
       "23182  Stress_responsive_A/B_Barrel_domain-containing...  \n",
       "23183                                               None  \n",
       "23184                                               None  \n",
       "27827  polynucleotide_kinase_3'-phosphatase-_putative...  \n",
       "27828  polynucleotide_kinase_3'-phosphatase-_putative...  \n",
       "27829                                               None  \n",
       "27830                                               None  \n",
       "31523            protein_of_unknown_function_-_conserved  \n",
       "31524            protein_of_unknown_function_-_conserved  \n",
       "31525                                               None  \n",
       "31526                                               None  \n",
       "31629                               hypothetical_protein  \n",
       "31630                               hypothetical_protein  \n",
       "31631                                               None  \n",
       "31632                                               None  \n",
       "42001                                  Memo-like_protein  \n",
       "42002                                  Memo-like_protein  \n",
       "42003                                               None  \n",
       "42004                                               None  \n",
       "42963                   hypothetical_protein_-_conserved  \n",
       "42964                   hypothetical_protein_-_conserved  \n",
       "42965                                               None  \n",
       "42966                                               None  \n",
       "44784   Protein_of_unknown_function_(DUF2763)_-_putative  \n",
       "44785   Protein_of_unknown_function_(DUF2763)_-_putative  \n",
       "44786                                               None  \n",
       "44787                                               None  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>feature</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>biotype</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>pseudogen</th>\n",
       "      <th>notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>23181</th>\n",
       "      <td>LinJ.27</td>\n",
       "      <td>gene</td>\n",
       "      <td>327990</td>\n",
       "      <td>328645</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_270013600</td>\n",
       "      <td>LINF_270013600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>None</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Stress_responsive_A/B_Barrel_domain-containing...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23182</th>\n",
       "      <td>LinJ.27</td>\n",
       "      <td>transcript</td>\n",
       "      <td>327990</td>\n",
       "      <td>328645</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_27T0013600</td>\n",
       "      <td>LINF_27T0013600</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>LINF_270013600</td>\n",
       "      <td>None</td>\n",
       "      <td>Stress_responsive_A/B_Barrel_domain-containing...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23183</th>\n",
       "      <td>LinJ.27</td>\n",
       "      <td>CDS</td>\n",
       "      <td>328114</td>\n",
       "      <td>328645</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_27T0013600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_27T0013600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23184</th>\n",
       "      <td>LinJ.27</td>\n",
       "      <td>5utr</td>\n",
       "      <td>327990</td>\n",
       "      <td>328113</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_27T0013600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27827</th>\n",
       "      <td>LinJ.30</td>\n",
       "      <td>gene</td>\n",
       "      <td>56144</td>\n",
       "      <td>57262</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_300006850</td>\n",
       "      <td>LINF_300006850</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>None</td>\n",
       "      <td>unknown</td>\n",
       "      <td>polynucleotide_kinase_3'-phosphatase-_putative...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27828</th>\n",
       "      <td>LinJ.30</td>\n",
       "      <td>transcript</td>\n",
       "      <td>56144</td>\n",
       "      <td>57262</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_30T0006850</td>\n",
       "      <td>LINF_30T0006850</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>LINF_300006850</td>\n",
       "      <td>None</td>\n",
       "      <td>polynucleotide_kinase_3'-phosphatase-_putative...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27829</th>\n",
       "      <td>LinJ.30</td>\n",
       "      <td>CDS</td>\n",
       "      <td>56787</td>\n",
       "      <td>57262</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_30T0006850</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_30T0006850</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27830</th>\n",
       "      <td>LinJ.30</td>\n",
       "      <td>3utr</td>\n",
       "      <td>56144</td>\n",
       "      <td>56786</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_30T0006850</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31523</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>gene</td>\n",
       "      <td>1404369</td>\n",
       "      <td>1405546</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_310037100</td>\n",
       "      <td>LINF_310037100</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31524</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>transcript</td>\n",
       "      <td>1404369</td>\n",
       "      <td>1405546</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_31T0037100</td>\n",
       "      <td>LINF_31T0037100</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>LINF_310037100</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31525</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1404572</td>\n",
       "      <td>1405546</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_31T0037100</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_31T0037100</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31526</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>3utr</td>\n",
       "      <td>1404369</td>\n",
       "      <td>1404571</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_31T0037100</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31629</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>gene</td>\n",
       "      <td>1473743</td>\n",
       "      <td>1475202</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_310039200</td>\n",
       "      <td>LINF_310039200</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>None</td>\n",
       "      <td>unknown</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31630</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>transcript</td>\n",
       "      <td>1473743</td>\n",
       "      <td>1475202</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_31T0039200</td>\n",
       "      <td>LINF_31T0039200</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>LINF_310039200</td>\n",
       "      <td>None</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31631</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1473788</td>\n",
       "      <td>1475202</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_31T0039200</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_31T0039200</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31632</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>3utr</td>\n",
       "      <td>1473743</td>\n",
       "      <td>1473787</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_31T0039200</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42001</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>gene</td>\n",
       "      <td>447893</td>\n",
       "      <td>450114</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_360017400</td>\n",
       "      <td>LINF_360017400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Memo-like_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42002</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>transcript</td>\n",
       "      <td>447893</td>\n",
       "      <td>450114</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0017400</td>\n",
       "      <td>LINF_36T0017400</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>LINF_360017400</td>\n",
       "      <td>None</td>\n",
       "      <td>Memo-like_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42003</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>CDS</td>\n",
       "      <td>447893</td>\n",
       "      <td>449005</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0017400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0017400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42004</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>3utr</td>\n",
       "      <td>449006</td>\n",
       "      <td>450114</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0017400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42963</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>gene</td>\n",
       "      <td>1180267</td>\n",
       "      <td>1182741</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_360036000</td>\n",
       "      <td>LINF_360036000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42964</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>transcript</td>\n",
       "      <td>1180267</td>\n",
       "      <td>1182741</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0036000</td>\n",
       "      <td>LINF_36T0036000</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>LINF_360036000</td>\n",
       "      <td>None</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42965</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1180804</td>\n",
       "      <td>1182741</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0036000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0036000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42966</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>3utr</td>\n",
       "      <td>1180267</td>\n",
       "      <td>1180803</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0036000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44784</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>gene</td>\n",
       "      <td>2392103</td>\n",
       "      <td>2394184</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_360071100</td>\n",
       "      <td>LINF_360071100</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF2763)_-_putative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44785</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>transcript</td>\n",
       "      <td>2392103</td>\n",
       "      <td>2394184</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0071100</td>\n",
       "      <td>LINF_36T0071100</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>LINF_360071100</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF2763)_-_putative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44786</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2393831</td>\n",
       "      <td>2394184</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0071100</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0071100</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44787</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>3utr</td>\n",
       "      <td>2392103</td>\n",
       "      <td>2393830</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_36T0071100</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 99
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Negative elements inside GTF elements"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's check how is the data in the dictionary. \n",
    "\n",
    "This way we can every NEGATIVE ELEMENT that is inside each GTF element."
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:53.022292Z",
     "start_time": "2024-10-03T08:01:52.975336Z"
    }
   },
   "source": [
    "# Create the boolean columns for each category in \"feature\"\n",
    "boolean_df = pd.get_dummies(gtf_df['feature'], prefix='', prefix_sep='').astype(bool)\n",
    "\n",
    "gtf_df = pd.concat([gtf_df, boolean_df], axis=1)\n",
    "gtf_df.head()"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "     chrom     feature  start   end strand         gene_id       gene_name  \\\n",
       "0  LinJ.01        gene   1520  5066      -  LINF_010005000  LINF_010005000   \n",
       "1  LinJ.01  transcript   1520  5066      -            None            None   \n",
       "2  LinJ.01         CDS   3710  4711      -            None            None   \n",
       "3  LinJ.01        3utr   1520  3709      -            None            None   \n",
       "4  LinJ.01        5utr   4712  5066      -            None            None   \n",
       "\n",
       "     transcript_id  transcript_name         biotype        parent_id  \\\n",
       "0             None             None  protein_coding             None   \n",
       "1  LINF_01T0005000  LINF_01T0005000  protein_coding   LINF_010005000   \n",
       "2  LINF_01T0005000             None            None  LINF_01T0005000   \n",
       "3             None             None            None  LINF_01T0005000   \n",
       "4             None             None            None  LINF_01T0005000   \n",
       "\n",
       "  pseudogen                                  notes   3utr   5utr    CDS  \\\n",
       "0      None  Protein_of_unknown_function_(DUF2946)  False  False  False   \n",
       "1      None  Protein_of_unknown_function_(DUF2946)  False  False  False   \n",
       "2      None                                   None  False  False   True   \n",
       "3      None                                   None   True  False  False   \n",
       "4      None                                   None  False   True  False   \n",
       "\n",
       "    gene  transcript  \n",
       "0   True       False  \n",
       "1  False        True  \n",
       "2  False       False  \n",
       "3  False       False  \n",
       "4  False       False  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>feature</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>biotype</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>pseudogen</th>\n",
       "      <th>notes</th>\n",
       "      <th>3utr</th>\n",
       "      <th>5utr</th>\n",
       "      <th>CDS</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>gene</td>\n",
       "      <td>1520</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF2946)</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>transcript</td>\n",
       "      <td>1520</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF2946)</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3710</td>\n",
       "      <td>4711</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>3utr</td>\n",
       "      <td>1520</td>\n",
       "      <td>3709</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>5utr</td>\n",
       "      <td>4712</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 100
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:53.922842Z",
     "start_time": "2024-10-03T08:01:53.897161Z"
    }
   },
   "source": [
    " # Let's drop the original \"feature\" column and reorder the columns\n",
    "gtf_df.drop(columns=\"feature\", inplace=True)\n",
    "gtf_df = gtf_df[[\"chrom\", \"start\", \"end\", \"strand\", \"gene_id\", \"transcript_id\", \"parent_id\", \"gene\", \"transcript\", \"CDS\", \"3utr\", \"5utr\", \"pseudogen\", \"notes\"]]\n",
    "gtf_df.head()"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "     chrom  start   end strand         gene_id    transcript_id  \\\n",
       "0  LinJ.01   1520  5066      -  LINF_010005000             None   \n",
       "1  LinJ.01   1520  5066      -            None  LINF_01T0005000   \n",
       "2  LinJ.01   3710  4711      -            None  LINF_01T0005000   \n",
       "3  LinJ.01   1520  3709      -            None             None   \n",
       "4  LinJ.01   4712  5066      -            None             None   \n",
       "\n",
       "         parent_id   gene  transcript    CDS   3utr   5utr pseudogen  \\\n",
       "0             None   True       False  False  False  False      None   \n",
       "1   LINF_010005000  False        True  False  False  False      None   \n",
       "2  LINF_01T0005000  False       False   True  False  False      None   \n",
       "3  LINF_01T0005000  False       False  False   True  False      None   \n",
       "4  LINF_01T0005000  False       False  False  False   True      None   \n",
       "\n",
       "                                   notes  \n",
       "0  Protein_of_unknown_function_(DUF2946)  \n",
       "1  Protein_of_unknown_function_(DUF2946)  \n",
       "2                                   None  \n",
       "3                                   None  \n",
       "4                                   None  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>CDS</th>\n",
       "      <th>3utr</th>\n",
       "      <th>5utr</th>\n",
       "      <th>pseudogen</th>\n",
       "      <th>notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>1520</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF2946)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>1520</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF2946)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>3710</td>\n",
       "      <td>4711</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>1520</td>\n",
       "      <td>3709</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>4712</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 101
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's make sure that we use the same column names for `gtf_df` and `neg_df`"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:55.277157Z",
     "start_time": "2024-10-03T08:01:55.261689Z"
    }
   },
   "source": [
    "print(gtf_df.columns)\n",
    "print(gtf_df.shape)\n",
    "gtf_df.head()"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['chrom', 'start', 'end', 'strand', 'gene_id', 'transcript_id',\n",
      "       'parent_id', 'gene', 'transcript', 'CDS', '3utr', '5utr', 'pseudogen',\n",
      "       'notes'],\n",
      "      dtype='object')\n",
      "(45368, 14)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "     chrom  start   end strand         gene_id    transcript_id  \\\n",
       "0  LinJ.01   1520  5066      -  LINF_010005000             None   \n",
       "1  LinJ.01   1520  5066      -            None  LINF_01T0005000   \n",
       "2  LinJ.01   3710  4711      -            None  LINF_01T0005000   \n",
       "3  LinJ.01   1520  3709      -            None             None   \n",
       "4  LinJ.01   4712  5066      -            None             None   \n",
       "\n",
       "         parent_id   gene  transcript    CDS   3utr   5utr pseudogen  \\\n",
       "0             None   True       False  False  False  False      None   \n",
       "1   LINF_010005000  False        True  False  False  False      None   \n",
       "2  LINF_01T0005000  False       False   True  False  False      None   \n",
       "3  LINF_01T0005000  False       False  False   True  False      None   \n",
       "4  LINF_01T0005000  False       False  False  False   True      None   \n",
       "\n",
       "                                   notes  \n",
       "0  Protein_of_unknown_function_(DUF2946)  \n",
       "1  Protein_of_unknown_function_(DUF2946)  \n",
       "2                                   None  \n",
       "3                                   None  \n",
       "4                                   None  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>CDS</th>\n",
       "      <th>3utr</th>\n",
       "      <th>5utr</th>\n",
       "      <th>pseudogen</th>\n",
       "      <th>notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>1520</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF2946)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>1520</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF2946)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>3710</td>\n",
       "      <td>4711</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>1520</td>\n",
       "      <td>3709</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>4712</td>\n",
       "      <td>5066</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 102
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### Clean \"notes\" column"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:56.814903Z",
     "start_time": "2024-10-03T08:01:56.802078Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gtf_functions = gtf_df[\"notes\"].value_counts()\n",
    "gtf_functions"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "notes\n",
       "hypothetical_protein_-_conserved                                            4043\n",
       "protein_of_unknown_function_-_conserved                                     1536\n",
       "hypothetical_protein                                                         372\n",
       "protein_kinase                                                               140\n",
       "hypothetical_protein_-__conserved                                             86\n",
       "                                                                            ... \n",
       "tb-292_membrane_associated_protein-like_protein_conflicted_zone_in_study       1\n",
       "tRNA                                                                           1\n",
       "tRNA-seC                                                                       1\n",
       "tRNA-val|Anticodon_gac                                                         1\n",
       "tRNA-Cys                                                                       1\n",
       "Name: count, Length: 4436, dtype: int64"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 103
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:58.132225Z",
     "start_time": "2024-10-03T08:01:58.064560Z"
    }
   },
   "source": [
    "# In `gtf_functions` filter all names with \"protein\" and \"hypothetical\" inside using a regex:\n",
    "gtf_functions_protein = gtf_functions[gtf_functions.index.str.contains(r\"(?=.*protein)(?=.*hypothetical)\", case=False)]\n",
    "gtf_functions_protein"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "notes\n",
       "hypothetical_protein_-_conserved                             4043\n",
       "hypothetical_protein                                          372\n",
       "hypothetical_protein_-__conserved                              86\n",
       "hypothetical_protein_-_unknown_function                        10\n",
       "Hypothetical_protein                                            8\n",
       "hypothetical_protein_-_conserved_                               4\n",
       "Conserved_hypothetical_ATP_binding_protein_-_putative           4\n",
       "hypothetical_protein_-_conserved__                              2\n",
       "hypothetical_protein_-_conserved_conflicted_zone_in_study       2\n",
       "hypothetical_protein_-_conserved|GF1                            2\n",
       "hypothetical_protein,_conserved                                 2\n",
       "hypothetical_protein_pseudogene                                 2\n",
       "hypothetical_protein_conserved                                  2\n",
       "hypothetical_protein-unknown_function                           2\n",
       "hypothetical_protein_conflicted_zone_in_study                   1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 104
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:01:59.065583Z",
     "start_time": "2024-10-03T08:01:59.061291Z"
    }
   },
   "cell_type": "code",
   "source": "# Let's change ",
   "outputs": [],
   "execution_count": 105
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:00.052146Z",
     "start_time": "2024-10-03T08:02:00.046186Z"
    }
   },
   "source": [
    "# Let's rename in `neg_df` the next columns:\n",
    "# 'sseqid' to 'chrom'\n",
    "# 'sstart' to 'start'\n",
    "# 'send' to 'end'\n",
    "neg_df.rename(columns={\"sseqid\": \"chrom\", \"sstart\": \"start\", \"send\": \"end\"}, inplace=True)\n",
    "print(neg_df.columns)"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['chrom', 'start', 'end', 'sstrand', 'sseq', 'name'], dtype='object')\n"
     ]
    }
   ],
   "execution_count": 106
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now let's repeat the dictionary process again:\n",
    "The next dictionary will have elements only COMPLETELY inside GTF elements"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:03.968786Z",
     "start_time": "2024-10-03T08:02:00.772439Z"
    }
   },
   "source": [
    "# # Create interval columns\n",
    "neg_df[\"interval\"] = pd.IntervalIndex.from_arrays(neg_df[\"start\"], neg_df[\"end\"], closed=\"both\")\n",
    "gtf_df[\"interval\"] = pd.IntervalIndex.from_arrays(gtf_df[\"start\"], gtf_df[\"end\"], closed=\"both\")\n",
    "\n",
    "# initialize dict\n",
    "neg_gtf_dict = {neg_name: [] for neg_name in neg_df[\"name\"].unique()}\n",
    "\n",
    "# Find elements in neg_df that are inside gtf_df\n",
    "# Find contains using boolean indexing\n",
    "for i, neg_row in neg_df.iterrows():\n",
    "    # Boolean mask for intervals that contain the neg_df interval\n",
    "    print(f\"Analyzing elem {i + 1}/{neg_df.shape[0]}\")\n",
    "\n",
    "    mask = (gtf_df['chrom'] == neg_row['chrom']) & \\\n",
    "           (gtf_df['start'] <= neg_row['start']) & \\\n",
    "           (gtf_df['end'] >= neg_row['end'])\n",
    "    contains = gtf_df[mask]\n",
    "    for j, gtf_row in contains.iterrows():\n",
    "        neg_gtf_dict[neg_row['name']].append(gtf_row.to_dict())"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analyzing elem 1/682\n",
      "Analyzing elem 2/682\n",
      "Analyzing elem 3/682\n",
      "Analyzing elem 4/682\n",
      "Analyzing elem 5/682\n",
      "Analyzing elem 6/682\n",
      "Analyzing elem 7/682\n",
      "Analyzing elem 8/682\n",
      "Analyzing elem 9/682\n",
      "Analyzing elem 10/682\n",
      "Analyzing elem 11/682\n",
      "Analyzing elem 12/682\n",
      "Analyzing elem 13/682\n",
      "Analyzing elem 14/682\n",
      "Analyzing elem 15/682\n",
      "Analyzing elem 16/682\n",
      "Analyzing elem 17/682\n",
      "Analyzing elem 18/682\n",
      "Analyzing elem 19/682\n",
      "Analyzing elem 20/682\n",
      "Analyzing elem 21/682\n",
      "Analyzing elem 22/682\n",
      "Analyzing elem 23/682\n",
      "Analyzing elem 24/682\n",
      "Analyzing elem 25/682\n",
      "Analyzing elem 26/682\n",
      "Analyzing elem 27/682\n",
      "Analyzing elem 28/682\n",
      "Analyzing elem 29/682\n",
      "Analyzing elem 30/682\n",
      "Analyzing elem 31/682\n",
      "Analyzing elem 32/682\n",
      "Analyzing elem 33/682\n",
      "Analyzing elem 34/682\n",
      "Analyzing elem 35/682\n",
      "Analyzing elem 36/682\n",
      "Analyzing elem 37/682\n",
      "Analyzing elem 38/682\n",
      "Analyzing elem 39/682\n",
      "Analyzing elem 40/682\n",
      "Analyzing elem 41/682\n",
      "Analyzing elem 42/682\n",
      "Analyzing elem 43/682\n",
      "Analyzing elem 44/682\n",
      "Analyzing elem 45/682\n",
      "Analyzing elem 46/682\n",
      "Analyzing elem 47/682\n",
      "Analyzing elem 48/682\n",
      "Analyzing elem 49/682\n",
      "Analyzing elem 50/682\n",
      "Analyzing elem 51/682\n",
      "Analyzing elem 52/682\n",
      "Analyzing elem 53/682\n",
      "Analyzing elem 54/682\n",
      "Analyzing elem 55/682\n",
      "Analyzing elem 56/682\n",
      "Analyzing elem 57/682\n",
      "Analyzing elem 58/682\n",
      "Analyzing elem 59/682\n",
      "Analyzing elem 60/682\n",
      "Analyzing elem 61/682\n",
      "Analyzing elem 62/682\n",
      "Analyzing elem 63/682\n",
      "Analyzing elem 64/682\n",
      "Analyzing elem 65/682\n",
      "Analyzing elem 66/682\n",
      "Analyzing elem 67/682\n",
      "Analyzing elem 68/682\n",
      "Analyzing elem 69/682\n",
      "Analyzing elem 70/682\n",
      "Analyzing elem 71/682\n",
      "Analyzing elem 72/682\n",
      "Analyzing elem 73/682\n",
      "Analyzing elem 74/682\n",
      "Analyzing elem 75/682\n",
      "Analyzing elem 76/682\n",
      "Analyzing elem 77/682\n",
      "Analyzing elem 78/682\n",
      "Analyzing elem 79/682\n",
      "Analyzing elem 80/682\n",
      "Analyzing elem 81/682\n",
      "Analyzing elem 82/682\n",
      "Analyzing elem 83/682\n",
      "Analyzing elem 84/682\n",
      "Analyzing elem 85/682\n",
      "Analyzing elem 86/682\n",
      "Analyzing elem 87/682\n",
      "Analyzing elem 88/682\n",
      "Analyzing elem 89/682\n",
      "Analyzing elem 90/682\n",
      "Analyzing elem 91/682\n",
      "Analyzing elem 92/682\n",
      "Analyzing elem 93/682\n",
      "Analyzing elem 94/682\n",
      "Analyzing elem 95/682\n",
      "Analyzing elem 96/682\n",
      "Analyzing elem 97/682\n",
      "Analyzing elem 98/682\n",
      "Analyzing elem 99/682\n",
      "Analyzing elem 100/682\n",
      "Analyzing elem 101/682\n",
      "Analyzing elem 102/682\n",
      "Analyzing elem 103/682\n",
      "Analyzing elem 104/682\n",
      "Analyzing elem 105/682\n",
      "Analyzing elem 106/682\n",
      "Analyzing elem 107/682\n",
      "Analyzing elem 108/682\n",
      "Analyzing elem 109/682\n",
      "Analyzing elem 110/682\n",
      "Analyzing elem 111/682\n",
      "Analyzing elem 112/682\n",
      "Analyzing elem 113/682\n",
      "Analyzing elem 114/682\n",
      "Analyzing elem 115/682\n",
      "Analyzing elem 116/682\n",
      "Analyzing elem 117/682\n",
      "Analyzing elem 118/682\n",
      "Analyzing elem 119/682\n",
      "Analyzing elem 120/682\n",
      "Analyzing elem 121/682\n",
      "Analyzing elem 122/682\n",
      "Analyzing elem 123/682\n",
      "Analyzing elem 124/682\n",
      "Analyzing elem 125/682\n",
      "Analyzing elem 126/682\n",
      "Analyzing elem 127/682\n",
      "Analyzing elem 128/682\n",
      "Analyzing elem 129/682\n",
      "Analyzing elem 130/682\n",
      "Analyzing elem 131/682\n",
      "Analyzing elem 132/682\n",
      "Analyzing elem 133/682\n",
      "Analyzing elem 134/682\n",
      "Analyzing elem 135/682\n",
      "Analyzing elem 136/682\n",
      "Analyzing elem 137/682\n",
      "Analyzing elem 138/682\n",
      "Analyzing elem 139/682\n",
      "Analyzing elem 140/682\n",
      "Analyzing elem 141/682\n",
      "Analyzing elem 142/682\n",
      "Analyzing elem 143/682\n",
      "Analyzing elem 144/682\n",
      "Analyzing elem 145/682\n",
      "Analyzing elem 146/682\n",
      "Analyzing elem 147/682\n",
      "Analyzing elem 148/682\n",
      "Analyzing elem 149/682\n",
      "Analyzing elem 150/682\n",
      "Analyzing elem 151/682\n",
      "Analyzing elem 152/682\n",
      "Analyzing elem 153/682\n",
      "Analyzing elem 154/682\n",
      "Analyzing elem 155/682\n",
      "Analyzing elem 156/682\n",
      "Analyzing elem 157/682\n",
      "Analyzing elem 158/682\n",
      "Analyzing elem 159/682\n",
      "Analyzing elem 160/682\n",
      "Analyzing elem 161/682\n",
      "Analyzing elem 162/682\n",
      "Analyzing elem 163/682\n",
      "Analyzing elem 164/682\n",
      "Analyzing elem 165/682\n",
      "Analyzing elem 166/682\n",
      "Analyzing elem 167/682\n",
      "Analyzing elem 168/682\n",
      "Analyzing elem 169/682\n",
      "Analyzing elem 170/682\n",
      "Analyzing elem 171/682\n",
      "Analyzing elem 172/682\n",
      "Analyzing elem 173/682\n",
      "Analyzing elem 174/682\n",
      "Analyzing elem 175/682\n",
      "Analyzing elem 176/682\n",
      "Analyzing elem 177/682\n",
      "Analyzing elem 178/682\n",
      "Analyzing elem 179/682\n",
      "Analyzing elem 180/682\n",
      "Analyzing elem 181/682\n",
      "Analyzing elem 182/682\n",
      "Analyzing elem 183/682\n",
      "Analyzing elem 184/682\n",
      "Analyzing elem 185/682\n",
      "Analyzing elem 186/682\n",
      "Analyzing elem 187/682\n",
      "Analyzing elem 188/682\n",
      "Analyzing elem 189/682\n",
      "Analyzing elem 190/682\n",
      "Analyzing elem 191/682\n",
      "Analyzing elem 192/682\n",
      "Analyzing elem 193/682\n",
      "Analyzing elem 194/682\n",
      "Analyzing elem 195/682\n",
      "Analyzing elem 196/682\n",
      "Analyzing elem 197/682\n",
      "Analyzing elem 198/682\n",
      "Analyzing elem 199/682\n",
      "Analyzing elem 200/682\n",
      "Analyzing elem 201/682\n",
      "Analyzing elem 202/682\n",
      "Analyzing elem 203/682\n",
      "Analyzing elem 204/682\n",
      "Analyzing elem 205/682\n",
      "Analyzing elem 206/682\n",
      "Analyzing elem 207/682\n",
      "Analyzing elem 208/682\n",
      "Analyzing elem 209/682\n",
      "Analyzing elem 210/682\n",
      "Analyzing elem 211/682\n",
      "Analyzing elem 212/682\n",
      "Analyzing elem 213/682\n",
      "Analyzing elem 214/682\n",
      "Analyzing elem 215/682\n",
      "Analyzing elem 216/682\n",
      "Analyzing elem 217/682\n",
      "Analyzing elem 218/682\n",
      "Analyzing elem 219/682\n",
      "Analyzing elem 220/682\n",
      "Analyzing elem 221/682\n",
      "Analyzing elem 222/682\n",
      "Analyzing elem 223/682\n",
      "Analyzing elem 224/682\n",
      "Analyzing elem 225/682\n",
      "Analyzing elem 226/682\n",
      "Analyzing elem 227/682\n",
      "Analyzing elem 228/682\n",
      "Analyzing elem 229/682\n",
      "Analyzing elem 230/682\n",
      "Analyzing elem 231/682\n",
      "Analyzing elem 232/682\n",
      "Analyzing elem 233/682\n",
      "Analyzing elem 234/682\n",
      "Analyzing elem 235/682\n",
      "Analyzing elem 236/682\n",
      "Analyzing elem 237/682\n",
      "Analyzing elem 238/682\n",
      "Analyzing elem 239/682\n",
      "Analyzing elem 240/682\n",
      "Analyzing elem 241/682\n",
      "Analyzing elem 242/682\n",
      "Analyzing elem 243/682\n",
      "Analyzing elem 244/682\n",
      "Analyzing elem 245/682\n",
      "Analyzing elem 246/682\n",
      "Analyzing elem 247/682\n",
      "Analyzing elem 248/682\n",
      "Analyzing elem 249/682\n",
      "Analyzing elem 250/682\n",
      "Analyzing elem 251/682\n",
      "Analyzing elem 252/682\n",
      "Analyzing elem 253/682\n",
      "Analyzing elem 254/682\n",
      "Analyzing elem 255/682\n",
      "Analyzing elem 256/682\n",
      "Analyzing elem 257/682\n",
      "Analyzing elem 258/682\n",
      "Analyzing elem 259/682\n",
      "Analyzing elem 260/682\n",
      "Analyzing elem 261/682\n",
      "Analyzing elem 262/682\n",
      "Analyzing elem 263/682\n",
      "Analyzing elem 264/682\n",
      "Analyzing elem 265/682\n",
      "Analyzing elem 266/682\n",
      "Analyzing elem 267/682\n",
      "Analyzing elem 268/682\n",
      "Analyzing elem 269/682\n",
      "Analyzing elem 270/682\n",
      "Analyzing elem 271/682\n",
      "Analyzing elem 272/682\n",
      "Analyzing elem 273/682\n",
      "Analyzing elem 274/682\n",
      "Analyzing elem 275/682\n",
      "Analyzing elem 276/682\n",
      "Analyzing elem 277/682\n",
      "Analyzing elem 278/682\n",
      "Analyzing elem 279/682\n",
      "Analyzing elem 280/682\n",
      "Analyzing elem 281/682\n",
      "Analyzing elem 282/682\n",
      "Analyzing elem 283/682\n",
      "Analyzing elem 284/682\n",
      "Analyzing elem 285/682\n",
      "Analyzing elem 286/682\n",
      "Analyzing elem 287/682\n",
      "Analyzing elem 288/682\n",
      "Analyzing elem 289/682\n",
      "Analyzing elem 290/682\n",
      "Analyzing elem 291/682\n",
      "Analyzing elem 292/682\n",
      "Analyzing elem 293/682\n",
      "Analyzing elem 294/682\n",
      "Analyzing elem 295/682\n",
      "Analyzing elem 296/682\n",
      "Analyzing elem 297/682\n",
      "Analyzing elem 298/682\n",
      "Analyzing elem 299/682\n",
      "Analyzing elem 300/682\n",
      "Analyzing elem 301/682\n",
      "Analyzing elem 302/682\n",
      "Analyzing elem 303/682\n",
      "Analyzing elem 304/682\n",
      "Analyzing elem 305/682\n",
      "Analyzing elem 306/682\n",
      "Analyzing elem 307/682\n",
      "Analyzing elem 308/682\n",
      "Analyzing elem 309/682\n",
      "Analyzing elem 310/682\n",
      "Analyzing elem 311/682\n",
      "Analyzing elem 312/682\n",
      "Analyzing elem 313/682\n",
      "Analyzing elem 314/682\n",
      "Analyzing elem 315/682\n",
      "Analyzing elem 316/682\n",
      "Analyzing elem 317/682\n",
      "Analyzing elem 318/682\n",
      "Analyzing elem 319/682\n",
      "Analyzing elem 320/682\n",
      "Analyzing elem 321/682\n",
      "Analyzing elem 322/682\n",
      "Analyzing elem 323/682\n",
      "Analyzing elem 324/682\n",
      "Analyzing elem 325/682\n",
      "Analyzing elem 326/682\n",
      "Analyzing elem 327/682\n",
      "Analyzing elem 328/682\n",
      "Analyzing elem 329/682\n",
      "Analyzing elem 330/682\n",
      "Analyzing elem 331/682\n",
      "Analyzing elem 332/682\n",
      "Analyzing elem 333/682\n",
      "Analyzing elem 334/682\n",
      "Analyzing elem 335/682\n",
      "Analyzing elem 336/682\n",
      "Analyzing elem 337/682\n",
      "Analyzing elem 338/682\n",
      "Analyzing elem 339/682\n",
      "Analyzing elem 340/682\n",
      "Analyzing elem 341/682\n",
      "Analyzing elem 342/682\n",
      "Analyzing elem 343/682\n",
      "Analyzing elem 344/682\n",
      "Analyzing elem 345/682\n",
      "Analyzing elem 346/682\n",
      "Analyzing elem 347/682\n",
      "Analyzing elem 348/682\n",
      "Analyzing elem 349/682\n",
      "Analyzing elem 350/682\n",
      "Analyzing elem 351/682\n",
      "Analyzing elem 352/682\n",
      "Analyzing elem 353/682\n",
      "Analyzing elem 354/682\n",
      "Analyzing elem 355/682\n",
      "Analyzing elem 356/682\n",
      "Analyzing elem 357/682\n",
      "Analyzing elem 358/682\n",
      "Analyzing elem 359/682\n",
      "Analyzing elem 360/682\n",
      "Analyzing elem 361/682\n",
      "Analyzing elem 362/682\n",
      "Analyzing elem 363/682\n",
      "Analyzing elem 364/682\n",
      "Analyzing elem 365/682\n",
      "Analyzing elem 366/682\n",
      "Analyzing elem 367/682\n",
      "Analyzing elem 368/682\n",
      "Analyzing elem 369/682\n",
      "Analyzing elem 370/682\n",
      "Analyzing elem 371/682\n",
      "Analyzing elem 372/682\n",
      "Analyzing elem 373/682\n",
      "Analyzing elem 374/682\n",
      "Analyzing elem 375/682\n",
      "Analyzing elem 376/682\n",
      "Analyzing elem 377/682\n",
      "Analyzing elem 378/682\n",
      "Analyzing elem 379/682\n",
      "Analyzing elem 380/682\n",
      "Analyzing elem 381/682\n",
      "Analyzing elem 382/682\n",
      "Analyzing elem 383/682\n",
      "Analyzing elem 384/682\n",
      "Analyzing elem 385/682\n",
      "Analyzing elem 386/682\n",
      "Analyzing elem 387/682\n",
      "Analyzing elem 388/682\n",
      "Analyzing elem 389/682\n",
      "Analyzing elem 390/682\n",
      "Analyzing elem 391/682\n",
      "Analyzing elem 392/682\n",
      "Analyzing elem 393/682\n",
      "Analyzing elem 394/682\n",
      "Analyzing elem 395/682\n",
      "Analyzing elem 396/682\n",
      "Analyzing elem 397/682\n",
      "Analyzing elem 398/682\n",
      "Analyzing elem 399/682\n",
      "Analyzing elem 400/682\n",
      "Analyzing elem 401/682\n",
      "Analyzing elem 402/682\n",
      "Analyzing elem 403/682\n",
      "Analyzing elem 404/682\n",
      "Analyzing elem 405/682\n",
      "Analyzing elem 406/682\n",
      "Analyzing elem 407/682\n",
      "Analyzing elem 408/682\n",
      "Analyzing elem 409/682\n",
      "Analyzing elem 410/682\n",
      "Analyzing elem 411/682\n",
      "Analyzing elem 412/682\n",
      "Analyzing elem 413/682\n",
      "Analyzing elem 414/682\n",
      "Analyzing elem 415/682\n",
      "Analyzing elem 416/682\n",
      "Analyzing elem 417/682\n",
      "Analyzing elem 418/682\n",
      "Analyzing elem 419/682\n",
      "Analyzing elem 420/682\n",
      "Analyzing elem 421/682\n",
      "Analyzing elem 422/682\n",
      "Analyzing elem 423/682\n",
      "Analyzing elem 424/682\n",
      "Analyzing elem 425/682\n",
      "Analyzing elem 426/682\n",
      "Analyzing elem 427/682\n",
      "Analyzing elem 428/682\n",
      "Analyzing elem 429/682\n",
      "Analyzing elem 430/682\n",
      "Analyzing elem 431/682\n",
      "Analyzing elem 432/682\n",
      "Analyzing elem 433/682\n",
      "Analyzing elem 434/682\n",
      "Analyzing elem 435/682\n",
      "Analyzing elem 436/682\n",
      "Analyzing elem 437/682\n",
      "Analyzing elem 438/682\n",
      "Analyzing elem 439/682\n",
      "Analyzing elem 440/682\n",
      "Analyzing elem 441/682\n",
      "Analyzing elem 442/682\n",
      "Analyzing elem 443/682\n",
      "Analyzing elem 444/682\n",
      "Analyzing elem 445/682\n",
      "Analyzing elem 446/682\n",
      "Analyzing elem 447/682\n",
      "Analyzing elem 448/682\n",
      "Analyzing elem 449/682\n",
      "Analyzing elem 450/682\n",
      "Analyzing elem 451/682\n",
      "Analyzing elem 452/682\n",
      "Analyzing elem 453/682\n",
      "Analyzing elem 454/682\n",
      "Analyzing elem 455/682\n",
      "Analyzing elem 456/682\n",
      "Analyzing elem 457/682\n",
      "Analyzing elem 458/682\n",
      "Analyzing elem 459/682\n",
      "Analyzing elem 460/682\n",
      "Analyzing elem 461/682\n",
      "Analyzing elem 462/682\n",
      "Analyzing elem 463/682\n",
      "Analyzing elem 464/682\n",
      "Analyzing elem 465/682\n",
      "Analyzing elem 466/682\n",
      "Analyzing elem 467/682\n",
      "Analyzing elem 468/682\n",
      "Analyzing elem 469/682\n",
      "Analyzing elem 470/682\n",
      "Analyzing elem 471/682\n",
      "Analyzing elem 472/682\n",
      "Analyzing elem 473/682\n",
      "Analyzing elem 474/682\n",
      "Analyzing elem 475/682\n",
      "Analyzing elem 476/682\n",
      "Analyzing elem 477/682\n",
      "Analyzing elem 478/682\n",
      "Analyzing elem 479/682\n",
      "Analyzing elem 480/682\n",
      "Analyzing elem 481/682\n",
      "Analyzing elem 482/682\n",
      "Analyzing elem 483/682\n",
      "Analyzing elem 484/682\n",
      "Analyzing elem 485/682\n",
      "Analyzing elem 486/682\n",
      "Analyzing elem 487/682\n",
      "Analyzing elem 488/682\n",
      "Analyzing elem 489/682\n",
      "Analyzing elem 490/682\n",
      "Analyzing elem 491/682\n",
      "Analyzing elem 492/682\n",
      "Analyzing elem 493/682\n",
      "Analyzing elem 494/682\n",
      "Analyzing elem 495/682\n",
      "Analyzing elem 496/682\n",
      "Analyzing elem 497/682\n",
      "Analyzing elem 498/682\n",
      "Analyzing elem 499/682\n",
      "Analyzing elem 500/682\n",
      "Analyzing elem 501/682\n",
      "Analyzing elem 502/682\n",
      "Analyzing elem 503/682\n",
      "Analyzing elem 504/682\n",
      "Analyzing elem 505/682\n",
      "Analyzing elem 506/682\n",
      "Analyzing elem 507/682\n",
      "Analyzing elem 508/682\n",
      "Analyzing elem 509/682\n",
      "Analyzing elem 510/682\n",
      "Analyzing elem 511/682\n",
      "Analyzing elem 512/682\n",
      "Analyzing elem 513/682\n",
      "Analyzing elem 514/682\n",
      "Analyzing elem 515/682\n",
      "Analyzing elem 516/682\n",
      "Analyzing elem 517/682\n",
      "Analyzing elem 518/682\n",
      "Analyzing elem 519/682\n",
      "Analyzing elem 520/682\n",
      "Analyzing elem 521/682\n",
      "Analyzing elem 522/682\n",
      "Analyzing elem 523/682\n",
      "Analyzing elem 524/682\n",
      "Analyzing elem 525/682\n",
      "Analyzing elem 526/682\n",
      "Analyzing elem 527/682\n",
      "Analyzing elem 528/682\n",
      "Analyzing elem 529/682\n",
      "Analyzing elem 530/682\n",
      "Analyzing elem 531/682\n",
      "Analyzing elem 532/682\n",
      "Analyzing elem 533/682\n",
      "Analyzing elem 534/682\n",
      "Analyzing elem 535/682\n",
      "Analyzing elem 536/682\n",
      "Analyzing elem 537/682\n",
      "Analyzing elem 538/682\n",
      "Analyzing elem 539/682\n",
      "Analyzing elem 540/682\n",
      "Analyzing elem 541/682\n",
      "Analyzing elem 542/682\n",
      "Analyzing elem 543/682\n",
      "Analyzing elem 544/682\n",
      "Analyzing elem 545/682\n",
      "Analyzing elem 546/682\n",
      "Analyzing elem 547/682\n",
      "Analyzing elem 548/682\n",
      "Analyzing elem 549/682\n",
      "Analyzing elem 550/682\n",
      "Analyzing elem 551/682\n",
      "Analyzing elem 552/682\n",
      "Analyzing elem 553/682\n",
      "Analyzing elem 554/682\n",
      "Analyzing elem 555/682\n",
      "Analyzing elem 556/682\n",
      "Analyzing elem 557/682\n",
      "Analyzing elem 558/682\n",
      "Analyzing elem 559/682\n",
      "Analyzing elem 560/682\n",
      "Analyzing elem 561/682\n",
      "Analyzing elem 562/682\n",
      "Analyzing elem 563/682\n",
      "Analyzing elem 564/682\n",
      "Analyzing elem 565/682\n",
      "Analyzing elem 566/682\n",
      "Analyzing elem 567/682\n",
      "Analyzing elem 568/682\n",
      "Analyzing elem 569/682\n",
      "Analyzing elem 570/682\n",
      "Analyzing elem 571/682\n",
      "Analyzing elem 572/682\n",
      "Analyzing elem 573/682\n",
      "Analyzing elem 574/682\n",
      "Analyzing elem 575/682\n",
      "Analyzing elem 576/682\n",
      "Analyzing elem 577/682\n",
      "Analyzing elem 578/682\n",
      "Analyzing elem 579/682\n",
      "Analyzing elem 580/682\n",
      "Analyzing elem 581/682\n",
      "Analyzing elem 582/682\n",
      "Analyzing elem 583/682\n",
      "Analyzing elem 584/682\n",
      "Analyzing elem 585/682\n",
      "Analyzing elem 586/682\n",
      "Analyzing elem 587/682\n",
      "Analyzing elem 588/682\n",
      "Analyzing elem 589/682\n",
      "Analyzing elem 590/682\n",
      "Analyzing elem 591/682\n",
      "Analyzing elem 592/682\n",
      "Analyzing elem 593/682\n",
      "Analyzing elem 594/682\n",
      "Analyzing elem 595/682\n",
      "Analyzing elem 596/682\n",
      "Analyzing elem 597/682\n",
      "Analyzing elem 598/682\n",
      "Analyzing elem 599/682\n",
      "Analyzing elem 600/682\n",
      "Analyzing elem 601/682\n",
      "Analyzing elem 602/682\n",
      "Analyzing elem 603/682\n",
      "Analyzing elem 604/682\n",
      "Analyzing elem 605/682\n",
      "Analyzing elem 606/682\n",
      "Analyzing elem 607/682\n",
      "Analyzing elem 608/682\n",
      "Analyzing elem 609/682\n",
      "Analyzing elem 610/682\n",
      "Analyzing elem 611/682\n",
      "Analyzing elem 612/682\n",
      "Analyzing elem 613/682\n",
      "Analyzing elem 614/682\n",
      "Analyzing elem 615/682\n",
      "Analyzing elem 616/682\n",
      "Analyzing elem 617/682\n",
      "Analyzing elem 618/682\n",
      "Analyzing elem 619/682\n",
      "Analyzing elem 620/682\n",
      "Analyzing elem 621/682\n",
      "Analyzing elem 622/682\n",
      "Analyzing elem 623/682\n",
      "Analyzing elem 624/682\n",
      "Analyzing elem 625/682\n",
      "Analyzing elem 626/682\n",
      "Analyzing elem 627/682\n",
      "Analyzing elem 628/682\n",
      "Analyzing elem 629/682\n",
      "Analyzing elem 630/682\n",
      "Analyzing elem 631/682\n",
      "Analyzing elem 632/682\n",
      "Analyzing elem 633/682\n",
      "Analyzing elem 634/682\n",
      "Analyzing elem 635/682\n",
      "Analyzing elem 636/682\n",
      "Analyzing elem 637/682\n",
      "Analyzing elem 638/682\n",
      "Analyzing elem 639/682\n",
      "Analyzing elem 640/682\n",
      "Analyzing elem 641/682\n",
      "Analyzing elem 642/682\n",
      "Analyzing elem 643/682\n",
      "Analyzing elem 644/682\n",
      "Analyzing elem 645/682\n",
      "Analyzing elem 646/682\n",
      "Analyzing elem 647/682\n",
      "Analyzing elem 648/682\n",
      "Analyzing elem 649/682\n",
      "Analyzing elem 650/682\n",
      "Analyzing elem 651/682\n",
      "Analyzing elem 652/682\n",
      "Analyzing elem 653/682\n",
      "Analyzing elem 654/682\n",
      "Analyzing elem 655/682\n",
      "Analyzing elem 656/682\n",
      "Analyzing elem 657/682\n",
      "Analyzing elem 658/682\n",
      "Analyzing elem 659/682\n",
      "Analyzing elem 660/682\n",
      "Analyzing elem 661/682\n",
      "Analyzing elem 662/682\n",
      "Analyzing elem 663/682\n",
      "Analyzing elem 664/682\n",
      "Analyzing elem 665/682\n",
      "Analyzing elem 666/682\n",
      "Analyzing elem 667/682\n",
      "Analyzing elem 668/682\n",
      "Analyzing elem 669/682\n",
      "Analyzing elem 670/682\n",
      "Analyzing elem 671/682\n",
      "Analyzing elem 672/682\n",
      "Analyzing elem 673/682\n",
      "Analyzing elem 674/682\n",
      "Analyzing elem 675/682\n",
      "Analyzing elem 676/682\n",
      "Analyzing elem 677/682\n",
      "Analyzing elem 678/682\n",
      "Analyzing elem 679/682\n",
      "Analyzing elem 680/682\n",
      "Analyzing elem 681/682\n",
      "Analyzing elem 682/682\n"
     ]
    }
   ],
   "execution_count": 107
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:04.004760Z",
     "start_time": "2024-10-03T08:02:03.999750Z"
    }
   },
   "source": [
    "# Prepare a pre JSON dict to not alter the original dict\n",
    "neg_gtf_relation_pre_json = neg_gtf_dict\n",
    "\n",
    "# Let's check the data\n",
    "# print(neg_gtf_relation_pre_json)"
   ],
   "outputs": [],
   "execution_count": 108
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:04.108611Z",
     "start_time": "2024-10-03T08:02:04.079928Z"
    }
   },
   "source": [
    "# Let's count the data\n",
    "counter_neg_inside = 0\n",
    "counter_neg_not_inside = 0\n",
    "for key, value in neg_gtf_dict.items():\n",
    "    print(\"=\"*50)\n",
    "    print(f\"{key}:\")\n",
    "    if len(value) > 0:\n",
    "        counter_neg_inside += 1\n",
    "        for elem in value:\n",
    "            print(f\"\\t{elem}\")\n",
    "    else:\n",
    "        counter_neg_not_inside += 1"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==================================================\n",
      "rejected_noCDS_c01.10:\n",
      "\t{'chrom': 'LinJ.01', 'start': 34736, 'end': 37218, 'strand': '-', 'gene_id': 'LINF_010006300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved__', 'interval': Interval(34736, 37218, closed='both')}\n",
      "\t{'chrom': 'LinJ.01', 'start': 34736, 'end': 37218, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_01T0006300', 'parent_id': 'LINF_010006300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved__', 'interval': Interval(34736, 37218, closed='both')}\n",
      "\t{'chrom': 'LinJ.01', 'start': 34736, 'end': 36818, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_01T0006300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(34736, 36818, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c01.20:\n",
      "==================================================\n",
      "rejected_noCDS_c01.30:\n",
      "\t{'chrom': 'LinJ.01', 'start': 145613, 'end': 146868, 'strand': '+', 'gene_id': 'LINF_010010350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(145613, 146868, closed='both')}\n",
      "\t{'chrom': 'LinJ.01', 'start': 145613, 'end': 146868, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_01T0010350', 'parent_id': 'LINF_010010350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(145613, 146868, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c01.40:\n",
      "==================================================\n",
      "rejected_noCDS_c01.50:\n",
      "==================================================\n",
      "rejected_noCDS_c01.60A:\n",
      "==================================================\n",
      "rejected_noCDS_c02.70:\n",
      "\t{'chrom': 'LinJ.02', 'start': 16154, 'end': 18430, 'strand': '-', 'gene_id': 'LINF_020005400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'cytochrome_b-domain_protein', 'interval': Interval(16154, 18430, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 16154, 'end': 18430, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_02T0005400', 'parent_id': 'LINF_020005400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'cytochrome_b-domain_protein', 'interval': Interval(16154, 18430, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 16154, 'end': 17546, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_02T0005400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(16154, 17546, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c02.80:\n",
      "\t{'chrom': 'LinJ.02', 'start': 73083, 'end': 83537, 'strand': '-', 'gene_id': 'LINF_020006700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'phosphoglycan_beta_1-3_galactosyltransferase', 'interval': Interval(73083, 83537, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 73083, 'end': 83537, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_02T0006700', 'parent_id': 'LINF_020006700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'phosphoglycan_beta_1-3_galactosyltransferase', 'interval': Interval(73083, 83537, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 73083, 'end': 79710, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_02T0006700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(73083, 79710, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c02.90B:\n",
      "\t{'chrom': 'LinJ.02', 'start': 93941, 'end': 102762, 'strand': '-', 'gene_id': 'LINF_020007000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'phosphoglycan_beta_1-3_galactosyltransferase', 'interval': Interval(93941, 102762, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 93941, 'end': 102762, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_02T0007000', 'parent_id': 'LINF_020007000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'phosphoglycan_beta_1-3_galactosyltransferase', 'interval': Interval(93941, 102762, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 93941, 'end': 99642, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_02T0007000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(93941, 99642, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c02.100:\n",
      "\t{'chrom': 'LinJ.02', 'start': 93941, 'end': 102762, 'strand': '-', 'gene_id': 'LINF_020007000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'phosphoglycan_beta_1-3_galactosyltransferase', 'interval': Interval(93941, 102762, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 93941, 'end': 102762, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_02T0007000', 'parent_id': 'LINF_020007000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'phosphoglycan_beta_1-3_galactosyltransferase', 'interval': Interval(93941, 102762, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 93941, 'end': 99642, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_02T0007000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(93941, 99642, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c02.110:\n",
      "\t{'chrom': 'LinJ.02', 'start': 93941, 'end': 102762, 'strand': '-', 'gene_id': 'LINF_020007000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'phosphoglycan_beta_1-3_galactosyltransferase', 'interval': Interval(93941, 102762, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 93941, 'end': 102762, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_02T0007000', 'parent_id': 'LINF_020007000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'phosphoglycan_beta_1-3_galactosyltransferase', 'interval': Interval(93941, 102762, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 93941, 'end': 99642, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_02T0007000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(93941, 99642, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c02.120:\n",
      "\t{'chrom': 'LinJ.02', 'start': 93941, 'end': 102762, 'strand': '-', 'gene_id': 'LINF_020007000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'phosphoglycan_beta_1-3_galactosyltransferase', 'interval': Interval(93941, 102762, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 93941, 'end': 102762, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_02T0007000', 'parent_id': 'LINF_020007000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'phosphoglycan_beta_1-3_galactosyltransferase', 'interval': Interval(93941, 102762, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 93941, 'end': 99642, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_02T0007000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(93941, 99642, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c02.130B:\n",
      "\t{'chrom': 'LinJ.02', 'start': 137296, 'end': 139716, 'strand': '-', 'gene_id': 'LINF_020007950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(137296, 139716, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 137296, 'end': 139716, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_02T0007950', 'parent_id': 'LINF_020007950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(137296, 139716, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c02.140:\n",
      "==================================================\n",
      "rejected_noCDS_c02.150:\n",
      "==================================================\n",
      "rejected_noCDS_c02.160B:\n",
      "\t{'chrom': 'LinJ.02', 'start': 196594, 'end': 201033, 'strand': '-', 'gene_id': 'LINF_020009400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': 'unknown', 'notes': 'Transmembrane_amino_acid_transporter_protein_-_putative', 'interval': Interval(196594, 201033, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 196594, 'end': 201033, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_02T0009400', 'parent_id': 'LINF_020009400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Transmembrane_amino_acid_transporter_protein_-_putative', 'interval': Interval(196594, 201033, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 196594, 'end': 199091, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_02T0009400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(196594, 199091, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c02.170:\n",
      "==================================================\n",
      "rejected_noCDS_c03.180:\n",
      "==================================================\n",
      "rejected_noCDS_c03.190:\n",
      "\t{'chrom': 'LinJ.03', 'start': 235179, 'end': 238900, 'strand': '+', 'gene_id': 'LINF_030011300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(235179, 238900, closed='both')}\n",
      "\t{'chrom': 'LinJ.03', 'start': 235179, 'end': 238900, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_03T0011300', 'parent_id': 'LINF_030011300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(235179, 238900, closed='both')}\n",
      "\t{'chrom': 'LinJ.03', 'start': 237321, 'end': 238900, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_03T0011300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(237321, 238900, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c04.200C:\n",
      "==================================================\n",
      "rejected_noCDS_c04.210C:\n",
      "==================================================\n",
      "rejected_noCDS_c04.220:\n",
      "\t{'chrom': 'LinJ.04', 'start': 139396, 'end': 142427, 'strand': '-', 'gene_id': 'LINF_040009200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Tetratricopeptide_repeat-containing_protein', 'interval': Interval(139396, 142427, closed='both')}\n",
      "\t{'chrom': 'LinJ.04', 'start': 139396, 'end': 142427, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_04T0009200', 'parent_id': 'LINF_040009200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Tetratricopeptide_repeat-containing_protein', 'interval': Interval(139396, 142427, closed='both')}\n",
      "\t{'chrom': 'LinJ.04', 'start': 142101, 'end': 142427, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_04T0009200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(142101, 142427, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c04.230:\n",
      "==================================================\n",
      "rejected_noCDS_c04.240C:\n",
      "\t{'chrom': 'LinJ.04', 'start': 208282, 'end': 210473, 'strand': '-', 'gene_id': 'LINF_040010300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(208282, 210473, closed='both')}\n",
      "\t{'chrom': 'LinJ.04', 'start': 208282, 'end': 210473, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_04T0010300', 'parent_id': 'LINF_040010300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(208282, 210473, closed='both')}\n",
      "\t{'chrom': 'LinJ.04', 'start': 208282, 'end': 209330, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_04T0010300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(208282, 209330, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c04.250:\n",
      "\t{'chrom': 'LinJ.04', 'start': 331044, 'end': 338857, 'strand': '-', 'gene_id': 'LINF_040013700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(331044, 338857, closed='both')}\n",
      "\t{'chrom': 'LinJ.04', 'start': 331044, 'end': 338857, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_04T0013700', 'parent_id': 'LINF_040013700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(331044, 338857, closed='both')}\n",
      "\t{'chrom': 'LinJ.04', 'start': 331044, 'end': 334299, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_04T0013700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(331044, 334299, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c04.260:\n",
      "\t{'chrom': 'LinJ.04', 'start': 331044, 'end': 338857, 'strand': '-', 'gene_id': 'LINF_040013700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(331044, 338857, closed='both')}\n",
      "\t{'chrom': 'LinJ.04', 'start': 331044, 'end': 338857, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_04T0013700', 'parent_id': 'LINF_040013700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(331044, 338857, closed='both')}\n",
      "\t{'chrom': 'LinJ.04', 'start': 331044, 'end': 334299, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_04T0013700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(331044, 334299, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c04.270:\n",
      "\t{'chrom': 'LinJ.04', 'start': 437650, 'end': 446664, 'strand': '-', 'gene_id': 'LINF_040017000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(437650, 446664, closed='both')}\n",
      "\t{'chrom': 'LinJ.04', 'start': 437650, 'end': 446664, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_04T0017000', 'parent_id': 'LINF_040017000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(437650, 446664, closed='both')}\n",
      "\t{'chrom': 'LinJ.04', 'start': 437650, 'end': 438600, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_04T0017000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(437650, 438600, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c05.280:\n",
      "==================================================\n",
      "rejected_noCDS_c05.290:\n",
      "\t{'chrom': 'LinJ.05', 'start': 36373, 'end': 40873, 'strand': '+', 'gene_id': 'LINF_050006200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_kinase', 'interval': Interval(36373, 40873, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 36373, 'end': 40873, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_05T0006200', 'parent_id': 'LINF_050006200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_kinase', 'interval': Interval(36373, 40873, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 38529, 'end': 40873, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_05T0006200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(38529, 40873, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c05.300D:\n",
      "==================================================\n",
      "rejected_noCDS_c05.310:\n",
      "\t{'chrom': 'LinJ.05', 'start': 66289, 'end': 73912, 'strand': '+', 'gene_id': 'LINF_050007100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'dual_specificity_phosphatase-like_protein', 'interval': Interval(66289, 73912, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 66289, 'end': 73912, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_05T0007100', 'parent_id': 'LINF_050007100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'dual_specificity_phosphatase-like_protein', 'interval': Interval(66289, 73912, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 69379, 'end': 73912, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_05T0007100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(69379, 73912, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c05.320D:\n",
      "==================================================\n",
      "rejected_noCDS_c05.330E:\n",
      "\t{'chrom': 'LinJ.05', 'start': 130378, 'end': 134939, 'strand': '+', 'gene_id': 'LINF_050009000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'SMC_(Structural_maintenance_of_chromosomes)_protein', 'interval': Interval(130378, 134939, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 130378, 'end': 134939, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_05T0009000', 'parent_id': 'LINF_050009000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'SMC_(Structural_maintenance_of_chromosomes)_protein', 'interval': Interval(130378, 134939, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 134289, 'end': 134939, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_05T0009000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(134289, 134939, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c05.340E:\n",
      "\t{'chrom': 'LinJ.05', 'start': 144533, 'end': 148377, 'strand': '+', 'gene_id': 'LINF_050009400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(144533, 148377, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 144533, 'end': 148377, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_05T0009400', 'parent_id': 'LINF_050009400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(144533, 148377, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 147220, 'end': 148377, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_05T0009400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(147220, 148377, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c05.350:\n",
      "==================================================\n",
      "rejected_noCDS_c05.360F:\n",
      "==================================================\n",
      "rejected_noCDS_c06.370:\n",
      "\t{'chrom': 'LinJ.06', 'start': 6602, 'end': 7870, 'strand': '-', 'gene_id': 'LINF_060005150', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(6602, 7870, closed='both')}\n",
      "\t{'chrom': 'LinJ.06', 'start': 6602, 'end': 7870, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_06T0005150', 'parent_id': 'LINF_060005150', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(6602, 7870, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c06.380:\n",
      "==================================================\n",
      "rejected_noCDS_c06.390:\n",
      "\t{'chrom': 'LinJ.06', 'start': 101708, 'end': 107301, 'strand': '-', 'gene_id': 'LINF_060008400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'pteridine_transporter_(truncated)_-_putative', 'interval': Interval(101708, 107301, closed='both')}\n",
      "\t{'chrom': 'LinJ.06', 'start': 101708, 'end': 107301, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_06T0008400', 'parent_id': 'LINF_060008400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'pteridine_transporter_(truncated)_-_putative', 'interval': Interval(101708, 107301, closed='both')}\n",
      "\t{'chrom': 'LinJ.06', 'start': 101708, 'end': 105155, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_06T0008400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(101708, 105155, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c06.400:\n",
      "==================================================\n",
      "rejected_noCDS_c06.410:\n",
      "\t{'chrom': 'LinJ.06', 'start': 402727, 'end': 406088, 'strand': '+', 'gene_id': 'LINF_060014900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(402727, 406088, closed='both')}\n",
      "\t{'chrom': 'LinJ.06', 'start': 402727, 'end': 406088, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_06T0014900', 'parent_id': 'LINF_060014900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(402727, 406088, closed='both')}\n",
      "\t{'chrom': 'LinJ.06', 'start': 404237, 'end': 406088, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_06T0014900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(404237, 406088, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c07.420:\n",
      "\t{'chrom': 'LinJ.07', 'start': 164694, 'end': 173256, 'strand': '-', 'gene_id': 'LINF_070009100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ring_finger_domain/Zinc_finger_-_C3HC4_type_(RING_finger)_containing_protein', 'interval': Interval(164694, 173256, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 164694, 'end': 173256, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_07T0009100', 'parent_id': 'LINF_070009100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ring_finger_domain/Zinc_finger_-_C3HC4_type_(RING_finger)_containing_protein', 'interval': Interval(164694, 173256, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 164694, 'end': 169404, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_07T0009100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(164694, 169404, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c07.430G:\n",
      "==================================================\n",
      "rejected_noCDS_c07.440H:\n",
      "\t{'chrom': 'LinJ.07', 'start': 211951, 'end': 214227, 'strand': '+', 'gene_id': 'LINF_070010160', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(211951, 214227, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 211951, 'end': 214227, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_07T0010160', 'parent_id': 'LINF_070010160', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(211951, 214227, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c07.450G:\n",
      "\t{'chrom': 'LinJ.07', 'start': 251481, 'end': 252207, 'strand': '+', 'gene_id': 'LINF_070010950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(251481, 252207, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 251481, 'end': 252207, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_07T0010950', 'parent_id': 'LINF_070010950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(251481, 252207, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c07.460G:\n",
      "\t{'chrom': 'LinJ.07', 'start': 297019, 'end': 304430, 'strand': '+', 'gene_id': 'LINF_070012100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'vacuolar-type_Ca2+-ATPase_-_putative', 'interval': Interval(297019, 304430, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 297019, 'end': 304430, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_07T0012100-150', 'parent_id': 'LINF_070012100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'vacuolar-type_Ca2+-ATPase_-_putative', 'interval': Interval(297019, 304430, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 300978, 'end': 304430, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_07T0012100-150', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(300978, 304430, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c07.470H:\n",
      "\t{'chrom': 'LinJ.07', 'start': 297019, 'end': 304430, 'strand': '+', 'gene_id': 'LINF_070012100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'vacuolar-type_Ca2+-ATPase_-_putative', 'interval': Interval(297019, 304430, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 297019, 'end': 304430, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_07T0012100-150', 'parent_id': 'LINF_070012100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'vacuolar-type_Ca2+-ATPase_-_putative', 'interval': Interval(297019, 304430, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 300978, 'end': 304430, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_07T0012100-150', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(300978, 304430, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c07.480:\n",
      "\t{'chrom': 'LinJ.07', 'start': 304431, 'end': 306253, 'strand': '+', 'gene_id': 'LINF_070012175', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(304431, 306253, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 304431, 'end': 306253, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_07T0012175', 'parent_id': 'LINF_070012175', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(304431, 306253, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c07.490:\n",
      "\t{'chrom': 'LinJ.07', 'start': 334632, 'end': 339908, 'strand': '+', 'gene_id': 'LINF_070013000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'malonyl-coa_decarboxylase-like_protein', 'interval': Interval(334632, 339908, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 334632, 'end': 339908, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_07T0013000', 'parent_id': 'LINF_070013000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'malonyl-coa_decarboxylase-like_protein', 'interval': Interval(334632, 339908, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 337534, 'end': 339908, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_07T0013000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(337534, 339908, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c07.500:\n",
      "\t{'chrom': 'LinJ.07', 'start': 334632, 'end': 339908, 'strand': '+', 'gene_id': 'LINF_070013000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'malonyl-coa_decarboxylase-like_protein', 'interval': Interval(334632, 339908, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 334632, 'end': 339908, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_07T0013000', 'parent_id': 'LINF_070013000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'malonyl-coa_decarboxylase-like_protein', 'interval': Interval(334632, 339908, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 337534, 'end': 339908, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_07T0013000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(337534, 339908, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c07.510:\n",
      "==================================================\n",
      "rejected_noCDS_c07.520:\n",
      "\t{'chrom': 'LinJ.07', 'start': 568971, 'end': 570497, 'strand': '+', 'gene_id': 'LINF_070018050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(568971, 570497, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 568971, 'end': 570497, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_07T0018050', 'parent_id': 'LINF_070018050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(568971, 570497, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.530:\n",
      "\t{'chrom': 'LinJ.08', 'start': 55028, 'end': 67549, 'strand': '+', 'gene_id': 'LINF_080006700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(55028, 67549, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 55028, 'end': 67549, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0006700', 'parent_id': 'LINF_080006700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(55028, 67549, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 62506, 'end': 67549, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_08T0006700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(62506, 67549, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.540:\n",
      "\t{'chrom': 'LinJ.08', 'start': 74182, 'end': 77746, 'strand': '+', 'gene_id': 'LINF_080007000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'inositol_phosphosphingolipid_phospholipase_C-Like|ISCL', 'interval': Interval(74182, 77746, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 74182, 'end': 77746, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0007000', 'parent_id': 'LINF_080007000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'inositol_phosphosphingolipid_phospholipase_C-Like|ISCL', 'interval': Interval(74182, 77746, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 76652, 'end': 77746, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_08T0007000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(76652, 77746, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.550:\n",
      "==================================================\n",
      "rejected_noCDS_c08.560I:\n",
      "==================================================\n",
      "rejected_noCDS_c08.570J:\n",
      "\t{'chrom': 'LinJ.08', 'start': 298159, 'end': 299531, 'strand': '+', 'gene_id': 'LINF_080012050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(298159, 299531, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 298159, 'end': 299531, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012050', 'parent_id': 'LINF_080012050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(298159, 299531, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.580J:\n",
      "\t{'chrom': 'LinJ.08', 'start': 298159, 'end': 299531, 'strand': '+', 'gene_id': 'LINF_080012050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(298159, 299531, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 298159, 'end': 299531, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012050', 'parent_id': 'LINF_080012050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(298159, 299531, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.590I:\n",
      "==================================================\n",
      "rejected_noCDS_c08.600J:\n",
      "\t{'chrom': 'LinJ.08', 'start': 302787, 'end': 304302, 'strand': '+', 'gene_id': 'LINF_080012150', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(302787, 304302, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 302787, 'end': 304302, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012150', 'parent_id': 'LINF_080012150', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(302787, 304302, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.610J:\n",
      "\t{'chrom': 'LinJ.08', 'start': 302787, 'end': 304302, 'strand': '+', 'gene_id': 'LINF_080012150', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(302787, 304302, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 302787, 'end': 304302, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012150', 'parent_id': 'LINF_080012150', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(302787, 304302, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.620I:\n",
      "==================================================\n",
      "rejected_noCDS_c08.630J:\n",
      "\t{'chrom': 'LinJ.08', 'start': 307412, 'end': 308886, 'strand': '+', 'gene_id': 'LINF_080012250', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(307412, 308886, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 307412, 'end': 308886, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012250', 'parent_id': 'LINF_080012250', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(307412, 308886, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.640I:\n",
      "==================================================\n",
      "rejected_noCDS_c08.650J:\n",
      "\t{'chrom': 'LinJ.08', 'start': 311996, 'end': 313821, 'strand': '+', 'gene_id': 'LINF_080012350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(311996, 313821, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 311996, 'end': 313821, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012350', 'parent_id': 'LINF_080012350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(311996, 313821, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.660J:\n",
      "\t{'chrom': 'LinJ.08', 'start': 311996, 'end': 313821, 'strand': '+', 'gene_id': 'LINF_080012350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(311996, 313821, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 311996, 'end': 313821, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012350', 'parent_id': 'LINF_080012350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(311996, 313821, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.670:\n",
      "==================================================\n",
      "rejected_noCDS_c08.680K:\n",
      "==================================================\n",
      "rejected_noCDS_c08.690L:\n",
      "\t{'chrom': 'LinJ.08', 'start': 330313, 'end': 332015, 'strand': '+', 'gene_id': 'LINF_080012825', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(330313, 332015, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 330313, 'end': 332015, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012825', 'parent_id': 'LINF_080012825', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(330313, 332015, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.700M:\n",
      "\t{'chrom': 'LinJ.08', 'start': 330313, 'end': 332015, 'strand': '+', 'gene_id': 'LINF_080012825', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(330313, 332015, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 330313, 'end': 332015, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012825', 'parent_id': 'LINF_080012825', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(330313, 332015, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.710K:\n",
      "==================================================\n",
      "rejected_noCDS_c08.720N:\n",
      "==================================================\n",
      "rejected_noCDS_c08.730:\n",
      "\t{'chrom': 'LinJ.08', 'start': 332481, 'end': 335118, 'strand': '+', 'gene_id': 'LINF_080012850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(332481, 335118, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 332481, 'end': 335118, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012850', 'parent_id': 'LINF_080012850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(332481, 335118, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.740L:\n",
      "\t{'chrom': 'LinJ.08', 'start': 335486, 'end': 337341, 'strand': '+', 'gene_id': 'LINF_080012875', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(335486, 337341, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 335486, 'end': 337341, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012875', 'parent_id': 'LINF_080012875', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(335486, 337341, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.750M:\n",
      "\t{'chrom': 'LinJ.08', 'start': 335486, 'end': 337341, 'strand': '+', 'gene_id': 'LINF_080012875', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(335486, 337341, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 335486, 'end': 337341, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012875', 'parent_id': 'LINF_080012875', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(335486, 337341, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.760K:\n",
      "==================================================\n",
      "rejected_noCDS_c08.770N:\n",
      "==================================================\n",
      "rejected_noCDS_c08.780L:\n",
      "\t{'chrom': 'LinJ.08', 'start': 340660, 'end': 342358, 'strand': '+', 'gene_id': 'LINF_080012925', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(340660, 342358, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 340660, 'end': 342358, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012925', 'parent_id': 'LINF_080012925', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(340660, 342358, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.790M:\n",
      "\t{'chrom': 'LinJ.08', 'start': 340660, 'end': 342358, 'strand': '+', 'gene_id': 'LINF_080012925', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(340660, 342358, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 340660, 'end': 342358, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012925', 'parent_id': 'LINF_080012925', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(340660, 342358, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.800K:\n",
      "==================================================\n",
      "rejected_noCDS_c08.810L:\n",
      "\t{'chrom': 'LinJ.08', 'start': 345835, 'end': 348001, 'strand': '+', 'gene_id': 'LINF_080012975', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(345835, 348001, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 345835, 'end': 348001, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012975', 'parent_id': 'LINF_080012975', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(345835, 348001, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.820M:\n",
      "\t{'chrom': 'LinJ.08', 'start': 345835, 'end': 348001, 'strand': '+', 'gene_id': 'LINF_080012975', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(345835, 348001, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 345835, 'end': 348001, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012975', 'parent_id': 'LINF_080012975', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(345835, 348001, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.830K:\n",
      "\t{'chrom': 'LinJ.08', 'start': 345835, 'end': 348001, 'strand': '+', 'gene_id': 'LINF_080012975', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(345835, 348001, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 345835, 'end': 348001, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012975', 'parent_id': 'LINF_080012975', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(345835, 348001, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.840:\n",
      "\t{'chrom': 'LinJ.08', 'start': 357622, 'end': 360413, 'strand': '-', 'gene_id': 'LINF_080013300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'mitochondrial_DNA_polymerase_beta', 'interval': Interval(357622, 360413, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 357622, 'end': 360413, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_08T0013300', 'parent_id': 'LINF_080013300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'mitochondrial_DNA_polymerase_beta', 'interval': Interval(357622, 360413, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 357622, 'end': 358794, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_08T0013300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(357622, 358794, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.850:\n",
      "==================================================\n",
      "rejected_noCDS_c08.860:\n",
      "==================================================\n",
      "rejected_noCDS_c08.870O:\n",
      "\t{'chrom': 'LinJ.08', 'start': 511823, 'end': 513727, 'strand': '+', 'gene_id': 'LINF_080017730', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(511823, 513727, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 511823, 'end': 513727, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0017730', 'parent_id': 'LINF_080017730', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(511823, 513727, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c09.880:\n",
      "\t{'chrom': 'LinJ.09', 'start': 34342, 'end': 47348, 'strand': '+', 'gene_id': 'LINF_090006000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RNA-binding_protein_5-like_protein', 'interval': Interval(34342, 47348, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 34342, 'end': 47348, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_09T0006000', 'parent_id': 'LINF_090006000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RNA-binding_protein_5-like_protein', 'interval': Interval(34342, 47348, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 35729, 'end': 47348, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_09T0006000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(35729, 47348, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c09.890P:\n",
      "\t{'chrom': 'LinJ.09', 'start': 208973, 'end': 214334, 'strand': '+', 'gene_id': 'LINF_090011100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Flagellum_attachment_zone_protein_3|FAZ3', 'interval': Interval(208973, 214334, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 208973, 'end': 214334, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_09T0011100', 'parent_id': 'LINF_090011100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Flagellum_attachment_zone_protein_3|FAZ3', 'interval': Interval(208973, 214334, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 211395, 'end': 214334, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_09T0011100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(211395, 214334, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c09.900P:\n",
      "\t{'chrom': 'LinJ.09', 'start': 214789, 'end': 219664, 'strand': '+', 'gene_id': 'LINF_090011200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'leucine-rich_repeat_protein', 'interval': Interval(214789, 219664, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 214789, 'end': 219664, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_09T0011200', 'parent_id': 'LINF_090011200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'leucine-rich_repeat_protein', 'interval': Interval(214789, 219664, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 218236, 'end': 219664, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_09T0011200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(218236, 219664, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c09.910P:\n",
      "\t{'chrom': 'LinJ.09', 'start': 228107, 'end': 229683, 'strand': '+', 'gene_id': 'LINF_090011500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'histidine_phosphatase', 'interval': Interval(228107, 229683, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 228107, 'end': 229683, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_09T0011500', 'parent_id': 'LINF_090011500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'histidine_phosphatase', 'interval': Interval(228107, 229683, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 229153, 'end': 229683, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_09T0011500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(229153, 229683, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c09.920:\n",
      "\t{'chrom': 'LinJ.09', 'start': 414998, 'end': 417392, 'strand': '+', 'gene_id': 'LINF_090017000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(414998, 417392, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 414998, 'end': 417392, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_09T0017000', 'parent_id': 'LINF_090017000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(414998, 417392, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 415420, 'end': 417392, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_09T0017000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(415420, 417392, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c09.930:\n",
      "\t{'chrom': 'LinJ.09', 'start': 497511, 'end': 498961, 'strand': '-', 'gene_id': 'LINF_090019850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(497511, 498961, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 497511, 'end': 498961, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_09T0019850', 'parent_id': 'LINF_090019850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(497511, 498961, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c09.940:\n",
      "\t{'chrom': 'LinJ.09', 'start': 497511, 'end': 498961, 'strand': '-', 'gene_id': 'LINF_090019850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(497511, 498961, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 497511, 'end': 498961, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_09T0019850', 'parent_id': 'LINF_090019850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(497511, 498961, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c09.950:\n",
      "\t{'chrom': 'LinJ.09', 'start': 505372, 'end': 507986, 'strand': '-', 'gene_id': 'LINF_090020200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-__conserved', 'interval': Interval(505372, 507986, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 505372, 'end': 507986, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_09T0020200', 'parent_id': 'LINF_090020200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-__conserved', 'interval': Interval(505372, 507986, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 505372, 'end': 506812, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_09T0020200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(505372, 506812, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c09.960:\n",
      "\t{'chrom': 'LinJ.09', 'start': 521466, 'end': 525002, 'strand': '-', 'gene_id': 'LINF_090020800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-__conserved', 'interval': Interval(521466, 525002, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 521466, 'end': 525002, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_09T0020800', 'parent_id': 'LINF_090020800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-__conserved', 'interval': Interval(521466, 525002, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 521466, 'end': 522445, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_09T0020800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(521466, 522445, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c09.970:\n",
      "\t{'chrom': 'LinJ.09', 'start': 526420, 'end': 528154, 'strand': '-', 'gene_id': 'LINF_090021000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-__conserved', 'interval': Interval(526420, 528154, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 526420, 'end': 528154, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_09T0021000', 'parent_id': 'LINF_090021000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-__conserved', 'interval': Interval(526420, 528154, closed='both')}\n",
      "\t{'chrom': 'LinJ.09', 'start': 526420, 'end': 527451, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_09T0021000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(526420, 527451, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c10.980O:\n",
      "==================================================\n",
      "rejected_noCDS_c10.990:\n",
      "\t{'chrom': 'LinJ.10', 'start': 33221, 'end': 35701, 'strand': '+', 'gene_id': 'LINF_100005700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(33221, 35701, closed='both')}\n",
      "\t{'chrom': 'LinJ.10', 'start': 33221, 'end': 35701, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_10T0005700', 'parent_id': 'LINF_100005700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(33221, 35701, closed='both')}\n",
      "\t{'chrom': 'LinJ.10', 'start': 34440, 'end': 35701, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_10T0005700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(34440, 35701, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c10.1000:\n",
      "\t{'chrom': 'LinJ.10', 'start': 71683, 'end': 73187, 'strand': '+', 'gene_id': 'LINF_100006550', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(71683, 73187, closed='both')}\n",
      "\t{'chrom': 'LinJ.10', 'start': 71683, 'end': 73187, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_10T0006550', 'parent_id': 'LINF_100006550', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(71683, 73187, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c10.1010:\n",
      "==================================================\n",
      "rejected_noCDS_c10.1020:\n",
      "\t{'chrom': 'LinJ.10', 'start': 215790, 'end': 217677, 'strand': '+', 'gene_id': 'LINF_100010050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(215790, 217677, closed='both')}\n",
      "\t{'chrom': 'LinJ.10', 'start': 215790, 'end': 217677, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_10T0010050', 'parent_id': 'LINF_100010050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(215790, 217677, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c10.1030:\n",
      "\t{'chrom': 'LinJ.10', 'start': 215790, 'end': 217677, 'strand': '+', 'gene_id': 'LINF_100010050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(215790, 217677, closed='both')}\n",
      "\t{'chrom': 'LinJ.10', 'start': 215790, 'end': 217677, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_10T0010050', 'parent_id': 'LINF_100010050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(215790, 217677, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c10.1040:\n",
      "\t{'chrom': 'LinJ.10', 'start': 266644, 'end': 271932, 'strand': '+', 'gene_id': 'LINF_100011600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'mitogen-activated_protein_kinase_3|MPK3|MAPK3', 'interval': Interval(266644, 271932, closed='both')}\n",
      "\t{'chrom': 'LinJ.10', 'start': 266644, 'end': 271932, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_10T0011600', 'parent_id': 'LINF_100011600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'mitogen-activated_protein_kinase_3|MPK3|MAPK3', 'interval': Interval(266644, 271932, closed='both')}\n",
      "\t{'chrom': 'LinJ.10', 'start': 269486, 'end': 271932, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_10T0011600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(269486, 271932, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c10.1050:\n",
      "\t{'chrom': 'LinJ.10', 'start': 266644, 'end': 271932, 'strand': '+', 'gene_id': 'LINF_100011600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'mitogen-activated_protein_kinase_3|MPK3|MAPK3', 'interval': Interval(266644, 271932, closed='both')}\n",
      "\t{'chrom': 'LinJ.10', 'start': 266644, 'end': 271932, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_10T0011600', 'parent_id': 'LINF_100011600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'mitogen-activated_protein_kinase_3|MPK3|MAPK3', 'interval': Interval(266644, 271932, closed='both')}\n",
      "\t{'chrom': 'LinJ.10', 'start': 269486, 'end': 271932, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_10T0011600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(269486, 271932, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c10.1060:\n",
      "\t{'chrom': 'LinJ.10', 'start': 382159, 'end': 391117, 'strand': '-', 'gene_id': 'LINF_100014200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(382159, 391117, closed='both')}\n",
      "\t{'chrom': 'LinJ.10', 'start': 382159, 'end': 391117, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_10T0014200', 'parent_id': 'LINF_100014200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(382159, 391117, closed='both')}\n",
      "\t{'chrom': 'LinJ.10', 'start': 382159, 'end': 384377, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_10T0014200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(382159, 384377, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c11.1070:\n",
      "==================================================\n",
      "rejected_noCDS_c11.1080:\n",
      "==================================================\n",
      "rejected_noCDS_c11.1090:\n",
      "\t{'chrom': 'LinJ.11', 'start': 205593, 'end': 211090, 'strand': '+', 'gene_id': 'LINF_110011500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'cytochrome_b5', 'interval': Interval(205593, 211090, closed='both')}\n",
      "\t{'chrom': 'LinJ.11', 'start': 205593, 'end': 211090, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_11T0011500', 'parent_id': 'LINF_110011500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'cytochrome_b5', 'interval': Interval(205593, 211090, closed='both')}\n",
      "\t{'chrom': 'LinJ.11', 'start': 206757, 'end': 211090, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_11T0011500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(206757, 211090, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1100:\n",
      "\t{'chrom': 'LinJ.12', 'start': 48004, 'end': 51759, 'strand': '+', 'gene_id': 'LINF_120006700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ALG9|alpha-1,2-mannosyltransferase', 'interval': Interval(48004, 51759, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 48004, 'end': 51759, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0006700', 'parent_id': 'LINF_120006700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ALG9|alpha-1,2-mannosyltransferase', 'interval': Interval(48004, 51759, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 50529, 'end': 51759, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0006700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(50529, 51759, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1110:\n",
      "\t{'chrom': 'LinJ.12', 'start': 57672, 'end': 63572, 'strand': '+', 'gene_id': 'LINF_120007000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ubiquitin_hydrolase_-_putative', 'interval': Interval(57672, 63572, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 57672, 'end': 63572, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0007000', 'parent_id': 'LINF_120007000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ubiquitin_hydrolase_-_putative', 'interval': Interval(57672, 63572, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 62813, 'end': 63572, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0007000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(62813, 63572, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1120:\n",
      "==================================================\n",
      "rejected_noCDS_c12.1130Q:\n",
      "\t{'chrom': 'LinJ.12', 'start': 229328, 'end': 233569, 'strand': '-', 'gene_id': 'LINF_120010050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(229328, 233569, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 229328, 'end': 233569, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_12T0010050', 'parent_id': 'LINF_120010050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(229328, 233569, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1140R:\n",
      "\t{'chrom': 'LinJ.12', 'start': 229328, 'end': 233569, 'strand': '-', 'gene_id': 'LINF_120010050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(229328, 233569, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 229328, 'end': 233569, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_12T0010050', 'parent_id': 'LINF_120010050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(229328, 233569, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1150Q:\n",
      "\t{'chrom': 'LinJ.12', 'start': 280128, 'end': 284643, 'strand': '-', 'gene_id': 'LINF_120010425', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(280128, 284643, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 280128, 'end': 284643, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_12T0010425', 'parent_id': 'LINF_120010425', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(280128, 284643, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1160R:\n",
      "\t{'chrom': 'LinJ.12', 'start': 280128, 'end': 284643, 'strand': '-', 'gene_id': 'LINF_120010425', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(280128, 284643, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 280128, 'end': 284643, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_12T0010425', 'parent_id': 'LINF_120010425', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(280128, 284643, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1170:\n",
      "==================================================\n",
      "rejected_noCDS_c12.1180S:\n",
      "==================================================\n",
      "rejected_noCDS_c12.1190:\n",
      "==================================================\n",
      "rejected_noCDS_c12.1200S:\n",
      "==================================================\n",
      "rejected_noCDS_c12.1210S:\n",
      "==================================================\n",
      "rejected_noCDS_c12.1220S:\n",
      "==================================================\n",
      "rejected_noCDS_c12.1230:\n",
      "\t{'chrom': 'LinJ.12', 'start': 417691, 'end': 421308, 'strand': '+', 'gene_id': 'LINF_120013560', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(417691, 421308, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 417691, 'end': 421308, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0013560', 'parent_id': 'LINF_120013560', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(417691, 421308, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1240S:\n",
      "==================================================\n",
      "rejected_noCDS_c12.1250S:\n",
      "==================================================\n",
      "rejected_noCDS_c12.1260:\n",
      "\t{'chrom': 'LinJ.12', 'start': 466522, 'end': 469621, 'strand': '+', 'gene_id': 'LINF_120015050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(466522, 469621, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 466522, 'end': 469621, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0015050', 'parent_id': 'LINF_120015050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(466522, 469621, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1270:\n",
      "\t{'chrom': 'LinJ.12', 'start': 466522, 'end': 469621, 'strand': '+', 'gene_id': 'LINF_120015050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(466522, 469621, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 466522, 'end': 469621, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0015050', 'parent_id': 'LINF_120015050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(466522, 469621, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1280:\n",
      "\t{'chrom': 'LinJ.12', 'start': 509381, 'end': 514021, 'strand': '+', 'gene_id': 'LINF_120015800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Putative_integral_membrane_protein_conserved_region_(DUF2404)_-_putative', 'interval': Interval(509381, 514021, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 509381, 'end': 514021, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0015800', 'parent_id': 'LINF_120015800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Putative_integral_membrane_protein_conserved_region_(DUF2404)_-_putative', 'interval': Interval(509381, 514021, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 512950, 'end': 514021, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0015800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(512950, 514021, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1290T:\n",
      "\t{'chrom': 'LinJ.12', 'start': 528652, 'end': 535332, 'strand': '+', 'gene_id': 'LINF_120016200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(528652, 535332, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 528652, 'end': 535332, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0016200', 'parent_id': 'LINF_120016200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(528652, 535332, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 530614, 'end': 535332, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0016200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(530614, 535332, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1300U:\n",
      "\t{'chrom': 'LinJ.12', 'start': 528652, 'end': 535332, 'strand': '+', 'gene_id': 'LINF_120016200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(528652, 535332, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 528652, 'end': 535332, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0016200', 'parent_id': 'LINF_120016200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(528652, 535332, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 530614, 'end': 535332, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0016200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(530614, 535332, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1310:\n",
      "\t{'chrom': 'LinJ.12', 'start': 528652, 'end': 535332, 'strand': '+', 'gene_id': 'LINF_120016200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(528652, 535332, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 528652, 'end': 535332, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0016200', 'parent_id': 'LINF_120016200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(528652, 535332, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 530614, 'end': 535332, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0016200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(530614, 535332, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1320U:\n",
      "\t{'chrom': 'LinJ.12', 'start': 545453, 'end': 547275, 'strand': '+', 'gene_id': 'LINF_120016450', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(545453, 547275, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 545453, 'end': 547275, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0016450', 'parent_id': 'LINF_120016450', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(545453, 547275, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1330:\n",
      "\t{'chrom': 'LinJ.12', 'start': 581574, 'end': 586501, 'strand': '+', 'gene_id': 'LINF_120017600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'alpha/beta_hydrolase', 'interval': Interval(581574, 586501, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 581574, 'end': 586501, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0017600', 'parent_id': 'LINF_120017600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'alpha/beta_hydrolase', 'interval': Interval(581574, 586501, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 583472, 'end': 586501, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0017600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(583472, 586501, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1340T:\n",
      "\t{'chrom': 'LinJ.12', 'start': 581574, 'end': 586501, 'strand': '+', 'gene_id': 'LINF_120017600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'alpha/beta_hydrolase', 'interval': Interval(581574, 586501, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 581574, 'end': 586501, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0017600', 'parent_id': 'LINF_120017600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'alpha/beta_hydrolase', 'interval': Interval(581574, 586501, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 583472, 'end': 586501, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0017600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(583472, 586501, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1350U:\n",
      "\t{'chrom': 'LinJ.12', 'start': 581574, 'end': 586501, 'strand': '+', 'gene_id': 'LINF_120017600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'alpha/beta_hydrolase', 'interval': Interval(581574, 586501, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 581574, 'end': 586501, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0017600', 'parent_id': 'LINF_120017600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'alpha/beta_hydrolase', 'interval': Interval(581574, 586501, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 583472, 'end': 586501, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0017600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(583472, 586501, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c13.1360:\n",
      "\t{'chrom': 'LinJ.13', 'start': 21177, 'end': 25140, 'strand': '-', 'gene_id': 'LINF_130005800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'carboxypeptidase|CP1', 'interval': Interval(21177, 25140, closed='both')}\n",
      "\t{'chrom': 'LinJ.13', 'start': 21177, 'end': 25140, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_13T0005800', 'parent_id': 'LINF_130005800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'carboxypeptidase|CP1', 'interval': Interval(21177, 25140, closed='both')}\n",
      "\t{'chrom': 'LinJ.13', 'start': 21177, 'end': 23361, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_13T0005800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(21177, 23361, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c13.1370:\n",
      "\t{'chrom': 'LinJ.13', 'start': 25141, 'end': 30558, 'strand': '-', 'gene_id': 'LINF_130005900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(25141, 30558, closed='both')}\n",
      "\t{'chrom': 'LinJ.13', 'start': 25141, 'end': 30558, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_13T0005900', 'parent_id': 'LINF_130005900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(25141, 30558, closed='both')}\n",
      "\t{'chrom': 'LinJ.13', 'start': 25141, 'end': 28875, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_13T0005900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(25141, 28875, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c13.1380:\n",
      "\t{'chrom': 'LinJ.13', 'start': 64984, 'end': 69230, 'strand': '-', 'gene_id': 'LINF_130007000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Triglyceride_lipase_-_putative', 'interval': Interval(64984, 69230, closed='both')}\n",
      "\t{'chrom': 'LinJ.13', 'start': 64984, 'end': 69230, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_13T0007000', 'parent_id': 'LINF_130007000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Triglyceride_lipase_-_putative', 'interval': Interval(64984, 69230, closed='both')}\n",
      "\t{'chrom': 'LinJ.13', 'start': 64984, 'end': 66322, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_13T0007000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(64984, 66322, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c13.1390:\n",
      "\t{'chrom': 'LinJ.13', 'start': 144368, 'end': 146861, 'strand': '+', 'gene_id': 'LINF_130009450', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(144368, 146861, closed='both')}\n",
      "\t{'chrom': 'LinJ.13', 'start': 144368, 'end': 146861, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_13T0009450', 'parent_id': 'LINF_130009450', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(144368, 146861, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c13.1400:\n",
      "==================================================\n",
      "rejected_noCDS_c14.1410:\n",
      "\t{'chrom': 'LinJ.14', 'start': 162580, 'end': 166213, 'strand': '+', 'gene_id': 'LINF_140010000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Cystathionine_beta-lyase', 'interval': Interval(162580, 166213, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 162580, 'end': 166213, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_14T0010000', 'parent_id': 'LINF_140010000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Cystathionine_beta-lyase', 'interval': Interval(162580, 166213, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 164234, 'end': 166213, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_14T0010000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(164234, 166213, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c14.1420:\n",
      "==================================================\n",
      "rejected_noCDS_c14.1430V:\n",
      "\t{'chrom': 'LinJ.14', 'start': 255152, 'end': 267139, 'strand': '-', 'gene_id': 'LINF_140012400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ketoacyl-CoA_synthase|Fatty_acid_elongase|ELO-3', 'interval': Interval(255152, 267139, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 255152, 'end': 267139, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_14T0012400', 'parent_id': 'LINF_140012400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ketoacyl-CoA_synthase|Fatty_acid_elongase|ELO-3', 'interval': Interval(255152, 267139, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 255152, 'end': 266070, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_14T0012400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(255152, 266070, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c14.1440V:\n",
      "\t{'chrom': 'LinJ.14', 'start': 255152, 'end': 267139, 'strand': '-', 'gene_id': 'LINF_140012400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ketoacyl-CoA_synthase|Fatty_acid_elongase|ELO-3', 'interval': Interval(255152, 267139, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 255152, 'end': 267139, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_14T0012400', 'parent_id': 'LINF_140012400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ketoacyl-CoA_synthase|Fatty_acid_elongase|ELO-3', 'interval': Interval(255152, 267139, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 255152, 'end': 266070, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_14T0012400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(255152, 266070, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c14.1450V:\n",
      "\t{'chrom': 'LinJ.14', 'start': 255152, 'end': 267139, 'strand': '-', 'gene_id': 'LINF_140012400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ketoacyl-CoA_synthase|Fatty_acid_elongase|ELO-3', 'interval': Interval(255152, 267139, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 255152, 'end': 267139, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_14T0012400', 'parent_id': 'LINF_140012400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ketoacyl-CoA_synthase|Fatty_acid_elongase|ELO-3', 'interval': Interval(255152, 267139, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 255152, 'end': 266070, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_14T0012400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(255152, 266070, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c14.1460V:\n",
      "\t{'chrom': 'LinJ.14', 'start': 255152, 'end': 267139, 'strand': '-', 'gene_id': 'LINF_140012400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ketoacyl-CoA_synthase|Fatty_acid_elongase|ELO-3', 'interval': Interval(255152, 267139, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 255152, 'end': 267139, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_14T0012400', 'parent_id': 'LINF_140012400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ketoacyl-CoA_synthase|Fatty_acid_elongase|ELO-3', 'interval': Interval(255152, 267139, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 255152, 'end': 266070, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_14T0012400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(255152, 266070, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c14.1470V:\n",
      "\t{'chrom': 'LinJ.14', 'start': 255152, 'end': 267139, 'strand': '-', 'gene_id': 'LINF_140012400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ketoacyl-CoA_synthase|Fatty_acid_elongase|ELO-3', 'interval': Interval(255152, 267139, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 255152, 'end': 267139, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_14T0012400', 'parent_id': 'LINF_140012400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ketoacyl-CoA_synthase|Fatty_acid_elongase|ELO-3', 'interval': Interval(255152, 267139, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 255152, 'end': 266070, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_14T0012400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(255152, 266070, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c14.1480:\n",
      "\t{'chrom': 'LinJ.14', 'start': 280945, 'end': 285205, 'strand': '-', 'gene_id': 'LINF_140012800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ketoacyl-CoA_synthase|fatty_acid_elongase|ELO-3', 'interval': Interval(280945, 285205, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 280945, 'end': 285205, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_14T0012800', 'parent_id': 'LINF_140012800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ketoacyl-CoA_synthase|fatty_acid_elongase|ELO-3', 'interval': Interval(280945, 285205, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 280945, 'end': 283797, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_14T0012800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(280945, 283797, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c14.1490V:\n",
      "\t{'chrom': 'LinJ.14', 'start': 285611, 'end': 287461, 'strand': '-', 'gene_id': 'LINF_140012850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(285611, 287461, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 285611, 'end': 287461, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_14T0012850', 'parent_id': 'LINF_140012850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(285611, 287461, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c14.1500:\n",
      "\t{'chrom': 'LinJ.14', 'start': 402446, 'end': 408212, 'strand': '-', 'gene_id': 'LINF_140015500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(402446, 408212, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 402446, 'end': 408212, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_14T0015500', 'parent_id': 'LINF_140015500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(402446, 408212, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 402446, 'end': 404022, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_14T0015500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(402446, 404022, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c14.1510:\n",
      "\t{'chrom': 'LinJ.14', 'start': 575947, 'end': 582294, 'strand': '+', 'gene_id': 'LINF_140019400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_kinase_(STE_family)', 'interval': Interval(575947, 582294, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 575947, 'end': 582294, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_14T0019400', 'parent_id': 'LINF_140019400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_kinase_(STE_family)', 'interval': Interval(575947, 582294, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 579893, 'end': 582294, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_14T0019400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(579893, 582294, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c14.1520:\n",
      "\t{'chrom': 'LinJ.14', 'start': 598416, 'end': 600364, 'strand': '+', 'gene_id': 'LINF_140019950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': 'unknown', 'notes': 'pteridine_transporter', 'interval': Interval(598416, 600364, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 598416, 'end': 600364, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_14T0019950', 'parent_id': 'LINF_140019950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'pteridine_transporter', 'interval': Interval(598416, 600364, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 599003, 'end': 599842, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_14T0019950', 'parent_id': 'LINF_14T0019950', 'gene': False, 'transcript': False, 'CDS': True, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(599003, 599842, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c14.1530:\n",
      "==================================================\n",
      "rejected_noCDS_c14.1540W:\n",
      "==================================================\n",
      "rejected_noCDS_c15.1550:\n",
      "\t{'chrom': 'LinJ.15', 'start': 38563, 'end': 40418, 'strand': '-', 'gene_id': 'LINF_150006150', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(38563, 40418, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 38563, 'end': 40418, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_15T0006150', 'parent_id': 'LINF_150006150', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(38563, 40418, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c15.1560:\n",
      "\t{'chrom': 'LinJ.15', 'start': 77396, 'end': 79634, 'strand': '-', 'gene_id': 'LINF_150007000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(77396, 79634, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 77396, 'end': 79634, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_15T0007000', 'parent_id': 'LINF_150007000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(77396, 79634, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 77396, 'end': 78628, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_15T0007000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(77396, 78628, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c15.1570:\n",
      "\t{'chrom': 'LinJ.15', 'start': 81381, 'end': 85480, 'strand': '-', 'gene_id': 'LINF_150007200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(81381, 85480, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 81381, 'end': 85480, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_15T0007200', 'parent_id': 'LINF_150007200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(81381, 85480, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 81381, 'end': 83746, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_15T0007200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(81381, 83746, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c15.1580:\n",
      "==================================================\n",
      "rejected_noCDS_c15.1590X:\n",
      "==================================================\n",
      "rejected_noCDS_c15.1600:\n",
      "\t{'chrom': 'LinJ.15', 'start': 286973, 'end': 288730, 'strand': '+', 'gene_id': 'LINF_150013750', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(286973, 288730, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 286973, 'end': 288730, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_15T0013750', 'parent_id': 'LINF_150013750', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(286973, 288730, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c15.1610X:\n",
      "==================================================\n",
      "rejected_noCDS_c15.1620:\n",
      "\t{'chrom': 'LinJ.15', 'start': 341184, 'end': 342366, 'strand': '+', 'gene_id': 'LINF_150015150', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(341184, 342366, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 341184, 'end': 342366, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_15T0015150', 'parent_id': 'LINF_150015150', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(341184, 342366, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c15.1630:\n",
      "\t{'chrom': 'LinJ.15', 'start': 353060, 'end': 362858, 'strand': '-', 'gene_id': 'LINF_150015400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATP-binding_cassette_protein_subfamily_A,_member_7|ABCA7', 'interval': Interval(353060, 362858, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 353060, 'end': 362858, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_15T0015400', 'parent_id': 'LINF_150015400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATP-binding_cassette_protein_subfamily_A,_member_7|ABCA7', 'interval': Interval(353060, 362858, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 353060, 'end': 354079, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_15T0015400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(353060, 354079, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c15.1640:\n",
      "\t{'chrom': 'LinJ.15', 'start': 353060, 'end': 362858, 'strand': '-', 'gene_id': 'LINF_150015400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATP-binding_cassette_protein_subfamily_A,_member_7|ABCA7', 'interval': Interval(353060, 362858, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 353060, 'end': 362858, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_15T0015400', 'parent_id': 'LINF_150015400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATP-binding_cassette_protein_subfamily_A,_member_7|ABCA7', 'interval': Interval(353060, 362858, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 353060, 'end': 354079, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_15T0015400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(353060, 354079, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c15.1650:\n",
      "\t{'chrom': 'LinJ.15', 'start': 403941, 'end': 408129, 'strand': '-', 'gene_id': 'LINF_150016600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(403941, 408129, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 403941, 'end': 408129, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_15T0016600', 'parent_id': 'LINF_150016600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(403941, 408129, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 403941, 'end': 405017, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_15T0016600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(403941, 405017, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c15.1660:\n",
      "\t{'chrom': 'LinJ.15', 'start': 516435, 'end': 518062, 'strand': '-', 'gene_id': 'LINF_150019850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(516435, 518062, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 516435, 'end': 518062, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_15T0019850', 'parent_id': 'LINF_150019850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(516435, 518062, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c15.1670:\n",
      "\t{'chrom': 'LinJ.15', 'start': 516435, 'end': 518062, 'strand': '-', 'gene_id': 'LINF_150019850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(516435, 518062, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 516435, 'end': 518062, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_15T0019850', 'parent_id': 'LINF_150019850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(516435, 518062, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c15.1680:\n",
      "\t{'chrom': 'LinJ.15', 'start': 533299, 'end': 538251, 'strand': '-', 'gene_id': 'LINF_150020400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'SET13', 'interval': Interval(533299, 538251, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 533299, 'end': 538251, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_15T0020400', 'parent_id': 'LINF_150020400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'SET13', 'interval': Interval(533299, 538251, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 533299, 'end': 535790, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_15T0020400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(533299, 535790, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c16.1690:\n",
      "\t{'chrom': 'LinJ.16', 'start': 39007, 'end': 40952, 'strand': '-', 'gene_id': 'LINF_160006600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(39007, 40952, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 39007, 'end': 40952, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_16T0006600', 'parent_id': 'LINF_160006600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(39007, 40952, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 39007, 'end': 40031, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_16T0006600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(39007, 40031, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c16.1700:\n",
      "\t{'chrom': 'LinJ.16', 'start': 79709, 'end': 81720, 'strand': '-', 'gene_id': 'LINF_160007600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_tyrosine_phosphatase_PRL-2', 'interval': Interval(79709, 81720, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 79709, 'end': 81720, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_16T0007600', 'parent_id': 'LINF_160007600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_tyrosine_phosphatase_PRL-2', 'interval': Interval(79709, 81720, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 79709, 'end': 80768, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_16T0007600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(79709, 80768, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c16.1710:\n",
      "\t{'chrom': 'LinJ.16', 'start': 119725, 'end': 123450, 'strand': '-', 'gene_id': 'LINF_160008700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(119725, 123450, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 119725, 'end': 123450, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_16T0008700', 'parent_id': 'LINF_160008700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(119725, 123450, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 119725, 'end': 121232, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_16T0008700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(119725, 121232, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c16.1720:\n",
      "==================================================\n",
      "rejected_noCDS_c16.1730:\n",
      "\t{'chrom': 'LinJ.16', 'start': 152179, 'end': 154521, 'strand': '-', 'gene_id': 'LINF_160009550', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(152179, 154521, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 152179, 'end': 154521, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_16T0009550', 'parent_id': 'LINF_160009550', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(152179, 154521, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c16.1740:\n",
      "\t{'chrom': 'LinJ.16', 'start': 199606, 'end': 202122, 'strand': '-', 'gene_id': 'LINF_160011100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'dihydroorotase', 'interval': Interval(199606, 202122, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 199606, 'end': 202122, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_16T0011100', 'parent_id': 'LINF_160011100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'dihydroorotase', 'interval': Interval(199606, 202122, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 199606, 'end': 200796, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_16T0011100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(199606, 200796, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c16.1750:\n",
      "\t{'chrom': 'LinJ.16', 'start': 261021, 'end': 264527, 'strand': '-', 'gene_id': 'LINF_160013000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'transaldolase', 'interval': Interval(261021, 264527, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 261021, 'end': 264527, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_16T0013000', 'parent_id': 'LINF_160013000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'transaldolase', 'interval': Interval(261021, 264527, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 261021, 'end': 263376, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_16T0013000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(261021, 263376, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c16.1760Y:\n",
      "\t{'chrom': 'LinJ.16', 'start': 371773, 'end': 376703, 'strand': '+', 'gene_id': 'LINF_160015800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_(DUF3184)', 'interval': Interval(371773, 376703, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 371773, 'end': 376703, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_16T0015800', 'parent_id': 'LINF_160015800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_(DUF3184)', 'interval': Interval(371773, 376703, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 375214, 'end': 376703, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_16T0015800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(375214, 376703, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c16.1770Y:\n",
      "\t{'chrom': 'LinJ.16', 'start': 377060, 'end': 383329, 'strand': '+', 'gene_id': 'LINF_160015820', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_(DUF3184)', 'interval': Interval(377060, 383329, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 377060, 'end': 383329, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_16T0015820', 'parent_id': 'LINF_160015820', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_(DUF3184)', 'interval': Interval(377060, 383329, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 380258, 'end': 383329, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_16T0015820', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(380258, 383329, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c16.1780:\n",
      "\t{'chrom': 'LinJ.16', 'start': 383706, 'end': 385908, 'strand': '+', 'gene_id': 'LINF_160015840', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(383706, 385908, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 383706, 'end': 385908, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_16T0015840', 'parent_id': 'LINF_160015840', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(383706, 385908, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c16.1790:\n",
      "\t{'chrom': 'LinJ.16', 'start': 555859, 'end': 561408, 'strand': '-', 'gene_id': 'LINF_160019700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'OTU-like_cysteine_protease_-_putative', 'interval': Interval(555859, 561408, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 555859, 'end': 561408, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_16T0019700', 'parent_id': 'LINF_160019700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'OTU-like_cysteine_protease_-_putative', 'interval': Interval(555859, 561408, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 555859, 'end': 557567, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_16T0019700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(555859, 557567, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c16.1800:\n",
      "\t{'chrom': 'LinJ.16', 'start': 603494, 'end': 611700, 'strand': '-', 'gene_id': 'LINF_160020900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(603494, 611700, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 603494, 'end': 611700, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_16T0020900', 'parent_id': 'LINF_160020900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(603494, 611700, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 603494, 'end': 608917, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_16T0020900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(603494, 608917, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c16.1810:\n",
      "\t{'chrom': 'LinJ.16', 'start': 603494, 'end': 611700, 'strand': '-', 'gene_id': 'LINF_160020900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(603494, 611700, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 603494, 'end': 611700, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_16T0020900', 'parent_id': 'LINF_160020900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(603494, 611700, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 603494, 'end': 608917, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_16T0020900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(603494, 608917, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c16.1820:\n",
      "\t{'chrom': 'LinJ.16', 'start': 686309, 'end': 687633, 'strand': '+', 'gene_id': 'LINF_160022925', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(686309, 687633, closed='both')}\n",
      "\t{'chrom': 'LinJ.16', 'start': 686309, 'end': 687633, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_16T0022925', 'parent_id': 'LINF_160022925', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(686309, 687633, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c17.1830Z:\n",
      "==================================================\n",
      "rejected_noCDS_c17.1840:\n",
      "\t{'chrom': 'LinJ.17', 'start': 62105, 'end': 66524, 'strand': '-', 'gene_id': 'LINF_170007350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(62105, 66524, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 62105, 'end': 66524, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_17T0007350', 'parent_id': 'LINF_170007350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(62105, 66524, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c17.1850F:\n",
      "\t{'chrom': 'LinJ.17', 'start': 62105, 'end': 66524, 'strand': '-', 'gene_id': 'LINF_170007350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(62105, 66524, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 62105, 'end': 66524, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_17T0007350', 'parent_id': 'LINF_170007350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(62105, 66524, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c17.1860:\n",
      "\t{'chrom': 'LinJ.17', 'start': 177727, 'end': 181043, 'strand': '-', 'gene_id': 'LINF_170010000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(177727, 181043, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 177727, 'end': 181043, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_17T0010000', 'parent_id': 'LINF_170010000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(177727, 181043, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 177727, 'end': 178250, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_17T0010000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(177727, 178250, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c17.1870:\n",
      "\t{'chrom': 'LinJ.17', 'start': 389710, 'end': 393557, 'strand': '-', 'gene_id': 'LINF_170014800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'guanosine_monophosphate_reductase|GMPR', 'interval': Interval(389710, 393557, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 389710, 'end': 393557, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_17T0014800', 'parent_id': 'LINF_170014800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'guanosine_monophosphate_reductase|GMPR', 'interval': Interval(389710, 393557, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 389710, 'end': 391987, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_17T0014800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(389710, 391987, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c17.1880:\n",
      "\t{'chrom': 'LinJ.17', 'start': 389710, 'end': 393557, 'strand': '-', 'gene_id': 'LINF_170014800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'guanosine_monophosphate_reductase|GMPR', 'interval': Interval(389710, 393557, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 389710, 'end': 393557, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_17T0014800', 'parent_id': 'LINF_170014800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'guanosine_monophosphate_reductase|GMPR', 'interval': Interval(389710, 393557, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 389710, 'end': 391987, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_17T0014800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(389710, 391987, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c17.1890:\n",
      "\t{'chrom': 'LinJ.17', 'start': 389710, 'end': 393557, 'strand': '-', 'gene_id': 'LINF_170014800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'guanosine_monophosphate_reductase|GMPR', 'interval': Interval(389710, 393557, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 389710, 'end': 393557, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_17T0014800', 'parent_id': 'LINF_170014800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'guanosine_monophosphate_reductase|GMPR', 'interval': Interval(389710, 393557, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 389710, 'end': 391987, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_17T0014800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(389710, 391987, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c17.1900:\n",
      "==================================================\n",
      "rejected_noCDS_c17.1910:\n",
      "==================================================\n",
      "rejected_noCDS_c17.1920:\n",
      "\t{'chrom': 'LinJ.17', 'start': 525829, 'end': 527199, 'strand': '+', 'gene_id': 'LINF_170017750', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(525829, 527199, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 525829, 'end': 527199, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_17T0017750', 'parent_id': 'LINF_170017750', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(525829, 527199, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c17.1930:\n",
      "\t{'chrom': 'LinJ.17', 'start': 527563, 'end': 531143, 'strand': '+', 'gene_id': 'LINF_170017800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'esterase-like_protein', 'interval': Interval(527563, 531143, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 527563, 'end': 531143, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_17T0017800', 'parent_id': 'LINF_170017800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'esterase-like_protein', 'interval': Interval(527563, 531143, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 527563, 'end': 528167, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_17T0017800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(527563, 528167, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c17.1940AA:\n",
      "==================================================\n",
      "rejected_noCDS_c17.1950AA:\n",
      "\t{'chrom': 'LinJ.17', 'start': 580021, 'end': 585507, 'strand': '+', 'gene_id': 'LINF_170019100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': 'unknown', 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(580021, 585507, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 580021, 'end': 585507, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_17T0019100', 'parent_id': 'LINF_170019100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(580021, 585507, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 583054, 'end': 585507, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_17T0019100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(583054, 585507, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c17.1960:\n",
      "\t{'chrom': 'LinJ.17', 'start': 585508, 'end': 591702, 'strand': '+', 'gene_id': 'LINF_170019200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(585508, 591702, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 585508, 'end': 591702, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_17T0019200', 'parent_id': 'LINF_170019200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(585508, 591702, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 588165, 'end': 591702, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_17T0019200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(588165, 591702, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c17.1970Z:\n",
      "==================================================\n",
      "rejected_noCDS_c18.1980:\n",
      "==================================================\n",
      "rejected_noCDS_c18.1990:\n",
      "==================================================\n",
      "rejected_noCDS_c18.2000:\n",
      "==================================================\n",
      "rejected_noCDS_c18.2010:\n",
      "\t{'chrom': 'LinJ.18', 'start': 284231, 'end': 290595, 'strand': '+', 'gene_id': 'LINF_180012200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'citrate_synthase', 'interval': Interval(284231, 290595, closed='both')}\n",
      "\t{'chrom': 'LinJ.18', 'start': 284231, 'end': 290595, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_18T0012200', 'parent_id': 'LINF_180012200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'citrate_synthase', 'interval': Interval(284231, 290595, closed='both')}\n",
      "\t{'chrom': 'LinJ.18', 'start': 286061, 'end': 290595, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_18T0012200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(286061, 290595, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c18.2020:\n",
      "\t{'chrom': 'LinJ.18', 'start': 479622, 'end': 482195, 'strand': '+', 'gene_id': 'LINF_180017050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(479622, 482195, closed='both')}\n",
      "\t{'chrom': 'LinJ.18', 'start': 479622, 'end': 482195, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_18T0017050', 'parent_id': 'LINF_180017050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(479622, 482195, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c18.2030:\n",
      "\t{'chrom': 'LinJ.18', 'start': 547000, 'end': 554693, 'strand': '+', 'gene_id': 'LINF_180018700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'zinc_finger_protein', 'interval': Interval(547000, 554693, closed='both')}\n",
      "\t{'chrom': 'LinJ.18', 'start': 547000, 'end': 554693, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_18T0018700', 'parent_id': 'LINF_180018700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'zinc_finger_protein', 'interval': Interval(547000, 554693, closed='both')}\n",
      "\t{'chrom': 'LinJ.18', 'start': 551897, 'end': 554693, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_18T0018700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(551897, 554693, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c18.2040:\n",
      "\t{'chrom': 'LinJ.18', 'start': 712068, 'end': 716116, 'strand': '+', 'gene_id': 'LINF_180022400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(712068, 716116, closed='both')}\n",
      "\t{'chrom': 'LinJ.18', 'start': 712068, 'end': 716116, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_18T0022400', 'parent_id': 'LINF_180022400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(712068, 716116, closed='both')}\n",
      "\t{'chrom': 'LinJ.18', 'start': 714187, 'end': 716116, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_18T0022400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(714187, 716116, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2050:\n",
      "\t{'chrom': 'LinJ.19', 'start': 51525, 'end': 56640, 'strand': '-', 'gene_id': 'LINF_190006900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ADP-_ATP_carrier_protein_1|ANC2|MCP5a', 'interval': Interval(51525, 56640, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 51525, 'end': 56640, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_19T0006900', 'parent_id': 'LINF_190006900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ADP-_ATP_carrier_protein_1|ANC2|MCP5a', 'interval': Interval(51525, 56640, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 51525, 'end': 55616, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_19T0006900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(51525, 55616, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2060AB:\n",
      "\t{'chrom': 'LinJ.19', 'start': 182279, 'end': 187638, 'strand': '+', 'gene_id': 'LINF_190009700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(182279, 187638, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 182279, 'end': 187638, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_19T0009700', 'parent_id': 'LINF_190009700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(182279, 187638, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 183711, 'end': 187638, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_19T0009700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(183711, 187638, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2070AC:\n",
      "\t{'chrom': 'LinJ.19', 'start': 207856, 'end': 211386, 'strand': '+', 'gene_id': 'LINF_190010300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Methionine_aminopeptidase_1|MetAP1', 'interval': Interval(207856, 211386, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 207856, 'end': 211386, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_19T0010300', 'parent_id': 'LINF_190010300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Methionine_aminopeptidase_1|MetAP1', 'interval': Interval(207856, 211386, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 209327, 'end': 211386, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_19T0010300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(209327, 211386, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2080AB:\n",
      "\t{'chrom': 'LinJ.19', 'start': 207856, 'end': 211386, 'strand': '+', 'gene_id': 'LINF_190010300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Methionine_aminopeptidase_1|MetAP1', 'interval': Interval(207856, 211386, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 207856, 'end': 211386, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_19T0010300', 'parent_id': 'LINF_190010300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Methionine_aminopeptidase_1|MetAP1', 'interval': Interval(207856, 211386, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 209327, 'end': 211386, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_19T0010300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(209327, 211386, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2090AC:\n",
      "\t{'chrom': 'LinJ.19', 'start': 224264, 'end': 231735, 'strand': '+', 'gene_id': 'LINF_190010600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(224264, 231735, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 224264, 'end': 231735, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_19T0010600', 'parent_id': 'LINF_190010600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(224264, 231735, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 227685, 'end': 231735, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_19T0010600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(227685, 231735, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2100:\n",
      "\t{'chrom': 'LinJ.19', 'start': 266291, 'end': 267888, 'strand': '+', 'gene_id': 'LINF_190010950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(266291, 267888, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 266291, 'end': 267888, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_19T0010950', 'parent_id': 'LINF_190010950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(266291, 267888, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2110:\n",
      "==================================================\n",
      "rejected_noCDS_c19.2120AD:\n",
      "==================================================\n",
      "rejected_noCDS_c19.2130AD:\n",
      "==================================================\n",
      "rejected_noCDS_c19.2140:\n",
      "\t{'chrom': 'LinJ.19', 'start': 407076, 'end': 408586, 'strand': '+', 'gene_id': 'LINF_190015250', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(407076, 408586, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 407076, 'end': 408586, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_19T0015250', 'parent_id': 'LINF_190015250', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(407076, 408586, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2150:\n",
      "\t{'chrom': 'LinJ.19', 'start': 407076, 'end': 408586, 'strand': '+', 'gene_id': 'LINF_190015250', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(407076, 408586, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 407076, 'end': 408586, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_19T0015250', 'parent_id': 'LINF_190015250', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(407076, 408586, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2160:\n",
      "==================================================\n",
      "rejected_noCDS_c19.2170AE:\n",
      "\t{'chrom': 'LinJ.19', 'start': 523131, 'end': 525327, 'strand': '+', 'gene_id': 'LINF_190017400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'actin-related_protein_2|Arp2', 'interval': Interval(523131, 525327, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 523131, 'end': 525327, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_19T0017400', 'parent_id': 'LINF_190017400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'actin-related_protein_2|Arp2', 'interval': Interval(523131, 525327, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 524714, 'end': 525327, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_19T0017400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(524714, 525327, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2180:\n",
      "\t{'chrom': 'LinJ.19', 'start': 529008, 'end': 529966, 'strand': '+', 'gene_id': 'LINF_190017650', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(529008, 529966, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 529008, 'end': 529966, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_19T0017650', 'parent_id': 'LINF_190017650', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(529008, 529966, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2190AE:\n",
      "\t{'chrom': 'LinJ.19', 'start': 567389, 'end': 571451, 'strand': '+', 'gene_id': 'LINF_190018700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glycerol_uptake_protein_-_putative', 'interval': Interval(567389, 571451, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 567389, 'end': 571451, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_19T0018700', 'parent_id': 'LINF_190018700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glycerol_uptake_protein_-_putative', 'interval': Interval(567389, 571451, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 570052, 'end': 571451, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_19T0018700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(570052, 571451, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2200AE:\n",
      "\t{'chrom': 'LinJ.19', 'start': 580307, 'end': 583972, 'strand': '+', 'gene_id': 'LINF_190019000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'MBOAT_domain-containing_protein', 'interval': Interval(580307, 583972, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 580307, 'end': 583972, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_19T0019000', 'parent_id': 'LINF_190019000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'MBOAT_domain-containing_protein', 'interval': Interval(580307, 583972, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 582943, 'end': 583972, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_19T0019000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(582943, 583972, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2210AF:\n",
      "\t{'chrom': 'LinJ.19', 'start': 615371, 'end': 617193, 'strand': '+', 'gene_id': 'LINF_190019900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(615371, 617193, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 615371, 'end': 617193, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_19T0019900', 'parent_id': 'LINF_190019900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(615371, 617193, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 615856, 'end': 617193, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_19T0019900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(615856, 617193, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2220AF:\n",
      "\t{'chrom': 'LinJ.19', 'start': 636887, 'end': 639685, 'strand': '+', 'gene_id': 'LINF_190020600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'cysteine_peptidase_A_(CPA)', 'interval': Interval(636887, 639685, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 636887, 'end': 639685, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_19T0020600', 'parent_id': 'LINF_190020600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'cysteine_peptidase_A_(CPA)', 'interval': Interval(636887, 639685, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 638293, 'end': 639685, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_19T0020600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(638293, 639685, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2230:\n",
      "==================================================\n",
      "rejected_noCDS_c20.2240:\n",
      "\t{'chrom': 'LinJ.20', 'start': 55241, 'end': 59296, 'strand': '-', 'gene_id': 'LINF_200006600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'wd40_repeat_domain-containing_protein', 'interval': Interval(55241, 59296, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 55241, 'end': 59296, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_20T0006600', 'parent_id': 'LINF_200006600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'wd40_repeat_domain-containing_protein', 'interval': Interval(55241, 59296, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 55241, 'end': 56713, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_20T0006600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(55241, 56713, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c20.2250:\n",
      "==================================================\n",
      "rejected_noCDS_c20.2260:\n",
      "==================================================\n",
      "rejected_noCDS_c20.2270:\n",
      "\t{'chrom': 'LinJ.20', 'start': 334910, 'end': 336234, 'strand': '+', 'gene_id': 'LINF_200013200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Holo-[acyl-carrier-protein]_synthase', 'interval': Interval(334910, 336234, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 334910, 'end': 336234, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_20T0013200', 'parent_id': 'LINF_200013200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Holo-[acyl-carrier-protein]_synthase', 'interval': Interval(334910, 336234, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 334910, 'end': 335367, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_20T0013200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(334910, 335367, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c20.2280:\n",
      "\t{'chrom': 'LinJ.20', 'start': 413052, 'end': 414543, 'strand': '+', 'gene_id': 'LINF_200015250', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(413052, 414543, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 413052, 'end': 414543, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_20T0015250', 'parent_id': 'LINF_200015250', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(413052, 414543, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c20.2290:\n",
      "\t{'chrom': 'LinJ.20', 'start': 413052, 'end': 414543, 'strand': '+', 'gene_id': 'LINF_200015250', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(413052, 414543, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 413052, 'end': 414543, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_20T0015250', 'parent_id': 'LINF_200015250', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(413052, 414543, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c20.2300:\n",
      "\t{'chrom': 'LinJ.20', 'start': 413052, 'end': 414543, 'strand': '+', 'gene_id': 'LINF_200015250', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(413052, 414543, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 413052, 'end': 414543, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_20T0015250', 'parent_id': 'LINF_200015250', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(413052, 414543, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c20.2310:\n",
      "==================================================\n",
      "rejected_noCDS_c20.2320:\n",
      "\t{'chrom': 'LinJ.20', 'start': 549192, 'end': 552336, 'strand': '+', 'gene_id': 'LINF_200017300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'calpain-like_cysteine_peptidase_-_Clan_CA_-_family_C2', 'interval': Interval(549192, 552336, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 549192, 'end': 552336, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_20T0017300', 'parent_id': 'LINF_200017300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'calpain-like_cysteine_peptidase_-_Clan_CA_-_family_C2', 'interval': Interval(549192, 552336, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 549986, 'end': 552336, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_20T0017300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(549986, 552336, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c20.2330:\n",
      "\t{'chrom': 'LinJ.20', 'start': 634334, 'end': 640436, 'strand': '+', 'gene_id': 'LINF_200019500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(634334, 640436, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 634334, 'end': 640436, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_20T0019500', 'parent_id': 'LINF_200019500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(634334, 640436, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 638486, 'end': 640436, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_20T0019500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(638486, 640436, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c21.2340:\n",
      "\t{'chrom': 'LinJ.21', 'start': 41211, 'end': 50310, 'strand': '+', 'gene_id': 'LINF_210006700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'serine/threonine_protein_kinase', 'interval': Interval(41211, 50310, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 41211, 'end': 50310, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_21T0006700', 'parent_id': 'LINF_210006700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'serine/threonine_protein_kinase', 'interval': Interval(41211, 50310, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 47904, 'end': 50310, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_21T0006700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(47904, 50310, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c21.2350:\n",
      "\t{'chrom': 'LinJ.21', 'start': 306597, 'end': 310529, 'strand': '+', 'gene_id': 'LINF_210014700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'methionine_aminopeptidase_2|MetAP2', 'interval': Interval(306597, 310529, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 306597, 'end': 310529, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_21T0014700', 'parent_id': 'LINF_210014700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'methionine_aminopeptidase_2|MetAP2', 'interval': Interval(306597, 310529, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 308457, 'end': 310529, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_21T0014700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(308457, 310529, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c21.2360:\n",
      "\t{'chrom': 'LinJ.21', 'start': 306597, 'end': 310529, 'strand': '+', 'gene_id': 'LINF_210014700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'methionine_aminopeptidase_2|MetAP2', 'interval': Interval(306597, 310529, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 306597, 'end': 310529, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_21T0014700', 'parent_id': 'LINF_210014700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'methionine_aminopeptidase_2|MetAP2', 'interval': Interval(306597, 310529, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 308457, 'end': 310529, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_21T0014700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(308457, 310529, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c21.2370:\n",
      "\t{'chrom': 'LinJ.21', 'start': 306597, 'end': 310529, 'strand': '+', 'gene_id': 'LINF_210014700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'methionine_aminopeptidase_2|MetAP2', 'interval': Interval(306597, 310529, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 306597, 'end': 310529, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_21T0014700', 'parent_id': 'LINF_210014700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'methionine_aminopeptidase_2|MetAP2', 'interval': Interval(306597, 310529, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 308457, 'end': 310529, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_21T0014700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(308457, 310529, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c21.2380:\n",
      "\t{'chrom': 'LinJ.21', 'start': 448700, 'end': 449476, 'strand': '+', 'gene_id': 'LINF_210018850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(448700, 449476, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 448700, 'end': 449476, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_21T0018850', 'parent_id': 'LINF_210018850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(448700, 449476, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c21.2390:\n",
      "==================================================\n",
      "rejected_noCDS_c21.2400:\n",
      "\t{'chrom': 'LinJ.21', 'start': 536332, 'end': 537886, 'strand': '-', 'gene_id': 'LINF_210021150', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(536332, 537886, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 536332, 'end': 537886, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_21T0021150', 'parent_id': 'LINF_210021150', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(536332, 537886, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c21.2410:\n",
      "\t{'chrom': 'LinJ.21', 'start': 563826, 'end': 569062, 'strand': '-', 'gene_id': 'LINF_210022000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(563826, 569062, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 563826, 'end': 569062, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_21T0022000', 'parent_id': 'LINF_210022000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(563826, 569062, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 563826, 'end': 564956, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_21T0022000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(563826, 564956, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c21.2420:\n",
      "\t{'chrom': 'LinJ.21', 'start': 563826, 'end': 569062, 'strand': '-', 'gene_id': 'LINF_210022000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(563826, 569062, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 563826, 'end': 569062, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_21T0022000', 'parent_id': 'LINF_210022000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(563826, 569062, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 563826, 'end': 564956, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_21T0022000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(563826, 564956, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c21.2430:\n",
      "\t{'chrom': 'LinJ.21', 'start': 579494, 'end': 582385, 'strand': '-', 'gene_id': 'LINF_210022900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': '2-oxoisovalerate_dehydrogenase_alpha_subunit', 'interval': Interval(579494, 582385, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 579494, 'end': 582385, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_21T0022900', 'parent_id': 'LINF_210022900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': '2-oxoisovalerate_dehydrogenase_alpha_subunit', 'interval': Interval(579494, 582385, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 579494, 'end': 580860, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_21T0022900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(579494, 580860, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c21.2440:\n",
      "\t{'chrom': 'LinJ.21', 'start': 621660, 'end': 628792, 'strand': '-', 'gene_id': 'LINF_210024000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_of_unknown_function_N-terminal_domain_(DUF2450)/Sec8_exocyst_complex_component_specific_domain_containing_protein_-_putative', 'interval': Interval(621660, 628792, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 621660, 'end': 628792, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_21T0024000', 'parent_id': 'LINF_210024000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_of_unknown_function_N-terminal_domain_(DUF2450)/Sec8_exocyst_complex_component_specific_domain_containing_protein_-_putative', 'interval': Interval(621660, 628792, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 621660, 'end': 622578, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_21T0024000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(621660, 622578, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c21.2450:\n",
      "\t{'chrom': 'LinJ.21', 'start': 686906, 'end': 690339, 'strand': '-', 'gene_id': 'LINF_210025500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'F-box_domain/F-box-like_-_putative', 'interval': Interval(686906, 690339, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 686906, 'end': 690339, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_21T0025500', 'parent_id': 'LINF_210025500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'F-box_domain/F-box-like_-_putative', 'interval': Interval(686906, 690339, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 686906, 'end': 687535, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_21T0025500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(686906, 687535, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c21.2460Z:\n",
      "==================================================\n",
      "rejected_noCDS_c22.2470:\n",
      "\t{'chrom': 'LinJ.22', 'start': 29384, 'end': 44839, 'strand': '-', 'gene_id': 'LINF_220006000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(29384, 44839, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 29384, 'end': 44839, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_22T0006000', 'parent_id': 'LINF_220006000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(29384, 44839, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 29384, 'end': 41969, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_22T0006000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(29384, 41969, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c22.2480:\n",
      "==================================================\n",
      "rejected_noCDS_c22.2490:\n",
      "\t{'chrom': 'LinJ.22', 'start': 73126, 'end': 77776, 'strand': '-', 'gene_id': 'LINF_220006900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(73126, 77776, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 73126, 'end': 77776, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_22T0006900', 'parent_id': 'LINF_220006900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(73126, 77776, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 73126, 'end': 76037, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_22T0006900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(73126, 76037, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c22.2500:\n",
      "\t{'chrom': 'LinJ.22', 'start': 227298, 'end': 232489, 'strand': '-', 'gene_id': 'LINF_220010400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Glycogen_synthase_kinase-3_alpha|GSKA', 'interval': Interval(227298, 232489, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 227298, 'end': 232489, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_22T0010400', 'parent_id': 'LINF_220010400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Glycogen_synthase_kinase-3_alpha|GSKA', 'interval': Interval(227298, 232489, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 227298, 'end': 229649, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_22T0010400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(227298, 229649, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c22.2510:\n",
      "\t{'chrom': 'LinJ.22', 'start': 430883, 'end': 435297, 'strand': '+', 'gene_id': 'LINF_220015300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'AAA+_ATPase', 'interval': Interval(430883, 435297, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 430883, 'end': 435297, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_22T0015300', 'parent_id': 'LINF_220015300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'AAA+_ATPase', 'interval': Interval(430883, 435297, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 433814, 'end': 435297, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_22T0015300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(433814, 435297, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c22.2520:\n",
      "\t{'chrom': 'LinJ.22', 'start': 625728, 'end': 629885, 'strand': '-', 'gene_id': 'LINF_220019700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'farnesyl_pyrophosphate_synthase|FPPS', 'interval': Interval(625728, 629885, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 625728, 'end': 629885, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_22T0019700', 'parent_id': 'LINF_220019700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'farnesyl_pyrophosphate_synthase|FPPS', 'interval': Interval(625728, 629885, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 625728, 'end': 628628, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_22T0019700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(625728, 628628, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c22.2530A:\n",
      "\t{'chrom': 'LinJ.22', 'start': 744694, 'end': 748242, 'strand': '+', 'gene_id': 'LINF_220023125', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(744694, 748242, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 744694, 'end': 748242, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_22T0023125', 'parent_id': 'LINF_220023125', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(744694, 748242, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c22.2540A:\n",
      "==================================================\n",
      "rejected_noCDS_c22.2550:\n",
      "\t{'chrom': 'LinJ.22', 'start': 754539, 'end': 782139, 'strand': '+', 'gene_id': 'LINF_220023200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(754539, 782139, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 754539, 'end': 782139, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_22T0023200', 'parent_id': 'LINF_220023200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(754539, 782139, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c23.2560:\n",
      "==================================================\n",
      "rejected_noCDS_c23.2570:\n",
      "==================================================\n",
      "rejected_noCDS_c23.2580AG:\n",
      "\t{'chrom': 'LinJ.23', 'start': 93890, 'end': 100632, 'strand': '+', 'gene_id': 'LINF_230007900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'argininosuccinate_synthase|Citrulline-aspartate_ligase', 'interval': Interval(93890, 100632, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 93890, 'end': 100632, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_23T0007900', 'parent_id': 'LINF_230007900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'argininosuccinate_synthase|Citrulline-aspartate_ligase', 'interval': Interval(93890, 100632, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 95523, 'end': 100632, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_23T0007900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(95523, 100632, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c23.2590AG:\n",
      "\t{'chrom': 'LinJ.23', 'start': 93890, 'end': 100632, 'strand': '+', 'gene_id': 'LINF_230007900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'argininosuccinate_synthase|Citrulline-aspartate_ligase', 'interval': Interval(93890, 100632, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 93890, 'end': 100632, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_23T0007900', 'parent_id': 'LINF_230007900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'argininosuccinate_synthase|Citrulline-aspartate_ligase', 'interval': Interval(93890, 100632, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 95523, 'end': 100632, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_23T0007900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(95523, 100632, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c23.2600AG:\n",
      "\t{'chrom': 'LinJ.23', 'start': 93890, 'end': 100632, 'strand': '+', 'gene_id': 'LINF_230007900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'argininosuccinate_synthase|Citrulline-aspartate_ligase', 'interval': Interval(93890, 100632, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 93890, 'end': 100632, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_23T0007900', 'parent_id': 'LINF_230007900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'argininosuccinate_synthase|Citrulline-aspartate_ligase', 'interval': Interval(93890, 100632, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 95523, 'end': 100632, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_23T0007900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(95523, 100632, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c23.2610:\n",
      "\t{'chrom': 'LinJ.23', 'start': 241846, 'end': 243240, 'strand': '-', 'gene_id': 'LINF_230012950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(241846, 243240, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 241846, 'end': 243240, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_23T0012950', 'parent_id': 'LINF_230012950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(241846, 243240, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c23.2620AH:\n",
      "\t{'chrom': 'LinJ.23', 'start': 286043, 'end': 296768, 'strand': '-', 'gene_id': 'LINF_230013900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(286043, 296768, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 286043, 'end': 296768, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_23T0013900', 'parent_id': 'LINF_230013900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(286043, 296768, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 286043, 'end': 290757, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_23T0013900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(286043, 290757, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c23.2630AH:\n",
      "\t{'chrom': 'LinJ.23', 'start': 297075, 'end': 302972, 'strand': '-', 'gene_id': 'LINF_230014000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Exocyst_complex_component_Sec10', 'interval': Interval(297075, 302972, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 297075, 'end': 302972, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_23T0014000', 'parent_id': 'LINF_230014000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Exocyst_complex_component_Sec10', 'interval': Interval(297075, 302972, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 297075, 'end': 298752, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_23T0014000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(297075, 298752, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c23.2640AI:\n",
      "==================================================\n",
      "rejected_noCDS_c23.2650AI:\n",
      "==================================================\n",
      "rejected_noCDS_c23.2660:\n",
      "\t{'chrom': 'LinJ.23', 'start': 491115, 'end': 493877, 'strand': '-', 'gene_id': 'LINF_230018300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(491115, 493877, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 491115, 'end': 493877, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_23T0018300', 'parent_id': 'LINF_230018300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(491115, 493877, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 491115, 'end': 492674, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_23T0018300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(491115, 492674, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c23.2670:\n",
      "\t{'chrom': 'LinJ.23', 'start': 673878, 'end': 677745, 'strand': '+', 'gene_id': 'LINF_230023000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'mitochondrial_carrier_protein|MCP18', 'interval': Interval(673878, 677745, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 673878, 'end': 677745, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_23T0023000', 'parent_id': 'LINF_230023000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'mitochondrial_carrier_protein|MCP18', 'interval': Interval(673878, 677745, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 675821, 'end': 677745, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_23T0023000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(675821, 677745, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c23.2680:\n",
      "\t{'chrom': 'LinJ.23', 'start': 673878, 'end': 677745, 'strand': '+', 'gene_id': 'LINF_230023000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'mitochondrial_carrier_protein|MCP18', 'interval': Interval(673878, 677745, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 673878, 'end': 677745, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_23T0023000', 'parent_id': 'LINF_230023000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'mitochondrial_carrier_protein|MCP18', 'interval': Interval(673878, 677745, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 675821, 'end': 677745, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_23T0023000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(675821, 677745, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c23.2690:\n",
      "==================================================\n",
      "rejected_noCDS_c23.2700:\n",
      "\t{'chrom': 'LinJ.23', 'start': 772064, 'end': 775977, 'strand': '+', 'gene_id': 'LINF_230026200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Splicing_factor_SYF1', 'interval': Interval(772064, 775977, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 772064, 'end': 775977, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_23T0026200', 'parent_id': 'LINF_230026200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Splicing_factor_SYF1', 'interval': Interval(772064, 775977, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 774664, 'end': 775977, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_23T0026200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(774664, 775977, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c24.2710AJ:\n",
      "==================================================\n",
      "rejected_noCDS_c24.2720:\n",
      "\t{'chrom': 'LinJ.24', 'start': 110205, 'end': 117720, 'strand': '+', 'gene_id': 'LINF_240008300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5A', 'interval': Interval(110205, 117720, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 110205, 'end': 117720, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_24T0008300', 'parent_id': 'LINF_240008300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5A', 'interval': Interval(110205, 117720, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 111675, 'end': 117720, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_24T0008300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(111675, 117720, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c24.2730:\n",
      "\t{'chrom': 'LinJ.24', 'start': 110205, 'end': 117720, 'strand': '+', 'gene_id': 'LINF_240008300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5A', 'interval': Interval(110205, 117720, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 110205, 'end': 117720, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_24T0008300', 'parent_id': 'LINF_240008300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5A', 'interval': Interval(110205, 117720, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 111675, 'end': 117720, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_24T0008300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(111675, 117720, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c24.2740:\n",
      "\t{'chrom': 'LinJ.24', 'start': 110205, 'end': 117720, 'strand': '+', 'gene_id': 'LINF_240008300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5A', 'interval': Interval(110205, 117720, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 110205, 'end': 117720, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_24T0008300', 'parent_id': 'LINF_240008300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5A', 'interval': Interval(110205, 117720, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 111675, 'end': 117720, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_24T0008300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(111675, 117720, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c24.2750:\n",
      "\t{'chrom': 'LinJ.24', 'start': 110205, 'end': 117720, 'strand': '+', 'gene_id': 'LINF_240008300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5A', 'interval': Interval(110205, 117720, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 110205, 'end': 117720, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_24T0008300', 'parent_id': 'LINF_240008300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5A', 'interval': Interval(110205, 117720, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 111675, 'end': 117720, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_24T0008300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(111675, 117720, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c24.2760:\n",
      "\t{'chrom': 'LinJ.24', 'start': 110205, 'end': 117720, 'strand': '+', 'gene_id': 'LINF_240008300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5A', 'interval': Interval(110205, 117720, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 110205, 'end': 117720, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_24T0008300', 'parent_id': 'LINF_240008300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5A', 'interval': Interval(110205, 117720, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 111675, 'end': 117720, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_24T0008300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(111675, 117720, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c24.2770AJ:\n",
      "==================================================\n",
      "rejected_noCDS_c24.2780:\n",
      "\t{'chrom': 'LinJ.24', 'start': 127166, 'end': 132439, 'strand': '+', 'gene_id': 'LINF_240008600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ppg3-related_protein-like_protein', 'interval': Interval(127166, 132439, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 127166, 'end': 132439, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_24T0008600', 'parent_id': 'LINF_240008600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ppg3-related_protein-like_protein', 'interval': Interval(127166, 132439, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 130571, 'end': 132439, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_24T0008600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(130571, 132439, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c24.2790AJ:\n",
      "\t{'chrom': 'LinJ.24', 'start': 145556, 'end': 151128, 'strand': '+', 'gene_id': 'LINF_240009100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(145556, 151128, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 145556, 'end': 151128, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_24T0009100', 'parent_id': 'LINF_240009100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(145556, 151128, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 149748, 'end': 151128, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_24T0009100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(149748, 151128, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c24.2800:\n",
      "\t{'chrom': 'LinJ.24', 'start': 151576, 'end': 158682, 'strand': '+', 'gene_id': 'LINF_240009200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_of_unknown_function_(DUF3184)_-_putative', 'interval': Interval(151576, 158682, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 151576, 'end': 158682, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_24T0009200', 'parent_id': 'LINF_240009200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_of_unknown_function_(DUF3184)_-_putative', 'interval': Interval(151576, 158682, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 151576, 'end': 151869, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_24T0009200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(151576, 151869, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c24.2810:\n",
      "\t{'chrom': 'LinJ.24', 'start': 463694, 'end': 465142, 'strand': '+', 'gene_id': 'LINF_240017950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(463694, 465142, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 463694, 'end': 465142, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_24T0017950', 'parent_id': 'LINF_240017950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(463694, 465142, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c24.2820:\n",
      "\t{'chrom': 'LinJ.24', 'start': 705162, 'end': 706009, 'strand': '-', 'gene_id': 'LINF_240024650', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(705162, 706009, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 705162, 'end': 706009, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_24T0024650', 'parent_id': 'LINF_240024650', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(705162, 706009, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c24.2830:\n",
      "\t{'chrom': 'LinJ.24', 'start': 715366, 'end': 723695, 'strand': '-', 'gene_id': 'LINF_240024900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'cyclin_11_-_putative', 'interval': Interval(715366, 723695, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 715366, 'end': 723695, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_24T0024900', 'parent_id': 'LINF_240024900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'cyclin_11_-_putative', 'interval': Interval(715366, 723695, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 715366, 'end': 718947, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_24T0024900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(715366, 718947, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c24.2840:\n",
      "\t{'chrom': 'LinJ.24', 'start': 806844, 'end': 812023, 'strand': '-', 'gene_id': 'LINF_240027700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(806844, 812023, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 806844, 'end': 812023, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_24T0027700', 'parent_id': 'LINF_240027700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(806844, 812023, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 806844, 'end': 811297, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_24T0027700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(806844, 811297, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c25.2850AK:\n",
      "==================================================\n",
      "rejected_noCDS_c25.2860AL:\n",
      "\t{'chrom': 'LinJ.25', 'start': 799, 'end': 5363, 'strand': '-', 'gene_id': 'LINF_250005000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'beta_galactofuranosyl_transferase|LPG1', 'interval': Interval(799, 5363, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 799, 'end': 5363, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_25T0005000', 'parent_id': 'LINF_250005000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'beta_galactofuranosyl_transferase|LPG1', 'interval': Interval(799, 5363, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 799, 'end': 3926, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_25T0005000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(799, 3926, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c25.2870AM:\n",
      "\t{'chrom': 'LinJ.25', 'start': 799, 'end': 5363, 'strand': '-', 'gene_id': 'LINF_250005000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'beta_galactofuranosyl_transferase|LPG1', 'interval': Interval(799, 5363, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 799, 'end': 5363, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_25T0005000', 'parent_id': 'LINF_250005000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'beta_galactofuranosyl_transferase|LPG1', 'interval': Interval(799, 5363, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 799, 'end': 3926, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_25T0005000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(799, 3926, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c25.2880AL:\n",
      "\t{'chrom': 'LinJ.25', 'start': 29864, 'end': 33171, 'strand': '-', 'gene_id': 'LINF_250006300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'electron_transfer_flavoprotein', 'interval': Interval(29864, 33171, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 29864, 'end': 33171, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_25T0006300', 'parent_id': 'LINF_250006300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'electron_transfer_flavoprotein', 'interval': Interval(29864, 33171, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 29864, 'end': 32167, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_25T0006300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(29864, 32167, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c25.2890AN:\n",
      "\t{'chrom': 'LinJ.25', 'start': 29864, 'end': 33171, 'strand': '-', 'gene_id': 'LINF_250006300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'electron_transfer_flavoprotein', 'interval': Interval(29864, 33171, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 29864, 'end': 33171, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_25T0006300', 'parent_id': 'LINF_250006300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'electron_transfer_flavoprotein', 'interval': Interval(29864, 33171, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 29864, 'end': 32167, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_25T0006300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(29864, 32167, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c25.2900AL:\n",
      "\t{'chrom': 'LinJ.25', 'start': 81625, 'end': 83195, 'strand': '-', 'gene_id': 'LINF_250007950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(81625, 83195, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 81625, 'end': 83195, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_25T0007950', 'parent_id': 'LINF_250007950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(81625, 83195, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c25.2910AM:\n",
      "\t{'chrom': 'LinJ.25', 'start': 81625, 'end': 83195, 'strand': '-', 'gene_id': 'LINF_250007950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(81625, 83195, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 81625, 'end': 83195, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_25T0007950', 'parent_id': 'LINF_250007950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(81625, 83195, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c25.2920AN:\n",
      "\t{'chrom': 'LinJ.25', 'start': 81625, 'end': 83195, 'strand': '-', 'gene_id': 'LINF_250007950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(81625, 83195, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 81625, 'end': 83195, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_25T0007950', 'parent_id': 'LINF_250007950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(81625, 83195, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c25.2930:\n",
      "\t{'chrom': 'LinJ.25', 'start': 290803, 'end': 297802, 'strand': '+', 'gene_id': 'LINF_250013900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(290803, 297802, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 290803, 'end': 297802, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_25T0013900', 'parent_id': 'LINF_250013900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(290803, 297802, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 296849, 'end': 297802, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_25T0013900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(296849, 297802, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c25.2940:\n",
      "==================================================\n",
      "rejected_noCDS_c25.2950:\n",
      "\t{'chrom': 'LinJ.25', 'start': 449408, 'end': 451511, 'strand': '-', 'gene_id': 'LINF_250017550', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(449408, 451511, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 449408, 'end': 451511, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_25T0017550', 'parent_id': 'LINF_250017550', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(449408, 451511, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c25.2960:\n",
      "\t{'chrom': 'LinJ.25', 'start': 449408, 'end': 451511, 'strand': '-', 'gene_id': 'LINF_250017550', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(449408, 451511, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 449408, 'end': 451511, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_25T0017550', 'parent_id': 'LINF_250017550', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(449408, 451511, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c25.2970:\n",
      "\t{'chrom': 'LinJ.25', 'start': 574890, 'end': 575856, 'strand': '-', 'gene_id': 'LINF_250021230', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(574890, 575856, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 574890, 'end': 575856, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_25T0021230', 'parent_id': 'LINF_250021230', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(574890, 575856, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c25.2980:\n",
      "\t{'chrom': 'LinJ.25', 'start': 603830, 'end': 611714, 'strand': '-', 'gene_id': 'LINF_250021950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(603830, 611714, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 603830, 'end': 611714, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_25T0021950', 'parent_id': 'LINF_250021950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(603830, 611714, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c25.2990:\n",
      "\t{'chrom': 'LinJ.25', 'start': 626555, 'end': 631894, 'strand': '-', 'gene_id': 'LINF_250022300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Casein_kinase_I|CK1.5', 'interval': Interval(626555, 631894, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 626555, 'end': 631894, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_25T0022300', 'parent_id': 'LINF_250022300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Casein_kinase_I|CK1.5', 'interval': Interval(626555, 631894, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 626555, 'end': 629636, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_25T0022300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(626555, 629636, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3000:\n",
      "\t{'chrom': 'LinJ.26', 'start': 9244, 'end': 11310, 'strand': '-', 'gene_id': 'LINF_260005400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Cdc2-related_protein_kinase_7|CRK7', 'interval': Interval(9244, 11310, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 9244, 'end': 11310, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_26T0005400', 'parent_id': 'LINF_260005400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Cdc2-related_protein_kinase_7|CRK7', 'interval': Interval(9244, 11310, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 9244, 'end': 10189, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_26T0005400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(9244, 10189, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3010:\n",
      "\t{'chrom': 'LinJ.26', 'start': 34801, 'end': 41086, 'strand': '-', 'gene_id': 'LINF_260006500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'nuclear_lim_interactor-interacting_factor-like_protein', 'interval': Interval(34801, 41086, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 34801, 'end': 41086, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_26T0006500', 'parent_id': 'LINF_260006500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'nuclear_lim_interactor-interacting_factor-like_protein', 'interval': Interval(34801, 41086, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 34801, 'end': 37979, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_26T0006500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(34801, 37979, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3020:\n",
      "\t{'chrom': 'LinJ.26', 'start': 168807, 'end': 171144, 'strand': '-', 'gene_id': 'LINF_260011300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(168807, 171144, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 168807, 'end': 171144, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_26T0011300', 'parent_id': 'LINF_260011300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(168807, 171144, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 168807, 'end': 169810, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_26T0011300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(168807, 169810, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3030:\n",
      "\t{'chrom': 'LinJ.26', 'start': 224623, 'end': 225662, 'strand': '-', 'gene_id': 'LINF_260012750', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(224623, 225662, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 224623, 'end': 225662, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_26T0012750', 'parent_id': 'LINF_260012750', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(224623, 225662, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3040:\n",
      "==================================================\n",
      "rejected_noCDS_c26.3050:\n",
      "\t{'chrom': 'LinJ.26', 'start': 359633, 'end': 367293, 'strand': '+', 'gene_id': 'LINF_260016300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(359633, 367293, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 359633, 'end': 367293, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_26T0016300', 'parent_id': 'LINF_260016300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(359633, 367293, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 365550, 'end': 367293, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_26T0016300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(365550, 367293, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3060AO:\n",
      "\t{'chrom': 'LinJ.26', 'start': 497725, 'end': 504267, 'strand': '+', 'gene_id': 'LINF_260019400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(497725, 504267, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 497725, 'end': 504267, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_26T0019400', 'parent_id': 'LINF_260019400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(497725, 504267, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 500505, 'end': 504267, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_26T0019400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(500505, 504267, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3070AP:\n",
      "\t{'chrom': 'LinJ.26', 'start': 598501, 'end': 599917, 'strand': '+', 'gene_id': 'LINF_260021850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(598501, 599917, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 598501, 'end': 599917, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_26T0021850', 'parent_id': 'LINF_260021850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(598501, 599917, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3080:\n",
      "==================================================\n",
      "rejected_noCDS_c26.3090AP:\n",
      "\t{'chrom': 'LinJ.26', 'start': 605032, 'end': 607263, 'strand': '+', 'gene_id': 'LINF_260022000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'sphingolipid_delta_4_desaturase', 'interval': Interval(605032, 607263, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 605032, 'end': 607263, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_26T0022000', 'parent_id': 'LINF_260022000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'sphingolipid_delta_4_desaturase', 'interval': Interval(605032, 607263, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 605032, 'end': 605835, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_26T0022000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(605032, 605835, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3100AP:\n",
      "\t{'chrom': 'LinJ.26', 'start': 607528, 'end': 609700, 'strand': '+', 'gene_id': 'LINF_260022100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'sphingolipid_delta_4_desaturase', 'interval': Interval(607528, 609700, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 607528, 'end': 609700, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_26T0022100', 'parent_id': 'LINF_260022100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'sphingolipid_delta_4_desaturase', 'interval': Interval(607528, 609700, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 607528, 'end': 608331, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_26T0022100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(607528, 608331, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3110:\n",
      "==================================================\n",
      "rejected_noCDS_c26.3120AO:\n",
      "\t{'chrom': 'LinJ.26', 'start': 716644, 'end': 717576, 'strand': '+', 'gene_id': 'LINF_260024450', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(716644, 717576, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 716644, 'end': 717576, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_26T0024450', 'parent_id': 'LINF_260024450', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(716644, 717576, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3130:\n",
      "\t{'chrom': 'LinJ.26', 'start': 729077, 'end': 729908, 'strand': '+', 'gene_id': 'LINF_260024750', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(729077, 729908, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 729077, 'end': 729908, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_26T0024750', 'parent_id': 'LINF_260024750', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(729077, 729908, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3140:\n",
      "\t{'chrom': 'LinJ.26', 'start': 741070, 'end': 748289, 'strand': '+', 'gene_id': 'LINF_260024900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'nucleolar_protein_86|NOP86', 'interval': Interval(741070, 748289, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 741070, 'end': 748289, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_26T0024900', 'parent_id': 'LINF_260024900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'nucleolar_protein_86|NOP86', 'interval': Interval(741070, 748289, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 743868, 'end': 748289, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_26T0024900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(743868, 748289, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3150:\n",
      "==================================================\n",
      "rejected_noCDS_c26.3160AQ:\n",
      "\t{'chrom': 'LinJ.26', 'start': 984908, 'end': 989950, 'strand': '+', 'gene_id': 'LINF_260031100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'serine/threonine_protein_phosphatase_-_putative', 'interval': Interval(984908, 989950, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 984908, 'end': 989950, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_26T0031100', 'parent_id': 'LINF_260031100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'serine/threonine_protein_phosphatase_-_putative', 'interval': Interval(984908, 989950, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 986864, 'end': 989950, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_26T0031100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(986864, 989950, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3170AQ:\n",
      "\t{'chrom': 'LinJ.26', 'start': 997548, 'end': 1001508, 'strand': '+', 'gene_id': 'LINF_260031400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'SET25', 'interval': Interval(997548, 1001508, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 997548, 'end': 1001508, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_26T0031400', 'parent_id': 'LINF_260031400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'SET25', 'interval': Interval(997548, 1001508, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 999331, 'end': 1001508, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_26T0031400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(999331, 1001508, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3180:\n",
      "\t{'chrom': 'LinJ.26', 'start': 1021429, 'end': 1025061, 'strand': '+', 'gene_id': 'LINF_260032200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(1021429, 1025061, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 1021429, 'end': 1025061, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_26T0032200', 'parent_id': 'LINF_260032200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(1021429, 1025061, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 1022711, 'end': 1025061, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_26T0032200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1022711, 1025061, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3190:\n",
      "\t{'chrom': 'LinJ.26', 'start': 1034309, 'end': 1041738, 'strand': '+', 'gene_id': 'LINF_260032600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATP-binding_cassette_protein_subfamily_B,_member_2|ABCB2', 'interval': Interval(1034309, 1041738, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 1034309, 'end': 1041738, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_26T0032600', 'parent_id': 'LINF_260032600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATP-binding_cassette_protein_subfamily_B,_member_2|ABCB2', 'interval': Interval(1034309, 1041738, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 1038222, 'end': 1041738, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_26T0032600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1038222, 1041738, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_BLAST_ERROR:\n",
      "==================================================\n",
      "rejected_noCDS_c27.3200:\n",
      "\t{'chrom': 'LinJ.27', 'start': 171066, 'end': 172367, 'strand': '+', 'gene_id': 'LINF_270010160', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(171066, 172367, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 171066, 'end': 172367, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_27T0010160', 'parent_id': 'LINF_270010160', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(171066, 172367, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c27.3210AR:\n",
      "\t{'chrom': 'LinJ.27', 'start': 373959, 'end': 378139, 'strand': '+', 'gene_id': 'LINF_270014800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'WD_domain_-_G-beta_repeat_-_containing_protein', 'interval': Interval(373959, 378139, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 373959, 'end': 378139, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_27T0014800', 'parent_id': 'LINF_270014800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'WD_domain_-_G-beta_repeat_-_containing_protein', 'interval': Interval(373959, 378139, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 377264, 'end': 378139, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_27T0014800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(377264, 378139, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c27.3220AR:\n",
      "\t{'chrom': 'LinJ.27', 'start': 396554, 'end': 398111, 'strand': '+', 'gene_id': 'LINF_270015300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_of_unknown_function_(DUF778)_-_putative', 'interval': Interval(396554, 398111, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 396554, 'end': 398111, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_27T0015300', 'parent_id': 'LINF_270015300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_of_unknown_function_(DUF778)_-_putative', 'interval': Interval(396554, 398111, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 397383, 'end': 398111, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_27T0015300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(397383, 398111, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c27.3230AR:\n",
      "\t{'chrom': 'LinJ.27', 'start': 442614, 'end': 445289, 'strand': '-', 'gene_id': 'LINF_270015600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'isovaleryl-coA_dehydrogenase', 'interval': Interval(442614, 445289, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 442614, 'end': 445289, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_27T0015600', 'parent_id': 'LINF_270015600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'isovaleryl-coA_dehydrogenase', 'interval': Interval(442614, 445289, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 442614, 'end': 443869, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_27T0015600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(442614, 443869, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c27.3240AR:\n",
      "\t{'chrom': 'LinJ.27', 'start': 448298, 'end': 450150, 'strand': '-', 'gene_id': 'LINF_270015750', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(448298, 450150, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 448298, 'end': 450150, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_27T0015750', 'parent_id': 'LINF_270015750', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(448298, 450150, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c27.3250AR:\n",
      "\t{'chrom': 'LinJ.27', 'start': 454644, 'end': 461051, 'strand': '-', 'gene_id': 'LINF_270015900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(454644, 461051, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 454644, 'end': 461051, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_27T0015900', 'parent_id': 'LINF_270015900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(454644, 461051, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 459392, 'end': 461051, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_27T0015900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(459392, 461051, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c27.3260AR:\n",
      "\t{'chrom': 'LinJ.27', 'start': 508851, 'end': 513552, 'strand': '-', 'gene_id': 'LINF_270017100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'vesicular-fusion_ATPase-like_protein', 'interval': Interval(508851, 513552, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 508851, 'end': 513552, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_27T0017100', 'parent_id': 'LINF_270017100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'vesicular-fusion_ATPase-like_protein', 'interval': Interval(508851, 513552, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 508851, 'end': 509701, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_27T0017100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(508851, 509701, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c27.3270:\n",
      "\t{'chrom': 'LinJ.27', 'start': 536880, 'end': 541850, 'strand': '-', 'gene_id': 'LINF_270017800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(536880, 541850, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 536880, 'end': 541850, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_27T0017800', 'parent_id': 'LINF_270017800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(536880, 541850, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 536880, 'end': 538957, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_27T0017800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(536880, 538957, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c27.3280AR:\n",
      "\t{'chrom': 'LinJ.27', 'start': 550325, 'end': 553841, 'strand': '-', 'gene_id': 'LINF_270018200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(550325, 553841, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 550325, 'end': 553841, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_27T0018200', 'parent_id': 'LINF_270018200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(550325, 553841, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 550325, 'end': 552400, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_27T0018200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(550325, 552400, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c27.3290AS:\n",
      "==================================================\n",
      "rejected_noCDS_c27.3300AS:\n",
      "==================================================\n",
      "rejected_noCDS_c27.3310:\n",
      "==================================================\n",
      "rejected_noCDS_c27.3320:\n",
      "\t{'chrom': 'LinJ.27', 'start': 711026, 'end': 716531, 'strand': '+', 'gene_id': 'LINF_270022600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Amino_acid_transporter|AAT24.1', 'interval': Interval(711026, 716531, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 711026, 'end': 716531, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_27T0022600', 'parent_id': 'LINF_270022600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Amino_acid_transporter|AAT24.1', 'interval': Interval(711026, 716531, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 712740, 'end': 716531, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_27T0022600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(712740, 716531, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c27.3330:\n",
      "\t{'chrom': 'LinJ.27', 'start': 711026, 'end': 716531, 'strand': '+', 'gene_id': 'LINF_270022600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Amino_acid_transporter|AAT24.1', 'interval': Interval(711026, 716531, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 711026, 'end': 716531, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_27T0022600', 'parent_id': 'LINF_270022600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Amino_acid_transporter|AAT24.1', 'interval': Interval(711026, 716531, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 712740, 'end': 716531, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_27T0022600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(712740, 716531, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c27.3340:\n",
      "\t{'chrom': 'LinJ.27', 'start': 738939, 'end': 742140, 'strand': '+', 'gene_id': 'LINF_270023150', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(738939, 742140, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 738939, 'end': 742140, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_27T0023150', 'parent_id': 'LINF_270023150', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(738939, 742140, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c27.3350:\n",
      "\t{'chrom': 'LinJ.27', 'start': 828921, 'end': 830579, 'strand': '-', 'gene_id': 'LINF_270024650', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(828921, 830579, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 828921, 'end': 830579, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_27T0024650', 'parent_id': 'LINF_270024650', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(828921, 830579, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c27.3360:\n",
      "\t{'chrom': 'LinJ.27', 'start': 950247, 'end': 954489, 'strand': '-', 'gene_id': 'LINF_270027600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ribonucleoside-diphosphate_reductase_small_chain', 'interval': Interval(950247, 954489, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 950247, 'end': 954489, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_27T0027600', 'parent_id': 'LINF_270027600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ribonucleoside-diphosphate_reductase_small_chain', 'interval': Interval(950247, 954489, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 950247, 'end': 953079, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_27T0027600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(950247, 953079, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c28.3370:\n",
      "==================================================\n",
      "rejected_noCDS_c28.3380:\n",
      "\t{'chrom': 'LinJ.28', 'start': 115419, 'end': 119569, 'strand': '+', 'gene_id': 'LINF_280008350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(115419, 119569, closed='both')}\n",
      "\t{'chrom': 'LinJ.28', 'start': 115419, 'end': 119569, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_28T0008350', 'parent_id': 'LINF_280008350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(115419, 119569, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c28.3390:\n",
      "\t{'chrom': 'LinJ.28', 'start': 200847, 'end': 204526, 'strand': '-', 'gene_id': 'LINF_280010700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Major_surface_protease_gp63|Leishmanolysin', 'interval': Interval(200847, 204526, closed='both')}\n",
      "\t{'chrom': 'LinJ.28', 'start': 200847, 'end': 204526, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_28T0010700', 'parent_id': 'LINF_280010700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Major_surface_protease_gp63|Leishmanolysin', 'interval': Interval(200847, 204526, closed='both')}\n",
      "\t{'chrom': 'LinJ.28', 'start': 200847, 'end': 202095, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_28T0010700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(200847, 202095, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c28.3400:\n",
      "==================================================\n",
      "rejected_noCDS_c28.3410:\n",
      "==================================================\n",
      "rejected_noCDS_c28.3420:\n",
      "\t{'chrom': 'LinJ.28', 'start': 989071, 'end': 993325, 'strand': '+', 'gene_id': 'LINF_280032200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Eukaryotic_translation_initiation_factor_4E-3|eIF4E3', 'interval': Interval(989071, 993325, closed='both')}\n",
      "\t{'chrom': 'LinJ.28', 'start': 989071, 'end': 993325, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_28T0032200', 'parent_id': 'LINF_280032200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Eukaryotic_translation_initiation_factor_4E-3|eIF4E3', 'interval': Interval(989071, 993325, closed='both')}\n",
      "\t{'chrom': 'LinJ.28', 'start': 990316, 'end': 993325, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_28T0032200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(990316, 993325, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c28.3430Z:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3440:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3450:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3460AT:\n",
      "\t{'chrom': 'LinJ.29', 'start': 205977, 'end': 209251, 'strand': '-', 'gene_id': 'LINF_290010600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Exonuclease_-_putative', 'interval': Interval(205977, 209251, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 205977, 'end': 209251, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0010600', 'parent_id': 'LINF_290010600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Exonuclease_-_putative', 'interval': Interval(205977, 209251, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 205977, 'end': 207931, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0010600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(205977, 207931, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3470AU:\n",
      "\t{'chrom': 'LinJ.29', 'start': 205977, 'end': 209251, 'strand': '-', 'gene_id': 'LINF_290010600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Exonuclease_-_putative', 'interval': Interval(205977, 209251, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 205977, 'end': 209251, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0010600', 'parent_id': 'LINF_290010600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Exonuclease_-_putative', 'interval': Interval(205977, 209251, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 205977, 'end': 207931, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0010600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(205977, 207931, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3480AT:\n",
      "\t{'chrom': 'LinJ.29', 'start': 315865, 'end': 322387, 'strand': '-', 'gene_id': 'LINF_290013800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(315865, 322387, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 315865, 'end': 322387, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0013800', 'parent_id': 'LINF_290013800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(315865, 322387, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 315865, 'end': 320545, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0013800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(315865, 320545, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3490AU:\n",
      "\t{'chrom': 'LinJ.29', 'start': 315865, 'end': 322387, 'strand': '-', 'gene_id': 'LINF_290013800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(315865, 322387, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 315865, 'end': 322387, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0013800', 'parent_id': 'LINF_290013800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(315865, 322387, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 315865, 'end': 320545, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0013800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(315865, 320545, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3500AV:\n",
      "\t{'chrom': 'LinJ.29', 'start': 315865, 'end': 322387, 'strand': '-', 'gene_id': 'LINF_290013800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(315865, 322387, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 315865, 'end': 322387, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0013800', 'parent_id': 'LINF_290013800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(315865, 322387, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 315865, 'end': 320545, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0013800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(315865, 320545, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3510:\n",
      "\t{'chrom': 'LinJ.29', 'start': 315865, 'end': 322387, 'strand': '-', 'gene_id': 'LINF_290013800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(315865, 322387, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 315865, 'end': 322387, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0013800', 'parent_id': 'LINF_290013800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(315865, 322387, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 315865, 'end': 320545, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0013800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(315865, 320545, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3520AV:\n",
      "\t{'chrom': 'LinJ.29', 'start': 340306, 'end': 343004, 'strand': '+', 'gene_id': 'LINF_290014550', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(340306, 343004, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 340306, 'end': 343004, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0014550', 'parent_id': 'LINF_290014550', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(340306, 343004, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3530AW:\n",
      "\t{'chrom': 'LinJ.29', 'start': 556193, 'end': 560127, 'strand': '+', 'gene_id': 'LINF_290019350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(556193, 560127, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 556193, 'end': 560127, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019350', 'parent_id': 'LINF_290019350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(556193, 560127, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 556193, 'end': 556765, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0019350', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(556193, 556765, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3540:\n",
      "\t{'chrom': 'LinJ.29', 'start': 556193, 'end': 560127, 'strand': '+', 'gene_id': 'LINF_290019350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(556193, 560127, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 556193, 'end': 560127, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019350', 'parent_id': 'LINF_290019350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(556193, 560127, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 557537, 'end': 560127, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0019350', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(557537, 560127, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3550AX:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3560AY:\n",
      "\t{'chrom': 'LinJ.29', 'start': 563252, 'end': 564329, 'strand': '+', 'gene_id': 'LINF_290019415', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(563252, 564329, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 563252, 'end': 564329, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019415', 'parent_id': 'LINF_290019415', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(563252, 564329, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3570AZ:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3580AX:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3590:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3600:\n",
      "\t{'chrom': 'LinJ.29', 'start': 567113, 'end': 570359, 'strand': '+', 'gene_id': 'LINF_290019445', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(567113, 570359, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 567113, 'end': 570359, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019445', 'parent_id': 'LINF_290019445', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(567113, 570359, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3610AY:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3620AZ:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3630AX:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3640AY:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3650AZ:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3660AX:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3670AY:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3680AW:\n",
      "\t{'chrom': 'LinJ.29', 'start': 583499, 'end': 587978, 'strand': '+', 'gene_id': 'LINF_290019700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(583499, 587978, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 583499, 'end': 587978, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019700', 'parent_id': 'LINF_290019700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(583499, 587978, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 584181, 'end': 587978, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0019700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(584181, 587978, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3690:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3700:\n",
      "\t{'chrom': 'LinJ.29', 'start': 596044, 'end': 599094, 'strand': '+', 'gene_id': 'LINF_290019925', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(596044, 599094, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 596044, 'end': 599094, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019925', 'parent_id': 'LINF_290019925', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(596044, 599094, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3710BA:\n",
      "\t{'chrom': 'LinJ.29', 'start': 596044, 'end': 599094, 'strand': '+', 'gene_id': 'LINF_290019925', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(596044, 599094, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 596044, 'end': 599094, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019925', 'parent_id': 'LINF_290019925', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(596044, 599094, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3720BA:\n",
      "\t{'chrom': 'LinJ.29', 'start': 664074, 'end': 666968, 'strand': '+', 'gene_id': 'LINF_290020550', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(664074, 666968, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 664074, 'end': 666968, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0020550', 'parent_id': 'LINF_290020550', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(664074, 666968, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3730:\n",
      "\t{'chrom': 'LinJ.29', 'start': 670544, 'end': 676210, 'strand': '-', 'gene_id': 'LINF_290020800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(670544, 676210, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 670544, 'end': 676210, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0020800', 'parent_id': 'LINF_290020800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(670544, 676210, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 674876, 'end': 676210, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0020800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(674876, 676210, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3740:\n",
      "\t{'chrom': 'LinJ.29', 'start': 670544, 'end': 676210, 'strand': '-', 'gene_id': 'LINF_290020800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(670544, 676210, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 670544, 'end': 676210, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0020800', 'parent_id': 'LINF_290020800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(670544, 676210, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 674876, 'end': 676210, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0020800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(674876, 676210, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3750:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3760:\n",
      "\t{'chrom': 'LinJ.29', 'start': 883457, 'end': 886087, 'strand': '-', 'gene_id': 'LINF_290026300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(883457, 886087, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 883457, 'end': 886087, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0026300', 'parent_id': 'LINF_290026300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(883457, 886087, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 883457, 'end': 885564, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0026300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(883457, 885564, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3770:\n",
      "\t{'chrom': 'LinJ.29', 'start': 886088, 'end': 890961, 'strand': '-', 'gene_id': 'LINF_290026400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(886088, 890961, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 886088, 'end': 890961, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0026400', 'parent_id': 'LINF_290026400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(886088, 890961, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 886088, 'end': 887303, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0026400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(886088, 887303, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3780:\n",
      "\t{'chrom': 'LinJ.29', 'start': 948645, 'end': 958282, 'strand': '-', 'gene_id': 'LINF_290028200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(948645, 958282, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 948645, 'end': 958282, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0028200', 'parent_id': 'LINF_290028200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(948645, 958282, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 948645, 'end': 952540, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0028200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(948645, 952540, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3790:\n",
      "\t{'chrom': 'LinJ.29', 'start': 1065275, 'end': 1075994, 'strand': '-', 'gene_id': 'LINF_290030600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1065275, 1075994, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 1065275, 'end': 1075994, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0030600', 'parent_id': 'LINF_290030600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1065275, 1075994, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 1065275, 'end': 1073997, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0030600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1065275, 1073997, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3800:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3810:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3820BB:\n",
      "\t{'chrom': 'LinJ.29', 'start': 1225185, 'end': 1227361, 'strand': '+', 'gene_id': 'LINF_290035050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1225185, 1227361, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 1225185, 'end': 1227361, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0035050', 'parent_id': 'LINF_290035050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1225185, 1227361, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3830:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3840BB:\n",
      "\t{'chrom': 'LinJ.29', 'start': 1243962, 'end': 1247286, 'strand': '+', 'gene_id': 'LINF_290035800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'inosine-adenosine-guanosine-nucleoside_hydrolase', 'interval': Interval(1243962, 1247286, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 1243962, 'end': 1247286, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0035800', 'parent_id': 'LINF_290035800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'inosine-adenosine-guanosine-nucleoside_hydrolase', 'interval': Interval(1243962, 1247286, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 1245161, 'end': 1247286, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0035800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1245161, 1247286, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.3850:\n",
      "\t{'chrom': 'LinJ.30', 'start': 2996, 'end': 5515, 'strand': '-', 'gene_id': 'LINF_300005050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2996, 5515, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 2996, 'end': 5515, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_30T0005050', 'parent_id': 'LINF_300005050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2996, 5515, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.3860BC:\n",
      "==================================================\n",
      "rejected_noCDS_c30.3870:\n",
      "\t{'chrom': 'LinJ.30', 'start': 274753, 'end': 277962, 'strand': '+', 'gene_id': 'LINF_300014200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_protein', 'interval': Interval(274753, 277962, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 274753, 'end': 277962, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_30T0014200', 'parent_id': 'LINF_300014200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_protein', 'interval': Interval(274753, 277962, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 275497, 'end': 277962, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0014200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(275497, 277962, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.3880:\n",
      "\t{'chrom': 'LinJ.30', 'start': 285194, 'end': 289972, 'strand': '+', 'gene_id': 'LINF_300014500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'adenosine_kinase-like_protein', 'interval': Interval(285194, 289972, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 285194, 'end': 289972, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_30T0014500', 'parent_id': 'LINF_300014500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'adenosine_kinase-like_protein', 'interval': Interval(285194, 289972, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 287016, 'end': 289972, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0014500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(287016, 289972, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.3890BD:\n",
      "\t{'chrom': 'LinJ.30', 'start': 285194, 'end': 289972, 'strand': '+', 'gene_id': 'LINF_300014500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'adenosine_kinase-like_protein', 'interval': Interval(285194, 289972, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 285194, 'end': 289972, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_30T0014500', 'parent_id': 'LINF_300014500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'adenosine_kinase-like_protein', 'interval': Interval(285194, 289972, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 287016, 'end': 289972, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0014500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(287016, 289972, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.3900BD:\n",
      "\t{'chrom': 'LinJ.30', 'start': 313068, 'end': 318963, 'strand': '+', 'gene_id': 'LINF_300015100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Histone-binding_protein_RBBP4_or_subunit_C_of_CAF1_complex/WD_domain_-_G-beta_repeat_-_putative', 'interval': Interval(313068, 318963, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 313068, 'end': 318963, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_30T0015100', 'parent_id': 'LINF_300015100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Histone-binding_protein_RBBP4_or_subunit_C_of_CAF1_complex/WD_domain_-_G-beta_repeat_-_putative', 'interval': Interval(313068, 318963, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 315510, 'end': 318963, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0015100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(315510, 318963, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.3910:\n",
      "\t{'chrom': 'LinJ.30', 'start': 368894, 'end': 372785, 'strand': '+', 'gene_id': 'LINF_300016600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'DREV_methyltransferase_-_putative', 'interval': Interval(368894, 372785, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 368894, 'end': 372785, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_30T0016600', 'parent_id': 'LINF_300016600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'DREV_methyltransferase_-_putative', 'interval': Interval(368894, 372785, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 370247, 'end': 372785, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0016600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(370247, 372785, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.3920BE:\n",
      "\t{'chrom': 'LinJ.30', 'start': 477196, 'end': 481025, 'strand': '+', 'gene_id': 'LINF_300019400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amidohydrolase', 'interval': Interval(477196, 481025, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 477196, 'end': 481025, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_30T0019400', 'parent_id': 'LINF_300019400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amidohydrolase', 'interval': Interval(477196, 481025, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 478506, 'end': 481025, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0019400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(478506, 481025, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.3930:\n",
      "==================================================\n",
      "rejected_noCDS_c30.3940BE:\n",
      "\t{'chrom': 'LinJ.30', 'start': 507455, 'end': 508919, 'strand': '+', 'gene_id': 'LINF_300020050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(507455, 508919, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 507455, 'end': 508919, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_30T0020050', 'parent_id': 'LINF_300020050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(507455, 508919, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.3950BE:\n",
      "\t{'chrom': 'LinJ.30', 'start': 509128, 'end': 516446, 'strand': '+', 'gene_id': 'LINF_300020100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'kinesin', 'interval': Interval(509128, 516446, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 509128, 'end': 516446, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_30T0020100', 'parent_id': 'LINF_300020100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'kinesin', 'interval': Interval(509128, 516446, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 513204, 'end': 516446, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0020100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(513204, 516446, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.3960BE:\n",
      "\t{'chrom': 'LinJ.30', 'start': 516824, 'end': 519567, 'strand': '+', 'gene_id': 'LINF_300020200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'p1/s1_nuclease', 'interval': Interval(516824, 519567, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 516824, 'end': 519567, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_30T0020200', 'parent_id': 'LINF_300020200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'p1/s1_nuclease', 'interval': Interval(516824, 519567, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 518090, 'end': 519567, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0020200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(518090, 519567, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.3970:\n",
      "\t{'chrom': 'LinJ.30', 'start': 573800, 'end': 576362, 'strand': '+', 'gene_id': 'LINF_300021650', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(573800, 576362, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 573800, 'end': 576362, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_30T0021650', 'parent_id': 'LINF_300021650', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(573800, 576362, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.3980:\n",
      "\t{'chrom': 'LinJ.30', 'start': 743271, 'end': 745229, 'strand': '+', 'gene_id': 'LINF_300026050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(743271, 745229, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 743271, 'end': 745229, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_30T0026050', 'parent_id': 'LINF_300026050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(743271, 745229, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.3990:\n",
      "\t{'chrom': 'LinJ.30', 'start': 776492, 'end': 786702, 'strand': '-', 'gene_id': 'LINF_300027200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RNA_recognition_motif._(a.k.a._RRM_-_RBD_-_or_RNP_domain)/RNA_recognition_motif_(a.k.a._RRM_-_RBD_-_or_RNP_domain)_-_putative', 'interval': Interval(776492, 786702, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 776492, 'end': 786702, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_30T0027200', 'parent_id': 'LINF_300027200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RNA_recognition_motif._(a.k.a._RRM_-_RBD_-_or_RNP_domain)/RNA_recognition_motif_(a.k.a._RRM_-_RBD_-_or_RNP_domain)_-_putative', 'interval': Interval(776492, 786702, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 776492, 'end': 785021, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0027200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(776492, 785021, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.4000:\n",
      "\t{'chrom': 'LinJ.30', 'start': 940325, 'end': 943038, 'strand': '-', 'gene_id': 'LINF_300030650', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(940325, 943038, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 940325, 'end': 943038, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_30T0030650', 'parent_id': 'LINF_300030650', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(940325, 943038, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.4010:\n",
      "\t{'chrom': 'LinJ.30', 'start': 948227, 'end': 959563, 'strand': '-', 'gene_id': 'LINF_300030800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(948227, 959563, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 948227, 'end': 959563, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_30T0030800', 'parent_id': 'LINF_300030800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(948227, 959563, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 948227, 'end': 956908, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0030800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(948227, 956908, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.4020:\n",
      "\t{'chrom': 'LinJ.30', 'start': 948227, 'end': 959563, 'strand': '-', 'gene_id': 'LINF_300030800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(948227, 959563, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 948227, 'end': 959563, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_30T0030800', 'parent_id': 'LINF_300030800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(948227, 959563, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 948227, 'end': 956908, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0030800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(948227, 956908, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.4030:\n",
      "\t{'chrom': 'LinJ.30', 'start': 1007205, 'end': 1011606, 'strand': '-', 'gene_id': 'LINF_300032500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(1007205, 1011606, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 1007205, 'end': 1011606, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_30T0032500', 'parent_id': 'LINF_300032500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(1007205, 1011606, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 1007205, 'end': 1008651, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0032500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1007205, 1008651, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.4040:\n",
      "\t{'chrom': 'LinJ.30', 'start': 1016746, 'end': 1018962, 'strand': '-', 'gene_id': 'LINF_300032900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1016746, 1018962, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 1016746, 'end': 1018962, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_30T0032900', 'parent_id': 'LINF_300032900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1016746, 1018962, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 1016746, 'end': 1018508, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0032900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1016746, 1018508, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.4050:\n",
      "\t{'chrom': 'LinJ.30', 'start': 1132567, 'end': 1136504, 'strand': '-', 'gene_id': 'LINF_300036800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'S-adenosylmethionine_decarboxylase_proenzyme|ADOMETDC_prozyme', 'interval': Interval(1132567, 1136504, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 1132567, 'end': 1136504, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_30T0036800', 'parent_id': 'LINF_300036800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'S-adenosylmethionine_decarboxylase_proenzyme|ADOMETDC_prozyme', 'interval': Interval(1132567, 1136504, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 1132567, 'end': 1135219, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0036800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1132567, 1135219, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.4060:\n",
      "\t{'chrom': 'LinJ.30', 'start': 1132567, 'end': 1136504, 'strand': '-', 'gene_id': 'LINF_300036800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'S-adenosylmethionine_decarboxylase_proenzyme|ADOMETDC_prozyme', 'interval': Interval(1132567, 1136504, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 1132567, 'end': 1136504, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_30T0036800', 'parent_id': 'LINF_300036800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'S-adenosylmethionine_decarboxylase_proenzyme|ADOMETDC_prozyme', 'interval': Interval(1132567, 1136504, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 1132567, 'end': 1135219, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0036800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1132567, 1135219, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.4070:\n",
      "\t{'chrom': 'LinJ.30', 'start': 1221291, 'end': 1232212, 'strand': '+', 'gene_id': 'LINF_300039400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Zinc_finger_protein|ZC3H15', 'interval': Interval(1221291, 1232212, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 1221291, 'end': 1232212, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_30T0039400', 'parent_id': 'LINF_300039400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Zinc_finger_protein|ZC3H15', 'interval': Interval(1221291, 1232212, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 1225222, 'end': 1232212, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0039400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1225222, 1232212, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.4080:\n",
      "\t{'chrom': 'LinJ.30', 'start': 1295522, 'end': 1297484, 'strand': '+', 'gene_id': 'LINF_300041200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RF-1_domain_containing_protein_-_putative', 'interval': Interval(1295522, 1297484, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 1295522, 'end': 1297484, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_30T0041200', 'parent_id': 'LINF_300041200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RF-1_domain_containing_protein_-_putative', 'interval': Interval(1295522, 1297484, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 1295522, 'end': 1295780, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0041200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(1295522, 1295780, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.4090:\n",
      "\t{'chrom': 'LinJ.30', 'start': 1333478, 'end': 1340150, 'strand': '+', 'gene_id': 'LINF_300042700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_of_unknown_function_(DUF3808)', 'interval': Interval(1333478, 1340150, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 1333478, 'end': 1340150, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_30T0042700', 'parent_id': 'LINF_300042700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_of_unknown_function_(DUF3808)', 'interval': Interval(1333478, 1340150, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 1335691, 'end': 1340150, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0042700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1335691, 1340150, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4100W:\n",
      "\t{'chrom': 'LinJ.31', 'start': 3, 'end': 2694, 'strand': '-', 'gene_id': 'LINF_310004900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(3, 2694, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 3, 'end': 2694, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0004900', 'parent_id': 'LINF_310004900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(3, 2694, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4110:\n",
      "==================================================\n",
      "rejected_noCDS_c31.4120:\n",
      "==================================================\n",
      "rejected_noCDS_c31.4130:\n",
      "\t{'chrom': 'LinJ.31', 'start': 150123, 'end': 152831, 'strand': '-', 'gene_id': 'LINF_310009800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein', 'interval': Interval(150123, 152831, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 150123, 'end': 152831, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0009800', 'parent_id': 'LINF_310009800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein', 'interval': Interval(150123, 152831, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 150123, 'end': 152088, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0009800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(150123, 152088, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4140:\n",
      "==================================================\n",
      "rejected_noCDS_c31.4150:\n",
      "\t{'chrom': 'LinJ.31', 'start': 153423, 'end': 156946, 'strand': '-', 'gene_id': 'LINF_310009850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': 'unknown', 'notes': 'tuzin_like_protein', 'interval': Interval(153423, 156946, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 153423, 'end': 156946, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0009850', 'parent_id': 'LINF_310009850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'tuzin_like_protein', 'interval': Interval(153423, 156946, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4160:\n",
      "\t{'chrom': 'LinJ.31', 'start': 253033, 'end': 254591, 'strand': '-', 'gene_id': 'LINF_310012730', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(253033, 254591, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 253033, 'end': 254591, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0012730', 'parent_id': 'LINF_310012730', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(253033, 254591, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4170:\n",
      "\t{'chrom': 'LinJ.31', 'start': 299542, 'end': 304099, 'strand': '-', 'gene_id': 'LINF_310013900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(299542, 304099, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 299542, 'end': 304099, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0013900', 'parent_id': 'LINF_310013900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(299542, 304099, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 302099, 'end': 304099, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0013900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(302099, 304099, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4180:\n",
      "==================================================\n",
      "rejected_noCDS_c31.4190BF:\n",
      "\t{'chrom': 'LinJ.31', 'start': 317136, 'end': 317964, 'strand': '-', 'gene_id': 'LINF_310014250', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(317136, 317964, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 317136, 'end': 317964, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0014250', 'parent_id': 'LINF_310014250', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(317136, 317964, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4200BF:\n",
      "\t{'chrom': 'LinJ.31', 'start': 322998, 'end': 328575, 'strand': '-', 'gene_id': 'LINF_310014500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amino_acid_permease_3|arginine_transporter|AAP3|aap3_4.7', 'interval': Interval(322998, 328575, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 322998, 'end': 328575, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0014500', 'parent_id': 'LINF_310014500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amino_acid_permease_3|arginine_transporter|AAP3|aap3_4.7', 'interval': Interval(322998, 328575, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 322998, 'end': 326836, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0014500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(322998, 326836, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4210BF:\n",
      "\t{'chrom': 'LinJ.31', 'start': 405961, 'end': 406903, 'strand': '-', 'gene_id': 'LINF_310016350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(405961, 406903, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 405961, 'end': 406903, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0016350', 'parent_id': 'LINF_310016350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(405961, 406903, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4220:\n",
      "==================================================\n",
      "rejected_noCDS_c31.4230BF:\n",
      "==================================================\n",
      "rejected_noCDS_c31.4240BG:\n",
      "==================================================\n",
      "rejected_noCDS_c31.4250BH:\n",
      "\t{'chrom': 'LinJ.31', 'start': 440694, 'end': 447113, 'strand': '-', 'gene_id': 'LINF_310017000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(440694, 447113, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 440694, 'end': 447113, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0017000', 'parent_id': 'LINF_310017000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(440694, 447113, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 440694, 'end': 442935, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0017000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(440694, 442935, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4260BF:\n",
      "\t{'chrom': 'LinJ.31', 'start': 440694, 'end': 447113, 'strand': '-', 'gene_id': 'LINF_310017000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(440694, 447113, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 440694, 'end': 447113, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0017000', 'parent_id': 'LINF_310017000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(440694, 447113, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 440694, 'end': 442935, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0017000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(440694, 442935, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4270BH:\n",
      "\t{'chrom': 'LinJ.31', 'start': 462000, 'end': 463926, 'strand': '-', 'gene_id': 'LINF_310017800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'monoglyceride_lipase', 'interval': Interval(462000, 463926, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 462000, 'end': 463926, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0017800', 'parent_id': 'LINF_310017800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'monoglyceride_lipase', 'interval': Interval(462000, 463926, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 462000, 'end': 462786, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0017800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(462000, 462786, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4280BG:\n",
      "\t{'chrom': 'LinJ.31', 'start': 500464, 'end': 506449, 'strand': '-', 'gene_id': 'LINF_310018700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'H(+)-exporting_diphosphatase', 'interval': Interval(500464, 506449, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 500464, 'end': 506449, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0018700', 'parent_id': 'LINF_310018700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'H(+)-exporting_diphosphatase', 'interval': Interval(500464, 506449, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 500464, 'end': 503903, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0018700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(500464, 503903, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4290BF:\n",
      "\t{'chrom': 'LinJ.31', 'start': 500464, 'end': 506449, 'strand': '-', 'gene_id': 'LINF_310018700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'H(+)-exporting_diphosphatase', 'interval': Interval(500464, 506449, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 500464, 'end': 506449, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0018700', 'parent_id': 'LINF_310018700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'H(+)-exporting_diphosphatase', 'interval': Interval(500464, 506449, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 500464, 'end': 503903, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0018700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(500464, 503903, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4300:\n",
      "\t{'chrom': 'LinJ.31', 'start': 509624, 'end': 510846, 'strand': '-', 'gene_id': 'LINF_310018950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(509624, 510846, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 509624, 'end': 510846, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0018950', 'parent_id': 'LINF_310018950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(509624, 510846, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4310:\n",
      "==================================================\n",
      "rejected_noCDS_c31.4320:\n",
      "\t{'chrom': 'LinJ.31', 'start': 645169, 'end': 657844, 'strand': '-', 'gene_id': 'LINF_310020900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(645169, 657844, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 645169, 'end': 657844, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0020900', 'parent_id': 'LINF_310020900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(645169, 657844, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 645169, 'end': 648937, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0020900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(645169, 648937, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4330:\n",
      "\t{'chrom': 'LinJ.31', 'start': 658211, 'end': 665517, 'strand': '-', 'gene_id': 'LINF_310021000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(658211, 665517, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 658211, 'end': 665517, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0021000', 'parent_id': 'LINF_310021000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(658211, 665517, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 658211, 'end': 661621, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0021000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(658211, 661621, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4340:\n",
      "\t{'chrom': 'LinJ.31', 'start': 751329, 'end': 760680, 'strand': '-', 'gene_id': 'LINF_310022400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(751329, 760680, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 751329, 'end': 760680, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0022400', 'parent_id': 'LINF_310022400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(751329, 760680, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 751329, 'end': 753145, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0022400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(751329, 753145, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4350BI:\n",
      "\t{'chrom': 'LinJ.31', 'start': 786541, 'end': 795054, 'strand': '-', 'gene_id': 'LINF_310022700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(786541, 795054, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 786541, 'end': 795054, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0022700', 'parent_id': 'LINF_310022700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(786541, 795054, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 786541, 'end': 793015, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0022700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(786541, 793015, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4360:\n",
      "\t{'chrom': 'LinJ.31', 'start': 786541, 'end': 795054, 'strand': '-', 'gene_id': 'LINF_310022700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(786541, 795054, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 786541, 'end': 795054, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0022700', 'parent_id': 'LINF_310022700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(786541, 795054, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 786541, 'end': 793015, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0022700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(786541, 793015, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4370:\n",
      "\t{'chrom': 'LinJ.31', 'start': 850219, 'end': 851351, 'strand': '-', 'gene_id': 'LINF_310024830', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(850219, 851351, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 850219, 'end': 851351, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0024830', 'parent_id': 'LINF_310024830', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(850219, 851351, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4380BI:\n",
      "\t{'chrom': 'LinJ.31', 'start': 855264, 'end': 861966, 'strand': '-', 'gene_id': 'LINF_310025000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(855264, 861966, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 855264, 'end': 861966, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0025000', 'parent_id': 'LINF_310025000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(855264, 861966, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 855264, 'end': 860716, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0025000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(855264, 860716, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4390:\n",
      "\t{'chrom': 'LinJ.31', 'start': 855264, 'end': 861966, 'strand': '-', 'gene_id': 'LINF_310025000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(855264, 861966, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 855264, 'end': 861966, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0025000', 'parent_id': 'LINF_310025000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(855264, 861966, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 855264, 'end': 860716, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0025000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(855264, 860716, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4400:\n",
      "==================================================\n",
      "rejected_noCDS_c31.4410BJ:\n",
      "\t{'chrom': 'LinJ.31', 'start': 967493, 'end': 971919, 'strand': '-', 'gene_id': 'LINF_310026700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'peptidase_m20/m25/m40_family-like_protein', 'interval': Interval(967493, 971919, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 967493, 'end': 971919, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0026700', 'parent_id': 'LINF_310026700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'peptidase_m20/m25/m40_family-like_protein', 'interval': Interval(967493, 971919, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 967493, 'end': 970298, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0026700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(967493, 970298, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4420BK:\n",
      "\t{'chrom': 'LinJ.31', 'start': 973364, 'end': 979320, 'strand': '-', 'gene_id': 'LINF_310026900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(973364, 979320, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 973364, 'end': 979320, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0026900', 'parent_id': 'LINF_310026900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(973364, 979320, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 973364, 'end': 975996, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0026900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(973364, 975996, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4430BK:\n",
      "\t{'chrom': 'LinJ.31', 'start': 979748, 'end': 981420, 'strand': '-', 'gene_id': 'LINF_310026950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(979748, 981420, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 979748, 'end': 981420, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0026950', 'parent_id': 'LINF_310026950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(979748, 981420, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4440:\n",
      "\t{'chrom': 'LinJ.31', 'start': 981791, 'end': 987828, 'strand': '-', 'gene_id': 'LINF_310027000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RING-variant_domain_containing_protein_-_putative', 'interval': Interval(981791, 987828, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 981791, 'end': 987828, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0027000', 'parent_id': 'LINF_310027000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RING-variant_domain_containing_protein_-_putative', 'interval': Interval(981791, 987828, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 981791, 'end': 983117, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0027000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(981791, 983117, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4450BK:\n",
      "\t{'chrom': 'LinJ.31', 'start': 991470, 'end': 996088, 'strand': '-', 'gene_id': 'LINF_310027200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'transcription_like_protein_nupm1', 'interval': Interval(991470, 996088, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 991470, 'end': 996088, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0027200', 'parent_id': 'LINF_310027200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'transcription_like_protein_nupm1', 'interval': Interval(991470, 996088, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 991470, 'end': 993701, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0027200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(991470, 993701, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4460BJ:\n",
      "\t{'chrom': 'LinJ.31', 'start': 1029887, 'end': 1033101, 'strand': '-', 'gene_id': 'LINF_310028200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'peptidase_m20/m25/m40_family-like_protein', 'interval': Interval(1029887, 1033101, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 1029887, 'end': 1033101, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0028200', 'parent_id': 'LINF_310028200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'peptidase_m20/m25/m40_family-like_protein', 'interval': Interval(1029887, 1033101, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 1029887, 'end': 1031480, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0028200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1029887, 1031480, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4470:\n",
      "==================================================\n",
      "rejected_noCDS_c31.4480:\n",
      "\t{'chrom': 'LinJ.31', 'start': 1159458, 'end': 1161477, 'strand': '-', 'gene_id': 'LINF_310030550', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1159458, 1161477, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 1159458, 'end': 1161477, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0030550', 'parent_id': 'LINF_310030550', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1159458, 1161477, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4490:\n",
      "\t{'chrom': 'LinJ.31', 'start': 1169541, 'end': 1173343, 'strand': '-', 'gene_id': 'LINF_310031000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ADP-ribosylation_factor|ARF1', 'interval': Interval(1169541, 1173343, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 1169541, 'end': 1173343, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0031000', 'parent_id': 'LINF_310031000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ADP-ribosylation_factor|ARF1', 'interval': Interval(1169541, 1173343, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 1169541, 'end': 1172681, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0031000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1169541, 1172681, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4500:\n",
      "\t{'chrom': 'LinJ.31', 'start': 1192515, 'end': 1193940, 'strand': '-', 'gene_id': 'LINF_310031550', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1192515, 1193940, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 1192515, 'end': 1193940, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0031550', 'parent_id': 'LINF_310031550', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1192515, 1193940, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4510:\n",
      "\t{'chrom': 'LinJ.31', 'start': 1195831, 'end': 1201297, 'strand': '-', 'gene_id': 'LINF_310031700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'p-nitrophenylphosphatase', 'interval': Interval(1195831, 1201297, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 1195831, 'end': 1201297, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0031700', 'parent_id': 'LINF_310031700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'p-nitrophenylphosphatase', 'interval': Interval(1195831, 1201297, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 1195831, 'end': 1200124, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0031700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1195831, 1200124, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4520:\n",
      "\t{'chrom': 'LinJ.31', 'start': 1195831, 'end': 1201297, 'strand': '-', 'gene_id': 'LINF_310031700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'p-nitrophenylphosphatase', 'interval': Interval(1195831, 1201297, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 1195831, 'end': 1201297, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0031700', 'parent_id': 'LINF_310031700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'p-nitrophenylphosphatase', 'interval': Interval(1195831, 1201297, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 1195831, 'end': 1200124, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0031700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1195831, 1200124, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4530:\n",
      "\t{'chrom': 'LinJ.31', 'start': 1201569, 'end': 1205179, 'strand': '-', 'gene_id': 'LINF_310031725', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1201569, 1205179, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 1201569, 'end': 1205179, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0031725', 'parent_id': 'LINF_310031725', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1201569, 1205179, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4540:\n",
      "\t{'chrom': 'LinJ.31', 'start': 1205762, 'end': 1208156, 'strand': '-', 'gene_id': 'LINF_310031750', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1205762, 1208156, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 1205762, 'end': 1208156, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0031750', 'parent_id': 'LINF_310031750', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1205762, 1208156, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 1205762, 'end': 1207023, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0031750', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1205762, 1207023, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4550:\n",
      "==================================================\n",
      "rejected_noCDS_c31.4560:\n",
      "\t{'chrom': 'LinJ.31', 'start': 1419515, 'end': 1421616, 'strand': '-', 'gene_id': 'LINF_310037900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1419515, 1421616, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 1419515, 'end': 1421616, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0037900', 'parent_id': 'LINF_310037900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1419515, 1421616, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 1419515, 'end': 1420693, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0037900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1419515, 1420693, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4570O:\n",
      "==================================================\n",
      "rejected_noCDS_c32.4580:\n",
      "\t{'chrom': 'LinJ.32', 'start': 239876, 'end': 246114, 'strand': '+', 'gene_id': 'LINF_320012400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ubiquitin-conjugating_enzyme_E2|UBC8', 'interval': Interval(239876, 246114, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 239876, 'end': 246114, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_32T0012400', 'parent_id': 'LINF_320012400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ubiquitin-conjugating_enzyme_E2|UBC8', 'interval': Interval(239876, 246114, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 241498, 'end': 246114, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_32T0012400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(241498, 246114, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4590:\n",
      "\t{'chrom': 'LinJ.32', 'start': 239876, 'end': 246114, 'strand': '+', 'gene_id': 'LINF_320012400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ubiquitin-conjugating_enzyme_E2|UBC8', 'interval': Interval(239876, 246114, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 239876, 'end': 246114, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_32T0012400', 'parent_id': 'LINF_320012400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ubiquitin-conjugating_enzyme_E2|UBC8', 'interval': Interval(239876, 246114, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 241498, 'end': 246114, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_32T0012400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(241498, 246114, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4600BL:\n",
      "\t{'chrom': 'LinJ.32', 'start': 345025, 'end': 351432, 'strand': '+', 'gene_id': 'LINF_320014830', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(345025, 351432, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 345025, 'end': 351432, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_32T0014830', 'parent_id': 'LINF_320014830', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(345025, 351432, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4610BM:\n",
      "\t{'chrom': 'LinJ.32', 'start': 345025, 'end': 351432, 'strand': '+', 'gene_id': 'LINF_320014830', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(345025, 351432, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 345025, 'end': 351432, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_32T0014830', 'parent_id': 'LINF_320014830', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(345025, 351432, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4620BL:\n",
      "\t{'chrom': 'LinJ.32', 'start': 374515, 'end': 391521, 'strand': '+', 'gene_id': 'LINF_320015400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(374515, 391521, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 374515, 'end': 391521, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_32T0015400', 'parent_id': 'LINF_320015400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(374515, 391521, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 380402, 'end': 391521, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_32T0015400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(380402, 391521, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4630BM:\n",
      "\t{'chrom': 'LinJ.32', 'start': 374515, 'end': 391521, 'strand': '+', 'gene_id': 'LINF_320015400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(374515, 391521, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 374515, 'end': 391521, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_32T0015400', 'parent_id': 'LINF_320015400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(374515, 391521, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 380402, 'end': 391521, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_32T0015400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(380402, 391521, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4640BL:\n",
      "\t{'chrom': 'LinJ.32', 'start': 391856, 'end': 395482, 'strand': '+', 'gene_id': 'LINF_320015500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Vps23_core_domain_containing_protein|Li1040', 'interval': Interval(391856, 395482, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 391856, 'end': 395482, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_32T0015500', 'parent_id': 'LINF_320015500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Vps23_core_domain_containing_protein|Li1040', 'interval': Interval(391856, 395482, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 393597, 'end': 395482, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_32T0015500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(393597, 395482, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4650BM:\n",
      "\t{'chrom': 'LinJ.32', 'start': 391856, 'end': 395482, 'strand': '+', 'gene_id': 'LINF_320015500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Vps23_core_domain_containing_protein|Li1040', 'interval': Interval(391856, 395482, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 391856, 'end': 395482, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_32T0015500', 'parent_id': 'LINF_320015500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Vps23_core_domain_containing_protein|Li1040', 'interval': Interval(391856, 395482, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 393597, 'end': 395482, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_32T0015500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(393597, 395482, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4660BN:\n",
      "\t{'chrom': 'LinJ.32', 'start': 514098, 'end': 517396, 'strand': '+', 'gene_id': 'LINF_320019000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'peptidase_PPPDE4', 'interval': Interval(514098, 517396, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 514098, 'end': 517396, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_32T0019000', 'parent_id': 'LINF_320019000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'peptidase_PPPDE4', 'interval': Interval(514098, 517396, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 516117, 'end': 517396, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_32T0019000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(516117, 517396, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4670BN:\n",
      "\t{'chrom': 'LinJ.32', 'start': 538303, 'end': 541717, 'strand': '-', 'gene_id': 'LINF_320019500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(538303, 541717, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 538303, 'end': 541717, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_32T0019500', 'parent_id': 'LINF_320019500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(538303, 541717, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 538303, 'end': 540475, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_32T0019500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(538303, 540475, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4680:\n",
      "\t{'chrom': 'LinJ.32', 'start': 656828, 'end': 662803, 'strand': '-', 'gene_id': 'LINF_320022600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_phosphatase_2C_-_putative', 'interval': Interval(656828, 662803, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 656828, 'end': 662803, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_32T0022600', 'parent_id': 'LINF_320022600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_phosphatase_2C_-_putative', 'interval': Interval(656828, 662803, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 656828, 'end': 659442, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_32T0022600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(656828, 659442, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4690:\n",
      "==================================================\n",
      "rejected_noCDS_c32.4700:\n",
      "\t{'chrom': 'LinJ.32', 'start': 768213, 'end': 771091, 'strand': '-', 'gene_id': 'LINF_320025600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(768213, 771091, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 768213, 'end': 771091, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_32T0025600', 'parent_id': 'LINF_320025600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(768213, 771091, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 768213, 'end': 769143, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_32T0025600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(768213, 769143, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4710BO:\n",
      "==================================================\n",
      "rejected_noCDS_c32.4720BO:\n",
      "\t{'chrom': 'LinJ.32', 'start': 896918, 'end': 901896, 'strand': '-', 'gene_id': 'LINF_320029300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Sphingosine-1-phosphate_phosphatase', 'interval': Interval(896918, 901896, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 896918, 'end': 901896, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_32T0029300', 'parent_id': 'LINF_320029300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Sphingosine-1-phosphate_phosphatase', 'interval': Interval(896918, 901896, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 896918, 'end': 898981, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_32T0029300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(896918, 898981, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4730:\n",
      "\t{'chrom': 'LinJ.32', 'start': 965256, 'end': 970325, 'strand': '-', 'gene_id': 'LINF_320031600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(965256, 970325, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 965256, 'end': 970325, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_32T0031600', 'parent_id': 'LINF_320031600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(965256, 970325, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 965256, 'end': 969846, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_32T0031600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(965256, 969846, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4740:\n",
      "\t{'chrom': 'LinJ.32', 'start': 1025806, 'end': 1030115, 'strand': '-', 'gene_id': 'LINF_320033250', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1025806, 1030115, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 1025806, 'end': 1030115, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_32T0033250', 'parent_id': 'LINF_320033250', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1025806, 1030115, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4750:\n",
      "\t{'chrom': 'LinJ.32', 'start': 1122643, 'end': 1126624, 'strand': '-', 'gene_id': 'LINF_320035800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Phosphatidate_cytidylyltransferase', 'interval': Interval(1122643, 1126624, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 1122643, 'end': 1126624, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_32T0035800', 'parent_id': 'LINF_320035800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Phosphatidate_cytidylyltransferase', 'interval': Interval(1122643, 1126624, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 1122643, 'end': 1124698, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_32T0035800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1122643, 1124698, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4760:\n",
      "\t{'chrom': 'LinJ.32', 'start': 1265342, 'end': 1282221, 'strand': '+', 'gene_id': 'LINF_320039700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1265342, 1282221, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 1265342, 'end': 1282221, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_32T0039700', 'parent_id': 'LINF_320039700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1265342, 1282221, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 1273960, 'end': 1282221, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_32T0039700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1273960, 1282221, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4770:\n",
      "\t{'chrom': 'LinJ.32', 'start': 1539270, 'end': 1541079, 'strand': '+', 'gene_id': 'LINF_320047040', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Esterase', 'interval': Interval(1539270, 1541079, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 1539270, 'end': 1541079, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_32T0047040', 'parent_id': 'LINF_320047040', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Esterase', 'interval': Interval(1539270, 1541079, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4780:\n",
      "\t{'chrom': 'LinJ.33', 'start': 55248, 'end': 63390, 'strand': '-', 'gene_id': 'LINF_330007200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(55248, 63390, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 55248, 'end': 63390, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_33T0007200', 'parent_id': 'LINF_330007200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(55248, 63390, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 55248, 'end': 59816, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_33T0007200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(55248, 59816, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4790:\n",
      "\t{'chrom': 'LinJ.33', 'start': 81344, 'end': 85743, 'strand': '-', 'gene_id': 'LINF_330007900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Peroxisomal_membrane_protein', 'interval': Interval(81344, 85743, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 81344, 'end': 85743, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_33T0007900', 'parent_id': 'LINF_330007900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Peroxisomal_membrane_protein', 'interval': Interval(81344, 85743, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 81344, 'end': 84280, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_33T0007900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(81344, 84280, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4800:\n",
      "\t{'chrom': 'LinJ.33', 'start': 81344, 'end': 85743, 'strand': '-', 'gene_id': 'LINF_330007900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Peroxisomal_membrane_protein', 'interval': Interval(81344, 85743, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 81344, 'end': 85743, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_33T0007900', 'parent_id': 'LINF_330007900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Peroxisomal_membrane_protein', 'interval': Interval(81344, 85743, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 81344, 'end': 84280, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_33T0007900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(81344, 84280, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4810:\n",
      "\t{'chrom': 'LinJ.33', 'start': 221375, 'end': 224292, 'strand': '-', 'gene_id': 'LINF_330012200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'tRNA-splicing_ligase_RtcB_-_putative', 'interval': Interval(221375, 224292, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 221375, 'end': 224292, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_33T0012200', 'parent_id': 'LINF_330012200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'tRNA-splicing_ligase_RtcB_-_putative', 'interval': Interval(221375, 224292, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 221375, 'end': 222727, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_33T0012200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(221375, 222727, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4820:\n",
      "\t{'chrom': 'LinJ.33', 'start': 341332, 'end': 345439, 'strand': '+', 'gene_id': 'LINF_330015600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': '2,4-dienoyl-CoA_reductase', 'interval': Interval(341332, 345439, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 341332, 'end': 345439, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0015600', 'parent_id': 'LINF_330015600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': '2,4-dienoyl-CoA_reductase', 'interval': Interval(341332, 345439, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 343658, 'end': 345439, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_33T0015600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(343658, 345439, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4830:\n",
      "\t{'chrom': 'LinJ.33', 'start': 415430, 'end': 428290, 'strand': '+', 'gene_id': 'LINF_330017600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': '\\xa0PSP1-like_protein', 'interval': Interval(415430, 428290, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 415430, 'end': 428290, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0017600', 'parent_id': 'LINF_330017600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': '\\xa0PSP1-like_protein', 'interval': Interval(415430, 428290, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 426033, 'end': 428290, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_33T0017600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(426033, 428290, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4840:\n",
      "\t{'chrom': 'LinJ.33', 'start': 488917, 'end': 491391, 'strand': '+', 'gene_id': 'LINF_330019200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Succinate_dehydrogenase_subunit_4|SDH4', 'interval': Interval(488917, 491391, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 488917, 'end': 491391, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0019200', 'parent_id': 'LINF_330019200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Succinate_dehydrogenase_subunit_4|SDH4', 'interval': Interval(488917, 491391, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 489579, 'end': 491391, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_33T0019200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(489579, 491391, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4850:\n",
      "==================================================\n",
      "rejected_noCDS_c33.4860:\n",
      "\t{'chrom': 'LinJ.33', 'start': 609978, 'end': 620397, 'strand': '+', 'gene_id': 'LINF_330022800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RNA_recognition_motif._(a.k.a._RRM_-_RBD_-_or_RNP_domain)/RNA_recognition_motif_(a.k.a._RRM_-_RBD_-_or_RNP_domain)_-_putative', 'interval': Interval(609978, 620397, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 609978, 'end': 620397, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0022800', 'parent_id': 'LINF_330022800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RNA_recognition_motif._(a.k.a._RRM_-_RBD_-_or_RNP_domain)/RNA_recognition_motif_(a.k.a._RRM_-_RBD_-_or_RNP_domain)_-_putative', 'interval': Interval(609978, 620397, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 611038, 'end': 620397, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_33T0022800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(611038, 620397, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4870:\n",
      "\t{'chrom': 'LinJ.33', 'start': 680400, 'end': 681392, 'strand': '+', 'gene_id': 'LINF_330023450', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(680400, 681392, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 680400, 'end': 681392, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0023450', 'parent_id': 'LINF_330023450', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(680400, 681392, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4880:\n",
      "==================================================\n",
      "rejected_noCDS_c33.4890BC:\n",
      "==================================================\n",
      "rejected_noCDS_c33.4900:\n",
      "\t{'chrom': 'LinJ.33', 'start': 1085307, 'end': 1089417, 'strand': '+', 'gene_id': 'LINF_330034700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Isocitrate_dehydrogenase|IDH-50', 'interval': Interval(1085307, 1089417, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1085307, 'end': 1089417, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0034700', 'parent_id': 'LINF_330034700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Isocitrate_dehydrogenase|IDH-50', 'interval': Interval(1085307, 1089417, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1086964, 'end': 1089417, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_33T0034700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1086964, 1089417, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4910:\n",
      "==================================================\n",
      "rejected_noCDS_c33.4920BP:\n",
      "\t{'chrom': 'LinJ.33', 'start': 1156039, 'end': 1172306, 'strand': '+', 'gene_id': 'LINF_330036550', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1156039, 1172306, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1156039, 'end': 1172306, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0036550', 'parent_id': 'LINF_330036550', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1156039, 1172306, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4930BP:\n",
      "\t{'chrom': 'LinJ.33', 'start': 1209450, 'end': 1213712, 'strand': '+', 'gene_id': 'LINF_330037200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'GTP-binding_elongation_factor_tu_family_protein_-_putative', 'interval': Interval(1209450, 1213712, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1209450, 'end': 1213712, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0037200', 'parent_id': 'LINF_330037200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'GTP-binding_elongation_factor_tu_family_protein_-_putative', 'interval': Interval(1209450, 1213712, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1212542, 'end': 1213712, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_33T0037200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1212542, 1213712, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4940BQ:\n",
      "==================================================\n",
      "rejected_noCDS_c33.4950BP:\n",
      "\t{'chrom': 'LinJ.33', 'start': 1261542, 'end': 1263979, 'strand': '+', 'gene_id': 'LINF_330038475', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1261542, 1263979, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1261542, 'end': 1263979, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0038475', 'parent_id': 'LINF_330038475', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1261542, 1263979, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4960BQ:\n",
      "==================================================\n",
      "rejected_noCDS_c33.4970BP:\n",
      "\t{'chrom': 'LinJ.33', 'start': 1279488, 'end': 1286836, 'strand': '+', 'gene_id': 'LINF_330038590', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1279488, 1286836, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1279488, 'end': 1286836, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0038590', 'parent_id': 'LINF_330038590', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1279488, 1286836, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4980:\n",
      "\t{'chrom': 'LinJ.33', 'start': 1335137, 'end': 1342813, 'strand': '+', 'gene_id': 'LINF_330039000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(1335137, 1342813, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1335137, 'end': 1342813, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0039000', 'parent_id': 'LINF_330039000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(1335137, 1342813, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1336830, 'end': 1342813, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_33T0039000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1336830, 1342813, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4990:\n",
      "\t{'chrom': 'LinJ.33', 'start': 1384772, 'end': 1390052, 'strand': '+', 'gene_id': 'LINF_330039700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1384772, 1390052, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1384772, 'end': 1390052, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0039700', 'parent_id': 'LINF_330039700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1384772, 1390052, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1386156, 'end': 1390052, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_33T0039700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1386156, 1390052, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.5000:\n",
      "\t{'chrom': 'LinJ.33', 'start': 1396085, 'end': 1399188, 'strand': '+', 'gene_id': 'LINF_330039850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1396085, 1399188, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1396085, 'end': 1399188, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0039850', 'parent_id': 'LINF_330039850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1396085, 1399188, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.5010:\n",
      "\t{'chrom': 'LinJ.33', 'start': 1396085, 'end': 1399188, 'strand': '+', 'gene_id': 'LINF_330039850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1396085, 1399188, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1396085, 'end': 1399188, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0039850', 'parent_id': 'LINF_330039850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1396085, 1399188, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.5020:\n",
      "\t{'chrom': 'LinJ.33', 'start': 1407699, 'end': 1416872, 'strand': '+', 'gene_id': 'LINF_330040000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1407699, 1416872, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1407699, 'end': 1416872, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0040000', 'parent_id': 'LINF_330040000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1407699, 1416872, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1412018, 'end': 1416872, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_33T0040000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1412018, 1416872, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.5030AK:\n",
      "==================================================\n",
      "rejected_noCDS_c33.5040AK:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5050:\n",
      "\t{'chrom': 'LinJ.34', 'start': 197352, 'end': 200525, 'strand': '+', 'gene_id': 'LINF_340010350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(197352, 200525, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 197352, 'end': 200525, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0010350', 'parent_id': 'LINF_340010350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(197352, 200525, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5060:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5070:\n",
      "\t{'chrom': 'LinJ.34', 'start': 363327, 'end': 367140, 'strand': '+', 'gene_id': 'LINF_340013700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Serine/threonine-protein_phosphatase_PP1|PP1-6c', 'interval': Interval(363327, 367140, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 363327, 'end': 367140, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0013700', 'parent_id': 'LINF_340013700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Serine/threonine-protein_phosphatase_PP1|PP1-6c', 'interval': Interval(363327, 367140, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 365214, 'end': 367140, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0013700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(365214, 367140, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5080BS:\n",
      "\t{'chrom': 'LinJ.34', 'start': 422379, 'end': 423723, 'strand': '+', 'gene_id': 'LINF_340015350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(422379, 423723, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 422379, 'end': 423723, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0015350', 'parent_id': 'LINF_340015350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(422379, 423723, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5090BR:\n",
      "\t{'chrom': 'LinJ.34', 'start': 423724, 'end': 426392, 'strand': '+', 'gene_id': 'LINF_340015400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_surface_protein_-_putative', 'interval': Interval(423724, 426392, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 423724, 'end': 426392, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0015400', 'parent_id': 'LINF_340015400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_surface_protein_-_putative', 'interval': Interval(423724, 426392, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 424495, 'end': 426392, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0015400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(424495, 426392, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5100BR:\n",
      "\t{'chrom': 'LinJ.34', 'start': 428116, 'end': 431093, 'strand': '+', 'gene_id': 'LINF_340015450', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(428116, 431093, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 428116, 'end': 431093, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0015450', 'parent_id': 'LINF_340015450', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(428116, 431093, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5110BR:\n",
      "\t{'chrom': 'LinJ.34', 'start': 432504, 'end': 435210, 'strand': '+', 'gene_id': 'LINF_340015500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_protein', 'interval': Interval(432504, 435210, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 432504, 'end': 435210, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0015500', 'parent_id': 'LINF_340015500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_protein', 'interval': Interval(432504, 435210, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 433308, 'end': 435210, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0015500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(433308, 435210, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5120BR:\n",
      "\t{'chrom': 'LinJ.34', 'start': 445685, 'end': 448392, 'strand': '+', 'gene_id': 'LINF_340015800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_protein', 'interval': Interval(445685, 448392, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 445685, 'end': 448392, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0015800', 'parent_id': 'LINF_340015800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_protein', 'interval': Interval(445685, 448392, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 446495, 'end': 448392, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0015800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(446495, 448392, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5130BR:\n",
      "\t{'chrom': 'LinJ.34', 'start': 450104, 'end': 453599, 'strand': '+', 'gene_id': 'LINF_340015900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_protein', 'interval': Interval(450104, 453599, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 450104, 'end': 453599, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0015900', 'parent_id': 'LINF_340015900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_protein', 'interval': Interval(450104, 453599, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 450914, 'end': 453599, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0015900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(450914, 453599, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5140BS:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5150BT:\n",
      "\t{'chrom': 'LinJ.34', 'start': 557626, 'end': 562306, 'strand': '-', 'gene_id': 'LINF_340018600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(557626, 562306, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 557626, 'end': 562306, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0018600', 'parent_id': 'LINF_340018600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(557626, 562306, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 557626, 'end': 559478, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0018600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(557626, 559478, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5160:\n",
      "\t{'chrom': 'LinJ.34', 'start': 686693, 'end': 692468, 'strand': '-', 'gene_id': 'LINF_340021600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(686693, 692468, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 686693, 'end': 692468, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0021600', 'parent_id': 'LINF_340021600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(686693, 692468, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 686693, 'end': 688437, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0021600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(686693, 688437, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5170:\n",
      "\t{'chrom': 'LinJ.34', 'start': 735203, 'end': 739818, 'strand': '-', 'gene_id': 'LINF_340022850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(735203, 739818, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 735203, 'end': 739818, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0022850', 'parent_id': 'LINF_340022850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(735203, 739818, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5180BS:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5190:\n",
      "\t{'chrom': 'LinJ.34', 'start': 739819, 'end': 743551, 'strand': '-', 'gene_id': 'LINF_340022900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_protein', 'interval': Interval(739819, 743551, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 739819, 'end': 743551, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0022900', 'parent_id': 'LINF_340022900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_protein', 'interval': Interval(739819, 743551, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 739819, 'end': 742807, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0022900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(739819, 742807, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5200BR:\n",
      "\t{'chrom': 'LinJ.34', 'start': 739819, 'end': 743551, 'strand': '-', 'gene_id': 'LINF_340022900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_protein', 'interval': Interval(739819, 743551, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 739819, 'end': 743551, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0022900', 'parent_id': 'LINF_340022900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_protein', 'interval': Interval(739819, 743551, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 739819, 'end': 742807, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0022900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(739819, 742807, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5210BT:\n",
      "\t{'chrom': 'LinJ.34', 'start': 754280, 'end': 757294, 'strand': '-', 'gene_id': 'LINF_340023220', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(754280, 757294, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 754280, 'end': 757294, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023220', 'parent_id': 'LINF_340023220', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(754280, 757294, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5220BT:\n",
      "\t{'chrom': 'LinJ.34', 'start': 754280, 'end': 757294, 'strand': '-', 'gene_id': 'LINF_340023220', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(754280, 757294, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 754280, 'end': 757294, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023220', 'parent_id': 'LINF_340023220', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(754280, 757294, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5230BU:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5240BT:\n",
      "\t{'chrom': 'LinJ.34', 'start': 758277, 'end': 761292, 'strand': '-', 'gene_id': 'LINF_340023260', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(758277, 761292, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 758277, 'end': 761292, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023260', 'parent_id': 'LINF_340023260', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(758277, 761292, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5250BT:\n",
      "\t{'chrom': 'LinJ.34', 'start': 758277, 'end': 761292, 'strand': '-', 'gene_id': 'LINF_340023260', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(758277, 761292, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 758277, 'end': 761292, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023260', 'parent_id': 'LINF_340023260', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(758277, 761292, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5260BU:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5270BT:\n",
      "\t{'chrom': 'LinJ.34', 'start': 761293, 'end': 765197, 'strand': '-', 'gene_id': 'LINF_340023280', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(761293, 765197, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 761293, 'end': 765197, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023280', 'parent_id': 'LINF_340023280', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(761293, 765197, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5280BT:\n",
      "\t{'chrom': 'LinJ.34', 'start': 761293, 'end': 765197, 'strand': '-', 'gene_id': 'LINF_340023280', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(761293, 765197, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 761293, 'end': 765197, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023280', 'parent_id': 'LINF_340023280', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(761293, 765197, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5290BU:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5300BT:\n",
      "\t{'chrom': 'LinJ.34', 'start': 772123, 'end': 775066, 'strand': '-', 'gene_id': 'LINF_340023415', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(772123, 775066, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 772123, 'end': 775066, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023415', 'parent_id': 'LINF_340023415', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(772123, 775066, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5310BU:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5320BT:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5330BU:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5340BU:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5350BU:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5360BT:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5370BU:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5380BU:\n",
      "\t{'chrom': 'LinJ.34', 'start': 805687, 'end': 808315, 'strand': '-', 'gene_id': 'LINF_340024200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(805687, 808315, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 805687, 'end': 808315, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0024200', 'parent_id': 'LINF_340024200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(805687, 808315, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 805687, 'end': 807555, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0024200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(805687, 807555, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5390BU:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5400BT:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5410BT:\n",
      "\t{'chrom': 'LinJ.34', 'start': 814218, 'end': 816590, 'strand': '-', 'gene_id': 'LINF_340024500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_surface_protein_-_putative', 'interval': Interval(814218, 816590, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 814218, 'end': 816590, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0024500', 'parent_id': 'LINF_340024500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_surface_protein_-_putative', 'interval': Interval(814218, 816590, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 814218, 'end': 815826, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0024500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(814218, 815826, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5420BT:\n",
      "\t{'chrom': 'LinJ.34', 'start': 816591, 'end': 819357, 'strand': '-', 'gene_id': 'LINF_340024600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_surface_protein_-_putative', 'interval': Interval(816591, 819357, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 816591, 'end': 819357, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0024600', 'parent_id': 'LINF_340024600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_surface_protein_-_putative', 'interval': Interval(816591, 819357, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 816591, 'end': 818593, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0024600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(816591, 818593, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5430BT:\n",
      "\t{'chrom': 'LinJ.34', 'start': 816591, 'end': 819357, 'strand': '-', 'gene_id': 'LINF_340024600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_surface_protein_-_putative', 'interval': Interval(816591, 819357, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 816591, 'end': 819357, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0024600', 'parent_id': 'LINF_340024600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_surface_protein_-_putative', 'interval': Interval(816591, 819357, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 816591, 'end': 818593, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0024600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(816591, 818593, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5440BV:\n",
      "\t{'chrom': 'LinJ.34', 'start': 889011, 'end': 895326, 'strand': '-', 'gene_id': 'LINF_340026200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Cardiolipin_synthetase', 'interval': Interval(889011, 895326, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 889011, 'end': 895326, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0026200', 'parent_id': 'LINF_340026200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Cardiolipin_synthetase', 'interval': Interval(889011, 895326, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 889011, 'end': 891956, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0026200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(889011, 891956, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5450BW:\n",
      "\t{'chrom': 'LinJ.34', 'start': 889011, 'end': 895326, 'strand': '-', 'gene_id': 'LINF_340026200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Cardiolipin_synthetase', 'interval': Interval(889011, 895326, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 889011, 'end': 895326, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0026200', 'parent_id': 'LINF_340026200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Cardiolipin_synthetase', 'interval': Interval(889011, 895326, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 889011, 'end': 891956, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0026200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(889011, 891956, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5460BV:\n",
      "\t{'chrom': 'LinJ.34', 'start': 902943, 'end': 911956, 'strand': '-', 'gene_id': 'LINF_340026700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'eukaryotic_translation_initiation_factor_2-alpha_kinase|eIF2alpha_kinase|PERK|EIF2AK2', 'interval': Interval(902943, 911956, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 902943, 'end': 911956, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0026700', 'parent_id': 'LINF_340026700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'eukaryotic_translation_initiation_factor_2-alpha_kinase|eIF2alpha_kinase|PERK|EIF2AK2', 'interval': Interval(902943, 911956, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 902943, 'end': 906520, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0026700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(902943, 906520, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5470BW:\n",
      "\t{'chrom': 'LinJ.34', 'start': 902943, 'end': 911956, 'strand': '-', 'gene_id': 'LINF_340026700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'eukaryotic_translation_initiation_factor_2-alpha_kinase|eIF2alpha_kinase|PERK|EIF2AK2', 'interval': Interval(902943, 911956, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 902943, 'end': 911956, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0026700', 'parent_id': 'LINF_340026700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'eukaryotic_translation_initiation_factor_2-alpha_kinase|eIF2alpha_kinase|PERK|EIF2AK2', 'interval': Interval(902943, 911956, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 902943, 'end': 906520, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0026700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(902943, 906520, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5480:\n",
      "\t{'chrom': 'LinJ.34', 'start': 914428, 'end': 917327, 'strand': '-', 'gene_id': 'LINF_340026900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': '8-oxoguanine_DNA_glycosylase_-_putative', 'interval': Interval(914428, 917327, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 914428, 'end': 917327, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0026900', 'parent_id': 'LINF_340026900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': '8-oxoguanine_DNA_glycosylase_-_putative', 'interval': Interval(914428, 917327, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 914428, 'end': 915466, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0026900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(914428, 915466, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5490BV:\n",
      "\t{'chrom': 'LinJ.34', 'start': 923279, 'end': 937617, 'strand': '-', 'gene_id': 'LINF_340027100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Dual_specificity_protein_phosphatase', 'interval': Interval(923279, 937617, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 923279, 'end': 937617, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0027100', 'parent_id': 'LINF_340027100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Dual_specificity_protein_phosphatase', 'interval': Interval(923279, 937617, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 923279, 'end': 930391, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0027100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(923279, 930391, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5500BW:\n",
      "\t{'chrom': 'LinJ.34', 'start': 923279, 'end': 937617, 'strand': '-', 'gene_id': 'LINF_340027100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Dual_specificity_protein_phosphatase', 'interval': Interval(923279, 937617, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 923279, 'end': 937617, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0027100', 'parent_id': 'LINF_340027100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Dual_specificity_protein_phosphatase', 'interval': Interval(923279, 937617, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 923279, 'end': 930391, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0027100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(923279, 930391, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5510:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1001422, 'end': 1012594, 'strand': '-', 'gene_id': 'LINF_340029200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(1001422, 1012594, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1001422, 'end': 1012594, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0029200', 'parent_id': 'LINF_340029200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(1001422, 1012594, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1001422, 'end': 1009395, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0029200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1001422, 1009395, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5520:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1067136, 'end': 1077919, 'strand': '-', 'gene_id': 'LINF_340030800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_phosphatase_2C-like_protein', 'interval': Interval(1067136, 1077919, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1067136, 'end': 1077919, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0030800', 'parent_id': 'LINF_340030800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_phosphatase_2C-like_protein', 'interval': Interval(1067136, 1077919, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1067136, 'end': 1075956, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0030800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1067136, 1075956, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5530:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1210394, 'end': 1214718, 'strand': '+', 'gene_id': 'LINF_340033475', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1210394, 1214718, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1210394, 'end': 1214718, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0033475', 'parent_id': 'LINF_340033475', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1210394, 1214718, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5540:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1225952, 'end': 1230078, 'strand': '+', 'gene_id': 'LINF_340033800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'calcineurin-like_phosphoesterase-like_protein', 'interval': Interval(1225952, 1230078, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1225952, 'end': 1230078, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0033800', 'parent_id': 'LINF_340033800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'calcineurin-like_phosphoesterase-like_protein', 'interval': Interval(1225952, 1230078, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1225952, 'end': 1226312, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0033800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(1225952, 1226312, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5550BX:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1225952, 'end': 1230078, 'strand': '+', 'gene_id': 'LINF_340033800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'calcineurin-like_phosphoesterase-like_protein', 'interval': Interval(1225952, 1230078, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1225952, 'end': 1230078, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0033800', 'parent_id': 'LINF_340033800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'calcineurin-like_phosphoesterase-like_protein', 'interval': Interval(1225952, 1230078, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1228197, 'end': 1230078, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0033800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1228197, 1230078, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5560BY:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1240773, 'end': 1245780, 'strand': '+', 'gene_id': 'LINF_340034300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1240773, 1245780, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1240773, 'end': 1245780, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034300', 'parent_id': 'LINF_340034300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1240773, 1245780, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1241492, 'end': 1245780, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1241492, 1245780, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5570BZ:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1240773, 'end': 1245780, 'strand': '+', 'gene_id': 'LINF_340034300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1240773, 1245780, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1240773, 'end': 1245780, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034300', 'parent_id': 'LINF_340034300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1240773, 1245780, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1241492, 'end': 1245780, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1241492, 1245780, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5580CA:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1240773, 'end': 1245780, 'strand': '+', 'gene_id': 'LINF_340034300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1240773, 1245780, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1240773, 'end': 1245780, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034300', 'parent_id': 'LINF_340034300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1240773, 1245780, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1241492, 'end': 1245780, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1241492, 1245780, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5590BY:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1246328, 'end': 1249925, 'strand': '+', 'gene_id': 'LINF_340034400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1246328, 1249925, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1246328, 'end': 1249925, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034400', 'parent_id': 'LINF_340034400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1246328, 1249925, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1247056, 'end': 1249925, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1247056, 1249925, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5600BZ:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1246328, 'end': 1249925, 'strand': '+', 'gene_id': 'LINF_340034400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1246328, 1249925, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1246328, 'end': 1249925, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034400', 'parent_id': 'LINF_340034400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1246328, 1249925, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1247056, 'end': 1249925, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1247056, 1249925, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5610CA:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1246328, 'end': 1249925, 'strand': '+', 'gene_id': 'LINF_340034400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1246328, 1249925, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1246328, 'end': 1249925, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034400', 'parent_id': 'LINF_340034400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1246328, 1249925, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1247056, 'end': 1249925, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1247056, 1249925, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5620CB:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1246328, 'end': 1249925, 'strand': '+', 'gene_id': 'LINF_340034400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1246328, 1249925, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1246328, 'end': 1249925, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034400', 'parent_id': 'LINF_340034400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1246328, 1249925, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1247056, 'end': 1249925, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1247056, 1249925, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5630J:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1250241, 'end': 1251653, 'strand': '+', 'gene_id': 'LINF_340034450', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1250241, 1251653, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1250241, 'end': 1251653, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034450', 'parent_id': 'LINF_340034450', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1250241, 1251653, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5640BY:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1252201, 'end': 1255794, 'strand': '+', 'gene_id': 'LINF_340034500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1252201, 1255794, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1252201, 'end': 1255794, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034500', 'parent_id': 'LINF_340034500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1252201, 1255794, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1252943, 'end': 1255794, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1252943, 1255794, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5650BZ:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1252201, 'end': 1255794, 'strand': '+', 'gene_id': 'LINF_340034500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1252201, 1255794, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1252201, 'end': 1255794, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034500', 'parent_id': 'LINF_340034500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1252201, 1255794, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1252943, 'end': 1255794, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1252943, 1255794, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5660CA:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1252201, 'end': 1255794, 'strand': '+', 'gene_id': 'LINF_340034500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1252201, 1255794, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1252201, 'end': 1255794, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034500', 'parent_id': 'LINF_340034500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1252201, 1255794, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1252943, 'end': 1255794, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1252943, 1255794, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5670CB:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1252201, 'end': 1255794, 'strand': '+', 'gene_id': 'LINF_340034500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1252201, 1255794, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1252201, 'end': 1255794, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034500', 'parent_id': 'LINF_340034500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1252201, 1255794, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1252943, 'end': 1255794, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1252943, 1255794, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5680J:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1256128, 'end': 1257539, 'strand': '+', 'gene_id': 'LINF_340034550', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1256128, 1257539, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1256128, 'end': 1257539, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034550', 'parent_id': 'LINF_340034550', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1256128, 1257539, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5690BY:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1258088, 'end': 1261692, 'strand': '+', 'gene_id': 'LINF_340034600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1258088, 1261692, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1258088, 'end': 1261692, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034600', 'parent_id': 'LINF_340034600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1258088, 1261692, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1258830, 'end': 1261692, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1258830, 1261692, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5700BZ:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1258088, 'end': 1261692, 'strand': '+', 'gene_id': 'LINF_340034600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1258088, 1261692, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1258088, 'end': 1261692, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034600', 'parent_id': 'LINF_340034600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1258088, 1261692, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1258830, 'end': 1261692, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1258830, 1261692, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5710CA:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1258088, 'end': 1261692, 'strand': '+', 'gene_id': 'LINF_340034600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1258088, 1261692, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1258088, 'end': 1261692, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034600', 'parent_id': 'LINF_340034600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1258088, 1261692, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1258830, 'end': 1261692, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1258830, 1261692, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5720CB:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1258088, 'end': 1261692, 'strand': '+', 'gene_id': 'LINF_340034600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1258088, 1261692, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1258088, 'end': 1261692, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034600', 'parent_id': 'LINF_340034600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1258088, 1261692, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1258830, 'end': 1261692, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1258830, 1261692, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5730J:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1262015, 'end': 1263719, 'strand': '+', 'gene_id': 'LINF_340034650', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1262015, 1263719, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1262015, 'end': 1263719, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034650', 'parent_id': 'LINF_340034650', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1262015, 1263719, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5740BY:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1263977, 'end': 1267600, 'strand': '+', 'gene_id': 'LINF_340034700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1263977, 1267600, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1263977, 'end': 1267600, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034700', 'parent_id': 'LINF_340034700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1263977, 1267600, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1264705, 'end': 1267600, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1264705, 1267600, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5750BZ:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1263977, 'end': 1267600, 'strand': '+', 'gene_id': 'LINF_340034700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1263977, 1267600, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1263977, 'end': 1267600, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034700', 'parent_id': 'LINF_340034700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1263977, 1267600, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1264705, 'end': 1267600, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1264705, 1267600, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5760CA:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1263977, 'end': 1267600, 'strand': '+', 'gene_id': 'LINF_340034700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1263977, 1267600, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1263977, 'end': 1267600, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034700', 'parent_id': 'LINF_340034700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1263977, 1267600, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1264705, 'end': 1267600, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1264705, 1267600, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5770CB:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1263977, 'end': 1267600, 'strand': '+', 'gene_id': 'LINF_340034700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1263977, 1267600, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1263977, 'end': 1267600, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034700', 'parent_id': 'LINF_340034700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1263977, 1267600, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1264705, 'end': 1267600, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1264705, 1267600, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5780J:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1267915, 'end': 1269327, 'strand': '+', 'gene_id': 'LINF_340034760', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1267915, 1269327, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1267915, 'end': 1269327, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034760', 'parent_id': 'LINF_340034760', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1267915, 1269327, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5790BZ:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1269875, 'end': 1273542, 'strand': '+', 'gene_id': 'LINF_340034800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein', 'interval': Interval(1269875, 1273542, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1269875, 'end': 1273542, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034800', 'parent_id': 'LINF_340034800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein', 'interval': Interval(1269875, 1273542, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1270608, 'end': 1273542, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1270608, 1273542, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5800CA:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1269875, 'end': 1273542, 'strand': '+', 'gene_id': 'LINF_340034800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein', 'interval': Interval(1269875, 1273542, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1269875, 'end': 1273542, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034800', 'parent_id': 'LINF_340034800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein', 'interval': Interval(1269875, 1273542, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1270608, 'end': 1273542, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1270608, 1273542, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5810J:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1273855, 'end': 1275269, 'strand': '+', 'gene_id': 'LINF_340034850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1273855, 1275269, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1273855, 'end': 1275269, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034850', 'parent_id': 'LINF_340034850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1273855, 1275269, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5820BY:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1275817, 'end': 1279464, 'strand': '+', 'gene_id': 'LINF_340034900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1275817, 1279464, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1275817, 'end': 1279464, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034900', 'parent_id': 'LINF_340034900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1275817, 1279464, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1276550, 'end': 1279464, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1276550, 1279464, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5830BZ:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1275817, 'end': 1279464, 'strand': '+', 'gene_id': 'LINF_340034900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1275817, 1279464, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1275817, 'end': 1279464, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034900', 'parent_id': 'LINF_340034900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1275817, 1279464, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1276550, 'end': 1279464, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1276550, 1279464, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5840CA:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1275817, 'end': 1279464, 'strand': '+', 'gene_id': 'LINF_340034900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1275817, 1279464, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1275817, 'end': 1279464, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034900', 'parent_id': 'LINF_340034900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein_-_putative', 'interval': Interval(1275817, 1279464, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1276550, 'end': 1279464, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0034900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1276550, 1279464, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5850J:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1279799, 'end': 1281587, 'strand': '+', 'gene_id': 'LINF_340034950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1279799, 1281587, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1279799, 'end': 1281587, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0034950', 'parent_id': 'LINF_340034950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1279799, 1281587, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5860:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1316728, 'end': 1319613, 'strand': '+', 'gene_id': 'LINF_340036050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1316728, 1319613, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1316728, 'end': 1319613, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0036050', 'parent_id': 'LINF_340036050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1316728, 1319613, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5870:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1316728, 'end': 1319613, 'strand': '+', 'gene_id': 'LINF_340036050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1316728, 1319613, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1316728, 'end': 1319613, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0036050', 'parent_id': 'LINF_340036050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1316728, 1319613, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5880CC:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1354107, 'end': 1357619, 'strand': '+', 'gene_id': 'LINF_340037300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'mitochondrial_carrier_protein|MCP6', 'interval': Interval(1354107, 1357619, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1354107, 'end': 1357619, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0037300', 'parent_id': 'LINF_340037300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'mitochondrial_carrier_protein|MCP6', 'interval': Interval(1354107, 1357619, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1355719, 'end': 1357619, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0037300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1355719, 1357619, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5890CC:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1358089, 'end': 1365799, 'strand': '+', 'gene_id': 'LINF_340037350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1358089, 1365799, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1358089, 'end': 1365799, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0037350', 'parent_id': 'LINF_340037350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1358089, 1365799, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1360468, 'end': 1365799, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0037350', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1360468, 1365799, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5900CD:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1372364, 'end': 1376363, 'strand': '+', 'gene_id': 'LINF_340037600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Tripartite_attachment_complex_40|TAC40', 'interval': Interval(1372364, 1376363, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1372364, 'end': 1376363, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0037600', 'parent_id': 'LINF_340037600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Tripartite_attachment_complex_40|TAC40', 'interval': Interval(1372364, 1376363, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1374067, 'end': 1376363, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0037600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1374067, 1376363, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5910CE:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1372364, 'end': 1376363, 'strand': '+', 'gene_id': 'LINF_340037600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Tripartite_attachment_complex_40|TAC40', 'interval': Interval(1372364, 1376363, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1372364, 'end': 1376363, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0037600', 'parent_id': 'LINF_340037600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Tripartite_attachment_complex_40|TAC40', 'interval': Interval(1372364, 1376363, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1374067, 'end': 1376363, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0037600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1374067, 1376363, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5920CC:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5930CC:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5940CD:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1391972, 'end': 1395962, 'strand': '+', 'gene_id': 'LINF_340038100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Tripartite_attachment_complex_40|TAC40', 'interval': Interval(1391972, 1395962, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1391972, 'end': 1395962, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0038100', 'parent_id': 'LINF_340038100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Tripartite_attachment_complex_40|TAC40', 'interval': Interval(1391972, 1395962, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1393676, 'end': 1395962, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0038100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1393676, 1395962, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5950CE:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1391972, 'end': 1395962, 'strand': '+', 'gene_id': 'LINF_340038100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Tripartite_attachment_complex_40|TAC40', 'interval': Interval(1391972, 1395962, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1391972, 'end': 1395962, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0038100', 'parent_id': 'LINF_340038100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Tripartite_attachment_complex_40|TAC40', 'interval': Interval(1391972, 1395962, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1393676, 'end': 1395962, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0038100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1393676, 1395962, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5960CC:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5970CC:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5980CC:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1420070, 'end': 1428451, 'strand': '+', 'gene_id': 'LINF_340038800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'NAD_dependent_epimerase/dehydratase_family', 'interval': Interval(1420070, 1428451, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1420070, 'end': 1428451, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0038800', 'parent_id': 'LINF_340038800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'NAD_dependent_epimerase/dehydratase_family', 'interval': Interval(1420070, 1428451, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1423024, 'end': 1428451, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0038800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1423024, 1428451, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5990CF:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1450417, 'end': 1455670, 'strand': '+', 'gene_id': 'LINF_340039400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Alpha-keto-acid_decarboxylase', 'interval': Interval(1450417, 1455670, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1450417, 'end': 1455670, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0039400', 'parent_id': 'LINF_340039400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Alpha-keto-acid_decarboxylase', 'interval': Interval(1450417, 1455670, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1452466, 'end': 1455670, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0039400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1452466, 1455670, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.6000CF:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1473802, 'end': 1479828, 'strand': '+', 'gene_id': 'LINF_340040300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Cytochrome_p450-like_protein|Sterol_C22_desaturase-like', 'interval': Interval(1473802, 1479828, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1473802, 'end': 1479828, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0040300', 'parent_id': 'LINF_340040300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Cytochrome_p450-like_protein|Sterol_C22_desaturase-like', 'interval': Interval(1473802, 1479828, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1476185, 'end': 1479828, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0040300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1476185, 1479828, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.6010:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1509524, 'end': 1513042, 'strand': '+', 'gene_id': 'LINF_340041200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'CPSF73|cleavage_and_polyadenylation_specificity_factor_subunit', 'interval': Interval(1509524, 1513042, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1509524, 'end': 1513042, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0041200', 'parent_id': 'LINF_340041200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'CPSF73|cleavage_and_polyadenylation_specificity_factor_subunit', 'interval': Interval(1509524, 1513042, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1512033, 'end': 1513042, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0041200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1512033, 1513042, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.6020:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1574276, 'end': 1578126, 'strand': '+', 'gene_id': 'LINF_340043700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Metallophos_domain-containing_protein', 'interval': Interval(1574276, 1578126, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1574276, 'end': 1578126, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0043700', 'parent_id': 'LINF_340043700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Metallophos_domain-containing_protein', 'interval': Interval(1574276, 1578126, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1576000, 'end': 1578126, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0043700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1576000, 1578126, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.6030:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1732476, 'end': 1737014, 'strand': '+', 'gene_id': 'LINF_340048400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ubiquitin_carboxyl-terminal_hydrolase', 'interval': Interval(1732476, 1737014, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1732476, 'end': 1737014, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0048400', 'parent_id': 'LINF_340048400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ubiquitin_carboxyl-terminal_hydrolase', 'interval': Interval(1732476, 1737014, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1736067, 'end': 1737014, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0048400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1736067, 1737014, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.6040:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1742910, 'end': 1745873, 'strand': '+', 'gene_id': 'LINF_340048700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'GRIP_domain_containing_protein', 'interval': Interval(1742910, 1745873, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1742910, 'end': 1745873, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0048700', 'parent_id': 'LINF_340048700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'GRIP_domain_containing_protein', 'interval': Interval(1742910, 1745873, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1744650, 'end': 1745873, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0048700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1744650, 1745873, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6050CG:\n",
      "\t{'chrom': 'LinJ.35', 'start': 249811, 'end': 254421, 'strand': '+', 'gene_id': 'LINF_350011700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Endonuclease/Exonuclease/phosphatase_family_-_putative', 'interval': Interval(249811, 254421, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 249811, 'end': 254421, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0011700', 'parent_id': 'LINF_350011700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Endonuclease/Exonuclease/phosphatase_family_-_putative', 'interval': Interval(249811, 254421, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 252159, 'end': 254421, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0011700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(252159, 254421, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6060CH:\n",
      "\t{'chrom': 'LinJ.35', 'start': 285400, 'end': 292518, 'strand': '+', 'gene_id': 'LINF_350012800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(285400, 292518, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 285400, 'end': 292518, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0012800', 'parent_id': 'LINF_350012800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(285400, 292518, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 285400, 'end': 287443, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0012800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(285400, 287443, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6070CI:\n",
      "\t{'chrom': 'LinJ.35', 'start': 296633, 'end': 298023, 'strand': '+', 'gene_id': 'LINF_350012950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(296633, 298023, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 296633, 'end': 298023, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0012950', 'parent_id': 'LINF_350012950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(296633, 298023, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6080:\n",
      "==================================================\n",
      "rejected_noCDS_c35.6090CI:\n",
      "\t{'chrom': 'LinJ.35', 'start': 307833, 'end': 309899, 'strand': '+', 'gene_id': 'LINF_350013300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(307833, 309899, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 307833, 'end': 309899, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0013300', 'parent_id': 'LINF_350013300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(307833, 309899, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 309226, 'end': 309899, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0013300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(309226, 309899, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6100:\n",
      "==================================================\n",
      "rejected_noCDS_c35.6110CH:\n",
      "\t{'chrom': 'LinJ.35', 'start': 372999, 'end': 385156, 'strand': '+', 'gene_id': 'LINF_350014930', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(372999, 385156, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 372999, 'end': 385156, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0014930', 'parent_id': 'LINF_350014930', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(372999, 385156, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6120CG:\n",
      "\t{'chrom': 'LinJ.35', 'start': 462165, 'end': 468419, 'strand': '+', 'gene_id': 'LINF_350016700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'NADH-dependent_fumarate_reductase', 'interval': Interval(462165, 468419, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 462165, 'end': 468419, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0016700', 'parent_id': 'LINF_350016700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'NADH-dependent_fumarate_reductase', 'interval': Interval(462165, 468419, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 462165, 'end': 464646, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0016700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(462165, 464646, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6130:\n",
      "\t{'chrom': 'LinJ.35', 'start': 462165, 'end': 468419, 'strand': '+', 'gene_id': 'LINF_350016700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'NADH-dependent_fumarate_reductase', 'interval': Interval(462165, 468419, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 462165, 'end': 468419, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0016700', 'parent_id': 'LINF_350016700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'NADH-dependent_fumarate_reductase', 'interval': Interval(462165, 468419, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 466135, 'end': 468419, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0016700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(466135, 468419, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6140:\n",
      "\t{'chrom': 'LinJ.35', 'start': 759907, 'end': 764052, 'strand': '+', 'gene_id': 'LINF_350025700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Deubiquitinating_enzyme_MINDY-3/4-like_protein', 'interval': Interval(759907, 764052, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 759907, 'end': 764052, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0025700', 'parent_id': 'LINF_350025700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Deubiquitinating_enzyme_MINDY-3/4-like_protein', 'interval': Interval(759907, 764052, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 762204, 'end': 764052, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0025700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(762204, 764052, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6150:\n",
      "\t{'chrom': 'LinJ.35', 'start': 796669, 'end': 802441, 'strand': '+', 'gene_id': 'LINF_350026500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(796669, 802441, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 796669, 'end': 802441, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0026500', 'parent_id': 'LINF_350026500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(796669, 802441, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 798792, 'end': 802441, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0026500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(798792, 802441, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6160:\n",
      "\t{'chrom': 'LinJ.35', 'start': 848557, 'end': 852531, 'strand': '+', 'gene_id': 'LINF_350027900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATP-grasp_domain_containing_protein_-_putative', 'interval': Interval(848557, 852531, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 848557, 'end': 852531, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0027900', 'parent_id': 'LINF_350027900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATP-grasp_domain_containing_protein_-_putative', 'interval': Interval(848557, 852531, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 850859, 'end': 852531, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0027900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(850859, 852531, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6170:\n",
      "\t{'chrom': 'LinJ.35', 'start': 995475, 'end': 997673, 'strand': '-', 'gene_id': 'LINF_350031200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(995475, 997673, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 995475, 'end': 997673, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_35T0031200', 'parent_id': 'LINF_350031200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(995475, 997673, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 996296, 'end': 997673, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0031200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(996296, 997673, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6180CJ:\n",
      "==================================================\n",
      "rejected_noCDS_c35.6190:\n",
      "\t{'chrom': 'LinJ.35', 'start': 1154361, 'end': 1158302, 'strand': '-', 'gene_id': 'LINF_350035400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'trypanothione-dependent_glyoxalase_I|GLO1', 'interval': Interval(1154361, 1158302, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1154361, 'end': 1158302, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_35T0035400', 'parent_id': 'LINF_350035400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'trypanothione-dependent_glyoxalase_I|GLO1', 'interval': Interval(1154361, 1158302, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1154361, 'end': 1157720, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0035400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1154361, 1157720, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6200BX:\n",
      "\t{'chrom': 'LinJ.35', 'start': 1182104, 'end': 1185965, 'strand': '-', 'gene_id': 'LINF_350036100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glycerol_kinase_-_glycosomal', 'interval': Interval(1182104, 1185965, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1182104, 'end': 1185965, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_35T0036100', 'parent_id': 'LINF_350036100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glycerol_kinase_-_glycosomal', 'interval': Interval(1182104, 1185965, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1182104, 'end': 1184202, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0036100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1182104, 1184202, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6210CK:\n",
      "\t{'chrom': 'LinJ.35', 'start': 1320709, 'end': 1324559, 'strand': '-', 'gene_id': 'LINF_350039900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'DNA-repair_protein_-_putative', 'interval': Interval(1320709, 1324559, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1320709, 'end': 1324559, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_35T0039900', 'parent_id': 'LINF_350039900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'DNA-repair_protein_-_putative', 'interval': Interval(1320709, 1324559, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1320709, 'end': 1321513, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0039900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1320709, 1321513, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6220CL:\n",
      "==================================================\n",
      "rejected_noCDS_c35.6230:\n",
      "==================================================\n",
      "rejected_noCDS_c35.6240CK:\n",
      "\t{'chrom': 'LinJ.35', 'start': 1331548, 'end': 1334812, 'strand': '-', 'gene_id': 'LINF_350040200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1331548, 1334812, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1331548, 'end': 1334812, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_35T0040200', 'parent_id': 'LINF_350040200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1331548, 1334812, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1331548, 'end': 1332252, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0040200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1331548, 1332252, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6250CJ:\n",
      "==================================================\n",
      "rejected_noCDS_c35.6260CL:\n",
      "\t{'chrom': 'LinJ.35', 'start': 1374147, 'end': 1380051, 'strand': '-', 'gene_id': 'LINF_350041200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'membrane_transporter_protein_-_putative', 'interval': Interval(1374147, 1380051, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1374147, 'end': 1380051, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_35T0041200', 'parent_id': 'LINF_350041200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'membrane_transporter_protein_-_putative', 'interval': Interval(1374147, 1380051, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1374147, 'end': 1377165, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0041200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1374147, 1377165, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6270:\n",
      "==================================================\n",
      "rejected_noCDS_c35.6280:\n",
      "\t{'chrom': 'LinJ.35', 'start': 1624799, 'end': 1632256, 'strand': '+', 'gene_id': 'LINF_350048000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1624799, 1632256, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1624799, 'end': 1632256, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0048000', 'parent_id': 'LINF_350048000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1624799, 1632256, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1628467, 'end': 1632256, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0048000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1628467, 1632256, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6290:\n",
      "\t{'chrom': 'LinJ.35', 'start': 1745256, 'end': 1748582, 'strand': '+', 'gene_id': 'LINF_350051200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Uncharacterised_conserved_protein_-_putative', 'interval': Interval(1745256, 1748582, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1745256, 'end': 1748582, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0051200', 'parent_id': 'LINF_350051200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Uncharacterised_conserved_protein_-_putative', 'interval': Interval(1745256, 1748582, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1745256, 'end': 1745523, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0051200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(1745256, 1745523, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6300:\n",
      "\t{'chrom': 'LinJ.35', 'start': 1903392, 'end': 1908031, 'strand': '+', 'gene_id': 'LINF_350056000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1903392, 1908031, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1903392, 'end': 1908031, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0056000', 'parent_id': 'LINF_350056000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1903392, 1908031, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1906719, 'end': 1908031, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0056000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1906719, 1908031, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6310:\n",
      "==================================================\n",
      "rejected_noCDS_c35.6320CM:\n",
      "\t{'chrom': 'LinJ.35', 'start': 1965874, 'end': 1966729, 'strand': '+', 'gene_id': 'LINF_350058030', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1965874, 1966729, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1965874, 'end': 1966729, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0058030', 'parent_id': 'LINF_350058030', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1965874, 1966729, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6330CM:\n",
      "\t{'chrom': 'LinJ.35', 'start': 1994576, 'end': 1997700, 'strand': '+', 'gene_id': 'LINF_350059400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(1994576, 1997700, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1994576, 'end': 1997700, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0059400', 'parent_id': 'LINF_350059400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(1994576, 1997700, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1995398, 'end': 1997700, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0059400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1995398, 1997700, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6340CN:\n",
      "\t{'chrom': 'LinJ.36', 'start': 267960, 'end': 275257, 'strand': '+', 'gene_id': 'LINF_360013100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(267960, 275257, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 267960, 'end': 275257, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_36T0013100', 'parent_id': 'LINF_360013100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(267960, 275257, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 273184, 'end': 275257, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0013100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(273184, 275257, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6350CN:\n",
      "\t{'chrom': 'LinJ.36', 'start': 312865, 'end': 320529, 'strand': '+', 'gene_id': 'LINF_360014100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Zinc_finger_protein|ZC3H32', 'interval': Interval(312865, 320529, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 312865, 'end': 320529, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_36T0014100', 'parent_id': 'LINF_360014100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Zinc_finger_protein|ZC3H32', 'interval': Interval(312865, 320529, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 315253, 'end': 320529, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0014100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(315253, 320529, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6360CO:\n",
      "\t{'chrom': 'LinJ.36', 'start': 493588, 'end': 494799, 'strand': '+', 'gene_id': 'LINF_360019350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(493588, 494799, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 493588, 'end': 494799, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_36T0019350', 'parent_id': 'LINF_360019350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(493588, 494799, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6370CO:\n",
      "==================================================\n",
      "rejected_noCDS_c36.6380:\n",
      "\t{'chrom': 'LinJ.36', 'start': 495605, 'end': 497054, 'strand': '-', 'gene_id': 'LINF_360019630', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(495605, 497054, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 495605, 'end': 497054, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0019630', 'parent_id': 'LINF_360019630', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(495605, 497054, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6390CO:\n",
      "\t{'chrom': 'LinJ.36', 'start': 513482, 'end': 516676, 'strand': '-', 'gene_id': 'LINF_360020150', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(513482, 516676, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 513482, 'end': 516676, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0020150', 'parent_id': 'LINF_360020150', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(513482, 516676, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6400CP:\n",
      "\t{'chrom': 'LinJ.36', 'start': 565414, 'end': 566152, 'strand': '-', 'gene_id': 'LINF_360021150', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(565414, 566152, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 565414, 'end': 566152, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0021150', 'parent_id': 'LINF_360021150', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(565414, 566152, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6410:\n",
      "\t{'chrom': 'LinJ.36', 'start': 571329, 'end': 578055, 'strand': '-', 'gene_id': 'LINF_360021400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'serine/threonine_protein_kinase', 'interval': Interval(571329, 578055, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 571329, 'end': 578055, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0021400', 'parent_id': 'LINF_360021400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'serine/threonine_protein_kinase', 'interval': Interval(571329, 578055, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 571329, 'end': 576358, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0021400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(571329, 576358, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6420CP:\n",
      "\t{'chrom': 'LinJ.36', 'start': 585484, 'end': 588074, 'strand': '-', 'gene_id': 'LINF_360021550', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(585484, 588074, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 585484, 'end': 588074, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0021550', 'parent_id': 'LINF_360021550', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(585484, 588074, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6430CP:\n",
      "\t{'chrom': 'LinJ.36', 'start': 588342, 'end': 593204, 'strand': '-', 'gene_id': 'LINF_360021600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'engulfment_and_cell_motility_domain_2_-_putative', 'interval': Interval(588342, 593204, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 588342, 'end': 593204, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0021600', 'parent_id': 'LINF_360021600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'engulfment_and_cell_motility_domain_2_-_putative', 'interval': Interval(588342, 593204, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 588342, 'end': 589725, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0021600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(588342, 589725, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6440CO:\n",
      "\t{'chrom': 'LinJ.36', 'start': 599331, 'end': 600508, 'strand': '-', 'gene_id': 'LINF_360021750', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(599331, 600508, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 599331, 'end': 600508, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0021750', 'parent_id': 'LINF_360021750', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(599331, 600508, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6450:\n",
      "\t{'chrom': 'LinJ.36', 'start': 632898, 'end': 640647, 'strand': '-', 'gene_id': 'LINF_360022600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'clathrin_heavy_chain', 'interval': Interval(632898, 640647, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 632898, 'end': 640647, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0022600', 'parent_id': 'LINF_360022600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'clathrin_heavy_chain', 'interval': Interval(632898, 640647, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 632898, 'end': 634891, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0022600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(632898, 634891, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6460:\n",
      "\t{'chrom': 'LinJ.36', 'start': 632898, 'end': 640647, 'strand': '-', 'gene_id': 'LINF_360022600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'clathrin_heavy_chain', 'interval': Interval(632898, 640647, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 632898, 'end': 640647, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0022600', 'parent_id': 'LINF_360022600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'clathrin_heavy_chain', 'interval': Interval(632898, 640647, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 632898, 'end': 634891, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0022600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(632898, 634891, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6470:\n",
      "\t{'chrom': 'LinJ.36', 'start': 973249, 'end': 974992, 'strand': '+', 'gene_id': 'LINF_360030800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(973249, 974992, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 973249, 'end': 974992, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_36T0030800', 'parent_id': 'LINF_360030800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(973249, 974992, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6480:\n",
      "\t{'chrom': 'LinJ.36', 'start': 1420740, 'end': 1424437, 'strand': '-', 'gene_id': 'LINF_360044300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'signal_recognition_particle_receptor_like_protein', 'interval': Interval(1420740, 1424437, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 1420740, 'end': 1424437, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0044300', 'parent_id': 'LINF_360044300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'signal_recognition_particle_receptor_like_protein', 'interval': Interval(1420740, 1424437, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 1420740, 'end': 1422543, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0044300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1420740, 1422543, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6490:\n",
      "==================================================\n",
      "rejected_noCDS_c36.6500:\n",
      "\t{'chrom': 'LinJ.36', 'start': 1587489, 'end': 1588827, 'strand': '+', 'gene_id': 'LINF_360049750', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1587489, 1588827, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 1587489, 'end': 1588827, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_36T0049750', 'parent_id': 'LINF_360049750', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1587489, 1588827, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6510:\n",
      "\t{'chrom': 'LinJ.36', 'start': 1675388, 'end': 1677071, 'strand': '+', 'gene_id': 'LINF_360051850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1675388, 1677071, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 1675388, 'end': 1677071, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_36T0051850', 'parent_id': 'LINF_360051850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1675388, 1677071, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6520:\n",
      "\t{'chrom': 'LinJ.36', 'start': 1702974, 'end': 1712480, 'strand': '+', 'gene_id': 'LINF_360052500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'predicted_C2_domain_protein', 'interval': Interval(1702974, 1712480, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 1702974, 'end': 1712480, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_36T0052500', 'parent_id': 'LINF_360052500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'predicted_C2_domain_protein', 'interval': Interval(1702974, 1712480, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 1710551, 'end': 1712480, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0052500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1710551, 1712480, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6530CQ:\n",
      "\t{'chrom': 'LinJ.36', 'start': 1944919, 'end': 1946935, 'strand': '-', 'gene_id': 'LINF_360058750', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1944919, 1946935, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 1944919, 'end': 1946935, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0058750', 'parent_id': 'LINF_360058750', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1944919, 1946935, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6540CR:\n",
      "\t{'chrom': 'LinJ.36', 'start': 1944919, 'end': 1946935, 'strand': '-', 'gene_id': 'LINF_360058750', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1944919, 1946935, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 1944919, 'end': 1946935, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0058750', 'parent_id': 'LINF_360058750', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1944919, 1946935, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6550CQ:\n",
      "\t{'chrom': 'LinJ.36', 'start': 1963667, 'end': 1973734, 'strand': '-', 'gene_id': 'LINF_360059300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1963667, 1973734, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 1963667, 'end': 1973734, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0059300', 'parent_id': 'LINF_360059300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1963667, 1973734, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 1963667, 'end': 1971662, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0059300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1963667, 1971662, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6560CR:\n",
      "\t{'chrom': 'LinJ.36', 'start': 1963667, 'end': 1973734, 'strand': '-', 'gene_id': 'LINF_360059300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1963667, 1973734, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 1963667, 'end': 1973734, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0059300', 'parent_id': 'LINF_360059300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1963667, 1973734, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 1963667, 'end': 1971662, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0059300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1963667, 1971662, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6570:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2031905, 'end': 2035330, 'strand': '-', 'gene_id': 'LINF_360061300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'aldehyde_dehydrogenase', 'interval': Interval(2031905, 2035330, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2031905, 'end': 2035330, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0061300', 'parent_id': 'LINF_360061300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'aldehyde_dehydrogenase', 'interval': Interval(2031905, 2035330, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2031905, 'end': 2033288, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0061300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2031905, 2033288, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6580CR:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2087103, 'end': 2094370, 'strand': '-', 'gene_id': 'LINF_360062500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_kinase_domain-containing_protein', 'interval': Interval(2087103, 2094370, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2087103, 'end': 2094370, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0062500', 'parent_id': 'LINF_360062500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_kinase_domain-containing_protein', 'interval': Interval(2087103, 2094370, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2087103, 'end': 2089585, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0062500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2087103, 2089585, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6590:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2087103, 'end': 2094370, 'strand': '-', 'gene_id': 'LINF_360062500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_kinase_domain-containing_protein', 'interval': Interval(2087103, 2094370, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2087103, 'end': 2094370, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0062500', 'parent_id': 'LINF_360062500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_kinase_domain-containing_protein', 'interval': Interval(2087103, 2094370, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2087103, 'end': 2089585, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0062500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2087103, 2089585, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6600CR:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2117939, 'end': 2120182, 'strand': '-', 'gene_id': 'LINF_360063300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(2117939, 2120182, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2117939, 'end': 2120182, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0063300', 'parent_id': 'LINF_360063300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(2117939, 2120182, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2117939, 'end': 2119655, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0063300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2117939, 2119655, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6610:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2174216, 'end': 2179241, 'strand': '-', 'gene_id': 'LINF_360064900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'adiponectin_receptor_protein_1_-_putative', 'interval': Interval(2174216, 2179241, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2174216, 'end': 2179241, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0064900', 'parent_id': 'LINF_360064900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'adiponectin_receptor_protein_1_-_putative', 'interval': Interval(2174216, 2179241, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2174216, 'end': 2177389, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0064900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2174216, 2177389, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6620:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2306975, 'end': 2310975, 'strand': '-', 'gene_id': 'LINF_360069000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Haloacid_dehalogenase-like_hydrolase', 'interval': Interval(2306975, 2310975, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2306975, 'end': 2310975, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0069000', 'parent_id': 'LINF_360069000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Haloacid_dehalogenase-like_hydrolase', 'interval': Interval(2306975, 2310975, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2306975, 'end': 2308975, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0069000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2306975, 2308975, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6630:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2322277, 'end': 2323858, 'strand': '-', 'gene_id': 'LINF_360069350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2322277, 2323858, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2322277, 'end': 2323858, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0069350', 'parent_id': 'LINF_360069350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2322277, 2323858, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6640:\n",
      "==================================================\n",
      "rejected_noCDS_c36.6650CS:\n",
      "==================================================\n",
      "rejected_noCDS_c36.6660CS:\n",
      "==================================================\n",
      "rejected_noCDS_c36.6670CS:\n",
      "==================================================\n",
      "rejected_noCDS_c36.6680CS:\n",
      "==================================================\n",
      "rejected_noCDS_c36.6690CS:\n",
      "==================================================\n",
      "rejected_noCDS_c36.6700CS:\n",
      "==================================================\n",
      "rejected_noCDS_c36.6710CS:\n",
      "==================================================\n",
      "rejected_noCDS_c36.6720CS:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2481878, 'end': 2486596, 'strand': '-', 'gene_id': 'LINF_360073600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2481878, 2486596, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2481878, 'end': 2486596, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0073600', 'parent_id': 'LINF_360073600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2481878, 2486596, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2481878, 'end': 2484386, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0073600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2481878, 2484386, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6730:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2481878, 'end': 2486596, 'strand': '-', 'gene_id': 'LINF_360073600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2481878, 2486596, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2481878, 'end': 2486596, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0073600', 'parent_id': 'LINF_360073600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2481878, 2486596, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2481878, 'end': 2484386, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0073600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2481878, 2484386, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6740CT:\n",
      "==================================================\n",
      "rejected_noCDS_c36.6750CU:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2633506, 'end': 2636050, 'strand': '-', 'gene_id': 'LINF_360078400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': '2′-O-ribose_methyltransferase|MTr1', 'interval': Interval(2633506, 2636050, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2633506, 'end': 2636050, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0078400', 'parent_id': 'LINF_360078400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': '2′-O-ribose_methyltransferase|MTr1', 'interval': Interval(2633506, 2636050, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2633506, 'end': 2634788, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0078400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2633506, 2634788, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6760CV:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2647886, 'end': 2650599, 'strand': '-', 'gene_id': 'LINF_360079000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'mitochondrial_carrier_protein_-_putative', 'interval': Interval(2647886, 2650599, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2647886, 'end': 2650599, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0079000', 'parent_id': 'LINF_360079000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'mitochondrial_carrier_protein_-_putative', 'interval': Interval(2647886, 2650599, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2647886, 'end': 2649497, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0079000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2647886, 2649497, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6770CT:\n",
      "==================================================\n",
      "rejected_noCDS_c36.6780CV:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2664678, 'end': 2668468, 'strand': '-', 'gene_id': 'LINF_360079500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'adaptin_complex_1_subunit|beta_adaptin\\xa0', 'interval': Interval(2664678, 2668468, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2664678, 'end': 2668468, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0079500', 'parent_id': 'LINF_360079500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'adaptin_complex_1_subunit|beta_adaptin\\xa0', 'interval': Interval(2664678, 2668468, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2664678, 'end': 2665863, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0079500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2664678, 2665863, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6790CU:\n",
      "==================================================\n",
      "rejected_noCDS_c36.6800CU:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2706448, 'end': 2707984, 'strand': '-', 'gene_id': 'LINF_360080850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2706448, 2707984, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2706448, 'end': 2707984, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0080850', 'parent_id': 'LINF_360080850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2706448, 2707984, closed='both')}\n"
     ]
    }
   ],
   "execution_count": 109
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:04.326113Z",
     "start_time": "2024-10-03T08:02:04.320333Z"
    }
   },
   "source": [
    "print(f\"From the total of {len(neg_gtf_dict)} NEGATIVE ELEMENTS, {counter_neg_inside} are inside GTF elements and {counter_neg_not_inside} are not inside GTF elements.\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "From the total of 681 NEGATIVE ELEMENTS, 503 are inside GTF elements and 178 are not inside GTF elements.\n"
     ]
    }
   ],
   "execution_count": 110
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's get the elements in different dictionaries depending on if the length of \"values\" is > 0 or not:"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:04.674759Z",
     "start_time": "2024-10-03T08:02:04.668070Z"
    }
   },
   "source": [
    "# Get the elements which value is > 0\n",
    "neg_inside_gtf_dict = {key: value for key, value in neg_gtf_dict.items() if len(value) > 0}\n",
    "print(len(neg_inside_gtf_dict))\n",
    "\n",
    "# Get the elements which value is == 0\n",
    "neg_not_inside_gtf_dict = {key: value for key, value in neg_gtf_dict.items() if len(value) == 0}\n",
    "print(len(neg_not_inside_gtf_dict))"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "503\n",
      "178\n"
     ]
    }
   ],
   "execution_count": 111
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "let's transform it in a data frame"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:05.033369Z",
     "start_time": "2024-10-03T08:02:05.004520Z"
    }
   },
   "source": [
    "neg_inside_gtf_list = []\n",
    "for key, value in neg_inside_gtf_dict.items():\n",
    "    for elem in value:\n",
    "        new_record = {'neg_name' : key}  # Create dict of 1 element\n",
    "        new_record.update(elem)  # Update the dict with the values from elem, this way \"neg_name\" goes first\n",
    "        neg_inside_gtf_list.append(new_record)\n",
    "\n",
    "neg_inside_gtf_df = pd.DataFrame(neg_inside_gtf_list)\n",
    "\n",
    "# Let's check the df\n",
    "print(neg_inside_gtf_df.shape)\n",
    "print(neg_inside_gtf_df.dtypes)\n",
    "print(neg_inside_gtf_df['neg_name'].nunique())\n",
    "neg_inside_gtf_df.head()"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1358, 16)\n",
      "neg_name                        object\n",
      "chrom                           object\n",
      "start                            int64\n",
      "end                              int64\n",
      "strand                          object\n",
      "gene_id                         object\n",
      "transcript_id                   object\n",
      "parent_id                       object\n",
      "gene                              bool\n",
      "transcript                        bool\n",
      "CDS                               bool\n",
      "3utr                              bool\n",
      "5utr                              bool\n",
      "pseudogen                       object\n",
      "notes                           object\n",
      "interval         interval[int64, both]\n",
      "dtype: object\n",
      "503\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "                neg_name    chrom   start     end strand         gene_id  \\\n",
       "0  rejected_noCDS_c01.10  LinJ.01   34736   37218      -  LINF_010006300   \n",
       "1  rejected_noCDS_c01.10  LinJ.01   34736   37218      -            None   \n",
       "2  rejected_noCDS_c01.10  LinJ.01   34736   36818      -            None   \n",
       "3  rejected_noCDS_c01.30  LinJ.01  145613  146868      +  LINF_010010350   \n",
       "4  rejected_noCDS_c01.30  LinJ.01  145613  146868      +            None   \n",
       "\n",
       "     transcript_id        parent_id   gene  transcript    CDS   3utr   5utr  \\\n",
       "0             None             None   True       False  False  False  False   \n",
       "1  LINF_01T0006300   LINF_010006300  False        True  False  False  False   \n",
       "2             None  LINF_01T0006300  False       False  False   True  False   \n",
       "3             None             None   True       False  False  False  False   \n",
       "4  LINF_01T0010350   LINF_010010350  False        True  False  False  False   \n",
       "\n",
       "  pseudogen                               notes          interval  \n",
       "0      None  hypothetical_protein_-_conserved__    [34736, 37218]  \n",
       "1      None  hypothetical_protein_-_conserved__    [34736, 37218]  \n",
       "2      None                                None    [34736, 36818]  \n",
       "3      None                                None  [145613, 146868]  \n",
       "4      None                                None  [145613, 146868]  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>neg_name</th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>CDS</th>\n",
       "      <th>3utr</th>\n",
       "      <th>5utr</th>\n",
       "      <th>pseudogen</th>\n",
       "      <th>notes</th>\n",
       "      <th>interval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rejected_noCDS_c01.10</td>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>34736</td>\n",
       "      <td>37218</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_010006300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>hypothetical_protein_-_conserved__</td>\n",
       "      <td>[34736, 37218]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rejected_noCDS_c01.10</td>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>34736</td>\n",
       "      <td>37218</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0006300</td>\n",
       "      <td>LINF_010006300</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>hypothetical_protein_-_conserved__</td>\n",
       "      <td>[34736, 37218]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rejected_noCDS_c01.10</td>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>34736</td>\n",
       "      <td>36818</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0006300</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[34736, 36818]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rejected_noCDS_c01.30</td>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>145613</td>\n",
       "      <td>146868</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_010010350</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[145613, 146868]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rejected_noCDS_c01.30</td>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>145613</td>\n",
       "      <td>146868</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0010350</td>\n",
       "      <td>LINF_010010350</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[145613, 146868]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 112
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now with these data we can analyze a lot of things"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## NEGATIVE ELEMENTS intergenic\n",
    "\n",
    "Now we will check the NEGATIVE ELEMENTS that are not overlapping any GTF element using `neg_not_inside_gtf_dict`"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:05.584005Z",
     "start_time": "2024-10-03T08:02:05.574378Z"
    }
   },
   "source": [
    "# Let's check again the dict:\n",
    "print(len(neg_not_inside_gtf_dict))\n",
    "neg_not_inside_gtf_dict"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "178\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'rejected_noCDS_c01.20': [],\n",
       " 'rejected_noCDS_c01.40': [],\n",
       " 'rejected_noCDS_c01.50': [],\n",
       " 'rejected_noCDS_c01.60A': [],\n",
       " 'rejected_noCDS_c02.140': [],\n",
       " 'rejected_noCDS_c02.150': [],\n",
       " 'rejected_noCDS_c02.170': [],\n",
       " 'rejected_noCDS_c03.180': [],\n",
       " 'rejected_noCDS_c04.200C': [],\n",
       " 'rejected_noCDS_c04.210C': [],\n",
       " 'rejected_noCDS_c04.230': [],\n",
       " 'rejected_noCDS_c05.280': [],\n",
       " 'rejected_noCDS_c05.300D': [],\n",
       " 'rejected_noCDS_c05.320D': [],\n",
       " 'rejected_noCDS_c05.350': [],\n",
       " 'rejected_noCDS_c05.360F': [],\n",
       " 'rejected_noCDS_c06.380': [],\n",
       " 'rejected_noCDS_c06.400': [],\n",
       " 'rejected_noCDS_c07.430G': [],\n",
       " 'rejected_noCDS_c07.510': [],\n",
       " 'rejected_noCDS_c08.550': [],\n",
       " 'rejected_noCDS_c08.560I': [],\n",
       " 'rejected_noCDS_c08.590I': [],\n",
       " 'rejected_noCDS_c08.620I': [],\n",
       " 'rejected_noCDS_c08.640I': [],\n",
       " 'rejected_noCDS_c08.670': [],\n",
       " 'rejected_noCDS_c08.680K': [],\n",
       " 'rejected_noCDS_c08.710K': [],\n",
       " 'rejected_noCDS_c08.720N': [],\n",
       " 'rejected_noCDS_c08.760K': [],\n",
       " 'rejected_noCDS_c08.770N': [],\n",
       " 'rejected_noCDS_c08.800K': [],\n",
       " 'rejected_noCDS_c08.850': [],\n",
       " 'rejected_noCDS_c08.860': [],\n",
       " 'rejected_noCDS_c10.980O': [],\n",
       " 'rejected_noCDS_c10.1010': [],\n",
       " 'rejected_noCDS_c11.1070': [],\n",
       " 'rejected_noCDS_c11.1080': [],\n",
       " 'rejected_noCDS_c12.1120': [],\n",
       " 'rejected_noCDS_c12.1170': [],\n",
       " 'rejected_noCDS_c12.1180S': [],\n",
       " 'rejected_noCDS_c12.1190': [],\n",
       " 'rejected_noCDS_c12.1200S': [],\n",
       " 'rejected_noCDS_c12.1210S': [],\n",
       " 'rejected_noCDS_c12.1220S': [],\n",
       " 'rejected_noCDS_c12.1240S': [],\n",
       " 'rejected_noCDS_c12.1250S': [],\n",
       " 'rejected_noCDS_c13.1400': [],\n",
       " 'rejected_noCDS_c14.1420': [],\n",
       " 'rejected_noCDS_c14.1530': [],\n",
       " 'rejected_noCDS_c14.1540W': [],\n",
       " 'rejected_noCDS_c15.1580': [],\n",
       " 'rejected_noCDS_c15.1590X': [],\n",
       " 'rejected_noCDS_c15.1610X': [],\n",
       " 'rejected_noCDS_c16.1720': [],\n",
       " 'rejected_noCDS_c17.1830Z': [],\n",
       " 'rejected_noCDS_c17.1900': [],\n",
       " 'rejected_noCDS_c17.1910': [],\n",
       " 'rejected_noCDS_c17.1940AA': [],\n",
       " 'rejected_noCDS_c17.1970Z': [],\n",
       " 'rejected_noCDS_c18.1980': [],\n",
       " 'rejected_noCDS_c18.1990': [],\n",
       " 'rejected_noCDS_c18.2000': [],\n",
       " 'rejected_noCDS_c19.2110': [],\n",
       " 'rejected_noCDS_c19.2120AD': [],\n",
       " 'rejected_noCDS_c19.2130AD': [],\n",
       " 'rejected_noCDS_c19.2160': [],\n",
       " 'rejected_noCDS_c19.2230': [],\n",
       " 'rejected_noCDS_c20.2250': [],\n",
       " 'rejected_noCDS_c20.2260': [],\n",
       " 'rejected_noCDS_c20.2310': [],\n",
       " 'rejected_noCDS_c21.2390': [],\n",
       " 'rejected_noCDS_c21.2460Z': [],\n",
       " 'rejected_noCDS_c22.2480': [],\n",
       " 'rejected_noCDS_c22.2540A': [],\n",
       " 'rejected_noCDS_c23.2560': [],\n",
       " 'rejected_noCDS_c23.2570': [],\n",
       " 'rejected_noCDS_c23.2640AI': [],\n",
       " 'rejected_noCDS_c23.2650AI': [],\n",
       " 'rejected_noCDS_c23.2690': [],\n",
       " 'rejected_noCDS_c24.2710AJ': [],\n",
       " 'rejected_noCDS_c24.2770AJ': [],\n",
       " 'rejected_noCDS_c25.2850AK': [],\n",
       " 'rejected_noCDS_c25.2940': [],\n",
       " 'rejected_noCDS_c26.3040': [],\n",
       " 'rejected_noCDS_c26.3080': [],\n",
       " 'rejected_noCDS_c26.3110': [],\n",
       " 'rejected_noCDS_c26.3150': [],\n",
       " 'rejected_noCDS_BLAST_ERROR': [],\n",
       " 'rejected_noCDS_c27.3290AS': [],\n",
       " 'rejected_noCDS_c27.3300AS': [],\n",
       " 'rejected_noCDS_c27.3310': [],\n",
       " 'rejected_noCDS_c28.3370': [],\n",
       " 'rejected_noCDS_c28.3400': [],\n",
       " 'rejected_noCDS_c28.3410': [],\n",
       " 'rejected_noCDS_c28.3430Z': [],\n",
       " 'rejected_noCDS_c29.3440': [],\n",
       " 'rejected_noCDS_c29.3450': [],\n",
       " 'rejected_noCDS_c29.3550AX': [],\n",
       " 'rejected_noCDS_c29.3570AZ': [],\n",
       " 'rejected_noCDS_c29.3580AX': [],\n",
       " 'rejected_noCDS_c29.3590': [],\n",
       " 'rejected_noCDS_c29.3610AY': [],\n",
       " 'rejected_noCDS_c29.3620AZ': [],\n",
       " 'rejected_noCDS_c29.3630AX': [],\n",
       " 'rejected_noCDS_c29.3640AY': [],\n",
       " 'rejected_noCDS_c29.3650AZ': [],\n",
       " 'rejected_noCDS_c29.3660AX': [],\n",
       " 'rejected_noCDS_c29.3670AY': [],\n",
       " 'rejected_noCDS_c29.3690': [],\n",
       " 'rejected_noCDS_c29.3750': [],\n",
       " 'rejected_noCDS_c29.3800': [],\n",
       " 'rejected_noCDS_c29.3810': [],\n",
       " 'rejected_noCDS_c29.3830': [],\n",
       " 'rejected_noCDS_c30.3860BC': [],\n",
       " 'rejected_noCDS_c30.3930': [],\n",
       " 'rejected_noCDS_c31.4110': [],\n",
       " 'rejected_noCDS_c31.4120': [],\n",
       " 'rejected_noCDS_c31.4140': [],\n",
       " 'rejected_noCDS_c31.4180': [],\n",
       " 'rejected_noCDS_c31.4220': [],\n",
       " 'rejected_noCDS_c31.4230BF': [],\n",
       " 'rejected_noCDS_c31.4240BG': [],\n",
       " 'rejected_noCDS_c31.4310': [],\n",
       " 'rejected_noCDS_c31.4400': [],\n",
       " 'rejected_noCDS_c31.4470': [],\n",
       " 'rejected_noCDS_c31.4550': [],\n",
       " 'rejected_noCDS_c32.4570O': [],\n",
       " 'rejected_noCDS_c32.4690': [],\n",
       " 'rejected_noCDS_c32.4710BO': [],\n",
       " 'rejected_noCDS_c33.4850': [],\n",
       " 'rejected_noCDS_c33.4880': [],\n",
       " 'rejected_noCDS_c33.4890BC': [],\n",
       " 'rejected_noCDS_c33.4910': [],\n",
       " 'rejected_noCDS_c33.4940BQ': [],\n",
       " 'rejected_noCDS_c33.4960BQ': [],\n",
       " 'rejected_noCDS_c33.5030AK': [],\n",
       " 'rejected_noCDS_c33.5040AK': [],\n",
       " 'rejected_noCDS_c34.5060': [],\n",
       " 'rejected_noCDS_c34.5140BS': [],\n",
       " 'rejected_noCDS_c34.5180BS': [],\n",
       " 'rejected_noCDS_c34.5230BU': [],\n",
       " 'rejected_noCDS_c34.5260BU': [],\n",
       " 'rejected_noCDS_c34.5290BU': [],\n",
       " 'rejected_noCDS_c34.5310BU': [],\n",
       " 'rejected_noCDS_c34.5320BT': [],\n",
       " 'rejected_noCDS_c34.5330BU': [],\n",
       " 'rejected_noCDS_c34.5340BU': [],\n",
       " 'rejected_noCDS_c34.5350BU': [],\n",
       " 'rejected_noCDS_c34.5360BT': [],\n",
       " 'rejected_noCDS_c34.5370BU': [],\n",
       " 'rejected_noCDS_c34.5390BU': [],\n",
       " 'rejected_noCDS_c34.5400BT': [],\n",
       " 'rejected_noCDS_c34.5920CC': [],\n",
       " 'rejected_noCDS_c34.5930CC': [],\n",
       " 'rejected_noCDS_c34.5960CC': [],\n",
       " 'rejected_noCDS_c34.5970CC': [],\n",
       " 'rejected_noCDS_c35.6080': [],\n",
       " 'rejected_noCDS_c35.6100': [],\n",
       " 'rejected_noCDS_c35.6180CJ': [],\n",
       " 'rejected_noCDS_c35.6220CL': [],\n",
       " 'rejected_noCDS_c35.6230': [],\n",
       " 'rejected_noCDS_c35.6250CJ': [],\n",
       " 'rejected_noCDS_c35.6270': [],\n",
       " 'rejected_noCDS_c35.6310': [],\n",
       " 'rejected_noCDS_c36.6370CO': [],\n",
       " 'rejected_noCDS_c36.6490': [],\n",
       " 'rejected_noCDS_c36.6640': [],\n",
       " 'rejected_noCDS_c36.6650CS': [],\n",
       " 'rejected_noCDS_c36.6660CS': [],\n",
       " 'rejected_noCDS_c36.6670CS': [],\n",
       " 'rejected_noCDS_c36.6680CS': [],\n",
       " 'rejected_noCDS_c36.6690CS': [],\n",
       " 'rejected_noCDS_c36.6700CS': [],\n",
       " 'rejected_noCDS_c36.6710CS': [],\n",
       " 'rejected_noCDS_c36.6740CT': [],\n",
       " 'rejected_noCDS_c36.6770CT': [],\n",
       " 'rejected_noCDS_c36.6790CU': []}"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 113
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:06.490563Z",
     "start_time": "2024-10-03T08:02:06.484687Z"
    }
   },
   "source": [
    "# Let's take the keys from the dict as a list:\n",
    "neg_not_inside_gtf_dict_list = list(neg_not_inside_gtf_dict.keys())\n",
    "print(neg_not_inside_gtf_dict_list)"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['rejected_noCDS_c01.20', 'rejected_noCDS_c01.40', 'rejected_noCDS_c01.50', 'rejected_noCDS_c01.60A', 'rejected_noCDS_c02.140', 'rejected_noCDS_c02.150', 'rejected_noCDS_c02.170', 'rejected_noCDS_c03.180', 'rejected_noCDS_c04.200C', 'rejected_noCDS_c04.210C', 'rejected_noCDS_c04.230', 'rejected_noCDS_c05.280', 'rejected_noCDS_c05.300D', 'rejected_noCDS_c05.320D', 'rejected_noCDS_c05.350', 'rejected_noCDS_c05.360F', 'rejected_noCDS_c06.380', 'rejected_noCDS_c06.400', 'rejected_noCDS_c07.430G', 'rejected_noCDS_c07.510', 'rejected_noCDS_c08.550', 'rejected_noCDS_c08.560I', 'rejected_noCDS_c08.590I', 'rejected_noCDS_c08.620I', 'rejected_noCDS_c08.640I', 'rejected_noCDS_c08.670', 'rejected_noCDS_c08.680K', 'rejected_noCDS_c08.710K', 'rejected_noCDS_c08.720N', 'rejected_noCDS_c08.760K', 'rejected_noCDS_c08.770N', 'rejected_noCDS_c08.800K', 'rejected_noCDS_c08.850', 'rejected_noCDS_c08.860', 'rejected_noCDS_c10.980O', 'rejected_noCDS_c10.1010', 'rejected_noCDS_c11.1070', 'rejected_noCDS_c11.1080', 'rejected_noCDS_c12.1120', 'rejected_noCDS_c12.1170', 'rejected_noCDS_c12.1180S', 'rejected_noCDS_c12.1190', 'rejected_noCDS_c12.1200S', 'rejected_noCDS_c12.1210S', 'rejected_noCDS_c12.1220S', 'rejected_noCDS_c12.1240S', 'rejected_noCDS_c12.1250S', 'rejected_noCDS_c13.1400', 'rejected_noCDS_c14.1420', 'rejected_noCDS_c14.1530', 'rejected_noCDS_c14.1540W', 'rejected_noCDS_c15.1580', 'rejected_noCDS_c15.1590X', 'rejected_noCDS_c15.1610X', 'rejected_noCDS_c16.1720', 'rejected_noCDS_c17.1830Z', 'rejected_noCDS_c17.1900', 'rejected_noCDS_c17.1910', 'rejected_noCDS_c17.1940AA', 'rejected_noCDS_c17.1970Z', 'rejected_noCDS_c18.1980', 'rejected_noCDS_c18.1990', 'rejected_noCDS_c18.2000', 'rejected_noCDS_c19.2110', 'rejected_noCDS_c19.2120AD', 'rejected_noCDS_c19.2130AD', 'rejected_noCDS_c19.2160', 'rejected_noCDS_c19.2230', 'rejected_noCDS_c20.2250', 'rejected_noCDS_c20.2260', 'rejected_noCDS_c20.2310', 'rejected_noCDS_c21.2390', 'rejected_noCDS_c21.2460Z', 'rejected_noCDS_c22.2480', 'rejected_noCDS_c22.2540A', 'rejected_noCDS_c23.2560', 'rejected_noCDS_c23.2570', 'rejected_noCDS_c23.2640AI', 'rejected_noCDS_c23.2650AI', 'rejected_noCDS_c23.2690', 'rejected_noCDS_c24.2710AJ', 'rejected_noCDS_c24.2770AJ', 'rejected_noCDS_c25.2850AK', 'rejected_noCDS_c25.2940', 'rejected_noCDS_c26.3040', 'rejected_noCDS_c26.3080', 'rejected_noCDS_c26.3110', 'rejected_noCDS_c26.3150', 'rejected_noCDS_BLAST_ERROR', 'rejected_noCDS_c27.3290AS', 'rejected_noCDS_c27.3300AS', 'rejected_noCDS_c27.3310', 'rejected_noCDS_c28.3370', 'rejected_noCDS_c28.3400', 'rejected_noCDS_c28.3410', 'rejected_noCDS_c28.3430Z', 'rejected_noCDS_c29.3440', 'rejected_noCDS_c29.3450', 'rejected_noCDS_c29.3550AX', 'rejected_noCDS_c29.3570AZ', 'rejected_noCDS_c29.3580AX', 'rejected_noCDS_c29.3590', 'rejected_noCDS_c29.3610AY', 'rejected_noCDS_c29.3620AZ', 'rejected_noCDS_c29.3630AX', 'rejected_noCDS_c29.3640AY', 'rejected_noCDS_c29.3650AZ', 'rejected_noCDS_c29.3660AX', 'rejected_noCDS_c29.3670AY', 'rejected_noCDS_c29.3690', 'rejected_noCDS_c29.3750', 'rejected_noCDS_c29.3800', 'rejected_noCDS_c29.3810', 'rejected_noCDS_c29.3830', 'rejected_noCDS_c30.3860BC', 'rejected_noCDS_c30.3930', 'rejected_noCDS_c31.4110', 'rejected_noCDS_c31.4120', 'rejected_noCDS_c31.4140', 'rejected_noCDS_c31.4180', 'rejected_noCDS_c31.4220', 'rejected_noCDS_c31.4230BF', 'rejected_noCDS_c31.4240BG', 'rejected_noCDS_c31.4310', 'rejected_noCDS_c31.4400', 'rejected_noCDS_c31.4470', 'rejected_noCDS_c31.4550', 'rejected_noCDS_c32.4570O', 'rejected_noCDS_c32.4690', 'rejected_noCDS_c32.4710BO', 'rejected_noCDS_c33.4850', 'rejected_noCDS_c33.4880', 'rejected_noCDS_c33.4890BC', 'rejected_noCDS_c33.4910', 'rejected_noCDS_c33.4940BQ', 'rejected_noCDS_c33.4960BQ', 'rejected_noCDS_c33.5030AK', 'rejected_noCDS_c33.5040AK', 'rejected_noCDS_c34.5060', 'rejected_noCDS_c34.5140BS', 'rejected_noCDS_c34.5180BS', 'rejected_noCDS_c34.5230BU', 'rejected_noCDS_c34.5260BU', 'rejected_noCDS_c34.5290BU', 'rejected_noCDS_c34.5310BU', 'rejected_noCDS_c34.5320BT', 'rejected_noCDS_c34.5330BU', 'rejected_noCDS_c34.5340BU', 'rejected_noCDS_c34.5350BU', 'rejected_noCDS_c34.5360BT', 'rejected_noCDS_c34.5370BU', 'rejected_noCDS_c34.5390BU', 'rejected_noCDS_c34.5400BT', 'rejected_noCDS_c34.5920CC', 'rejected_noCDS_c34.5930CC', 'rejected_noCDS_c34.5960CC', 'rejected_noCDS_c34.5970CC', 'rejected_noCDS_c35.6080', 'rejected_noCDS_c35.6100', 'rejected_noCDS_c35.6180CJ', 'rejected_noCDS_c35.6220CL', 'rejected_noCDS_c35.6230', 'rejected_noCDS_c35.6250CJ', 'rejected_noCDS_c35.6270', 'rejected_noCDS_c35.6310', 'rejected_noCDS_c36.6370CO', 'rejected_noCDS_c36.6490', 'rejected_noCDS_c36.6640', 'rejected_noCDS_c36.6650CS', 'rejected_noCDS_c36.6660CS', 'rejected_noCDS_c36.6670CS', 'rejected_noCDS_c36.6680CS', 'rejected_noCDS_c36.6690CS', 'rejected_noCDS_c36.6700CS', 'rejected_noCDS_c36.6710CS', 'rejected_noCDS_c36.6740CT', 'rejected_noCDS_c36.6770CT', 'rejected_noCDS_c36.6790CU']\n"
     ]
    }
   ],
   "execution_count": 114
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "Let's find the overlapping elements:"
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:08.265979Z",
     "start_time": "2024-10-03T08:02:07.049581Z"
    }
   },
   "source": [
    "# Call the old code but only using the keys from the list\n",
    "# initialize dict\n",
    "neg_gtf_dict_2 = {name: [] for name in neg_not_inside_gtf_dict_list}\n",
    "\n",
    "# Find elements in neg_df that overlap with a gtf_df\n",
    "# Find overlaps using boolean indexing\n",
    "for i, neg_row in neg_df.iterrows():\n",
    "    # Boolean mask for intervals that overlaps the neg interval\n",
    "    print(f\"Analyzing elem {i+1}/{neg_df.shape[0]}\")\n",
    "\n",
    "    if neg_row['name'] not in neg_not_inside_gtf_dict_list:\n",
    "        continue\n",
    "\n",
    "    mask = (gtf_df['chrom'] == neg_row['chrom']) & \\\n",
    "            (\n",
    "                ((neg_row['start'] >= gtf_df['start']) & (neg_row['start'] <= gtf_df['end'])) |\n",
    "                ((neg_row['end'] >= gtf_df['start']) & (neg_row['end'] <= gtf_df['end'])) |\n",
    "                ((gtf_df['start'] >= neg_row['start']) & (gtf_df['start'] <= neg_row['end'])) |\n",
    "                ((gtf_df['end'] >= neg_row['start']) & (gtf_df['end'] <= neg_row['end']))\n",
    "                )\n",
    "    \n",
    "    overlaps = gtf_df[mask]\n",
    "    \n",
    "    for j, gtf_row in overlaps.iterrows():\n",
    "        neg_gtf_dict_2[neg_row['name']].append(gtf_row.to_dict())"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analyzing elem 1/682\n",
      "Analyzing elem 2/682\n",
      "Analyzing elem 3/682\n",
      "Analyzing elem 4/682\n",
      "Analyzing elem 5/682\n",
      "Analyzing elem 6/682\n",
      "Analyzing elem 7/682\n",
      "Analyzing elem 8/682\n",
      "Analyzing elem 9/682\n",
      "Analyzing elem 10/682\n",
      "Analyzing elem 11/682\n",
      "Analyzing elem 12/682\n",
      "Analyzing elem 13/682\n",
      "Analyzing elem 14/682\n",
      "Analyzing elem 15/682\n",
      "Analyzing elem 16/682\n",
      "Analyzing elem 17/682\n",
      "Analyzing elem 18/682\n",
      "Analyzing elem 19/682\n",
      "Analyzing elem 20/682\n",
      "Analyzing elem 21/682\n",
      "Analyzing elem 22/682\n",
      "Analyzing elem 23/682\n",
      "Analyzing elem 24/682\n",
      "Analyzing elem 25/682\n",
      "Analyzing elem 26/682\n",
      "Analyzing elem 27/682\n",
      "Analyzing elem 28/682\n",
      "Analyzing elem 29/682\n",
      "Analyzing elem 30/682\n",
      "Analyzing elem 31/682\n",
      "Analyzing elem 32/682\n",
      "Analyzing elem 33/682\n",
      "Analyzing elem 34/682\n",
      "Analyzing elem 35/682\n",
      "Analyzing elem 36/682\n",
      "Analyzing elem 37/682\n",
      "Analyzing elem 38/682\n",
      "Analyzing elem 39/682\n",
      "Analyzing elem 40/682\n",
      "Analyzing elem 41/682\n",
      "Analyzing elem 42/682\n",
      "Analyzing elem 43/682\n",
      "Analyzing elem 44/682\n",
      "Analyzing elem 45/682\n",
      "Analyzing elem 46/682\n",
      "Analyzing elem 47/682\n",
      "Analyzing elem 48/682\n",
      "Analyzing elem 49/682\n",
      "Analyzing elem 50/682\n",
      "Analyzing elem 51/682\n",
      "Analyzing elem 52/682\n",
      "Analyzing elem 53/682\n",
      "Analyzing elem 54/682\n",
      "Analyzing elem 55/682\n",
      "Analyzing elem 56/682\n",
      "Analyzing elem 57/682\n",
      "Analyzing elem 58/682\n",
      "Analyzing elem 59/682\n",
      "Analyzing elem 60/682\n",
      "Analyzing elem 61/682\n",
      "Analyzing elem 62/682\n",
      "Analyzing elem 63/682\n",
      "Analyzing elem 64/682\n",
      "Analyzing elem 65/682\n",
      "Analyzing elem 66/682\n",
      "Analyzing elem 67/682\n",
      "Analyzing elem 68/682\n",
      "Analyzing elem 69/682\n",
      "Analyzing elem 70/682\n",
      "Analyzing elem 71/682\n",
      "Analyzing elem 72/682\n",
      "Analyzing elem 73/682\n",
      "Analyzing elem 74/682\n",
      "Analyzing elem 75/682\n",
      "Analyzing elem 76/682\n",
      "Analyzing elem 77/682\n",
      "Analyzing elem 78/682\n",
      "Analyzing elem 79/682\n",
      "Analyzing elem 80/682\n",
      "Analyzing elem 81/682\n",
      "Analyzing elem 82/682\n",
      "Analyzing elem 83/682\n",
      "Analyzing elem 84/682\n",
      "Analyzing elem 85/682\n",
      "Analyzing elem 86/682\n",
      "Analyzing elem 87/682\n",
      "Analyzing elem 88/682\n",
      "Analyzing elem 89/682\n",
      "Analyzing elem 90/682\n",
      "Analyzing elem 91/682\n",
      "Analyzing elem 92/682\n",
      "Analyzing elem 93/682\n",
      "Analyzing elem 94/682\n",
      "Analyzing elem 95/682\n",
      "Analyzing elem 96/682\n",
      "Analyzing elem 97/682\n",
      "Analyzing elem 98/682\n",
      "Analyzing elem 99/682\n",
      "Analyzing elem 100/682\n",
      "Analyzing elem 101/682\n",
      "Analyzing elem 102/682\n",
      "Analyzing elem 103/682\n",
      "Analyzing elem 104/682\n",
      "Analyzing elem 105/682\n",
      "Analyzing elem 106/682\n",
      "Analyzing elem 107/682\n",
      "Analyzing elem 108/682\n",
      "Analyzing elem 109/682\n",
      "Analyzing elem 110/682\n",
      "Analyzing elem 111/682\n",
      "Analyzing elem 112/682\n",
      "Analyzing elem 113/682\n",
      "Analyzing elem 114/682\n",
      "Analyzing elem 115/682\n",
      "Analyzing elem 116/682\n",
      "Analyzing elem 117/682\n",
      "Analyzing elem 118/682\n",
      "Analyzing elem 119/682\n",
      "Analyzing elem 120/682\n",
      "Analyzing elem 121/682\n",
      "Analyzing elem 122/682\n",
      "Analyzing elem 123/682\n",
      "Analyzing elem 124/682\n",
      "Analyzing elem 125/682\n",
      "Analyzing elem 126/682\n",
      "Analyzing elem 127/682\n",
      "Analyzing elem 128/682\n",
      "Analyzing elem 129/682\n",
      "Analyzing elem 130/682\n",
      "Analyzing elem 131/682\n",
      "Analyzing elem 132/682\n",
      "Analyzing elem 133/682\n",
      "Analyzing elem 134/682\n",
      "Analyzing elem 135/682\n",
      "Analyzing elem 136/682\n",
      "Analyzing elem 137/682\n",
      "Analyzing elem 138/682\n",
      "Analyzing elem 139/682\n",
      "Analyzing elem 140/682\n",
      "Analyzing elem 141/682\n",
      "Analyzing elem 142/682\n",
      "Analyzing elem 143/682\n",
      "Analyzing elem 144/682\n",
      "Analyzing elem 145/682\n",
      "Analyzing elem 146/682\n",
      "Analyzing elem 147/682\n",
      "Analyzing elem 148/682\n",
      "Analyzing elem 149/682\n",
      "Analyzing elem 150/682\n",
      "Analyzing elem 151/682\n",
      "Analyzing elem 152/682\n",
      "Analyzing elem 153/682\n",
      "Analyzing elem 154/682\n",
      "Analyzing elem 155/682\n",
      "Analyzing elem 156/682\n",
      "Analyzing elem 157/682\n",
      "Analyzing elem 158/682\n",
      "Analyzing elem 159/682\n",
      "Analyzing elem 160/682\n",
      "Analyzing elem 161/682\n",
      "Analyzing elem 162/682\n",
      "Analyzing elem 163/682\n",
      "Analyzing elem 164/682\n",
      "Analyzing elem 165/682\n",
      "Analyzing elem 166/682\n",
      "Analyzing elem 167/682\n",
      "Analyzing elem 168/682\n",
      "Analyzing elem 169/682\n",
      "Analyzing elem 170/682\n",
      "Analyzing elem 171/682\n",
      "Analyzing elem 172/682\n",
      "Analyzing elem 173/682\n",
      "Analyzing elem 174/682\n",
      "Analyzing elem 175/682\n",
      "Analyzing elem 176/682\n",
      "Analyzing elem 177/682\n",
      "Analyzing elem 178/682\n",
      "Analyzing elem 179/682\n",
      "Analyzing elem 180/682\n",
      "Analyzing elem 181/682\n",
      "Analyzing elem 182/682\n",
      "Analyzing elem 183/682\n",
      "Analyzing elem 184/682\n",
      "Analyzing elem 185/682\n",
      "Analyzing elem 186/682\n",
      "Analyzing elem 187/682\n",
      "Analyzing elem 188/682\n",
      "Analyzing elem 189/682\n",
      "Analyzing elem 190/682\n",
      "Analyzing elem 191/682\n",
      "Analyzing elem 192/682\n",
      "Analyzing elem 193/682\n",
      "Analyzing elem 194/682\n",
      "Analyzing elem 195/682\n",
      "Analyzing elem 196/682\n",
      "Analyzing elem 197/682\n",
      "Analyzing elem 198/682\n",
      "Analyzing elem 199/682\n",
      "Analyzing elem 200/682\n",
      "Analyzing elem 201/682\n",
      "Analyzing elem 202/682\n",
      "Analyzing elem 203/682\n",
      "Analyzing elem 204/682\n",
      "Analyzing elem 205/682\n",
      "Analyzing elem 206/682\n",
      "Analyzing elem 207/682\n",
      "Analyzing elem 208/682\n",
      "Analyzing elem 209/682\n",
      "Analyzing elem 210/682\n",
      "Analyzing elem 211/682\n",
      "Analyzing elem 212/682\n",
      "Analyzing elem 213/682\n",
      "Analyzing elem 214/682\n",
      "Analyzing elem 215/682\n",
      "Analyzing elem 216/682\n",
      "Analyzing elem 217/682\n",
      "Analyzing elem 218/682\n",
      "Analyzing elem 219/682\n",
      "Analyzing elem 220/682\n",
      "Analyzing elem 221/682\n",
      "Analyzing elem 222/682\n",
      "Analyzing elem 223/682\n",
      "Analyzing elem 224/682\n",
      "Analyzing elem 225/682\n",
      "Analyzing elem 226/682\n",
      "Analyzing elem 227/682\n",
      "Analyzing elem 228/682\n",
      "Analyzing elem 229/682\n",
      "Analyzing elem 230/682\n",
      "Analyzing elem 231/682\n",
      "Analyzing elem 232/682\n",
      "Analyzing elem 233/682\n",
      "Analyzing elem 234/682\n",
      "Analyzing elem 235/682\n",
      "Analyzing elem 236/682\n",
      "Analyzing elem 237/682\n",
      "Analyzing elem 238/682\n",
      "Analyzing elem 239/682\n",
      "Analyzing elem 240/682\n",
      "Analyzing elem 241/682\n",
      "Analyzing elem 242/682\n",
      "Analyzing elem 243/682\n",
      "Analyzing elem 244/682\n",
      "Analyzing elem 245/682\n",
      "Analyzing elem 246/682\n",
      "Analyzing elem 247/682\n",
      "Analyzing elem 248/682\n",
      "Analyzing elem 249/682\n",
      "Analyzing elem 250/682\n",
      "Analyzing elem 251/682\n",
      "Analyzing elem 252/682\n",
      "Analyzing elem 253/682\n",
      "Analyzing elem 254/682\n",
      "Analyzing elem 255/682\n",
      "Analyzing elem 256/682\n",
      "Analyzing elem 257/682\n",
      "Analyzing elem 258/682\n",
      "Analyzing elem 259/682\n",
      "Analyzing elem 260/682\n",
      "Analyzing elem 261/682\n",
      "Analyzing elem 262/682\n",
      "Analyzing elem 263/682\n",
      "Analyzing elem 264/682\n",
      "Analyzing elem 265/682\n",
      "Analyzing elem 266/682\n",
      "Analyzing elem 267/682\n",
      "Analyzing elem 268/682\n",
      "Analyzing elem 269/682\n",
      "Analyzing elem 270/682\n",
      "Analyzing elem 271/682\n",
      "Analyzing elem 272/682\n",
      "Analyzing elem 273/682\n",
      "Analyzing elem 274/682\n",
      "Analyzing elem 275/682\n",
      "Analyzing elem 276/682\n",
      "Analyzing elem 277/682\n",
      "Analyzing elem 278/682\n",
      "Analyzing elem 279/682\n",
      "Analyzing elem 280/682\n",
      "Analyzing elem 281/682\n",
      "Analyzing elem 282/682\n",
      "Analyzing elem 283/682\n",
      "Analyzing elem 284/682\n",
      "Analyzing elem 285/682\n",
      "Analyzing elem 286/682\n",
      "Analyzing elem 287/682\n",
      "Analyzing elem 288/682\n",
      "Analyzing elem 289/682\n",
      "Analyzing elem 290/682\n",
      "Analyzing elem 291/682\n",
      "Analyzing elem 292/682\n",
      "Analyzing elem 293/682\n",
      "Analyzing elem 294/682\n",
      "Analyzing elem 295/682\n",
      "Analyzing elem 296/682\n",
      "Analyzing elem 297/682\n",
      "Analyzing elem 298/682\n",
      "Analyzing elem 299/682\n",
      "Analyzing elem 300/682\n",
      "Analyzing elem 301/682\n",
      "Analyzing elem 302/682\n",
      "Analyzing elem 303/682\n",
      "Analyzing elem 304/682\n",
      "Analyzing elem 305/682\n",
      "Analyzing elem 306/682\n",
      "Analyzing elem 307/682\n",
      "Analyzing elem 308/682\n",
      "Analyzing elem 309/682\n",
      "Analyzing elem 310/682\n",
      "Analyzing elem 311/682\n",
      "Analyzing elem 312/682\n",
      "Analyzing elem 313/682\n",
      "Analyzing elem 314/682\n",
      "Analyzing elem 315/682\n",
      "Analyzing elem 316/682\n",
      "Analyzing elem 317/682\n",
      "Analyzing elem 318/682\n",
      "Analyzing elem 319/682\n",
      "Analyzing elem 320/682\n",
      "Analyzing elem 321/682\n",
      "Analyzing elem 322/682\n",
      "Analyzing elem 323/682\n",
      "Analyzing elem 324/682\n",
      "Analyzing elem 325/682\n",
      "Analyzing elem 326/682\n",
      "Analyzing elem 327/682\n",
      "Analyzing elem 328/682\n",
      "Analyzing elem 329/682\n",
      "Analyzing elem 330/682\n",
      "Analyzing elem 331/682\n",
      "Analyzing elem 332/682\n",
      "Analyzing elem 333/682\n",
      "Analyzing elem 334/682\n",
      "Analyzing elem 335/682\n",
      "Analyzing elem 336/682\n",
      "Analyzing elem 337/682\n",
      "Analyzing elem 338/682\n",
      "Analyzing elem 339/682\n",
      "Analyzing elem 340/682\n",
      "Analyzing elem 341/682\n",
      "Analyzing elem 342/682\n",
      "Analyzing elem 343/682\n",
      "Analyzing elem 344/682\n",
      "Analyzing elem 345/682\n",
      "Analyzing elem 346/682\n",
      "Analyzing elem 347/682\n",
      "Analyzing elem 348/682\n",
      "Analyzing elem 349/682\n",
      "Analyzing elem 350/682\n",
      "Analyzing elem 351/682\n",
      "Analyzing elem 352/682\n",
      "Analyzing elem 353/682\n",
      "Analyzing elem 354/682\n",
      "Analyzing elem 355/682\n",
      "Analyzing elem 356/682\n",
      "Analyzing elem 357/682\n",
      "Analyzing elem 358/682\n",
      "Analyzing elem 359/682\n",
      "Analyzing elem 360/682\n",
      "Analyzing elem 361/682\n",
      "Analyzing elem 362/682\n",
      "Analyzing elem 363/682\n",
      "Analyzing elem 364/682\n",
      "Analyzing elem 365/682\n",
      "Analyzing elem 366/682\n",
      "Analyzing elem 367/682\n",
      "Analyzing elem 368/682\n",
      "Analyzing elem 369/682\n",
      "Analyzing elem 370/682\n",
      "Analyzing elem 371/682\n",
      "Analyzing elem 372/682\n",
      "Analyzing elem 373/682\n",
      "Analyzing elem 374/682\n",
      "Analyzing elem 375/682\n",
      "Analyzing elem 376/682\n",
      "Analyzing elem 377/682\n",
      "Analyzing elem 378/682\n",
      "Analyzing elem 379/682\n",
      "Analyzing elem 380/682\n",
      "Analyzing elem 381/682\n",
      "Analyzing elem 382/682\n",
      "Analyzing elem 383/682\n",
      "Analyzing elem 384/682\n",
      "Analyzing elem 385/682\n",
      "Analyzing elem 386/682\n",
      "Analyzing elem 387/682\n",
      "Analyzing elem 388/682\n",
      "Analyzing elem 389/682\n",
      "Analyzing elem 390/682\n",
      "Analyzing elem 391/682\n",
      "Analyzing elem 392/682\n",
      "Analyzing elem 393/682\n",
      "Analyzing elem 394/682\n",
      "Analyzing elem 395/682\n",
      "Analyzing elem 396/682\n",
      "Analyzing elem 397/682\n",
      "Analyzing elem 398/682\n",
      "Analyzing elem 399/682\n",
      "Analyzing elem 400/682\n",
      "Analyzing elem 401/682\n",
      "Analyzing elem 402/682\n",
      "Analyzing elem 403/682\n",
      "Analyzing elem 404/682\n",
      "Analyzing elem 405/682\n",
      "Analyzing elem 406/682\n",
      "Analyzing elem 407/682\n",
      "Analyzing elem 408/682\n",
      "Analyzing elem 409/682\n",
      "Analyzing elem 410/682\n",
      "Analyzing elem 411/682\n",
      "Analyzing elem 412/682\n",
      "Analyzing elem 413/682\n",
      "Analyzing elem 414/682\n",
      "Analyzing elem 415/682\n",
      "Analyzing elem 416/682\n",
      "Analyzing elem 417/682\n",
      "Analyzing elem 418/682\n",
      "Analyzing elem 419/682\n",
      "Analyzing elem 420/682\n",
      "Analyzing elem 421/682\n",
      "Analyzing elem 422/682\n",
      "Analyzing elem 423/682\n",
      "Analyzing elem 424/682\n",
      "Analyzing elem 425/682\n",
      "Analyzing elem 426/682\n",
      "Analyzing elem 427/682\n",
      "Analyzing elem 428/682\n",
      "Analyzing elem 429/682\n",
      "Analyzing elem 430/682\n",
      "Analyzing elem 431/682\n",
      "Analyzing elem 432/682\n",
      "Analyzing elem 433/682\n",
      "Analyzing elem 434/682\n",
      "Analyzing elem 435/682\n",
      "Analyzing elem 436/682\n",
      "Analyzing elem 437/682\n",
      "Analyzing elem 438/682\n",
      "Analyzing elem 439/682\n",
      "Analyzing elem 440/682\n",
      "Analyzing elem 441/682\n",
      "Analyzing elem 442/682\n",
      "Analyzing elem 443/682\n",
      "Analyzing elem 444/682\n",
      "Analyzing elem 445/682\n",
      "Analyzing elem 446/682\n",
      "Analyzing elem 447/682\n",
      "Analyzing elem 448/682\n",
      "Analyzing elem 449/682\n",
      "Analyzing elem 450/682\n",
      "Analyzing elem 451/682\n",
      "Analyzing elem 452/682\n",
      "Analyzing elem 453/682\n",
      "Analyzing elem 454/682\n",
      "Analyzing elem 455/682\n",
      "Analyzing elem 456/682\n",
      "Analyzing elem 457/682\n",
      "Analyzing elem 458/682\n",
      "Analyzing elem 459/682\n",
      "Analyzing elem 460/682\n",
      "Analyzing elem 461/682\n",
      "Analyzing elem 462/682\n",
      "Analyzing elem 463/682\n",
      "Analyzing elem 464/682\n",
      "Analyzing elem 465/682\n",
      "Analyzing elem 466/682\n",
      "Analyzing elem 467/682\n",
      "Analyzing elem 468/682\n",
      "Analyzing elem 469/682\n",
      "Analyzing elem 470/682\n",
      "Analyzing elem 471/682\n",
      "Analyzing elem 472/682\n",
      "Analyzing elem 473/682\n",
      "Analyzing elem 474/682\n",
      "Analyzing elem 475/682\n",
      "Analyzing elem 476/682\n",
      "Analyzing elem 477/682\n",
      "Analyzing elem 478/682\n",
      "Analyzing elem 479/682\n",
      "Analyzing elem 480/682\n",
      "Analyzing elem 481/682\n",
      "Analyzing elem 482/682\n",
      "Analyzing elem 483/682\n",
      "Analyzing elem 484/682\n",
      "Analyzing elem 485/682\n",
      "Analyzing elem 486/682\n",
      "Analyzing elem 487/682\n",
      "Analyzing elem 488/682\n",
      "Analyzing elem 489/682\n",
      "Analyzing elem 490/682\n",
      "Analyzing elem 491/682\n",
      "Analyzing elem 492/682\n",
      "Analyzing elem 493/682\n",
      "Analyzing elem 494/682\n",
      "Analyzing elem 495/682\n",
      "Analyzing elem 496/682\n",
      "Analyzing elem 497/682\n",
      "Analyzing elem 498/682\n",
      "Analyzing elem 499/682\n",
      "Analyzing elem 500/682\n",
      "Analyzing elem 501/682\n",
      "Analyzing elem 502/682\n",
      "Analyzing elem 503/682\n",
      "Analyzing elem 504/682\n",
      "Analyzing elem 505/682\n",
      "Analyzing elem 506/682\n",
      "Analyzing elem 507/682\n",
      "Analyzing elem 508/682\n",
      "Analyzing elem 509/682\n",
      "Analyzing elem 510/682\n",
      "Analyzing elem 511/682\n",
      "Analyzing elem 512/682\n",
      "Analyzing elem 513/682\n",
      "Analyzing elem 514/682\n",
      "Analyzing elem 515/682\n",
      "Analyzing elem 516/682\n",
      "Analyzing elem 517/682\n",
      "Analyzing elem 518/682\n",
      "Analyzing elem 519/682\n",
      "Analyzing elem 520/682\n",
      "Analyzing elem 521/682\n",
      "Analyzing elem 522/682\n",
      "Analyzing elem 523/682\n",
      "Analyzing elem 524/682\n",
      "Analyzing elem 525/682\n",
      "Analyzing elem 526/682\n",
      "Analyzing elem 527/682\n",
      "Analyzing elem 528/682\n",
      "Analyzing elem 529/682\n",
      "Analyzing elem 530/682\n",
      "Analyzing elem 531/682\n",
      "Analyzing elem 532/682\n",
      "Analyzing elem 533/682\n",
      "Analyzing elem 534/682\n",
      "Analyzing elem 535/682\n",
      "Analyzing elem 536/682\n",
      "Analyzing elem 537/682\n",
      "Analyzing elem 538/682\n",
      "Analyzing elem 539/682\n",
      "Analyzing elem 540/682\n",
      "Analyzing elem 541/682\n",
      "Analyzing elem 542/682\n",
      "Analyzing elem 543/682\n",
      "Analyzing elem 544/682\n",
      "Analyzing elem 545/682\n",
      "Analyzing elem 546/682\n",
      "Analyzing elem 547/682\n",
      "Analyzing elem 548/682\n",
      "Analyzing elem 549/682\n",
      "Analyzing elem 550/682\n",
      "Analyzing elem 551/682\n",
      "Analyzing elem 552/682\n",
      "Analyzing elem 553/682\n",
      "Analyzing elem 554/682\n",
      "Analyzing elem 555/682\n",
      "Analyzing elem 556/682\n",
      "Analyzing elem 557/682\n",
      "Analyzing elem 558/682\n",
      "Analyzing elem 559/682\n",
      "Analyzing elem 560/682\n",
      "Analyzing elem 561/682\n",
      "Analyzing elem 562/682\n",
      "Analyzing elem 563/682\n",
      "Analyzing elem 564/682\n",
      "Analyzing elem 565/682\n",
      "Analyzing elem 566/682\n",
      "Analyzing elem 567/682\n",
      "Analyzing elem 568/682\n",
      "Analyzing elem 569/682\n",
      "Analyzing elem 570/682\n",
      "Analyzing elem 571/682\n",
      "Analyzing elem 572/682\n",
      "Analyzing elem 573/682\n",
      "Analyzing elem 574/682\n",
      "Analyzing elem 575/682\n",
      "Analyzing elem 576/682\n",
      "Analyzing elem 577/682\n",
      "Analyzing elem 578/682\n",
      "Analyzing elem 579/682\n",
      "Analyzing elem 580/682\n",
      "Analyzing elem 581/682\n",
      "Analyzing elem 582/682\n",
      "Analyzing elem 583/682\n",
      "Analyzing elem 584/682\n",
      "Analyzing elem 585/682\n",
      "Analyzing elem 586/682\n",
      "Analyzing elem 587/682\n",
      "Analyzing elem 588/682\n",
      "Analyzing elem 589/682\n",
      "Analyzing elem 590/682\n",
      "Analyzing elem 591/682\n",
      "Analyzing elem 592/682\n",
      "Analyzing elem 593/682\n",
      "Analyzing elem 594/682\n",
      "Analyzing elem 595/682\n",
      "Analyzing elem 596/682\n",
      "Analyzing elem 597/682\n",
      "Analyzing elem 598/682\n",
      "Analyzing elem 599/682\n",
      "Analyzing elem 600/682\n",
      "Analyzing elem 601/682\n",
      "Analyzing elem 602/682\n",
      "Analyzing elem 603/682\n",
      "Analyzing elem 604/682\n",
      "Analyzing elem 605/682\n",
      "Analyzing elem 606/682\n",
      "Analyzing elem 607/682\n",
      "Analyzing elem 608/682\n",
      "Analyzing elem 609/682\n",
      "Analyzing elem 610/682\n",
      "Analyzing elem 611/682\n",
      "Analyzing elem 612/682\n",
      "Analyzing elem 613/682\n",
      "Analyzing elem 614/682\n",
      "Analyzing elem 615/682\n",
      "Analyzing elem 616/682\n",
      "Analyzing elem 617/682\n",
      "Analyzing elem 618/682\n",
      "Analyzing elem 619/682\n",
      "Analyzing elem 620/682\n",
      "Analyzing elem 621/682\n",
      "Analyzing elem 622/682\n",
      "Analyzing elem 623/682\n",
      "Analyzing elem 624/682\n",
      "Analyzing elem 625/682\n",
      "Analyzing elem 626/682\n",
      "Analyzing elem 627/682\n",
      "Analyzing elem 628/682\n",
      "Analyzing elem 629/682\n",
      "Analyzing elem 630/682\n",
      "Analyzing elem 631/682\n",
      "Analyzing elem 632/682\n",
      "Analyzing elem 633/682\n",
      "Analyzing elem 634/682\n",
      "Analyzing elem 635/682\n",
      "Analyzing elem 636/682\n",
      "Analyzing elem 637/682\n",
      "Analyzing elem 638/682\n",
      "Analyzing elem 639/682\n",
      "Analyzing elem 640/682\n",
      "Analyzing elem 641/682\n",
      "Analyzing elem 642/682\n",
      "Analyzing elem 643/682\n",
      "Analyzing elem 644/682\n",
      "Analyzing elem 645/682\n",
      "Analyzing elem 646/682\n",
      "Analyzing elem 647/682\n",
      "Analyzing elem 648/682\n",
      "Analyzing elem 649/682\n",
      "Analyzing elem 650/682\n",
      "Analyzing elem 651/682\n",
      "Analyzing elem 652/682\n",
      "Analyzing elem 653/682\n",
      "Analyzing elem 654/682\n",
      "Analyzing elem 655/682\n",
      "Analyzing elem 656/682\n",
      "Analyzing elem 657/682\n",
      "Analyzing elem 658/682\n",
      "Analyzing elem 659/682\n",
      "Analyzing elem 660/682\n",
      "Analyzing elem 661/682\n",
      "Analyzing elem 662/682\n",
      "Analyzing elem 663/682\n",
      "Analyzing elem 664/682\n",
      "Analyzing elem 665/682\n",
      "Analyzing elem 666/682\n",
      "Analyzing elem 667/682\n",
      "Analyzing elem 668/682\n",
      "Analyzing elem 669/682\n",
      "Analyzing elem 670/682\n",
      "Analyzing elem 671/682\n",
      "Analyzing elem 672/682\n",
      "Analyzing elem 673/682\n",
      "Analyzing elem 674/682\n",
      "Analyzing elem 675/682\n",
      "Analyzing elem 676/682\n",
      "Analyzing elem 677/682\n",
      "Analyzing elem 678/682\n",
      "Analyzing elem 679/682\n",
      "Analyzing elem 680/682\n",
      "Analyzing elem 681/682\n",
      "Analyzing elem 682/682\n"
     ]
    }
   ],
   "execution_count": 115
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:08.326410Z",
     "start_time": "2024-10-03T08:02:08.313399Z"
    }
   },
   "source": [
    "counter_neg_overlaps = 0\n",
    "counter_neg_not_overlaps = 0\n",
    "for key, value in neg_gtf_dict_2.items():\n",
    "    print(f\"{'='*50}\")\n",
    "    print(f\"{key}:\")\n",
    "    if len(value) > 0:\n",
    "        counter_neg_overlaps += 1\n",
    "        for elem in value:\n",
    "            print(f\"\\t{elem}\")\n",
    "    else:\n",
    "        counter_neg_not_overlaps += 1"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==================================================\n",
      "rejected_noCDS_c01.20:\n",
      "\t{'chrom': 'LinJ.01', 'start': 114146, 'end': 116224, 'strand': '+', 'gene_id': 'LINF_010009600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'nicotinamidase|PNC1', 'interval': Interval(114146, 116224, closed='both')}\n",
      "\t{'chrom': 'LinJ.01', 'start': 114146, 'end': 116224, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_01T0009600', 'parent_id': 'LINF_010009600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'nicotinamidase|PNC1', 'interval': Interval(114146, 116224, closed='both')}\n",
      "\t{'chrom': 'LinJ.01', 'start': 114146, 'end': 114423, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_01T0009600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(114146, 114423, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c01.40:\n",
      "\t{'chrom': 'LinJ.01', 'start': 262069, 'end': 262979, 'strand': '+', 'gene_id': 'LINF_010013350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(262069, 262979, closed='both')}\n",
      "\t{'chrom': 'LinJ.01', 'start': 262069, 'end': 262979, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_01T0013350', 'parent_id': 'LINF_010013350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(262069, 262979, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c01.50:\n",
      "==================================================\n",
      "rejected_noCDS_c01.60A:\n",
      "==================================================\n",
      "rejected_noCDS_c02.140:\n",
      "\t{'chrom': 'LinJ.02', 'start': 137296, 'end': 139716, 'strand': '-', 'gene_id': 'LINF_020007950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(137296, 139716, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 137296, 'end': 139716, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_02T0007950', 'parent_id': 'LINF_020007950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(137296, 139716, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 139717, 'end': 150152, 'strand': '-', 'gene_id': 'LINF_020008000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATP-binding_cassette_subfamily_A_-_member_1|ABCA1', 'interval': Interval(139717, 150152, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 139717, 'end': 150152, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_02T0008000', 'parent_id': 'LINF_020008000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATP-binding_cassette_subfamily_A_-_member_1|ABCA1', 'interval': Interval(139717, 150152, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 139717, 'end': 145477, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_02T0008000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(139717, 145477, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c02.150:\n",
      "\t{'chrom': 'LinJ.02', 'start': 139717, 'end': 150152, 'strand': '-', 'gene_id': 'LINF_020008000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATP-binding_cassette_subfamily_A_-_member_1|ABCA1', 'interval': Interval(139717, 150152, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 139717, 'end': 150152, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_02T0008000', 'parent_id': 'LINF_020008000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATP-binding_cassette_subfamily_A_-_member_1|ABCA1', 'interval': Interval(139717, 150152, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 149222, 'end': 150152, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_02T0008000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(149222, 150152, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 150375, 'end': 155425, 'strand': '-', 'gene_id': 'LINF_020008100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'paraflagellar_rod_component_11|PFC11', 'interval': Interval(150375, 155425, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 150375, 'end': 155425, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_02T0008100', 'parent_id': 'LINF_020008100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'paraflagellar_rod_component_11|PFC11', 'interval': Interval(150375, 155425, closed='both')}\n",
      "\t{'chrom': 'LinJ.02', 'start': 150375, 'end': 153057, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_02T0008100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(150375, 153057, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c02.170:\n",
      "==================================================\n",
      "rejected_noCDS_c03.180:\n",
      "\t{'chrom': 'LinJ.03', 'start': 168578, 'end': 173520, 'strand': '+', 'gene_id': 'LINF_030009600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ankyrin_repeats_(3_copies)/Ankyrin_repeats_(many_copies)/Ankyrin_repeat/DHHC_palmitoyltransferase_-_putative', 'interval': Interval(168578, 173520, closed='both')}\n",
      "\t{'chrom': 'LinJ.03', 'start': 168578, 'end': 173520, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_03T0009600', 'parent_id': 'LINF_030009600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ankyrin_repeats_(3_copies)/Ankyrin_repeats_(many_copies)/Ankyrin_repeat/DHHC_palmitoyltransferase_-_putative', 'interval': Interval(168578, 173520, closed='both')}\n",
      "\t{'chrom': 'LinJ.03', 'start': 172508, 'end': 173520, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_03T0009600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(172508, 173520, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c04.200C:\n",
      "==================================================\n",
      "rejected_noCDS_c04.210C:\n",
      "\t{'chrom': 'LinJ.04', 'start': 126105, 'end': 126374, 'strand': '-', 'gene_id': 'LINF_040008875', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(126105, 126374, closed='both')}\n",
      "\t{'chrom': 'LinJ.04', 'start': 126105, 'end': 126374, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_04T0008875', 'parent_id': 'LINF_040008875', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(126105, 126374, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c04.230:\n",
      "\t{'chrom': 'LinJ.04', 'start': 139396, 'end': 142427, 'strand': '-', 'gene_id': 'LINF_040009200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Tetratricopeptide_repeat-containing_protein', 'interval': Interval(139396, 142427, closed='both')}\n",
      "\t{'chrom': 'LinJ.04', 'start': 139396, 'end': 142427, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_04T0009200', 'parent_id': 'LINF_040009200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Tetratricopeptide_repeat-containing_protein', 'interval': Interval(139396, 142427, closed='both')}\n",
      "\t{'chrom': 'LinJ.04', 'start': 142101, 'end': 142427, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_04T0009200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(142101, 142427, closed='both')}\n",
      "\t{'chrom': 'LinJ.04', 'start': 142650, 'end': 156872, 'strand': '-', 'gene_id': 'LINF_040009300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(142650, 156872, closed='both')}\n",
      "\t{'chrom': 'LinJ.04', 'start': 142650, 'end': 156872, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_04T0009300', 'parent_id': 'LINF_040009300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(142650, 156872, closed='both')}\n",
      "\t{'chrom': 'LinJ.04', 'start': 142650, 'end': 143425, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_04T0009300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(142650, 143425, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c05.280:\n",
      "\t{'chrom': 'LinJ.05', 'start': 1093, 'end': 2315, 'strand': '-', 'gene_id': 'LINF_050004700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1093, 2315, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 1093, 'end': 2315, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_05T0004700', 'parent_id': 'LINF_050004700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1093, 2315, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c05.300D:\n",
      "\t{'chrom': 'LinJ.05', 'start': 66289, 'end': 73912, 'strand': '+', 'gene_id': 'LINF_050007100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'dual_specificity_phosphatase-like_protein', 'interval': Interval(66289, 73912, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 66289, 'end': 73912, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_05T0007100', 'parent_id': 'LINF_050007100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'dual_specificity_phosphatase-like_protein', 'interval': Interval(66289, 73912, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 66289, 'end': 67362, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_05T0007100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(66289, 67362, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c05.320D:\n",
      "\t{'chrom': 'LinJ.05', 'start': 104200, 'end': 106507, 'strand': '+', 'gene_id': 'LINF_050008400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'mitoribosomal_protein_uS11m', 'interval': Interval(104200, 106507, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 104200, 'end': 106507, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_05T0008400', 'parent_id': 'LINF_050008400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'mitoribosomal_protein_uS11m', 'interval': Interval(104200, 106507, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 105430, 'end': 106507, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_05T0008400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(105430, 106507, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 106508, 'end': 108915, 'strand': '+', 'gene_id': 'LINF_050008500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'trypanothione_reductase_(TRYR)', 'interval': Interval(106508, 108915, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 106508, 'end': 108915, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_05T0008500', 'parent_id': 'LINF_050008500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'trypanothione_reductase_(TRYR)', 'interval': Interval(106508, 108915, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 106508, 'end': 106885, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_05T0008500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(106508, 106885, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c05.350:\n",
      "\t{'chrom': 'LinJ.05', 'start': 357004, 'end': 360467, 'strand': '+', 'gene_id': 'LINF_050014700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glutaminyl_cyclase', 'interval': Interval(357004, 360467, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 357004, 'end': 360467, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_05T0014700', 'parent_id': 'LINF_050014700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glutaminyl_cyclase', 'interval': Interval(357004, 360467, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 359831, 'end': 360467, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_05T0014700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(359831, 360467, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 360956, 'end': 363428, 'strand': '+', 'gene_id': 'LINF_050014800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'dipeptidyl-peptidase_III|metallo-peptidase_-_Clan_M_-_Family_M49', 'interval': Interval(360956, 363428, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 360956, 'end': 363428, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_05T0014800', 'parent_id': 'LINF_050014800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'dipeptidyl-peptidase_III|metallo-peptidase_-_Clan_M_-_Family_M49', 'interval': Interval(360956, 363428, closed='both')}\n",
      "\t{'chrom': 'LinJ.05', 'start': 360956, 'end': 361110, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_05T0014800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(360956, 361110, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c05.360F:\n",
      "==================================================\n",
      "rejected_noCDS_c06.380:\n",
      "\t{'chrom': 'LinJ.06', 'start': 59844, 'end': 62540, 'strand': '-', 'gene_id': 'LINF_060006900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'kinesin-like_protein', 'interval': Interval(59844, 62540, closed='both')}\n",
      "\t{'chrom': 'LinJ.06', 'start': 59844, 'end': 62540, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_06T0006900', 'parent_id': 'LINF_060006900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'kinesin-like_protein', 'interval': Interval(59844, 62540, closed='both')}\n",
      "\t{'chrom': 'LinJ.06', 'start': 61911, 'end': 62540, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_06T0006900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(61911, 62540, closed='both')}\n",
      "\t{'chrom': 'LinJ.06', 'start': 62983, 'end': 65883, 'strand': '-', 'gene_id': 'LINF_060007000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(62983, 65883, closed='both')}\n",
      "\t{'chrom': 'LinJ.06', 'start': 62983, 'end': 65883, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_06T0007000', 'parent_id': 'LINF_060007000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(62983, 65883, closed='both')}\n",
      "\t{'chrom': 'LinJ.06', 'start': 62983, 'end': 63490, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_06T0007000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(62983, 63490, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c06.400:\n",
      "==================================================\n",
      "rejected_noCDS_c07.430G:\n",
      "\t{'chrom': 'LinJ.07', 'start': 211168, 'end': 211744, 'strand': '+', 'gene_id': 'LINF_070010140', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(211168, 211744, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 211168, 'end': 211744, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_07T0010140', 'parent_id': 'LINF_070010140', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(211168, 211744, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 211951, 'end': 214227, 'strand': '+', 'gene_id': 'LINF_070010160', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(211951, 214227, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 211951, 'end': 214227, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_07T0010160', 'parent_id': 'LINF_070010160', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(211951, 214227, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c07.510:\n",
      "\t{'chrom': 'LinJ.07', 'start': 384350, 'end': 384937, 'strand': '+', 'gene_id': 'LINF_070013950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(384350, 384937, closed='both')}\n",
      "\t{'chrom': 'LinJ.07', 'start': 384350, 'end': 384937, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_07T0013950', 'parent_id': 'LINF_070013950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(384350, 384937, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.550:\n",
      "\t{'chrom': 'LinJ.08', 'start': 190115, 'end': 192664, 'strand': '+', 'gene_id': 'LINF_080009800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'small_ubiquitin-like_modifier_(SUMO)', 'interval': Interval(190115, 192664, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 190115, 'end': 192664, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0009800', 'parent_id': 'LINF_080009800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'small_ubiquitin-like_modifier_(SUMO)', 'interval': Interval(190115, 192664, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 190647, 'end': 192664, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_08T0009800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(190647, 192664, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.560I:\n",
      "\t{'chrom': 'LinJ.08', 'start': 298159, 'end': 299531, 'strand': '+', 'gene_id': 'LINF_080012050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(298159, 299531, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 298159, 'end': 299531, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012050', 'parent_id': 'LINF_080012050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(298159, 299531, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.590I:\n",
      "\t{'chrom': 'LinJ.08', 'start': 302787, 'end': 304302, 'strand': '+', 'gene_id': 'LINF_080012150', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(302787, 304302, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 302787, 'end': 304302, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012150', 'parent_id': 'LINF_080012150', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(302787, 304302, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.620I:\n",
      "\t{'chrom': 'LinJ.08', 'start': 307412, 'end': 308886, 'strand': '+', 'gene_id': 'LINF_080012250', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(307412, 308886, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 307412, 'end': 308886, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012250', 'parent_id': 'LINF_080012250', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(307412, 308886, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.640I:\n",
      "\t{'chrom': 'LinJ.08', 'start': 311996, 'end': 313821, 'strand': '+', 'gene_id': 'LINF_080012350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(311996, 313821, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 311996, 'end': 313821, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012350', 'parent_id': 'LINF_080012350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(311996, 313821, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.670:\n",
      "\t{'chrom': 'LinJ.08', 'start': 316292, 'end': 318363, 'strand': '+', 'gene_id': 'LINF_080012450', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(316292, 318363, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 316292, 'end': 318363, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012450', 'parent_id': 'LINF_080012450', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(316292, 318363, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.680K:\n",
      "\t{'chrom': 'LinJ.08', 'start': 324605, 'end': 326842, 'strand': '+', 'gene_id': 'LINF_080012700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'tuzin', 'interval': Interval(324605, 326842, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 324605, 'end': 326842, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012700', 'parent_id': 'LINF_080012700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'tuzin', 'interval': Interval(324605, 326842, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 326394, 'end': 326842, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_08T0012700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(326394, 326842, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.710K:\n",
      "\t{'chrom': 'LinJ.08', 'start': 330313, 'end': 332015, 'strand': '+', 'gene_id': 'LINF_080012825', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(330313, 332015, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 330313, 'end': 332015, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012825', 'parent_id': 'LINF_080012825', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(330313, 332015, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.720N:\n",
      "\t{'chrom': 'LinJ.08', 'start': 332481, 'end': 335118, 'strand': '+', 'gene_id': 'LINF_080012850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(332481, 335118, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 332481, 'end': 335118, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012850', 'parent_id': 'LINF_080012850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(332481, 335118, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.760K:\n",
      "\t{'chrom': 'LinJ.08', 'start': 335486, 'end': 337341, 'strand': '+', 'gene_id': 'LINF_080012875', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(335486, 337341, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 335486, 'end': 337341, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012875', 'parent_id': 'LINF_080012875', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(335486, 337341, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.770N:\n",
      "\t{'chrom': 'LinJ.08', 'start': 337654, 'end': 340312, 'strand': '+', 'gene_id': 'LINF_080012900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_protein', 'interval': Interval(337654, 340312, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 337654, 'end': 340312, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012900', 'parent_id': 'LINF_080012900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_protein', 'interval': Interval(337654, 340312, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 337654, 'end': 337858, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_08T0012900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(337654, 337858, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.800K:\n",
      "\t{'chrom': 'LinJ.08', 'start': 340660, 'end': 342358, 'strand': '+', 'gene_id': 'LINF_080012925', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(340660, 342358, closed='both')}\n",
      "\t{'chrom': 'LinJ.08', 'start': 340660, 'end': 342358, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_08T0012925', 'parent_id': 'LINF_080012925', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(340660, 342358, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c08.850:\n",
      "==================================================\n",
      "rejected_noCDS_c08.860:\n",
      "==================================================\n",
      "rejected_noCDS_c10.980O:\n",
      "==================================================\n",
      "rejected_noCDS_c10.1010:\n",
      "\t{'chrom': 'LinJ.10', 'start': 71683, 'end': 73187, 'strand': '+', 'gene_id': 'LINF_100006550', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(71683, 73187, closed='both')}\n",
      "\t{'chrom': 'LinJ.10', 'start': 71683, 'end': 73187, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_10T0006550', 'parent_id': 'LINF_100006550', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(71683, 73187, closed='both')}\n",
      "\t{'chrom': 'LinJ.10', 'start': 73188, 'end': 76489, 'strand': '+', 'gene_id': 'LINF_100006600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'guanosine_diphosphatase|NTPD2', 'interval': Interval(73188, 76489, closed='both')}\n",
      "\t{'chrom': 'LinJ.10', 'start': 73188, 'end': 76489, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_10T0006600', 'parent_id': 'LINF_100006600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'guanosine_diphosphatase|NTPD2', 'interval': Interval(73188, 76489, closed='both')}\n",
      "\t{'chrom': 'LinJ.10', 'start': 73188, 'end': 73389, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_10T0006600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(73188, 73389, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c11.1070:\n",
      "\t{'chrom': 'LinJ.11', 'start': 161231, 'end': 163190, 'strand': '-', 'gene_id': 'LINF_110009720', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(161231, 163190, closed='both')}\n",
      "\t{'chrom': 'LinJ.11', 'start': 161231, 'end': 163190, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_11T0009720', 'parent_id': 'LINF_110009720', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(161231, 163190, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c11.1080:\n",
      "\t{'chrom': 'LinJ.11', 'start': 190410, 'end': 194177, 'strand': '+', 'gene_id': 'LINF_110011200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'purine_nucleobase_transporter|NT4', 'interval': Interval(190410, 194177, closed='both')}\n",
      "\t{'chrom': 'LinJ.11', 'start': 190410, 'end': 194177, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_11T0011200', 'parent_id': 'LINF_110011200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'purine_nucleobase_transporter|NT4', 'interval': Interval(190410, 194177, closed='both')}\n",
      "\t{'chrom': 'LinJ.11', 'start': 192469, 'end': 194177, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_11T0011200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(192469, 194177, closed='both')}\n",
      "\t{'chrom': 'LinJ.11', 'start': 194178, 'end': 196953, 'strand': '+', 'gene_id': 'LINF_110011250', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(194178, 196953, closed='both')}\n",
      "\t{'chrom': 'LinJ.11', 'start': 194178, 'end': 196953, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_11T0011250', 'parent_id': 'LINF_110011250', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(194178, 196953, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1120:\n",
      "==================================================\n",
      "rejected_noCDS_c12.1170:\n",
      "==================================================\n",
      "rejected_noCDS_c12.1180S:\n",
      "\t{'chrom': 'LinJ.12', 'start': 372226, 'end': 374050, 'strand': '+', 'gene_id': 'LINF_120012350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_of_unknown_function_(DUF962),_putative', 'interval': Interval(372226, 374050, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 372226, 'end': 374050, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0012350', 'parent_id': 'LINF_120012350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_of_unknown_function_(DUF962),_putative', 'interval': Interval(372226, 374050, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 372831, 'end': 374050, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0012350', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(372831, 374050, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 374051, 'end': 377775, 'strand': '+', 'gene_id': 'LINF_120012400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(374051, 377775, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 374051, 'end': 377775, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0012400', 'parent_id': 'LINF_120012400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(374051, 377775, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 374051, 'end': 375027, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0012400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(374051, 375027, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1190:\n",
      "\t{'chrom': 'LinJ.12', 'start': 374051, 'end': 377775, 'strand': '+', 'gene_id': 'LINF_120012400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(374051, 377775, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 374051, 'end': 377775, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0012400', 'parent_id': 'LINF_120012400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(374051, 377775, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 375514, 'end': 377775, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0012400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(375514, 377775, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1200S:\n",
      "\t{'chrom': 'LinJ.12', 'start': 389565, 'end': 393477, 'strand': '+', 'gene_id': 'LINF_120012800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ribonuclease_inhibitor', 'interval': Interval(389565, 393477, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 389565, 'end': 393477, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0012800', 'parent_id': 'LINF_120012800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Ribonuclease_inhibitor', 'interval': Interval(389565, 393477, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 390949, 'end': 393477, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0012800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(390949, 393477, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 393788, 'end': 397393, 'strand': '+', 'gene_id': 'LINF_120012900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(393788, 397393, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 393788, 'end': 397393, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0012900', 'parent_id': 'LINF_120012900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(393788, 397393, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 393788, 'end': 394522, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0012900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(393788, 394522, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1210S:\n",
      "\t{'chrom': 'LinJ.12', 'start': 401779, 'end': 406034, 'strand': '+', 'gene_id': 'LINF_120013100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'promastigote_surface_antigen_38S', 'interval': Interval(401779, 406034, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 401779, 'end': 406034, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0013100', 'parent_id': 'LINF_120013100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'promastigote_surface_antigen_38S', 'interval': Interval(401779, 406034, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 403504, 'end': 406034, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0013100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(403504, 406034, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 406344, 'end': 409449, 'strand': '+', 'gene_id': 'LINF_120013300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(406344, 409449, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 406344, 'end': 409449, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0013300', 'parent_id': 'LINF_120013300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(406344, 409449, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 406344, 'end': 407078, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0013300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(406344, 407078, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1220S:\n",
      "\t{'chrom': 'LinJ.12', 'start': 413336, 'end': 417374, 'strand': '+', 'gene_id': 'LINF_120013500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Surface_antigen_protein_2', 'interval': Interval(413336, 417374, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 413336, 'end': 417374, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0013500', 'parent_id': 'LINF_120013500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Surface_antigen_protein_2', 'interval': Interval(413336, 417374, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 415192, 'end': 417374, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0013500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(415192, 417374, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 417691, 'end': 421308, 'strand': '+', 'gene_id': 'LINF_120013560', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(417691, 421308, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 417691, 'end': 421308, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0013560', 'parent_id': 'LINF_120013560', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(417691, 421308, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1240S:\n",
      "\t{'chrom': 'LinJ.12', 'start': 425243, 'end': 429221, 'strand': '+', 'gene_id': 'LINF_120013700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'surface_antigen_protein_2_-_putative', 'interval': Interval(425243, 429221, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 425243, 'end': 429221, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0013700', 'parent_id': 'LINF_120013700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'surface_antigen_protein_2_-_putative', 'interval': Interval(425243, 429221, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 426862, 'end': 429221, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0013700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(426862, 429221, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 429222, 'end': 431627, 'strand': '+', 'gene_id': 'LINF_120013800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(429222, 431627, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 429222, 'end': 431627, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0013800', 'parent_id': 'LINF_120013800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(429222, 431627, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 429222, 'end': 429627, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0013800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(429222, 429627, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c12.1250S:\n",
      "\t{'chrom': 'LinJ.12', 'start': 461643, 'end': 466202, 'strand': '+', 'gene_id': 'LINF_120015000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Surface_antigenic_protein|PSA-2', 'interval': Interval(461643, 466202, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 461643, 'end': 466202, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0015000', 'parent_id': 'LINF_120015000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Surface_antigenic_protein|PSA-2', 'interval': Interval(461643, 466202, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 463757, 'end': 466202, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_12T0015000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(463757, 466202, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 466522, 'end': 469621, 'strand': '+', 'gene_id': 'LINF_120015050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(466522, 469621, closed='both')}\n",
      "\t{'chrom': 'LinJ.12', 'start': 466522, 'end': 469621, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_12T0015050', 'parent_id': 'LINF_120015050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(466522, 469621, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c13.1400:\n",
      "\t{'chrom': 'LinJ.13', 'start': 413449, 'end': 414617, 'strand': '-', 'gene_id': 'LINF_130016950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(413449, 414617, closed='both')}\n",
      "\t{'chrom': 'LinJ.13', 'start': 413449, 'end': 414617, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_13T0016950', 'parent_id': 'LINF_130016950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(413449, 414617, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c14.1420:\n",
      "==================================================\n",
      "rejected_noCDS_c14.1530:\n",
      "==================================================\n",
      "rejected_noCDS_c14.1540W:\n",
      "\t{'chrom': 'LinJ.14', 'start': 655102, 'end': 656077, 'strand': '+', 'gene_id': 'LINF_140021450', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(655102, 656077, closed='both')}\n",
      "\t{'chrom': 'LinJ.14', 'start': 655102, 'end': 656077, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_14T0021450', 'parent_id': 'LINF_140021450', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(655102, 656077, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c15.1580:\n",
      "\t{'chrom': 'LinJ.15', 'start': 151001, 'end': 152466, 'strand': '+', 'gene_id': 'LINF_150009600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(151001, 152466, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 151001, 'end': 152466, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_15T0009600', 'parent_id': 'LINF_150009600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(151001, 152466, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 151500, 'end': 152466, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_15T0009600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(151500, 152466, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 152817, 'end': 154532, 'strand': '+', 'gene_id': 'LINF_150009700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(152817, 154532, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 152817, 'end': 154532, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_15T0009700', 'parent_id': 'LINF_150009700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(152817, 154532, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 152817, 'end': 153094, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_15T0009700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(152817, 153094, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c15.1590X:\n",
      "\t{'chrom': 'LinJ.15', 'start': 286973, 'end': 288730, 'strand': '+', 'gene_id': 'LINF_150013750', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(286973, 288730, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 286973, 'end': 288730, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_15T0013750', 'parent_id': 'LINF_150013750', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(286973, 288730, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c15.1610X:\n",
      "\t{'chrom': 'LinJ.15', 'start': 306215, 'end': 309145, 'strand': '+', 'gene_id': 'LINF_150014000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(306215, 309145, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 306215, 'end': 309145, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_15T0014000', 'parent_id': 'LINF_150014000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(306215, 309145, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 308258, 'end': 309145, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_15T0014000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(308258, 309145, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 309377, 'end': 310768, 'strand': '+', 'gene_id': 'LINF_150014050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(309377, 310768, closed='both')}\n",
      "\t{'chrom': 'LinJ.15', 'start': 309377, 'end': 310768, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_15T0014050', 'parent_id': 'LINF_150014050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(309377, 310768, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c16.1720:\n",
      "==================================================\n",
      "rejected_noCDS_c17.1830Z:\n",
      "\t{'chrom': 'LinJ.17', 'start': 931, 'end': 2725, 'strand': '-', 'gene_id': 'LINF_170004900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(931, 2725, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 931, 'end': 2725, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_17T0004900', 'parent_id': 'LINF_170004900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(931, 2725, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c17.1900:\n",
      "\t{'chrom': 'LinJ.17', 'start': 454894, 'end': 459103, 'strand': '+', 'gene_id': 'LINF_170016500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Phosphorylated_CTD_interacting_factor_1_WW_domain_containing_protein_-_putative', 'interval': Interval(454894, 459103, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 454894, 'end': 459103, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_17T0016500', 'parent_id': 'LINF_170016500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Phosphorylated_CTD_interacting_factor_1_WW_domain_containing_protein_-_putative', 'interval': Interval(454894, 459103, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 458150, 'end': 459103, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_17T0016500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(458150, 459103, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c17.1910:\n",
      "==================================================\n",
      "rejected_noCDS_c17.1940AA:\n",
      "\t{'chrom': 'LinJ.17', 'start': 562538, 'end': 568030, 'strand': '+', 'gene_id': 'LINF_170018700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'WW_domain_containing_protein_-_putative', 'interval': Interval(562538, 568030, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 562538, 'end': 568030, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_17T0018700', 'parent_id': 'LINF_170018700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'WW_domain_containing_protein_-_putative', 'interval': Interval(562538, 568030, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 562538, 'end': 562587, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_17T0018700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(562538, 562587, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c17.1970Z:\n",
      "\t{'chrom': 'LinJ.17', 'start': 687597, 'end': 689967, 'strand': '+', 'gene_id': 'LINF_170022200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(687597, 689967, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 687597, 'end': 689967, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_17T0022200', 'parent_id': 'LINF_170022200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(687597, 689967, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 688415, 'end': 689967, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_17T0022200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(688415, 689967, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 690110, 'end': 690871, 'strand': '+', 'gene_id': 'LINF_170022300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(690110, 690871, closed='both')}\n",
      "\t{'chrom': 'LinJ.17', 'start': 690110, 'end': 690871, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_17T0022300', 'parent_id': 'LINF_170022300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(690110, 690871, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c18.1980:\n",
      "==================================================\n",
      "rejected_noCDS_c18.1990:\n",
      "\t{'chrom': 'LinJ.18', 'start': 162150, 'end': 167192, 'strand': '-', 'gene_id': 'LINF_180009000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5B', 'interval': Interval(162150, 167192, closed='both')}\n",
      "\t{'chrom': 'LinJ.18', 'start': 162150, 'end': 167192, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_18T0009000', 'parent_id': 'LINF_180009000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5B', 'interval': Interval(162150, 167192, closed='both')}\n",
      "\t{'chrom': 'LinJ.18', 'start': 165272, 'end': 167192, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_18T0009000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(165272, 167192, closed='both')}\n",
      "\t{'chrom': 'LinJ.18', 'start': 167193, 'end': 169438, 'strand': '-', 'gene_id': 'LINF_180009100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(167193, 169438, closed='both')}\n",
      "\t{'chrom': 'LinJ.18', 'start': 167193, 'end': 169438, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_18T0009100', 'parent_id': 'LINF_180009100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(167193, 169438, closed='both')}\n",
      "\t{'chrom': 'LinJ.18', 'start': 167193, 'end': 168037, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_18T0009100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(167193, 168037, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c18.2000:\n",
      "\t{'chrom': 'LinJ.18', 'start': 238666, 'end': 239602, 'strand': '+', 'gene_id': 'LINF_180010950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(238666, 239602, closed='both')}\n",
      "\t{'chrom': 'LinJ.18', 'start': 238666, 'end': 239602, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_18T0010950', 'parent_id': 'LINF_180010950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(238666, 239602, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2110:\n",
      "\t{'chrom': 'LinJ.19', 'start': 354993, 'end': 355737, 'strand': '+', 'gene_id': 'LINF_190013715', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATG8/AUT7/APG8/PAZ2', 'interval': Interval(354993, 355737, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 354993, 'end': 355737, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_19T0013715', 'parent_id': 'LINF_190013715', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATG8/AUT7/APG8/PAZ2', 'interval': Interval(354993, 355737, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 355034, 'end': 355399, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_19T0013715', 'parent_id': 'LINF_19T0013715', 'gene': False, 'transcript': False, 'CDS': True, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(355034, 355399, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 354993, 'end': 355033, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_19T0013715', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(354993, 355033, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 355400, 'end': 355737, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_19T0013715', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(355400, 355737, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2120AD:\n",
      "\t{'chrom': 'LinJ.19', 'start': 361081, 'end': 362554, 'strand': '+', 'gene_id': 'LINF_190013900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATG8/AUT7/APG8/PAZ2', 'interval': Interval(361081, 362554, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 361081, 'end': 362554, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_19T0013900', 'parent_id': 'LINF_190013900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATG8/AUT7/APG8/PAZ2', 'interval': Interval(361081, 362554, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 361488, 'end': 362554, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_19T0013900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(361488, 362554, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2130AD:\n",
      "\t{'chrom': 'LinJ.19', 'start': 364663, 'end': 366138, 'strand': '+', 'gene_id': 'LINF_190014100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATG8/AUT7/APG8/PAZ2', 'interval': Interval(364663, 366138, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 364663, 'end': 366138, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_19T0014100', 'parent_id': 'LINF_190014100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'ATG8/AUT7/APG8/PAZ2', 'interval': Interval(364663, 366138, closed='both')}\n",
      "\t{'chrom': 'LinJ.19', 'start': 365070, 'end': 366138, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_19T0014100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(365070, 366138, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c19.2160:\n",
      "==================================================\n",
      "rejected_noCDS_c19.2230:\n",
      "==================================================\n",
      "rejected_noCDS_c20.2250:\n",
      "\t{'chrom': 'LinJ.20', 'start': 134609, 'end': 135924, 'strand': '+', 'gene_id': 'LINF_200008850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(134609, 135924, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 134609, 'end': 135924, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_20T0008850', 'parent_id': 'LINF_200008850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(134609, 135924, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 135925, 'end': 140156, 'strand': '+', 'gene_id': 'LINF_200008900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_of_unknown_function_(DUF2946)_-_putative', 'interval': Interval(135925, 140156, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 135925, 'end': 140156, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_20T0008900', 'parent_id': 'LINF_200008900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_of_unknown_function_(DUF2946)_-_putative', 'interval': Interval(135925, 140156, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 135925, 'end': 136313, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_20T0008900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(135925, 136313, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c20.2260:\n",
      "\t{'chrom': 'LinJ.20', 'start': 245715, 'end': 247334, 'strand': '+', 'gene_id': 'LINF_200012000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(245715, 247334, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 245715, 'end': 247334, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_20T0012000', 'parent_id': 'LINF_200012000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(245715, 247334, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 246773, 'end': 247334, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_20T0012000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(246773, 247334, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 247335, 'end': 249516, 'strand': '+', 'gene_id': 'LINF_200012100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_of_unknown_function_(DUF775)_-_putative', 'interval': Interval(247335, 249516, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 247335, 'end': 249516, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_20T0012100', 'parent_id': 'LINF_200012100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_of_unknown_function_(DUF775)_-_putative', 'interval': Interval(247335, 249516, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 247335, 'end': 248221, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_20T0012100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(247335, 248221, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c20.2310:\n",
      "\t{'chrom': 'LinJ.20', 'start': 432463, 'end': 433723, 'strand': '+', 'gene_id': 'LINF_200015450', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(432463, 433723, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 432463, 'end': 433723, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_20T0015450', 'parent_id': 'LINF_200015450', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(432463, 433723, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 433724, 'end': 439654, 'strand': '+', 'gene_id': 'LINF_200015500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(433724, 439654, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 433724, 'end': 439654, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_20T0015500', 'parent_id': 'LINF_200015500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(433724, 439654, closed='both')}\n",
      "\t{'chrom': 'LinJ.20', 'start': 433724, 'end': 434033, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_20T0015500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(433724, 434033, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c21.2390:\n",
      "\t{'chrom': 'LinJ.21', 'start': 488393, 'end': 489994, 'strand': '-', 'gene_id': 'LINF_210019900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_of_unknown_function_-_conserved_(L1p/L10e_family)', 'interval': Interval(488393, 489994, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 488393, 'end': 489994, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_21T0019900', 'parent_id': 'LINF_210019900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Protein_of_unknown_function_-_conserved_(L1p/L10e_family)', 'interval': Interval(488393, 489994, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 488393, 'end': 489031, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_21T0019900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(488393, 489031, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c21.2460Z:\n",
      "\t{'chrom': 'LinJ.21', 'start': 763461, 'end': 764506, 'strand': '-', 'gene_id': 'LINF_210028800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(763461, 764506, closed='both')}\n",
      "\t{'chrom': 'LinJ.21', 'start': 763461, 'end': 764506, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_21T0028800', 'parent_id': 'LINF_210028800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(763461, 764506, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c22.2480:\n",
      "\t{'chrom': 'LinJ.22', 'start': 61724, 'end': 62907, 'strand': '-', 'gene_id': 'LINF_220006650', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(61724, 62907, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 61724, 'end': 62907, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_22T0006650', 'parent_id': 'LINF_220006650', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(61724, 62907, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 63270, 'end': 67965, 'strand': '-', 'gene_id': 'LINF_220006700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(63270, 67965, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 63270, 'end': 67965, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_22T0006700', 'parent_id': 'LINF_220006700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(63270, 67965, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 63270, 'end': 64084, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_22T0006700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(63270, 64084, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c22.2540A:\n",
      "\t{'chrom': 'LinJ.22', 'start': 751020, 'end': 754538, 'strand': '+', 'gene_id': 'LINF_220023175', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(751020, 754538, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 751020, 'end': 754538, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_22T0023175', 'parent_id': 'LINF_220023175', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(751020, 754538, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 754539, 'end': 782139, 'strand': '+', 'gene_id': 'LINF_220023200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(754539, 782139, closed='both')}\n",
      "\t{'chrom': 'LinJ.22', 'start': 754539, 'end': 782139, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_22T0023200', 'parent_id': 'LINF_220023200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(754539, 782139, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c23.2560:\n",
      "\t{'chrom': 'LinJ.23', 'start': 55800, 'end': 57308, 'strand': '+', 'gene_id': 'LINF_230006950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(55800, 57308, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 55800, 'end': 57308, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_23T0006950', 'parent_id': 'LINF_230006950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(55800, 57308, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c23.2570:\n",
      "\t{'chrom': 'LinJ.23', 'start': 85852, 'end': 87903, 'strand': '+', 'gene_id': 'LINF_230007700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'terbinafine_resistance_locus_protein_(yip1)|HTBF', 'interval': Interval(85852, 87903, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 85852, 'end': 87903, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_23T0007700', 'parent_id': 'LINF_230007700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'terbinafine_resistance_locus_protein_(yip1)|HTBF', 'interval': Interval(85852, 87903, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 86943, 'end': 87903, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_23T0007700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(86943, 87903, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c23.2640AI:\n",
      "==================================================\n",
      "rejected_noCDS_c23.2650AI:\n",
      "==================================================\n",
      "rejected_noCDS_c23.2690:\n",
      "\t{'chrom': 'LinJ.23', 'start': 733585, 'end': 736847, 'strand': '+', 'gene_id': 'LINF_230024900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(733585, 736847, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 733585, 'end': 736847, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_23T0024900', 'parent_id': 'LINF_230024900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(733585, 736847, closed='both')}\n",
      "\t{'chrom': 'LinJ.23', 'start': 734379, 'end': 736847, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_23T0024900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(734379, 736847, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c24.2710AJ:\n",
      "\t{'chrom': 'LinJ.24', 'start': 108132, 'end': 110204, 'strand': '+', 'gene_id': 'LINF_240008250', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(108132, 110204, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 108132, 'end': 110204, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_24T0008250', 'parent_id': 'LINF_240008250', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(108132, 110204, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 110205, 'end': 117720, 'strand': '+', 'gene_id': 'LINF_240008300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5A', 'interval': Interval(110205, 117720, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 110205, 'end': 117720, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_24T0008300', 'parent_id': 'LINF_240008300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5A', 'interval': Interval(110205, 117720, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 110205, 'end': 110267, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_24T0008300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(110205, 110267, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c24.2770AJ:\n",
      "\t{'chrom': 'LinJ.24', 'start': 110205, 'end': 117720, 'strand': '+', 'gene_id': 'LINF_240008300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5A', 'interval': Interval(110205, 117720, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 110205, 'end': 117720, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_24T0008300', 'parent_id': 'LINF_240008300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5A', 'interval': Interval(110205, 117720, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 111675, 'end': 117720, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_24T0008300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(111675, 117720, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 117721, 'end': 122337, 'strand': '+', 'gene_id': 'LINF_240008400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5A', 'interval': Interval(117721, 122337, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 117721, 'end': 122337, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_24T0008400', 'parent_id': 'LINF_240008400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'UDP-galactose_transporter|LPG5A', 'interval': Interval(117721, 122337, closed='both')}\n",
      "\t{'chrom': 'LinJ.24', 'start': 117721, 'end': 117783, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_24T0008400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(117721, 117783, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c25.2850AK:\n",
      "\t{'chrom': 'LinJ.25', 'start': 799, 'end': 5363, 'strand': '-', 'gene_id': 'LINF_250005000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'beta_galactofuranosyl_transferase|LPG1', 'interval': Interval(799, 5363, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 799, 'end': 5363, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_25T0005000', 'parent_id': 'LINF_250005000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'beta_galactofuranosyl_transferase|LPG1', 'interval': Interval(799, 5363, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 799, 'end': 3926, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_25T0005000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(799, 3926, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c25.2940:\n",
      "\t{'chrom': 'LinJ.25', 'start': 373220, 'end': 374443, 'strand': '+', 'gene_id': 'LINF_250015760', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(373220, 374443, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 373220, 'end': 374443, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_25T0015760', 'parent_id': 'LINF_250015760', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(373220, 374443, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 374444, 'end': 376365, 'strand': '+', 'gene_id': 'LINF_250015800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'LYR_domain-containing_protein', 'interval': Interval(374444, 376365, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 374444, 'end': 376365, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_25T0015800', 'parent_id': 'LINF_250015800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'LYR_domain-containing_protein', 'interval': Interval(374444, 376365, closed='both')}\n",
      "\t{'chrom': 'LinJ.25', 'start': 374444, 'end': 374856, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_25T0015800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(374444, 374856, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3040:\n",
      "\t{'chrom': 'LinJ.26', 'start': 224623, 'end': 225662, 'strand': '-', 'gene_id': 'LINF_260012750', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(224623, 225662, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 224623, 'end': 225662, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_26T0012750', 'parent_id': 'LINF_260012750', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(224623, 225662, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 225663, 'end': 226736, 'strand': '-', 'gene_id': 'LINF_260012800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RNA-binding_protein|Y14', 'interval': Interval(225663, 226736, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 225663, 'end': 226736, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_26T0012800', 'parent_id': 'LINF_260012800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RNA-binding_protein|Y14', 'interval': Interval(225663, 226736, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 225663, 'end': 226143, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_26T0012800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(225663, 226143, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3080:\n",
      "\t{'chrom': 'LinJ.26', 'start': 598501, 'end': 599917, 'strand': '+', 'gene_id': 'LINF_260021850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(598501, 599917, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 598501, 'end': 599917, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_26T0021850', 'parent_id': 'LINF_260021850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(598501, 599917, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3110:\n",
      "\t{'chrom': 'LinJ.26', 'start': 662292, 'end': 664196, 'strand': '+', 'gene_id': 'LINF_260023450', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(662292, 664196, closed='both')}\n",
      "\t{'chrom': 'LinJ.26', 'start': 662292, 'end': 664196, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_26T0023450', 'parent_id': 'LINF_260023450', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(662292, 664196, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c26.3150:\n",
      "==================================================\n",
      "rejected_noCDS_BLAST_ERROR:\n",
      "==================================================\n",
      "rejected_noCDS_c27.3290AS:\n",
      "\t{'chrom': 'LinJ.27', 'start': 557803, 'end': 565292, 'strand': '-', 'gene_id': 'LINF_270018400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RING-H2_zinc_finger/Anaphase-promoting_complex_subunit_11_RING-H2_finger/Ring_finger_domain_containing_protein_-_putative', 'interval': Interval(557803, 565292, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 557803, 'end': 565292, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_27T0018400', 'parent_id': 'LINF_270018400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RING-H2_zinc_finger/Anaphase-promoting_complex_subunit_11_RING-H2_finger/Ring_finger_domain_containing_protein_-_putative', 'interval': Interval(557803, 565292, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 557803, 'end': 561174, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_27T0018400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(557803, 561174, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c27.3300AS:\n",
      "\t{'chrom': 'LinJ.27', 'start': 577188, 'end': 579826, 'strand': '-', 'gene_id': 'LINF_270018850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': 'unknown', 'notes': 'Carboxypeptidase', 'interval': Interval(577188, 579826, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 577188, 'end': 579826, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_27T0018850', 'parent_id': 'LINF_270018850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Carboxypeptidase', 'interval': Interval(577188, 579826, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 577719, 'end': 579429, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_27T0018850', 'parent_id': 'LINF_27T0018850', 'gene': False, 'transcript': False, 'CDS': True, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(577719, 579429, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 577188, 'end': 577718, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_27T0018850', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(577188, 577718, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c27.3310:\n",
      "\t{'chrom': 'LinJ.27', 'start': 701702, 'end': 703017, 'strand': '+', 'gene_id': 'LINF_270022350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(701702, 703017, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 701702, 'end': 703017, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_27T0022350', 'parent_id': 'LINF_270022350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(701702, 703017, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 703018, 'end': 708994, 'strand': '+', 'gene_id': 'LINF_270022400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'diacylglycerol_acyltransferase_-_putative', 'interval': Interval(703018, 708994, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 703018, 'end': 708994, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_27T0022400', 'parent_id': 'LINF_270022400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'diacylglycerol_acyltransferase_-_putative', 'interval': Interval(703018, 708994, closed='both')}\n",
      "\t{'chrom': 'LinJ.27', 'start': 703018, 'end': 703558, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_27T0022400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(703018, 703558, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c28.3370:\n",
      "\t{'chrom': 'LinJ.28', 'start': 57936, 'end': 60119, 'strand': '+', 'gene_id': 'LINF_280006850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(57936, 60119, closed='both')}\n",
      "\t{'chrom': 'LinJ.28', 'start': 57936, 'end': 60119, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_28T0006850', 'parent_id': 'LINF_280006850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(57936, 60119, closed='both')}\n",
      "\t{'chrom': 'LinJ.28', 'start': 60120, 'end': 63587, 'strand': '+', 'gene_id': 'LINF_280006900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(60120, 63587, closed='both')}\n",
      "\t{'chrom': 'LinJ.28', 'start': 60120, 'end': 63587, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_28T0006900', 'parent_id': 'LINF_280006900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(60120, 63587, closed='both')}\n",
      "\t{'chrom': 'LinJ.28', 'start': 60120, 'end': 60396, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_28T0006900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(60120, 60396, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c28.3400:\n",
      "\t{'chrom': 'LinJ.28', 'start': 592222, 'end': 595702, 'strand': '+', 'gene_id': 'LINF_280021850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(592222, 595702, closed='both')}\n",
      "\t{'chrom': 'LinJ.28', 'start': 592222, 'end': 595702, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_28T0021850', 'parent_id': 'LINF_280021850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(592222, 595702, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c28.3410:\n",
      "\t{'chrom': 'LinJ.28', 'start': 654749, 'end': 663755, 'strand': '-', 'gene_id': 'LINF_280023400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(654749, 663755, closed='both')}\n",
      "\t{'chrom': 'LinJ.28', 'start': 654749, 'end': 663755, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_28T0023400', 'parent_id': 'LINF_280023400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(654749, 663755, closed='both')}\n",
      "\t{'chrom': 'LinJ.28', 'start': 654749, 'end': 656104, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_28T0023400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(654749, 656104, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c28.3430Z:\n",
      "\t{'chrom': 'LinJ.28', 'start': 1202044, 'end': 1204457, 'strand': '-', 'gene_id': 'LINF_280039900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1202044, 1204457, closed='both')}\n",
      "\t{'chrom': 'LinJ.28', 'start': 1202044, 'end': 1204457, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_28T0039900', 'parent_id': 'LINF_280039900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1202044, 1204457, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3440:\n",
      "\t{'chrom': 'LinJ.29', 'start': 94100, 'end': 97001, 'strand': '-', 'gene_id': 'LINF_290007800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'D-lactate_dehydrogenase-like_protein', 'interval': Interval(94100, 97001, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 94100, 'end': 97001, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0007800', 'parent_id': 'LINF_290007800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'D-lactate_dehydrogenase-like_protein', 'interval': Interval(94100, 97001, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 94100, 'end': 95303, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0007800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(94100, 95303, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3450:\n",
      "\t{'chrom': 'LinJ.29', 'start': 154596, 'end': 160193, 'strand': '-', 'gene_id': 'LINF_290009600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(154596, 160193, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 154596, 'end': 160193, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0009600', 'parent_id': 'LINF_290009600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(154596, 160193, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 154596, 'end': 156174, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0009600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(154596, 156174, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3550AX:\n",
      "\t{'chrom': 'LinJ.29', 'start': 556193, 'end': 560127, 'strand': '+', 'gene_id': 'LINF_290019350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(556193, 560127, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 556193, 'end': 560127, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019350', 'parent_id': 'LINF_290019350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(556193, 560127, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 557537, 'end': 560127, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0019350', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(557537, 560127, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3570AZ:\n",
      "\t{'chrom': 'LinJ.29', 'start': 563252, 'end': 564329, 'strand': '+', 'gene_id': 'LINF_290019415', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(563252, 564329, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 563252, 'end': 564329, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019415', 'parent_id': 'LINF_290019415', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(563252, 564329, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 564330, 'end': 566488, 'strand': '+', 'gene_id': 'LINF_290019430', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(564330, 566488, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 564330, 'end': 566488, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019430', 'parent_id': 'LINF_290019430', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(564330, 566488, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3580AX:\n",
      "\t{'chrom': 'LinJ.29', 'start': 564330, 'end': 566488, 'strand': '+', 'gene_id': 'LINF_290019430', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(564330, 566488, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 564330, 'end': 566488, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019430', 'parent_id': 'LINF_290019430', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(564330, 566488, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3590:\n",
      "\t{'chrom': 'LinJ.29', 'start': 567113, 'end': 570359, 'strand': '+', 'gene_id': 'LINF_290019445', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(567113, 570359, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 567113, 'end': 570359, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019445', 'parent_id': 'LINF_290019445', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(567113, 570359, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3610AY:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3620AZ:\n",
      "\t{'chrom': 'LinJ.29', 'start': 570696, 'end': 572858, 'strand': '+', 'gene_id': 'LINF_290019475', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(570696, 572858, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 570696, 'end': 572858, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019475', 'parent_id': 'LINF_290019475', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(570696, 572858, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3630AX:\n",
      "\t{'chrom': 'LinJ.29', 'start': 570696, 'end': 572858, 'strand': '+', 'gene_id': 'LINF_290019475', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(570696, 572858, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 570696, 'end': 572858, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019475', 'parent_id': 'LINF_290019475', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(570696, 572858, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3640AY:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3650AZ:\n",
      "\t{'chrom': 'LinJ.29', 'start': 577099, 'end': 579261, 'strand': '+', 'gene_id': 'LINF_290019560', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(577099, 579261, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 577099, 'end': 579261, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019560', 'parent_id': 'LINF_290019560', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(577099, 579261, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3660AX:\n",
      "\t{'chrom': 'LinJ.29', 'start': 577099, 'end': 579261, 'strand': '+', 'gene_id': 'LINF_290019560', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(577099, 579261, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 577099, 'end': 579261, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019560', 'parent_id': 'LINF_290019560', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(577099, 579261, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3670AY:\n",
      "==================================================\n",
      "rejected_noCDS_c29.3690:\n",
      "\t{'chrom': 'LinJ.29', 'start': 583499, 'end': 587978, 'strand': '+', 'gene_id': 'LINF_290019700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(583499, 587978, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 583499, 'end': 587978, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019700', 'parent_id': 'LINF_290019700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(583499, 587978, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 584181, 'end': 587978, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0019700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(584181, 587978, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 587979, 'end': 596043, 'strand': '+', 'gene_id': 'LINF_290019800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RNA_binding_protein_-_putative', 'interval': Interval(587979, 596043, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 587979, 'end': 596043, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0019800', 'parent_id': 'LINF_290019800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RNA_binding_protein_-_putative', 'interval': Interval(587979, 596043, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 587979, 'end': 592487, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0019800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(587979, 592487, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3750:\n",
      "\t{'chrom': 'LinJ.29', 'start': 670544, 'end': 676210, 'strand': '-', 'gene_id': 'LINF_290020800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(670544, 676210, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 670544, 'end': 676210, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0020800', 'parent_id': 'LINF_290020800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein', 'interval': Interval(670544, 676210, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 674876, 'end': 676210, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0020800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(674876, 676210, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 676211, 'end': 685529, 'strand': '-', 'gene_id': 'LINF_290020900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Phosphatidylinositol_kinase|PIK', 'interval': Interval(676211, 685529, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 676211, 'end': 685529, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0020900', 'parent_id': 'LINF_290020900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Phosphatidylinositol_kinase|PIK', 'interval': Interval(676211, 685529, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 676211, 'end': 677181, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0020900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(676211, 677181, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3800:\n",
      "\t{'chrom': 'LinJ.29', 'start': 1065275, 'end': 1075994, 'strand': '-', 'gene_id': 'LINF_290030600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1065275, 1075994, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 1065275, 'end': 1075994, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0030600', 'parent_id': 'LINF_290030600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1065275, 1075994, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 1075237, 'end': 1075994, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0030600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(1075237, 1075994, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 1076374, 'end': 1079215, 'strand': '-', 'gene_id': 'LINF_290030650', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1076374, 1079215, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 1076374, 'end': 1079215, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_29T0030650', 'parent_id': 'LINF_290030650', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1076374, 1079215, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3810:\n",
      "\t{'chrom': 'LinJ.29', 'start': 1187676, 'end': 1189124, 'strand': '+', 'gene_id': 'LINF_290034000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1187676, 1189124, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 1187676, 'end': 1189124, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0034000', 'parent_id': 'LINF_290034000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1187676, 1189124, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 1188055, 'end': 1189124, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_29T0034000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1188055, 1189124, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 1189252, 'end': 1190272, 'strand': '+', 'gene_id': 'LINF_290034030', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1189252, 1190272, closed='both')}\n",
      "\t{'chrom': 'LinJ.29', 'start': 1189252, 'end': 1190272, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_29T0034030', 'parent_id': 'LINF_290034030', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1189252, 1190272, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c29.3830:\n",
      "==================================================\n",
      "rejected_noCDS_c30.3860BC:\n",
      "\t{'chrom': 'LinJ.30', 'start': 102141, 'end': 105323, 'strand': '-', 'gene_id': 'LINF_300008300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Rab-GTPase-TBC_domain_containing_protein', 'interval': Interval(102141, 105323, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 102141, 'end': 105323, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_30T0008300', 'parent_id': 'LINF_300008300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Rab-GTPase-TBC_domain_containing_protein', 'interval': Interval(102141, 105323, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 104949, 'end': 105323, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0008300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(104949, 105323, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 105722, 'end': 109074, 'strand': '-', 'gene_id': 'LINF_300008400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'KU80_protein', 'interval': Interval(105722, 109074, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 105722, 'end': 109074, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_30T0008400', 'parent_id': 'LINF_300008400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'KU80_protein', 'interval': Interval(105722, 109074, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 105722, 'end': 106621, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_30T0008400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(105722, 106621, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c30.3930:\n",
      "\t{'chrom': 'LinJ.30', 'start': 486069, 'end': 487146, 'strand': '+', 'gene_id': 'LINF_300019650', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(486069, 487146, closed='both')}\n",
      "\t{'chrom': 'LinJ.30', 'start': 486069, 'end': 487146, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_30T0019650', 'parent_id': 'LINF_300019650', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(486069, 487146, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4110:\n",
      "==================================================\n",
      "rejected_noCDS_c31.4120:\n",
      "\t{'chrom': 'LinJ.31', 'start': 148538, 'end': 149817, 'strand': '-', 'gene_id': 'LINF_310009750', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(148538, 149817, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 148538, 'end': 149817, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0009750', 'parent_id': 'LINF_310009750', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(148538, 149817, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 150123, 'end': 152831, 'strand': '-', 'gene_id': 'LINF_310009800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein', 'interval': Interval(150123, 152831, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 150123, 'end': 152831, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0009800', 'parent_id': 'LINF_310009800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin_surface_glycoprotein', 'interval': Interval(150123, 152831, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 150123, 'end': 152088, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0009800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(150123, 152088, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4140:\n",
      "\t{'chrom': 'LinJ.31', 'start': 153423, 'end': 156946, 'strand': '-', 'gene_id': 'LINF_310009850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': 'unknown', 'notes': 'tuzin_like_protein', 'interval': Interval(153423, 156946, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 153423, 'end': 156946, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0009850', 'parent_id': 'LINF_310009850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'tuzin_like_protein', 'interval': Interval(153423, 156946, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 153423, 'end': 154161, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0009850', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(153423, 154161, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4180:\n",
      "\t{'chrom': 'LinJ.31', 'start': 313930, 'end': 317135, 'strand': '-', 'gene_id': 'LINF_310014225', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(313930, 317135, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 313930, 'end': 317135, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0014225', 'parent_id': 'LINF_310014225', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(313930, 317135, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4220:\n",
      "==================================================\n",
      "rejected_noCDS_c31.4230BF:\n",
      "==================================================\n",
      "rejected_noCDS_c31.4240BG:\n",
      "==================================================\n",
      "rejected_noCDS_c31.4310:\n",
      "\t{'chrom': 'LinJ.31', 'start': 633369, 'end': 634601, 'strand': '-', 'gene_id': 'LINF_310020750', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(633369, 634601, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 633369, 'end': 634601, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0020750', 'parent_id': 'LINF_310020750', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(633369, 634601, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4400:\n",
      "\t{'chrom': 'LinJ.31', 'start': 908256, 'end': 910777, 'strand': '-', 'gene_id': 'LINF_310025650', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(908256, 910777, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 908256, 'end': 910777, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0025650', 'parent_id': 'LINF_310025650', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(908256, 910777, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 910778, 'end': 916576, 'strand': '-', 'gene_id': 'LINF_310025700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_kinase', 'interval': Interval(910778, 916576, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 910778, 'end': 916576, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0025700', 'parent_id': 'LINF_310025700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_kinase', 'interval': Interval(910778, 916576, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 910778, 'end': 912292, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_31T0025700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(910778, 912292, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4470:\n",
      "\t{'chrom': 'LinJ.31', 'start': 1038721, 'end': 1039235, 'strand': '-', 'gene_id': 'LINF_310028450', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1038721, 1039235, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 1038721, 'end': 1039235, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0028450', 'parent_id': 'LINF_310028450', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1038721, 1039235, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c31.4550:\n",
      "\t{'chrom': 'LinJ.31', 'start': 1325203, 'end': 1326083, 'strand': '-', 'gene_id': 'LINF_310034550', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1325203, 1326083, closed='both')}\n",
      "\t{'chrom': 'LinJ.31', 'start': 1325203, 'end': 1326083, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_31T0034550', 'parent_id': 'LINF_310034550', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1325203, 1326083, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4570O:\n",
      "\t{'chrom': 'LinJ.32', 'start': 968, 'end': 1399, 'strand': '+', 'gene_id': 'LINF_320004800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(968, 1399, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 968, 'end': 1399, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_32T0004800', 'parent_id': 'LINF_320004800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(968, 1399, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 1943, 'end': 2745, 'strand': '-', 'gene_id': 'LINF_320004900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1943, 2745, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 1943, 'end': 2745, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_32T0004900', 'parent_id': 'LINF_320004900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1943, 2745, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4690:\n",
      "\t{'chrom': 'LinJ.32', 'start': 746881, 'end': 751164, 'strand': '-', 'gene_id': 'LINF_320025100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RUS_family_member', 'interval': Interval(746881, 751164, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 746881, 'end': 751164, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_32T0025100', 'parent_id': 'LINF_320025100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'RUS_family_member', 'interval': Interval(746881, 751164, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 746881, 'end': 749098, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_32T0025100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(746881, 749098, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c32.4710BO:\n",
      "\t{'chrom': 'LinJ.32', 'start': 802344, 'end': 803500, 'strand': '-', 'gene_id': 'LINF_320026750', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(802344, 803500, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 802344, 'end': 803500, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_32T0026750', 'parent_id': 'LINF_320026750', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(802344, 803500, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 803501, 'end': 807153, 'strand': '-', 'gene_id': 'LINF_320026800', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Cytidylyltransferase', 'interval': Interval(803501, 807153, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 803501, 'end': 807153, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_32T0026800', 'parent_id': 'LINF_320026800', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Cytidylyltransferase', 'interval': Interval(803501, 807153, closed='both')}\n",
      "\t{'chrom': 'LinJ.32', 'start': 803501, 'end': 805717, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_32T0026800', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(803501, 805717, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4850:\n",
      "\t{'chrom': 'LinJ.33', 'start': 530154, 'end': 533949, 'strand': '+', 'gene_id': 'LINF_330020600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Major_Facilitator_Superfamily/Nucleoside_H+_symporter/LacY_proton/sugar_symporter_-_putative', 'interval': Interval(530154, 533949, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 530154, 'end': 533949, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0020600', 'parent_id': 'LINF_330020600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Major_Facilitator_Superfamily/Nucleoside_H+_symporter/LacY_proton/sugar_symporter_-_putative', 'interval': Interval(530154, 533949, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 530154, 'end': 530693, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_33T0020600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(530154, 530693, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4880:\n",
      "==================================================\n",
      "rejected_noCDS_c33.4890BC:\n",
      "\t{'chrom': 'LinJ.33', 'start': 1002245, 'end': 1005303, 'strand': '+', 'gene_id': 'LINF_330032400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Acetate:succinate_CoA_transferase', 'interval': Interval(1002245, 1005303, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1002245, 'end': 1005303, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0032400', 'parent_id': 'LINF_330032400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Acetate:succinate_CoA_transferase', 'interval': Interval(1002245, 1005303, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1004088, 'end': 1005303, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_33T0032400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1004088, 1005303, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1005687, 'end': 1009527, 'strand': '+', 'gene_id': 'LINF_330032500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Rab-GTPase-TBC_domain_containing_protein', 'interval': Interval(1005687, 1009527, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1005687, 'end': 1009527, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0032500', 'parent_id': 'LINF_330032500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Rab-GTPase-TBC_domain_containing_protein', 'interval': Interval(1005687, 1009527, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1005687, 'end': 1005824, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_33T0032500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(1005687, 1005824, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4910:\n",
      "==================================================\n",
      "rejected_noCDS_c33.4940BQ:\n",
      "\t{'chrom': 'LinJ.33', 'start': 1254377, 'end': 1257191, 'strand': '+', 'gene_id': 'LINF_330038400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1254377, 1257191, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1254377, 'end': 1257191, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0038400', 'parent_id': 'LINF_330038400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1254377, 1257191, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1255198, 'end': 1257191, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_33T0038400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1255198, 1257191, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.4960BQ:\n",
      "\t{'chrom': 'LinJ.33', 'start': 1272320, 'end': 1275156, 'strand': '+', 'gene_id': 'LINF_330038545', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1272320, 1275156, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1272320, 'end': 1275156, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0038545', 'parent_id': 'LINF_330038545', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1272320, 1275156, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c33.5030AK:\n",
      "==================================================\n",
      "rejected_noCDS_c33.5040AK:\n",
      "\t{'chrom': 'LinJ.33', 'start': 1531735, 'end': 1532106, 'strand': '+', 'gene_id': 'LINF_330042920', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1531735, 1532106, closed='both')}\n",
      "\t{'chrom': 'LinJ.33', 'start': 1531735, 'end': 1532106, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_33T0042920', 'parent_id': 'LINF_330042920', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1531735, 1532106, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5060:\n",
      "\t{'chrom': 'LinJ.34', 'start': 329434, 'end': 331016, 'strand': '-', 'gene_id': 'LINF_340012450', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(329434, 331016, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 329434, 'end': 331016, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0012450', 'parent_id': 'LINF_340012450', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(329434, 331016, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 331111, 'end': 331968, 'strand': '+', 'gene_id': 'LINF_340012500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(331111, 331968, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 331111, 'end': 331968, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0012500', 'parent_id': 'LINF_340012500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(331111, 331968, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5140BS:\n",
      "\t{'chrom': 'LinJ.34', 'start': 511354, 'end': 513171, 'strand': '-', 'gene_id': 'LINF_340017475', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(511354, 513171, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 511354, 'end': 513171, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0017475', 'parent_id': 'LINF_340017475', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(511354, 513171, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 513172, 'end': 516089, 'strand': '-', 'gene_id': 'LINF_340017500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_protein', 'interval': Interval(513172, 516089, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 513172, 'end': 516089, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0017500', 'parent_id': 'LINF_340017500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_protein', 'interval': Interval(513172, 516089, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 513172, 'end': 515349, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0017500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(513172, 515349, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5180BS:\n",
      "\t{'chrom': 'LinJ.34', 'start': 735203, 'end': 739818, 'strand': '-', 'gene_id': 'LINF_340022850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(735203, 739818, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 735203, 'end': 739818, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0022850', 'parent_id': 'LINF_340022850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(735203, 739818, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 739819, 'end': 743551, 'strand': '-', 'gene_id': 'LINF_340022900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_protein', 'interval': Interval(739819, 743551, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 739819, 'end': 743551, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0022900', 'parent_id': 'LINF_340022900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_protein', 'interval': Interval(739819, 743551, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 739819, 'end': 742807, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0022900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(739819, 742807, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5230BU:\n",
      "\t{'chrom': 'LinJ.34', 'start': 754280, 'end': 757294, 'strand': '-', 'gene_id': 'LINF_340023220', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(754280, 757294, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 754280, 'end': 757294, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023220', 'parent_id': 'LINF_340023220', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(754280, 757294, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 758277, 'end': 761292, 'strand': '-', 'gene_id': 'LINF_340023260', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(758277, 761292, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 758277, 'end': 761292, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023260', 'parent_id': 'LINF_340023260', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(758277, 761292, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5260BU:\n",
      "\t{'chrom': 'LinJ.34', 'start': 758277, 'end': 761292, 'strand': '-', 'gene_id': 'LINF_340023260', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(758277, 761292, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 758277, 'end': 761292, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023260', 'parent_id': 'LINF_340023260', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(758277, 761292, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 761293, 'end': 765197, 'strand': '-', 'gene_id': 'LINF_340023280', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(761293, 765197, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 761293, 'end': 765197, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023280', 'parent_id': 'LINF_340023280', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(761293, 765197, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5290BU:\n",
      "\t{'chrom': 'LinJ.34', 'start': 761293, 'end': 765197, 'strand': '-', 'gene_id': 'LINF_340023280', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(761293, 765197, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 761293, 'end': 765197, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023280', 'parent_id': 'LINF_340023280', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(761293, 765197, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 766090, 'end': 768423, 'strand': '-', 'gene_id': 'LINF_340023300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_surface_protein_-_putative', 'interval': Interval(766090, 768423, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 766090, 'end': 768423, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023300', 'parent_id': 'LINF_340023300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_surface_protein_-_putative', 'interval': Interval(766090, 768423, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 766090, 'end': 767657, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0023300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(766090, 767657, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5310BU:\n",
      "\t{'chrom': 'LinJ.34', 'start': 772123, 'end': 775066, 'strand': '-', 'gene_id': 'LINF_340023415', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(772123, 775066, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 772123, 'end': 775066, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023415', 'parent_id': 'LINF_340023415', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(772123, 775066, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 775900, 'end': 777200, 'strand': '-', 'gene_id': 'LINF_340023430', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(775900, 777200, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 775900, 'end': 777200, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023430', 'parent_id': 'LINF_340023430', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(775900, 777200, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5320BT:\n",
      "\t{'chrom': 'LinJ.34', 'start': 775900, 'end': 777200, 'strand': '-', 'gene_id': 'LINF_340023430', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(775900, 777200, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 775900, 'end': 777200, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023430', 'parent_id': 'LINF_340023430', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(775900, 777200, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 777201, 'end': 778297, 'strand': '-', 'gene_id': 'LINF_340023445', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(777201, 778297, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 777201, 'end': 778297, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023445', 'parent_id': 'LINF_340023445', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(777201, 778297, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5330BU:\n",
      "\t{'chrom': 'LinJ.34', 'start': 777201, 'end': 778297, 'strand': '-', 'gene_id': 'LINF_340023445', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(777201, 778297, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 777201, 'end': 778297, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023445', 'parent_id': 'LINF_340023445', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(777201, 778297, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5340BU:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5350BU:\n",
      "\t{'chrom': 'LinJ.34', 'start': 778924, 'end': 780014, 'strand': '-', 'gene_id': 'LINF_340023460', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(778924, 780014, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 778924, 'end': 780014, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023460', 'parent_id': 'LINF_340023460', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(778924, 780014, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5360BT:\n",
      "\t{'chrom': 'LinJ.34', 'start': 778924, 'end': 780014, 'strand': '-', 'gene_id': 'LINF_340023460', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(778924, 780014, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 778924, 'end': 780014, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023460', 'parent_id': 'LINF_340023460', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(778924, 780014, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 780144, 'end': 781532, 'strand': '-', 'gene_id': 'LINF_340023475', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(780144, 781532, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 780144, 'end': 781532, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023475', 'parent_id': 'LINF_340023475', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(780144, 781532, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5370BU:\n",
      "\t{'chrom': 'LinJ.34', 'start': 780144, 'end': 781532, 'strand': '-', 'gene_id': 'LINF_340023475', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(780144, 781532, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 780144, 'end': 781532, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023475', 'parent_id': 'LINF_340023475', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(780144, 781532, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 781533, 'end': 782303, 'strand': '-', 'gene_id': 'LINF_340023490', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(781533, 782303, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 781533, 'end': 782303, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023490', 'parent_id': 'LINF_340023490', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(781533, 782303, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 782521, 'end': 784862, 'strand': '-', 'gene_id': 'LINF_340023500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_surface_protein', 'interval': Interval(782521, 784862, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 782521, 'end': 784862, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0023500', 'parent_id': 'LINF_340023500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_surface_protein', 'interval': Interval(782521, 784862, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 782521, 'end': 784096, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0023500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(782521, 784096, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5390BU:\n",
      "==================================================\n",
      "rejected_noCDS_c34.5400BT:\n",
      "\t{'chrom': 'LinJ.34', 'start': 814218, 'end': 816590, 'strand': '-', 'gene_id': 'LINF_340024500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_surface_protein_-_putative', 'interval': Interval(814218, 816590, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 814218, 'end': 816590, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_34T0024500', 'parent_id': 'LINF_340024500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'amastin-like_surface_protein_-_putative', 'interval': Interval(814218, 816590, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 814218, 'end': 815826, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0024500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(814218, 815826, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5920CC:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1372364, 'end': 1376363, 'strand': '+', 'gene_id': 'LINF_340037600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Tripartite_attachment_complex_40|TAC40', 'interval': Interval(1372364, 1376363, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1372364, 'end': 1376363, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0037600', 'parent_id': 'LINF_340037600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Tripartite_attachment_complex_40|TAC40', 'interval': Interval(1372364, 1376363, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1374067, 'end': 1376363, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0037600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1374067, 1376363, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5930CC:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1381967, 'end': 1383514, 'strand': '+', 'gene_id': 'LINF_340037850', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1381967, 1383514, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1381967, 'end': 1383514, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0037850', 'parent_id': 'LINF_340037850', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1381967, 1383514, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5960CC:\n",
      "\t{'chrom': 'LinJ.34', 'start': 1391972, 'end': 1395962, 'strand': '+', 'gene_id': 'LINF_340038100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Tripartite_attachment_complex_40|TAC40', 'interval': Interval(1391972, 1395962, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1391972, 'end': 1395962, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_34T0038100', 'parent_id': 'LINF_340038100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Tripartite_attachment_complex_40|TAC40', 'interval': Interval(1391972, 1395962, closed='both')}\n",
      "\t{'chrom': 'LinJ.34', 'start': 1393676, 'end': 1395962, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_34T0038100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1393676, 1395962, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c34.5970CC:\n",
      "==================================================\n",
      "rejected_noCDS_c35.6080:\n",
      "\t{'chrom': 'LinJ.35', 'start': 296633, 'end': 298023, 'strand': '+', 'gene_id': 'LINF_350012950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(296633, 298023, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 296633, 'end': 298023, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0012950', 'parent_id': 'LINF_350012950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(296633, 298023, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6100:\n",
      "\t{'chrom': 'LinJ.35', 'start': 332062, 'end': 334222, 'strand': '+', 'gene_id': 'LINF_350013950', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(332062, 334222, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 332062, 'end': 334222, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0013950', 'parent_id': 'LINF_350013950', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(332062, 334222, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 334360, 'end': 336208, 'strand': '+', 'gene_id': 'LINF_350014000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(334360, 336208, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 334360, 'end': 336208, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0014000', 'parent_id': 'LINF_350014000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein_of_unknown_function_-_conserved', 'interval': Interval(334360, 336208, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 334360, 'end': 334792, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0014000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(334360, 334792, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6180CJ:\n",
      "\t{'chrom': 'LinJ.35', 'start': 1125055, 'end': 1126995, 'strand': '-', 'gene_id': 'LINF_350034650', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1125055, 1126995, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1125055, 'end': 1126995, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_35T0034650', 'parent_id': 'LINF_350034650', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1125055, 1126995, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6220CL:\n",
      "==================================================\n",
      "rejected_noCDS_c35.6230:\n",
      "==================================================\n",
      "rejected_noCDS_c35.6250CJ:\n",
      "\t{'chrom': 'LinJ.35', 'start': 1366385, 'end': 1367810, 'strand': '-', 'gene_id': 'LINF_350041050', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1366385, 1367810, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1366385, 'end': 1367810, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_35T0041050', 'parent_id': 'LINF_350041050', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1366385, 1367810, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6270:\n",
      "\t{'chrom': 'LinJ.35', 'start': 1550851, 'end': 1552427, 'strand': '+', 'gene_id': 'LINF_350046600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Splicing_factor_3B_subunit_10_(SF3b10)_-_putative', 'interval': Interval(1550851, 1552427, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1550851, 'end': 1552427, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0046600', 'parent_id': 'LINF_350046600', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'Splicing_factor_3B_subunit_10_(SF3b10)_-_putative', 'interval': Interval(1550851, 1552427, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1551239, 'end': 1552427, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0046600', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(1551239, 1552427, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1552428, 'end': 1554684, 'strand': '+', 'gene_id': 'LINF_350046700', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1552428, 1554684, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1552428, 'end': 1554684, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_35T0046700', 'parent_id': 'LINF_350046700', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(1552428, 1554684, closed='both')}\n",
      "\t{'chrom': 'LinJ.35', 'start': 1552428, 'end': 1552651, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_35T0046700', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(1552428, 1552651, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c35.6310:\n",
      "==================================================\n",
      "rejected_noCDS_c36.6370CO:\n",
      "\t{'chrom': 'LinJ.36', 'start': 493588, 'end': 494799, 'strand': '+', 'gene_id': 'LINF_360019350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(493588, 494799, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 493588, 'end': 494799, 'strand': '+', 'gene_id': None, 'transcript_id': 'LINF_36T0019350', 'parent_id': 'LINF_360019350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(493588, 494799, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 494822, 'end': 494893, 'strand': '+', 'gene_id': 'LINF_360019500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'tRNA-Thr', 'interval': Interval(494822, 494893, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 494822, 'end': 494893, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_360019500', 'gene': False, 'transcript': False, 'CDS': True, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(494822, 494893, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 494972, 'end': 495370, 'strand': '-', 'gene_id': 'LINF_360019550', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(494972, 495370, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 494972, 'end': 495370, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0019550', 'parent_id': 'LINF_360019550', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(494972, 495370, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 494975, 'end': 495059, 'strand': '+', 'gene_id': 'LINF_360019600', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'tRNA-Leu', 'interval': Interval(494975, 495059, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 494975, 'end': 495059, 'strand': '+', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_360019600', 'gene': False, 'transcript': False, 'CDS': True, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(494975, 495059, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6490:\n",
      "==================================================\n",
      "rejected_noCDS_c36.6640:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2322277, 'end': 2323858, 'strand': '-', 'gene_id': 'LINF_360069350', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2322277, 2323858, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2322277, 'end': 2323858, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0069350', 'parent_id': 'LINF_360069350', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2322277, 2323858, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2323859, 'end': 2326194, 'strand': '-', 'gene_id': 'LINF_360069400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(2323859, 2326194, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2323859, 'end': 2326194, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0069400', 'parent_id': 'LINF_360069400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(2323859, 2326194, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2323859, 'end': 2324793, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0069400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2323859, 2324793, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6650CS:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2456799, 'end': 2460093, 'strand': '-', 'gene_id': 'LINF_360072900', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2456799, 2460093, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2456799, 'end': 2460093, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0072900', 'parent_id': 'LINF_360072900', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2456799, 2460093, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2456799, 'end': 2458220, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0072900', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2456799, 2458220, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6660CS:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2460371, 'end': 2463651, 'strand': '-', 'gene_id': 'LINF_360073000', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2460371, 2463651, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2460371, 'end': 2463651, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0073000', 'parent_id': 'LINF_360073000', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2460371, 2463651, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2460371, 'end': 2461778, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0073000', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2460371, 2461778, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6670CS:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2463918, 'end': 2467210, 'strand': '-', 'gene_id': 'LINF_360073100', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2463918, 2467210, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2463918, 'end': 2467210, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0073100', 'parent_id': 'LINF_360073100', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2463918, 2467210, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2463918, 'end': 2465337, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0073100', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2463918, 2465337, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6680CS:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2467477, 'end': 2470767, 'strand': '-', 'gene_id': 'LINF_360073200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2467477, 2470767, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2467477, 'end': 2470767, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0073200', 'parent_id': 'LINF_360073200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2467477, 2470767, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2467477, 'end': 2468894, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0073200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2467477, 2468894, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6690CS:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2471045, 'end': 2474323, 'strand': '-', 'gene_id': 'LINF_360073300', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2471045, 2474323, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2471045, 'end': 2474323, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0073300', 'parent_id': 'LINF_360073300', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2471045, 2474323, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2471045, 'end': 2472450, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0073300', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2471045, 2472450, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6700CS:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2474601, 'end': 2477880, 'strand': '-', 'gene_id': 'LINF_360073400', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2474601, 2477880, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2474601, 'end': 2477880, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0073400', 'parent_id': 'LINF_360073400', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2474601, 2477880, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2474601, 'end': 2476007, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0073400', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2474601, 2476007, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6710CS:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2478158, 'end': 2481440, 'strand': '-', 'gene_id': 'LINF_360073500', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2478158, 2481440, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2478158, 'end': 2481440, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0073500', 'parent_id': 'LINF_360073500', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'glucose_transporter', 'interval': Interval(2478158, 2481440, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2478158, 'end': 2479567, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0073500', 'gene': False, 'transcript': False, 'CDS': False, '3utr': True, '5utr': False, 'pseudogen': None, 'notes': None, 'interval': Interval(2478158, 2479567, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6740CT:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2597511, 'end': 2600258, 'strand': '-', 'gene_id': 'LINF_360077200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein-l-isoaspartate_o-methyltransferase_-_putative', 'interval': Interval(2597511, 2600258, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2597511, 'end': 2600258, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0077200', 'parent_id': 'LINF_360077200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'protein-l-isoaspartate_o-methyltransferase_-_putative', 'interval': Interval(2597511, 2600258, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2599967, 'end': 2600258, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0077200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(2599967, 2600258, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6770CT:\n",
      "\t{'chrom': 'LinJ.36', 'start': 2655844, 'end': 2658753, 'strand': '-', 'gene_id': 'LINF_360079200', 'transcript_id': None, 'parent_id': None, 'gene': True, 'transcript': False, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(2655844, 2658753, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2655844, 'end': 2658753, 'strand': '-', 'gene_id': None, 'transcript_id': 'LINF_36T0079200', 'parent_id': 'LINF_360079200', 'gene': False, 'transcript': True, 'CDS': False, '3utr': False, '5utr': False, 'pseudogen': None, 'notes': 'hypothetical_protein_-_conserved', 'interval': Interval(2655844, 2658753, closed='both')}\n",
      "\t{'chrom': 'LinJ.36', 'start': 2657303, 'end': 2658753, 'strand': '-', 'gene_id': None, 'transcript_id': None, 'parent_id': 'LINF_36T0079200', 'gene': False, 'transcript': False, 'CDS': False, '3utr': False, '5utr': True, 'pseudogen': None, 'notes': None, 'interval': Interval(2657303, 2658753, closed='both')}\n",
      "==================================================\n",
      "rejected_noCDS_c36.6790CU:\n"
     ]
    }
   ],
   "execution_count": 116
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:08.521150Z",
     "start_time": "2024-10-03T08:02:08.516095Z"
    }
   },
   "source": [
    "print(f\"From the total of {len(neg_not_inside_gtf_dict_list)} NEGATIVE ELEMENTS, {counter_neg_overlaps} are overlapping the GTF elements and {counter_neg_not_overlaps} are not overlapping the GTF elements\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "From the total of 178 NEGATIVE ELEMENTS, 137 are overlapping the GTF elements and 41 are not overlapping the GTF elements\n"
     ]
    }
   ],
   "execution_count": 117
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's join the two \"neg_gtf_dict\" dictionaries to save them as a JSON file"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:08.840496Z",
     "start_time": "2024-10-03T08:02:08.834466Z"
    }
   },
   "source": [
    "dict_neg_full_inside = neg_gtf_dict.copy()\n",
    "dict_neg_overlap = neg_gtf_dict_2.copy()\n",
    "\n",
    "# Let's join them\n",
    "for key in dict_neg_overlap.keys():\n",
    "    if key in dict_neg_full_inside.keys():\n",
    "        if len(dict_neg_overlap[key]) > 0:\n",
    "            dict_neg_full_inside[key].extend(dict_neg_overlap[key])\n"
   ],
   "outputs": [],
   "execution_count": 118
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:09.011108Z",
     "start_time": "2024-10-03T08:02:08.949260Z"
    }
   },
   "source": [
    "# Save the data to a json file\n",
    "path_gtf_neg_relation_json = \"./data/neg_gtf_relation.json\" # Path to save the json file\n",
    "\n",
    "# The problem will be the pandas Interval type and the JSON package. We need to create a custom serializer\n",
    "def custom_serializer(obj):\n",
    "    if isinstance(obj, pd.Interval):\n",
    "        return {\n",
    "            'left': int(obj.left) if isinstance(obj.left, np.integer) else obj.left,  # JSOn package can't process int64; transforming it\n",
    "            'right': int(obj.right) if isinstance(obj.right, np.integer) else obj.right,  # JSOn package can't process int64; transforming it\n",
    "            'closed': obj.closed\n",
    "        }\n",
    "    elif isinstance(obj, np.integer):  # Check for numpy integer types\n",
    "        return int(obj)  # Convert to a standard Python int\n",
    "    raise TypeError(f\"Object of type {type(obj).__name__} is not JSON serializable\")\n",
    "\n",
    "\n",
    "# Save the data to a json file\n",
    "with open(path_gtf_neg_relation_json, \"w\") as f:\n",
    "    json.dump(dict_neg_full_inside, f, default=custom_serializer)"
   ],
   "outputs": [],
   "execution_count": 119
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:09.086138Z",
     "start_time": "2024-10-03T08:02:09.079607Z"
    }
   },
   "source": [
    "# Let's get the elements in different dictionaries\n",
    "neg_overlaps_gtf_dict = {key: value for key, value in neg_gtf_dict_2.items() if len(value) > 0}\n",
    "\n",
    "# And now for the INTERGENIC elements\n",
    "neg_intergenic_gtf_dict = {key: value for key, value in neg_gtf_dict_2.items() if len(value) == 0}"
   ],
   "outputs": [],
   "execution_count": 120
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:09.183739Z",
     "start_time": "2024-10-03T08:02:09.176093Z"
    }
   },
   "source": [
    "print(len(neg_intergenic_gtf_dict))\n",
    "list(neg_intergenic_gtf_dict.keys())"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "41\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['rejected_noCDS_c01.50',\n",
       " 'rejected_noCDS_c01.60A',\n",
       " 'rejected_noCDS_c02.170',\n",
       " 'rejected_noCDS_c04.200C',\n",
       " 'rejected_noCDS_c05.360F',\n",
       " 'rejected_noCDS_c06.400',\n",
       " 'rejected_noCDS_c08.850',\n",
       " 'rejected_noCDS_c08.860',\n",
       " 'rejected_noCDS_c10.980O',\n",
       " 'rejected_noCDS_c12.1120',\n",
       " 'rejected_noCDS_c12.1170',\n",
       " 'rejected_noCDS_c14.1420',\n",
       " 'rejected_noCDS_c14.1530',\n",
       " 'rejected_noCDS_c16.1720',\n",
       " 'rejected_noCDS_c17.1910',\n",
       " 'rejected_noCDS_c18.1980',\n",
       " 'rejected_noCDS_c19.2160',\n",
       " 'rejected_noCDS_c19.2230',\n",
       " 'rejected_noCDS_c23.2640AI',\n",
       " 'rejected_noCDS_c23.2650AI',\n",
       " 'rejected_noCDS_c26.3150',\n",
       " 'rejected_noCDS_BLAST_ERROR',\n",
       " 'rejected_noCDS_c29.3610AY',\n",
       " 'rejected_noCDS_c29.3640AY',\n",
       " 'rejected_noCDS_c29.3670AY',\n",
       " 'rejected_noCDS_c29.3830',\n",
       " 'rejected_noCDS_c31.4110',\n",
       " 'rejected_noCDS_c31.4220',\n",
       " 'rejected_noCDS_c31.4230BF',\n",
       " 'rejected_noCDS_c31.4240BG',\n",
       " 'rejected_noCDS_c33.4880',\n",
       " 'rejected_noCDS_c33.4910',\n",
       " 'rejected_noCDS_c33.5030AK',\n",
       " 'rejected_noCDS_c34.5340BU',\n",
       " 'rejected_noCDS_c34.5390BU',\n",
       " 'rejected_noCDS_c34.5970CC',\n",
       " 'rejected_noCDS_c35.6220CL',\n",
       " 'rejected_noCDS_c35.6230',\n",
       " 'rejected_noCDS_c35.6310',\n",
       " 'rejected_noCDS_c36.6490',\n",
       " 'rejected_noCDS_c36.6790CU']"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 121
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## NEG ELEMENTS OVERLAPPING"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:09.648408Z",
     "start_time": "2024-10-03T08:02:09.496126Z"
    }
   },
   "source": [
    "# Check the Dict\n",
    "print(len(neg_overlaps_gtf_dict))\n",
    "neg_overlaps_gtf_dict"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "137\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'rejected_noCDS_c01.20': [{'chrom': 'LinJ.01',\n",
       "   'start': 114146,\n",
       "   'end': 116224,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_010009600',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'nicotinamidase|PNC1',\n",
       "   'interval': Interval(114146, 116224, closed='both')},\n",
       "  {'chrom': 'LinJ.01',\n",
       "   'start': 114146,\n",
       "   'end': 116224,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_01T0009600',\n",
       "   'parent_id': 'LINF_010009600',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'nicotinamidase|PNC1',\n",
       "   'interval': Interval(114146, 116224, closed='both')},\n",
       "  {'chrom': 'LinJ.01',\n",
       "   'start': 114146,\n",
       "   'end': 114423,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_01T0009600',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(114146, 114423, closed='both')}],\n",
       " 'rejected_noCDS_c01.40': [{'chrom': 'LinJ.01',\n",
       "   'start': 262069,\n",
       "   'end': 262979,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_010013350',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(262069, 262979, closed='both')},\n",
       "  {'chrom': 'LinJ.01',\n",
       "   'start': 262069,\n",
       "   'end': 262979,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_01T0013350',\n",
       "   'parent_id': 'LINF_010013350',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(262069, 262979, closed='both')}],\n",
       " 'rejected_noCDS_c02.140': [{'chrom': 'LinJ.02',\n",
       "   'start': 137296,\n",
       "   'end': 139716,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_020007950',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(137296, 139716, closed='both')},\n",
       "  {'chrom': 'LinJ.02',\n",
       "   'start': 137296,\n",
       "   'end': 139716,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_02T0007950',\n",
       "   'parent_id': 'LINF_020007950',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(137296, 139716, closed='both')},\n",
       "  {'chrom': 'LinJ.02',\n",
       "   'start': 139717,\n",
       "   'end': 150152,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_020008000',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'ATP-binding_cassette_subfamily_A_-_member_1|ABCA1',\n",
       "   'interval': Interval(139717, 150152, closed='both')},\n",
       "  {'chrom': 'LinJ.02',\n",
       "   'start': 139717,\n",
       "   'end': 150152,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_02T0008000',\n",
       "   'parent_id': 'LINF_020008000',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'ATP-binding_cassette_subfamily_A_-_member_1|ABCA1',\n",
       "   'interval': Interval(139717, 150152, closed='both')},\n",
       "  {'chrom': 'LinJ.02',\n",
       "   'start': 139717,\n",
       "   'end': 145477,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_02T0008000',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(139717, 145477, closed='both')}],\n",
       " 'rejected_noCDS_c02.150': [{'chrom': 'LinJ.02',\n",
       "   'start': 139717,\n",
       "   'end': 150152,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_020008000',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'ATP-binding_cassette_subfamily_A_-_member_1|ABCA1',\n",
       "   'interval': Interval(139717, 150152, closed='both')},\n",
       "  {'chrom': 'LinJ.02',\n",
       "   'start': 139717,\n",
       "   'end': 150152,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_02T0008000',\n",
       "   'parent_id': 'LINF_020008000',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'ATP-binding_cassette_subfamily_A_-_member_1|ABCA1',\n",
       "   'interval': Interval(139717, 150152, closed='both')},\n",
       "  {'chrom': 'LinJ.02',\n",
       "   'start': 149222,\n",
       "   'end': 150152,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_02T0008000',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(149222, 150152, closed='both')},\n",
       "  {'chrom': 'LinJ.02',\n",
       "   'start': 150375,\n",
       "   'end': 155425,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_020008100',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'paraflagellar_rod_component_11|PFC11',\n",
       "   'interval': Interval(150375, 155425, closed='both')},\n",
       "  {'chrom': 'LinJ.02',\n",
       "   'start': 150375,\n",
       "   'end': 155425,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_02T0008100',\n",
       "   'parent_id': 'LINF_020008100',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'paraflagellar_rod_component_11|PFC11',\n",
       "   'interval': Interval(150375, 155425, closed='both')},\n",
       "  {'chrom': 'LinJ.02',\n",
       "   'start': 150375,\n",
       "   'end': 153057,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_02T0008100',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(150375, 153057, closed='both')}],\n",
       " 'rejected_noCDS_c03.180': [{'chrom': 'LinJ.03',\n",
       "   'start': 168578,\n",
       "   'end': 173520,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_030009600',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Ankyrin_repeats_(3_copies)/Ankyrin_repeats_(many_copies)/Ankyrin_repeat/DHHC_palmitoyltransferase_-_putative',\n",
       "   'interval': Interval(168578, 173520, closed='both')},\n",
       "  {'chrom': 'LinJ.03',\n",
       "   'start': 168578,\n",
       "   'end': 173520,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_03T0009600',\n",
       "   'parent_id': 'LINF_030009600',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Ankyrin_repeats_(3_copies)/Ankyrin_repeats_(many_copies)/Ankyrin_repeat/DHHC_palmitoyltransferase_-_putative',\n",
       "   'interval': Interval(168578, 173520, closed='both')},\n",
       "  {'chrom': 'LinJ.03',\n",
       "   'start': 172508,\n",
       "   'end': 173520,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_03T0009600',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(172508, 173520, closed='both')}],\n",
       " 'rejected_noCDS_c04.210C': [{'chrom': 'LinJ.04',\n",
       "   'start': 126105,\n",
       "   'end': 126374,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_040008875',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(126105, 126374, closed='both')},\n",
       "  {'chrom': 'LinJ.04',\n",
       "   'start': 126105,\n",
       "   'end': 126374,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_04T0008875',\n",
       "   'parent_id': 'LINF_040008875',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(126105, 126374, closed='both')}],\n",
       " 'rejected_noCDS_c04.230': [{'chrom': 'LinJ.04',\n",
       "   'start': 139396,\n",
       "   'end': 142427,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_040009200',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Tetratricopeptide_repeat-containing_protein',\n",
       "   'interval': Interval(139396, 142427, closed='both')},\n",
       "  {'chrom': 'LinJ.04',\n",
       "   'start': 139396,\n",
       "   'end': 142427,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_04T0009200',\n",
       "   'parent_id': 'LINF_040009200',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Tetratricopeptide_repeat-containing_protein',\n",
       "   'interval': Interval(139396, 142427, closed='both')},\n",
       "  {'chrom': 'LinJ.04',\n",
       "   'start': 142101,\n",
       "   'end': 142427,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_04T0009200',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(142101, 142427, closed='both')},\n",
       "  {'chrom': 'LinJ.04',\n",
       "   'start': 142650,\n",
       "   'end': 156872,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_040009300',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(142650, 156872, closed='both')},\n",
       "  {'chrom': 'LinJ.04',\n",
       "   'start': 142650,\n",
       "   'end': 156872,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_04T0009300',\n",
       "   'parent_id': 'LINF_040009300',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(142650, 156872, closed='both')},\n",
       "  {'chrom': 'LinJ.04',\n",
       "   'start': 142650,\n",
       "   'end': 143425,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_04T0009300',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(142650, 143425, closed='both')}],\n",
       " 'rejected_noCDS_c05.280': [{'chrom': 'LinJ.05',\n",
       "   'start': 1093,\n",
       "   'end': 2315,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_050004700',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1093, 2315, closed='both')},\n",
       "  {'chrom': 'LinJ.05',\n",
       "   'start': 1093,\n",
       "   'end': 2315,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_05T0004700',\n",
       "   'parent_id': 'LINF_050004700',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1093, 2315, closed='both')}],\n",
       " 'rejected_noCDS_c05.300D': [{'chrom': 'LinJ.05',\n",
       "   'start': 66289,\n",
       "   'end': 73912,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_050007100',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'dual_specificity_phosphatase-like_protein',\n",
       "   'interval': Interval(66289, 73912, closed='both')},\n",
       "  {'chrom': 'LinJ.05',\n",
       "   'start': 66289,\n",
       "   'end': 73912,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_05T0007100',\n",
       "   'parent_id': 'LINF_050007100',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'dual_specificity_phosphatase-like_protein',\n",
       "   'interval': Interval(66289, 73912, closed='both')},\n",
       "  {'chrom': 'LinJ.05',\n",
       "   'start': 66289,\n",
       "   'end': 67362,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_05T0007100',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(66289, 67362, closed='both')}],\n",
       " 'rejected_noCDS_c05.320D': [{'chrom': 'LinJ.05',\n",
       "   'start': 104200,\n",
       "   'end': 106507,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_050008400',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'mitoribosomal_protein_uS11m',\n",
       "   'interval': Interval(104200, 106507, closed='both')},\n",
       "  {'chrom': 'LinJ.05',\n",
       "   'start': 104200,\n",
       "   'end': 106507,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_05T0008400',\n",
       "   'parent_id': 'LINF_050008400',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'mitoribosomal_protein_uS11m',\n",
       "   'interval': Interval(104200, 106507, closed='both')},\n",
       "  {'chrom': 'LinJ.05',\n",
       "   'start': 105430,\n",
       "   'end': 106507,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_05T0008400',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(105430, 106507, closed='both')},\n",
       "  {'chrom': 'LinJ.05',\n",
       "   'start': 106508,\n",
       "   'end': 108915,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_050008500',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'trypanothione_reductase_(TRYR)',\n",
       "   'interval': Interval(106508, 108915, closed='both')},\n",
       "  {'chrom': 'LinJ.05',\n",
       "   'start': 106508,\n",
       "   'end': 108915,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_05T0008500',\n",
       "   'parent_id': 'LINF_050008500',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'trypanothione_reductase_(TRYR)',\n",
       "   'interval': Interval(106508, 108915, closed='both')},\n",
       "  {'chrom': 'LinJ.05',\n",
       "   'start': 106508,\n",
       "   'end': 106885,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_05T0008500',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(106508, 106885, closed='both')}],\n",
       " 'rejected_noCDS_c05.350': [{'chrom': 'LinJ.05',\n",
       "   'start': 357004,\n",
       "   'end': 360467,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_050014700',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'glutaminyl_cyclase',\n",
       "   'interval': Interval(357004, 360467, closed='both')},\n",
       "  {'chrom': 'LinJ.05',\n",
       "   'start': 357004,\n",
       "   'end': 360467,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_05T0014700',\n",
       "   'parent_id': 'LINF_050014700',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'glutaminyl_cyclase',\n",
       "   'interval': Interval(357004, 360467, closed='both')},\n",
       "  {'chrom': 'LinJ.05',\n",
       "   'start': 359831,\n",
       "   'end': 360467,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_05T0014700',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(359831, 360467, closed='both')},\n",
       "  {'chrom': 'LinJ.05',\n",
       "   'start': 360956,\n",
       "   'end': 363428,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_050014800',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'dipeptidyl-peptidase_III|metallo-peptidase_-_Clan_M_-_Family_M49',\n",
       "   'interval': Interval(360956, 363428, closed='both')},\n",
       "  {'chrom': 'LinJ.05',\n",
       "   'start': 360956,\n",
       "   'end': 363428,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_05T0014800',\n",
       "   'parent_id': 'LINF_050014800',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'dipeptidyl-peptidase_III|metallo-peptidase_-_Clan_M_-_Family_M49',\n",
       "   'interval': Interval(360956, 363428, closed='both')},\n",
       "  {'chrom': 'LinJ.05',\n",
       "   'start': 360956,\n",
       "   'end': 361110,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_05T0014800',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(360956, 361110, closed='both')}],\n",
       " 'rejected_noCDS_c06.380': [{'chrom': 'LinJ.06',\n",
       "   'start': 59844,\n",
       "   'end': 62540,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_060006900',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'kinesin-like_protein',\n",
       "   'interval': Interval(59844, 62540, closed='both')},\n",
       "  {'chrom': 'LinJ.06',\n",
       "   'start': 59844,\n",
       "   'end': 62540,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_06T0006900',\n",
       "   'parent_id': 'LINF_060006900',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'kinesin-like_protein',\n",
       "   'interval': Interval(59844, 62540, closed='both')},\n",
       "  {'chrom': 'LinJ.06',\n",
       "   'start': 61911,\n",
       "   'end': 62540,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_06T0006900',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(61911, 62540, closed='both')},\n",
       "  {'chrom': 'LinJ.06',\n",
       "   'start': 62983,\n",
       "   'end': 65883,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_060007000',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_of_unknown_function_-_conserved',\n",
       "   'interval': Interval(62983, 65883, closed='both')},\n",
       "  {'chrom': 'LinJ.06',\n",
       "   'start': 62983,\n",
       "   'end': 65883,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_06T0007000',\n",
       "   'parent_id': 'LINF_060007000',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_of_unknown_function_-_conserved',\n",
       "   'interval': Interval(62983, 65883, closed='both')},\n",
       "  {'chrom': 'LinJ.06',\n",
       "   'start': 62983,\n",
       "   'end': 63490,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_06T0007000',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(62983, 63490, closed='both')}],\n",
       " 'rejected_noCDS_c07.430G': [{'chrom': 'LinJ.07',\n",
       "   'start': 211168,\n",
       "   'end': 211744,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_070010140',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(211168, 211744, closed='both')},\n",
       "  {'chrom': 'LinJ.07',\n",
       "   'start': 211168,\n",
       "   'end': 211744,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_07T0010140',\n",
       "   'parent_id': 'LINF_070010140',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(211168, 211744, closed='both')},\n",
       "  {'chrom': 'LinJ.07',\n",
       "   'start': 211951,\n",
       "   'end': 214227,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_070010160',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(211951, 214227, closed='both')},\n",
       "  {'chrom': 'LinJ.07',\n",
       "   'start': 211951,\n",
       "   'end': 214227,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_07T0010160',\n",
       "   'parent_id': 'LINF_070010160',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(211951, 214227, closed='both')}],\n",
       " 'rejected_noCDS_c07.510': [{'chrom': 'LinJ.07',\n",
       "   'start': 384350,\n",
       "   'end': 384937,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_070013950',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(384350, 384937, closed='both')},\n",
       "  {'chrom': 'LinJ.07',\n",
       "   'start': 384350,\n",
       "   'end': 384937,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_07T0013950',\n",
       "   'parent_id': 'LINF_070013950',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(384350, 384937, closed='both')}],\n",
       " 'rejected_noCDS_c08.550': [{'chrom': 'LinJ.08',\n",
       "   'start': 190115,\n",
       "   'end': 192664,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_080009800',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'small_ubiquitin-like_modifier_(SUMO)',\n",
       "   'interval': Interval(190115, 192664, closed='both')},\n",
       "  {'chrom': 'LinJ.08',\n",
       "   'start': 190115,\n",
       "   'end': 192664,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_08T0009800',\n",
       "   'parent_id': 'LINF_080009800',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'small_ubiquitin-like_modifier_(SUMO)',\n",
       "   'interval': Interval(190115, 192664, closed='both')},\n",
       "  {'chrom': 'LinJ.08',\n",
       "   'start': 190647,\n",
       "   'end': 192664,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_08T0009800',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(190647, 192664, closed='both')}],\n",
       " 'rejected_noCDS_c08.560I': [{'chrom': 'LinJ.08',\n",
       "   'start': 298159,\n",
       "   'end': 299531,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_080012050',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(298159, 299531, closed='both')},\n",
       "  {'chrom': 'LinJ.08',\n",
       "   'start': 298159,\n",
       "   'end': 299531,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_08T0012050',\n",
       "   'parent_id': 'LINF_080012050',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(298159, 299531, closed='both')}],\n",
       " 'rejected_noCDS_c08.590I': [{'chrom': 'LinJ.08',\n",
       "   'start': 302787,\n",
       "   'end': 304302,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_080012150',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(302787, 304302, closed='both')},\n",
       "  {'chrom': 'LinJ.08',\n",
       "   'start': 302787,\n",
       "   'end': 304302,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_08T0012150',\n",
       "   'parent_id': 'LINF_080012150',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(302787, 304302, closed='both')}],\n",
       " 'rejected_noCDS_c08.620I': [{'chrom': 'LinJ.08',\n",
       "   'start': 307412,\n",
       "   'end': 308886,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_080012250',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(307412, 308886, closed='both')},\n",
       "  {'chrom': 'LinJ.08',\n",
       "   'start': 307412,\n",
       "   'end': 308886,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_08T0012250',\n",
       "   'parent_id': 'LINF_080012250',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(307412, 308886, closed='both')}],\n",
       " 'rejected_noCDS_c08.640I': [{'chrom': 'LinJ.08',\n",
       "   'start': 311996,\n",
       "   'end': 313821,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_080012350',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(311996, 313821, closed='both')},\n",
       "  {'chrom': 'LinJ.08',\n",
       "   'start': 311996,\n",
       "   'end': 313821,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_08T0012350',\n",
       "   'parent_id': 'LINF_080012350',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(311996, 313821, closed='both')}],\n",
       " 'rejected_noCDS_c08.670': [{'chrom': 'LinJ.08',\n",
       "   'start': 316292,\n",
       "   'end': 318363,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_080012450',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(316292, 318363, closed='both')},\n",
       "  {'chrom': 'LinJ.08',\n",
       "   'start': 316292,\n",
       "   'end': 318363,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_08T0012450',\n",
       "   'parent_id': 'LINF_080012450',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(316292, 318363, closed='both')}],\n",
       " 'rejected_noCDS_c08.680K': [{'chrom': 'LinJ.08',\n",
       "   'start': 324605,\n",
       "   'end': 326842,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_080012700',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'tuzin',\n",
       "   'interval': Interval(324605, 326842, closed='both')},\n",
       "  {'chrom': 'LinJ.08',\n",
       "   'start': 324605,\n",
       "   'end': 326842,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_08T0012700',\n",
       "   'parent_id': 'LINF_080012700',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'tuzin',\n",
       "   'interval': Interval(324605, 326842, closed='both')},\n",
       "  {'chrom': 'LinJ.08',\n",
       "   'start': 326394,\n",
       "   'end': 326842,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_08T0012700',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(326394, 326842, closed='both')}],\n",
       " 'rejected_noCDS_c08.710K': [{'chrom': 'LinJ.08',\n",
       "   'start': 330313,\n",
       "   'end': 332015,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_080012825',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(330313, 332015, closed='both')},\n",
       "  {'chrom': 'LinJ.08',\n",
       "   'start': 330313,\n",
       "   'end': 332015,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_08T0012825',\n",
       "   'parent_id': 'LINF_080012825',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(330313, 332015, closed='both')}],\n",
       " 'rejected_noCDS_c08.720N': [{'chrom': 'LinJ.08',\n",
       "   'start': 332481,\n",
       "   'end': 335118,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_080012850',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(332481, 335118, closed='both')},\n",
       "  {'chrom': 'LinJ.08',\n",
       "   'start': 332481,\n",
       "   'end': 335118,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_08T0012850',\n",
       "   'parent_id': 'LINF_080012850',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(332481, 335118, closed='both')}],\n",
       " 'rejected_noCDS_c08.760K': [{'chrom': 'LinJ.08',\n",
       "   'start': 335486,\n",
       "   'end': 337341,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_080012875',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(335486, 337341, closed='both')},\n",
       "  {'chrom': 'LinJ.08',\n",
       "   'start': 335486,\n",
       "   'end': 337341,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_08T0012875',\n",
       "   'parent_id': 'LINF_080012875',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(335486, 337341, closed='both')}],\n",
       " 'rejected_noCDS_c08.770N': [{'chrom': 'LinJ.08',\n",
       "   'start': 337654,\n",
       "   'end': 340312,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_080012900',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'amastin-like_protein',\n",
       "   'interval': Interval(337654, 340312, closed='both')},\n",
       "  {'chrom': 'LinJ.08',\n",
       "   'start': 337654,\n",
       "   'end': 340312,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_08T0012900',\n",
       "   'parent_id': 'LINF_080012900',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'amastin-like_protein',\n",
       "   'interval': Interval(337654, 340312, closed='both')},\n",
       "  {'chrom': 'LinJ.08',\n",
       "   'start': 337654,\n",
       "   'end': 337858,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_08T0012900',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(337654, 337858, closed='both')}],\n",
       " 'rejected_noCDS_c08.800K': [{'chrom': 'LinJ.08',\n",
       "   'start': 340660,\n",
       "   'end': 342358,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_080012925',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(340660, 342358, closed='both')},\n",
       "  {'chrom': 'LinJ.08',\n",
       "   'start': 340660,\n",
       "   'end': 342358,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_08T0012925',\n",
       "   'parent_id': 'LINF_080012925',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(340660, 342358, closed='both')}],\n",
       " 'rejected_noCDS_c10.1010': [{'chrom': 'LinJ.10',\n",
       "   'start': 71683,\n",
       "   'end': 73187,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_100006550',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(71683, 73187, closed='both')},\n",
       "  {'chrom': 'LinJ.10',\n",
       "   'start': 71683,\n",
       "   'end': 73187,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_10T0006550',\n",
       "   'parent_id': 'LINF_100006550',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(71683, 73187, closed='both')},\n",
       "  {'chrom': 'LinJ.10',\n",
       "   'start': 73188,\n",
       "   'end': 76489,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_100006600',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'guanosine_diphosphatase|NTPD2',\n",
       "   'interval': Interval(73188, 76489, closed='both')},\n",
       "  {'chrom': 'LinJ.10',\n",
       "   'start': 73188,\n",
       "   'end': 76489,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_10T0006600',\n",
       "   'parent_id': 'LINF_100006600',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'guanosine_diphosphatase|NTPD2',\n",
       "   'interval': Interval(73188, 76489, closed='both')},\n",
       "  {'chrom': 'LinJ.10',\n",
       "   'start': 73188,\n",
       "   'end': 73389,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_10T0006600',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(73188, 73389, closed='both')}],\n",
       " 'rejected_noCDS_c11.1070': [{'chrom': 'LinJ.11',\n",
       "   'start': 161231,\n",
       "   'end': 163190,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_110009720',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(161231, 163190, closed='both')},\n",
       "  {'chrom': 'LinJ.11',\n",
       "   'start': 161231,\n",
       "   'end': 163190,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_11T0009720',\n",
       "   'parent_id': 'LINF_110009720',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(161231, 163190, closed='both')}],\n",
       " 'rejected_noCDS_c11.1080': [{'chrom': 'LinJ.11',\n",
       "   'start': 190410,\n",
       "   'end': 194177,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_110011200',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'purine_nucleobase_transporter|NT4',\n",
       "   'interval': Interval(190410, 194177, closed='both')},\n",
       "  {'chrom': 'LinJ.11',\n",
       "   'start': 190410,\n",
       "   'end': 194177,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_11T0011200',\n",
       "   'parent_id': 'LINF_110011200',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'purine_nucleobase_transporter|NT4',\n",
       "   'interval': Interval(190410, 194177, closed='both')},\n",
       "  {'chrom': 'LinJ.11',\n",
       "   'start': 192469,\n",
       "   'end': 194177,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_11T0011200',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(192469, 194177, closed='both')},\n",
       "  {'chrom': 'LinJ.11',\n",
       "   'start': 194178,\n",
       "   'end': 196953,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_110011250',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(194178, 196953, closed='both')},\n",
       "  {'chrom': 'LinJ.11',\n",
       "   'start': 194178,\n",
       "   'end': 196953,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_11T0011250',\n",
       "   'parent_id': 'LINF_110011250',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(194178, 196953, closed='both')}],\n",
       " 'rejected_noCDS_c12.1180S': [{'chrom': 'LinJ.12',\n",
       "   'start': 372226,\n",
       "   'end': 374050,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_120012350',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Protein_of_unknown_function_(DUF962),_putative',\n",
       "   'interval': Interval(372226, 374050, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 372226,\n",
       "   'end': 374050,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_12T0012350',\n",
       "   'parent_id': 'LINF_120012350',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Protein_of_unknown_function_(DUF962),_putative',\n",
       "   'interval': Interval(372226, 374050, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 372831,\n",
       "   'end': 374050,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_12T0012350',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(372831, 374050, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 374051,\n",
       "   'end': 377775,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_120012400',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_of_unknown_function_-_conserved',\n",
       "   'interval': Interval(374051, 377775, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 374051,\n",
       "   'end': 377775,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_12T0012400',\n",
       "   'parent_id': 'LINF_120012400',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_of_unknown_function_-_conserved',\n",
       "   'interval': Interval(374051, 377775, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 374051,\n",
       "   'end': 375027,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_12T0012400',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(374051, 375027, closed='both')}],\n",
       " 'rejected_noCDS_c12.1190': [{'chrom': 'LinJ.12',\n",
       "   'start': 374051,\n",
       "   'end': 377775,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_120012400',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_of_unknown_function_-_conserved',\n",
       "   'interval': Interval(374051, 377775, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 374051,\n",
       "   'end': 377775,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_12T0012400',\n",
       "   'parent_id': 'LINF_120012400',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_of_unknown_function_-_conserved',\n",
       "   'interval': Interval(374051, 377775, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 375514,\n",
       "   'end': 377775,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_12T0012400',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(375514, 377775, closed='both')}],\n",
       " 'rejected_noCDS_c12.1200S': [{'chrom': 'LinJ.12',\n",
       "   'start': 389565,\n",
       "   'end': 393477,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_120012800',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Ribonuclease_inhibitor',\n",
       "   'interval': Interval(389565, 393477, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 389565,\n",
       "   'end': 393477,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_12T0012800',\n",
       "   'parent_id': 'LINF_120012800',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Ribonuclease_inhibitor',\n",
       "   'interval': Interval(389565, 393477, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 390949,\n",
       "   'end': 393477,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_12T0012800',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(390949, 393477, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 393788,\n",
       "   'end': 397393,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_120012900',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(393788, 397393, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 393788,\n",
       "   'end': 397393,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_12T0012900',\n",
       "   'parent_id': 'LINF_120012900',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(393788, 397393, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 393788,\n",
       "   'end': 394522,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_12T0012900',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(393788, 394522, closed='both')}],\n",
       " 'rejected_noCDS_c12.1210S': [{'chrom': 'LinJ.12',\n",
       "   'start': 401779,\n",
       "   'end': 406034,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_120013100',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'promastigote_surface_antigen_38S',\n",
       "   'interval': Interval(401779, 406034, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 401779,\n",
       "   'end': 406034,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_12T0013100',\n",
       "   'parent_id': 'LINF_120013100',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'promastigote_surface_antigen_38S',\n",
       "   'interval': Interval(401779, 406034, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 403504,\n",
       "   'end': 406034,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_12T0013100',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(403504, 406034, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 406344,\n",
       "   'end': 409449,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_120013300',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(406344, 409449, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 406344,\n",
       "   'end': 409449,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_12T0013300',\n",
       "   'parent_id': 'LINF_120013300',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(406344, 409449, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 406344,\n",
       "   'end': 407078,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_12T0013300',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(406344, 407078, closed='both')}],\n",
       " 'rejected_noCDS_c12.1220S': [{'chrom': 'LinJ.12',\n",
       "   'start': 413336,\n",
       "   'end': 417374,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_120013500',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Surface_antigen_protein_2',\n",
       "   'interval': Interval(413336, 417374, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 413336,\n",
       "   'end': 417374,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_12T0013500',\n",
       "   'parent_id': 'LINF_120013500',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Surface_antigen_protein_2',\n",
       "   'interval': Interval(413336, 417374, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 415192,\n",
       "   'end': 417374,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_12T0013500',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(415192, 417374, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 417691,\n",
       "   'end': 421308,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_120013560',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(417691, 421308, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 417691,\n",
       "   'end': 421308,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_12T0013560',\n",
       "   'parent_id': 'LINF_120013560',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(417691, 421308, closed='both')}],\n",
       " 'rejected_noCDS_c12.1240S': [{'chrom': 'LinJ.12',\n",
       "   'start': 425243,\n",
       "   'end': 429221,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_120013700',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'surface_antigen_protein_2_-_putative',\n",
       "   'interval': Interval(425243, 429221, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 425243,\n",
       "   'end': 429221,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_12T0013700',\n",
       "   'parent_id': 'LINF_120013700',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'surface_antigen_protein_2_-_putative',\n",
       "   'interval': Interval(425243, 429221, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 426862,\n",
       "   'end': 429221,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_12T0013700',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(426862, 429221, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 429222,\n",
       "   'end': 431627,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_120013800',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_of_unknown_function_-_conserved',\n",
       "   'interval': Interval(429222, 431627, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 429222,\n",
       "   'end': 431627,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_12T0013800',\n",
       "   'parent_id': 'LINF_120013800',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_of_unknown_function_-_conserved',\n",
       "   'interval': Interval(429222, 431627, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 429222,\n",
       "   'end': 429627,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_12T0013800',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(429222, 429627, closed='both')}],\n",
       " 'rejected_noCDS_c12.1250S': [{'chrom': 'LinJ.12',\n",
       "   'start': 461643,\n",
       "   'end': 466202,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_120015000',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Surface_antigenic_protein|PSA-2',\n",
       "   'interval': Interval(461643, 466202, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 461643,\n",
       "   'end': 466202,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_12T0015000',\n",
       "   'parent_id': 'LINF_120015000',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Surface_antigenic_protein|PSA-2',\n",
       "   'interval': Interval(461643, 466202, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 463757,\n",
       "   'end': 466202,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_12T0015000',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(463757, 466202, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 466522,\n",
       "   'end': 469621,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_120015050',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(466522, 469621, closed='both')},\n",
       "  {'chrom': 'LinJ.12',\n",
       "   'start': 466522,\n",
       "   'end': 469621,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_12T0015050',\n",
       "   'parent_id': 'LINF_120015050',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(466522, 469621, closed='both')}],\n",
       " 'rejected_noCDS_c13.1400': [{'chrom': 'LinJ.13',\n",
       "   'start': 413449,\n",
       "   'end': 414617,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_130016950',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(413449, 414617, closed='both')},\n",
       "  {'chrom': 'LinJ.13',\n",
       "   'start': 413449,\n",
       "   'end': 414617,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_13T0016950',\n",
       "   'parent_id': 'LINF_130016950',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(413449, 414617, closed='both')}],\n",
       " 'rejected_noCDS_c14.1540W': [{'chrom': 'LinJ.14',\n",
       "   'start': 655102,\n",
       "   'end': 656077,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_140021450',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(655102, 656077, closed='both')},\n",
       "  {'chrom': 'LinJ.14',\n",
       "   'start': 655102,\n",
       "   'end': 656077,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_14T0021450',\n",
       "   'parent_id': 'LINF_140021450',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(655102, 656077, closed='both')}],\n",
       " 'rejected_noCDS_c15.1580': [{'chrom': 'LinJ.15',\n",
       "   'start': 151001,\n",
       "   'end': 152466,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_150009600',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_of_unknown_function_-_conserved',\n",
       "   'interval': Interval(151001, 152466, closed='both')},\n",
       "  {'chrom': 'LinJ.15',\n",
       "   'start': 151001,\n",
       "   'end': 152466,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_15T0009600',\n",
       "   'parent_id': 'LINF_150009600',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_of_unknown_function_-_conserved',\n",
       "   'interval': Interval(151001, 152466, closed='both')},\n",
       "  {'chrom': 'LinJ.15',\n",
       "   'start': 151500,\n",
       "   'end': 152466,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_15T0009600',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(151500, 152466, closed='both')},\n",
       "  {'chrom': 'LinJ.15',\n",
       "   'start': 152817,\n",
       "   'end': 154532,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_150009700',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_of_unknown_function_-_conserved',\n",
       "   'interval': Interval(152817, 154532, closed='both')},\n",
       "  {'chrom': 'LinJ.15',\n",
       "   'start': 152817,\n",
       "   'end': 154532,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_15T0009700',\n",
       "   'parent_id': 'LINF_150009700',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_of_unknown_function_-_conserved',\n",
       "   'interval': Interval(152817, 154532, closed='both')},\n",
       "  {'chrom': 'LinJ.15',\n",
       "   'start': 152817,\n",
       "   'end': 153094,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_15T0009700',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(152817, 153094, closed='both')}],\n",
       " 'rejected_noCDS_c15.1590X': [{'chrom': 'LinJ.15',\n",
       "   'start': 286973,\n",
       "   'end': 288730,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_150013750',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(286973, 288730, closed='both')},\n",
       "  {'chrom': 'LinJ.15',\n",
       "   'start': 286973,\n",
       "   'end': 288730,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_15T0013750',\n",
       "   'parent_id': 'LINF_150013750',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(286973, 288730, closed='both')}],\n",
       " 'rejected_noCDS_c15.1610X': [{'chrom': 'LinJ.15',\n",
       "   'start': 306215,\n",
       "   'end': 309145,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_150014000',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(306215, 309145, closed='both')},\n",
       "  {'chrom': 'LinJ.15',\n",
       "   'start': 306215,\n",
       "   'end': 309145,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_15T0014000',\n",
       "   'parent_id': 'LINF_150014000',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(306215, 309145, closed='both')},\n",
       "  {'chrom': 'LinJ.15',\n",
       "   'start': 308258,\n",
       "   'end': 309145,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_15T0014000',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(308258, 309145, closed='both')},\n",
       "  {'chrom': 'LinJ.15',\n",
       "   'start': 309377,\n",
       "   'end': 310768,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_150014050',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(309377, 310768, closed='both')},\n",
       "  {'chrom': 'LinJ.15',\n",
       "   'start': 309377,\n",
       "   'end': 310768,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_15T0014050',\n",
       "   'parent_id': 'LINF_150014050',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(309377, 310768, closed='both')}],\n",
       " 'rejected_noCDS_c17.1830Z': [{'chrom': 'LinJ.17',\n",
       "   'start': 931,\n",
       "   'end': 2725,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_170004900',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(931, 2725, closed='both')},\n",
       "  {'chrom': 'LinJ.17',\n",
       "   'start': 931,\n",
       "   'end': 2725,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_17T0004900',\n",
       "   'parent_id': 'LINF_170004900',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(931, 2725, closed='both')}],\n",
       " 'rejected_noCDS_c17.1900': [{'chrom': 'LinJ.17',\n",
       "   'start': 454894,\n",
       "   'end': 459103,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_170016500',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Phosphorylated_CTD_interacting_factor_1_WW_domain_containing_protein_-_putative',\n",
       "   'interval': Interval(454894, 459103, closed='both')},\n",
       "  {'chrom': 'LinJ.17',\n",
       "   'start': 454894,\n",
       "   'end': 459103,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_17T0016500',\n",
       "   'parent_id': 'LINF_170016500',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Phosphorylated_CTD_interacting_factor_1_WW_domain_containing_protein_-_putative',\n",
       "   'interval': Interval(454894, 459103, closed='both')},\n",
       "  {'chrom': 'LinJ.17',\n",
       "   'start': 458150,\n",
       "   'end': 459103,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_17T0016500',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(458150, 459103, closed='both')}],\n",
       " 'rejected_noCDS_c17.1940AA': [{'chrom': 'LinJ.17',\n",
       "   'start': 562538,\n",
       "   'end': 568030,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_170018700',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'WW_domain_containing_protein_-_putative',\n",
       "   'interval': Interval(562538, 568030, closed='both')},\n",
       "  {'chrom': 'LinJ.17',\n",
       "   'start': 562538,\n",
       "   'end': 568030,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_17T0018700',\n",
       "   'parent_id': 'LINF_170018700',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'WW_domain_containing_protein_-_putative',\n",
       "   'interval': Interval(562538, 568030, closed='both')},\n",
       "  {'chrom': 'LinJ.17',\n",
       "   'start': 562538,\n",
       "   'end': 562587,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_17T0018700',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(562538, 562587, closed='both')}],\n",
       " 'rejected_noCDS_c17.1970Z': [{'chrom': 'LinJ.17',\n",
       "   'start': 687597,\n",
       "   'end': 689967,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_170022200',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(687597, 689967, closed='both')},\n",
       "  {'chrom': 'LinJ.17',\n",
       "   'start': 687597,\n",
       "   'end': 689967,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_17T0022200',\n",
       "   'parent_id': 'LINF_170022200',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(687597, 689967, closed='both')},\n",
       "  {'chrom': 'LinJ.17',\n",
       "   'start': 688415,\n",
       "   'end': 689967,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_17T0022200',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(688415, 689967, closed='both')},\n",
       "  {'chrom': 'LinJ.17',\n",
       "   'start': 690110,\n",
       "   'end': 690871,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_170022300',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(690110, 690871, closed='both')},\n",
       "  {'chrom': 'LinJ.17',\n",
       "   'start': 690110,\n",
       "   'end': 690871,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_17T0022300',\n",
       "   'parent_id': 'LINF_170022300',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(690110, 690871, closed='both')}],\n",
       " 'rejected_noCDS_c18.1990': [{'chrom': 'LinJ.18',\n",
       "   'start': 162150,\n",
       "   'end': 167192,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_180009000',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'UDP-galactose_transporter|LPG5B',\n",
       "   'interval': Interval(162150, 167192, closed='both')},\n",
       "  {'chrom': 'LinJ.18',\n",
       "   'start': 162150,\n",
       "   'end': 167192,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_18T0009000',\n",
       "   'parent_id': 'LINF_180009000',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'UDP-galactose_transporter|LPG5B',\n",
       "   'interval': Interval(162150, 167192, closed='both')},\n",
       "  {'chrom': 'LinJ.18',\n",
       "   'start': 165272,\n",
       "   'end': 167192,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_18T0009000',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(165272, 167192, closed='both')},\n",
       "  {'chrom': 'LinJ.18',\n",
       "   'start': 167193,\n",
       "   'end': 169438,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_180009100',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(167193, 169438, closed='both')},\n",
       "  {'chrom': 'LinJ.18',\n",
       "   'start': 167193,\n",
       "   'end': 169438,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_18T0009100',\n",
       "   'parent_id': 'LINF_180009100',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(167193, 169438, closed='both')},\n",
       "  {'chrom': 'LinJ.18',\n",
       "   'start': 167193,\n",
       "   'end': 168037,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_18T0009100',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(167193, 168037, closed='both')}],\n",
       " 'rejected_noCDS_c18.2000': [{'chrom': 'LinJ.18',\n",
       "   'start': 238666,\n",
       "   'end': 239602,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_180010950',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(238666, 239602, closed='both')},\n",
       "  {'chrom': 'LinJ.18',\n",
       "   'start': 238666,\n",
       "   'end': 239602,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_18T0010950',\n",
       "   'parent_id': 'LINF_180010950',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(238666, 239602, closed='both')}],\n",
       " 'rejected_noCDS_c19.2110': [{'chrom': 'LinJ.19',\n",
       "   'start': 354993,\n",
       "   'end': 355737,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_190013715',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'ATG8/AUT7/APG8/PAZ2',\n",
       "   'interval': Interval(354993, 355737, closed='both')},\n",
       "  {'chrom': 'LinJ.19',\n",
       "   'start': 354993,\n",
       "   'end': 355737,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_19T0013715',\n",
       "   'parent_id': 'LINF_190013715',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'ATG8/AUT7/APG8/PAZ2',\n",
       "   'interval': Interval(354993, 355737, closed='both')},\n",
       "  {'chrom': 'LinJ.19',\n",
       "   'start': 355034,\n",
       "   'end': 355399,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_19T0013715',\n",
       "   'parent_id': 'LINF_19T0013715',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': True,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(355034, 355399, closed='both')},\n",
       "  {'chrom': 'LinJ.19',\n",
       "   'start': 354993,\n",
       "   'end': 355033,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_19T0013715',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(354993, 355033, closed='both')},\n",
       "  {'chrom': 'LinJ.19',\n",
       "   'start': 355400,\n",
       "   'end': 355737,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_19T0013715',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(355400, 355737, closed='both')}],\n",
       " 'rejected_noCDS_c19.2120AD': [{'chrom': 'LinJ.19',\n",
       "   'start': 361081,\n",
       "   'end': 362554,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_190013900',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'ATG8/AUT7/APG8/PAZ2',\n",
       "   'interval': Interval(361081, 362554, closed='both')},\n",
       "  {'chrom': 'LinJ.19',\n",
       "   'start': 361081,\n",
       "   'end': 362554,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_19T0013900',\n",
       "   'parent_id': 'LINF_190013900',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'ATG8/AUT7/APG8/PAZ2',\n",
       "   'interval': Interval(361081, 362554, closed='both')},\n",
       "  {'chrom': 'LinJ.19',\n",
       "   'start': 361488,\n",
       "   'end': 362554,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_19T0013900',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(361488, 362554, closed='both')}],\n",
       " 'rejected_noCDS_c19.2130AD': [{'chrom': 'LinJ.19',\n",
       "   'start': 364663,\n",
       "   'end': 366138,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_190014100',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'ATG8/AUT7/APG8/PAZ2',\n",
       "   'interval': Interval(364663, 366138, closed='both')},\n",
       "  {'chrom': 'LinJ.19',\n",
       "   'start': 364663,\n",
       "   'end': 366138,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_19T0014100',\n",
       "   'parent_id': 'LINF_190014100',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'ATG8/AUT7/APG8/PAZ2',\n",
       "   'interval': Interval(364663, 366138, closed='both')},\n",
       "  {'chrom': 'LinJ.19',\n",
       "   'start': 365070,\n",
       "   'end': 366138,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_19T0014100',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(365070, 366138, closed='both')}],\n",
       " 'rejected_noCDS_c20.2250': [{'chrom': 'LinJ.20',\n",
       "   'start': 134609,\n",
       "   'end': 135924,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_200008850',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(134609, 135924, closed='both')},\n",
       "  {'chrom': 'LinJ.20',\n",
       "   'start': 134609,\n",
       "   'end': 135924,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_20T0008850',\n",
       "   'parent_id': 'LINF_200008850',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(134609, 135924, closed='both')},\n",
       "  {'chrom': 'LinJ.20',\n",
       "   'start': 135925,\n",
       "   'end': 140156,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_200008900',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Protein_of_unknown_function_(DUF2946)_-_putative',\n",
       "   'interval': Interval(135925, 140156, closed='both')},\n",
       "  {'chrom': 'LinJ.20',\n",
       "   'start': 135925,\n",
       "   'end': 140156,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_20T0008900',\n",
       "   'parent_id': 'LINF_200008900',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Protein_of_unknown_function_(DUF2946)_-_putative',\n",
       "   'interval': Interval(135925, 140156, closed='both')},\n",
       "  {'chrom': 'LinJ.20',\n",
       "   'start': 135925,\n",
       "   'end': 136313,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_20T0008900',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(135925, 136313, closed='both')}],\n",
       " 'rejected_noCDS_c20.2260': [{'chrom': 'LinJ.20',\n",
       "   'start': 245715,\n",
       "   'end': 247334,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_200012000',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_of_unknown_function_-_conserved',\n",
       "   'interval': Interval(245715, 247334, closed='both')},\n",
       "  {'chrom': 'LinJ.20',\n",
       "   'start': 245715,\n",
       "   'end': 247334,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_20T0012000',\n",
       "   'parent_id': 'LINF_200012000',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_of_unknown_function_-_conserved',\n",
       "   'interval': Interval(245715, 247334, closed='both')},\n",
       "  {'chrom': 'LinJ.20',\n",
       "   'start': 246773,\n",
       "   'end': 247334,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_20T0012000',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(246773, 247334, closed='both')},\n",
       "  {'chrom': 'LinJ.20',\n",
       "   'start': 247335,\n",
       "   'end': 249516,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_200012100',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Protein_of_unknown_function_(DUF775)_-_putative',\n",
       "   'interval': Interval(247335, 249516, closed='both')},\n",
       "  {'chrom': 'LinJ.20',\n",
       "   'start': 247335,\n",
       "   'end': 249516,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_20T0012100',\n",
       "   'parent_id': 'LINF_200012100',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Protein_of_unknown_function_(DUF775)_-_putative',\n",
       "   'interval': Interval(247335, 249516, closed='both')},\n",
       "  {'chrom': 'LinJ.20',\n",
       "   'start': 247335,\n",
       "   'end': 248221,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_20T0012100',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(247335, 248221, closed='both')}],\n",
       " 'rejected_noCDS_c20.2310': [{'chrom': 'LinJ.20',\n",
       "   'start': 432463,\n",
       "   'end': 433723,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_200015450',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(432463, 433723, closed='both')},\n",
       "  {'chrom': 'LinJ.20',\n",
       "   'start': 432463,\n",
       "   'end': 433723,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_20T0015450',\n",
       "   'parent_id': 'LINF_200015450',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(432463, 433723, closed='both')},\n",
       "  {'chrom': 'LinJ.20',\n",
       "   'start': 433724,\n",
       "   'end': 439654,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_200015500',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(433724, 439654, closed='both')},\n",
       "  {'chrom': 'LinJ.20',\n",
       "   'start': 433724,\n",
       "   'end': 439654,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_20T0015500',\n",
       "   'parent_id': 'LINF_200015500',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(433724, 439654, closed='both')},\n",
       "  {'chrom': 'LinJ.20',\n",
       "   'start': 433724,\n",
       "   'end': 434033,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_20T0015500',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(433724, 434033, closed='both')}],\n",
       " 'rejected_noCDS_c21.2390': [{'chrom': 'LinJ.21',\n",
       "   'start': 488393,\n",
       "   'end': 489994,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_210019900',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Protein_of_unknown_function_-_conserved_(L1p/L10e_family)',\n",
       "   'interval': Interval(488393, 489994, closed='both')},\n",
       "  {'chrom': 'LinJ.21',\n",
       "   'start': 488393,\n",
       "   'end': 489994,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_21T0019900',\n",
       "   'parent_id': 'LINF_210019900',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Protein_of_unknown_function_-_conserved_(L1p/L10e_family)',\n",
       "   'interval': Interval(488393, 489994, closed='both')},\n",
       "  {'chrom': 'LinJ.21',\n",
       "   'start': 488393,\n",
       "   'end': 489031,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_21T0019900',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(488393, 489031, closed='both')}],\n",
       " 'rejected_noCDS_c21.2460Z': [{'chrom': 'LinJ.21',\n",
       "   'start': 763461,\n",
       "   'end': 764506,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_210028800',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(763461, 764506, closed='both')},\n",
       "  {'chrom': 'LinJ.21',\n",
       "   'start': 763461,\n",
       "   'end': 764506,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_21T0028800',\n",
       "   'parent_id': 'LINF_210028800',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(763461, 764506, closed='both')}],\n",
       " 'rejected_noCDS_c22.2480': [{'chrom': 'LinJ.22',\n",
       "   'start': 61724,\n",
       "   'end': 62907,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_220006650',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(61724, 62907, closed='both')},\n",
       "  {'chrom': 'LinJ.22',\n",
       "   'start': 61724,\n",
       "   'end': 62907,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_22T0006650',\n",
       "   'parent_id': 'LINF_220006650',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(61724, 62907, closed='both')},\n",
       "  {'chrom': 'LinJ.22',\n",
       "   'start': 63270,\n",
       "   'end': 67965,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_220006700',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(63270, 67965, closed='both')},\n",
       "  {'chrom': 'LinJ.22',\n",
       "   'start': 63270,\n",
       "   'end': 67965,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_22T0006700',\n",
       "   'parent_id': 'LINF_220006700',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(63270, 67965, closed='both')},\n",
       "  {'chrom': 'LinJ.22',\n",
       "   'start': 63270,\n",
       "   'end': 64084,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_22T0006700',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(63270, 64084, closed='both')}],\n",
       " 'rejected_noCDS_c22.2540A': [{'chrom': 'LinJ.22',\n",
       "   'start': 751020,\n",
       "   'end': 754538,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_220023175',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(751020, 754538, closed='both')},\n",
       "  {'chrom': 'LinJ.22',\n",
       "   'start': 751020,\n",
       "   'end': 754538,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_22T0023175',\n",
       "   'parent_id': 'LINF_220023175',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(751020, 754538, closed='both')},\n",
       "  {'chrom': 'LinJ.22',\n",
       "   'start': 754539,\n",
       "   'end': 782139,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_220023200',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(754539, 782139, closed='both')},\n",
       "  {'chrom': 'LinJ.22',\n",
       "   'start': 754539,\n",
       "   'end': 782139,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_22T0023200',\n",
       "   'parent_id': 'LINF_220023200',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(754539, 782139, closed='both')}],\n",
       " 'rejected_noCDS_c23.2560': [{'chrom': 'LinJ.23',\n",
       "   'start': 55800,\n",
       "   'end': 57308,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_230006950',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(55800, 57308, closed='both')},\n",
       "  {'chrom': 'LinJ.23',\n",
       "   'start': 55800,\n",
       "   'end': 57308,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_23T0006950',\n",
       "   'parent_id': 'LINF_230006950',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(55800, 57308, closed='both')}],\n",
       " 'rejected_noCDS_c23.2570': [{'chrom': 'LinJ.23',\n",
       "   'start': 85852,\n",
       "   'end': 87903,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_230007700',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'terbinafine_resistance_locus_protein_(yip1)|HTBF',\n",
       "   'interval': Interval(85852, 87903, closed='both')},\n",
       "  {'chrom': 'LinJ.23',\n",
       "   'start': 85852,\n",
       "   'end': 87903,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_23T0007700',\n",
       "   'parent_id': 'LINF_230007700',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'terbinafine_resistance_locus_protein_(yip1)|HTBF',\n",
       "   'interval': Interval(85852, 87903, closed='both')},\n",
       "  {'chrom': 'LinJ.23',\n",
       "   'start': 86943,\n",
       "   'end': 87903,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_23T0007700',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(86943, 87903, closed='both')}],\n",
       " 'rejected_noCDS_c23.2690': [{'chrom': 'LinJ.23',\n",
       "   'start': 733585,\n",
       "   'end': 736847,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_230024900',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(733585, 736847, closed='both')},\n",
       "  {'chrom': 'LinJ.23',\n",
       "   'start': 733585,\n",
       "   'end': 736847,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_23T0024900',\n",
       "   'parent_id': 'LINF_230024900',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(733585, 736847, closed='both')},\n",
       "  {'chrom': 'LinJ.23',\n",
       "   'start': 734379,\n",
       "   'end': 736847,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_23T0024900',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(734379, 736847, closed='both')}],\n",
       " 'rejected_noCDS_c24.2710AJ': [{'chrom': 'LinJ.24',\n",
       "   'start': 108132,\n",
       "   'end': 110204,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_240008250',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(108132, 110204, closed='both')},\n",
       "  {'chrom': 'LinJ.24',\n",
       "   'start': 108132,\n",
       "   'end': 110204,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_24T0008250',\n",
       "   'parent_id': 'LINF_240008250',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(108132, 110204, closed='both')},\n",
       "  {'chrom': 'LinJ.24',\n",
       "   'start': 110205,\n",
       "   'end': 117720,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_240008300',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'UDP-galactose_transporter|LPG5A',\n",
       "   'interval': Interval(110205, 117720, closed='both')},\n",
       "  {'chrom': 'LinJ.24',\n",
       "   'start': 110205,\n",
       "   'end': 117720,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_24T0008300',\n",
       "   'parent_id': 'LINF_240008300',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'UDP-galactose_transporter|LPG5A',\n",
       "   'interval': Interval(110205, 117720, closed='both')},\n",
       "  {'chrom': 'LinJ.24',\n",
       "   'start': 110205,\n",
       "   'end': 110267,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_24T0008300',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(110205, 110267, closed='both')}],\n",
       " 'rejected_noCDS_c24.2770AJ': [{'chrom': 'LinJ.24',\n",
       "   'start': 110205,\n",
       "   'end': 117720,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_240008300',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'UDP-galactose_transporter|LPG5A',\n",
       "   'interval': Interval(110205, 117720, closed='both')},\n",
       "  {'chrom': 'LinJ.24',\n",
       "   'start': 110205,\n",
       "   'end': 117720,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_24T0008300',\n",
       "   'parent_id': 'LINF_240008300',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'UDP-galactose_transporter|LPG5A',\n",
       "   'interval': Interval(110205, 117720, closed='both')},\n",
       "  {'chrom': 'LinJ.24',\n",
       "   'start': 111675,\n",
       "   'end': 117720,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_24T0008300',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(111675, 117720, closed='both')},\n",
       "  {'chrom': 'LinJ.24',\n",
       "   'start': 117721,\n",
       "   'end': 122337,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_240008400',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'UDP-galactose_transporter|LPG5A',\n",
       "   'interval': Interval(117721, 122337, closed='both')},\n",
       "  {'chrom': 'LinJ.24',\n",
       "   'start': 117721,\n",
       "   'end': 122337,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_24T0008400',\n",
       "   'parent_id': 'LINF_240008400',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'UDP-galactose_transporter|LPG5A',\n",
       "   'interval': Interval(117721, 122337, closed='both')},\n",
       "  {'chrom': 'LinJ.24',\n",
       "   'start': 117721,\n",
       "   'end': 117783,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_24T0008400',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(117721, 117783, closed='both')}],\n",
       " 'rejected_noCDS_c25.2850AK': [{'chrom': 'LinJ.25',\n",
       "   'start': 799,\n",
       "   'end': 5363,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_250005000',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'beta_galactofuranosyl_transferase|LPG1',\n",
       "   'interval': Interval(799, 5363, closed='both')},\n",
       "  {'chrom': 'LinJ.25',\n",
       "   'start': 799,\n",
       "   'end': 5363,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_25T0005000',\n",
       "   'parent_id': 'LINF_250005000',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'beta_galactofuranosyl_transferase|LPG1',\n",
       "   'interval': Interval(799, 5363, closed='both')},\n",
       "  {'chrom': 'LinJ.25',\n",
       "   'start': 799,\n",
       "   'end': 3926,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_25T0005000',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(799, 3926, closed='both')}],\n",
       " 'rejected_noCDS_c25.2940': [{'chrom': 'LinJ.25',\n",
       "   'start': 373220,\n",
       "   'end': 374443,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_250015760',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(373220, 374443, closed='both')},\n",
       "  {'chrom': 'LinJ.25',\n",
       "   'start': 373220,\n",
       "   'end': 374443,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_25T0015760',\n",
       "   'parent_id': 'LINF_250015760',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(373220, 374443, closed='both')},\n",
       "  {'chrom': 'LinJ.25',\n",
       "   'start': 374444,\n",
       "   'end': 376365,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_250015800',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'LYR_domain-containing_protein',\n",
       "   'interval': Interval(374444, 376365, closed='both')},\n",
       "  {'chrom': 'LinJ.25',\n",
       "   'start': 374444,\n",
       "   'end': 376365,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_25T0015800',\n",
       "   'parent_id': 'LINF_250015800',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'LYR_domain-containing_protein',\n",
       "   'interval': Interval(374444, 376365, closed='both')},\n",
       "  {'chrom': 'LinJ.25',\n",
       "   'start': 374444,\n",
       "   'end': 374856,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_25T0015800',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(374444, 374856, closed='both')}],\n",
       " 'rejected_noCDS_c26.3040': [{'chrom': 'LinJ.26',\n",
       "   'start': 224623,\n",
       "   'end': 225662,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_260012750',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(224623, 225662, closed='both')},\n",
       "  {'chrom': 'LinJ.26',\n",
       "   'start': 224623,\n",
       "   'end': 225662,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_26T0012750',\n",
       "   'parent_id': 'LINF_260012750',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(224623, 225662, closed='both')},\n",
       "  {'chrom': 'LinJ.26',\n",
       "   'start': 225663,\n",
       "   'end': 226736,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_260012800',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'RNA-binding_protein|Y14',\n",
       "   'interval': Interval(225663, 226736, closed='both')},\n",
       "  {'chrom': 'LinJ.26',\n",
       "   'start': 225663,\n",
       "   'end': 226736,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_26T0012800',\n",
       "   'parent_id': 'LINF_260012800',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'RNA-binding_protein|Y14',\n",
       "   'interval': Interval(225663, 226736, closed='both')},\n",
       "  {'chrom': 'LinJ.26',\n",
       "   'start': 225663,\n",
       "   'end': 226143,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_26T0012800',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(225663, 226143, closed='both')}],\n",
       " 'rejected_noCDS_c26.3080': [{'chrom': 'LinJ.26',\n",
       "   'start': 598501,\n",
       "   'end': 599917,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_260021850',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(598501, 599917, closed='both')},\n",
       "  {'chrom': 'LinJ.26',\n",
       "   'start': 598501,\n",
       "   'end': 599917,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_26T0021850',\n",
       "   'parent_id': 'LINF_260021850',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(598501, 599917, closed='both')}],\n",
       " 'rejected_noCDS_c26.3110': [{'chrom': 'LinJ.26',\n",
       "   'start': 662292,\n",
       "   'end': 664196,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_260023450',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(662292, 664196, closed='both')},\n",
       "  {'chrom': 'LinJ.26',\n",
       "   'start': 662292,\n",
       "   'end': 664196,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_26T0023450',\n",
       "   'parent_id': 'LINF_260023450',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(662292, 664196, closed='both')}],\n",
       " 'rejected_noCDS_c27.3290AS': [{'chrom': 'LinJ.27',\n",
       "   'start': 557803,\n",
       "   'end': 565292,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_270018400',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'RING-H2_zinc_finger/Anaphase-promoting_complex_subunit_11_RING-H2_finger/Ring_finger_domain_containing_protein_-_putative',\n",
       "   'interval': Interval(557803, 565292, closed='both')},\n",
       "  {'chrom': 'LinJ.27',\n",
       "   'start': 557803,\n",
       "   'end': 565292,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_27T0018400',\n",
       "   'parent_id': 'LINF_270018400',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'RING-H2_zinc_finger/Anaphase-promoting_complex_subunit_11_RING-H2_finger/Ring_finger_domain_containing_protein_-_putative',\n",
       "   'interval': Interval(557803, 565292, closed='both')},\n",
       "  {'chrom': 'LinJ.27',\n",
       "   'start': 557803,\n",
       "   'end': 561174,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_27T0018400',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(557803, 561174, closed='both')}],\n",
       " 'rejected_noCDS_c27.3300AS': [{'chrom': 'LinJ.27',\n",
       "   'start': 577188,\n",
       "   'end': 579826,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_270018850',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': 'unknown',\n",
       "   'notes': 'Carboxypeptidase',\n",
       "   'interval': Interval(577188, 579826, closed='both')},\n",
       "  {'chrom': 'LinJ.27',\n",
       "   'start': 577188,\n",
       "   'end': 579826,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_27T0018850',\n",
       "   'parent_id': 'LINF_270018850',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Carboxypeptidase',\n",
       "   'interval': Interval(577188, 579826, closed='both')},\n",
       "  {'chrom': 'LinJ.27',\n",
       "   'start': 577719,\n",
       "   'end': 579429,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_27T0018850',\n",
       "   'parent_id': 'LINF_27T0018850',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': True,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(577719, 579429, closed='both')},\n",
       "  {'chrom': 'LinJ.27',\n",
       "   'start': 577188,\n",
       "   'end': 577718,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_27T0018850',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(577188, 577718, closed='both')}],\n",
       " 'rejected_noCDS_c27.3310': [{'chrom': 'LinJ.27',\n",
       "   'start': 701702,\n",
       "   'end': 703017,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_270022350',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(701702, 703017, closed='both')},\n",
       "  {'chrom': 'LinJ.27',\n",
       "   'start': 701702,\n",
       "   'end': 703017,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_27T0022350',\n",
       "   'parent_id': 'LINF_270022350',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(701702, 703017, closed='both')},\n",
       "  {'chrom': 'LinJ.27',\n",
       "   'start': 703018,\n",
       "   'end': 708994,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_270022400',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'diacylglycerol_acyltransferase_-_putative',\n",
       "   'interval': Interval(703018, 708994, closed='both')},\n",
       "  {'chrom': 'LinJ.27',\n",
       "   'start': 703018,\n",
       "   'end': 708994,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_27T0022400',\n",
       "   'parent_id': 'LINF_270022400',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'diacylglycerol_acyltransferase_-_putative',\n",
       "   'interval': Interval(703018, 708994, closed='both')},\n",
       "  {'chrom': 'LinJ.27',\n",
       "   'start': 703018,\n",
       "   'end': 703558,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_27T0022400',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(703018, 703558, closed='both')}],\n",
       " 'rejected_noCDS_c28.3370': [{'chrom': 'LinJ.28',\n",
       "   'start': 57936,\n",
       "   'end': 60119,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_280006850',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(57936, 60119, closed='both')},\n",
       "  {'chrom': 'LinJ.28',\n",
       "   'start': 57936,\n",
       "   'end': 60119,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_28T0006850',\n",
       "   'parent_id': 'LINF_280006850',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(57936, 60119, closed='both')},\n",
       "  {'chrom': 'LinJ.28',\n",
       "   'start': 60120,\n",
       "   'end': 63587,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_280006900',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(60120, 63587, closed='both')},\n",
       "  {'chrom': 'LinJ.28',\n",
       "   'start': 60120,\n",
       "   'end': 63587,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_28T0006900',\n",
       "   'parent_id': 'LINF_280006900',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(60120, 63587, closed='both')},\n",
       "  {'chrom': 'LinJ.28',\n",
       "   'start': 60120,\n",
       "   'end': 60396,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_28T0006900',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(60120, 60396, closed='both')}],\n",
       " 'rejected_noCDS_c28.3400': [{'chrom': 'LinJ.28',\n",
       "   'start': 592222,\n",
       "   'end': 595702,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_280021850',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(592222, 595702, closed='both')},\n",
       "  {'chrom': 'LinJ.28',\n",
       "   'start': 592222,\n",
       "   'end': 595702,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_28T0021850',\n",
       "   'parent_id': 'LINF_280021850',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(592222, 595702, closed='both')}],\n",
       " 'rejected_noCDS_c28.3410': [{'chrom': 'LinJ.28',\n",
       "   'start': 654749,\n",
       "   'end': 663755,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_280023400',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(654749, 663755, closed='both')},\n",
       "  {'chrom': 'LinJ.28',\n",
       "   'start': 654749,\n",
       "   'end': 663755,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_28T0023400',\n",
       "   'parent_id': 'LINF_280023400',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(654749, 663755, closed='both')},\n",
       "  {'chrom': 'LinJ.28',\n",
       "   'start': 654749,\n",
       "   'end': 656104,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_28T0023400',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(654749, 656104, closed='both')}],\n",
       " 'rejected_noCDS_c28.3430Z': [{'chrom': 'LinJ.28',\n",
       "   'start': 1202044,\n",
       "   'end': 1204457,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_280039900',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1202044, 1204457, closed='both')},\n",
       "  {'chrom': 'LinJ.28',\n",
       "   'start': 1202044,\n",
       "   'end': 1204457,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_28T0039900',\n",
       "   'parent_id': 'LINF_280039900',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1202044, 1204457, closed='both')}],\n",
       " 'rejected_noCDS_c29.3440': [{'chrom': 'LinJ.29',\n",
       "   'start': 94100,\n",
       "   'end': 97001,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_290007800',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'D-lactate_dehydrogenase-like_protein',\n",
       "   'interval': Interval(94100, 97001, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 94100,\n",
       "   'end': 97001,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0007800',\n",
       "   'parent_id': 'LINF_290007800',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'D-lactate_dehydrogenase-like_protein',\n",
       "   'interval': Interval(94100, 97001, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 94100,\n",
       "   'end': 95303,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_29T0007800',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(94100, 95303, closed='both')}],\n",
       " 'rejected_noCDS_c29.3450': [{'chrom': 'LinJ.29',\n",
       "   'start': 154596,\n",
       "   'end': 160193,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_290009600',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(154596, 160193, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 154596,\n",
       "   'end': 160193,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0009600',\n",
       "   'parent_id': 'LINF_290009600',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(154596, 160193, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 154596,\n",
       "   'end': 156174,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_29T0009600',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(154596, 156174, closed='both')}],\n",
       " 'rejected_noCDS_c29.3550AX': [{'chrom': 'LinJ.29',\n",
       "   'start': 556193,\n",
       "   'end': 560127,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_290019350',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein',\n",
       "   'interval': Interval(556193, 560127, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 556193,\n",
       "   'end': 560127,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0019350',\n",
       "   'parent_id': 'LINF_290019350',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein',\n",
       "   'interval': Interval(556193, 560127, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 557537,\n",
       "   'end': 560127,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_29T0019350',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(557537, 560127, closed='both')}],\n",
       " 'rejected_noCDS_c29.3570AZ': [{'chrom': 'LinJ.29',\n",
       "   'start': 563252,\n",
       "   'end': 564329,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_290019415',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(563252, 564329, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 563252,\n",
       "   'end': 564329,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0019415',\n",
       "   'parent_id': 'LINF_290019415',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(563252, 564329, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 564330,\n",
       "   'end': 566488,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_290019430',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(564330, 566488, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 564330,\n",
       "   'end': 566488,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0019430',\n",
       "   'parent_id': 'LINF_290019430',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(564330, 566488, closed='both')}],\n",
       " 'rejected_noCDS_c29.3580AX': [{'chrom': 'LinJ.29',\n",
       "   'start': 564330,\n",
       "   'end': 566488,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_290019430',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(564330, 566488, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 564330,\n",
       "   'end': 566488,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0019430',\n",
       "   'parent_id': 'LINF_290019430',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(564330, 566488, closed='both')}],\n",
       " 'rejected_noCDS_c29.3590': [{'chrom': 'LinJ.29',\n",
       "   'start': 567113,\n",
       "   'end': 570359,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_290019445',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(567113, 570359, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 567113,\n",
       "   'end': 570359,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0019445',\n",
       "   'parent_id': 'LINF_290019445',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(567113, 570359, closed='both')}],\n",
       " 'rejected_noCDS_c29.3620AZ': [{'chrom': 'LinJ.29',\n",
       "   'start': 570696,\n",
       "   'end': 572858,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_290019475',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(570696, 572858, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 570696,\n",
       "   'end': 572858,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0019475',\n",
       "   'parent_id': 'LINF_290019475',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(570696, 572858, closed='both')}],\n",
       " 'rejected_noCDS_c29.3630AX': [{'chrom': 'LinJ.29',\n",
       "   'start': 570696,\n",
       "   'end': 572858,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_290019475',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(570696, 572858, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 570696,\n",
       "   'end': 572858,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0019475',\n",
       "   'parent_id': 'LINF_290019475',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(570696, 572858, closed='both')}],\n",
       " 'rejected_noCDS_c29.3650AZ': [{'chrom': 'LinJ.29',\n",
       "   'start': 577099,\n",
       "   'end': 579261,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_290019560',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(577099, 579261, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 577099,\n",
       "   'end': 579261,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0019560',\n",
       "   'parent_id': 'LINF_290019560',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(577099, 579261, closed='both')}],\n",
       " 'rejected_noCDS_c29.3660AX': [{'chrom': 'LinJ.29',\n",
       "   'start': 577099,\n",
       "   'end': 579261,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_290019560',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(577099, 579261, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 577099,\n",
       "   'end': 579261,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0019560',\n",
       "   'parent_id': 'LINF_290019560',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(577099, 579261, closed='both')}],\n",
       " 'rejected_noCDS_c29.3690': [{'chrom': 'LinJ.29',\n",
       "   'start': 583499,\n",
       "   'end': 587978,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_290019700',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein',\n",
       "   'interval': Interval(583499, 587978, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 583499,\n",
       "   'end': 587978,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0019700',\n",
       "   'parent_id': 'LINF_290019700',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein',\n",
       "   'interval': Interval(583499, 587978, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 584181,\n",
       "   'end': 587978,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_29T0019700',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(584181, 587978, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 587979,\n",
       "   'end': 596043,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_290019800',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'RNA_binding_protein_-_putative',\n",
       "   'interval': Interval(587979, 596043, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 587979,\n",
       "   'end': 596043,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0019800',\n",
       "   'parent_id': 'LINF_290019800',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'RNA_binding_protein_-_putative',\n",
       "   'interval': Interval(587979, 596043, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 587979,\n",
       "   'end': 592487,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_29T0019800',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(587979, 592487, closed='both')}],\n",
       " 'rejected_noCDS_c29.3750': [{'chrom': 'LinJ.29',\n",
       "   'start': 670544,\n",
       "   'end': 676210,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_290020800',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein',\n",
       "   'interval': Interval(670544, 676210, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 670544,\n",
       "   'end': 676210,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0020800',\n",
       "   'parent_id': 'LINF_290020800',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein',\n",
       "   'interval': Interval(670544, 676210, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 674876,\n",
       "   'end': 676210,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_29T0020800',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(674876, 676210, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 676211,\n",
       "   'end': 685529,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_290020900',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Phosphatidylinositol_kinase|PIK',\n",
       "   'interval': Interval(676211, 685529, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 676211,\n",
       "   'end': 685529,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0020900',\n",
       "   'parent_id': 'LINF_290020900',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Phosphatidylinositol_kinase|PIK',\n",
       "   'interval': Interval(676211, 685529, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 676211,\n",
       "   'end': 677181,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_29T0020900',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(676211, 677181, closed='both')}],\n",
       " 'rejected_noCDS_c29.3800': [{'chrom': 'LinJ.29',\n",
       "   'start': 1065275,\n",
       "   'end': 1075994,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_290030600',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(1065275, 1075994, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 1065275,\n",
       "   'end': 1075994,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0030600',\n",
       "   'parent_id': 'LINF_290030600',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(1065275, 1075994, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 1075237,\n",
       "   'end': 1075994,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_29T0030600',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1075237, 1075994, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 1076374,\n",
       "   'end': 1079215,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_290030650',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1076374, 1079215, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 1076374,\n",
       "   'end': 1079215,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0030650',\n",
       "   'parent_id': 'LINF_290030650',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1076374, 1079215, closed='both')}],\n",
       " 'rejected_noCDS_c29.3810': [{'chrom': 'LinJ.29',\n",
       "   'start': 1187676,\n",
       "   'end': 1189124,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_290034000',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(1187676, 1189124, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 1187676,\n",
       "   'end': 1189124,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0034000',\n",
       "   'parent_id': 'LINF_290034000',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(1187676, 1189124, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 1188055,\n",
       "   'end': 1189124,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_29T0034000',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1188055, 1189124, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 1189252,\n",
       "   'end': 1190272,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_290034030',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1189252, 1190272, closed='both')},\n",
       "  {'chrom': 'LinJ.29',\n",
       "   'start': 1189252,\n",
       "   'end': 1190272,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_29T0034030',\n",
       "   'parent_id': 'LINF_290034030',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1189252, 1190272, closed='both')}],\n",
       " 'rejected_noCDS_c30.3860BC': [{'chrom': 'LinJ.30',\n",
       "   'start': 102141,\n",
       "   'end': 105323,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_300008300',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Rab-GTPase-TBC_domain_containing_protein',\n",
       "   'interval': Interval(102141, 105323, closed='both')},\n",
       "  {'chrom': 'LinJ.30',\n",
       "   'start': 102141,\n",
       "   'end': 105323,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_30T0008300',\n",
       "   'parent_id': 'LINF_300008300',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Rab-GTPase-TBC_domain_containing_protein',\n",
       "   'interval': Interval(102141, 105323, closed='both')},\n",
       "  {'chrom': 'LinJ.30',\n",
       "   'start': 104949,\n",
       "   'end': 105323,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_30T0008300',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(104949, 105323, closed='both')},\n",
       "  {'chrom': 'LinJ.30',\n",
       "   'start': 105722,\n",
       "   'end': 109074,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_300008400',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'KU80_protein',\n",
       "   'interval': Interval(105722, 109074, closed='both')},\n",
       "  {'chrom': 'LinJ.30',\n",
       "   'start': 105722,\n",
       "   'end': 109074,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_30T0008400',\n",
       "   'parent_id': 'LINF_300008400',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'KU80_protein',\n",
       "   'interval': Interval(105722, 109074, closed='both')},\n",
       "  {'chrom': 'LinJ.30',\n",
       "   'start': 105722,\n",
       "   'end': 106621,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_30T0008400',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(105722, 106621, closed='both')}],\n",
       " 'rejected_noCDS_c30.3930': [{'chrom': 'LinJ.30',\n",
       "   'start': 486069,\n",
       "   'end': 487146,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_300019650',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(486069, 487146, closed='both')},\n",
       "  {'chrom': 'LinJ.30',\n",
       "   'start': 486069,\n",
       "   'end': 487146,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_30T0019650',\n",
       "   'parent_id': 'LINF_300019650',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(486069, 487146, closed='both')}],\n",
       " 'rejected_noCDS_c31.4120': [{'chrom': 'LinJ.31',\n",
       "   'start': 148538,\n",
       "   'end': 149817,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_310009750',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(148538, 149817, closed='both')},\n",
       "  {'chrom': 'LinJ.31',\n",
       "   'start': 148538,\n",
       "   'end': 149817,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_31T0009750',\n",
       "   'parent_id': 'LINF_310009750',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(148538, 149817, closed='both')},\n",
       "  {'chrom': 'LinJ.31',\n",
       "   'start': 150123,\n",
       "   'end': 152831,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_310009800',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'amastin_surface_glycoprotein',\n",
       "   'interval': Interval(150123, 152831, closed='both')},\n",
       "  {'chrom': 'LinJ.31',\n",
       "   'start': 150123,\n",
       "   'end': 152831,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_31T0009800',\n",
       "   'parent_id': 'LINF_310009800',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'amastin_surface_glycoprotein',\n",
       "   'interval': Interval(150123, 152831, closed='both')},\n",
       "  {'chrom': 'LinJ.31',\n",
       "   'start': 150123,\n",
       "   'end': 152088,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_31T0009800',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(150123, 152088, closed='both')}],\n",
       " 'rejected_noCDS_c31.4140': [{'chrom': 'LinJ.31',\n",
       "   'start': 153423,\n",
       "   'end': 156946,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_310009850',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': 'unknown',\n",
       "   'notes': 'tuzin_like_protein',\n",
       "   'interval': Interval(153423, 156946, closed='both')},\n",
       "  {'chrom': 'LinJ.31',\n",
       "   'start': 153423,\n",
       "   'end': 156946,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_31T0009850',\n",
       "   'parent_id': 'LINF_310009850',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'tuzin_like_protein',\n",
       "   'interval': Interval(153423, 156946, closed='both')},\n",
       "  {'chrom': 'LinJ.31',\n",
       "   'start': 153423,\n",
       "   'end': 154161,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_31T0009850',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(153423, 154161, closed='both')}],\n",
       " 'rejected_noCDS_c31.4180': [{'chrom': 'LinJ.31',\n",
       "   'start': 313930,\n",
       "   'end': 317135,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_310014225',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(313930, 317135, closed='both')},\n",
       "  {'chrom': 'LinJ.31',\n",
       "   'start': 313930,\n",
       "   'end': 317135,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_31T0014225',\n",
       "   'parent_id': 'LINF_310014225',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(313930, 317135, closed='both')}],\n",
       " 'rejected_noCDS_c31.4310': [{'chrom': 'LinJ.31',\n",
       "   'start': 633369,\n",
       "   'end': 634601,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_310020750',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(633369, 634601, closed='both')},\n",
       "  {'chrom': 'LinJ.31',\n",
       "   'start': 633369,\n",
       "   'end': 634601,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_31T0020750',\n",
       "   'parent_id': 'LINF_310020750',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(633369, 634601, closed='both')}],\n",
       " 'rejected_noCDS_c31.4400': [{'chrom': 'LinJ.31',\n",
       "   'start': 908256,\n",
       "   'end': 910777,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_310025650',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(908256, 910777, closed='both')},\n",
       "  {'chrom': 'LinJ.31',\n",
       "   'start': 908256,\n",
       "   'end': 910777,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_31T0025650',\n",
       "   'parent_id': 'LINF_310025650',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(908256, 910777, closed='both')},\n",
       "  {'chrom': 'LinJ.31',\n",
       "   'start': 910778,\n",
       "   'end': 916576,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_310025700',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_kinase',\n",
       "   'interval': Interval(910778, 916576, closed='both')},\n",
       "  {'chrom': 'LinJ.31',\n",
       "   'start': 910778,\n",
       "   'end': 916576,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_31T0025700',\n",
       "   'parent_id': 'LINF_310025700',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_kinase',\n",
       "   'interval': Interval(910778, 916576, closed='both')},\n",
       "  {'chrom': 'LinJ.31',\n",
       "   'start': 910778,\n",
       "   'end': 912292,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_31T0025700',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(910778, 912292, closed='both')}],\n",
       " 'rejected_noCDS_c31.4470': [{'chrom': 'LinJ.31',\n",
       "   'start': 1038721,\n",
       "   'end': 1039235,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_310028450',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1038721, 1039235, closed='both')},\n",
       "  {'chrom': 'LinJ.31',\n",
       "   'start': 1038721,\n",
       "   'end': 1039235,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_31T0028450',\n",
       "   'parent_id': 'LINF_310028450',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1038721, 1039235, closed='both')}],\n",
       " 'rejected_noCDS_c31.4550': [{'chrom': 'LinJ.31',\n",
       "   'start': 1325203,\n",
       "   'end': 1326083,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_310034550',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1325203, 1326083, closed='both')},\n",
       "  {'chrom': 'LinJ.31',\n",
       "   'start': 1325203,\n",
       "   'end': 1326083,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_31T0034550',\n",
       "   'parent_id': 'LINF_310034550',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1325203, 1326083, closed='both')}],\n",
       " 'rejected_noCDS_c32.4570O': [{'chrom': 'LinJ.32',\n",
       "   'start': 968,\n",
       "   'end': 1399,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_320004800',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(968, 1399, closed='both')},\n",
       "  {'chrom': 'LinJ.32',\n",
       "   'start': 968,\n",
       "   'end': 1399,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_32T0004800',\n",
       "   'parent_id': 'LINF_320004800',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(968, 1399, closed='both')},\n",
       "  {'chrom': 'LinJ.32',\n",
       "   'start': 1943,\n",
       "   'end': 2745,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_320004900',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1943, 2745, closed='both')},\n",
       "  {'chrom': 'LinJ.32',\n",
       "   'start': 1943,\n",
       "   'end': 2745,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_32T0004900',\n",
       "   'parent_id': 'LINF_320004900',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1943, 2745, closed='both')}],\n",
       " 'rejected_noCDS_c32.4690': [{'chrom': 'LinJ.32',\n",
       "   'start': 746881,\n",
       "   'end': 751164,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_320025100',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'RUS_family_member',\n",
       "   'interval': Interval(746881, 751164, closed='both')},\n",
       "  {'chrom': 'LinJ.32',\n",
       "   'start': 746881,\n",
       "   'end': 751164,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_32T0025100',\n",
       "   'parent_id': 'LINF_320025100',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'RUS_family_member',\n",
       "   'interval': Interval(746881, 751164, closed='both')},\n",
       "  {'chrom': 'LinJ.32',\n",
       "   'start': 746881,\n",
       "   'end': 749098,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_32T0025100',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(746881, 749098, closed='both')}],\n",
       " 'rejected_noCDS_c32.4710BO': [{'chrom': 'LinJ.32',\n",
       "   'start': 802344,\n",
       "   'end': 803500,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_320026750',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(802344, 803500, closed='both')},\n",
       "  {'chrom': 'LinJ.32',\n",
       "   'start': 802344,\n",
       "   'end': 803500,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_32T0026750',\n",
       "   'parent_id': 'LINF_320026750',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(802344, 803500, closed='both')},\n",
       "  {'chrom': 'LinJ.32',\n",
       "   'start': 803501,\n",
       "   'end': 807153,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_320026800',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Cytidylyltransferase',\n",
       "   'interval': Interval(803501, 807153, closed='both')},\n",
       "  {'chrom': 'LinJ.32',\n",
       "   'start': 803501,\n",
       "   'end': 807153,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_32T0026800',\n",
       "   'parent_id': 'LINF_320026800',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Cytidylyltransferase',\n",
       "   'interval': Interval(803501, 807153, closed='both')},\n",
       "  {'chrom': 'LinJ.32',\n",
       "   'start': 803501,\n",
       "   'end': 805717,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_32T0026800',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(803501, 805717, closed='both')}],\n",
       " 'rejected_noCDS_c33.4850': [{'chrom': 'LinJ.33',\n",
       "   'start': 530154,\n",
       "   'end': 533949,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_330020600',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Major_Facilitator_Superfamily/Nucleoside_H+_symporter/LacY_proton/sugar_symporter_-_putative',\n",
       "   'interval': Interval(530154, 533949, closed='both')},\n",
       "  {'chrom': 'LinJ.33',\n",
       "   'start': 530154,\n",
       "   'end': 533949,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_33T0020600',\n",
       "   'parent_id': 'LINF_330020600',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Major_Facilitator_Superfamily/Nucleoside_H+_symporter/LacY_proton/sugar_symporter_-_putative',\n",
       "   'interval': Interval(530154, 533949, closed='both')},\n",
       "  {'chrom': 'LinJ.33',\n",
       "   'start': 530154,\n",
       "   'end': 530693,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_33T0020600',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(530154, 530693, closed='both')}],\n",
       " 'rejected_noCDS_c33.4890BC': [{'chrom': 'LinJ.33',\n",
       "   'start': 1002245,\n",
       "   'end': 1005303,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_330032400',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Acetate:succinate_CoA_transferase',\n",
       "   'interval': Interval(1002245, 1005303, closed='both')},\n",
       "  {'chrom': 'LinJ.33',\n",
       "   'start': 1002245,\n",
       "   'end': 1005303,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_33T0032400',\n",
       "   'parent_id': 'LINF_330032400',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Acetate:succinate_CoA_transferase',\n",
       "   'interval': Interval(1002245, 1005303, closed='both')},\n",
       "  {'chrom': 'LinJ.33',\n",
       "   'start': 1004088,\n",
       "   'end': 1005303,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_33T0032400',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1004088, 1005303, closed='both')},\n",
       "  {'chrom': 'LinJ.33',\n",
       "   'start': 1005687,\n",
       "   'end': 1009527,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_330032500',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Rab-GTPase-TBC_domain_containing_protein',\n",
       "   'interval': Interval(1005687, 1009527, closed='both')},\n",
       "  {'chrom': 'LinJ.33',\n",
       "   'start': 1005687,\n",
       "   'end': 1009527,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_33T0032500',\n",
       "   'parent_id': 'LINF_330032500',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Rab-GTPase-TBC_domain_containing_protein',\n",
       "   'interval': Interval(1005687, 1009527, closed='both')},\n",
       "  {'chrom': 'LinJ.33',\n",
       "   'start': 1005687,\n",
       "   'end': 1005824,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_33T0032500',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1005687, 1005824, closed='both')}],\n",
       " 'rejected_noCDS_c33.4940BQ': [{'chrom': 'LinJ.33',\n",
       "   'start': 1254377,\n",
       "   'end': 1257191,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_330038400',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(1254377, 1257191, closed='both')},\n",
       "  {'chrom': 'LinJ.33',\n",
       "   'start': 1254377,\n",
       "   'end': 1257191,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_33T0038400',\n",
       "   'parent_id': 'LINF_330038400',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(1254377, 1257191, closed='both')},\n",
       "  {'chrom': 'LinJ.33',\n",
       "   'start': 1255198,\n",
       "   'end': 1257191,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_33T0038400',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1255198, 1257191, closed='both')}],\n",
       " 'rejected_noCDS_c33.4960BQ': [{'chrom': 'LinJ.33',\n",
       "   'start': 1272320,\n",
       "   'end': 1275156,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_330038545',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1272320, 1275156, closed='both')},\n",
       "  {'chrom': 'LinJ.33',\n",
       "   'start': 1272320,\n",
       "   'end': 1275156,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_33T0038545',\n",
       "   'parent_id': 'LINF_330038545',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1272320, 1275156, closed='both')}],\n",
       " 'rejected_noCDS_c33.5040AK': [{'chrom': 'LinJ.33',\n",
       "   'start': 1531735,\n",
       "   'end': 1532106,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_330042920',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1531735, 1532106, closed='both')},\n",
       "  {'chrom': 'LinJ.33',\n",
       "   'start': 1531735,\n",
       "   'end': 1532106,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_33T0042920',\n",
       "   'parent_id': 'LINF_330042920',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1531735, 1532106, closed='both')}],\n",
       " 'rejected_noCDS_c34.5060': [{'chrom': 'LinJ.34',\n",
       "   'start': 329434,\n",
       "   'end': 331016,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340012450',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(329434, 331016, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 329434,\n",
       "   'end': 331016,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0012450',\n",
       "   'parent_id': 'LINF_340012450',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(329434, 331016, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 331111,\n",
       "   'end': 331968,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_340012500',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(331111, 331968, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 331111,\n",
       "   'end': 331968,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0012500',\n",
       "   'parent_id': 'LINF_340012500',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(331111, 331968, closed='both')}],\n",
       " 'rejected_noCDS_c34.5140BS': [{'chrom': 'LinJ.34',\n",
       "   'start': 511354,\n",
       "   'end': 513171,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340017475',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(511354, 513171, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 511354,\n",
       "   'end': 513171,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0017475',\n",
       "   'parent_id': 'LINF_340017475',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(511354, 513171, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 513172,\n",
       "   'end': 516089,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340017500',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'amastin-like_protein',\n",
       "   'interval': Interval(513172, 516089, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 513172,\n",
       "   'end': 516089,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0017500',\n",
       "   'parent_id': 'LINF_340017500',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'amastin-like_protein',\n",
       "   'interval': Interval(513172, 516089, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 513172,\n",
       "   'end': 515349,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_34T0017500',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(513172, 515349, closed='both')}],\n",
       " 'rejected_noCDS_c34.5180BS': [{'chrom': 'LinJ.34',\n",
       "   'start': 735203,\n",
       "   'end': 739818,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340022850',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(735203, 739818, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 735203,\n",
       "   'end': 739818,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0022850',\n",
       "   'parent_id': 'LINF_340022850',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(735203, 739818, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 739819,\n",
       "   'end': 743551,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340022900',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'amastin-like_protein',\n",
       "   'interval': Interval(739819, 743551, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 739819,\n",
       "   'end': 743551,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0022900',\n",
       "   'parent_id': 'LINF_340022900',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'amastin-like_protein',\n",
       "   'interval': Interval(739819, 743551, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 739819,\n",
       "   'end': 742807,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_34T0022900',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(739819, 742807, closed='both')}],\n",
       " 'rejected_noCDS_c34.5230BU': [{'chrom': 'LinJ.34',\n",
       "   'start': 754280,\n",
       "   'end': 757294,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340023220',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(754280, 757294, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 754280,\n",
       "   'end': 757294,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0023220',\n",
       "   'parent_id': 'LINF_340023220',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(754280, 757294, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 758277,\n",
       "   'end': 761292,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340023260',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(758277, 761292, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 758277,\n",
       "   'end': 761292,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0023260',\n",
       "   'parent_id': 'LINF_340023260',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(758277, 761292, closed='both')}],\n",
       " 'rejected_noCDS_c34.5260BU': [{'chrom': 'LinJ.34',\n",
       "   'start': 758277,\n",
       "   'end': 761292,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340023260',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(758277, 761292, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 758277,\n",
       "   'end': 761292,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0023260',\n",
       "   'parent_id': 'LINF_340023260',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(758277, 761292, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 761293,\n",
       "   'end': 765197,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340023280',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(761293, 765197, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 761293,\n",
       "   'end': 765197,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0023280',\n",
       "   'parent_id': 'LINF_340023280',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(761293, 765197, closed='both')}],\n",
       " 'rejected_noCDS_c34.5290BU': [{'chrom': 'LinJ.34',\n",
       "   'start': 761293,\n",
       "   'end': 765197,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340023280',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(761293, 765197, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 761293,\n",
       "   'end': 765197,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0023280',\n",
       "   'parent_id': 'LINF_340023280',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(761293, 765197, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 766090,\n",
       "   'end': 768423,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340023300',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'amastin-like_surface_protein_-_putative',\n",
       "   'interval': Interval(766090, 768423, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 766090,\n",
       "   'end': 768423,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0023300',\n",
       "   'parent_id': 'LINF_340023300',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'amastin-like_surface_protein_-_putative',\n",
       "   'interval': Interval(766090, 768423, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 766090,\n",
       "   'end': 767657,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_34T0023300',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(766090, 767657, closed='both')}],\n",
       " 'rejected_noCDS_c34.5310BU': [{'chrom': 'LinJ.34',\n",
       "   'start': 772123,\n",
       "   'end': 775066,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340023415',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(772123, 775066, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 772123,\n",
       "   'end': 775066,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0023415',\n",
       "   'parent_id': 'LINF_340023415',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(772123, 775066, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 775900,\n",
       "   'end': 777200,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340023430',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(775900, 777200, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 775900,\n",
       "   'end': 777200,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0023430',\n",
       "   'parent_id': 'LINF_340023430',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(775900, 777200, closed='both')}],\n",
       " 'rejected_noCDS_c34.5320BT': [{'chrom': 'LinJ.34',\n",
       "   'start': 775900,\n",
       "   'end': 777200,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340023430',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(775900, 777200, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 775900,\n",
       "   'end': 777200,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0023430',\n",
       "   'parent_id': 'LINF_340023430',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(775900, 777200, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 777201,\n",
       "   'end': 778297,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340023445',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(777201, 778297, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 777201,\n",
       "   'end': 778297,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0023445',\n",
       "   'parent_id': 'LINF_340023445',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(777201, 778297, closed='both')}],\n",
       " 'rejected_noCDS_c34.5330BU': [{'chrom': 'LinJ.34',\n",
       "   'start': 777201,\n",
       "   'end': 778297,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340023445',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(777201, 778297, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 777201,\n",
       "   'end': 778297,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0023445',\n",
       "   'parent_id': 'LINF_340023445',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(777201, 778297, closed='both')}],\n",
       " 'rejected_noCDS_c34.5350BU': [{'chrom': 'LinJ.34',\n",
       "   'start': 778924,\n",
       "   'end': 780014,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340023460',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(778924, 780014, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 778924,\n",
       "   'end': 780014,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0023460',\n",
       "   'parent_id': 'LINF_340023460',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(778924, 780014, closed='both')}],\n",
       " 'rejected_noCDS_c34.5360BT': [{'chrom': 'LinJ.34',\n",
       "   'start': 778924,\n",
       "   'end': 780014,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340023460',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(778924, 780014, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 778924,\n",
       "   'end': 780014,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0023460',\n",
       "   'parent_id': 'LINF_340023460',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(778924, 780014, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 780144,\n",
       "   'end': 781532,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340023475',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(780144, 781532, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 780144,\n",
       "   'end': 781532,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0023475',\n",
       "   'parent_id': 'LINF_340023475',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(780144, 781532, closed='both')}],\n",
       " 'rejected_noCDS_c34.5370BU': [{'chrom': 'LinJ.34',\n",
       "   'start': 780144,\n",
       "   'end': 781532,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340023475',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(780144, 781532, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 780144,\n",
       "   'end': 781532,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0023475',\n",
       "   'parent_id': 'LINF_340023475',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(780144, 781532, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 781533,\n",
       "   'end': 782303,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340023490',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(781533, 782303, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 781533,\n",
       "   'end': 782303,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0023490',\n",
       "   'parent_id': 'LINF_340023490',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(781533, 782303, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 782521,\n",
       "   'end': 784862,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340023500',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'amastin-like_surface_protein',\n",
       "   'interval': Interval(782521, 784862, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 782521,\n",
       "   'end': 784862,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0023500',\n",
       "   'parent_id': 'LINF_340023500',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'amastin-like_surface_protein',\n",
       "   'interval': Interval(782521, 784862, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 782521,\n",
       "   'end': 784096,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_34T0023500',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(782521, 784096, closed='both')}],\n",
       " 'rejected_noCDS_c34.5400BT': [{'chrom': 'LinJ.34',\n",
       "   'start': 814218,\n",
       "   'end': 816590,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_340024500',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'amastin-like_surface_protein_-_putative',\n",
       "   'interval': Interval(814218, 816590, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 814218,\n",
       "   'end': 816590,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0024500',\n",
       "   'parent_id': 'LINF_340024500',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'amastin-like_surface_protein_-_putative',\n",
       "   'interval': Interval(814218, 816590, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 814218,\n",
       "   'end': 815826,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_34T0024500',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(814218, 815826, closed='both')}],\n",
       " 'rejected_noCDS_c34.5920CC': [{'chrom': 'LinJ.34',\n",
       "   'start': 1372364,\n",
       "   'end': 1376363,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_340037600',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Tripartite_attachment_complex_40|TAC40',\n",
       "   'interval': Interval(1372364, 1376363, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 1372364,\n",
       "   'end': 1376363,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0037600',\n",
       "   'parent_id': 'LINF_340037600',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Tripartite_attachment_complex_40|TAC40',\n",
       "   'interval': Interval(1372364, 1376363, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 1374067,\n",
       "   'end': 1376363,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_34T0037600',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1374067, 1376363, closed='both')}],\n",
       " 'rejected_noCDS_c34.5930CC': [{'chrom': 'LinJ.34',\n",
       "   'start': 1381967,\n",
       "   'end': 1383514,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_340037850',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1381967, 1383514, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 1381967,\n",
       "   'end': 1383514,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0037850',\n",
       "   'parent_id': 'LINF_340037850',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1381967, 1383514, closed='both')}],\n",
       " 'rejected_noCDS_c34.5960CC': [{'chrom': 'LinJ.34',\n",
       "   'start': 1391972,\n",
       "   'end': 1395962,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_340038100',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Tripartite_attachment_complex_40|TAC40',\n",
       "   'interval': Interval(1391972, 1395962, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 1391972,\n",
       "   'end': 1395962,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_34T0038100',\n",
       "   'parent_id': 'LINF_340038100',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Tripartite_attachment_complex_40|TAC40',\n",
       "   'interval': Interval(1391972, 1395962, closed='both')},\n",
       "  {'chrom': 'LinJ.34',\n",
       "   'start': 1393676,\n",
       "   'end': 1395962,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_34T0038100',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1393676, 1395962, closed='both')}],\n",
       " 'rejected_noCDS_c35.6080': [{'chrom': 'LinJ.35',\n",
       "   'start': 296633,\n",
       "   'end': 298023,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_350012950',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(296633, 298023, closed='both')},\n",
       "  {'chrom': 'LinJ.35',\n",
       "   'start': 296633,\n",
       "   'end': 298023,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_35T0012950',\n",
       "   'parent_id': 'LINF_350012950',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(296633, 298023, closed='both')}],\n",
       " 'rejected_noCDS_c35.6100': [{'chrom': 'LinJ.35',\n",
       "   'start': 332062,\n",
       "   'end': 334222,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_350013950',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(332062, 334222, closed='both')},\n",
       "  {'chrom': 'LinJ.35',\n",
       "   'start': 332062,\n",
       "   'end': 334222,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_35T0013950',\n",
       "   'parent_id': 'LINF_350013950',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(332062, 334222, closed='both')},\n",
       "  {'chrom': 'LinJ.35',\n",
       "   'start': 334360,\n",
       "   'end': 336208,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_350014000',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_of_unknown_function_-_conserved',\n",
       "   'interval': Interval(334360, 336208, closed='both')},\n",
       "  {'chrom': 'LinJ.35',\n",
       "   'start': 334360,\n",
       "   'end': 336208,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_35T0014000',\n",
       "   'parent_id': 'LINF_350014000',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein_of_unknown_function_-_conserved',\n",
       "   'interval': Interval(334360, 336208, closed='both')},\n",
       "  {'chrom': 'LinJ.35',\n",
       "   'start': 334360,\n",
       "   'end': 334792,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_35T0014000',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(334360, 334792, closed='both')}],\n",
       " 'rejected_noCDS_c35.6180CJ': [{'chrom': 'LinJ.35',\n",
       "   'start': 1125055,\n",
       "   'end': 1126995,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_350034650',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1125055, 1126995, closed='both')},\n",
       "  {'chrom': 'LinJ.35',\n",
       "   'start': 1125055,\n",
       "   'end': 1126995,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_35T0034650',\n",
       "   'parent_id': 'LINF_350034650',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1125055, 1126995, closed='both')}],\n",
       " 'rejected_noCDS_c35.6250CJ': [{'chrom': 'LinJ.35',\n",
       "   'start': 1366385,\n",
       "   'end': 1367810,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_350041050',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1366385, 1367810, closed='both')},\n",
       "  {'chrom': 'LinJ.35',\n",
       "   'start': 1366385,\n",
       "   'end': 1367810,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_35T0041050',\n",
       "   'parent_id': 'LINF_350041050',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1366385, 1367810, closed='both')}],\n",
       " 'rejected_noCDS_c35.6270': [{'chrom': 'LinJ.35',\n",
       "   'start': 1550851,\n",
       "   'end': 1552427,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_350046600',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Splicing_factor_3B_subunit_10_(SF3b10)_-_putative',\n",
       "   'interval': Interval(1550851, 1552427, closed='both')},\n",
       "  {'chrom': 'LinJ.35',\n",
       "   'start': 1550851,\n",
       "   'end': 1552427,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_35T0046600',\n",
       "   'parent_id': 'LINF_350046600',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'Splicing_factor_3B_subunit_10_(SF3b10)_-_putative',\n",
       "   'interval': Interval(1550851, 1552427, closed='both')},\n",
       "  {'chrom': 'LinJ.35',\n",
       "   'start': 1551239,\n",
       "   'end': 1552427,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_35T0046600',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1551239, 1552427, closed='both')},\n",
       "  {'chrom': 'LinJ.35',\n",
       "   'start': 1552428,\n",
       "   'end': 1554684,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_350046700',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(1552428, 1554684, closed='both')},\n",
       "  {'chrom': 'LinJ.35',\n",
       "   'start': 1552428,\n",
       "   'end': 1554684,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_35T0046700',\n",
       "   'parent_id': 'LINF_350046700',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(1552428, 1554684, closed='both')},\n",
       "  {'chrom': 'LinJ.35',\n",
       "   'start': 1552428,\n",
       "   'end': 1552651,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_35T0046700',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(1552428, 1552651, closed='both')}],\n",
       " 'rejected_noCDS_c36.6370CO': [{'chrom': 'LinJ.36',\n",
       "   'start': 493588,\n",
       "   'end': 494799,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_360019350',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(493588, 494799, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 493588,\n",
       "   'end': 494799,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_36T0019350',\n",
       "   'parent_id': 'LINF_360019350',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(493588, 494799, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 494822,\n",
       "   'end': 494893,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_360019500',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'tRNA-Thr',\n",
       "   'interval': Interval(494822, 494893, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 494822,\n",
       "   'end': 494893,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_360019500',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': True,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(494822, 494893, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 494972,\n",
       "   'end': 495370,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_360019550',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(494972, 495370, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 494972,\n",
       "   'end': 495370,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_36T0019550',\n",
       "   'parent_id': 'LINF_360019550',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(494972, 495370, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 494975,\n",
       "   'end': 495059,\n",
       "   'strand': '+',\n",
       "   'gene_id': 'LINF_360019600',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'tRNA-Leu',\n",
       "   'interval': Interval(494975, 495059, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 494975,\n",
       "   'end': 495059,\n",
       "   'strand': '+',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_360019600',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': True,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(494975, 495059, closed='both')}],\n",
       " 'rejected_noCDS_c36.6640': [{'chrom': 'LinJ.36',\n",
       "   'start': 2322277,\n",
       "   'end': 2323858,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_360069350',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(2322277, 2323858, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2322277,\n",
       "   'end': 2323858,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_36T0069350',\n",
       "   'parent_id': 'LINF_360069350',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(2322277, 2323858, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2323859,\n",
       "   'end': 2326194,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_360069400',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(2323859, 2326194, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2323859,\n",
       "   'end': 2326194,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_36T0069400',\n",
       "   'parent_id': 'LINF_360069400',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(2323859, 2326194, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2323859,\n",
       "   'end': 2324793,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_36T0069400',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(2323859, 2324793, closed='both')}],\n",
       " 'rejected_noCDS_c36.6650CS': [{'chrom': 'LinJ.36',\n",
       "   'start': 2456799,\n",
       "   'end': 2460093,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_360072900',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'glucose_transporter',\n",
       "   'interval': Interval(2456799, 2460093, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2456799,\n",
       "   'end': 2460093,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_36T0072900',\n",
       "   'parent_id': 'LINF_360072900',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'glucose_transporter',\n",
       "   'interval': Interval(2456799, 2460093, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2456799,\n",
       "   'end': 2458220,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_36T0072900',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(2456799, 2458220, closed='both')}],\n",
       " 'rejected_noCDS_c36.6660CS': [{'chrom': 'LinJ.36',\n",
       "   'start': 2460371,\n",
       "   'end': 2463651,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_360073000',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'glucose_transporter',\n",
       "   'interval': Interval(2460371, 2463651, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2460371,\n",
       "   'end': 2463651,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_36T0073000',\n",
       "   'parent_id': 'LINF_360073000',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'glucose_transporter',\n",
       "   'interval': Interval(2460371, 2463651, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2460371,\n",
       "   'end': 2461778,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_36T0073000',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(2460371, 2461778, closed='both')}],\n",
       " 'rejected_noCDS_c36.6670CS': [{'chrom': 'LinJ.36',\n",
       "   'start': 2463918,\n",
       "   'end': 2467210,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_360073100',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'glucose_transporter',\n",
       "   'interval': Interval(2463918, 2467210, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2463918,\n",
       "   'end': 2467210,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_36T0073100',\n",
       "   'parent_id': 'LINF_360073100',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'glucose_transporter',\n",
       "   'interval': Interval(2463918, 2467210, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2463918,\n",
       "   'end': 2465337,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_36T0073100',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(2463918, 2465337, closed='both')}],\n",
       " 'rejected_noCDS_c36.6680CS': [{'chrom': 'LinJ.36',\n",
       "   'start': 2467477,\n",
       "   'end': 2470767,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_360073200',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'glucose_transporter',\n",
       "   'interval': Interval(2467477, 2470767, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2467477,\n",
       "   'end': 2470767,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_36T0073200',\n",
       "   'parent_id': 'LINF_360073200',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'glucose_transporter',\n",
       "   'interval': Interval(2467477, 2470767, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2467477,\n",
       "   'end': 2468894,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_36T0073200',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(2467477, 2468894, closed='both')}],\n",
       " 'rejected_noCDS_c36.6690CS': [{'chrom': 'LinJ.36',\n",
       "   'start': 2471045,\n",
       "   'end': 2474323,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_360073300',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'glucose_transporter',\n",
       "   'interval': Interval(2471045, 2474323, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2471045,\n",
       "   'end': 2474323,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_36T0073300',\n",
       "   'parent_id': 'LINF_360073300',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'glucose_transporter',\n",
       "   'interval': Interval(2471045, 2474323, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2471045,\n",
       "   'end': 2472450,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_36T0073300',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(2471045, 2472450, closed='both')}],\n",
       " 'rejected_noCDS_c36.6700CS': [{'chrom': 'LinJ.36',\n",
       "   'start': 2474601,\n",
       "   'end': 2477880,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_360073400',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'glucose_transporter',\n",
       "   'interval': Interval(2474601, 2477880, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2474601,\n",
       "   'end': 2477880,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_36T0073400',\n",
       "   'parent_id': 'LINF_360073400',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'glucose_transporter',\n",
       "   'interval': Interval(2474601, 2477880, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2474601,\n",
       "   'end': 2476007,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_36T0073400',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(2474601, 2476007, closed='both')}],\n",
       " 'rejected_noCDS_c36.6710CS': [{'chrom': 'LinJ.36',\n",
       "   'start': 2478158,\n",
       "   'end': 2481440,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_360073500',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'glucose_transporter',\n",
       "   'interval': Interval(2478158, 2481440, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2478158,\n",
       "   'end': 2481440,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_36T0073500',\n",
       "   'parent_id': 'LINF_360073500',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'glucose_transporter',\n",
       "   'interval': Interval(2478158, 2481440, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2478158,\n",
       "   'end': 2479567,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_36T0073500',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': True,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(2478158, 2479567, closed='both')}],\n",
       " 'rejected_noCDS_c36.6740CT': [{'chrom': 'LinJ.36',\n",
       "   'start': 2597511,\n",
       "   'end': 2600258,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_360077200',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein-l-isoaspartate_o-methyltransferase_-_putative',\n",
       "   'interval': Interval(2597511, 2600258, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2597511,\n",
       "   'end': 2600258,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_36T0077200',\n",
       "   'parent_id': 'LINF_360077200',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'protein-l-isoaspartate_o-methyltransferase_-_putative',\n",
       "   'interval': Interval(2597511, 2600258, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2599967,\n",
       "   'end': 2600258,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_36T0077200',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(2599967, 2600258, closed='both')}],\n",
       " 'rejected_noCDS_c36.6770CT': [{'chrom': 'LinJ.36',\n",
       "   'start': 2655844,\n",
       "   'end': 2658753,\n",
       "   'strand': '-',\n",
       "   'gene_id': 'LINF_360079200',\n",
       "   'transcript_id': None,\n",
       "   'parent_id': None,\n",
       "   'gene': True,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(2655844, 2658753, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2655844,\n",
       "   'end': 2658753,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': 'LINF_36T0079200',\n",
       "   'parent_id': 'LINF_360079200',\n",
       "   'gene': False,\n",
       "   'transcript': True,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': False,\n",
       "   'pseudogen': None,\n",
       "   'notes': 'hypothetical_protein_-_conserved',\n",
       "   'interval': Interval(2655844, 2658753, closed='both')},\n",
       "  {'chrom': 'LinJ.36',\n",
       "   'start': 2657303,\n",
       "   'end': 2658753,\n",
       "   'strand': '-',\n",
       "   'gene_id': None,\n",
       "   'transcript_id': None,\n",
       "   'parent_id': 'LINF_36T0079200',\n",
       "   'gene': False,\n",
       "   'transcript': False,\n",
       "   'CDS': False,\n",
       "   '3utr': False,\n",
       "   '5utr': True,\n",
       "   'pseudogen': None,\n",
       "   'notes': None,\n",
       "   'interval': Interval(2657303, 2658753, closed='both')}]}"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 122
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:09.980404Z",
     "start_time": "2024-10-03T08:02:09.968181Z"
    }
   },
   "source": [
    "# Transform it into a DataFrame\n",
    "neg_overlaps_gtf_list = []\n",
    "for key, value in neg_overlaps_gtf_dict.items():\n",
    "    for elem in value:\n",
    "        new_record = {'neg_name': key}  # Create dict of 1 element\n",
    "        new_record.update(elem)  # Update the dict with the values from elem, this way \"neg_name\" goes first\n",
    "        neg_overlaps_gtf_list.append(new_record)\n",
    "\n",
    "neg_overlaps_gtf_df = pd.DataFrame(neg_overlaps_gtf_list)"
   ],
   "outputs": [],
   "execution_count": 123
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:10.056402Z",
     "start_time": "2024-10-03T08:02:10.038447Z"
    }
   },
   "source": [
    "print(neg_overlaps_gtf_df.shape)\n",
    "print(neg_overlaps_gtf_df.dtypes)\n",
    "print(neg_overlaps_gtf_df['neg_name'].nunique())\n",
    "neg_overlaps_gtf_df.head()"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(493, 16)\n",
      "neg_name                        object\n",
      "chrom                           object\n",
      "start                            int64\n",
      "end                              int64\n",
      "strand                          object\n",
      "gene_id                         object\n",
      "transcript_id                   object\n",
      "parent_id                       object\n",
      "gene                              bool\n",
      "transcript                        bool\n",
      "CDS                               bool\n",
      "3utr                              bool\n",
      "5utr                              bool\n",
      "pseudogen                       object\n",
      "notes                           object\n",
      "interval         interval[int64, both]\n",
      "dtype: object\n",
      "137\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "                neg_name    chrom   start     end strand         gene_id  \\\n",
       "0  rejected_noCDS_c01.20  LinJ.01  114146  116224      +  LINF_010009600   \n",
       "1  rejected_noCDS_c01.20  LinJ.01  114146  116224      +            None   \n",
       "2  rejected_noCDS_c01.20  LinJ.01  114146  114423      +            None   \n",
       "3  rejected_noCDS_c01.40  LinJ.01  262069  262979      +  LINF_010013350   \n",
       "4  rejected_noCDS_c01.40  LinJ.01  262069  262979      +            None   \n",
       "\n",
       "     transcript_id        parent_id   gene  transcript    CDS   3utr   5utr  \\\n",
       "0             None             None   True       False  False  False  False   \n",
       "1  LINF_01T0009600   LINF_010009600  False        True  False  False  False   \n",
       "2             None  LINF_01T0009600  False       False  False  False   True   \n",
       "3             None             None   True       False  False  False  False   \n",
       "4  LINF_01T0013350   LINF_010013350  False        True  False  False  False   \n",
       "\n",
       "  pseudogen                notes          interval  \n",
       "0      None  nicotinamidase|PNC1  [114146, 116224]  \n",
       "1      None  nicotinamidase|PNC1  [114146, 116224]  \n",
       "2      None                 None  [114146, 114423]  \n",
       "3      None                 None  [262069, 262979]  \n",
       "4      None                 None  [262069, 262979]  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>neg_name</th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>CDS</th>\n",
       "      <th>3utr</th>\n",
       "      <th>5utr</th>\n",
       "      <th>pseudogen</th>\n",
       "      <th>notes</th>\n",
       "      <th>interval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rejected_noCDS_c01.20</td>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>114146</td>\n",
       "      <td>116224</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_010009600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>nicotinamidase|PNC1</td>\n",
       "      <td>[114146, 116224]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rejected_noCDS_c01.20</td>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>114146</td>\n",
       "      <td>116224</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0009600</td>\n",
       "      <td>LINF_010009600</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>nicotinamidase|PNC1</td>\n",
       "      <td>[114146, 116224]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rejected_noCDS_c01.20</td>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>114146</td>\n",
       "      <td>114423</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0009600</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[114146, 114423]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rejected_noCDS_c01.40</td>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>262069</td>\n",
       "      <td>262979</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_010013350</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[262069, 262979]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rejected_noCDS_c01.40</td>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>262069</td>\n",
       "      <td>262979</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0013350</td>\n",
       "      <td>LINF_010013350</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[262069, 262979]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 124
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Divide \"overlapping\" and \"overextended\" element\n",
    "In the `neg_overlaps_gtf_df` elements.There are some that touch more than one element in the GTF such as:\n",
    "* 3'UTR, CDS\n",
    "* 3'UTR transcript_1, 5'UTR transcript_2\n",
    "* etc.\n",
    "\n",
    "On the other part, there are some elements that don't touch more than one element. Instead, they touch one element and overextend to an intergenic zone"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:10.343386Z",
     "start_time": "2024-10-03T08:02:10.338438Z"
    }
   },
   "source": [
    "# Let's group the elements by \"neg_name\"\n",
    "groupy_neg_overlaps_gtf_df = neg_overlaps_gtf_df.groupby('neg_name')"
   ],
   "outputs": [],
   "execution_count": 125
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:10.888948Z",
     "start_time": "2024-10-03T08:02:10.700980Z"
    }
   },
   "source": [
    "# Create the pre-list to save the elements\n",
    "true_overlaps_gtf = []\n",
    "overextend_elements_gtf = []\n",
    "\n",
    "# Iterate over the groupy object\n",
    "for name, group in groupy_neg_overlaps_gtf_df:\n",
    "    location = group[[\"gene\", \"transcript\", \"CDS\", \"3utr\", \"5utr\"]].sum()\n",
    "    \n",
    "    # Get the elements that extend by 3'utr or 5'utr\n",
    "    if (location[\"gene\"] == 1) & (location[\"3utr\"] == 1 or location[\"5utr\"] == 1):\n",
    "        [overextend_elements_gtf.append(elem) for i, elem in group.iterrows()]\n",
    "    elif (location[\"gene\"] == 1) & (location[\"3utr\"] == 0 and location[\"5utr\"] == 0):  # Elements that overextend, but there are non-coding gentes.\n",
    "        [overextend_elements_gtf.append(elem) for i, elem in group.iterrows()]\n",
    "    else:  # The rest will be the elements that are truly overlapping more than one GTF element\n",
    "        [true_overlaps_gtf.append(elem) for i, elem in group.iterrows()]\n",
    "\n",
    "# Let's create the DataFrames\n",
    "true_overlaps_gtf_df = pd.DataFrame(true_overlaps_gtf, columns=neg_overlaps_gtf_df.columns)\n",
    "overextend_elements_gtf_df = pd.DataFrame(overextend_elements_gtf, columns=neg_overlaps_gtf_df.columns)"
   ],
   "outputs": [],
   "execution_count": 126
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:02:11.067665Z",
     "start_time": "2024-10-03T08:02:11.061291Z"
    }
   },
   "source": [
    "print(f\"True Overlaps: {true_overlaps_gtf_df.shape}\"\n",
    "      f\"\\n\\tUnique negs: {true_overlaps_gtf_df['neg_name'].nunique()}\")\n",
    "print(f\"Overextend Elements: {overextend_elements_gtf_df.shape}\"\n",
    "      f\"\\n\\tUnique negs: {overextend_elements_gtf_df['neg_name'].nunique()}\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True Overlaps: (288, 16)\n",
      "\tUnique negs: 55\n",
      "Overextend Elements: (205, 16)\n",
      "\tUnique negs: 82\n"
     ]
    }
   ],
   "execution_count": 127
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "# Analyze results\n",
    "\n",
    "* **Total NEGATIVE ELEMENTS**: 681\n",
    "* A) `neg_inside_gtf_df` ==> Data frames of the 503 negs that are inside the coordinates of a GTF element.\n",
    "* **Not completely inside GTF:** 178\n",
    "    * B) `neg_overlaps_gtf_df`==> Data frame of the 137 elements that overlap GTF elements.\n",
    "        * B.1)`true_overlaps_gtf_df`==> 55 elements\n",
    "        * B.)`overextend_elements_gtf_df`==> 82 elements\n",
    "    * C) `neg_intergenic_gtf_dict`==> Dictionary with the 41 INTERGENIC elements"
   ]
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "## Prepare data"
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "First, let's join the data frames `neg_inside_gtf_df` (503 elements) with `neg_overlaps_gtf_df` (137 elements) for a total of 640 elements"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:35:49.434203Z",
     "start_time": "2024-10-03T08:35:49.426243Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Let's join `neg_inside_gtf_df` with `neg_overlaps_gtf_df`\n",
    "neg_to_filter = pd.concat([neg_inside_gtf_df, neg_overlaps_gtf_df])\n",
    "\n",
    "# Now let's sort them by chrom and then by start\n",
    "neg_to_filter.sort_values(by=[\"chrom\", \"start\"], inplace=True)\n",
    "\n",
    "# Let's do some descriptive statistics\n",
    "print(neg_to_filter.shape)\n",
    "print(neg_to_filter['neg_name'].nunique())\n",
    "print(neg_to_filter.dtypes)\n",
    "neg_to_filter"
   ],
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'neg_inside_gtf_df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001B[0;31m---------------------------------------------------------------------------\u001B[0m",
      "\u001B[0;31mNameError\u001B[0m                                 Traceback (most recent call last)",
      "Cell \u001B[0;32mIn[21], line 2\u001B[0m\n\u001B[1;32m      1\u001B[0m \u001B[38;5;66;03m# Let's join `neg_inside_gtf_df` with `neg_overlaps_gtf_df`\u001B[39;00m\n\u001B[0;32m----> 2\u001B[0m neg_to_filter \u001B[38;5;241m=\u001B[39m pd\u001B[38;5;241m.\u001B[39mconcat([\u001B[43mneg_inside_gtf_df\u001B[49m, neg_overlaps_gtf_df])\n\u001B[1;32m      4\u001B[0m \u001B[38;5;66;03m# Now let's sort them by chrom and then by start\u001B[39;00m\n\u001B[1;32m      5\u001B[0m neg_to_filter\u001B[38;5;241m.\u001B[39msort_values(by\u001B[38;5;241m=\u001B[39m[\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mchrom\u001B[39m\u001B[38;5;124m\"\u001B[39m, \u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mstart\u001B[39m\u001B[38;5;124m\"\u001B[39m], inplace\u001B[38;5;241m=\u001B[39m\u001B[38;5;28;01mTrue\u001B[39;00m)\n",
      "\u001B[0;31mNameError\u001B[0m: name 'neg_inside_gtf_df' is not defined"
     ]
    }
   ],
   "execution_count": 21
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "## Cleaning phase"
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### Defining functions"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T08:36:08.691178Z",
     "start_time": "2024-10-03T08:36:08.686715Z"
    }
   },
   "cell_type": "code",
   "source": [
    "def search_string(data_frame, searching_string):\n",
    "    \"\"\"\n",
    "    :param data_frame: The DataFrame to search within. Must contain a 'notes' column to perform string matching.\n",
    "    :param searching_string: The string to search for within the 'notes' column of the DataFrame.\n",
    "    :return: A filtered DataFrame that contains only rows where 'notes' contains the searching_string, ignoring case.\n",
    "    \"\"\"\n",
    "    filtered_df = data_frame[data_frame['notes'].fillna('').str.contains(searching_string, case=False)]\n",
    "    print(f\"The of filtered data: {filtered_df.shape}\")\n",
    "    print(f\"The unique values in column 'neg_name': {filtered_df['neg_name'].nunique()}\")\n",
    "    return filtered_df\n",
    "\n",
    "def checking_data(data_frame):\n",
    "    \"\"\"\n",
    "    :param data_frame: pandas DataFrame that is being checked\n",
    "    :return: None\n",
    "    \"\"\"\n",
    "    print(f\"Shape of the data frame is: {data_frame.shape}\")\n",
    "    print(f\"Number of unique values in column 'neg_name': {data_frame['neg_name'].nunique()} \")\n",
    "\n",
    "def group_and_count(data_frame, group_column):\n",
    "    \"\"\"\n",
    "    :param data_frame: The input DataFrame containing the data to be grouped and counted.\n",
    "    :param group_column: The column name used to group the data_frame.\n",
    "    :return: A sorted DataFrame with unique notes from the group_column and their associated counts.\n",
    "    \"\"\"\n",
    "    grouped_df = data_frame.groupby(group_column)\n",
    "    grouped_column_counter = {}\n",
    "    loc_counter_global = {}\n",
    "    for _, group_data in grouped_df:\n",
    "        notes = group_data['notes'].unique()\n",
    "        for element in notes:\n",
    "            if element is not None:\n",
    "                if element not in grouped_column_counter:\n",
    "                    grouped_column_counter[element] = 1\n",
    "                else:\n",
    "                    grouped_column_counter[element] += 1\n",
    "                    \n",
    "        loc_data = group_data[[\"gene\", \"transcript\", \"CDS\", \"3utr\", \"5utr\"]].sum()\n",
    "        for loc_element in loc_data.index:\n",
    "            if loc_data[loc_element] > 0:\n",
    "                if loc_element not in loc_counter_global:\n",
    "                    loc_counter_global[loc_element] = 1\n",
    "                else:\n",
    "                    loc_counter_global[loc_element] += 1\n",
    "        \n",
    "    notes_counter_global_sorted = dict(sorted(grouped_column_counter.items(), key=lambda x: x[1], reverse=True))\n",
    "    notes_counter_global_sorted_df = pd.DataFrame(notes_counter_global_sorted.items(), columns=[\"notes\", \"count\"])\n",
    "    print(loc_counter_global)\n",
    "    return notes_counter_global_sorted_df"
   ],
   "outputs": [],
   "execution_count": 22
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### Check function and location\n"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T09:00:19.609525Z",
     "start_time": "2024-10-03T09:00:19.599225Z"
    }
   },
   "cell_type": "code",
   "source": [
    "checking_data(neg_to_filter)\n",
    "### Defining functions\n",
    "neg_to_filter.head()"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of the data frame is: (1851, 16)\n",
      "Number of unique values in column 'neg_name': 640 \n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "                neg_name    chrom   start     end strand         gene_id  \\\n",
       "0  rejected_noCDS_c01.10  LinJ.01   34736   37218      -  LINF_010006300   \n",
       "1  rejected_noCDS_c01.10  LinJ.01   34736   37218      -            None   \n",
       "2  rejected_noCDS_c01.10  LinJ.01   34736   36818      -            None   \n",
       "0  rejected_noCDS_c01.20  LinJ.01  114146  116224      +  LINF_010009600   \n",
       "1  rejected_noCDS_c01.20  LinJ.01  114146  116224      +            None   \n",
       "\n",
       "     transcript_id        parent_id   gene  transcript    CDS   3utr   5utr  \\\n",
       "0             None             None   True       False  False  False  False   \n",
       "1  LINF_01T0006300   LINF_010006300  False        True  False  False  False   \n",
       "2             None  LINF_01T0006300  False       False  False   True  False   \n",
       "0             None             None   True       False  False  False  False   \n",
       "1  LINF_01T0009600   LINF_010009600  False        True  False  False  False   \n",
       "\n",
       "  pseudogen                               notes          interval  \n",
       "0      None  hypothetical_protein_-_conserved__    [34736, 37218]  \n",
       "1      None  hypothetical_protein_-_conserved__    [34736, 37218]  \n",
       "2      None                                None    [34736, 36818]  \n",
       "0      None                 nicotinamidase|PNC1  [114146, 116224]  \n",
       "1      None                 nicotinamidase|PNC1  [114146, 116224]  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>neg_name</th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>CDS</th>\n",
       "      <th>3utr</th>\n",
       "      <th>5utr</th>\n",
       "      <th>pseudogen</th>\n",
       "      <th>notes</th>\n",
       "      <th>interval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rejected_noCDS_c01.10</td>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>34736</td>\n",
       "      <td>37218</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_010006300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>hypothetical_protein_-_conserved__</td>\n",
       "      <td>[34736, 37218]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rejected_noCDS_c01.10</td>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>34736</td>\n",
       "      <td>37218</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0006300</td>\n",
       "      <td>LINF_010006300</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>hypothetical_protein_-_conserved__</td>\n",
       "      <td>[34736, 37218]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rejected_noCDS_c01.10</td>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>34736</td>\n",
       "      <td>36818</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0006300</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[34736, 36818]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rejected_noCDS_c01.20</td>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>114146</td>\n",
       "      <td>116224</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_010009600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>nicotinamidase|PNC1</td>\n",
       "      <td>[114146, 116224]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rejected_noCDS_c01.20</td>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>114146</td>\n",
       "      <td>116224</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>LINF_01T0009600</td>\n",
       "      <td>LINF_010009600</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>nicotinamidase|PNC1</td>\n",
       "      <td>[114146, 116224]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 54
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T09:00:22.851854Z",
     "start_time": "2024-10-03T09:00:22.645342Z"
    }
   },
   "cell_type": "code",
   "source": "group_and_count(neg_to_filter, \"neg_name\")",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'gene': 640, 'transcript': 640, '3utr': 400, '5utr': 52, 'CDS': 4}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "                                                 notes  count\n",
       "0                     hypothetical_protein_-_conserved     88\n",
       "1              protein_of_unknown_function_-_conserved     31\n",
       "2              amastin_surface_glycoprotein_-_putative     23\n",
       "3                                 hypothetical_protein     12\n",
       "4                                  glucose_transporter      9\n",
       "..                                                 ...    ...\n",
       "209               Haloacid_dehalogenase-like_hydrolase      1\n",
       "210  protein-l-isoaspartate_o-methyltransferase_-_p...      1\n",
       "211                 2′-O-ribose_methyltransferase|MTr1      1\n",
       "212           mitochondrial_carrier_protein_-_putative      1\n",
       "213            adaptin_complex_1_subunit|beta_adaptin       1\n",
       "\n",
       "[214 rows x 2 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>notes</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hypothetical_protein</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>glucose_transporter</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>Haloacid_dehalogenase-like_hydrolase</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>protein-l-isoaspartate_o-methyltransferase_-_p...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>211</th>\n",
       "      <td>2′-O-ribose_methyltransferase|MTr1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>mitochondrial_carrier_protein_-_putative</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>adaptin_complex_1_subunit|beta_adaptin</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>214 rows × 2 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 55
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "### Checking: Hypothetical protein\n",
    "\n",
    "We'll see what to do with this kind of data. In this case probably we won't do anything because we don't know what the gene does."
   ]
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T09:00:25.609293Z",
     "start_time": "2024-10-03T09:00:25.605915Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Let's search for \"hypothetical_protein\" and add them to a new data frame\n",
    "filter_data = search_string(neg_to_filter, \"hypothetical_protein\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The of filtered data: (208, 16)\n",
      "The unique values in column 'neg_name': 104\n"
     ]
    }
   ],
   "execution_count": 56
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T09:00:27.492324Z",
     "start_time": "2024-10-03T09:00:27.489233Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Now let's take from \"neg_to_filter\" all the rows that have the same \"neg_name\" as in \"good_negatives\"\n",
    "good_negatives = neg_to_filter[neg_to_filter['neg_name'].isin(filter_data['neg_name'])]\n",
    "checking_data(good_negatives)"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of the data frame is: (349, 16)\n",
      "Number of unique values in column 'neg_name': 104 \n"
     ]
    }
   ],
   "execution_count": 57
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T09:00:29.260321Z",
     "start_time": "2024-10-03T09:00:28.998071Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Remove them now:\n",
    "neg_filtered = neg_to_filter[~neg_to_filter['neg_name'].isin(good_negatives['neg_name'])]\n",
    "# Let's check the data frame now\n",
    "checking_data(neg_filtered)\n",
    "group_and_count(neg_filtered, \"neg_name\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of the data frame is: (1502, 16)\n",
      "Number of unique values in column 'neg_name': 536 \n",
      "{'gene': 536, 'transcript': 536, '5utr': 34, '3utr': 307, 'CDS': 4}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "                                                 notes  count\n",
       "0              protein_of_unknown_function_-_conserved     31\n",
       "1              amastin_surface_glycoprotein_-_putative     23\n",
       "2                                  glucose_transporter      9\n",
       "3                                 amastin-like_protein      8\n",
       "4                      UDP-galactose_transporter|LPG5A      7\n",
       "..                                                 ...    ...\n",
       "199               Haloacid_dehalogenase-like_hydrolase      1\n",
       "200  protein-l-isoaspartate_o-methyltransferase_-_p...      1\n",
       "201                 2′-O-ribose_methyltransferase|MTr1      1\n",
       "202           mitochondrial_carrier_protein_-_putative      1\n",
       "203            adaptin_complex_1_subunit|beta_adaptin       1\n",
       "\n",
       "[204 rows x 2 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>notes</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>glucose_transporter</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>amastin-like_protein</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>UDP-galactose_transporter|LPG5A</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>Haloacid_dehalogenase-like_hydrolase</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>protein-l-isoaspartate_o-methyltransferase_-_p...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>2′-O-ribose_methyltransferase|MTr1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>mitochondrial_carrier_protein_-_putative</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203</th>\n",
       "      <td>adaptin_complex_1_subunit|beta_adaptin</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>204 rows × 2 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 58
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### Checking: Protein of unknown function"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T09:07:00.826980Z",
     "start_time": "2024-10-03T09:07:00.821250Z"
    }
   },
   "cell_type": "code",
   "source": [
    "filter_data = search_string(neg_filtered, \"protein_of_unknown_function\")\n",
    "filter_data['notes'].value_counts()"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The of filtered data: (84, 16)\n",
      "The unique values in column 'neg_name': 39\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "notes\n",
       "protein_of_unknown_function_-_conserved                                                                                                 64\n",
       "protein_of_unknown_function_(DUF3184)                                                                                                    4\n",
       "Protein_of_unknown_function_(DUF962),_putative                                                                                           2\n",
       "Protein_of_unknown_function_(DUF2946)_-_putative                                                                                         2\n",
       "Protein_of_unknown_function_(DUF775)_-_putative                                                                                          2\n",
       "Protein_of_unknown_function_-_conserved_(L1p/L10e_family)                                                                                2\n",
       "Protein_of_unknown_function_N-terminal_domain_(DUF2450)/Sec8_exocyst_complex_component_specific_domain_containing_protein_-_putative     2\n",
       "Protein_of_unknown_function_(DUF3184)_-_putative                                                                                         2\n",
       "Protein_of_unknown_function_(DUF778)_-_putative                                                                                          2\n",
       "Protein_of_unknown_function_(DUF3808)                                                                                                    2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 59
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T10:17:07.277412Z",
     "start_time": "2024-10-03T10:17:07.265212Z"
    }
   },
   "cell_type": "code",
   "source": "filter_data[filter_data[\"gene\"] == True]",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                       neg_name    chrom    start      end strand  \\\n",
       "28       rejected_noCDS_c03.190  LinJ.03   235179   238900      +   \n",
       "44       rejected_noCDS_c06.380  LinJ.06    59844    62540      -   \n",
       "47       rejected_noCDS_c06.380  LinJ.06    62983    65883      -   \n",
       "63       rejected_noCDS_c06.410  LinJ.06   402727   406088      +   \n",
       "95     rejected_noCDS_c12.1180S  LinJ.12   372226   374050      +   \n",
       "98     rejected_noCDS_c12.1180S  LinJ.12   374051   377775      +   \n",
       "101     rejected_noCDS_c12.1190  LinJ.12   374051   377775      +   \n",
       "121    rejected_noCDS_c12.1240S  LinJ.12   425243   429221      +   \n",
       "124    rejected_noCDS_c12.1240S  LinJ.12   429222   431627      +   \n",
       "206    rejected_noCDS_c12.1290T  LinJ.12   528652   535332      +   \n",
       "209    rejected_noCDS_c12.1300U  LinJ.12   528652   535332      +   \n",
       "212     rejected_noCDS_c12.1310  LinJ.12   528652   535332      +   \n",
       "271     rejected_noCDS_c15.1560  LinJ.15    77396    79634      -   \n",
       "136     rejected_noCDS_c15.1580  LinJ.15   151001   152466      +   \n",
       "139     rejected_noCDS_c15.1580  LinJ.15   152817   154532      +   \n",
       "303     rejected_noCDS_c16.1710  LinJ.16   119725   123450      -   \n",
       "314    rejected_noCDS_c16.1760Y  LinJ.16   371773   376703      +   \n",
       "317    rejected_noCDS_c16.1770Y  LinJ.16   377060   383329      +   \n",
       "325     rejected_noCDS_c16.1800  LinJ.16   603494   611700      -   \n",
       "328     rejected_noCDS_c16.1810  LinJ.16   603494   611700      -   \n",
       "383   rejected_noCDS_c19.2090AC  LinJ.19   224264   231735      +   \n",
       "181     rejected_noCDS_c20.2250  LinJ.20   134609   135924      +   \n",
       "183     rejected_noCDS_c20.2250  LinJ.20   135925   140156      +   \n",
       "186     rejected_noCDS_c20.2260  LinJ.20   245715   247334      +   \n",
       "189     rejected_noCDS_c20.2260  LinJ.20   247335   249516      +   \n",
       "197     rejected_noCDS_c21.2390  LinJ.21   488393   489994      -   \n",
       "452     rejected_noCDS_c21.2440  LinJ.21   621660   628792      -   \n",
       "494     rejected_noCDS_c23.2660  LinJ.23   491115   493877      -   \n",
       "527     rejected_noCDS_c24.2800  LinJ.24   151576   158682      +   \n",
       "578     rejected_noCDS_c26.3020  LinJ.26   168807   171144      -   \n",
       "610     rejected_noCDS_c26.3180  LinJ.26  1021429  1025061      +   \n",
       "621   rejected_noCDS_c27.3220AR  LinJ.27   396554   398111      +   \n",
       "635     rejected_noCDS_c27.3270  LinJ.27   536880   541850      -   \n",
       "761     rejected_noCDS_c30.4010  LinJ.30   948227   959563      -   \n",
       "764     rejected_noCDS_c30.4020  LinJ.30   948227   959563      -   \n",
       "767     rejected_noCDS_c30.4030  LinJ.30  1007205  1011606      -   \n",
       "785     rejected_noCDS_c30.4090  LinJ.30  1333478  1340150      +   \n",
       "850   rejected_noCDS_c31.4420BK  LinJ.31   973364   979320      -   \n",
       "922     rejected_noCDS_c32.4730  LinJ.32   965256   970325      -   \n",
       "973     rejected_noCDS_c33.4980  LinJ.33  1335137  1342813      +   \n",
       "1007  rejected_noCDS_c34.5150BT  LinJ.34   557626   562306      -   \n",
       "1068    rejected_noCDS_c34.5510  LinJ.34  1001422  1012594      -   \n",
       "438     rejected_noCDS_c35.6100  LinJ.35   332062   334222      +   \n",
       "440     rejected_noCDS_c35.6100  LinJ.35   334360   336208      +   \n",
       "1231    rejected_noCDS_c35.6150  LinJ.35   796669   802441      +   \n",
       "1266  rejected_noCDS_c35.6330CM  LinJ.35  1994576  1997700      +   \n",
       "\n",
       "             gene_id transcript_id parent_id  gene  transcript    CDS   3utr  \\\n",
       "28    LINF_030011300          None      None  True       False  False  False   \n",
       "44    LINF_060006900          None      None  True       False  False  False   \n",
       "47    LINF_060007000          None      None  True       False  False  False   \n",
       "63    LINF_060014900          None      None  True       False  False  False   \n",
       "95    LINF_120012350          None      None  True       False  False  False   \n",
       "98    LINF_120012400          None      None  True       False  False  False   \n",
       "101   LINF_120012400          None      None  True       False  False  False   \n",
       "121   LINF_120013700          None      None  True       False  False  False   \n",
       "124   LINF_120013800          None      None  True       False  False  False   \n",
       "206   LINF_120016200          None      None  True       False  False  False   \n",
       "209   LINF_120016200          None      None  True       False  False  False   \n",
       "212   LINF_120016200          None      None  True       False  False  False   \n",
       "271   LINF_150007000          None      None  True       False  False  False   \n",
       "136   LINF_150009600          None      None  True       False  False  False   \n",
       "139   LINF_150009700          None      None  True       False  False  False   \n",
       "303   LINF_160008700          None      None  True       False  False  False   \n",
       "314   LINF_160015800          None      None  True       False  False  False   \n",
       "317   LINF_160015820          None      None  True       False  False  False   \n",
       "325   LINF_160020900          None      None  True       False  False  False   \n",
       "328   LINF_160020900          None      None  True       False  False  False   \n",
       "383   LINF_190010600          None      None  True       False  False  False   \n",
       "181   LINF_200008850          None      None  True       False  False  False   \n",
       "183   LINF_200008900          None      None  True       False  False  False   \n",
       "186   LINF_200012000          None      None  True       False  False  False   \n",
       "189   LINF_200012100          None      None  True       False  False  False   \n",
       "197   LINF_210019900          None      None  True       False  False  False   \n",
       "452   LINF_210024000          None      None  True       False  False  False   \n",
       "494   LINF_230018300          None      None  True       False  False  False   \n",
       "527   LINF_240009200          None      None  True       False  False  False   \n",
       "578   LINF_260011300          None      None  True       False  False  False   \n",
       "610   LINF_260032200          None      None  True       False  False  False   \n",
       "621   LINF_270015300          None      None  True       False  False  False   \n",
       "635   LINF_270017800          None      None  True       False  False  False   \n",
       "761   LINF_300030800          None      None  True       False  False  False   \n",
       "764   LINF_300030800          None      None  True       False  False  False   \n",
       "767   LINF_300032500          None      None  True       False  False  False   \n",
       "785   LINF_300042700          None      None  True       False  False  False   \n",
       "850   LINF_310026900          None      None  True       False  False  False   \n",
       "922   LINF_320031600          None      None  True       False  False  False   \n",
       "973   LINF_330039000          None      None  True       False  False  False   \n",
       "1007  LINF_340018600          None      None  True       False  False  False   \n",
       "1068  LINF_340029200          None      None  True       False  False  False   \n",
       "438   LINF_350013950          None      None  True       False  False  False   \n",
       "440   LINF_350014000          None      None  True       False  False  False   \n",
       "1231  LINF_350026500          None      None  True       False  False  False   \n",
       "1266  LINF_350059400          None      None  True       False  False  False   \n",
       "\n",
       "       5utr pseudogen                                              notes  \\\n",
       "28    False      None            protein_of_unknown_function_-_conserved   \n",
       "44    False      None                               kinesin-like_protein   \n",
       "47    False      None            protein_of_unknown_function_-_conserved   \n",
       "63    False      None            protein_of_unknown_function_-_conserved   \n",
       "95    False      None     Protein_of_unknown_function_(DUF962),_putative   \n",
       "98    False      None            protein_of_unknown_function_-_conserved   \n",
       "101   False      None            protein_of_unknown_function_-_conserved   \n",
       "121   False      None               surface_antigen_protein_2_-_putative   \n",
       "124   False      None            protein_of_unknown_function_-_conserved   \n",
       "206   False      None            protein_of_unknown_function_-_conserved   \n",
       "209   False      None            protein_of_unknown_function_-_conserved   \n",
       "212   False      None            protein_of_unknown_function_-_conserved   \n",
       "271   False      None            protein_of_unknown_function_-_conserved   \n",
       "136   False      None            protein_of_unknown_function_-_conserved   \n",
       "139   False      None            protein_of_unknown_function_-_conserved   \n",
       "303   False      None            protein_of_unknown_function_-_conserved   \n",
       "314   False      None              protein_of_unknown_function_(DUF3184)   \n",
       "317   False      None              protein_of_unknown_function_(DUF3184)   \n",
       "325   False      None            protein_of_unknown_function_-_conserved   \n",
       "328   False      None            protein_of_unknown_function_-_conserved   \n",
       "383   False      None            protein_of_unknown_function_-_conserved   \n",
       "181   False      None                                               None   \n",
       "183   False      None   Protein_of_unknown_function_(DUF2946)_-_putative   \n",
       "186   False      None            protein_of_unknown_function_-_conserved   \n",
       "189   False      None    Protein_of_unknown_function_(DUF775)_-_putative   \n",
       "197   False      None  Protein_of_unknown_function_-_conserved_(L1p/L...   \n",
       "452   False      None  Protein_of_unknown_function_N-terminal_domain_...   \n",
       "494   False      None            protein_of_unknown_function_-_conserved   \n",
       "527   False      None   Protein_of_unknown_function_(DUF3184)_-_putative   \n",
       "578   False      None            protein_of_unknown_function_-_conserved   \n",
       "610   False      None            protein_of_unknown_function_-_conserved   \n",
       "621   False      None    Protein_of_unknown_function_(DUF778)_-_putative   \n",
       "635   False      None            protein_of_unknown_function_-_conserved   \n",
       "761   False      None            protein_of_unknown_function_-_conserved   \n",
       "764   False      None            protein_of_unknown_function_-_conserved   \n",
       "767   False      None            protein_of_unknown_function_-_conserved   \n",
       "785   False      None              Protein_of_unknown_function_(DUF3808)   \n",
       "850   False      None            protein_of_unknown_function_-_conserved   \n",
       "922   False      None            protein_of_unknown_function_-_conserved   \n",
       "973   False      None            protein_of_unknown_function_-_conserved   \n",
       "1007  False      None            protein_of_unknown_function_-_conserved   \n",
       "1068  False      None            protein_of_unknown_function_-_conserved   \n",
       "438   False      None                                               None   \n",
       "440   False      None            protein_of_unknown_function_-_conserved   \n",
       "1231  False      None            protein_of_unknown_function_-_conserved   \n",
       "1266  False      None            protein_of_unknown_function_-_conserved   \n",
       "\n",
       "                interval  \n",
       "28      [235179, 238900]  \n",
       "44        [59844, 62540]  \n",
       "47        [62983, 65883]  \n",
       "63      [402727, 406088]  \n",
       "95      [372226, 374050]  \n",
       "98      [374051, 377775]  \n",
       "101     [374051, 377775]  \n",
       "121     [425243, 429221]  \n",
       "124     [429222, 431627]  \n",
       "206     [528652, 535332]  \n",
       "209     [528652, 535332]  \n",
       "212     [528652, 535332]  \n",
       "271       [77396, 79634]  \n",
       "136     [151001, 152466]  \n",
       "139     [152817, 154532]  \n",
       "303     [119725, 123450]  \n",
       "314     [371773, 376703]  \n",
       "317     [377060, 383329]  \n",
       "325     [603494, 611700]  \n",
       "328     [603494, 611700]  \n",
       "383     [224264, 231735]  \n",
       "181     [134609, 135924]  \n",
       "183     [135925, 140156]  \n",
       "186     [245715, 247334]  \n",
       "189     [247335, 249516]  \n",
       "197     [488393, 489994]  \n",
       "452     [621660, 628792]  \n",
       "494     [491115, 493877]  \n",
       "527     [151576, 158682]  \n",
       "578     [168807, 171144]  \n",
       "610   [1021429, 1025061]  \n",
       "621     [396554, 398111]  \n",
       "635     [536880, 541850]  \n",
       "761     [948227, 959563]  \n",
       "764     [948227, 959563]  \n",
       "767   [1007205, 1011606]  \n",
       "785   [1333478, 1340150]  \n",
       "850     [973364, 979320]  \n",
       "922     [965256, 970325]  \n",
       "973   [1335137, 1342813]  \n",
       "1007    [557626, 562306]  \n",
       "1068  [1001422, 1012594]  \n",
       "438     [332062, 334222]  \n",
       "440     [334360, 336208]  \n",
       "1231    [796669, 802441]  \n",
       "1266  [1994576, 1997700]  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>neg_name</th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>CDS</th>\n",
       "      <th>3utr</th>\n",
       "      <th>5utr</th>\n",
       "      <th>pseudogen</th>\n",
       "      <th>notes</th>\n",
       "      <th>interval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>rejected_noCDS_c03.190</td>\n",
       "      <td>LinJ.03</td>\n",
       "      <td>235179</td>\n",
       "      <td>238900</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_030011300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[235179, 238900]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>rejected_noCDS_c06.380</td>\n",
       "      <td>LinJ.06</td>\n",
       "      <td>59844</td>\n",
       "      <td>62540</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_060006900</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>kinesin-like_protein</td>\n",
       "      <td>[59844, 62540]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>rejected_noCDS_c06.380</td>\n",
       "      <td>LinJ.06</td>\n",
       "      <td>62983</td>\n",
       "      <td>65883</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_060007000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[62983, 65883]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>rejected_noCDS_c06.410</td>\n",
       "      <td>LinJ.06</td>\n",
       "      <td>402727</td>\n",
       "      <td>406088</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_060014900</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[402727, 406088]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>rejected_noCDS_c12.1180S</td>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>372226</td>\n",
       "      <td>374050</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_120012350</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF962),_putative</td>\n",
       "      <td>[372226, 374050]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>rejected_noCDS_c12.1180S</td>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>374051</td>\n",
       "      <td>377775</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_120012400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[374051, 377775]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>rejected_noCDS_c12.1190</td>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>374051</td>\n",
       "      <td>377775</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_120012400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[374051, 377775]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>rejected_noCDS_c12.1240S</td>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>425243</td>\n",
       "      <td>429221</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_120013700</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>surface_antigen_protein_2_-_putative</td>\n",
       "      <td>[425243, 429221]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>rejected_noCDS_c12.1240S</td>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>429222</td>\n",
       "      <td>431627</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_120013800</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[429222, 431627]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>rejected_noCDS_c12.1290T</td>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>528652</td>\n",
       "      <td>535332</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_120016200</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[528652, 535332]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>rejected_noCDS_c12.1300U</td>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>528652</td>\n",
       "      <td>535332</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_120016200</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[528652, 535332]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>rejected_noCDS_c12.1310</td>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>528652</td>\n",
       "      <td>535332</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_120016200</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[528652, 535332]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271</th>\n",
       "      <td>rejected_noCDS_c15.1560</td>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>77396</td>\n",
       "      <td>79634</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_150007000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[77396, 79634]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>rejected_noCDS_c15.1580</td>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>151001</td>\n",
       "      <td>152466</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_150009600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[151001, 152466]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>rejected_noCDS_c15.1580</td>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>152817</td>\n",
       "      <td>154532</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_150009700</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[152817, 154532]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303</th>\n",
       "      <td>rejected_noCDS_c16.1710</td>\n",
       "      <td>LinJ.16</td>\n",
       "      <td>119725</td>\n",
       "      <td>123450</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_160008700</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[119725, 123450]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314</th>\n",
       "      <td>rejected_noCDS_c16.1760Y</td>\n",
       "      <td>LinJ.16</td>\n",
       "      <td>371773</td>\n",
       "      <td>376703</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_160015800</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_(DUF3184)</td>\n",
       "      <td>[371773, 376703]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>317</th>\n",
       "      <td>rejected_noCDS_c16.1770Y</td>\n",
       "      <td>LinJ.16</td>\n",
       "      <td>377060</td>\n",
       "      <td>383329</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_160015820</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_(DUF3184)</td>\n",
       "      <td>[377060, 383329]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>rejected_noCDS_c16.1800</td>\n",
       "      <td>LinJ.16</td>\n",
       "      <td>603494</td>\n",
       "      <td>611700</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_160020900</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[603494, 611700]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>rejected_noCDS_c16.1810</td>\n",
       "      <td>LinJ.16</td>\n",
       "      <td>603494</td>\n",
       "      <td>611700</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_160020900</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[603494, 611700]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>383</th>\n",
       "      <td>rejected_noCDS_c19.2090AC</td>\n",
       "      <td>LinJ.19</td>\n",
       "      <td>224264</td>\n",
       "      <td>231735</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_190010600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[224264, 231735]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>rejected_noCDS_c20.2250</td>\n",
       "      <td>LinJ.20</td>\n",
       "      <td>134609</td>\n",
       "      <td>135924</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_200008850</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[134609, 135924]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>rejected_noCDS_c20.2250</td>\n",
       "      <td>LinJ.20</td>\n",
       "      <td>135925</td>\n",
       "      <td>140156</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_200008900</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF2946)_-_putative</td>\n",
       "      <td>[135925, 140156]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>rejected_noCDS_c20.2260</td>\n",
       "      <td>LinJ.20</td>\n",
       "      <td>245715</td>\n",
       "      <td>247334</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_200012000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[245715, 247334]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>rejected_noCDS_c20.2260</td>\n",
       "      <td>LinJ.20</td>\n",
       "      <td>247335</td>\n",
       "      <td>249516</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_200012100</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF775)_-_putative</td>\n",
       "      <td>[247335, 249516]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>rejected_noCDS_c21.2390</td>\n",
       "      <td>LinJ.21</td>\n",
       "      <td>488393</td>\n",
       "      <td>489994</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_210019900</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_-_conserved_(L1p/L...</td>\n",
       "      <td>[488393, 489994]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>452</th>\n",
       "      <td>rejected_noCDS_c21.2440</td>\n",
       "      <td>LinJ.21</td>\n",
       "      <td>621660</td>\n",
       "      <td>628792</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_210024000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_N-terminal_domain_...</td>\n",
       "      <td>[621660, 628792]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>rejected_noCDS_c23.2660</td>\n",
       "      <td>LinJ.23</td>\n",
       "      <td>491115</td>\n",
       "      <td>493877</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_230018300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[491115, 493877]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>527</th>\n",
       "      <td>rejected_noCDS_c24.2800</td>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>151576</td>\n",
       "      <td>158682</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_240009200</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF3184)_-_putative</td>\n",
       "      <td>[151576, 158682]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>rejected_noCDS_c26.3020</td>\n",
       "      <td>LinJ.26</td>\n",
       "      <td>168807</td>\n",
       "      <td>171144</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_260011300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[168807, 171144]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>610</th>\n",
       "      <td>rejected_noCDS_c26.3180</td>\n",
       "      <td>LinJ.26</td>\n",
       "      <td>1021429</td>\n",
       "      <td>1025061</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_260032200</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[1021429, 1025061]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>621</th>\n",
       "      <td>rejected_noCDS_c27.3220AR</td>\n",
       "      <td>LinJ.27</td>\n",
       "      <td>396554</td>\n",
       "      <td>398111</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_270015300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF778)_-_putative</td>\n",
       "      <td>[396554, 398111]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>635</th>\n",
       "      <td>rejected_noCDS_c27.3270</td>\n",
       "      <td>LinJ.27</td>\n",
       "      <td>536880</td>\n",
       "      <td>541850</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_270017800</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[536880, 541850]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>761</th>\n",
       "      <td>rejected_noCDS_c30.4010</td>\n",
       "      <td>LinJ.30</td>\n",
       "      <td>948227</td>\n",
       "      <td>959563</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_300030800</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[948227, 959563]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>764</th>\n",
       "      <td>rejected_noCDS_c30.4020</td>\n",
       "      <td>LinJ.30</td>\n",
       "      <td>948227</td>\n",
       "      <td>959563</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_300030800</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[948227, 959563]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>767</th>\n",
       "      <td>rejected_noCDS_c30.4030</td>\n",
       "      <td>LinJ.30</td>\n",
       "      <td>1007205</td>\n",
       "      <td>1011606</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_300032500</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[1007205, 1011606]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>785</th>\n",
       "      <td>rejected_noCDS_c30.4090</td>\n",
       "      <td>LinJ.30</td>\n",
       "      <td>1333478</td>\n",
       "      <td>1340150</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_300042700</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein_of_unknown_function_(DUF3808)</td>\n",
       "      <td>[1333478, 1340150]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>850</th>\n",
       "      <td>rejected_noCDS_c31.4420BK</td>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>973364</td>\n",
       "      <td>979320</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_310026900</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[973364, 979320]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>922</th>\n",
       "      <td>rejected_noCDS_c32.4730</td>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>965256</td>\n",
       "      <td>970325</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_320031600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[965256, 970325]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>973</th>\n",
       "      <td>rejected_noCDS_c33.4980</td>\n",
       "      <td>LinJ.33</td>\n",
       "      <td>1335137</td>\n",
       "      <td>1342813</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_330039000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[1335137, 1342813]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1007</th>\n",
       "      <td>rejected_noCDS_c34.5150BT</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>557626</td>\n",
       "      <td>562306</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_340018600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[557626, 562306]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1068</th>\n",
       "      <td>rejected_noCDS_c34.5510</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1001422</td>\n",
       "      <td>1012594</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_340029200</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[1001422, 1012594]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>rejected_noCDS_c35.6100</td>\n",
       "      <td>LinJ.35</td>\n",
       "      <td>332062</td>\n",
       "      <td>334222</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_350013950</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[332062, 334222]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440</th>\n",
       "      <td>rejected_noCDS_c35.6100</td>\n",
       "      <td>LinJ.35</td>\n",
       "      <td>334360</td>\n",
       "      <td>336208</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_350014000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[334360, 336208]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1231</th>\n",
       "      <td>rejected_noCDS_c35.6150</td>\n",
       "      <td>LinJ.35</td>\n",
       "      <td>796669</td>\n",
       "      <td>802441</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_350026500</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[796669, 802441]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1266</th>\n",
       "      <td>rejected_noCDS_c35.6330CM</td>\n",
       "      <td>LinJ.35</td>\n",
       "      <td>1994576</td>\n",
       "      <td>1997700</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_350059400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_-_conserved</td>\n",
       "      <td>[1994576, 1997700]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 62
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T09:08:53.092339Z",
     "start_time": "2024-10-03T09:08:53.085501Z"
    }
   },
   "cell_type": "code",
   "source": [
    "filter_data[\n",
    "    (filter_data[\"gene\"] == True) & (filter_data[\"notes\"] == \"protein_of_unknown_function_(DUF3184)\")\n",
    "]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                     neg_name    chrom   start     end strand         gene_id  \\\n",
       "314  rejected_noCDS_c16.1760Y  LinJ.16  371773  376703      +  LINF_160015800   \n",
       "317  rejected_noCDS_c16.1770Y  LinJ.16  377060  383329      +  LINF_160015820   \n",
       "\n",
       "    transcript_id parent_id  gene  transcript    CDS   3utr   5utr pseudogen  \\\n",
       "314          None      None  True       False  False  False  False      None   \n",
       "317          None      None  True       False  False  False  False      None   \n",
       "\n",
       "                                     notes          interval  \n",
       "314  protein_of_unknown_function_(DUF3184)  [371773, 376703]  \n",
       "317  protein_of_unknown_function_(DUF3184)  [377060, 383329]  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>neg_name</th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>CDS</th>\n",
       "      <th>3utr</th>\n",
       "      <th>5utr</th>\n",
       "      <th>pseudogen</th>\n",
       "      <th>notes</th>\n",
       "      <th>interval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>314</th>\n",
       "      <td>rejected_noCDS_c16.1760Y</td>\n",
       "      <td>LinJ.16</td>\n",
       "      <td>371773</td>\n",
       "      <td>376703</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_160015800</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_(DUF3184)</td>\n",
       "      <td>[371773, 376703]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>317</th>\n",
       "      <td>rejected_noCDS_c16.1770Y</td>\n",
       "      <td>LinJ.16</td>\n",
       "      <td>377060</td>\n",
       "      <td>383329</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_160015820</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>protein_of_unknown_function_(DUF3184)</td>\n",
       "      <td>[377060, 383329]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 60
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T10:17:12.210247Z",
     "start_time": "2024-10-03T10:17:12.206891Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Now let's take the all the elements with the same \"neg_name\" as `filter_data`\n",
    "filter_data = neg_filtered[neg_filtered['neg_name'].isin(filter_data['neg_name'])]\n",
    "checking_data(filter_data)"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of the data frame is: (136, 16)\n",
      "Number of unique values in column 'neg_name': 39 \n"
     ]
    }
   ],
   "execution_count": 63
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T10:17:13.618063Z",
     "start_time": "2024-10-03T10:17:13.614772Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Now let's add the lines in `filter_data` to `good_negatives`\n",
    "good_negatives = pd.concat([good_negatives, filter_data])\n",
    "checking_data(good_negatives)"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of the data frame is: (485, 16)\n",
      "Number of unique values in column 'neg_name': 143 \n"
     ]
    }
   ],
   "execution_count": 64
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T10:17:14.791717Z",
     "start_time": "2024-10-03T10:17:14.787954Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Now let's remove the `filter_data` data from `neg_filtered`\n",
    "neg_filtered = neg_filtered[~neg_filtered['neg_name'].isin(filter_data['neg_name'])]\n",
    "checking_data(neg_filtered)\n"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of the data frame is: (1366, 16)\n",
      "Number of unique values in column 'neg_name': 497 \n"
     ]
    }
   ],
   "execution_count": 65
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T10:17:16.473386Z",
     "start_time": "2024-10-03T10:17:16.155854Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Check the global functions:\n",
    "group_and_count(neg_filtered, \"neg_name\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'gene': 497, 'transcript': 497, '5utr': 26, '3utr': 271, 'CDS': 4}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "                                                 notes  count\n",
       "0              amastin_surface_glycoprotein_-_putative     23\n",
       "1                                  glucose_transporter      9\n",
       "2                                 amastin-like_protein      8\n",
       "3                      UDP-galactose_transporter|LPG5A      7\n",
       "4              amastin-like_surface_protein_-_putative      6\n",
       "..                                                 ...    ...\n",
       "187               Haloacid_dehalogenase-like_hydrolase      1\n",
       "188  protein-l-isoaspartate_o-methyltransferase_-_p...      1\n",
       "189                 2′-O-ribose_methyltransferase|MTr1      1\n",
       "190           mitochondrial_carrier_protein_-_putative      1\n",
       "191            adaptin_complex_1_subunit|beta_adaptin       1\n",
       "\n",
       "[192 rows x 2 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>notes</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>glucose_transporter</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>amastin-like_protein</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>UDP-galactose_transporter|LPG5A</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>amastin-like_surface_protein_-_putative</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>Haloacid_dehalogenase-like_hydrolase</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>protein-l-isoaspartate_o-methyltransferase_-_p...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>2′-O-ribose_methyltransferase|MTr1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>mitochondrial_carrier_protein_-_putative</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>adaptin_complex_1_subunit|beta_adaptin</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>192 rows × 2 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 66
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### Checking: amastin"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T10:21:53.664532Z",
     "start_time": "2024-10-03T10:21:53.659678Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Checking contents with amastin\n",
    "filter_data = search_string(neg_filtered, \"amastin\")\n",
    "filter_data['notes'].value_counts()"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The of filtered data: (86, 16)\n",
      "The unique values in column 'neg_name': 43\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "notes\n",
       "amastin_surface_glycoprotein_-_putative    46\n",
       "amastin-like_protein                       16\n",
       "amastin-like_surface_protein_-_putative    12\n",
       "amastin_surface_glycoprotein                8\n",
       "amastin_surface_protein                     2\n",
       "amastin-like_surface_protein                2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 67
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T10:22:03.543258Z",
     "start_time": "2024-10-03T10:22:03.532960Z"
    }
   },
   "cell_type": "code",
   "source": "filter_data[filter_data[\"gene\"] == True]",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                       neg_name    chrom    start      end strand  \\\n",
       "78      rejected_noCDS_c08.770N  LinJ.08   337654   340312      +   \n",
       "726     rejected_noCDS_c30.3870  LinJ.30   274753   277962      +   \n",
       "790     rejected_noCDS_c31.4130  LinJ.31   150123   152831      -   \n",
       "328     rejected_noCDS_c31.4120  LinJ.31   150123   152831      -   \n",
       "993   rejected_noCDS_c34.5090BR  LinJ.34   423724   426392      +   \n",
       "998   rejected_noCDS_c34.5110BR  LinJ.34   432504   435210      +   \n",
       "1001  rejected_noCDS_c34.5120BR  LinJ.34   445685   448392      +   \n",
       "1004  rejected_noCDS_c34.5130BR  LinJ.34   450104   453599      +   \n",
       "381   rejected_noCDS_c34.5140BS  LinJ.34   513172   516089      -   \n",
       "1015    rejected_noCDS_c34.5190  LinJ.34   739819   743551      -   \n",
       "1018  rejected_noCDS_c34.5200BR  LinJ.34   739819   743551      -   \n",
       "386   rejected_noCDS_c34.5180BS  LinJ.34   739819   743551      -   \n",
       "399   rejected_noCDS_c34.5290BU  LinJ.34   766090   768423      -   \n",
       "422   rejected_noCDS_c34.5370BU  LinJ.34   782521   784862      -   \n",
       "1035  rejected_noCDS_c34.5380BU  LinJ.34   805687   808315      -   \n",
       "1038  rejected_noCDS_c34.5410BT  LinJ.34   814218   816590      -   \n",
       "425   rejected_noCDS_c34.5400BT  LinJ.34   814218   816590      -   \n",
       "1041  rejected_noCDS_c34.5420BT  LinJ.34   816591   819357      -   \n",
       "1044  rejected_noCDS_c34.5430BT  LinJ.34   816591   819357      -   \n",
       "1082  rejected_noCDS_c34.5560BY  LinJ.34  1240773  1245780      +   \n",
       "1085  rejected_noCDS_c34.5570BZ  LinJ.34  1240773  1245780      +   \n",
       "1088  rejected_noCDS_c34.5580CA  LinJ.34  1240773  1245780      +   \n",
       "1091  rejected_noCDS_c34.5590BY  LinJ.34  1246328  1249925      +   \n",
       "1094  rejected_noCDS_c34.5600BZ  LinJ.34  1246328  1249925      +   \n",
       "1097  rejected_noCDS_c34.5610CA  LinJ.34  1246328  1249925      +   \n",
       "1100  rejected_noCDS_c34.5620CB  LinJ.34  1246328  1249925      +   \n",
       "1105  rejected_noCDS_c34.5640BY  LinJ.34  1252201  1255794      +   \n",
       "1108  rejected_noCDS_c34.5650BZ  LinJ.34  1252201  1255794      +   \n",
       "1111  rejected_noCDS_c34.5660CA  LinJ.34  1252201  1255794      +   \n",
       "1114  rejected_noCDS_c34.5670CB  LinJ.34  1252201  1255794      +   \n",
       "1119  rejected_noCDS_c34.5690BY  LinJ.34  1258088  1261692      +   \n",
       "1122  rejected_noCDS_c34.5700BZ  LinJ.34  1258088  1261692      +   \n",
       "1125  rejected_noCDS_c34.5710CA  LinJ.34  1258088  1261692      +   \n",
       "1128  rejected_noCDS_c34.5720CB  LinJ.34  1258088  1261692      +   \n",
       "1133  rejected_noCDS_c34.5740BY  LinJ.34  1263977  1267600      +   \n",
       "1136  rejected_noCDS_c34.5750BZ  LinJ.34  1263977  1267600      +   \n",
       "1139  rejected_noCDS_c34.5760CA  LinJ.34  1263977  1267600      +   \n",
       "1142  rejected_noCDS_c34.5770CB  LinJ.34  1263977  1267600      +   \n",
       "1147  rejected_noCDS_c34.5790BZ  LinJ.34  1269875  1273542      +   \n",
       "1150  rejected_noCDS_c34.5800CA  LinJ.34  1269875  1273542      +   \n",
       "1155  rejected_noCDS_c34.5820BY  LinJ.34  1275817  1279464      +   \n",
       "1158  rejected_noCDS_c34.5830BZ  LinJ.34  1275817  1279464      +   \n",
       "1161  rejected_noCDS_c34.5840CA  LinJ.34  1275817  1279464      +   \n",
       "\n",
       "             gene_id transcript_id parent_id  gene  transcript    CDS   3utr  \\\n",
       "78    LINF_080012900          None      None  True       False  False  False   \n",
       "726   LINF_300014200          None      None  True       False  False  False   \n",
       "790   LINF_310009800          None      None  True       False  False  False   \n",
       "328   LINF_310009800          None      None  True       False  False  False   \n",
       "993   LINF_340015400          None      None  True       False  False  False   \n",
       "998   LINF_340015500          None      None  True       False  False  False   \n",
       "1001  LINF_340015800          None      None  True       False  False  False   \n",
       "1004  LINF_340015900          None      None  True       False  False  False   \n",
       "381   LINF_340017500          None      None  True       False  False  False   \n",
       "1015  LINF_340022900          None      None  True       False  False  False   \n",
       "1018  LINF_340022900          None      None  True       False  False  False   \n",
       "386   LINF_340022900          None      None  True       False  False  False   \n",
       "399   LINF_340023300          None      None  True       False  False  False   \n",
       "422   LINF_340023500          None      None  True       False  False  False   \n",
       "1035  LINF_340024200          None      None  True       False  False  False   \n",
       "1038  LINF_340024500          None      None  True       False  False  False   \n",
       "425   LINF_340024500          None      None  True       False  False  False   \n",
       "1041  LINF_340024600          None      None  True       False  False  False   \n",
       "1044  LINF_340024600          None      None  True       False  False  False   \n",
       "1082  LINF_340034300          None      None  True       False  False  False   \n",
       "1085  LINF_340034300          None      None  True       False  False  False   \n",
       "1088  LINF_340034300          None      None  True       False  False  False   \n",
       "1091  LINF_340034400          None      None  True       False  False  False   \n",
       "1094  LINF_340034400          None      None  True       False  False  False   \n",
       "1097  LINF_340034400          None      None  True       False  False  False   \n",
       "1100  LINF_340034400          None      None  True       False  False  False   \n",
       "1105  LINF_340034500          None      None  True       False  False  False   \n",
       "1108  LINF_340034500          None      None  True       False  False  False   \n",
       "1111  LINF_340034500          None      None  True       False  False  False   \n",
       "1114  LINF_340034500          None      None  True       False  False  False   \n",
       "1119  LINF_340034600          None      None  True       False  False  False   \n",
       "1122  LINF_340034600          None      None  True       False  False  False   \n",
       "1125  LINF_340034600          None      None  True       False  False  False   \n",
       "1128  LINF_340034600          None      None  True       False  False  False   \n",
       "1133  LINF_340034700          None      None  True       False  False  False   \n",
       "1136  LINF_340034700          None      None  True       False  False  False   \n",
       "1139  LINF_340034700          None      None  True       False  False  False   \n",
       "1142  LINF_340034700          None      None  True       False  False  False   \n",
       "1147  LINF_340034800          None      None  True       False  False  False   \n",
       "1150  LINF_340034800          None      None  True       False  False  False   \n",
       "1155  LINF_340034900          None      None  True       False  False  False   \n",
       "1158  LINF_340034900          None      None  True       False  False  False   \n",
       "1161  LINF_340034900          None      None  True       False  False  False   \n",
       "\n",
       "       5utr pseudogen                                    notes  \\\n",
       "78    False      None                     amastin-like_protein   \n",
       "726   False      None                  amastin_surface_protein   \n",
       "790   False      None             amastin_surface_glycoprotein   \n",
       "328   False      None             amastin_surface_glycoprotein   \n",
       "993   False      None  amastin-like_surface_protein_-_putative   \n",
       "998   False      None                     amastin-like_protein   \n",
       "1001  False      None                     amastin-like_protein   \n",
       "1004  False      None                     amastin-like_protein   \n",
       "381   False      None                     amastin-like_protein   \n",
       "1015  False      None                     amastin-like_protein   \n",
       "1018  False      None                     amastin-like_protein   \n",
       "386   False      None                     amastin-like_protein   \n",
       "399   False      None  amastin-like_surface_protein_-_putative   \n",
       "422   False      None             amastin-like_surface_protein   \n",
       "1035  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1038  False      None  amastin-like_surface_protein_-_putative   \n",
       "425   False      None  amastin-like_surface_protein_-_putative   \n",
       "1041  False      None  amastin-like_surface_protein_-_putative   \n",
       "1044  False      None  amastin-like_surface_protein_-_putative   \n",
       "1082  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1085  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1088  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1091  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1094  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1097  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1100  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1105  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1108  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1111  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1114  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1119  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1122  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1125  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1128  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1133  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1136  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1139  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1142  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1147  False      None             amastin_surface_glycoprotein   \n",
       "1150  False      None             amastin_surface_glycoprotein   \n",
       "1155  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1158  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "1161  False      None  amastin_surface_glycoprotein_-_putative   \n",
       "\n",
       "                interval  \n",
       "78      [337654, 340312]  \n",
       "726     [274753, 277962]  \n",
       "790     [150123, 152831]  \n",
       "328     [150123, 152831]  \n",
       "993     [423724, 426392]  \n",
       "998     [432504, 435210]  \n",
       "1001    [445685, 448392]  \n",
       "1004    [450104, 453599]  \n",
       "381     [513172, 516089]  \n",
       "1015    [739819, 743551]  \n",
       "1018    [739819, 743551]  \n",
       "386     [739819, 743551]  \n",
       "399     [766090, 768423]  \n",
       "422     [782521, 784862]  \n",
       "1035    [805687, 808315]  \n",
       "1038    [814218, 816590]  \n",
       "425     [814218, 816590]  \n",
       "1041    [816591, 819357]  \n",
       "1044    [816591, 819357]  \n",
       "1082  [1240773, 1245780]  \n",
       "1085  [1240773, 1245780]  \n",
       "1088  [1240773, 1245780]  \n",
       "1091  [1246328, 1249925]  \n",
       "1094  [1246328, 1249925]  \n",
       "1097  [1246328, 1249925]  \n",
       "1100  [1246328, 1249925]  \n",
       "1105  [1252201, 1255794]  \n",
       "1108  [1252201, 1255794]  \n",
       "1111  [1252201, 1255794]  \n",
       "1114  [1252201, 1255794]  \n",
       "1119  [1258088, 1261692]  \n",
       "1122  [1258088, 1261692]  \n",
       "1125  [1258088, 1261692]  \n",
       "1128  [1258088, 1261692]  \n",
       "1133  [1263977, 1267600]  \n",
       "1136  [1263977, 1267600]  \n",
       "1139  [1263977, 1267600]  \n",
       "1142  [1263977, 1267600]  \n",
       "1147  [1269875, 1273542]  \n",
       "1150  [1269875, 1273542]  \n",
       "1155  [1275817, 1279464]  \n",
       "1158  [1275817, 1279464]  \n",
       "1161  [1275817, 1279464]  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>neg_name</th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>CDS</th>\n",
       "      <th>3utr</th>\n",
       "      <th>5utr</th>\n",
       "      <th>pseudogen</th>\n",
       "      <th>notes</th>\n",
       "      <th>interval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>rejected_noCDS_c08.770N</td>\n",
       "      <td>LinJ.08</td>\n",
       "      <td>337654</td>\n",
       "      <td>340312</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_080012900</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin-like_protein</td>\n",
       "      <td>[337654, 340312]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>726</th>\n",
       "      <td>rejected_noCDS_c30.3870</td>\n",
       "      <td>LinJ.30</td>\n",
       "      <td>274753</td>\n",
       "      <td>277962</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_300014200</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_protein</td>\n",
       "      <td>[274753, 277962]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>790</th>\n",
       "      <td>rejected_noCDS_c31.4130</td>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>150123</td>\n",
       "      <td>152831</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_310009800</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein</td>\n",
       "      <td>[150123, 152831]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>rejected_noCDS_c31.4120</td>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>150123</td>\n",
       "      <td>152831</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_310009800</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein</td>\n",
       "      <td>[150123, 152831]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>993</th>\n",
       "      <td>rejected_noCDS_c34.5090BR</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>423724</td>\n",
       "      <td>426392</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340015400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin-like_surface_protein_-_putative</td>\n",
       "      <td>[423724, 426392]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>rejected_noCDS_c34.5110BR</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>432504</td>\n",
       "      <td>435210</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340015500</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin-like_protein</td>\n",
       "      <td>[432504, 435210]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1001</th>\n",
       "      <td>rejected_noCDS_c34.5120BR</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>445685</td>\n",
       "      <td>448392</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340015800</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin-like_protein</td>\n",
       "      <td>[445685, 448392]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1004</th>\n",
       "      <td>rejected_noCDS_c34.5130BR</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>450104</td>\n",
       "      <td>453599</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340015900</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin-like_protein</td>\n",
       "      <td>[450104, 453599]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>381</th>\n",
       "      <td>rejected_noCDS_c34.5140BS</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>513172</td>\n",
       "      <td>516089</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_340017500</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin-like_protein</td>\n",
       "      <td>[513172, 516089]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1015</th>\n",
       "      <td>rejected_noCDS_c34.5190</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>739819</td>\n",
       "      <td>743551</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_340022900</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin-like_protein</td>\n",
       "      <td>[739819, 743551]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1018</th>\n",
       "      <td>rejected_noCDS_c34.5200BR</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>739819</td>\n",
       "      <td>743551</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_340022900</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin-like_protein</td>\n",
       "      <td>[739819, 743551]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>386</th>\n",
       "      <td>rejected_noCDS_c34.5180BS</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>739819</td>\n",
       "      <td>743551</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_340022900</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin-like_protein</td>\n",
       "      <td>[739819, 743551]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399</th>\n",
       "      <td>rejected_noCDS_c34.5290BU</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>766090</td>\n",
       "      <td>768423</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_340023300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin-like_surface_protein_-_putative</td>\n",
       "      <td>[766090, 768423]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>rejected_noCDS_c34.5370BU</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>782521</td>\n",
       "      <td>784862</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_340023500</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin-like_surface_protein</td>\n",
       "      <td>[782521, 784862]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1035</th>\n",
       "      <td>rejected_noCDS_c34.5380BU</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>805687</td>\n",
       "      <td>808315</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_340024200</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[805687, 808315]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1038</th>\n",
       "      <td>rejected_noCDS_c34.5410BT</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>814218</td>\n",
       "      <td>816590</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_340024500</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin-like_surface_protein_-_putative</td>\n",
       "      <td>[814218, 816590]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>rejected_noCDS_c34.5400BT</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>814218</td>\n",
       "      <td>816590</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_340024500</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin-like_surface_protein_-_putative</td>\n",
       "      <td>[814218, 816590]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1041</th>\n",
       "      <td>rejected_noCDS_c34.5420BT</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>816591</td>\n",
       "      <td>819357</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_340024600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin-like_surface_protein_-_putative</td>\n",
       "      <td>[816591, 819357]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044</th>\n",
       "      <td>rejected_noCDS_c34.5430BT</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>816591</td>\n",
       "      <td>819357</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_340024600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin-like_surface_protein_-_putative</td>\n",
       "      <td>[816591, 819357]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1082</th>\n",
       "      <td>rejected_noCDS_c34.5560BY</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1240773</td>\n",
       "      <td>1245780</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1240773, 1245780]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1085</th>\n",
       "      <td>rejected_noCDS_c34.5570BZ</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1240773</td>\n",
       "      <td>1245780</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1240773, 1245780]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1088</th>\n",
       "      <td>rejected_noCDS_c34.5580CA</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1240773</td>\n",
       "      <td>1245780</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1240773, 1245780]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1091</th>\n",
       "      <td>rejected_noCDS_c34.5590BY</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1246328</td>\n",
       "      <td>1249925</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1246328, 1249925]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1094</th>\n",
       "      <td>rejected_noCDS_c34.5600BZ</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1246328</td>\n",
       "      <td>1249925</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1246328, 1249925]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>rejected_noCDS_c34.5610CA</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1246328</td>\n",
       "      <td>1249925</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1246328, 1249925]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1100</th>\n",
       "      <td>rejected_noCDS_c34.5620CB</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1246328</td>\n",
       "      <td>1249925</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1246328, 1249925]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1105</th>\n",
       "      <td>rejected_noCDS_c34.5640BY</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1252201</td>\n",
       "      <td>1255794</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034500</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1252201, 1255794]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1108</th>\n",
       "      <td>rejected_noCDS_c34.5650BZ</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1252201</td>\n",
       "      <td>1255794</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034500</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1252201, 1255794]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1111</th>\n",
       "      <td>rejected_noCDS_c34.5660CA</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1252201</td>\n",
       "      <td>1255794</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034500</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1252201, 1255794]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1114</th>\n",
       "      <td>rejected_noCDS_c34.5670CB</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1252201</td>\n",
       "      <td>1255794</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034500</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1252201, 1255794]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1119</th>\n",
       "      <td>rejected_noCDS_c34.5690BY</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1258088</td>\n",
       "      <td>1261692</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1258088, 1261692]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1122</th>\n",
       "      <td>rejected_noCDS_c34.5700BZ</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1258088</td>\n",
       "      <td>1261692</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1258088, 1261692]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1125</th>\n",
       "      <td>rejected_noCDS_c34.5710CA</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1258088</td>\n",
       "      <td>1261692</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1258088, 1261692]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128</th>\n",
       "      <td>rejected_noCDS_c34.5720CB</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1258088</td>\n",
       "      <td>1261692</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1258088, 1261692]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1133</th>\n",
       "      <td>rejected_noCDS_c34.5740BY</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1263977</td>\n",
       "      <td>1267600</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034700</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1263977, 1267600]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1136</th>\n",
       "      <td>rejected_noCDS_c34.5750BZ</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1263977</td>\n",
       "      <td>1267600</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034700</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1263977, 1267600]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1139</th>\n",
       "      <td>rejected_noCDS_c34.5760CA</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1263977</td>\n",
       "      <td>1267600</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034700</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1263977, 1267600]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1142</th>\n",
       "      <td>rejected_noCDS_c34.5770CB</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1263977</td>\n",
       "      <td>1267600</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034700</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1263977, 1267600]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1147</th>\n",
       "      <td>rejected_noCDS_c34.5790BZ</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1269875</td>\n",
       "      <td>1273542</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034800</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein</td>\n",
       "      <td>[1269875, 1273542]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1150</th>\n",
       "      <td>rejected_noCDS_c34.5800CA</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1269875</td>\n",
       "      <td>1273542</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034800</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein</td>\n",
       "      <td>[1269875, 1273542]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1155</th>\n",
       "      <td>rejected_noCDS_c34.5820BY</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1275817</td>\n",
       "      <td>1279464</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034900</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1275817, 1279464]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1158</th>\n",
       "      <td>rejected_noCDS_c34.5830BZ</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1275817</td>\n",
       "      <td>1279464</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034900</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1275817, 1279464]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1161</th>\n",
       "      <td>rejected_noCDS_c34.5840CA</td>\n",
       "      <td>LinJ.34</td>\n",
       "      <td>1275817</td>\n",
       "      <td>1279464</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_340034900</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>amastin_surface_glycoprotein_-_putative</td>\n",
       "      <td>[1275817, 1279464]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 68
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "Since whe checked that these elements al from the repetitive elements \"amastin\", let's delete them"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T10:23:14.489445Z",
     "start_time": "2024-10-03T10:23:14.486049Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Now let's take the all the elements with the same \"neg_name\" as `filter_data`\n",
    "filter_data = neg_filtered[neg_filtered['neg_name'].isin(filter_data['neg_name'])]\n",
    "checking_data(filter_data)"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of the data frame is: (141, 16)\n",
      "Number of unique values in column 'neg_name': 43 \n"
     ]
    }
   ],
   "execution_count": 70
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T10:24:39.101776Z",
     "start_time": "2024-10-03T10:24:38.944159Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Let's remove them from `neg_filtered`\n",
    "neg_filtered = neg_filtered[~neg_filtered['neg_name'].isin(filter_data['neg_name'])]\n",
    "checking_data(neg_filtered)\n",
    "group_and_count(neg_filtered, \"neg_name\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of the data frame is: (1225, 16)\n",
      "Number of unique values in column 'neg_name': 454 \n",
      "{'gene': 454, 'transcript': 454, '5utr': 25, '3utr': 229, 'CDS': 4}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "                                                 notes  count\n",
       "0                                  glucose_transporter      9\n",
       "1                      UDP-galactose_transporter|LPG5A      7\n",
       "2               Tripartite_attachment_complex_40|TAC40      6\n",
       "3         phosphoglycan_beta_1-3_galactosyltransferase      5\n",
       "4      Ketoacyl-CoA_synthase|Fatty_acid_elongase|ELO-3      5\n",
       "..                                                 ...    ...\n",
       "181               Haloacid_dehalogenase-like_hydrolase      1\n",
       "182  protein-l-isoaspartate_o-methyltransferase_-_p...      1\n",
       "183                 2′-O-ribose_methyltransferase|MTr1      1\n",
       "184           mitochondrial_carrier_protein_-_putative      1\n",
       "185            adaptin_complex_1_subunit|beta_adaptin       1\n",
       "\n",
       "[186 rows x 2 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>notes</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>glucose_transporter</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>UDP-galactose_transporter|LPG5A</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Tripartite_attachment_complex_40|TAC40</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>phosphoglycan_beta_1-3_galactosyltransferase</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Ketoacyl-CoA_synthase|Fatty_acid_elongase|ELO-3</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>Haloacid_dehalogenase-like_hydrolase</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182</th>\n",
       "      <td>protein-l-isoaspartate_o-methyltransferase_-_p...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>2′-O-ribose_methyltransferase|MTr1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>mitochondrial_carrier_protein_-_putative</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>adaptin_complex_1_subunit|beta_adaptin</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>186 rows × 2 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 72
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### Checking: glucose"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T10:30:17.695742Z",
     "start_time": "2024-10-03T10:30:17.689424Z"
    }
   },
   "cell_type": "code",
   "source": [
    "filter_data = search_string(neg_filtered, \"glucose\")\n",
    "filter_data['notes'].value_counts()"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The of filtered data: (18, 16)\n",
      "The unique values in column 'neg_name': 9\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "notes\n",
       "glucose_transporter    18\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 74
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T10:30:40.237975Z",
     "start_time": "2024-10-03T10:30:40.230880Z"
    }
   },
   "cell_type": "code",
   "source": "filter_data[filter_data[\"gene\"] == True]",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                       neg_name    chrom    start      end strand  \\\n",
       "466   rejected_noCDS_c36.6650CS  LinJ.36  2456799  2460093      -   \n",
       "469   rejected_noCDS_c36.6660CS  LinJ.36  2460371  2463651      -   \n",
       "472   rejected_noCDS_c36.6670CS  LinJ.36  2463918  2467210      -   \n",
       "475   rejected_noCDS_c36.6680CS  LinJ.36  2467477  2470767      -   \n",
       "478   rejected_noCDS_c36.6690CS  LinJ.36  2471045  2474323      -   \n",
       "481   rejected_noCDS_c36.6700CS  LinJ.36  2474601  2477880      -   \n",
       "484   rejected_noCDS_c36.6710CS  LinJ.36  2478158  2481440      -   \n",
       "1341  rejected_noCDS_c36.6720CS  LinJ.36  2481878  2486596      -   \n",
       "1344    rejected_noCDS_c36.6730  LinJ.36  2481878  2486596      -   \n",
       "\n",
       "             gene_id transcript_id parent_id  gene  transcript    CDS   3utr  \\\n",
       "466   LINF_360072900          None      None  True       False  False  False   \n",
       "469   LINF_360073000          None      None  True       False  False  False   \n",
       "472   LINF_360073100          None      None  True       False  False  False   \n",
       "475   LINF_360073200          None      None  True       False  False  False   \n",
       "478   LINF_360073300          None      None  True       False  False  False   \n",
       "481   LINF_360073400          None      None  True       False  False  False   \n",
       "484   LINF_360073500          None      None  True       False  False  False   \n",
       "1341  LINF_360073600          None      None  True       False  False  False   \n",
       "1344  LINF_360073600          None      None  True       False  False  False   \n",
       "\n",
       "       5utr pseudogen                notes            interval  \n",
       "466   False      None  glucose_transporter  [2456799, 2460093]  \n",
       "469   False      None  glucose_transporter  [2460371, 2463651]  \n",
       "472   False      None  glucose_transporter  [2463918, 2467210]  \n",
       "475   False      None  glucose_transporter  [2467477, 2470767]  \n",
       "478   False      None  glucose_transporter  [2471045, 2474323]  \n",
       "481   False      None  glucose_transporter  [2474601, 2477880]  \n",
       "484   False      None  glucose_transporter  [2478158, 2481440]  \n",
       "1341  False      None  glucose_transporter  [2481878, 2486596]  \n",
       "1344  False      None  glucose_transporter  [2481878, 2486596]  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>neg_name</th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>CDS</th>\n",
       "      <th>3utr</th>\n",
       "      <th>5utr</th>\n",
       "      <th>pseudogen</th>\n",
       "      <th>notes</th>\n",
       "      <th>interval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>466</th>\n",
       "      <td>rejected_noCDS_c36.6650CS</td>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>2456799</td>\n",
       "      <td>2460093</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_360072900</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>glucose_transporter</td>\n",
       "      <td>[2456799, 2460093]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>469</th>\n",
       "      <td>rejected_noCDS_c36.6660CS</td>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>2460371</td>\n",
       "      <td>2463651</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_360073000</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>glucose_transporter</td>\n",
       "      <td>[2460371, 2463651]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>rejected_noCDS_c36.6670CS</td>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>2463918</td>\n",
       "      <td>2467210</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_360073100</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>glucose_transporter</td>\n",
       "      <td>[2463918, 2467210]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475</th>\n",
       "      <td>rejected_noCDS_c36.6680CS</td>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>2467477</td>\n",
       "      <td>2470767</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_360073200</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>glucose_transporter</td>\n",
       "      <td>[2467477, 2470767]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>478</th>\n",
       "      <td>rejected_noCDS_c36.6690CS</td>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>2471045</td>\n",
       "      <td>2474323</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_360073300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>glucose_transporter</td>\n",
       "      <td>[2471045, 2474323]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481</th>\n",
       "      <td>rejected_noCDS_c36.6700CS</td>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>2474601</td>\n",
       "      <td>2477880</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_360073400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>glucose_transporter</td>\n",
       "      <td>[2474601, 2477880]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>484</th>\n",
       "      <td>rejected_noCDS_c36.6710CS</td>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>2478158</td>\n",
       "      <td>2481440</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_360073500</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>glucose_transporter</td>\n",
       "      <td>[2478158, 2481440]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1341</th>\n",
       "      <td>rejected_noCDS_c36.6720CS</td>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>2481878</td>\n",
       "      <td>2486596</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_360073600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>glucose_transporter</td>\n",
       "      <td>[2481878, 2486596]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1344</th>\n",
       "      <td>rejected_noCDS_c36.6730</td>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>2481878</td>\n",
       "      <td>2486596</td>\n",
       "      <td>-</td>\n",
       "      <td>LINF_360073600</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>glucose_transporter</td>\n",
       "      <td>[2481878, 2486596]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 75
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "Here is all the CS family and 6730. Let's remove them"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T10:45:12.432290Z",
     "start_time": "2024-10-03T10:45:12.429509Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Take all elements with those names\n",
    "filter_data = neg_filtered[neg_filtered['neg_name'].isin(filter_data['neg_name'])]\n",
    "checking_data(filter_data)"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of the data frame is: (27, 16)\n",
      "Number of unique values in column 'neg_name': 9 \n"
     ]
    }
   ],
   "execution_count": 76
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T10:46:19.573898Z",
     "start_time": "2024-10-03T10:46:19.384453Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Let's remove them from `neg_filtered`\n",
    "neg_filtered = neg_filtered[~neg_filtered['neg_name'].isin(filter_data['neg_name'])]\n",
    "checking_data(neg_filtered)\n",
    "group_and_count(neg_filtered, \"neg_name\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of the data frame is: (1198, 16)\n",
      "Number of unique values in column 'neg_name': 445 \n",
      "{'gene': 445, 'transcript': 445, '5utr': 25, '3utr': 220, 'CDS': 4}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "                                                 notes  count\n",
       "0                      UDP-galactose_transporter|LPG5A      7\n",
       "1               Tripartite_attachment_complex_40|TAC40      6\n",
       "2         phosphoglycan_beta_1-3_galactosyltransferase      5\n",
       "3      Ketoacyl-CoA_synthase|Fatty_acid_elongase|ELO-3      5\n",
       "4                                 alpha/beta_hydrolase      3\n",
       "..                                                 ...    ...\n",
       "180               Haloacid_dehalogenase-like_hydrolase      1\n",
       "181  protein-l-isoaspartate_o-methyltransferase_-_p...      1\n",
       "182                 2′-O-ribose_methyltransferase|MTr1      1\n",
       "183           mitochondrial_carrier_protein_-_putative      1\n",
       "184            adaptin_complex_1_subunit|beta_adaptin       1\n",
       "\n",
       "[185 rows x 2 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>notes</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>UDP-galactose_transporter|LPG5A</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tripartite_attachment_complex_40|TAC40</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>phosphoglycan_beta_1-3_galactosyltransferase</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ketoacyl-CoA_synthase|Fatty_acid_elongase|ELO-3</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>alpha/beta_hydrolase</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>Haloacid_dehalogenase-like_hydrolase</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>protein-l-isoaspartate_o-methyltransferase_-_p...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182</th>\n",
       "      <td>2′-O-ribose_methyltransferase|MTr1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>mitochondrial_carrier_protein_-_putative</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>adaptin_complex_1_subunit|beta_adaptin</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>185 rows × 2 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 77
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### Checking: galactose"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T10:50:21.174581Z",
     "start_time": "2024-10-03T10:50:21.169851Z"
    }
   },
   "cell_type": "code",
   "source": [
    "filter_data = search_string(neg_filtered, \"galactose\")\n",
    "filter_data['notes'].value_counts()"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The of filtered data: (16, 16)\n",
      "The unique values in column 'neg_name': 7\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "notes\n",
       "UDP-galactose_transporter|LPG5A    16\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 79
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T10:52:42.744338Z",
     "start_time": "2024-10-03T10:52:42.736580Z"
    }
   },
   "cell_type": "code",
   "source": "filter_data[filter_data[\"gene\"] == True]",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                      neg_name    chrom   start     end strand  \\\n",
       "506    rejected_noCDS_c24.2720  LinJ.24  110205  117720      +   \n",
       "509    rejected_noCDS_c24.2730  LinJ.24  110205  117720      +   \n",
       "512    rejected_noCDS_c24.2740  LinJ.24  110205  117720      +   \n",
       "515    rejected_noCDS_c24.2750  LinJ.24  110205  117720      +   \n",
       "518    rejected_noCDS_c24.2760  LinJ.24  110205  117720      +   \n",
       "221  rejected_noCDS_c24.2710AJ  LinJ.24  110205  117720      +   \n",
       "224  rejected_noCDS_c24.2770AJ  LinJ.24  110205  117720      +   \n",
       "227  rejected_noCDS_c24.2770AJ  LinJ.24  117721  122337      +   \n",
       "\n",
       "            gene_id transcript_id parent_id  gene  transcript    CDS   3utr  \\\n",
       "506  LINF_240008300          None      None  True       False  False  False   \n",
       "509  LINF_240008300          None      None  True       False  False  False   \n",
       "512  LINF_240008300          None      None  True       False  False  False   \n",
       "515  LINF_240008300          None      None  True       False  False  False   \n",
       "518  LINF_240008300          None      None  True       False  False  False   \n",
       "221  LINF_240008300          None      None  True       False  False  False   \n",
       "224  LINF_240008300          None      None  True       False  False  False   \n",
       "227  LINF_240008400          None      None  True       False  False  False   \n",
       "\n",
       "      5utr pseudogen                            notes          interval  \n",
       "506  False      None  UDP-galactose_transporter|LPG5A  [110205, 117720]  \n",
       "509  False      None  UDP-galactose_transporter|LPG5A  [110205, 117720]  \n",
       "512  False      None  UDP-galactose_transporter|LPG5A  [110205, 117720]  \n",
       "515  False      None  UDP-galactose_transporter|LPG5A  [110205, 117720]  \n",
       "518  False      None  UDP-galactose_transporter|LPG5A  [110205, 117720]  \n",
       "221  False      None  UDP-galactose_transporter|LPG5A  [110205, 117720]  \n",
       "224  False      None  UDP-galactose_transporter|LPG5A  [110205, 117720]  \n",
       "227  False      None  UDP-galactose_transporter|LPG5A  [117721, 122337]  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>neg_name</th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>parent_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>CDS</th>\n",
       "      <th>3utr</th>\n",
       "      <th>5utr</th>\n",
       "      <th>pseudogen</th>\n",
       "      <th>notes</th>\n",
       "      <th>interval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>506</th>\n",
       "      <td>rejected_noCDS_c24.2720</td>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>110205</td>\n",
       "      <td>117720</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_240008300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>UDP-galactose_transporter|LPG5A</td>\n",
       "      <td>[110205, 117720]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>509</th>\n",
       "      <td>rejected_noCDS_c24.2730</td>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>110205</td>\n",
       "      <td>117720</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_240008300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>UDP-galactose_transporter|LPG5A</td>\n",
       "      <td>[110205, 117720]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>512</th>\n",
       "      <td>rejected_noCDS_c24.2740</td>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>110205</td>\n",
       "      <td>117720</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_240008300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>UDP-galactose_transporter|LPG5A</td>\n",
       "      <td>[110205, 117720]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>515</th>\n",
       "      <td>rejected_noCDS_c24.2750</td>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>110205</td>\n",
       "      <td>117720</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_240008300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>UDP-galactose_transporter|LPG5A</td>\n",
       "      <td>[110205, 117720]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>518</th>\n",
       "      <td>rejected_noCDS_c24.2760</td>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>110205</td>\n",
       "      <td>117720</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_240008300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>UDP-galactose_transporter|LPG5A</td>\n",
       "      <td>[110205, 117720]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>rejected_noCDS_c24.2710AJ</td>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>110205</td>\n",
       "      <td>117720</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_240008300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>UDP-galactose_transporter|LPG5A</td>\n",
       "      <td>[110205, 117720]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>rejected_noCDS_c24.2770AJ</td>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>110205</td>\n",
       "      <td>117720</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_240008300</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>UDP-galactose_transporter|LPG5A</td>\n",
       "      <td>[110205, 117720]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>rejected_noCDS_c24.2770AJ</td>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>117721</td>\n",
       "      <td>122337</td>\n",
       "      <td>+</td>\n",
       "      <td>LINF_240008400</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>UDP-galactose_transporter|LPG5A</td>\n",
       "      <td>[117721, 122337]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 80
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-03T11:29:35.667695Z",
     "start_time": "2024-10-03T11:29:35.663802Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Check all rows with the \"neg_names\"\n",
    "filter_data = neg_filtered[neg_filtered['neg_name'].isin(filter_data['neg_name'])]\n",
    "checking_data(filter_data)"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of the data frame is: (26, 16)\n",
      "Number of unique values in column 'neg_name': 7 \n"
     ]
    }
   ],
   "execution_count": 81
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {
    "height": "300px",
    "width": "603px"
   },
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "312.188px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
