{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as pd\n",
    "import pandas as pd\n",
    "import subprocess\n",
    "import os\n",
    "\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get path directions\n",
    "data_path = \"../BEDOPS_join_strands/filtered_data/positives_testing_elements.csv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>173</td>\n",
       "      <td>1</td>\n",
       "      <td>173</td>\n",
       "      <td>plus</td>\n",
       "      <td>ACACCAGTACACCAGTACACCAGTACACCAGTACACCAGTACACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>699</td>\n",
       "      <td>24093</td>\n",
       "      <td>24791</td>\n",
       "      <td>plus</td>\n",
       "      <td>GGGGGAGGCGGGGGAGGCGGGGGGCACGCACCTCCATGCGTGGCAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>845</td>\n",
       "      <td>35316</td>\n",
       "      <td>36160</td>\n",
       "      <td>plus</td>\n",
       "      <td>GAGGTGGAGGCCGCTCTGCCCCCCCCGCCGCCGAGTGCTGCAGGCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>892</td>\n",
       "      <td>39698</td>\n",
       "      <td>40589</td>\n",
       "      <td>plus</td>\n",
       "      <td>CTCACCCTCATCCCACCCCTCCTCGTCCATCGACGGGAGTGGGCGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>888</td>\n",
       "      <td>54885</td>\n",
       "      <td>55772</td>\n",
       "      <td>plus</td>\n",
       "      <td>TGTTTGGTCTTCCGCGTGTCCGTTTTCGCTGCCGCACACTGCGAGG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         0    1      2      3     4  \\\n",
       "0  LinJ.01  173      1    173  plus   \n",
       "1  LinJ.01  699  24093  24791  plus   \n",
       "2  LinJ.01  845  35316  36160  plus   \n",
       "3  LinJ.01  892  39698  40589  plus   \n",
       "4  LinJ.01  888  54885  55772  plus   \n",
       "\n",
       "                                                   5  \n",
       "0  ACACCAGTACACCAGTACACCAGTACACCAGTACACCAGTACACCA...  \n",
       "1  GGGGGAGGCGGGGGAGGCGGGGGGCACGCACCTCCATGCGTGGCAT...  \n",
       "2  GAGGTGGAGGCCGCTCTGCCCCCCCCGCCGCCGAGTGCTGCAGGCA...  \n",
       "3  CTCACCCTCATCCCACCCCTCCTCGTCCATCGACGGGAGTGGGCGG...  \n",
       "4  TGTTTGGTCTTCCGCGTGTCCGTTTTCGCTGCCGCACACTGCGAGG...  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get data\n",
    "data_df = pd.read_csv(data_path, sep=\",\", header=None)\n",
    "data_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>173</td>\n",
       "      <td>1</td>\n",
       "      <td>173</td>\n",
       "      <td>plus</td>\n",
       "      <td>ACACCAGTACACCAGTACACCAGTACACCAGTACACCAGTACACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>699</td>\n",
       "      <td>24093</td>\n",
       "      <td>24791</td>\n",
       "      <td>plus</td>\n",
       "      <td>GGGGGAGGCGGGGGAGGCGGGGGGCACGCACCTCCATGCGTGGCAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>845</td>\n",
       "      <td>35316</td>\n",
       "      <td>36160</td>\n",
       "      <td>plus</td>\n",
       "      <td>GAGGTGGAGGCCGCTCTGCCCCCCCCGCCGCCGAGTGCTGCAGGCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>892</td>\n",
       "      <td>39698</td>\n",
       "      <td>40589</td>\n",
       "      <td>plus</td>\n",
       "      <td>CTCACCCTCATCCCACCCCTCCTCGTCCATCGACGGGAGTGGGCGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>888</td>\n",
       "      <td>54885</td>\n",
       "      <td>55772</td>\n",
       "      <td>plus</td>\n",
       "      <td>TGTTTGGTCTTCCGCGTGTCCGTTTTCGCTGCCGCACACTGCGAGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>868</td>\n",
       "      <td>75659</td>\n",
       "      <td>76526</td>\n",
       "      <td>plus</td>\n",
       "      <td>GAGCGCCGCTGAGCAGGCAAGCGAGGCAACCTACGAAAACATGGCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>306</td>\n",
       "      <td>86582</td>\n",
       "      <td>86887</td>\n",
       "      <td>plus</td>\n",
       "      <td>ATTTGGGCACATGCAGCTGGCGCTGCTGGCCCGTCGCGTGCGCCTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>803</td>\n",
       "      <td>130118</td>\n",
       "      <td>130920</td>\n",
       "      <td>plus</td>\n",
       "      <td>CCTCTGCCGCGCAGCGCGAAGTCATCCCATCACGCCAACGCCGACT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>609</td>\n",
       "      <td>137465</td>\n",
       "      <td>138073</td>\n",
       "      <td>plus</td>\n",
       "      <td>GGGGTGGGGCGGCGGCGCACAGACACACACACACACACACACACAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>681</td>\n",
       "      <td>145352</td>\n",
       "      <td>146032</td>\n",
       "      <td>plus</td>\n",
       "      <td>GGGGTCGGGCACGCGCTGCAGGCCCTTGTACGGACTGGGCAAGTGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>424</td>\n",
       "      <td>205581</td>\n",
       "      <td>206004</td>\n",
       "      <td>plus</td>\n",
       "      <td>CACACACACACACGCACACACCTCCGTGCGTGGGGGGGGGGGTACC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>526</td>\n",
       "      <td>270740</td>\n",
       "      <td>271265</td>\n",
       "      <td>plus</td>\n",
       "      <td>CGATATGCCTCAGCACCTTCACCATCTCCGCCCCGTTCACGGCCCA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          0    1       2       3     4  \\\n",
       "0   LinJ.01  173       1     173  plus   \n",
       "1   LinJ.01  699   24093   24791  plus   \n",
       "2   LinJ.01  845   35316   36160  plus   \n",
       "3   LinJ.01  892   39698   40589  plus   \n",
       "4   LinJ.01  888   54885   55772  plus   \n",
       "5   LinJ.01  868   75659   76526  plus   \n",
       "6   LinJ.01  306   86582   86887  plus   \n",
       "7   LinJ.01  803  130118  130920  plus   \n",
       "8   LinJ.01  609  137465  138073  plus   \n",
       "9   LinJ.01  681  145352  146032  plus   \n",
       "10  LinJ.01  424  205581  206004  plus   \n",
       "11  LinJ.01  526  270740  271265  plus   \n",
       "\n",
       "                                                    5  \n",
       "0   ACACCAGTACACCAGTACACCAGTACACCAGTACACCAGTACACCA...  \n",
       "1   GGGGGAGGCGGGGGAGGCGGGGGGCACGCACCTCCATGCGTGGCAT...  \n",
       "2   GAGGTGGAGGCCGCTCTGCCCCCCCCGCCGCCGAGTGCTGCAGGCA...  \n",
       "3   CTCACCCTCATCCCACCCCTCCTCGTCCATCGACGGGAGTGGGCGG...  \n",
       "4   TGTTTGGTCTTCCGCGTGTCCGTTTTCGCTGCCGCACACTGCGAGG...  \n",
       "5   GAGCGCCGCTGAGCAGGCAAGCGAGGCAACCTACGAAAACATGGCA...  \n",
       "6   ATTTGGGCACATGCAGCTGGCGCTGCTGGCCCGTCGCGTGCGCCTC...  \n",
       "7   CCTCTGCCGCGCAGCGCGAAGTCATCCCATCACGCCAACGCCGACT...  \n",
       "8   GGGGTGGGGCGGCGGCGCACAGACACACACACACACACACACACAC...  \n",
       "9   GGGGTCGGGCACGCGCTGCAGGCCCTTGTACGGACTGGGCAAGTGG...  \n",
       "10  CACACACACACACGCACACACCTCCGTGCGTGGGGGGGGGGGTACC...  \n",
       "11  CGATATGCCTCAGCACCTTCACCATCTCCGCCCCGTTCACGGCCCA...  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Extract the first chromosome 1 from our elements\n",
    "data_df_chr1 = data_df[data_df[0] == \"LinJ.01\"]\n",
    "data_df_chr1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get fasta f# Define fasta_creator file from the first chromosome\n",
    "def fasta_creator(data, fasta_output_path):\n",
    "    matrix = []\n",
    "    for index, sequence in data.iterrows():\n",
    "        index += 1\n",
    "        rec = SeqRecord(Seq(sequence[5]),  # In the 5 position is the seq\n",
    "                        id=\"Seq_\" + str(index),\n",
    "                        description=\"Leishmania infantum\"\n",
    "                        )\n",
    "        matrix.append(rec)\n",
    "    SeqIO.write(matrix, fasta_output_path, \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create fasta from mine elements of chr1\n",
    "fasta_creator(data_df_chr1, \"./ingi_vs_mine/mine_chr1.fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Prepare functions\n",
    "# Let's define the BLASTn dictionary function\n",
    "def blastn_dic(path_input):\n",
    "    os.system(\"makeblastdb -in \" + path_input + \" -dbtype nucl -parse_seqids\")\n",
    "    print(\"\\nBlast Dictionary created in\", path_input)\n",
    "\n",
    "#And the blaster\n",
    "def blastn_blaster(query_path, dict_path, perc_identity):\n",
    "    cmd = \"blastn -word_size 11 -query \" \\\n",
    "        + query_path + \" -db \" \\\n",
    "        + dict_path \\\n",
    "        + \" -perc_identity \" + str(perc_identity) \\\n",
    "        + \" -outfmt 10\"\n",
    "    data = subprocess.check_output(cmd, shell=True, universal_newlines=True)  # Important the E value\n",
    "    return data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 04/17/2024 16:19:04\n",
      "New DB name:   /home/rfpacheco/Desktop/Projects/Testing_Leishmania_project/Compare_OrigPositvSIDER_vs_LastFiltered/ingi_vs_mine/dict/mine_chr1.fasta\n",
      "New DB title:  ./ingi_vs_mine/dict/mine_chr1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Desktop/Projects/Testing_Leishmania_project/Compare_OrigPositvSIDER_vs_LastFiltered/ingi_vs_mine/dict/mine_chr1.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 12 sequences in 0.0764821 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./ingi_vs_mine/dict/mine_chr1.fasta\n"
     ]
    }
   ],
   "source": [
    "# Create a dictionary\n",
    "blastn_dic(\"./ingi_vs_mine/dict/mine_chr1.fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(4, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ingi_1</td>\n",
       "      <td>Seq_6</td>\n",
       "      <td>85.185</td>\n",
       "      <td>27</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>18</td>\n",
       "      <td>42</td>\n",
       "      <td>265</td>\n",
       "      <td>291</td>\n",
       "      <td>0.004</td>\n",
       "      <td>27.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ingi_1</td>\n",
       "      <td>Seq_5</td>\n",
       "      <td>88.235</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>48</td>\n",
       "      <td>619</td>\n",
       "      <td>603</td>\n",
       "      <td>0.18</td>\n",
       "      <td>21.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ingi_1</td>\n",
       "      <td>Seq_3</td>\n",
       "      <td>88.235</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>48</td>\n",
       "      <td>577</td>\n",
       "      <td>561</td>\n",
       "      <td>0.18</td>\n",
       "      <td>21.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ingi_1</td>\n",
       "      <td>Seq_2</td>\n",
       "      <td>100.000</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>42</td>\n",
       "      <td>48</td>\n",
       "      <td>58</td>\n",
       "      <td>0.18</td>\n",
       "      <td>21.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       0      1        2   3  4  5   6   7    8    9      10    11\n",
       "0  ingi_1  Seq_6   85.185  27  2  1  18  42  265  291  0.004  27.0\n",
       "1  ingi_1  Seq_5   88.235  17  2  0  32  48  619  603   0.18  21.4\n",
       "2  ingi_1  Seq_3   88.235  17  2  0  32  48  577  561   0.18  21.4\n",
       "3  ingi_1  Seq_2  100.000  11  0  0  32  42   48   58   0.18  21.4"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Make blastn with ingi vs mine in chr1\n",
    "blastn_chr1_mine = blastn_blaster(\"./ingi.fasta\", \"./ingi_vs_mine/dict/mine_chr1.fasta\", 60)\n",
    "blastn_chr1_mine = pd.DataFrame([x.split(\",\") for x in blastn_chr1_mine.split(\"\\n\") if x])\n",
    "print(blastn_chr1_mine.shape)\n",
    "blastn_chr1_mine.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "Seq_6    1\n",
      "Seq_5    1\n",
      "Seq_3    1\n",
      "Seq_2    1\n",
      "Name: count, dtype: int64\n",
      "(4,)\n"
     ]
    }
   ],
   "source": [
    "print(blastn_chr1_mine[1].value_counts())\n",
    "print(blastn_chr1_mine[1].value_counts().shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 04/17/2024 16:25:34\n",
      "New DB name:   /home/rfpacheco/Desktop/Projects/Testing_Leishmania_project/Compare_OrigPositvSIDER_vs_LastFiltered/ingi_vs_TP/dict/SIDER2_Chr1_L_Infantum.fasta\n",
      "New DB title:  ./ingi_vs_TP/dict/SIDER2_Chr1_L_Infantum.fasta\n",
      "Sequence type: Nucleotide\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 7 sequences in 0.00102806 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./ingi_vs_TP/dict/SIDER2_Chr1_L_Infantum.fasta\n"
     ]
    }
   ],
   "source": [
    "# NOW LET'S MAKE THE SAME WITH THE TRUE POSITIVES\n",
    "blastn_dic(\"./ingi_vs_TP/dict/SIDER2_Chr1_L_Infantum.fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(3, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ingi_1</td>\n",
       "      <td>1-55910r</td>\n",
       "      <td>100.000</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>42</td>\n",
       "      <td>31</td>\n",
       "      <td>41</td>\n",
       "      <td>0.083</td>\n",
       "      <td>21.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ingi_1</td>\n",
       "      <td>1-36305r</td>\n",
       "      <td>100.000</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>52</td>\n",
       "      <td>0.083</td>\n",
       "      <td>21.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ingi_1</td>\n",
       "      <td>1-24471d</td>\n",
       "      <td>100.000</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>52</td>\n",
       "      <td>0.083</td>\n",
       "      <td>21.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       0         1        2   3  4  5   6   7   8   9      10    11\n",
       "0  ingi_1  1-55910r  100.000  11  0  0  32  42  31  41  0.083  21.4\n",
       "1  ingi_1  1-36305r  100.000  11  0  0  32  42  42  52  0.083  21.4\n",
       "2  ingi_1  1-24471d  100.000  11  0  0  32  42  42  52  0.083  21.4"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "blastn_chr1_TP = blastn_blaster(\"./ingi.fasta\", \"./ingi_vs_TP/dict/SIDER2_Chr1_L_Infantum.fasta\", 60)\n",
    "blastn_chr1_TP = pd.DataFrame([x.split(\",\") for x in blastn_chr1_TP.split(\"\\n\") if x])\n",
    "print(blastn_chr1_TP.shape)\n",
    "blastn_chr1_TP.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "1-55910r    1\n",
      "1-36305r    1\n",
      "1-24471d    1\n",
      "Name: count, dtype: int64\n",
      "(3,)\n"
     ]
    }
   ],
   "source": [
    "print(blastn_chr1_TP[1].value_counts())\n",
    "print(blastn_chr1_TP[1].value_counts().shape)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
