{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Author:** Ronny F. Pacheco\n",
    "**Date:** Jul 2024  \n",
    "**Copyright:** © 2024 Ronny Pacheco \n",
    "**License:** MIT License\n",
    "\n",
    "---\n",
    "\n",
    "MIT License\n",
    "\n",
    "Copyright (c) 2024 Ronny Pacheco\n",
    "\n",
    "Permission is hereby granted, free of charge, to any person obtaining a copy\n",
    "of this software and associated documentation files (the \"Software\"), to deal\n",
    "in the Software without restriction, including without limitation the rights\n",
    "to use, copy, modify, merge, publish, distribute, sublicense, and/or sell\n",
    "copies of the Software, and to permit persons to whom the Software is\n",
    "furnished to do so, subject to the following conditions:\n",
    "\n",
    "The above copyright notice and this permission notice shall be included in all\n",
    "copies or substantial portions of the Software.\n",
    "\n",
    "THE SOFTWARE IS PROVIDED \"AS IS\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR\n",
    "IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,\n",
    "FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE\n",
    "AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER\n",
    "LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,\n",
    "OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE\n",
    "SOFTWARE.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": [
     "hide"
    ]
   },
   "source": [
    "# Needed modules"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T09:57:44.648977Z",
     "start_time": "2024-09-27T09:57:44.647039Z"
    }
   },
   "source": [
    "# Import needed modules\n",
    "import pandas as pd  # type: ignore\n",
    "import re  # type: ignore\n",
    "import os\n",
    "import pickle\n",
    "import json\n"
   ],
   "outputs": [],
   "execution_count": 99
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:11:30.983455Z",
     "start_time": "2024-09-27T08:11:30.981535Z"
    }
   },
   "source": [
    "## https://kioku-space.com/en/jupyter-skip-execution/\n",
    "from IPython.core.magic import register_cell_magic # type: ignore\n",
    "\n",
    "@register_cell_magic\n",
    "def skip(line, cell):\n",
    "    print(\"Skipping cell\")\n",
    "    if line and cell:\n",
    "        pass\n",
    "    return"
   ],
   "outputs": [],
   "execution_count": 14
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "# Pickle save"
  },
  {
   "cell_type": "code",
   "metadata": {
    "tags": [
     "hide"
    ],
    "ExecuteTime": {
     "end_time": "2024-09-27T08:11:32.906684Z",
     "start_time": "2024-09-27T08:11:32.904061Z"
    }
   },
   "source": [
    "%%skip\n",
    "# =============================================================================\n",
    "# main function\n",
    "# =============================================================================\n",
    "def data_save_load(option):\n",
    "    \"\"\"\n",
    "    This function is used to save or load data for the Jupyter notebook.\n",
    "    \n",
    "    Parameters:\n",
    "    option (str): Either 'save' or 'load' to save or load variables.\n",
    "    \n",
    "    Returns:\n",
    "    dict: Dictionary of loaded variables (only when option is 'load').\n",
    "    \"\"\"\n",
    "    path_folder = \"ipynb_db\"  # Folder to save variables\n",
    "    os.makedirs(path_folder, exist_ok=True)  # Create folder if not exist\n",
    "    path_pickle = os.path.join(path_folder, f\"{os.path.basename(globals()['__session__'])}_variables.pkl\") # Path to save the variables\n",
    "\n",
    "    if option == \"save\":\n",
    "        with open(path_pickle, \"wb\") as file:\n",
    "            variables_dict = {}  # Dictionary to save the variables\n",
    "            pickle.dump(variables_dict, file)\n",
    "    elif option == \"load\":\n",
    "        with open(path_pickle, \"rb\") as file:\n",
    "            variables = pickle.load(file)\n",
    "        # Now load the variables\n",
    "        for var_key, var_value in variables.items():\n",
    "            variables[var_key] = var_value\n",
    "\n",
    "# =============================================================================\n",
    "# Call the function\n",
    "# =============================================================================\n",
    "data_save_load(option=\"save\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Skipping cell\n"
     ]
    }
   ],
   "execution_count": 15
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": ""
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "# Load_data"
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:14:38.718897Z",
     "start_time": "2024-09-27T08:14:36.784065Z"
    }
   },
   "source": [
    "# Getting path\n",
    "path_file = \"./data/20240703111001_LINF-Tabla_maestra_v3-2024.xlsx\"\n",
    "\n",
    "# Importing excel data with multiple tabs\n",
    "df = pd.read_excel(path_file, header=0)"
   ],
   "outputs": [],
   "execution_count": 17
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "## Checking data"
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:14:44.818426Z",
     "start_time": "2024-09-27T08:14:44.810749Z"
    }
   },
   "source": [
    "# Checking the table data\n",
    "print(df.shape)\n",
    "print(df.dtypes)\n",
    "df.head()"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(9871, 14)\n",
      "Chrom                 object\n",
      "Gene_ID               object\n",
      "Gene_coord            object\n",
      "Gen_seq               object\n",
      "Transcript_ID         object\n",
      "Transcript_coord      object\n",
      "Transcript_seq        object\n",
      "CDS/miscRNA_ID        object\n",
      "Pseudogen             object\n",
      "CDS/miscRNA_coord     object\n",
      "CDS/miscRNA_nt_seq    object\n",
      "CDS_aa_seq            object\n",
      "Function              object\n",
      "Notas-JMR             object\n",
      "dtype: object\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "     Chrom         Gene_ID             Gene_coord  \\\n",
       "0  LinJ.01  LINF_010005000    LinJ.01:1520:5066:-   \n",
       "1  LinJ.01  LINF_010005100    LinJ.01:5067:7570:-   \n",
       "2  LinJ.01  LINF_010005200   LinJ.01:8150:11640:-   \n",
       "3  LinJ.01  LINF_010005300  LinJ.01:11906:12886:-   \n",
       "4  LinJ.01  LINF_010005400  LinJ.01:13649:17111:-   \n",
       "\n",
       "                                             Gen_seq    Transcript_ID  \\\n",
       "0  TCCTAGCGGACCCTTGACGCAACCACTGCTCCGCCTTTCTCCTTCT...  LINF_01T0005000   \n",
       "1  TATTCCCTCTGCAAACGCATCGCTCACACCGACCCACCTACGCACA...  LINF_01T0005100   \n",
       "2  GGCCGTCTGACGCCCTCCGACCACACCCTCGTCCTCGACAGCACCC...  LINF_01T0005200   \n",
       "3  CGAGGAGAGGGGCGGACGCACCCGCCCCCCCTCACTCTCTGCACCG...  LINF_01T0005300   \n",
       "4  CCGCCGTGTCCCAGCTCTGCCTGTGCTCGTCTCTCCCCTCAACCCC...  LINF_01T0005400   \n",
       "\n",
       "        Transcript_coord                                     Transcript_seq  \\\n",
       "0    LinJ.01:1520:5066:-  TCCTAGCGGACCCTTGACGCAACCACTGCTCCGCCTTTCTCCTTCT...   \n",
       "1    LinJ.01:5067:7570:-  TATTCCCTCTGCAAACGCATCGCTCACACCGACCCACCTACGCACA...   \n",
       "2   LinJ.01:8150:11640:-  GGCCGTCTGACGCCCTCCGACCACACCCTCGTCCTCGACAGCACCC...   \n",
       "3  LinJ.01:11906:12886:-  CGAGGAGAGGGGCGGACGCACCCGCCCCCCCTCACTCTCTGCACCG...   \n",
       "4  LinJ.01:13649:17111:-  CCGCCGTGTCCCAGCTCTGCCTGTGCTCGTCTCTCCCCTCAACCCC...   \n",
       "\n",
       "   CDS/miscRNA_ID Pseudogen      CDS/miscRNA_coord  \\\n",
       "0  LINF_010005000        NO    LinJ.01:3710:4711:-   \n",
       "1  LINF_010005100        NO    LinJ.01:5804:7438:-   \n",
       "2  LINF_010005200        NO   LinJ.01:9038:11059:-   \n",
       "3  LINF_010005300        NO  LinJ.01:12041:12601:-   \n",
       "4  LINF_010005400        NO  LinJ.01:14957:16954:-   \n",
       "\n",
       "                                  CDS/miscRNA_nt_seq  \\\n",
       "0  ATGCGCTGCGCTCTCGTCTTCGTGCTGGTCGCGGCGGCGCTCGCGT...   \n",
       "1  ATGTTGAGACGCTGCTCAGCGCTGCTGAATGATCCGCCACTGGTGA...   \n",
       "2  ATGATGTCGGCCGAGCCGCCGTCGTCGCAGCCGTACATCAGCGACG...   \n",
       "3  ATGAAGCGAGCGCGTTCTCCGTCGGCCGACAGCACCGTCTCAGCCT...   \n",
       "4  ATGCCTGCTCCGATTTTCGACAAGGTTCTCGTCGCTAACCGCGGCG...   \n",
       "\n",
       "                                          CDS_aa_seq  \\\n",
       "0  MRCALVFVLVAAALACFAPAMTHAYSTTYAARSAGSGAFAFDPLEI...   \n",
       "1  MLRRCSALLNDPPLVSGGPVLGLMRDTGTRTTIRPSIMDAQITQSA...   \n",
       "2  MMSAEPPSSQPYISDVLRRYQLERFQSSFASSMTIKDLLALQPEDF...   \n",
       "3  MKRARSPSADSTVSASAPSAVSLSSSAVFDRSLSSSSSLSAIEEKE...   \n",
       "4  MPAPIFDKVLVANRGEIACRVMATCQRLGIKTVAVYSTADEQAKHV...   \n",
       "\n",
       "                                      Function Notas-JMR  \n",
       "0        Protein of unknown function (DUF2946)       NaN  \n",
       "1  Endonuclease/Exonuclease/phosphatase family       NaN  \n",
       "2                                   Kinesin-13       NaN  \n",
       "3             hypothetical protein - conserved       NaN  \n",
       "4        Propionyl-CoA carboxylase - α-subunit       NaN  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Gen_seq</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>Transcript_seq</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>CDS/miscRNA_nt_seq</th>\n",
       "      <th>CDS_aa_seq</th>\n",
       "      <th>Function</th>\n",
       "      <th>Notas-JMR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>LinJ.01:1520:5066:-</td>\n",
       "      <td>TCCTAGCGGACCCTTGACGCAACCACTGCTCCGCCTTTCTCCTTCT...</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>LinJ.01:1520:5066:-</td>\n",
       "      <td>TCCTAGCGGACCCTTGACGCAACCACTGCTCCGCCTTTCTCCTTCT...</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.01:3710:4711:-</td>\n",
       "      <td>ATGCGCTGCGCTCTCGTCTTCGTGCTGGTCGCGGCGGCGCTCGCGT...</td>\n",
       "      <td>MRCALVFVLVAAALACFAPAMTHAYSTTYAARSAGSGAFAFDPLEI...</td>\n",
       "      <td>Protein of unknown function (DUF2946)</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>LINF_010005100</td>\n",
       "      <td>LinJ.01:5067:7570:-</td>\n",
       "      <td>TATTCCCTCTGCAAACGCATCGCTCACACCGACCCACCTACGCACA...</td>\n",
       "      <td>LINF_01T0005100</td>\n",
       "      <td>LinJ.01:5067:7570:-</td>\n",
       "      <td>TATTCCCTCTGCAAACGCATCGCTCACACCGACCCACCTACGCACA...</td>\n",
       "      <td>LINF_010005100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.01:5804:7438:-</td>\n",
       "      <td>ATGTTGAGACGCTGCTCAGCGCTGCTGAATGATCCGCCACTGGTGA...</td>\n",
       "      <td>MLRRCSALLNDPPLVSGGPVLGLMRDTGTRTTIRPSIMDAQITQSA...</td>\n",
       "      <td>Endonuclease/Exonuclease/phosphatase family</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>LINF_010005200</td>\n",
       "      <td>LinJ.01:8150:11640:-</td>\n",
       "      <td>GGCCGTCTGACGCCCTCCGACCACACCCTCGTCCTCGACAGCACCC...</td>\n",
       "      <td>LINF_01T0005200</td>\n",
       "      <td>LinJ.01:8150:11640:-</td>\n",
       "      <td>GGCCGTCTGACGCCCTCCGACCACACCCTCGTCCTCGACAGCACCC...</td>\n",
       "      <td>LINF_010005200</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.01:9038:11059:-</td>\n",
       "      <td>ATGATGTCGGCCGAGCCGCCGTCGTCGCAGCCGTACATCAGCGACG...</td>\n",
       "      <td>MMSAEPPSSQPYISDVLRRYQLERFQSSFASSMTIKDLLALQPEDF...</td>\n",
       "      <td>Kinesin-13</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>LINF_010005300</td>\n",
       "      <td>LinJ.01:11906:12886:-</td>\n",
       "      <td>CGAGGAGAGGGGCGGACGCACCCGCCCCCCCTCACTCTCTGCACCG...</td>\n",
       "      <td>LINF_01T0005300</td>\n",
       "      <td>LinJ.01:11906:12886:-</td>\n",
       "      <td>CGAGGAGAGGGGCGGACGCACCCGCCCCCCCTCACTCTCTGCACCG...</td>\n",
       "      <td>LINF_010005300</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.01:12041:12601:-</td>\n",
       "      <td>ATGAAGCGAGCGCGTTCTCCGTCGGCCGACAGCACCGTCTCAGCCT...</td>\n",
       "      <td>MKRARSPSADSTVSASAPSAVSLSSSAVFDRSLSSSSSLSAIEEKE...</td>\n",
       "      <td>hypothetical protein - conserved</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>LINF_010005400</td>\n",
       "      <td>LinJ.01:13649:17111:-</td>\n",
       "      <td>CCGCCGTGTCCCAGCTCTGCCTGTGCTCGTCTCTCCCCTCAACCCC...</td>\n",
       "      <td>LINF_01T0005400</td>\n",
       "      <td>LinJ.01:13649:17111:-</td>\n",
       "      <td>CCGCCGTGTCCCAGCTCTGCCTGTGCTCGTCTCTCCCCTCAACCCC...</td>\n",
       "      <td>LINF_010005400</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.01:14957:16954:-</td>\n",
       "      <td>ATGCCTGCTCCGATTTTCGACAAGGTTCTCGTCGCTAACCGCGGCG...</td>\n",
       "      <td>MPAPIFDKVLVANRGEIACRVMATCQRLGIKTVAVYSTADEQAKHV...</td>\n",
       "      <td>Propionyl-CoA carboxylase - α-subunit</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 18
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For a GTF file we'll need the next columns:\n",
    "1. SeqID\n",
    "2. Source\n",
    "3. Feature\n",
    "4. Start\n",
    "5. End\n",
    "6. Score\n",
    "7. Strand\n",
    "8. Phase\n",
    "9. Attributes\n",
    "\n",
    "All but the last one need a value. If not value is needed, use the dot \".\"."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "# Cleaning/Preparing data"
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:15:27.756425Z",
     "start_time": "2024-09-27T08:15:27.754690Z"
    }
   },
   "source": [
    "# Needed columns to make the GTF\n",
    "needed_columns = ['Chrom', 'Gene_ID', 'Gene_coord', 'Transcript_ID', 'Transcript_coord', 'CDS/miscRNA_ID', 'Pseudogen', 'CDS/miscRNA_coord', 'Function']"
   ],
   "outputs": [],
   "execution_count": 19
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:15:37.571033Z",
     "start_time": "2024-09-27T08:15:37.561406Z"
    }
   },
   "source": [
    "# Selecting only the needed columns\n",
    "df_filtered = df[needed_columns].copy()\n",
    "print(df_filtered.shape)\n",
    "print(df_filtered.dtypes)\n",
    "df_filtered.head()"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(9871, 9)\n",
      "Chrom                object\n",
      "Gene_ID              object\n",
      "Gene_coord           object\n",
      "Transcript_ID        object\n",
      "Transcript_coord     object\n",
      "CDS/miscRNA_ID       object\n",
      "Pseudogen            object\n",
      "CDS/miscRNA_coord    object\n",
      "Function             object\n",
      "dtype: object\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "     Chrom         Gene_ID             Gene_coord    Transcript_ID  \\\n",
       "0  LinJ.01  LINF_010005000    LinJ.01:1520:5066:-  LINF_01T0005000   \n",
       "1  LinJ.01  LINF_010005100    LinJ.01:5067:7570:-  LINF_01T0005100   \n",
       "2  LinJ.01  LINF_010005200   LinJ.01:8150:11640:-  LINF_01T0005200   \n",
       "3  LinJ.01  LINF_010005300  LinJ.01:11906:12886:-  LINF_01T0005300   \n",
       "4  LinJ.01  LINF_010005400  LinJ.01:13649:17111:-  LINF_01T0005400   \n",
       "\n",
       "        Transcript_coord  CDS/miscRNA_ID Pseudogen      CDS/miscRNA_coord  \\\n",
       "0    LinJ.01:1520:5066:-  LINF_010005000        NO    LinJ.01:3710:4711:-   \n",
       "1    LinJ.01:5067:7570:-  LINF_010005100        NO    LinJ.01:5804:7438:-   \n",
       "2   LinJ.01:8150:11640:-  LINF_010005200        NO   LinJ.01:9038:11059:-   \n",
       "3  LinJ.01:11906:12886:-  LINF_010005300        NO  LinJ.01:12041:12601:-   \n",
       "4  LinJ.01:13649:17111:-  LINF_010005400        NO  LinJ.01:14957:16954:-   \n",
       "\n",
       "                                      Function  \n",
       "0        Protein of unknown function (DUF2946)  \n",
       "1  Endonuclease/Exonuclease/phosphatase family  \n",
       "2                                   Kinesin-13  \n",
       "3             hypothetical protein - conserved  \n",
       "4        Propionyl-CoA carboxylase - α-subunit  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>LinJ.01:1520:5066:-</td>\n",
       "      <td>LINF_01T0005000</td>\n",
       "      <td>LinJ.01:1520:5066:-</td>\n",
       "      <td>LINF_010005000</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.01:3710:4711:-</td>\n",
       "      <td>Protein of unknown function (DUF2946)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>LINF_010005100</td>\n",
       "      <td>LinJ.01:5067:7570:-</td>\n",
       "      <td>LINF_01T0005100</td>\n",
       "      <td>LinJ.01:5067:7570:-</td>\n",
       "      <td>LINF_010005100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.01:5804:7438:-</td>\n",
       "      <td>Endonuclease/Exonuclease/phosphatase family</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>LINF_010005200</td>\n",
       "      <td>LinJ.01:8150:11640:-</td>\n",
       "      <td>LINF_01T0005200</td>\n",
       "      <td>LinJ.01:8150:11640:-</td>\n",
       "      <td>LINF_010005200</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.01:9038:11059:-</td>\n",
       "      <td>Kinesin-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>LINF_010005300</td>\n",
       "      <td>LinJ.01:11906:12886:-</td>\n",
       "      <td>LINF_01T0005300</td>\n",
       "      <td>LinJ.01:11906:12886:-</td>\n",
       "      <td>LINF_010005300</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.01:12041:12601:-</td>\n",
       "      <td>hypothetical protein - conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>LINF_010005400</td>\n",
       "      <td>LinJ.01:13649:17111:-</td>\n",
       "      <td>LINF_01T0005400</td>\n",
       "      <td>LinJ.01:13649:17111:-</td>\n",
       "      <td>LINF_010005400</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.01:14957:16954:-</td>\n",
       "      <td>Propionyl-CoA carboxylase - α-subunit</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 20
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "## Missing, ND, N/A and unknown values"
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:16:04.938743Z",
     "start_time": "2024-09-27T08:16:04.915787Z"
    }
   },
   "source": [
    "# Check all the columns for missing values, \"ND\", \"N/A\", and \"unknown\"\n",
    "for column in df_filtered.columns:\n",
    "    if df_filtered[column].isnull().values.any():\n",
    "        print(f\"{column} has {df_filtered[column].isnull().sum()} <missing> values\")\n",
    "    if \"ND\" in df_filtered[column].values:\n",
    "        print(f\"{column} has {df_filtered[column].value_counts()['ND']} <ND> values\")\n",
    "    if \"N/A\" in df_filtered[column].values:\n",
    "        print(f\"{column} has {df_filtered[column].value_counts()['N/A']} <N/A> values\")\n",
    "    if \"unknown\" in df_filtered[column].values:\n",
    "        print(f\"{column} has {df_filtered[column].value_counts()['unknown']} <unknown> values\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene_coord has 1 <ND> values\n",
      "Transcript_ID has 210 <ND> values\n",
      "Transcript_coord has 210 <ND> values\n",
      "CDS/miscRNA_ID has 1103 <ND> values\n",
      "CDS/miscRNA_coord has 1100 <ND> values\n",
      "Function has 1116 <unknown> values\n"
     ]
    }
   ],
   "execution_count": 21
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:white; background-color:red;\">THERE IS ONE MISSING VALUE IN GENE_COORD</span>\n",
    "\n",
    "This element will be skipped in the future code, because without these coordinates we can't make the GTF"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:16:18.782968Z",
     "start_time": "2024-09-27T08:16:18.777641Z"
    }
   },
   "source": [
    "# \"Gene_coord\" is important, so let's check which element has the ND value\n",
    "print(df_filtered[df_filtered[\"Gene_coord\"] == \"ND\"])"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        Chrom         Gene_ID Gene_coord Transcript_ID Transcript_coord  \\\n",
      "3806  LinJ.22  LINF_220021000         ND            ND               ND   \n",
      "\n",
      "      CDS/miscRNA_ID Pseudogen        CDS/miscRNA_coord              Function  \n",
      "3806  LINF_220021000        NO  LinJ.22:664742:665152:-  hypothetical_protein  \n"
     ]
    }
   ],
   "execution_count": 22
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:17:02.742232Z",
     "start_time": "2024-09-27T08:17:02.736824Z"
    }
   },
   "source": [
    "# Let's remove \"LINF_220021000\" it form the dataframe\n",
    "df_filtered = df_filtered[df_filtered[\"Gene_ID\"] != \"LINF_220021000\"].copy()"
   ],
   "outputs": [],
   "execution_count": 23
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 5.1.1 Checking where \"Transcript_coord\" is \"ND\""
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:17:20.514972Z",
     "start_time": "2024-09-27T08:17:20.506418Z"
    }
   },
   "source": [
    "# Check elements where \"Transcript_coord\" or \"Transcript_ID\" is ND\n",
    "testing_trans_coord = df_filtered[(df_filtered[\"Transcript_ID\"] == \"ND\") | (df_filtered[\"Transcript_coord\"] == \"ND\")]\n",
    "testing_trans_coord"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom             Gene_ID                 Gene_coord Transcript_ID  \\\n",
       "152   LinJ.02      LINF_020010500    LinJ.02:235340:235431:+            ND   \n",
       "160   LinJ.02  LINF_02.SLRNA.0010    LinJ.02:273078:273181:+            ND   \n",
       "161   LinJ.02  LINF_02.SLRNA.0020    LinJ.02:273519:273622:+            ND   \n",
       "162   LinJ.02  LINF_02.SLRNA.0030    LinJ.02:273947:274050:+            ND   \n",
       "163   LinJ.02  LINF_02.SLRNA.0040    LinJ.02:274388:274491:+            ND   \n",
       "...       ...                 ...                        ...           ...   \n",
       "9522  LinJ.36      LINF_360051500  LinJ.36:1648079:1648152:-            ND   \n",
       "9783  LinJ.36      LINF_360074500  LinJ.36:2530973:2531044:+            ND   \n",
       "9785  LinJ.36      LINF_360074600  LinJ.36:2531835:2531916:-            ND   \n",
       "9786  LinJ.36      LINF_360074700  LinJ.36:2531978:2532049:-            ND   \n",
       "9787  LinJ.36      LINF_360074800  LinJ.36:2532123:2532195:+            ND   \n",
       "\n",
       "     Transcript_coord      CDS/miscRNA_ID Pseudogen  \\\n",
       "152                ND      LINF_020010500        NO   \n",
       "160                ND  LINF_02.SLRNA.0010        NO   \n",
       "161                ND  LINF_02.SLRNA.0020        NO   \n",
       "162                ND  LINF_02.SLRNA.0030        NO   \n",
       "163                ND  LINF_02.SLRNA.0040        NO   \n",
       "...               ...                 ...       ...   \n",
       "9522               ND      LINF_360051500        NO   \n",
       "9783               ND      LINF_360074500        NO   \n",
       "9785               ND      LINF_360074600        NO   \n",
       "9786               ND      LINF_360074700        NO   \n",
       "9787               ND      LINF_360074800        NO   \n",
       "\n",
       "              CDS/miscRNA_coord                Function  \n",
       "152     LinJ.02:235340:235431:+  tRNA-val|Anticodon gac  \n",
       "160     LinJ.02:273078:273181:+      spliced_leader_RNA  \n",
       "161     LinJ.02:273519:273622:+      spliced_leader_RNA  \n",
       "162     LinJ.02:273947:274050:+      spliced_leader_RNA  \n",
       "163     LinJ.02:274388:274491:+      spliced_leader_RNA  \n",
       "...                         ...                     ...  \n",
       "9522  LinJ.36:1648079:1648152:-                tRNA-Gln  \n",
       "9783  LinJ.36:2530973:2531044:+                tRNA-Cys  \n",
       "9785  LinJ.36:2531835:2531916:-                tRNA-Leu  \n",
       "9786  LinJ.36:2531978:2532049:-                tRNA-Met  \n",
       "9787  LinJ.36:2532123:2532195:+                tRNA-Pro  \n",
       "\n",
       "[209 rows x 9 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>LinJ.02</td>\n",
       "      <td>LINF_020010500</td>\n",
       "      <td>LinJ.02:235340:235431:+</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_020010500</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.02:235340:235431:+</td>\n",
       "      <td>tRNA-val|Anticodon gac</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>LinJ.02</td>\n",
       "      <td>LINF_02.SLRNA.0010</td>\n",
       "      <td>LinJ.02:273078:273181:+</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_02.SLRNA.0010</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.02:273078:273181:+</td>\n",
       "      <td>spliced_leader_RNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>LinJ.02</td>\n",
       "      <td>LINF_02.SLRNA.0020</td>\n",
       "      <td>LinJ.02:273519:273622:+</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_02.SLRNA.0020</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.02:273519:273622:+</td>\n",
       "      <td>spliced_leader_RNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>LinJ.02</td>\n",
       "      <td>LINF_02.SLRNA.0030</td>\n",
       "      <td>LinJ.02:273947:274050:+</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_02.SLRNA.0030</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.02:273947:274050:+</td>\n",
       "      <td>spliced_leader_RNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>LinJ.02</td>\n",
       "      <td>LINF_02.SLRNA.0040</td>\n",
       "      <td>LinJ.02:274388:274491:+</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_02.SLRNA.0040</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.02:274388:274491:+</td>\n",
       "      <td>spliced_leader_RNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9522</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>LINF_360051500</td>\n",
       "      <td>LinJ.36:1648079:1648152:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_360051500</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.36:1648079:1648152:-</td>\n",
       "      <td>tRNA-Gln</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9783</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>LINF_360074500</td>\n",
       "      <td>LinJ.36:2530973:2531044:+</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_360074500</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.36:2530973:2531044:+</td>\n",
       "      <td>tRNA-Cys</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9785</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>LINF_360074600</td>\n",
       "      <td>LinJ.36:2531835:2531916:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_360074600</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.36:2531835:2531916:-</td>\n",
       "      <td>tRNA-Leu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9786</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>LINF_360074700</td>\n",
       "      <td>LinJ.36:2531978:2532049:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_360074700</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.36:2531978:2532049:-</td>\n",
       "      <td>tRNA-Met</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9787</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>LINF_360074800</td>\n",
       "      <td>LinJ.36:2532123:2532195:+</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_360074800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.36:2532123:2532195:+</td>\n",
       "      <td>tRNA-Pro</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>209 rows × 9 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 24
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:17:22.760745Z",
     "start_time": "2024-09-27T08:17:22.756783Z"
    }
   },
   "source": [
    "# Check the functions\n",
    "testing_trans_coord['Function'].value_counts()"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Function\n",
       "spliced_leader_RNA                                 45\n",
       "snoRNA                                             25\n",
       "unknown                                            15\n",
       "tRNA-Leu                                           11\n",
       "5S rRNA                                            11\n",
       "spliced_leader-like_RNA                            10\n",
       "tRNA-Arg                                            9\n",
       "tRNA-Gly                                            7\n",
       "tRNA-Thr                                            6\n",
       "tRNA-Ala                                            6\n",
       "rRNA                                                6\n",
       "tRNA-Pro                                            5\n",
       "tRNA-Val                                            5\n",
       "tRNA-Ser                                            5\n",
       "tRNA-Lys                                            5\n",
       "tRNA-Gln                                            4\n",
       "tRNA-Asn                                            4\n",
       "tRNA-Met                                            4\n",
       "tRNA-Ile                                            4\n",
       "tRNA-Glu                                            3\n",
       "tRNA-Asp                                            3\n",
       "hypothetical protein - conserved                    2\n",
       "tRNA-His                                            2\n",
       "tRNA-Phe                                            2\n",
       "tRNA-val|Anticodon gac                              1\n",
       "tRNA-seC                                            1\n",
       "tRNA                                                1\n",
       "hypothetical protein                                1\n",
       "tb-292 membrane associated protein-like protein     1\n",
       "protein of unknown function - conserved             1\n",
       "snRNA U6                                            1\n",
       "tRNA-Trp                                            1\n",
       "snRNA U2                                            1\n",
       "tRNA-Cys                                            1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 25
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "They are specials RNA mainly, but let's check:\n",
    "* protein of unknown function - conserved\n",
    "* tb-292 membrane associated protein-like protein\n",
    "* hypothetical_protein\n",
    "* hypothetical protein\n",
    "* hypothetical protein - conserved\n",
    "* unknown"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:17:30.183554Z",
     "start_time": "2024-09-27T08:17:30.177138Z"
    }
   },
   "source": [
    "conflicted_zone_1 = testing_trans_coord[testing_trans_coord['Function'].isin([\"protein of unknown function - conserved\", \"tb-292 membrane associated protein-like protein\", \"hypothetical_protein\", \"hypothetical protein\", \"hypothetical protein - conserved\", \"unknown\"])]\n",
    "conflicted_zone_1"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID                 Gene_coord Transcript_ID  \\\n",
       "2227  LinJ.15  LINF_150010400    LinJ.15:186963:187592:+            ND   \n",
       "2228  LinJ.15  LINF_150010500    LinJ.15:187730:189344:-            ND   \n",
       "2229  LinJ.15  LINF_150010600    LinJ.15:189599:190309:-            ND   \n",
       "2230  LinJ.15  LINF_150010700    LinJ.15:190389:191303:-            ND   \n",
       "2231  LinJ.15  LINF_150010800    LinJ.15:191558:192268:-            ND   \n",
       "2232  LinJ.15  LINF_150010900    LinJ.15:192348:193262:-            ND   \n",
       "2233  LinJ.15  LINF_150011000    LinJ.15:193517:194227:-            ND   \n",
       "2234  LinJ.15  LINF_150011100    LinJ.15:194307:195221:-            ND   \n",
       "2235  LinJ.15  LINF_150011200    LinJ.15:195476:196186:-            ND   \n",
       "2236  LinJ.15  LINF_150011300    LinJ.15:196266:197180:-            ND   \n",
       "2237  LinJ.15  LINF_150011400    LinJ.15:197435:198145:-            ND   \n",
       "2238  LinJ.15  LINF_150011500    LinJ.15:198225:199139:-            ND   \n",
       "2239  LinJ.15  LINF_150011600    LinJ.15:199394:200104:-            ND   \n",
       "2240  LinJ.15  LINF_150011700    LinJ.15:200184:201098:-            ND   \n",
       "2241  LinJ.15  LINF_150011800    LinJ.15:201353:202063:-            ND   \n",
       "2242  LinJ.15  LINF_150011900    LinJ.15:202143:203057:-            ND   \n",
       "2243  LinJ.15  LINF_150012000    LinJ.15:203312:204151:-            ND   \n",
       "2244  LinJ.15  LINF_150012100    LinJ.15:204741:219545:+            ND   \n",
       "7793  LinJ.33  LINF_330040700  LinJ.33:1443861:1447517:+            ND   \n",
       "7794  LinJ.33  LINF_330040800  LinJ.33:1453827:1467362:+            ND   \n",
       "\n",
       "     Transcript_coord  CDS/miscRNA_ID Pseudogen          CDS/miscRNA_coord  \\\n",
       "2227               ND  LINF_150010400        NO    LinJ.15:186963:187592:+   \n",
       "2228               ND  LINF_150010500        NO    LinJ.15:187730:189344:-   \n",
       "2229               ND  LINF_150010600        NO    LinJ.15:189599:190309:-   \n",
       "2230               ND  LINF_150010700        NO    LinJ.15:190389:191303:-   \n",
       "2231               ND  LINF_150010800        NO    LinJ.15:191558:192268:-   \n",
       "2232               ND  LINF_150010900        NO    LinJ.15:192348:193262:-   \n",
       "2233               ND  LINF_150011000        NO    LinJ.15:193517:194227:-   \n",
       "2234               ND  LINF_150011100        NO    LinJ.15:194307:195221:-   \n",
       "2235               ND  LINF_150011200        NO    LinJ.15:195476:196186:-   \n",
       "2236               ND  LINF_150011300        NO    LinJ.15:196266:197180:-   \n",
       "2237               ND  LINF_150011400        NO    LinJ.15:197435:198145:-   \n",
       "2238               ND  LINF_150011500        NO    LinJ.15:198225:199139:-   \n",
       "2239               ND  LINF_150011600        NO    LinJ.15:199394:200104:-   \n",
       "2240               ND  LINF_150011700        NO    LinJ.15:200184:201098:-   \n",
       "2241               ND  LINF_150011800        NO    LinJ.15:201353:202063:-   \n",
       "2242               ND  LINF_150011900        NO    LinJ.15:202143:203057:-   \n",
       "2243               ND  LINF_150012000        NO    LinJ.15:203312:204151:-   \n",
       "2244               ND  LINF_150012100        NO      LinJ.15:204741:219545   \n",
       "7793               ND  LINF_330040700        NO  LinJ.33:1443861:1447517:+   \n",
       "7794               ND  LINF_330040800        NO  LinJ.33:1453827:1467362:+   \n",
       "\n",
       "                                             Function  \n",
       "2227          protein of unknown function - conserved  \n",
       "2228                             hypothetical protein  \n",
       "2229                                          unknown  \n",
       "2230                                          unknown  \n",
       "2231                                          unknown  \n",
       "2232                                          unknown  \n",
       "2233                                          unknown  \n",
       "2234                                          unknown  \n",
       "2235                                          unknown  \n",
       "2236                                          unknown  \n",
       "2237                                          unknown  \n",
       "2238                                          unknown  \n",
       "2239                                          unknown  \n",
       "2240                                          unknown  \n",
       "2241                                          unknown  \n",
       "2242                                          unknown  \n",
       "2243                                          unknown  \n",
       "2244  tb-292 membrane associated protein-like protein  \n",
       "7793                 hypothetical protein - conserved  \n",
       "7794                 hypothetical protein - conserved  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2227</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150010400</td>\n",
       "      <td>LinJ.15:186963:187592:+</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150010400</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:186963:187592:+</td>\n",
       "      <td>protein of unknown function - conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2228</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150010500</td>\n",
       "      <td>LinJ.15:187730:189344:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150010500</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:187730:189344:-</td>\n",
       "      <td>hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2229</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150010600</td>\n",
       "      <td>LinJ.15:189599:190309:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150010600</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:189599:190309:-</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2230</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150010700</td>\n",
       "      <td>LinJ.15:190389:191303:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150010700</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:190389:191303:-</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2231</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150010800</td>\n",
       "      <td>LinJ.15:191558:192268:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150010800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:191558:192268:-</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2232</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150010900</td>\n",
       "      <td>LinJ.15:192348:193262:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150010900</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:192348:193262:-</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2233</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011000</td>\n",
       "      <td>LinJ.15:193517:194227:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150011000</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:193517:194227:-</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2234</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011100</td>\n",
       "      <td>LinJ.15:194307:195221:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150011100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:194307:195221:-</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2235</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011200</td>\n",
       "      <td>LinJ.15:195476:196186:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150011200</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:195476:196186:-</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2236</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011300</td>\n",
       "      <td>LinJ.15:196266:197180:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150011300</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:196266:197180:-</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2237</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011400</td>\n",
       "      <td>LinJ.15:197435:198145:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150011400</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:197435:198145:-</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2238</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011500</td>\n",
       "      <td>LinJ.15:198225:199139:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150011500</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:198225:199139:-</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2239</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011600</td>\n",
       "      <td>LinJ.15:199394:200104:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150011600</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:199394:200104:-</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2240</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011700</td>\n",
       "      <td>LinJ.15:200184:201098:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150011700</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:200184:201098:-</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2241</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011800</td>\n",
       "      <td>LinJ.15:201353:202063:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150011800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:201353:202063:-</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2242</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011900</td>\n",
       "      <td>LinJ.15:202143:203057:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150011900</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:202143:203057:-</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2243</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150012000</td>\n",
       "      <td>LinJ.15:203312:204151:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150012000</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:203312:204151:-</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2244</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150012100</td>\n",
       "      <td>LinJ.15:204741:219545:+</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_150012100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:204741:219545</td>\n",
       "      <td>tb-292 membrane associated protein-like protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7793</th>\n",
       "      <td>LinJ.33</td>\n",
       "      <td>LINF_330040700</td>\n",
       "      <td>LinJ.33:1443861:1447517:+</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_330040700</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.33:1443861:1447517:+</td>\n",
       "      <td>hypothetical protein - conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7794</th>\n",
       "      <td>LinJ.33</td>\n",
       "      <td>LINF_330040800</td>\n",
       "      <td>LinJ.33:1453827:1467362:+</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>LINF_330040800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.33:1453827:1467362:+</td>\n",
       "      <td>hypothetical protein - conserved</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 26
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Shouldn't they have a transcript?\n",
    "\n",
    "<span style=\"color:red;\">@JMR told us to remove the CDS/misRNA_coord and note them</span>\n",
    "\n",
    "Apart from that: Speaking with @Javi and @JMR (16/07/24) this zone is a conflicted zone. Should add to \"Notes/Function\" sth like - \"conflicted_zone_in_study\" for example"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:17:35.934698Z",
     "start_time": "2024-09-27T08:17:35.932036Z"
    }
   },
   "source": [
    "# Take the \"Gene_ID\" from the elements in the last code cell\n",
    "testing_list = testing_trans_coord[testing_trans_coord['Function'].isin([\"protein of unknown function - conserved\", \"tb-292 membrane associated protein-like protein\", \"hypothetical_protein\", \"hypothetical protein\", \"hypothetical protein - conserved\", \"unknown\"])]\\\n",
    "    [\"Gene_ID\"].tolist()\n",
    "\n",
    "# Check list\n",
    "print(testing_list)"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['LINF_150010400', 'LINF_150010500', 'LINF_150010600', 'LINF_150010700', 'LINF_150010800', 'LINF_150010900', 'LINF_150011000', 'LINF_150011100', 'LINF_150011200', 'LINF_150011300', 'LINF_150011400', 'LINF_150011500', 'LINF_150011600', 'LINF_150011700', 'LINF_150011800', 'LINF_150011900', 'LINF_150012000', 'LINF_150012100', 'LINF_330040700', 'LINF_330040800']\n"
     ]
    }
   ],
   "execution_count": 27
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:17:38.594274Z",
     "start_time": "2024-09-27T08:17:38.579610Z"
    }
   },
   "source": [
    "# Remove in df_filtered, in the \"testing_list\" the elements \"CDS/miscRNA_ID\", and \"CDS/miscRNA_coord\" to \"ND\" \n",
    "df_filtered.loc[df_filtered[\"Gene_ID\"].isin(testing_list), [\"CDS/miscRNA_ID\", \"CDS/miscRNA_coord\"]] = \"ND\"\n",
    "\n",
    "# To the \"Function\" column in the `testing_list` we will add to the original text the words: _conflicted_zone_in_study\n",
    "df_filtered.loc[df_filtered[\"Gene_ID\"].isin(testing_list), \"Function\"] = df_filtered.loc[df_filtered[\"Gene_ID\"].isin(testing_list), \"Function\"] + \"_conflicted_zone_in_study\"\n",
    "\n",
    "\n",
    "# Check the elements in the \"testing_list\"\n",
    "df_filtered[df_filtered[\"Gene_ID\"].isin(testing_list)]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID                 Gene_coord Transcript_ID  \\\n",
       "2227  LinJ.15  LINF_150010400    LinJ.15:186963:187592:+            ND   \n",
       "2228  LinJ.15  LINF_150010500    LinJ.15:187730:189344:-            ND   \n",
       "2229  LinJ.15  LINF_150010600    LinJ.15:189599:190309:-            ND   \n",
       "2230  LinJ.15  LINF_150010700    LinJ.15:190389:191303:-            ND   \n",
       "2231  LinJ.15  LINF_150010800    LinJ.15:191558:192268:-            ND   \n",
       "2232  LinJ.15  LINF_150010900    LinJ.15:192348:193262:-            ND   \n",
       "2233  LinJ.15  LINF_150011000    LinJ.15:193517:194227:-            ND   \n",
       "2234  LinJ.15  LINF_150011100    LinJ.15:194307:195221:-            ND   \n",
       "2235  LinJ.15  LINF_150011200    LinJ.15:195476:196186:-            ND   \n",
       "2236  LinJ.15  LINF_150011300    LinJ.15:196266:197180:-            ND   \n",
       "2237  LinJ.15  LINF_150011400    LinJ.15:197435:198145:-            ND   \n",
       "2238  LinJ.15  LINF_150011500    LinJ.15:198225:199139:-            ND   \n",
       "2239  LinJ.15  LINF_150011600    LinJ.15:199394:200104:-            ND   \n",
       "2240  LinJ.15  LINF_150011700    LinJ.15:200184:201098:-            ND   \n",
       "2241  LinJ.15  LINF_150011800    LinJ.15:201353:202063:-            ND   \n",
       "2242  LinJ.15  LINF_150011900    LinJ.15:202143:203057:-            ND   \n",
       "2243  LinJ.15  LINF_150012000    LinJ.15:203312:204151:-            ND   \n",
       "2244  LinJ.15  LINF_150012100    LinJ.15:204741:219545:+            ND   \n",
       "7793  LinJ.33  LINF_330040700  LinJ.33:1443861:1447517:+            ND   \n",
       "7794  LinJ.33  LINF_330040800  LinJ.33:1453827:1467362:+            ND   \n",
       "\n",
       "     Transcript_coord CDS/miscRNA_ID Pseudogen CDS/miscRNA_coord  \\\n",
       "2227               ND             ND        NO                ND   \n",
       "2228               ND             ND        NO                ND   \n",
       "2229               ND             ND        NO                ND   \n",
       "2230               ND             ND        NO                ND   \n",
       "2231               ND             ND        NO                ND   \n",
       "2232               ND             ND        NO                ND   \n",
       "2233               ND             ND        NO                ND   \n",
       "2234               ND             ND        NO                ND   \n",
       "2235               ND             ND        NO                ND   \n",
       "2236               ND             ND        NO                ND   \n",
       "2237               ND             ND        NO                ND   \n",
       "2238               ND             ND        NO                ND   \n",
       "2239               ND             ND        NO                ND   \n",
       "2240               ND             ND        NO                ND   \n",
       "2241               ND             ND        NO                ND   \n",
       "2242               ND             ND        NO                ND   \n",
       "2243               ND             ND        NO                ND   \n",
       "2244               ND             ND        NO                ND   \n",
       "7793               ND             ND        NO                ND   \n",
       "7794               ND             ND        NO                ND   \n",
       "\n",
       "                                               Function  \n",
       "2227  protein of unknown function - conserved_confli...  \n",
       "2228      hypothetical protein_conflicted_zone_in_study  \n",
       "2229                   unknown_conflicted_zone_in_study  \n",
       "2230                   unknown_conflicted_zone_in_study  \n",
       "2231                   unknown_conflicted_zone_in_study  \n",
       "2232                   unknown_conflicted_zone_in_study  \n",
       "2233                   unknown_conflicted_zone_in_study  \n",
       "2234                   unknown_conflicted_zone_in_study  \n",
       "2235                   unknown_conflicted_zone_in_study  \n",
       "2236                   unknown_conflicted_zone_in_study  \n",
       "2237                   unknown_conflicted_zone_in_study  \n",
       "2238                   unknown_conflicted_zone_in_study  \n",
       "2239                   unknown_conflicted_zone_in_study  \n",
       "2240                   unknown_conflicted_zone_in_study  \n",
       "2241                   unknown_conflicted_zone_in_study  \n",
       "2242                   unknown_conflicted_zone_in_study  \n",
       "2243                   unknown_conflicted_zone_in_study  \n",
       "2244  tb-292 membrane associated protein-like protei...  \n",
       "7793  hypothetical protein - conserved_conflicted_zo...  \n",
       "7794  hypothetical protein - conserved_conflicted_zo...  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2227</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150010400</td>\n",
       "      <td>LinJ.15:186963:187592:+</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>protein of unknown function - conserved_confli...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2228</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150010500</td>\n",
       "      <td>LinJ.15:187730:189344:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>hypothetical protein_conflicted_zone_in_study</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2229</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150010600</td>\n",
       "      <td>LinJ.15:189599:190309:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>unknown_conflicted_zone_in_study</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2230</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150010700</td>\n",
       "      <td>LinJ.15:190389:191303:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>unknown_conflicted_zone_in_study</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2231</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150010800</td>\n",
       "      <td>LinJ.15:191558:192268:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>unknown_conflicted_zone_in_study</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2232</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150010900</td>\n",
       "      <td>LinJ.15:192348:193262:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>unknown_conflicted_zone_in_study</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2233</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011000</td>\n",
       "      <td>LinJ.15:193517:194227:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>unknown_conflicted_zone_in_study</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2234</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011100</td>\n",
       "      <td>LinJ.15:194307:195221:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>unknown_conflicted_zone_in_study</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2235</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011200</td>\n",
       "      <td>LinJ.15:195476:196186:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>unknown_conflicted_zone_in_study</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2236</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011300</td>\n",
       "      <td>LinJ.15:196266:197180:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>unknown_conflicted_zone_in_study</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2237</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011400</td>\n",
       "      <td>LinJ.15:197435:198145:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>unknown_conflicted_zone_in_study</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2238</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011500</td>\n",
       "      <td>LinJ.15:198225:199139:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>unknown_conflicted_zone_in_study</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2239</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011600</td>\n",
       "      <td>LinJ.15:199394:200104:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>unknown_conflicted_zone_in_study</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2240</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011700</td>\n",
       "      <td>LinJ.15:200184:201098:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>unknown_conflicted_zone_in_study</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2241</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011800</td>\n",
       "      <td>LinJ.15:201353:202063:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>unknown_conflicted_zone_in_study</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2242</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150011900</td>\n",
       "      <td>LinJ.15:202143:203057:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>unknown_conflicted_zone_in_study</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2243</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150012000</td>\n",
       "      <td>LinJ.15:203312:204151:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>unknown_conflicted_zone_in_study</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2244</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150012100</td>\n",
       "      <td>LinJ.15:204741:219545:+</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>tb-292 membrane associated protein-like protei...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7793</th>\n",
       "      <td>LinJ.33</td>\n",
       "      <td>LINF_330040700</td>\n",
       "      <td>LinJ.33:1443861:1447517:+</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>hypothetical protein - conserved_conflicted_zo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7794</th>\n",
       "      <td>LinJ.33</td>\n",
       "      <td>LINF_330040800</td>\n",
       "      <td>LinJ.33:1453827:1467362:+</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>hypothetical protein - conserved_conflicted_zo...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 28
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "## Checking coordinates"
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Wrong coordinates step 1\n",
    "Normally the coordinates are like <span style=\"color:orange\">LinJ.01:1000:5000:-</span>:\n",
    "* The fields are \"chromosome\", \"start\", \"end\" and \"strand\" separated by \":\"\n",
    "\n",
    "In this part we are going to check if every coordinate have the same structure"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:18:11.208363Z",
     "start_time": "2024-09-27T08:18:11.195189Z"
    }
   },
   "source": [
    "# Checking which indices doesn't have the coordinate correct structure\n",
    "df_condition = df_filtered[(df_filtered['CDS/miscRNA_coord'] != \"ND\") & (df_filtered['Gene_coord'] != \"ND\")].copy()\n",
    "condition = df_condition['Gene_coord'].str.split(\":\").str[3] == df_condition['CDS/miscRNA_coord'].str.split(\":\").str[3]  # checking if the 3rd element is sense\n",
    "wrong_coord_1 = df_condition[~condition]['Gene_ID'].tolist()\n",
    "print(f\"Indices with incorrect coordinates structure: {wrong_coord_1}\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Indices with incorrect coordinates structure: ['LINF_260030600', 'LINF_330015900']\n"
     ]
    }
   ],
   "execution_count": 29
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:18:13.437330Z",
     "start_time": "2024-09-27T08:18:13.424655Z"
    }
   },
   "source": [
    "# Checking which indices doesn't have the coordinate correct structure\n",
    "df_condition = df_filtered[(df_filtered['CDS/miscRNA_coord'] != \"ND\") & (df_filtered['Gene_coord'] != \"ND\")].copy()\n",
    "condition = df_condition['Gene_coord'].str.split(\":\").str[3] == df_condition['CDS/miscRNA_coord'].str.split(\":\").str[3]  # checking if the 3rd element is sense\n",
    "wrong_coord_1 = df_condition[~condition]['Gene_ID'].tolist()\n",
    "print(f\"Indices with incorrect coordinates structure: {wrong_coord_1}\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Indices with incorrect coordinates structure: ['LINF_260030600', 'LINF_330015900']\n"
     ]
    }
   ],
   "execution_count": 30
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:18:15.201529Z",
     "start_time": "2024-09-27T08:18:15.193039Z"
    }
   },
   "source": [
    "df_condition.loc[\n",
    "    df_condition['Gene_ID'].isin(wrong_coord_1), \n",
    "    ['Gene_ID', 'Gene_coord', 'Transcript_coord', 'CDS/miscRNA_coord', 'Function']\n",
    "]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "             Gene_ID               Gene_coord         Transcript_coord  \\\n",
       "4926  LINF_260030600  LinJ.26:974904:977491:+  LinJ.26:974904:977491:+   \n",
       "7511  LINF_330015900  LinJ.33:351965:353644:+  LinJ.33:351965:353644:+   \n",
       "\n",
       "            CDS/miscRNA_coord                          Function  \n",
       "4926  LinJ.26:975665-977149:+              hypothetical protein  \n",
       "7511  LinJ.33:352089-353393:+  hypothetical protein - conserved  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4926</th>\n",
       "      <td>LINF_260030600</td>\n",
       "      <td>LinJ.26:974904:977491:+</td>\n",
       "      <td>LinJ.26:974904:977491:+</td>\n",
       "      <td>LinJ.26:975665-977149:+</td>\n",
       "      <td>hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7511</th>\n",
       "      <td>LINF_330015900</td>\n",
       "      <td>LinJ.33:351965:353644:+</td>\n",
       "      <td>LinJ.33:351965:353644:+</td>\n",
       "      <td>LinJ.33:352089-353393:+</td>\n",
       "      <td>hypothetical protein - conserved</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 31
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Seems there are some coordinates not separated by \":\", and instead by \"-\" so it doesn't follow the normal structure. We need to correct it for a correct script indexing, for the moment we save this indices"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "#### Repairing"
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:18:37.213513Z",
     "start_time": "2024-09-27T08:18:37.205557Z"
    }
   },
   "source": [
    "# Function to apply the regex substitution\n",
    "def replace_dash_with_colon(s):\n",
    "    return re.sub(r'(\\d+)-(\\d+)', r'\\1:\\2', s)\n",
    "\n",
    "# Apply the function to the Series\n",
    "df_filtered.loc[df_filtered['Gene_ID'].isin(wrong_coord_1), 'CDS/miscRNA_coord'] = df_filtered.loc[df_filtered['Gene_ID'].isin(wrong_coord_1), 'CDS/miscRNA_coord'].apply(replace_dash_with_colon)\n",
    "\n",
    "# Checking the change\n",
    "df_filtered.loc[df_filtered['Gene_ID'].isin(wrong_coord_1), ['Gene_ID', 'Gene_coord', 'Transcript_coord', 'CDS/miscRNA_coord', 'Function']]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "             Gene_ID               Gene_coord         Transcript_coord  \\\n",
       "4926  LINF_260030600  LinJ.26:974904:977491:+  LinJ.26:974904:977491:+   \n",
       "7511  LINF_330015900  LinJ.33:351965:353644:+  LinJ.33:351965:353644:+   \n",
       "\n",
       "            CDS/miscRNA_coord                          Function  \n",
       "4926  LinJ.26:975665:977149:+              hypothetical protein  \n",
       "7511  LinJ.33:352089:353393:+  hypothetical protein - conserved  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4926</th>\n",
       "      <td>LINF_260030600</td>\n",
       "      <td>LinJ.26:974904:977491:+</td>\n",
       "      <td>LinJ.26:974904:977491:+</td>\n",
       "      <td>LinJ.26:975665:977149:+</td>\n",
       "      <td>hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7511</th>\n",
       "      <td>LINF_330015900</td>\n",
       "      <td>LinJ.33:351965:353644:+</td>\n",
       "      <td>LinJ.33:351965:353644:+</td>\n",
       "      <td>LinJ.33:352089:353393:+</td>\n",
       "      <td>hypothetical protein - conserved</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 32
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Wrong coordinates step 2\n",
    "If we remember, the correct coordinate structure is <span style=\"color:orange\">LinJ.01:1000:5000:-</span>:\n",
    "* * The fields are \"chromosome\", \"start\", \"end\" and \"strand\" separated by \":\".\n",
    "\n",
    "Since there are 4 fields separated by \":\", when we split the data by \":\" there should be 4 elements. We are going to check if there are indeed 4 elements and where is not"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:18:51.818978Z",
     "start_time": "2024-09-27T08:18:51.773187Z"
    }
   },
   "source": [
    "# Check as well in df_condition for the values when .split[\":\"] is not 4\n",
    "df_condition = df_filtered[(df_filtered['CDS/miscRNA_coord'] != \"ND\") & (df_filtered['Gene_coord'] != \"ND\")].copy()\n",
    "df_condition['Gene_coord_split'] = df_condition['Gene_coord'].str.split(\":\")\n",
    "df_condition['CDS/miscRNA_coord_split'] = df_condition['CDS/miscRNA_coord'].str.split(\":\")\n",
    "condition = (df_condition['Gene_coord_split'].str.len() != 4) | (df_condition['CDS/miscRNA_coord_split'].str.len() != 4)\n",
    "wrong_coord_2 = df_condition[condition]['Gene_ID'].tolist()\n",
    "print(f\"False indices: {wrong_coord_2}\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False indices: []\n"
     ]
    }
   ],
   "execution_count": 33
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:18:53.651303Z",
     "start_time": "2024-09-27T08:18:53.645716Z"
    }
   },
   "source": [
    "# Check the elements\n",
    "df_condition.loc[\n",
    "    df_condition['Gene_ID'].isin(wrong_coord_2), \n",
    "    ['Gene_ID', 'Gene_coord', 'Transcript_coord', 'CDS/miscRNA_coord', 'Function']\n",
    "]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Gene_ID, Gene_coord, Transcript_coord, CDS/miscRNA_coord, Function]\n",
       "Index: []"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 34
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are no wrong elements in \"CDS/miscRNA_coord\" and \"Gene_coord\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Wrong coordinates step 3\n",
    "\n",
    "What if there were an structure like <span style=\"color:orange\">LinJ.01:1000-5000:10000-20000:-</span>?\n",
    "\n",
    "There would be 4 fields when we separate it by \":\":\n",
    "* LinJ.01\n",
    "* 1000-5000\n",
    "* 10000-20000\n",
    "* \"-\"\n",
    "\n",
    "But in the last script, there will be a problem since the start coordinates can't be \"1000-5000\", so let's check if this exists in the table"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:19:04.967103Z",
     "start_time": "2024-09-27T08:19:04.947762Z"
    }
   },
   "source": [
    "# search the regex condition where \"number\"-\"number\" in the Gene_coord and CDS/miscRNA_coord columns\n",
    "# import re\n",
    "df_condition = df_filtered[(df_filtered['CDS/miscRNA_coord'] != \"ND\") & (df_filtered['Gene_coord'] != \"ND\")].copy()\n",
    "# condition = df_condition['Gene_coord'].str.contains(r\"\\d+-\\d+\") | df_condition['CDS/miscRNA_coord'].str.contains(r\"\\d+-\\d+\")\n",
    "condition = df_condition['Gene_coord'].str.contains(r\"\\d+-\\d+\") | df_condition['Transcript_coord'].str.contains(r\"\\d+-\\d+\") | df_condition['CDS/miscRNA_coord'].str.contains(r\"\\d+-\\d+\")\n",
    "\n",
    "wrong_coord_3 = df_condition[condition]['Gene_ID'].tolist()\n",
    "print(f\"False indices: {wrong_coord_3}\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False indices: ['LINF_070008800', 'LINF_100016800', 'LINF_240023900', 'LINF_290033800']\n"
     ]
    }
   ],
   "execution_count": 35
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:19:06.770380Z",
     "start_time": "2024-09-27T08:19:06.762621Z"
    }
   },
   "source": [
    "df_condition.loc[\n",
    "    df_condition['Gene_ID'].isin(wrong_coord_3), \n",
    "    ['Gene_ID', 'Gene_coord', 'Transcript_coord', 'CDS/miscRNA_coord', 'Function']\n",
    "]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "             Gene_ID                 Gene_coord  \\\n",
       "842   LINF_070008800    LinJ.07:153281:160411:-   \n",
       "1431  LINF_100016800    LinJ.10:468436:469642:-   \n",
       "4288  LINF_240023900    LinJ.24:689866:693427:-   \n",
       "6002  LINF_290033800  LinJ.29:1181710:1184611:+   \n",
       "\n",
       "                               Transcript_coord  \\\n",
       "842       LinJ.07:153281-154334:158854-160411:-   \n",
       "1431      LinJ.10:468436-468904:469155-469642:-   \n",
       "4288      LinJ.24:689866-690901:692650-693427:-   \n",
       "6002  LinJ.29:1181710-1182263:1183113-1184611:+   \n",
       "\n",
       "                              CDS/miscRNA_coord  \\\n",
       "842       LinJ.07:153994-154334:158854-160240:-   \n",
       "1431                    LinJ.10:469169:469561:-   \n",
       "4288      LinJ.24:690626-690901:692650-693288:-   \n",
       "6002  LinJ.29:1182001-1182263:1183113-1184352:+   \n",
       "\n",
       "                                               Function  \n",
       "842                  ATP-dependent RNA helicase (DBP2B)  \n",
       "1431                                         histone H3  \n",
       "4288                                 nPABP-like protein  \n",
       "6002  polynucleotide adenylyltransferase|poly(A) pol...  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>842</th>\n",
       "      <td>LINF_070008800</td>\n",
       "      <td>LinJ.07:153281:160411:-</td>\n",
       "      <td>LinJ.07:153281-154334:158854-160411:-</td>\n",
       "      <td>LinJ.07:153994-154334:158854-160240:-</td>\n",
       "      <td>ATP-dependent RNA helicase (DBP2B)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1431</th>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>LinJ.10:468436:469642:-</td>\n",
       "      <td>LinJ.10:468436-468904:469155-469642:-</td>\n",
       "      <td>LinJ.10:469169:469561:-</td>\n",
       "      <td>histone H3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4288</th>\n",
       "      <td>LINF_240023900</td>\n",
       "      <td>LinJ.24:689866:693427:-</td>\n",
       "      <td>LinJ.24:689866-690901:692650-693427:-</td>\n",
       "      <td>LinJ.24:690626-690901:692650-693288:-</td>\n",
       "      <td>nPABP-like protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6002</th>\n",
       "      <td>LINF_290033800</td>\n",
       "      <td>LinJ.29:1181710:1184611:+</td>\n",
       "      <td>LinJ.29:1181710-1182263:1183113-1184611:+</td>\n",
       "      <td>LinJ.29:1182001-1182263:1183113-1184352:+</td>\n",
       "      <td>polynucleotide adenylyltransferase|poly(A) pol...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 36
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In cases like LINF_070008800, LINF_240023900, LINF_290033800 there seems to be 2 transcripts and 2 CDS.\n",
    "In the case of LINF_100016800 there 2 transcript but one CDS\n",
    "\n",
    "We'll have to improve the original table, so it can be parsed later, First let's repair the coordinates:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "#### Repairing"
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:20:42.885502Z",
     "start_time": "2024-09-27T08:20:42.874090Z"
    }
   },
   "source": [
    "# Repair \"Transcript_coord\"\n",
    "import re\n",
    "\n",
    "for elem in wrong_coord_3:\n",
    "    df_filtered.loc[df_filtered['Gene_ID'] == elem, 'Transcript_coord'] = df_filtered.loc[df_filtered['Gene_ID'] == elem, 'Transcript_coord'].apply(\n",
    "        lambda x: re.sub(r'(\\d+)-(\\d+):(\\d+)-(\\d+)', r'\\1:\\2-\\3:\\4', x) if isinstance(x, str) else x\n",
    "    )\n",
    "\n",
    "# Check the elements\n",
    "df_filtered[df_filtered['Gene_ID'].isin(wrong_coord_3)]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID                 Gene_coord    Transcript_ID  \\\n",
       "842   LinJ.07  LINF_070008800    LinJ.07:153281:160411:-  LINF_07T0008800   \n",
       "1431  LinJ.10  LINF_100016800    LinJ.10:468436:469642:-  LINF_10T0016800   \n",
       "4288  LinJ.24  LINF_240023900    LinJ.24:689866:693427:-  LINF_24T0023900   \n",
       "6002  LinJ.29  LINF_290033800  LinJ.29:1181710:1184611:+  LINF_29T0033800   \n",
       "\n",
       "                               Transcript_coord  CDS/miscRNA_ID Pseudogen  \\\n",
       "842       LinJ.07:153281:154334-158854:160411:-  LINF_070008800        NO   \n",
       "1431      LinJ.10:468436:468904-469155:469642:-  LINF_100016800        NO   \n",
       "4288      LinJ.24:689866:690901-692650:693427:-  LINF_240023900        NO   \n",
       "6002  LinJ.29:1181710:1182263-1183113:1184611:+  LINF_290033800        NO   \n",
       "\n",
       "                              CDS/miscRNA_coord  \\\n",
       "842       LinJ.07:153994-154334:158854-160240:-   \n",
       "1431                    LinJ.10:469169:469561:-   \n",
       "4288      LinJ.24:690626-690901:692650-693288:-   \n",
       "6002  LinJ.29:1182001-1182263:1183113-1184352:+   \n",
       "\n",
       "                                               Function  \n",
       "842                  ATP-dependent RNA helicase (DBP2B)  \n",
       "1431                                         histone H3  \n",
       "4288                                 nPABP-like protein  \n",
       "6002  polynucleotide adenylyltransferase|poly(A) pol...  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>842</th>\n",
       "      <td>LinJ.07</td>\n",
       "      <td>LINF_070008800</td>\n",
       "      <td>LinJ.07:153281:160411:-</td>\n",
       "      <td>LINF_07T0008800</td>\n",
       "      <td>LinJ.07:153281:154334-158854:160411:-</td>\n",
       "      <td>LINF_070008800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.07:153994-154334:158854-160240:-</td>\n",
       "      <td>ATP-dependent RNA helicase (DBP2B)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1431</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>LinJ.10:468436:469642:-</td>\n",
       "      <td>LINF_10T0016800</td>\n",
       "      <td>LinJ.10:468436:468904-469155:469642:-</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.10:469169:469561:-</td>\n",
       "      <td>histone H3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4288</th>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>LINF_240023900</td>\n",
       "      <td>LinJ.24:689866:693427:-</td>\n",
       "      <td>LINF_24T0023900</td>\n",
       "      <td>LinJ.24:689866:690901-692650:693427:-</td>\n",
       "      <td>LINF_240023900</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.24:690626-690901:692650-693288:-</td>\n",
       "      <td>nPABP-like protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6002</th>\n",
       "      <td>LinJ.29</td>\n",
       "      <td>LINF_290033800</td>\n",
       "      <td>LinJ.29:1181710:1184611:+</td>\n",
       "      <td>LINF_29T0033800</td>\n",
       "      <td>LinJ.29:1181710:1182263-1183113:1184611:+</td>\n",
       "      <td>LINF_290033800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.29:1182001-1182263:1183113-1184352:+</td>\n",
       "      <td>polynucleotide adenylyltransferase|poly(A) pol...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 37
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:21:27.053563Z",
     "start_time": "2024-09-27T08:21:27.043323Z"
    }
   },
   "source": [
    "# Repairing CDS/miscRNA_coord\n",
    "cds_coord_repaired = [\"LINF_070008800\", \"LINF_240023900\", \"LINF_290033800\"]\n",
    "for elem in cds_coord_repaired:\n",
    "    df_filtered.loc[df_filtered['Gene_ID'] == elem, 'CDS/miscRNA_coord'] = df_filtered.loc[df_filtered['Gene_ID'] == elem, 'CDS/miscRNA_coord'].apply(\n",
    "        lambda x: re.sub(r'(\\d+)-(\\d+):(\\d+)-(\\d+)', r'\\1:\\2-\\3:\\4', x) if isinstance(x, str) else x\n",
    "    )\n",
    "\n",
    "# Check the elements\n",
    "df_filtered[df_filtered['Gene_ID'].isin(cds_coord_repaired)]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID                 Gene_coord    Transcript_ID  \\\n",
       "842   LinJ.07  LINF_070008800    LinJ.07:153281:160411:-  LINF_07T0008800   \n",
       "4288  LinJ.24  LINF_240023900    LinJ.24:689866:693427:-  LINF_24T0023900   \n",
       "6002  LinJ.29  LINF_290033800  LinJ.29:1181710:1184611:+  LINF_29T0033800   \n",
       "\n",
       "                               Transcript_coord  CDS/miscRNA_ID Pseudogen  \\\n",
       "842       LinJ.07:153281:154334-158854:160411:-  LINF_070008800        NO   \n",
       "4288      LinJ.24:689866:690901-692650:693427:-  LINF_240023900        NO   \n",
       "6002  LinJ.29:1181710:1182263-1183113:1184611:+  LINF_290033800        NO   \n",
       "\n",
       "                              CDS/miscRNA_coord  \\\n",
       "842       LinJ.07:153994:154334-158854:160240:-   \n",
       "4288      LinJ.24:690626:690901-692650:693288:-   \n",
       "6002  LinJ.29:1182001:1182263-1183113:1184352:+   \n",
       "\n",
       "                                               Function  \n",
       "842                  ATP-dependent RNA helicase (DBP2B)  \n",
       "4288                                 nPABP-like protein  \n",
       "6002  polynucleotide adenylyltransferase|poly(A) pol...  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>842</th>\n",
       "      <td>LinJ.07</td>\n",
       "      <td>LINF_070008800</td>\n",
       "      <td>LinJ.07:153281:160411:-</td>\n",
       "      <td>LINF_07T0008800</td>\n",
       "      <td>LinJ.07:153281:154334-158854:160411:-</td>\n",
       "      <td>LINF_070008800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.07:153994:154334-158854:160240:-</td>\n",
       "      <td>ATP-dependent RNA helicase (DBP2B)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4288</th>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>LINF_240023900</td>\n",
       "      <td>LinJ.24:689866:693427:-</td>\n",
       "      <td>LINF_24T0023900</td>\n",
       "      <td>LinJ.24:689866:690901-692650:693427:-</td>\n",
       "      <td>LINF_240023900</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.24:690626:690901-692650:693288:-</td>\n",
       "      <td>nPABP-like protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6002</th>\n",
       "      <td>LinJ.29</td>\n",
       "      <td>LINF_290033800</td>\n",
       "      <td>LinJ.29:1181710:1184611:+</td>\n",
       "      <td>LINF_29T0033800</td>\n",
       "      <td>LinJ.29:1181710:1182263-1183113:1184611:+</td>\n",
       "      <td>LINF_290033800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.29:1182001:1182263-1183113:1184352:+</td>\n",
       "      <td>polynucleotide adenylyltransferase|poly(A) pol...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 38
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now let's divide the row, 2 rows for each transcript."
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:21:34.999781Z",
     "start_time": "2024-09-27T08:21:34.974420Z"
    }
   },
   "source": [
    "for gene in cds_coord_repaired:\n",
    "    # Get column values:\n",
    "    chrom = df_filtered[df_filtered['Gene_ID'] == gene]['Chrom'].values[0]\n",
    "    sense = df_filtered[df_filtered['Gene_ID'] == gene]['Gene_coord'].str.split(\":\").str[3].values[0]\n",
    "    gene_id = df_filtered[df_filtered['Gene_ID'] == gene]['Gene_ID'].values[0]\n",
    "    gene_coord = df_filtered[df_filtered['Gene_ID'] == gene]['Gene_coord'].values[0]\n",
    "    transcript_id = df_filtered[df_filtered['Gene_ID'] == gene]['Transcript_ID'].values[0]\n",
    "    transcript_coord = df_filtered[df_filtered['Gene_ID'] == gene]['Transcript_coord'].values[0]\n",
    "    cds_id = df_filtered[df_filtered['Gene_ID'] == gene]['CDS/miscRNA_ID'].values[0]\n",
    "    pseudogen = df_filtered[df_filtered['Gene_ID'] == gene]['Pseudogen'].values[0]\n",
    "    cds_coord = df_filtered[df_filtered['Gene_ID'] == gene]['CDS/miscRNA_coord'].values[0]\n",
    "    function = df_filtered[df_filtered['Gene_ID'] == gene]['Function'].values[0]\n",
    "\n",
    "\n",
    "    # Get the coordinates\n",
    "    first_trans_coord = transcript_coord.split(\"-\")[0].split(\":\")[1] + \":\" + transcript_coord.split(\"-\")[0].split(\":\")[2]\n",
    "    second_trans_coord = transcript_coord.split(\"-\")[1].split(\":\")[0] + \":\" + transcript_coord.split(\"-\")[1].split(\":\")[1]\n",
    "\n",
    "    first_cds_coord = cds_coord.split(\"-\")[0].split(\":\")[1] + \":\" + cds_coord.split(\"-\")[0].split(\":\")[2]\n",
    "    second_cds_coord = cds_coord.split(\"-\")[1].split(\":\")[0] + \":\" + cds_coord.split(\"-\")[1].split(\":\")[1]\n",
    "\n",
    "    # get the rows:\n",
    "    first_row = [\n",
    "        chrom,\n",
    "        gene_id,\n",
    "        gene_coord,\n",
    "        transcript_id + \"_1\",\n",
    "        chrom + \":\" + first_trans_coord + \":\" + sense,\n",
    "        cds_id + \"_1\",\n",
    "        pseudogen,\n",
    "        chrom + \":\" + first_cds_coord + \":\" + sense,\n",
    "        function\n",
    "    ]\n",
    "\n",
    "    second_row = [\n",
    "        chrom,\n",
    "        gene_id,\n",
    "        gene_coord,\n",
    "        transcript_id + \"_2\",\n",
    "        chrom + \":\" + second_trans_coord + \":\" + sense,\n",
    "        cds_id + \"_2\",\n",
    "        pseudogen,\n",
    "        chrom + \":\" + second_cds_coord + \":\" + sense,\n",
    "        function\n",
    "    ]\n",
    "\n",
    "    # Transforming the rows to dataframes\n",
    "    first_row = pd.DataFrame([first_row], columns=df_filtered.columns)\n",
    "    second_row = pd.DataFrame([second_row], columns=df_filtered.columns)\n",
    "\n",
    "    # Get the index of the row:\n",
    "    index_split = df_filtered[df_filtered['Gene_ID'] == gene].index[0]\n",
    "\n",
    "    # Split the data frame at the index_split\n",
    "    df_part1 = df_filtered.iloc[:index_split, :]\n",
    "    df_part2 = df_filtered.iloc[index_split+1:, :]\n",
    "\n",
    "    # Removing the last row of the first part, because it will be divided\n",
    "    df_part1 = df_part1.iloc[:-1, :]\n",
    "\n",
    "    # Inserting the new two rows\n",
    "    df_filtered = pd.concat([df_part1, first_row, second_row, df_part2], ignore_index=True)"
   ],
   "outputs": [],
   "execution_count": 39
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:21:41.507389Z",
     "start_time": "2024-09-27T08:21:41.500949Z"
    }
   },
   "source": [
    "# Let's check the results\n",
    "df_filtered[df_filtered['Gene_ID'].isin(cds_coord_repaired)]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID                 Gene_coord      Transcript_ID  \\\n",
       "841   LinJ.07  LINF_070008800    LinJ.07:153281:160411:-  LINF_07T0008800_1   \n",
       "842   LinJ.07  LINF_070008800    LinJ.07:153281:160411:-  LINF_07T0008800_2   \n",
       "4286  LinJ.24  LINF_240023900    LinJ.24:689866:693427:-  LINF_24T0023900_1   \n",
       "4287  LinJ.24  LINF_240023900    LinJ.24:689866:693427:-  LINF_24T0023900_2   \n",
       "6000  LinJ.29  LINF_290033800  LinJ.29:1181710:1184611:+  LINF_29T0033800_1   \n",
       "6001  LinJ.29  LINF_290033800  LinJ.29:1181710:1184611:+  LINF_29T0033800_2   \n",
       "\n",
       "               Transcript_coord    CDS/miscRNA_ID Pseudogen  \\\n",
       "841     LinJ.07:153281:154334:-  LINF_070008800_1        NO   \n",
       "842     LinJ.07:158854:160411:-  LINF_070008800_2        NO   \n",
       "4286    LinJ.24:689866:690901:-  LINF_240023900_1        NO   \n",
       "4287    LinJ.24:692650:693427:-  LINF_240023900_2        NO   \n",
       "6000  LinJ.29:1181710:1182263:+  LINF_290033800_1        NO   \n",
       "6001  LinJ.29:1183113:1184611:+  LINF_290033800_2        NO   \n",
       "\n",
       "              CDS/miscRNA_coord  \\\n",
       "841     LinJ.07:153994:154334:-   \n",
       "842     LinJ.07:158854:160240:-   \n",
       "4286    LinJ.24:690626:690901:-   \n",
       "4287    LinJ.24:692650:693288:-   \n",
       "6000  LinJ.29:1182001:1182263:+   \n",
       "6001  LinJ.29:1183113:1184352:+   \n",
       "\n",
       "                                               Function  \n",
       "841                  ATP-dependent RNA helicase (DBP2B)  \n",
       "842                  ATP-dependent RNA helicase (DBP2B)  \n",
       "4286                                 nPABP-like protein  \n",
       "4287                                 nPABP-like protein  \n",
       "6000  polynucleotide adenylyltransferase|poly(A) pol...  \n",
       "6001  polynucleotide adenylyltransferase|poly(A) pol...  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>841</th>\n",
       "      <td>LinJ.07</td>\n",
       "      <td>LINF_070008800</td>\n",
       "      <td>LinJ.07:153281:160411:-</td>\n",
       "      <td>LINF_07T0008800_1</td>\n",
       "      <td>LinJ.07:153281:154334:-</td>\n",
       "      <td>LINF_070008800_1</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.07:153994:154334:-</td>\n",
       "      <td>ATP-dependent RNA helicase (DBP2B)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>842</th>\n",
       "      <td>LinJ.07</td>\n",
       "      <td>LINF_070008800</td>\n",
       "      <td>LinJ.07:153281:160411:-</td>\n",
       "      <td>LINF_07T0008800_2</td>\n",
       "      <td>LinJ.07:158854:160411:-</td>\n",
       "      <td>LINF_070008800_2</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.07:158854:160240:-</td>\n",
       "      <td>ATP-dependent RNA helicase (DBP2B)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4286</th>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>LINF_240023900</td>\n",
       "      <td>LinJ.24:689866:693427:-</td>\n",
       "      <td>LINF_24T0023900_1</td>\n",
       "      <td>LinJ.24:689866:690901:-</td>\n",
       "      <td>LINF_240023900_1</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.24:690626:690901:-</td>\n",
       "      <td>nPABP-like protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4287</th>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>LINF_240023900</td>\n",
       "      <td>LinJ.24:689866:693427:-</td>\n",
       "      <td>LINF_24T0023900_2</td>\n",
       "      <td>LinJ.24:692650:693427:-</td>\n",
       "      <td>LINF_240023900_2</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.24:692650:693288:-</td>\n",
       "      <td>nPABP-like protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6000</th>\n",
       "      <td>LinJ.29</td>\n",
       "      <td>LINF_290033800</td>\n",
       "      <td>LinJ.29:1181710:1184611:+</td>\n",
       "      <td>LINF_29T0033800_1</td>\n",
       "      <td>LinJ.29:1181710:1182263:+</td>\n",
       "      <td>LINF_290033800_1</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.29:1182001:1182263:+</td>\n",
       "      <td>polynucleotide adenylyltransferase|poly(A) pol...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6001</th>\n",
       "      <td>LinJ.29</td>\n",
       "      <td>LINF_290033800</td>\n",
       "      <td>LinJ.29:1181710:1184611:+</td>\n",
       "      <td>LINF_29T0033800_2</td>\n",
       "      <td>LinJ.29:1183113:1184611:+</td>\n",
       "      <td>LINF_290033800_2</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.29:1183113:1184352:+</td>\n",
       "      <td>polynucleotide adenylyltransferase|poly(A) pol...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 40
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:21:44.027073Z",
     "start_time": "2024-09-27T08:21:44.018869Z"
    }
   },
   "source": [
    "df_filtered[df_filtered['Gene_ID'] == \"LINF_100016800\"]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID               Gene_coord    Transcript_ID  \\\n",
       "1431  LinJ.10  LINF_100016800  LinJ.10:468436:469642:-  LINF_10T0016800   \n",
       "\n",
       "                           Transcript_coord  CDS/miscRNA_ID Pseudogen  \\\n",
       "1431  LinJ.10:468436:468904-469155:469642:-  LINF_100016800        NO   \n",
       "\n",
       "            CDS/miscRNA_coord    Function  \n",
       "1431  LinJ.10:469169:469561:-  histone H3  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1431</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>LinJ.10:468436:469642:-</td>\n",
       "      <td>LINF_10T0016800</td>\n",
       "      <td>LinJ.10:468436:468904-469155:469642:-</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.10:469169:469561:-</td>\n",
       "      <td>histone H3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 41
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Seems in **LINF_100016800** the CDS (469169:469561) is inside the second transcript (469155:469642) only"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:22:01.086109Z",
     "start_time": "2024-09-27T08:22:01.073815Z"
    }
   },
   "source": [
    "# Let's reuse the code \n",
    "for gene in [\"LINF_100016800\"]:\n",
    "    # Get column values:\n",
    "    chrom = df_filtered[df_filtered['Gene_ID'] == gene]['Chrom'].values[0]\n",
    "    sense = df_filtered[df_filtered['Gene_ID'] == gene]['Gene_coord'].str.split(\":\").str[3].values[0]\n",
    "    gene_id = df_filtered[df_filtered['Gene_ID'] == gene]['Gene_ID'].values[0]\n",
    "    gene_coord = df_filtered[df_filtered['Gene_ID'] == gene]['Gene_coord'].values[0]\n",
    "    transcript_id = df_filtered[df_filtered['Gene_ID'] == gene]['Transcript_ID'].values[0]\n",
    "    transcript_coord = df_filtered[df_filtered['Gene_ID'] == gene]['Transcript_coord'].values[0]\n",
    "    cds_id = df_filtered[df_filtered['Gene_ID'] == gene]['CDS/miscRNA_ID'].values[0]\n",
    "    pseudogen = df_filtered[df_filtered['Gene_ID'] == gene]['Pseudogen'].values[0]\n",
    "    cds_coord = df_filtered[df_filtered['Gene_ID'] == gene]['CDS/miscRNA_coord'].values[0]\n",
    "    function = df_filtered[df_filtered['Gene_ID'] == gene]['Function'].values[0]\n",
    "\n",
    "\n",
    "    # Get the coordinates\n",
    "    first_trans_coord = transcript_coord.split(\"-\")[0].split(\":\")[1] + \":\" + transcript_coord.split(\"-\")[0].split(\":\")[2]\n",
    "    second_trans_coord = transcript_coord.split(\"-\")[1].split(\":\")[0] + \":\" + transcript_coord.split(\"-\")[1].split(\":\")[1]\n",
    "\n",
    "    second_cds_coord = df_filtered[df_filtered['Gene_ID'] == gene]['CDS/miscRNA_coord'].values[0]  # This is the only one that changes in the code and it keeps the same value\n",
    "\n",
    "    # get the rows:\n",
    "    first_row = [\n",
    "        chrom,\n",
    "        gene_id,\n",
    "        gene_coord,\n",
    "        transcript_id + \"_1\",\n",
    "        chrom + \":\" + first_trans_coord + \":\" + sense,\n",
    "        \"ND\",  # No CDS in this transcript\n",
    "        pseudogen,\n",
    "        \"ND\",  # No CDS in this transcript,\n",
    "        function\n",
    "    ]\n",
    "\n",
    "    second_row = [\n",
    "        chrom,\n",
    "        gene_id,\n",
    "        gene_coord,\n",
    "        transcript_id + \"_2\",\n",
    "        chrom + \":\" + second_trans_coord + \":\" + sense,\n",
    "        cds_id, # only one CDS in this gene\n",
    "        pseudogen,\n",
    "        second_cds_coord,  # only one CDS in this gene\n",
    "        function\n",
    "    ]\n",
    "\n",
    "    # Transforming the rows to dataframes\n",
    "    first_row = pd.DataFrame([first_row], columns=df_filtered.columns)\n",
    "    second_row = pd.DataFrame([second_row], columns=df_filtered.columns)\n",
    "\n",
    "    # Get the index of the row:\n",
    "    index_split = df_filtered[df_filtered['Gene_ID'] == gene].index[0]\n",
    "\n",
    "    # Split the data frame at the index_split\n",
    "    df_part1 = df_filtered.iloc[:index_split, :]\n",
    "    df_part2 = df_filtered.iloc[index_split+1:, :]\n",
    "\n",
    "    # Removing the last row of the first part, because it will be divided\n",
    "    df_part1 = df_part1.iloc[:-1, :]\n",
    "\n",
    "    # Inserting the new two rows\n",
    "    df_filtered = pd.concat([df_part1, first_row, second_row, df_part2], ignore_index=True)"
   ],
   "outputs": [],
   "execution_count": 42
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:22:04.184094Z",
     "start_time": "2024-09-27T08:22:04.177723Z"
    }
   },
   "source": [
    "df_filtered[df_filtered['Gene_ID'] == \"LINF_100016800\"]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID               Gene_coord      Transcript_ID  \\\n",
       "1430  LinJ.10  LINF_100016800  LinJ.10:468436:469642:-  LINF_10T0016800_1   \n",
       "1431  LinJ.10  LINF_100016800  LinJ.10:468436:469642:-  LINF_10T0016800_2   \n",
       "\n",
       "             Transcript_coord  CDS/miscRNA_ID Pseudogen  \\\n",
       "1430  LinJ.10:468436:468904:-              ND        NO   \n",
       "1431  LinJ.10:469155:469642:-  LINF_100016800        NO   \n",
       "\n",
       "            CDS/miscRNA_coord    Function  \n",
       "1430                       ND  histone H3  \n",
       "1431  LinJ.10:469169:469561:-  histone H3  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1430</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>LinJ.10:468436:469642:-</td>\n",
       "      <td>LINF_10T0016800_1</td>\n",
       "      <td>LinJ.10:468436:468904:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>histone H3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1431</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>LinJ.10:468436:469642:-</td>\n",
       "      <td>LINF_10T0016800_2</td>\n",
       "      <td>LinJ.10:469155:469642:-</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.10:469169:469561:-</td>\n",
       "      <td>histone H3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 43
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "# Other cleaning/preparing data"
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "## Check other possible problems"
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "### Pseudogen categories"
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:22:39.681965Z",
     "start_time": "2024-09-27T08:22:39.678504Z"
    }
   },
   "source": [
    "# Need to bear in mind the field \"pseudogen\"\n",
    "df_filtered['Pseudogen'].value_counts()"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Pseudogen\n",
       "NO     9821\n",
       "YES      40\n",
       "SI        9\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 44
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "In the case of \"pseudogene\" we need to have in mind that there are no only \"YES\" values as expected, but \"SI\" values as well."
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "### CDS/miscRNA with ND but with coordinates"
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:22:56.648710Z",
     "start_time": "2024-09-27T08:22:56.642198Z"
    }
   },
   "source": [
    "# In this case, there is no ID for the CDS/miscRNA, but there are coordinates. Let's check them\n",
    "df_filtered[(df_filtered['CDS/miscRNA_ID'] == \"ND\") & (df_filtered['CDS/miscRNA_coord'] != \"ND\")]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID                 Gene_coord    Transcript_ID  \\\n",
       "1462  LinJ.10  LINF_100019760    LinJ.10:549490:550353:-  LINF_10T0019760   \n",
       "1789  LinJ.12  LINF_120017725    LinJ.12:590295:591611:+  LINF_12T0017725   \n",
       "9702  LinJ.36  LINF_360067550  LinJ.36:2250628:2251942:-  LINF_36T0067550   \n",
       "\n",
       "               Transcript_coord CDS/miscRNA_ID Pseudogen  \\\n",
       "1462    LinJ.10:549490:550353:-             ND        NO   \n",
       "1789    LinJ.12:590295:591611:+             ND        NO   \n",
       "9702  LinJ.36:2250628:2251942:-             ND        NO   \n",
       "\n",
       "              CDS/miscRNA_coord                                 Function  \n",
       "1462    LinJ.10:549923:550327:-  Protein of unknown function - conserved  \n",
       "1789    LinJ.12:590379:591341:+                        methyltransferase  \n",
       "9702  LinJ.36:2251654:2251797:-              protein of unknown function  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1462</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100019760</td>\n",
       "      <td>LinJ.10:549490:550353:-</td>\n",
       "      <td>LINF_10T0019760</td>\n",
       "      <td>LinJ.10:549490:550353:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.10:549923:550327:-</td>\n",
       "      <td>Protein of unknown function - conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1789</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>LINF_120017725</td>\n",
       "      <td>LinJ.12:590295:591611:+</td>\n",
       "      <td>LINF_12T0017725</td>\n",
       "      <td>LinJ.12:590295:591611:+</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.12:590379:591341:+</td>\n",
       "      <td>methyltransferase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9702</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>LINF_360067550</td>\n",
       "      <td>LinJ.36:2250628:2251942:-</td>\n",
       "      <td>LINF_36T0067550</td>\n",
       "      <td>LinJ.36:2250628:2251942:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.36:2251654:2251797:-</td>\n",
       "      <td>protein of unknown function</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 45
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "Speaking with @Javi this seems to be an annotation error, let's add the \"Gene_ID\" column ID to the \"CDS/miscRNA_ID\""
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:23:00.626978Z",
     "start_time": "2024-09-27T08:23:00.618805Z"
    }
   },
   "source": [
    "# For the cases in the cell before, use the \"Gene_ID\" value in \"CDS/miscRNA_ID\"\n",
    "condition = (df_filtered['CDS/miscRNA_ID'] == \"ND\") & (df_filtered['CDS/miscRNA_coord'] != \"ND\")\n",
    "df_filtered.loc[condition, 'CDS/miscRNA_ID'] = df_filtered.loc[condition, 'Gene_ID']\n",
    "df_filtered.loc[condition]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID                 Gene_coord    Transcript_ID  \\\n",
       "1462  LinJ.10  LINF_100019760    LinJ.10:549490:550353:-  LINF_10T0019760   \n",
       "1789  LinJ.12  LINF_120017725    LinJ.12:590295:591611:+  LINF_12T0017725   \n",
       "9702  LinJ.36  LINF_360067550  LinJ.36:2250628:2251942:-  LINF_36T0067550   \n",
       "\n",
       "               Transcript_coord  CDS/miscRNA_ID Pseudogen  \\\n",
       "1462    LinJ.10:549490:550353:-  LINF_100019760        NO   \n",
       "1789    LinJ.12:590295:591611:+  LINF_120017725        NO   \n",
       "9702  LinJ.36:2250628:2251942:-  LINF_360067550        NO   \n",
       "\n",
       "              CDS/miscRNA_coord                                 Function  \n",
       "1462    LinJ.10:549923:550327:-  Protein of unknown function - conserved  \n",
       "1789    LinJ.12:590379:591341:+                        methyltransferase  \n",
       "9702  LinJ.36:2251654:2251797:-              protein of unknown function  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1462</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100019760</td>\n",
       "      <td>LinJ.10:549490:550353:-</td>\n",
       "      <td>LINF_10T0019760</td>\n",
       "      <td>LinJ.10:549490:550353:-</td>\n",
       "      <td>LINF_100019760</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.10:549923:550327:-</td>\n",
       "      <td>Protein of unknown function - conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1789</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>LINF_120017725</td>\n",
       "      <td>LinJ.12:590295:591611:+</td>\n",
       "      <td>LINF_12T0017725</td>\n",
       "      <td>LinJ.12:590295:591611:+</td>\n",
       "      <td>LINF_120017725</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.12:590379:591341:+</td>\n",
       "      <td>methyltransferase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9702</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>LINF_360067550</td>\n",
       "      <td>LinJ.36:2250628:2251942:-</td>\n",
       "      <td>LINF_36T0067550</td>\n",
       "      <td>LinJ.36:2250628:2251942:-</td>\n",
       "      <td>LINF_360067550</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.36:2251654:2251797:-</td>\n",
       "      <td>protein of unknown function</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 46
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "Now it's repaired"
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "### \"CDS/miscRNA_coord\" are not inside the coordinates in \"Gene_coord\""
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:23:22.874567Z",
     "start_time": "2024-09-27T08:23:22.866980Z"
    }
   },
   "source": [
    "df_condition[df_condition['CDS/miscRNA_coord'].str.split(\":\").str[1].values == \"+\"]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Chrom, Gene_ID, Gene_coord, Transcript_ID, Transcript_coord, CDS/miscRNA_ID, Pseudogen, CDS/miscRNA_coord, Function]\n",
       "Index: []"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 47
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:23:29.425677Z",
     "start_time": "2024-09-27T08:23:29.399224Z"
    }
   },
   "source": [
    "# Let's check where the coordinates  in \"CDS/miscRNA_coord\" are not inside the coordinates in \"Gene_coord\"\n",
    "## Prepare Data Frame\n",
    "condition = (df_filtered['CDS/miscRNA_coord'] != \"ND\") & (df_filtered['Gene_coord'] != \"ND\")  # Condition to filter, for when the coordinates are not ND\n",
    "df_condition = df_filtered[condition].copy()\n",
    "df_condition = df_condition.drop([842, 4288, 6002])  # Drop the rows with the LinJ.XX:100:1000-2000:3000 coordinates types\n",
    "\n",
    "# Prepare columns\n",
    "df_condition['Gene_coord_start'] = df_condition['Gene_coord'].str.split(\":\").str[1].astype(int)\n",
    "df_condition['Gene_coord_end'] = df_condition['Gene_coord'].str.split(\":\").str[2].astype(int)\n",
    "df_condition['CDS/miscRNA_coord_start'] = df_condition['CDS/miscRNA_coord'].str.split(\":\").str[1].astype(int)\n",
    "df_condition['CDS/miscRNA_coord_end'] = df_condition['CDS/miscRNA_coord'].str.split(\":\").str[2].astype(int)\n",
    "# print(df_condition[['Gene_ID', 'Gene_coord', 'CDS/miscRNA_coord', 'Gene_coord_start', 'Gene_coord_end', 'CDS/miscRNA_coord_start', 'CDS/miscRNA_coord_end']])  # Check the values\n",
    "\n",
    "# Now let's check that all the coordinates in \"CDS/miscRNA_coord\" are inside the coordinates in \"Gene_coord\"\n",
    "condition = (\n",
    "    (df_condition['CDS/miscRNA_coord_start'] >= df_condition['Gene_coord_start']) & (df_condition['CDS/miscRNA_coord_start'] <= df_condition['Gene_coord_end']) &\n",
    "    (df_condition['CDS/miscRNA_coord_end'] >= df_condition['Gene_coord_start']) & (df_condition['CDS/miscRNA_coord_end'] <= df_condition['Gene_coord_end'])\n",
    "    )\n",
    "\n",
    "# check the lines that don't fulfill the condition\n",
    "df_condition[~condition]\n"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID                 Gene_coord    Transcript_ID  \\\n",
       "1551  LinJ.11  LINF_110011400    LinJ.11:202887:204220:+  LINF_11T0011400   \n",
       "2067  LinJ.14  LINF_140012250    LinJ.14:248820:251321:-  LINF_14T0012250   \n",
       "2174  LinJ.15  LINF_150005000        LinJ.15:1108:1687:-  LINF_15T0005000   \n",
       "3949  LinJ.23  LINF_230015300    LinJ.23:366240:369541:-  LINF_23T0015300   \n",
       "5906  LinJ.29  LINF_290024900    LinJ.29:850028:850959:-  LINF_29T0024900   \n",
       "8710  LinJ.35  LINF_350034200  LinJ.35:1106085:1108479:-  LINF_35T0034200   \n",
       "8711  LinJ.35  LINF_350034300  LinJ.35:1108768:1111622:-  LINF_35T0034300   \n",
       "\n",
       "               Transcript_coord  CDS/miscRNA_ID Pseudogen  \\\n",
       "1551    LinJ.11:202887:204220:+  LINF_110011400       YES   \n",
       "2067    LinJ.14:248820:251321:-  LINF_140012250       YES   \n",
       "2174        LinJ.15:1108:1687:-  LINF_150005000        NO   \n",
       "3949    LinJ.23:366240:369541:-  LINF_230015300        NO   \n",
       "5906    LinJ.29:850028:850959:-  LINF_290024900        NO   \n",
       "8710  LinJ.35:1106085:1108479:-  LINF_350034200        NO   \n",
       "8711  LinJ.35:1108768:1111622:-  LINF_350034300        NO   \n",
       "\n",
       "              CDS/miscRNA_coord                                     Function  \\\n",
       "1551    LinJ.11:202513:204017:+          argonaute like protein (pseudogene)   \n",
       "2067    LinJ.14:250151:251383:-                          fatty acid elongase   \n",
       "2174        LinJ.15:1505:1715:-                                   histone H4   \n",
       "3949    LinJ.23:367028:369919:-             hypothetical protein - conserved   \n",
       "5906    LinJ.29:849730:850788:-               paraflagellar rod protein-like   \n",
       "8710  LinJ.35:1109345:1111528:-  mt-LAF5|mitoribosomal LSU assembly factor 5   \n",
       "8711  LinJ.35:1109345:1111843:-             Hypothetical protein - conserved   \n",
       "\n",
       "      Gene_coord_start  Gene_coord_end  CDS/miscRNA_coord_start  \\\n",
       "1551            202887          204220                   202513   \n",
       "2067            248820          251321                   250151   \n",
       "2174              1108            1687                     1505   \n",
       "3949            366240          369541                   367028   \n",
       "5906            850028          850959                   849730   \n",
       "8710           1106085         1108479                  1109345   \n",
       "8711           1108768         1111622                  1109345   \n",
       "\n",
       "      CDS/miscRNA_coord_end  \n",
       "1551                 204017  \n",
       "2067                 251383  \n",
       "2174                   1715  \n",
       "3949                 369919  \n",
       "5906                 850788  \n",
       "8710                1111528  \n",
       "8711                1111843  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "      <th>Gene_coord_start</th>\n",
       "      <th>Gene_coord_end</th>\n",
       "      <th>CDS/miscRNA_coord_start</th>\n",
       "      <th>CDS/miscRNA_coord_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1551</th>\n",
       "      <td>LinJ.11</td>\n",
       "      <td>LINF_110011400</td>\n",
       "      <td>LinJ.11:202887:204220:+</td>\n",
       "      <td>LINF_11T0011400</td>\n",
       "      <td>LinJ.11:202887:204220:+</td>\n",
       "      <td>LINF_110011400</td>\n",
       "      <td>YES</td>\n",
       "      <td>LinJ.11:202513:204017:+</td>\n",
       "      <td>argonaute like protein (pseudogene)</td>\n",
       "      <td>202887</td>\n",
       "      <td>204220</td>\n",
       "      <td>202513</td>\n",
       "      <td>204017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2067</th>\n",
       "      <td>LinJ.14</td>\n",
       "      <td>LINF_140012250</td>\n",
       "      <td>LinJ.14:248820:251321:-</td>\n",
       "      <td>LINF_14T0012250</td>\n",
       "      <td>LinJ.14:248820:251321:-</td>\n",
       "      <td>LINF_140012250</td>\n",
       "      <td>YES</td>\n",
       "      <td>LinJ.14:250151:251383:-</td>\n",
       "      <td>fatty acid elongase</td>\n",
       "      <td>248820</td>\n",
       "      <td>251321</td>\n",
       "      <td>250151</td>\n",
       "      <td>251383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2174</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150005000</td>\n",
       "      <td>LinJ.15:1108:1687:-</td>\n",
       "      <td>LINF_15T0005000</td>\n",
       "      <td>LinJ.15:1108:1687:-</td>\n",
       "      <td>LINF_150005000</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:1505:1715:-</td>\n",
       "      <td>histone H4</td>\n",
       "      <td>1108</td>\n",
       "      <td>1687</td>\n",
       "      <td>1505</td>\n",
       "      <td>1715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3949</th>\n",
       "      <td>LinJ.23</td>\n",
       "      <td>LINF_230015300</td>\n",
       "      <td>LinJ.23:366240:369541:-</td>\n",
       "      <td>LINF_23T0015300</td>\n",
       "      <td>LinJ.23:366240:369541:-</td>\n",
       "      <td>LINF_230015300</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.23:367028:369919:-</td>\n",
       "      <td>hypothetical protein - conserved</td>\n",
       "      <td>366240</td>\n",
       "      <td>369541</td>\n",
       "      <td>367028</td>\n",
       "      <td>369919</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5906</th>\n",
       "      <td>LinJ.29</td>\n",
       "      <td>LINF_290024900</td>\n",
       "      <td>LinJ.29:850028:850959:-</td>\n",
       "      <td>LINF_29T0024900</td>\n",
       "      <td>LinJ.29:850028:850959:-</td>\n",
       "      <td>LINF_290024900</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.29:849730:850788:-</td>\n",
       "      <td>paraflagellar rod protein-like</td>\n",
       "      <td>850028</td>\n",
       "      <td>850959</td>\n",
       "      <td>849730</td>\n",
       "      <td>850788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8710</th>\n",
       "      <td>LinJ.35</td>\n",
       "      <td>LINF_350034200</td>\n",
       "      <td>LinJ.35:1106085:1108479:-</td>\n",
       "      <td>LINF_35T0034200</td>\n",
       "      <td>LinJ.35:1106085:1108479:-</td>\n",
       "      <td>LINF_350034200</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.35:1109345:1111528:-</td>\n",
       "      <td>mt-LAF5|mitoribosomal LSU assembly factor 5</td>\n",
       "      <td>1106085</td>\n",
       "      <td>1108479</td>\n",
       "      <td>1109345</td>\n",
       "      <td>1111528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8711</th>\n",
       "      <td>LinJ.35</td>\n",
       "      <td>LINF_350034300</td>\n",
       "      <td>LinJ.35:1108768:1111622:-</td>\n",
       "      <td>LINF_35T0034300</td>\n",
       "      <td>LinJ.35:1108768:1111622:-</td>\n",
       "      <td>LINF_350034300</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.35:1109345:1111843:-</td>\n",
       "      <td>Hypothetical protein - conserved</td>\n",
       "      <td>1108768</td>\n",
       "      <td>1111622</td>\n",
       "      <td>1109345</td>\n",
       "      <td>1111843</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 48
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "At the end, we'll have to ignore:\n",
    "* LINF_360065300 ==> spoke with @Javi and @JMR, hypothetical bi-cistronic gene.\n",
    "* LINF_110011400, LINF_140012250, LINF_150005000, LINF_230015300, LINF_290024900, LINF_350034200, LINF_350034300  ==> CDS/miscRNA_coord\tare outside Gene_coord.\n",
    "* Elements where `['Gene_coord'] == \"ND\"` (Already dropped from the table)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "### Transcript_coord diff than gene_coord"
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Normally elements in **Transcript_coord** are the same as in **Gene_coord**, except when a gene has more than 1 transcript: let's check it"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:24:04.001694Z",
     "start_time": "2024-09-27T08:24:03.992844Z"
    }
   },
   "source": [
    "test_df = df_filtered.copy()\n",
    "test_df = test_df[test_df['Transcript_coord'] != \"ND\"].copy()\n",
    "test_df[test_df['Gene_coord'] != test_df['Transcript_coord']]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID                 Gene_coord      Transcript_ID  \\\n",
       "841   LinJ.07  LINF_070008800    LinJ.07:153281:160411:-  LINF_07T0008800_1   \n",
       "842   LinJ.07  LINF_070008800    LinJ.07:153281:160411:-  LINF_07T0008800_2   \n",
       "1430  LinJ.10  LINF_100016800    LinJ.10:468436:469642:-  LINF_10T0016800_1   \n",
       "1431  LinJ.10  LINF_100016800    LinJ.10:468436:469642:-  LINF_10T0016800_2   \n",
       "4286  LinJ.24  LINF_240023900    LinJ.24:689866:693427:-  LINF_24T0023900_1   \n",
       "4287  LinJ.24  LINF_240023900    LinJ.24:689866:693427:-  LINF_24T0023900_2   \n",
       "6000  LinJ.29  LINF_290033800  LinJ.29:1181710:1184611:+  LINF_29T0033800_1   \n",
       "6001  LinJ.29  LINF_290033800  LinJ.29:1181710:1184611:+  LINF_29T0033800_2   \n",
       "9840  LinJ.36  LINF_360079700  LinJ.36:2681995:2683375:-    LINF_36T0079700   \n",
       "\n",
       "               Transcript_coord    CDS/miscRNA_ID Pseudogen  \\\n",
       "841     LinJ.07:153281:154334:-  LINF_070008800_1        NO   \n",
       "842     LinJ.07:158854:160411:-  LINF_070008800_2        NO   \n",
       "1430    LinJ.10:468436:468904:-                ND        NO   \n",
       "1431    LinJ.10:469155:469642:-    LINF_100016800        NO   \n",
       "4286    LinJ.24:689866:690901:-  LINF_240023900_1        NO   \n",
       "4287    LinJ.24:692650:693427:-  LINF_240023900_2        NO   \n",
       "6000  LinJ.29:1181710:1182263:+  LINF_290033800_1        NO   \n",
       "6001  LinJ.29:1183113:1184611:+  LINF_290033800_2        NO   \n",
       "9840  LinJ.36:2681395:2683433:-    LINF_360079700        SI   \n",
       "\n",
       "              CDS/miscRNA_coord  \\\n",
       "841     LinJ.07:153994:154334:-   \n",
       "842     LinJ.07:158854:160240:-   \n",
       "1430                         ND   \n",
       "1431    LinJ.10:469169:469561:-   \n",
       "4286    LinJ.24:690626:690901:-   \n",
       "4287    LinJ.24:692650:693288:-   \n",
       "6000  LinJ.29:1182001:1182263:+   \n",
       "6001  LinJ.29:1183113:1184352:+   \n",
       "9840  LinJ.36:2682378:2682647:-   \n",
       "\n",
       "                                               Function  \n",
       "841                  ATP-dependent RNA helicase (DBP2B)  \n",
       "842                  ATP-dependent RNA helicase (DBP2B)  \n",
       "1430                                         histone H3  \n",
       "1431                                         histone H3  \n",
       "4286                                 nPABP-like protein  \n",
       "4287                                 nPABP-like protein  \n",
       "6000  polynucleotide adenylyltransferase|poly(A) pol...  \n",
       "6001  polynucleotide adenylyltransferase|poly(A) pol...  \n",
       "9840                                               SET8  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>841</th>\n",
       "      <td>LinJ.07</td>\n",
       "      <td>LINF_070008800</td>\n",
       "      <td>LinJ.07:153281:160411:-</td>\n",
       "      <td>LINF_07T0008800_1</td>\n",
       "      <td>LinJ.07:153281:154334:-</td>\n",
       "      <td>LINF_070008800_1</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.07:153994:154334:-</td>\n",
       "      <td>ATP-dependent RNA helicase (DBP2B)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>842</th>\n",
       "      <td>LinJ.07</td>\n",
       "      <td>LINF_070008800</td>\n",
       "      <td>LinJ.07:153281:160411:-</td>\n",
       "      <td>LINF_07T0008800_2</td>\n",
       "      <td>LinJ.07:158854:160411:-</td>\n",
       "      <td>LINF_070008800_2</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.07:158854:160240:-</td>\n",
       "      <td>ATP-dependent RNA helicase (DBP2B)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1430</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>LinJ.10:468436:469642:-</td>\n",
       "      <td>LINF_10T0016800_1</td>\n",
       "      <td>LinJ.10:468436:468904:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>histone H3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1431</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>LinJ.10:468436:469642:-</td>\n",
       "      <td>LINF_10T0016800_2</td>\n",
       "      <td>LinJ.10:469155:469642:-</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.10:469169:469561:-</td>\n",
       "      <td>histone H3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4286</th>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>LINF_240023900</td>\n",
       "      <td>LinJ.24:689866:693427:-</td>\n",
       "      <td>LINF_24T0023900_1</td>\n",
       "      <td>LinJ.24:689866:690901:-</td>\n",
       "      <td>LINF_240023900_1</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.24:690626:690901:-</td>\n",
       "      <td>nPABP-like protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4287</th>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>LINF_240023900</td>\n",
       "      <td>LinJ.24:689866:693427:-</td>\n",
       "      <td>LINF_24T0023900_2</td>\n",
       "      <td>LinJ.24:692650:693427:-</td>\n",
       "      <td>LINF_240023900_2</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.24:692650:693288:-</td>\n",
       "      <td>nPABP-like protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6000</th>\n",
       "      <td>LinJ.29</td>\n",
       "      <td>LINF_290033800</td>\n",
       "      <td>LinJ.29:1181710:1184611:+</td>\n",
       "      <td>LINF_29T0033800_1</td>\n",
       "      <td>LinJ.29:1181710:1182263:+</td>\n",
       "      <td>LINF_290033800_1</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.29:1182001:1182263:+</td>\n",
       "      <td>polynucleotide adenylyltransferase|poly(A) pol...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6001</th>\n",
       "      <td>LinJ.29</td>\n",
       "      <td>LINF_290033800</td>\n",
       "      <td>LinJ.29:1181710:1184611:+</td>\n",
       "      <td>LINF_29T0033800_2</td>\n",
       "      <td>LinJ.29:1183113:1184611:+</td>\n",
       "      <td>LINF_290033800_2</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.29:1183113:1184352:+</td>\n",
       "      <td>polynucleotide adenylyltransferase|poly(A) pol...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9840</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>LINF_360079700</td>\n",
       "      <td>LinJ.36:2681995:2683375:-</td>\n",
       "      <td>LINF_36T0079700</td>\n",
       "      <td>LinJ.36:2681395:2683433:-</td>\n",
       "      <td>LINF_360079700</td>\n",
       "      <td>SI</td>\n",
       "      <td>LinJ.36:2682378:2682647:-</td>\n",
       "      <td>SET8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 49
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We already checked all  of them except one:\n",
    "* LINF_360079700 ==> Strange case. Because the transcript goes beyond the gene coord. Well check it out later"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "### Duplicate \"Gene_ID\""
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:24:15.720167Z",
     "start_time": "2024-09-27T08:24:15.712294Z"
    }
   },
   "source": [
    "# Check elements where the \"Gene_ID\" value_counts() is greater than 1\n",
    "df_filtered['Gene_ID'].value_counts()[df_filtered['Gene_ID'].value_counts() > 1]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Gene_ID\n",
       "LINF_320037300-400     2\n",
       "LINF_320029000-100     2\n",
       "LINF_100016800         2\n",
       "LINF_290033800         2\n",
       "LINF_120014500-600     2\n",
       "LINF_260029800-900     2\n",
       "LINF_310033100-3000    2\n",
       "LINF_220018100-200     2\n",
       "LINF_070008800         2\n",
       "LINF_320030400-500     2\n",
       "LINF_100011400-500     2\n",
       "LINF_250017700-600     2\n",
       "LINF_240023900         2\n",
       "LINF_330021700-800     2\n",
       "LINF_250015300-400     2\n",
       "LINF_060017800-900     2\n",
       "LINF_270033600-700     2\n",
       "LINF_120014100-200     2\n",
       "LINF_310035100-5000    2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 50
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "All of  them are by pairs"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:24:21.026413Z",
     "start_time": "2024-09-27T08:24:21.014614Z"
    }
   },
   "source": [
    "# df_filtered['Gene_ID'].value_counts()[df_filtered['Gene_ID'].value_counts() > 1]\n",
    "df_filtered[df_filtered['Gene_ID'].value_counts()[df_filtered['Gene_ID']].values > 1]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom              Gene_ID                 Gene_coord  \\\n",
       "777   LinJ.06   LINF_060017800-900    LinJ.06:487685:490191:+   \n",
       "778   LinJ.06   LINF_060017800-900    LinJ.06:487685:490191:+   \n",
       "841   LinJ.07       LINF_070008800    LinJ.07:153281:160411:-   \n",
       "842   LinJ.07       LINF_070008800    LinJ.07:153281:160411:-   \n",
       "1376  LinJ.10   LINF_100011400-500    LinJ.10:263964:266643:+   \n",
       "1377  LinJ.10   LINF_100011400-500    LinJ.10:263964:266643:+   \n",
       "1430  LinJ.10       LINF_100016800    LinJ.10:468436:469642:-   \n",
       "1431  LinJ.10       LINF_100016800    LinJ.10:468436:469642:-   \n",
       "1748  LinJ.12   LINF_120014100-200    LinJ.12:440484:443559:+   \n",
       "1749  LinJ.12   LINF_120014100-200    LinJ.12:440484:443559:+   \n",
       "1752  LinJ.12   LINF_120014500-600    LinJ.12:450370:452072:+   \n",
       "1753  LinJ.12   LINF_120014500-600    LinJ.12:450370:452072:+   \n",
       "3774  LinJ.22   LINF_220018100-200    LinJ.22:557968:563169:+   \n",
       "3775  LinJ.22   LINF_220018100-200    LinJ.22:557968:563169:+   \n",
       "4286  LinJ.24       LINF_240023900    LinJ.24:689866:693427:-   \n",
       "4287  LinJ.24       LINF_240023900    LinJ.24:689866:693427:-   \n",
       "4469  LinJ.25   LINF_250015300-400    LinJ.25:347501:349146:+   \n",
       "4470  LinJ.25   LINF_250015300-400    LinJ.25:347501:349146:+   \n",
       "4495  LinJ.25   LINF_250017700-600    LinJ.25:451512:453695:-   \n",
       "4496  LinJ.25   LINF_250017700-600    LinJ.25:451512:453695:-   \n",
       "4917  LinJ.26   LINF_260029800-900    LinJ.26:953645:955999:+   \n",
       "4918  LinJ.26   LINF_260029800-900    LinJ.26:953645:955999:+   \n",
       "5286  LinJ.27   LINF_270033600-700  LinJ.27:1153377:1155197:+   \n",
       "5287  LinJ.27   LINF_270033600-700  LinJ.27:1153377:1155197:+   \n",
       "6000  LinJ.29       LINF_290033800  LinJ.29:1181710:1184611:+   \n",
       "6001  LinJ.29       LINF_290033800  LinJ.29:1181710:1184611:+   \n",
       "6820  LinJ.31  LINF_310033100-3000  LinJ.31:1273465:1275764:-   \n",
       "6821  LinJ.31  LINF_310033100-3000  LinJ.31:1273465:1275764:-   \n",
       "6844  LinJ.31  LINF_310035100-5000  LinJ.31:1343999:1346787:-   \n",
       "6845  LinJ.31  LINF_310035100-5000  LinJ.31:1343999:1346787:-   \n",
       "7181  LinJ.32   LINF_320029000-100    LinJ.32:889966:894771:-   \n",
       "7182  LinJ.32   LINF_320029000-100    LinJ.32:889966:894771:-   \n",
       "7195  LinJ.32   LINF_320030400-500    LinJ.32:933016:935642:-   \n",
       "7196  LinJ.32   LINF_320030400-500    LinJ.32:933016:935642:-   \n",
       "7275  LinJ.32   LINF_320037300-400  LinJ.32:1177409:1179063:+   \n",
       "7276  LinJ.32   LINF_320037300-400  LinJ.32:1177409:1179063:+   \n",
       "7568  LinJ.33   LINF_330021700-800    LinJ.33:553571:558357:+   \n",
       "7569  LinJ.33   LINF_330021700-800    LinJ.33:553571:558357:+   \n",
       "\n",
       "             Transcript_ID           Transcript_coord    CDS/miscRNA_ID  \\\n",
       "777    LINF_06T0017800-900    LinJ.06:487685:490191:+    LINF_060017800   \n",
       "778    LINF_06T0017800-900    LinJ.06:487685:490191:+    LINF_060017900   \n",
       "841      LINF_07T0008800_1    LinJ.07:153281:154334:-  LINF_070008800_1   \n",
       "842      LINF_07T0008800_2    LinJ.07:158854:160411:-  LINF_070008800_2   \n",
       "1376   LINF_10T0011400-500    LinJ.10:263964:266643:+    LINF_100011400   \n",
       "1377   LINF_10T0011400-500    LinJ.10:263964:266643:+    LINF_100011500   \n",
       "1430     LINF_10T0016800_1    LinJ.10:468436:468904:-                ND   \n",
       "1431     LINF_10T0016800_2    LinJ.10:469155:469642:-    LINF_100016800   \n",
       "1748   LINF_12T0014100-200    LinJ.12:440484:443559:+    LINF_120014100   \n",
       "1749   LINF_12T0014100-200    LinJ.12:440484:443559:+    LINF_120014200   \n",
       "1752   LINF_12T0014500-600    LinJ.12:450370:452072:+    LINF_120014500   \n",
       "1753   LINF_12T0014500-600    LinJ.12:450370:452072:+    LINF_120014600   \n",
       "3774   LINF_22T0018100-200    LinJ.22:557968:563169:+    LINF_220018100   \n",
       "3775   LINF_22T0018100-200    LinJ.22:557968:563169:+    LINF_220018200   \n",
       "4286     LINF_24T0023900_1    LinJ.24:689866:690901:-  LINF_240023900_1   \n",
       "4287     LINF_24T0023900_2    LinJ.24:692650:693427:-  LINF_240023900_2   \n",
       "4469   LINF_25T0015300-400    LinJ.25:347501:349146:+    LINF_250015300   \n",
       "4470   LINF_25T0015300-400    LinJ.25:347501:349146:+    LINF_250015400   \n",
       "4495   LINF_25T0017700-600    LinJ.25:451512:453695:-    LINF_250017600   \n",
       "4496   LINF_25T0017700-600    LinJ.25:451512:453695:-    LINF_250017700   \n",
       "4917   LINF_26T0029800-900    LinJ.26:953645:955999:+    LINF_260029800   \n",
       "4918   LINF_26T0029800-900    LinJ.26:953645:955999:+    LINF_260029900   \n",
       "5286   LINF_27T0033600-700  LinJ.27:1153377:1155197:+    LINF_270033600   \n",
       "5287   LINF_27T0033600-700  LinJ.27:1153377:1155197:+    LINF_270033700   \n",
       "6000     LINF_29T0033800_1  LinJ.29:1181710:1182263:+  LINF_290033800_1   \n",
       "6001     LINF_29T0033800_2  LinJ.29:1183113:1184611:+  LINF_290033800_2   \n",
       "6820  LINF_31T0033100-3000  LinJ.31:1273465:1275764:-    LINF_310033000   \n",
       "6821  LINF_31T0033100-3000  LinJ.31:1273465:1275764:-    LINF_310033100   \n",
       "6844  LINF_31T0035100-5000  LinJ.31:1343999:1346787:-    LINF_310035000   \n",
       "6845  LINF_31T0035100-5000  LinJ.31:1343999:1346787:-    LINF_310035100   \n",
       "7181   LINF_32T0029000-100    LinJ.32:889966:894771:-    LINF_320029000   \n",
       "7182   LINF_32T0029000-100    LinJ.32:889966:894771:-    LINF_320029100   \n",
       "7195   LINF_32T0030400-500    LinJ.32:933016:935642:-    LINF_320030400   \n",
       "7196   LINF_32T0030400-500    LinJ.32:933016:935642:-    LINF_320030500   \n",
       "7275   LINF_32T0037300-400  LinJ.32:1177409:1179063:+    LINF_320037300   \n",
       "7276   LINF_32T0037300-400  LinJ.32:1177409:1179063:+    LINF_320037400   \n",
       "7568   LINF_33T0021700-800    LinJ.33:553571:558357:+    LINF_330021700   \n",
       "7569   LINF_33T0021700-800    LinJ.33:553571:558357:+    LINF_330021800   \n",
       "\n",
       "     Pseudogen          CDS/miscRNA_coord  \\\n",
       "777         NO    LinJ.06:487818:488159:+   \n",
       "778         NO    LinJ.06:488373:488633:+   \n",
       "841         NO    LinJ.07:153994:154334:-   \n",
       "842         NO    LinJ.07:158854:160240:-   \n",
       "1376        NO    LinJ.10:264569:264949:+   \n",
       "1377        NO    LinJ.10:264981:265235:+   \n",
       "1430        NO                         ND   \n",
       "1431        NO    LinJ.10:469169:469561:-   \n",
       "1748        NO    LinJ.12:440748:441215:+   \n",
       "1749        NO    LinJ.12:441320:441805:+   \n",
       "1752        NO    LinJ.12:450545:451012:+   \n",
       "1753        NO    LinJ.12:451117:451827:+   \n",
       "3774        NO    LinJ.22:558160:559113:+   \n",
       "3775        NO    LinJ.22:559255:563064:+   \n",
       "4286        NO    LinJ.24:690626:690901:-   \n",
       "4287        NO    LinJ.24:692650:693288:-   \n",
       "4469        NO    LinJ.25:347718:348536:+   \n",
       "4470        NO    LinJ.25:348604:349041:+   \n",
       "4495        NO    LinJ.25:451989:452591:-   \n",
       "4496        NO    LinJ.25:452600:453340:-   \n",
       "4917        NO    LinJ.26:953976:954413:+   \n",
       "4918        NO    LinJ.26:954522:954851:+   \n",
       "5286        NO  LinJ.27:1153523:1154887:+   \n",
       "5287        NO  LinJ.27:1153982:1154887:+   \n",
       "6000        NO  LinJ.29:1182001:1182263:+   \n",
       "6001        NO  LinJ.29:1183113:1184352:+   \n",
       "6820        NO  LinJ.31:1274043:1274432:-   \n",
       "6821        NO  LinJ.31:1275097:1275579:-   \n",
       "6844        NO  LinJ.31:1344181:1344771:-   \n",
       "6845        NO  LinJ.31:1344807:1345379:-   \n",
       "7181        NO    LinJ.32:890313:890891:-   \n",
       "7182        NO    LinJ.32:890951:893725:-   \n",
       "7195        NO    LinJ.32:934389:934805:-   \n",
       "7196        NO    LinJ.32:934940:935455:-   \n",
       "7275        NO  LinJ.32:1177678:1178130:+   \n",
       "7276        NO  LinJ.32:1178292:1178738:+   \n",
       "7568        NO    LinJ.33:555988:556314:+   \n",
       "7569        NO    LinJ.33:557286:557831:+   \n",
       "\n",
       "                                               Function  \n",
       "777                                hypothetical_protein  \n",
       "778                                hypothetical_protein  \n",
       "841                  ATP-dependent RNA helicase (DBP2B)  \n",
       "842                  ATP-dependent RNA helicase (DBP2B)  \n",
       "1376                   hypothetical_protein_-_conserved  \n",
       "1377                               hypothetical_protein  \n",
       "1430                                         histone H3  \n",
       "1431                                         histone H3  \n",
       "1748                   hypothetical_protein_-_conserved  \n",
       "1749                   hypothetical_protein_-_conserved  \n",
       "1752                   hypothetical_protein_-_conserved  \n",
       "1753                   hypothetical_protein_-_conserved  \n",
       "3774                               hypothetical_protein  \n",
       "3775               WD_domain_-_G-beta_repeat_-_putative  \n",
       "4286                                 nPABP-like protein  \n",
       "4287                                 nPABP-like protein  \n",
       "4469                   hypothetical_protein_-_conserved  \n",
       "4470                               hypothetical_protein  \n",
       "4495                               hypothetical_protein  \n",
       "4496       alpha-tubulin_N-acetyltransferase_-_putative  \n",
       "4917   Lys-63-specific_deubiquitinase_BRCC36_-_putative  \n",
       "4918                               hypothetical_protein  \n",
       "5286                   hypothetical_protein_-_conserved  \n",
       "5287                   hypothetical_protein_-_conserved  \n",
       "6000  polynucleotide adenylyltransferase|poly(A) pol...  \n",
       "6001  polynucleotide adenylyltransferase|poly(A) pol...  \n",
       "6820                        uridine_kinase-like_protein  \n",
       "6821                     ferredoxin_2fe-2s-like_protein  \n",
       "6844                               hypothetical_protein  \n",
       "6845                   hypothetical_protein_-_conserved  \n",
       "7181                               hypothetical_protein  \n",
       "7182           membrane_associated_protein-like_protein  \n",
       "7195                               hypothetical_protein  \n",
       "7196                   hypothetical_protein_-_conserved  \n",
       "7275        Domain_of_unknown_function_DUF21_-_putative  \n",
       "7276                               hypothetical_protein  \n",
       "7568                               hypothetical_protein  \n",
       "7569                               hypothetical_protein  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>777</th>\n",
       "      <td>LinJ.06</td>\n",
       "      <td>LINF_060017800-900</td>\n",
       "      <td>LinJ.06:487685:490191:+</td>\n",
       "      <td>LINF_06T0017800-900</td>\n",
       "      <td>LinJ.06:487685:490191:+</td>\n",
       "      <td>LINF_060017800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.06:487818:488159:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>778</th>\n",
       "      <td>LinJ.06</td>\n",
       "      <td>LINF_060017800-900</td>\n",
       "      <td>LinJ.06:487685:490191:+</td>\n",
       "      <td>LINF_06T0017800-900</td>\n",
       "      <td>LinJ.06:487685:490191:+</td>\n",
       "      <td>LINF_060017900</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.06:488373:488633:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>841</th>\n",
       "      <td>LinJ.07</td>\n",
       "      <td>LINF_070008800</td>\n",
       "      <td>LinJ.07:153281:160411:-</td>\n",
       "      <td>LINF_07T0008800_1</td>\n",
       "      <td>LinJ.07:153281:154334:-</td>\n",
       "      <td>LINF_070008800_1</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.07:153994:154334:-</td>\n",
       "      <td>ATP-dependent RNA helicase (DBP2B)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>842</th>\n",
       "      <td>LinJ.07</td>\n",
       "      <td>LINF_070008800</td>\n",
       "      <td>LinJ.07:153281:160411:-</td>\n",
       "      <td>LINF_07T0008800_2</td>\n",
       "      <td>LinJ.07:158854:160411:-</td>\n",
       "      <td>LINF_070008800_2</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.07:158854:160240:-</td>\n",
       "      <td>ATP-dependent RNA helicase (DBP2B)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1376</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100011400-500</td>\n",
       "      <td>LinJ.10:263964:266643:+</td>\n",
       "      <td>LINF_10T0011400-500</td>\n",
       "      <td>LinJ.10:263964:266643:+</td>\n",
       "      <td>LINF_100011400</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.10:264569:264949:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1377</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100011400-500</td>\n",
       "      <td>LinJ.10:263964:266643:+</td>\n",
       "      <td>LINF_10T0011400-500</td>\n",
       "      <td>LinJ.10:263964:266643:+</td>\n",
       "      <td>LINF_100011500</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.10:264981:265235:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1430</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>LinJ.10:468436:469642:-</td>\n",
       "      <td>LINF_10T0016800_1</td>\n",
       "      <td>LinJ.10:468436:468904:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>histone H3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1431</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>LinJ.10:468436:469642:-</td>\n",
       "      <td>LINF_10T0016800_2</td>\n",
       "      <td>LinJ.10:469155:469642:-</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.10:469169:469561:-</td>\n",
       "      <td>histone H3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1748</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>LINF_120014100-200</td>\n",
       "      <td>LinJ.12:440484:443559:+</td>\n",
       "      <td>LINF_12T0014100-200</td>\n",
       "      <td>LinJ.12:440484:443559:+</td>\n",
       "      <td>LINF_120014100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.12:440748:441215:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1749</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>LINF_120014100-200</td>\n",
       "      <td>LinJ.12:440484:443559:+</td>\n",
       "      <td>LINF_12T0014100-200</td>\n",
       "      <td>LinJ.12:440484:443559:+</td>\n",
       "      <td>LINF_120014200</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.12:441320:441805:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1752</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>LINF_120014500-600</td>\n",
       "      <td>LinJ.12:450370:452072:+</td>\n",
       "      <td>LINF_12T0014500-600</td>\n",
       "      <td>LinJ.12:450370:452072:+</td>\n",
       "      <td>LINF_120014500</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.12:450545:451012:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1753</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>LINF_120014500-600</td>\n",
       "      <td>LinJ.12:450370:452072:+</td>\n",
       "      <td>LINF_12T0014500-600</td>\n",
       "      <td>LinJ.12:450370:452072:+</td>\n",
       "      <td>LINF_120014600</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.12:451117:451827:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3774</th>\n",
       "      <td>LinJ.22</td>\n",
       "      <td>LINF_220018100-200</td>\n",
       "      <td>LinJ.22:557968:563169:+</td>\n",
       "      <td>LINF_22T0018100-200</td>\n",
       "      <td>LinJ.22:557968:563169:+</td>\n",
       "      <td>LINF_220018100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.22:558160:559113:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3775</th>\n",
       "      <td>LinJ.22</td>\n",
       "      <td>LINF_220018100-200</td>\n",
       "      <td>LinJ.22:557968:563169:+</td>\n",
       "      <td>LINF_22T0018100-200</td>\n",
       "      <td>LinJ.22:557968:563169:+</td>\n",
       "      <td>LINF_220018200</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.22:559255:563064:+</td>\n",
       "      <td>WD_domain_-_G-beta_repeat_-_putative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4286</th>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>LINF_240023900</td>\n",
       "      <td>LinJ.24:689866:693427:-</td>\n",
       "      <td>LINF_24T0023900_1</td>\n",
       "      <td>LinJ.24:689866:690901:-</td>\n",
       "      <td>LINF_240023900_1</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.24:690626:690901:-</td>\n",
       "      <td>nPABP-like protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4287</th>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>LINF_240023900</td>\n",
       "      <td>LinJ.24:689866:693427:-</td>\n",
       "      <td>LINF_24T0023900_2</td>\n",
       "      <td>LinJ.24:692650:693427:-</td>\n",
       "      <td>LINF_240023900_2</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.24:692650:693288:-</td>\n",
       "      <td>nPABP-like protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4469</th>\n",
       "      <td>LinJ.25</td>\n",
       "      <td>LINF_250015300-400</td>\n",
       "      <td>LinJ.25:347501:349146:+</td>\n",
       "      <td>LINF_25T0015300-400</td>\n",
       "      <td>LinJ.25:347501:349146:+</td>\n",
       "      <td>LINF_250015300</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.25:347718:348536:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4470</th>\n",
       "      <td>LinJ.25</td>\n",
       "      <td>LINF_250015300-400</td>\n",
       "      <td>LinJ.25:347501:349146:+</td>\n",
       "      <td>LINF_25T0015300-400</td>\n",
       "      <td>LinJ.25:347501:349146:+</td>\n",
       "      <td>LINF_250015400</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.25:348604:349041:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4495</th>\n",
       "      <td>LinJ.25</td>\n",
       "      <td>LINF_250017700-600</td>\n",
       "      <td>LinJ.25:451512:453695:-</td>\n",
       "      <td>LINF_25T0017700-600</td>\n",
       "      <td>LinJ.25:451512:453695:-</td>\n",
       "      <td>LINF_250017600</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.25:451989:452591:-</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4496</th>\n",
       "      <td>LinJ.25</td>\n",
       "      <td>LINF_250017700-600</td>\n",
       "      <td>LinJ.25:451512:453695:-</td>\n",
       "      <td>LINF_25T0017700-600</td>\n",
       "      <td>LinJ.25:451512:453695:-</td>\n",
       "      <td>LINF_250017700</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.25:452600:453340:-</td>\n",
       "      <td>alpha-tubulin_N-acetyltransferase_-_putative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4917</th>\n",
       "      <td>LinJ.26</td>\n",
       "      <td>LINF_260029800-900</td>\n",
       "      <td>LinJ.26:953645:955999:+</td>\n",
       "      <td>LINF_26T0029800-900</td>\n",
       "      <td>LinJ.26:953645:955999:+</td>\n",
       "      <td>LINF_260029800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.26:953976:954413:+</td>\n",
       "      <td>Lys-63-specific_deubiquitinase_BRCC36_-_putative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4918</th>\n",
       "      <td>LinJ.26</td>\n",
       "      <td>LINF_260029800-900</td>\n",
       "      <td>LinJ.26:953645:955999:+</td>\n",
       "      <td>LINF_26T0029800-900</td>\n",
       "      <td>LinJ.26:953645:955999:+</td>\n",
       "      <td>LINF_260029900</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.26:954522:954851:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5286</th>\n",
       "      <td>LinJ.27</td>\n",
       "      <td>LINF_270033600-700</td>\n",
       "      <td>LinJ.27:1153377:1155197:+</td>\n",
       "      <td>LINF_27T0033600-700</td>\n",
       "      <td>LinJ.27:1153377:1155197:+</td>\n",
       "      <td>LINF_270033600</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.27:1153523:1154887:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5287</th>\n",
       "      <td>LinJ.27</td>\n",
       "      <td>LINF_270033600-700</td>\n",
       "      <td>LinJ.27:1153377:1155197:+</td>\n",
       "      <td>LINF_27T0033600-700</td>\n",
       "      <td>LinJ.27:1153377:1155197:+</td>\n",
       "      <td>LINF_270033700</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.27:1153982:1154887:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6000</th>\n",
       "      <td>LinJ.29</td>\n",
       "      <td>LINF_290033800</td>\n",
       "      <td>LinJ.29:1181710:1184611:+</td>\n",
       "      <td>LINF_29T0033800_1</td>\n",
       "      <td>LinJ.29:1181710:1182263:+</td>\n",
       "      <td>LINF_290033800_1</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.29:1182001:1182263:+</td>\n",
       "      <td>polynucleotide adenylyltransferase|poly(A) pol...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6001</th>\n",
       "      <td>LinJ.29</td>\n",
       "      <td>LINF_290033800</td>\n",
       "      <td>LinJ.29:1181710:1184611:+</td>\n",
       "      <td>LINF_29T0033800_2</td>\n",
       "      <td>LinJ.29:1183113:1184611:+</td>\n",
       "      <td>LINF_290033800_2</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.29:1183113:1184352:+</td>\n",
       "      <td>polynucleotide adenylyltransferase|poly(A) pol...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6820</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>LINF_310033100-3000</td>\n",
       "      <td>LinJ.31:1273465:1275764:-</td>\n",
       "      <td>LINF_31T0033100-3000</td>\n",
       "      <td>LinJ.31:1273465:1275764:-</td>\n",
       "      <td>LINF_310033000</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.31:1274043:1274432:-</td>\n",
       "      <td>uridine_kinase-like_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6821</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>LINF_310033100-3000</td>\n",
       "      <td>LinJ.31:1273465:1275764:-</td>\n",
       "      <td>LINF_31T0033100-3000</td>\n",
       "      <td>LinJ.31:1273465:1275764:-</td>\n",
       "      <td>LINF_310033100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.31:1275097:1275579:-</td>\n",
       "      <td>ferredoxin_2fe-2s-like_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6844</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>LINF_310035100-5000</td>\n",
       "      <td>LinJ.31:1343999:1346787:-</td>\n",
       "      <td>LINF_31T0035100-5000</td>\n",
       "      <td>LinJ.31:1343999:1346787:-</td>\n",
       "      <td>LINF_310035000</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.31:1344181:1344771:-</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6845</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>LINF_310035100-5000</td>\n",
       "      <td>LinJ.31:1343999:1346787:-</td>\n",
       "      <td>LINF_31T0035100-5000</td>\n",
       "      <td>LinJ.31:1343999:1346787:-</td>\n",
       "      <td>LINF_310035100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.31:1344807:1345379:-</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7181</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320029000-100</td>\n",
       "      <td>LinJ.32:889966:894771:-</td>\n",
       "      <td>LINF_32T0029000-100</td>\n",
       "      <td>LinJ.32:889966:894771:-</td>\n",
       "      <td>LINF_320029000</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:890313:890891:-</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7182</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320029000-100</td>\n",
       "      <td>LinJ.32:889966:894771:-</td>\n",
       "      <td>LINF_32T0029000-100</td>\n",
       "      <td>LinJ.32:889966:894771:-</td>\n",
       "      <td>LINF_320029100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:890951:893725:-</td>\n",
       "      <td>membrane_associated_protein-like_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7195</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320030400-500</td>\n",
       "      <td>LinJ.32:933016:935642:-</td>\n",
       "      <td>LINF_32T0030400-500</td>\n",
       "      <td>LinJ.32:933016:935642:-</td>\n",
       "      <td>LINF_320030400</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:934389:934805:-</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7196</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320030400-500</td>\n",
       "      <td>LinJ.32:933016:935642:-</td>\n",
       "      <td>LINF_32T0030400-500</td>\n",
       "      <td>LinJ.32:933016:935642:-</td>\n",
       "      <td>LINF_320030500</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:934940:935455:-</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7275</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320037300-400</td>\n",
       "      <td>LinJ.32:1177409:1179063:+</td>\n",
       "      <td>LINF_32T0037300-400</td>\n",
       "      <td>LinJ.32:1177409:1179063:+</td>\n",
       "      <td>LINF_320037300</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:1177678:1178130:+</td>\n",
       "      <td>Domain_of_unknown_function_DUF21_-_putative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7276</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320037300-400</td>\n",
       "      <td>LinJ.32:1177409:1179063:+</td>\n",
       "      <td>LINF_32T0037300-400</td>\n",
       "      <td>LinJ.32:1177409:1179063:+</td>\n",
       "      <td>LINF_320037400</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:1178292:1178738:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7568</th>\n",
       "      <td>LinJ.33</td>\n",
       "      <td>LINF_330021700-800</td>\n",
       "      <td>LinJ.33:553571:558357:+</td>\n",
       "      <td>LINF_33T0021700-800</td>\n",
       "      <td>LinJ.33:553571:558357:+</td>\n",
       "      <td>LINF_330021700</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.33:555988:556314:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7569</th>\n",
       "      <td>LinJ.33</td>\n",
       "      <td>LINF_330021700-800</td>\n",
       "      <td>LinJ.33:553571:558357:+</td>\n",
       "      <td>LINF_33T0021700-800</td>\n",
       "      <td>LinJ.33:553571:558357:+</td>\n",
       "      <td>LINF_330021800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.33:557286:557831:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 51
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can see that some \"Transcript_ID\" are divided by _1 and _2, but not all. So the ones that do not end in \"_1\" or \"_2\" we should change them"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:24:51.528427Z",
     "start_time": "2024-09-27T08:24:51.509581Z"
    }
   },
   "source": [
    "# Let's start testing\n",
    "duplicated_gene_id_df = df_filtered[df_filtered['Gene_ID'].value_counts()[df_filtered['Gene_ID']].values > 1].copy()\n",
    "print(duplicated_gene_id_df.shape)\n",
    "duplicated_gene_id_df"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(38, 9)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "        Chrom              Gene_ID                 Gene_coord  \\\n",
       "777   LinJ.06   LINF_060017800-900    LinJ.06:487685:490191:+   \n",
       "778   LinJ.06   LINF_060017800-900    LinJ.06:487685:490191:+   \n",
       "841   LinJ.07       LINF_070008800    LinJ.07:153281:160411:-   \n",
       "842   LinJ.07       LINF_070008800    LinJ.07:153281:160411:-   \n",
       "1376  LinJ.10   LINF_100011400-500    LinJ.10:263964:266643:+   \n",
       "1377  LinJ.10   LINF_100011400-500    LinJ.10:263964:266643:+   \n",
       "1430  LinJ.10       LINF_100016800    LinJ.10:468436:469642:-   \n",
       "1431  LinJ.10       LINF_100016800    LinJ.10:468436:469642:-   \n",
       "1748  LinJ.12   LINF_120014100-200    LinJ.12:440484:443559:+   \n",
       "1749  LinJ.12   LINF_120014100-200    LinJ.12:440484:443559:+   \n",
       "1752  LinJ.12   LINF_120014500-600    LinJ.12:450370:452072:+   \n",
       "1753  LinJ.12   LINF_120014500-600    LinJ.12:450370:452072:+   \n",
       "3774  LinJ.22   LINF_220018100-200    LinJ.22:557968:563169:+   \n",
       "3775  LinJ.22   LINF_220018100-200    LinJ.22:557968:563169:+   \n",
       "4286  LinJ.24       LINF_240023900    LinJ.24:689866:693427:-   \n",
       "4287  LinJ.24       LINF_240023900    LinJ.24:689866:693427:-   \n",
       "4469  LinJ.25   LINF_250015300-400    LinJ.25:347501:349146:+   \n",
       "4470  LinJ.25   LINF_250015300-400    LinJ.25:347501:349146:+   \n",
       "4495  LinJ.25   LINF_250017700-600    LinJ.25:451512:453695:-   \n",
       "4496  LinJ.25   LINF_250017700-600    LinJ.25:451512:453695:-   \n",
       "4917  LinJ.26   LINF_260029800-900    LinJ.26:953645:955999:+   \n",
       "4918  LinJ.26   LINF_260029800-900    LinJ.26:953645:955999:+   \n",
       "5286  LinJ.27   LINF_270033600-700  LinJ.27:1153377:1155197:+   \n",
       "5287  LinJ.27   LINF_270033600-700  LinJ.27:1153377:1155197:+   \n",
       "6000  LinJ.29       LINF_290033800  LinJ.29:1181710:1184611:+   \n",
       "6001  LinJ.29       LINF_290033800  LinJ.29:1181710:1184611:+   \n",
       "6820  LinJ.31  LINF_310033100-3000  LinJ.31:1273465:1275764:-   \n",
       "6821  LinJ.31  LINF_310033100-3000  LinJ.31:1273465:1275764:-   \n",
       "6844  LinJ.31  LINF_310035100-5000  LinJ.31:1343999:1346787:-   \n",
       "6845  LinJ.31  LINF_310035100-5000  LinJ.31:1343999:1346787:-   \n",
       "7181  LinJ.32   LINF_320029000-100    LinJ.32:889966:894771:-   \n",
       "7182  LinJ.32   LINF_320029000-100    LinJ.32:889966:894771:-   \n",
       "7195  LinJ.32   LINF_320030400-500    LinJ.32:933016:935642:-   \n",
       "7196  LinJ.32   LINF_320030400-500    LinJ.32:933016:935642:-   \n",
       "7275  LinJ.32   LINF_320037300-400  LinJ.32:1177409:1179063:+   \n",
       "7276  LinJ.32   LINF_320037300-400  LinJ.32:1177409:1179063:+   \n",
       "7568  LinJ.33   LINF_330021700-800    LinJ.33:553571:558357:+   \n",
       "7569  LinJ.33   LINF_330021700-800    LinJ.33:553571:558357:+   \n",
       "\n",
       "             Transcript_ID           Transcript_coord    CDS/miscRNA_ID  \\\n",
       "777    LINF_06T0017800-900    LinJ.06:487685:490191:+    LINF_060017800   \n",
       "778    LINF_06T0017800-900    LinJ.06:487685:490191:+    LINF_060017900   \n",
       "841      LINF_07T0008800_1    LinJ.07:153281:154334:-  LINF_070008800_1   \n",
       "842      LINF_07T0008800_2    LinJ.07:158854:160411:-  LINF_070008800_2   \n",
       "1376   LINF_10T0011400-500    LinJ.10:263964:266643:+    LINF_100011400   \n",
       "1377   LINF_10T0011400-500    LinJ.10:263964:266643:+    LINF_100011500   \n",
       "1430     LINF_10T0016800_1    LinJ.10:468436:468904:-                ND   \n",
       "1431     LINF_10T0016800_2    LinJ.10:469155:469642:-    LINF_100016800   \n",
       "1748   LINF_12T0014100-200    LinJ.12:440484:443559:+    LINF_120014100   \n",
       "1749   LINF_12T0014100-200    LinJ.12:440484:443559:+    LINF_120014200   \n",
       "1752   LINF_12T0014500-600    LinJ.12:450370:452072:+    LINF_120014500   \n",
       "1753   LINF_12T0014500-600    LinJ.12:450370:452072:+    LINF_120014600   \n",
       "3774   LINF_22T0018100-200    LinJ.22:557968:563169:+    LINF_220018100   \n",
       "3775   LINF_22T0018100-200    LinJ.22:557968:563169:+    LINF_220018200   \n",
       "4286     LINF_24T0023900_1    LinJ.24:689866:690901:-  LINF_240023900_1   \n",
       "4287     LINF_24T0023900_2    LinJ.24:692650:693427:-  LINF_240023900_2   \n",
       "4469   LINF_25T0015300-400    LinJ.25:347501:349146:+    LINF_250015300   \n",
       "4470   LINF_25T0015300-400    LinJ.25:347501:349146:+    LINF_250015400   \n",
       "4495   LINF_25T0017700-600    LinJ.25:451512:453695:-    LINF_250017600   \n",
       "4496   LINF_25T0017700-600    LinJ.25:451512:453695:-    LINF_250017700   \n",
       "4917   LINF_26T0029800-900    LinJ.26:953645:955999:+    LINF_260029800   \n",
       "4918   LINF_26T0029800-900    LinJ.26:953645:955999:+    LINF_260029900   \n",
       "5286   LINF_27T0033600-700  LinJ.27:1153377:1155197:+    LINF_270033600   \n",
       "5287   LINF_27T0033600-700  LinJ.27:1153377:1155197:+    LINF_270033700   \n",
       "6000     LINF_29T0033800_1  LinJ.29:1181710:1182263:+  LINF_290033800_1   \n",
       "6001     LINF_29T0033800_2  LinJ.29:1183113:1184611:+  LINF_290033800_2   \n",
       "6820  LINF_31T0033100-3000  LinJ.31:1273465:1275764:-    LINF_310033000   \n",
       "6821  LINF_31T0033100-3000  LinJ.31:1273465:1275764:-    LINF_310033100   \n",
       "6844  LINF_31T0035100-5000  LinJ.31:1343999:1346787:-    LINF_310035000   \n",
       "6845  LINF_31T0035100-5000  LinJ.31:1343999:1346787:-    LINF_310035100   \n",
       "7181   LINF_32T0029000-100    LinJ.32:889966:894771:-    LINF_320029000   \n",
       "7182   LINF_32T0029000-100    LinJ.32:889966:894771:-    LINF_320029100   \n",
       "7195   LINF_32T0030400-500    LinJ.32:933016:935642:-    LINF_320030400   \n",
       "7196   LINF_32T0030400-500    LinJ.32:933016:935642:-    LINF_320030500   \n",
       "7275   LINF_32T0037300-400  LinJ.32:1177409:1179063:+    LINF_320037300   \n",
       "7276   LINF_32T0037300-400  LinJ.32:1177409:1179063:+    LINF_320037400   \n",
       "7568   LINF_33T0021700-800    LinJ.33:553571:558357:+    LINF_330021700   \n",
       "7569   LINF_33T0021700-800    LinJ.33:553571:558357:+    LINF_330021800   \n",
       "\n",
       "     Pseudogen          CDS/miscRNA_coord  \\\n",
       "777         NO    LinJ.06:487818:488159:+   \n",
       "778         NO    LinJ.06:488373:488633:+   \n",
       "841         NO    LinJ.07:153994:154334:-   \n",
       "842         NO    LinJ.07:158854:160240:-   \n",
       "1376        NO    LinJ.10:264569:264949:+   \n",
       "1377        NO    LinJ.10:264981:265235:+   \n",
       "1430        NO                         ND   \n",
       "1431        NO    LinJ.10:469169:469561:-   \n",
       "1748        NO    LinJ.12:440748:441215:+   \n",
       "1749        NO    LinJ.12:441320:441805:+   \n",
       "1752        NO    LinJ.12:450545:451012:+   \n",
       "1753        NO    LinJ.12:451117:451827:+   \n",
       "3774        NO    LinJ.22:558160:559113:+   \n",
       "3775        NO    LinJ.22:559255:563064:+   \n",
       "4286        NO    LinJ.24:690626:690901:-   \n",
       "4287        NO    LinJ.24:692650:693288:-   \n",
       "4469        NO    LinJ.25:347718:348536:+   \n",
       "4470        NO    LinJ.25:348604:349041:+   \n",
       "4495        NO    LinJ.25:451989:452591:-   \n",
       "4496        NO    LinJ.25:452600:453340:-   \n",
       "4917        NO    LinJ.26:953976:954413:+   \n",
       "4918        NO    LinJ.26:954522:954851:+   \n",
       "5286        NO  LinJ.27:1153523:1154887:+   \n",
       "5287        NO  LinJ.27:1153982:1154887:+   \n",
       "6000        NO  LinJ.29:1182001:1182263:+   \n",
       "6001        NO  LinJ.29:1183113:1184352:+   \n",
       "6820        NO  LinJ.31:1274043:1274432:-   \n",
       "6821        NO  LinJ.31:1275097:1275579:-   \n",
       "6844        NO  LinJ.31:1344181:1344771:-   \n",
       "6845        NO  LinJ.31:1344807:1345379:-   \n",
       "7181        NO    LinJ.32:890313:890891:-   \n",
       "7182        NO    LinJ.32:890951:893725:-   \n",
       "7195        NO    LinJ.32:934389:934805:-   \n",
       "7196        NO    LinJ.32:934940:935455:-   \n",
       "7275        NO  LinJ.32:1177678:1178130:+   \n",
       "7276        NO  LinJ.32:1178292:1178738:+   \n",
       "7568        NO    LinJ.33:555988:556314:+   \n",
       "7569        NO    LinJ.33:557286:557831:+   \n",
       "\n",
       "                                               Function  \n",
       "777                                hypothetical_protein  \n",
       "778                                hypothetical_protein  \n",
       "841                  ATP-dependent RNA helicase (DBP2B)  \n",
       "842                  ATP-dependent RNA helicase (DBP2B)  \n",
       "1376                   hypothetical_protein_-_conserved  \n",
       "1377                               hypothetical_protein  \n",
       "1430                                         histone H3  \n",
       "1431                                         histone H3  \n",
       "1748                   hypothetical_protein_-_conserved  \n",
       "1749                   hypothetical_protein_-_conserved  \n",
       "1752                   hypothetical_protein_-_conserved  \n",
       "1753                   hypothetical_protein_-_conserved  \n",
       "3774                               hypothetical_protein  \n",
       "3775               WD_domain_-_G-beta_repeat_-_putative  \n",
       "4286                                 nPABP-like protein  \n",
       "4287                                 nPABP-like protein  \n",
       "4469                   hypothetical_protein_-_conserved  \n",
       "4470                               hypothetical_protein  \n",
       "4495                               hypothetical_protein  \n",
       "4496       alpha-tubulin_N-acetyltransferase_-_putative  \n",
       "4917   Lys-63-specific_deubiquitinase_BRCC36_-_putative  \n",
       "4918                               hypothetical_protein  \n",
       "5286                   hypothetical_protein_-_conserved  \n",
       "5287                   hypothetical_protein_-_conserved  \n",
       "6000  polynucleotide adenylyltransferase|poly(A) pol...  \n",
       "6001  polynucleotide adenylyltransferase|poly(A) pol...  \n",
       "6820                        uridine_kinase-like_protein  \n",
       "6821                     ferredoxin_2fe-2s-like_protein  \n",
       "6844                               hypothetical_protein  \n",
       "6845                   hypothetical_protein_-_conserved  \n",
       "7181                               hypothetical_protein  \n",
       "7182           membrane_associated_protein-like_protein  \n",
       "7195                               hypothetical_protein  \n",
       "7196                   hypothetical_protein_-_conserved  \n",
       "7275        Domain_of_unknown_function_DUF21_-_putative  \n",
       "7276                               hypothetical_protein  \n",
       "7568                               hypothetical_protein  \n",
       "7569                               hypothetical_protein  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>777</th>\n",
       "      <td>LinJ.06</td>\n",
       "      <td>LINF_060017800-900</td>\n",
       "      <td>LinJ.06:487685:490191:+</td>\n",
       "      <td>LINF_06T0017800-900</td>\n",
       "      <td>LinJ.06:487685:490191:+</td>\n",
       "      <td>LINF_060017800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.06:487818:488159:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>778</th>\n",
       "      <td>LinJ.06</td>\n",
       "      <td>LINF_060017800-900</td>\n",
       "      <td>LinJ.06:487685:490191:+</td>\n",
       "      <td>LINF_06T0017800-900</td>\n",
       "      <td>LinJ.06:487685:490191:+</td>\n",
       "      <td>LINF_060017900</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.06:488373:488633:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>841</th>\n",
       "      <td>LinJ.07</td>\n",
       "      <td>LINF_070008800</td>\n",
       "      <td>LinJ.07:153281:160411:-</td>\n",
       "      <td>LINF_07T0008800_1</td>\n",
       "      <td>LinJ.07:153281:154334:-</td>\n",
       "      <td>LINF_070008800_1</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.07:153994:154334:-</td>\n",
       "      <td>ATP-dependent RNA helicase (DBP2B)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>842</th>\n",
       "      <td>LinJ.07</td>\n",
       "      <td>LINF_070008800</td>\n",
       "      <td>LinJ.07:153281:160411:-</td>\n",
       "      <td>LINF_07T0008800_2</td>\n",
       "      <td>LinJ.07:158854:160411:-</td>\n",
       "      <td>LINF_070008800_2</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.07:158854:160240:-</td>\n",
       "      <td>ATP-dependent RNA helicase (DBP2B)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1376</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100011400-500</td>\n",
       "      <td>LinJ.10:263964:266643:+</td>\n",
       "      <td>LINF_10T0011400-500</td>\n",
       "      <td>LinJ.10:263964:266643:+</td>\n",
       "      <td>LINF_100011400</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.10:264569:264949:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1377</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100011400-500</td>\n",
       "      <td>LinJ.10:263964:266643:+</td>\n",
       "      <td>LINF_10T0011400-500</td>\n",
       "      <td>LinJ.10:263964:266643:+</td>\n",
       "      <td>LINF_100011500</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.10:264981:265235:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1430</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>LinJ.10:468436:469642:-</td>\n",
       "      <td>LINF_10T0016800_1</td>\n",
       "      <td>LinJ.10:468436:468904:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>histone H3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1431</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>LinJ.10:468436:469642:-</td>\n",
       "      <td>LINF_10T0016800_2</td>\n",
       "      <td>LinJ.10:469155:469642:-</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.10:469169:469561:-</td>\n",
       "      <td>histone H3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1748</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>LINF_120014100-200</td>\n",
       "      <td>LinJ.12:440484:443559:+</td>\n",
       "      <td>LINF_12T0014100-200</td>\n",
       "      <td>LinJ.12:440484:443559:+</td>\n",
       "      <td>LINF_120014100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.12:440748:441215:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1749</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>LINF_120014100-200</td>\n",
       "      <td>LinJ.12:440484:443559:+</td>\n",
       "      <td>LINF_12T0014100-200</td>\n",
       "      <td>LinJ.12:440484:443559:+</td>\n",
       "      <td>LINF_120014200</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.12:441320:441805:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1752</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>LINF_120014500-600</td>\n",
       "      <td>LinJ.12:450370:452072:+</td>\n",
       "      <td>LINF_12T0014500-600</td>\n",
       "      <td>LinJ.12:450370:452072:+</td>\n",
       "      <td>LINF_120014500</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.12:450545:451012:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1753</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>LINF_120014500-600</td>\n",
       "      <td>LinJ.12:450370:452072:+</td>\n",
       "      <td>LINF_12T0014500-600</td>\n",
       "      <td>LinJ.12:450370:452072:+</td>\n",
       "      <td>LINF_120014600</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.12:451117:451827:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3774</th>\n",
       "      <td>LinJ.22</td>\n",
       "      <td>LINF_220018100-200</td>\n",
       "      <td>LinJ.22:557968:563169:+</td>\n",
       "      <td>LINF_22T0018100-200</td>\n",
       "      <td>LinJ.22:557968:563169:+</td>\n",
       "      <td>LINF_220018100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.22:558160:559113:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3775</th>\n",
       "      <td>LinJ.22</td>\n",
       "      <td>LINF_220018100-200</td>\n",
       "      <td>LinJ.22:557968:563169:+</td>\n",
       "      <td>LINF_22T0018100-200</td>\n",
       "      <td>LinJ.22:557968:563169:+</td>\n",
       "      <td>LINF_220018200</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.22:559255:563064:+</td>\n",
       "      <td>WD_domain_-_G-beta_repeat_-_putative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4286</th>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>LINF_240023900</td>\n",
       "      <td>LinJ.24:689866:693427:-</td>\n",
       "      <td>LINF_24T0023900_1</td>\n",
       "      <td>LinJ.24:689866:690901:-</td>\n",
       "      <td>LINF_240023900_1</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.24:690626:690901:-</td>\n",
       "      <td>nPABP-like protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4287</th>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>LINF_240023900</td>\n",
       "      <td>LinJ.24:689866:693427:-</td>\n",
       "      <td>LINF_24T0023900_2</td>\n",
       "      <td>LinJ.24:692650:693427:-</td>\n",
       "      <td>LINF_240023900_2</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.24:692650:693288:-</td>\n",
       "      <td>nPABP-like protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4469</th>\n",
       "      <td>LinJ.25</td>\n",
       "      <td>LINF_250015300-400</td>\n",
       "      <td>LinJ.25:347501:349146:+</td>\n",
       "      <td>LINF_25T0015300-400</td>\n",
       "      <td>LinJ.25:347501:349146:+</td>\n",
       "      <td>LINF_250015300</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.25:347718:348536:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4470</th>\n",
       "      <td>LinJ.25</td>\n",
       "      <td>LINF_250015300-400</td>\n",
       "      <td>LinJ.25:347501:349146:+</td>\n",
       "      <td>LINF_25T0015300-400</td>\n",
       "      <td>LinJ.25:347501:349146:+</td>\n",
       "      <td>LINF_250015400</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.25:348604:349041:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4495</th>\n",
       "      <td>LinJ.25</td>\n",
       "      <td>LINF_250017700-600</td>\n",
       "      <td>LinJ.25:451512:453695:-</td>\n",
       "      <td>LINF_25T0017700-600</td>\n",
       "      <td>LinJ.25:451512:453695:-</td>\n",
       "      <td>LINF_250017600</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.25:451989:452591:-</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4496</th>\n",
       "      <td>LinJ.25</td>\n",
       "      <td>LINF_250017700-600</td>\n",
       "      <td>LinJ.25:451512:453695:-</td>\n",
       "      <td>LINF_25T0017700-600</td>\n",
       "      <td>LinJ.25:451512:453695:-</td>\n",
       "      <td>LINF_250017700</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.25:452600:453340:-</td>\n",
       "      <td>alpha-tubulin_N-acetyltransferase_-_putative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4917</th>\n",
       "      <td>LinJ.26</td>\n",
       "      <td>LINF_260029800-900</td>\n",
       "      <td>LinJ.26:953645:955999:+</td>\n",
       "      <td>LINF_26T0029800-900</td>\n",
       "      <td>LinJ.26:953645:955999:+</td>\n",
       "      <td>LINF_260029800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.26:953976:954413:+</td>\n",
       "      <td>Lys-63-specific_deubiquitinase_BRCC36_-_putative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4918</th>\n",
       "      <td>LinJ.26</td>\n",
       "      <td>LINF_260029800-900</td>\n",
       "      <td>LinJ.26:953645:955999:+</td>\n",
       "      <td>LINF_26T0029800-900</td>\n",
       "      <td>LinJ.26:953645:955999:+</td>\n",
       "      <td>LINF_260029900</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.26:954522:954851:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5286</th>\n",
       "      <td>LinJ.27</td>\n",
       "      <td>LINF_270033600-700</td>\n",
       "      <td>LinJ.27:1153377:1155197:+</td>\n",
       "      <td>LINF_27T0033600-700</td>\n",
       "      <td>LinJ.27:1153377:1155197:+</td>\n",
       "      <td>LINF_270033600</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.27:1153523:1154887:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5287</th>\n",
       "      <td>LinJ.27</td>\n",
       "      <td>LINF_270033600-700</td>\n",
       "      <td>LinJ.27:1153377:1155197:+</td>\n",
       "      <td>LINF_27T0033600-700</td>\n",
       "      <td>LinJ.27:1153377:1155197:+</td>\n",
       "      <td>LINF_270033700</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.27:1153982:1154887:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6000</th>\n",
       "      <td>LinJ.29</td>\n",
       "      <td>LINF_290033800</td>\n",
       "      <td>LinJ.29:1181710:1184611:+</td>\n",
       "      <td>LINF_29T0033800_1</td>\n",
       "      <td>LinJ.29:1181710:1182263:+</td>\n",
       "      <td>LINF_290033800_1</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.29:1182001:1182263:+</td>\n",
       "      <td>polynucleotide adenylyltransferase|poly(A) pol...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6001</th>\n",
       "      <td>LinJ.29</td>\n",
       "      <td>LINF_290033800</td>\n",
       "      <td>LinJ.29:1181710:1184611:+</td>\n",
       "      <td>LINF_29T0033800_2</td>\n",
       "      <td>LinJ.29:1183113:1184611:+</td>\n",
       "      <td>LINF_290033800_2</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.29:1183113:1184352:+</td>\n",
       "      <td>polynucleotide adenylyltransferase|poly(A) pol...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6820</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>LINF_310033100-3000</td>\n",
       "      <td>LinJ.31:1273465:1275764:-</td>\n",
       "      <td>LINF_31T0033100-3000</td>\n",
       "      <td>LinJ.31:1273465:1275764:-</td>\n",
       "      <td>LINF_310033000</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.31:1274043:1274432:-</td>\n",
       "      <td>uridine_kinase-like_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6821</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>LINF_310033100-3000</td>\n",
       "      <td>LinJ.31:1273465:1275764:-</td>\n",
       "      <td>LINF_31T0033100-3000</td>\n",
       "      <td>LinJ.31:1273465:1275764:-</td>\n",
       "      <td>LINF_310033100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.31:1275097:1275579:-</td>\n",
       "      <td>ferredoxin_2fe-2s-like_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6844</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>LINF_310035100-5000</td>\n",
       "      <td>LinJ.31:1343999:1346787:-</td>\n",
       "      <td>LINF_31T0035100-5000</td>\n",
       "      <td>LinJ.31:1343999:1346787:-</td>\n",
       "      <td>LINF_310035000</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.31:1344181:1344771:-</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6845</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>LINF_310035100-5000</td>\n",
       "      <td>LinJ.31:1343999:1346787:-</td>\n",
       "      <td>LINF_31T0035100-5000</td>\n",
       "      <td>LinJ.31:1343999:1346787:-</td>\n",
       "      <td>LINF_310035100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.31:1344807:1345379:-</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7181</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320029000-100</td>\n",
       "      <td>LinJ.32:889966:894771:-</td>\n",
       "      <td>LINF_32T0029000-100</td>\n",
       "      <td>LinJ.32:889966:894771:-</td>\n",
       "      <td>LINF_320029000</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:890313:890891:-</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7182</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320029000-100</td>\n",
       "      <td>LinJ.32:889966:894771:-</td>\n",
       "      <td>LINF_32T0029000-100</td>\n",
       "      <td>LinJ.32:889966:894771:-</td>\n",
       "      <td>LINF_320029100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:890951:893725:-</td>\n",
       "      <td>membrane_associated_protein-like_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7195</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320030400-500</td>\n",
       "      <td>LinJ.32:933016:935642:-</td>\n",
       "      <td>LINF_32T0030400-500</td>\n",
       "      <td>LinJ.32:933016:935642:-</td>\n",
       "      <td>LINF_320030400</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:934389:934805:-</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7196</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320030400-500</td>\n",
       "      <td>LinJ.32:933016:935642:-</td>\n",
       "      <td>LINF_32T0030400-500</td>\n",
       "      <td>LinJ.32:933016:935642:-</td>\n",
       "      <td>LINF_320030500</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:934940:935455:-</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7275</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320037300-400</td>\n",
       "      <td>LinJ.32:1177409:1179063:+</td>\n",
       "      <td>LINF_32T0037300-400</td>\n",
       "      <td>LinJ.32:1177409:1179063:+</td>\n",
       "      <td>LINF_320037300</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:1177678:1178130:+</td>\n",
       "      <td>Domain_of_unknown_function_DUF21_-_putative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7276</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320037300-400</td>\n",
       "      <td>LinJ.32:1177409:1179063:+</td>\n",
       "      <td>LINF_32T0037300-400</td>\n",
       "      <td>LinJ.32:1177409:1179063:+</td>\n",
       "      <td>LINF_320037400</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:1178292:1178738:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7568</th>\n",
       "      <td>LinJ.33</td>\n",
       "      <td>LINF_330021700-800</td>\n",
       "      <td>LinJ.33:553571:558357:+</td>\n",
       "      <td>LINF_33T0021700-800</td>\n",
       "      <td>LinJ.33:553571:558357:+</td>\n",
       "      <td>LINF_330021700</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.33:555988:556314:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7569</th>\n",
       "      <td>LinJ.33</td>\n",
       "      <td>LINF_330021700-800</td>\n",
       "      <td>LinJ.33:553571:558357:+</td>\n",
       "      <td>LINF_33T0021700-800</td>\n",
       "      <td>LinJ.33:553571:558357:+</td>\n",
       "      <td>LINF_330021800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.33:557286:557831:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 52
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:25:01.494899Z",
     "start_time": "2024-09-27T08:25:01.478225Z"
    }
   },
   "source": [
    "# Let's loop through them by pairs\n",
    "group_duplicated_gene_id_df = duplicated_gene_id_df.groupby('Gene_ID')\n",
    "duplicated_gene_id_df_2 = pd.DataFrame(columns=duplicated_gene_id_df.columns)  # Create empty DataFrame to store the results\n",
    "\n",
    "for group_name, group_df in group_duplicated_gene_id_df:\n",
    "    # Check if the string in column \"Transcript_ID\" ends with \"_1\" or \"_2\"\n",
    "    condition = group_df['Transcript_ID'].str.endswith(\"_1\") | group_df['Transcript_ID'].str.endswith(\"_2\") # Condition to filter\n",
    "    if condition.sum() == 2:  # Since both elements have the separator \"_1\" and \"_2\", let's skip the group.\n",
    "        print(f\"Skipping {group_name} because it has both _1 and _2\")\n",
    "        duplicated_gene_id_df_2 = pd.concat([duplicated_gene_id_df_2, group_df], ignore_index=True)\n",
    "        continue\n",
    "    else:  # Let's differentiate the elements in \"Transcript_ID\" with \"_1\" and \"_2\"\n",
    "        for i in range(2):\n",
    "            group_df.iloc[i, :].Transcript_ID = group_df.iloc[i, :].Transcript_ID + f\"_{i+1}\"\n",
    "        # print(group_df[\"Transcript_ID\"])\n",
    "        duplicated_gene_id_df_2 = pd.concat([duplicated_gene_id_df_2, group_df], ignore_index=True)"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Skipping LINF_070008800 because it has both _1 and _2\n",
      "Skipping LINF_100016800 because it has both _1 and _2\n",
      "Skipping LINF_240023900 because it has both _1 and _2\n",
      "Skipping LINF_290033800 because it has both _1 and _2\n"
     ]
    }
   ],
   "execution_count": 53
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:25:04.388616Z",
     "start_time": "2024-09-27T08:25:04.382069Z"
    }
   },
   "source": [
    "# In the original `filtered_df` replace the column \"Transcript_ID\" with the \"Transcript_ID\" values in `final_df`\n",
    "# Get the index to replace in the original data frame\n",
    "index_to_replace = df_filtered[df_filtered['Gene_ID'].value_counts()[df_filtered['Gene_ID']].values > 1].index\n",
    "\n",
    "# Create a mask\n",
    "mask = df_filtered.index.isin(index_to_replace)\n",
    "\n",
    "# Replace the \"Transcript_ID\" in the `filtered_df` with the \"Transcript_ID\" in `final_df`\n",
    "df_filtered.loc[mask, 'Transcript_ID'] = duplicated_gene_id_df_2.loc[:, 'Transcript_ID'].values"
   ],
   "outputs": [],
   "execution_count": 54
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:25:07.750649Z",
     "start_time": "2024-09-27T08:25:07.743187Z"
    }
   },
   "source": [
    "# Check results \n",
    "df_filtered.loc[mask, :]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom              Gene_ID                 Gene_coord  \\\n",
       "777   LinJ.06   LINF_060017800-900    LinJ.06:487685:490191:+   \n",
       "778   LinJ.06   LINF_060017800-900    LinJ.06:487685:490191:+   \n",
       "841   LinJ.07       LINF_070008800    LinJ.07:153281:160411:-   \n",
       "842   LinJ.07       LINF_070008800    LinJ.07:153281:160411:-   \n",
       "1376  LinJ.10   LINF_100011400-500    LinJ.10:263964:266643:+   \n",
       "1377  LinJ.10   LINF_100011400-500    LinJ.10:263964:266643:+   \n",
       "1430  LinJ.10       LINF_100016800    LinJ.10:468436:469642:-   \n",
       "1431  LinJ.10       LINF_100016800    LinJ.10:468436:469642:-   \n",
       "1748  LinJ.12   LINF_120014100-200    LinJ.12:440484:443559:+   \n",
       "1749  LinJ.12   LINF_120014100-200    LinJ.12:440484:443559:+   \n",
       "1752  LinJ.12   LINF_120014500-600    LinJ.12:450370:452072:+   \n",
       "1753  LinJ.12   LINF_120014500-600    LinJ.12:450370:452072:+   \n",
       "3774  LinJ.22   LINF_220018100-200    LinJ.22:557968:563169:+   \n",
       "3775  LinJ.22   LINF_220018100-200    LinJ.22:557968:563169:+   \n",
       "4286  LinJ.24       LINF_240023900    LinJ.24:689866:693427:-   \n",
       "4287  LinJ.24       LINF_240023900    LinJ.24:689866:693427:-   \n",
       "4469  LinJ.25   LINF_250015300-400    LinJ.25:347501:349146:+   \n",
       "4470  LinJ.25   LINF_250015300-400    LinJ.25:347501:349146:+   \n",
       "4495  LinJ.25   LINF_250017700-600    LinJ.25:451512:453695:-   \n",
       "4496  LinJ.25   LINF_250017700-600    LinJ.25:451512:453695:-   \n",
       "4917  LinJ.26   LINF_260029800-900    LinJ.26:953645:955999:+   \n",
       "4918  LinJ.26   LINF_260029800-900    LinJ.26:953645:955999:+   \n",
       "5286  LinJ.27   LINF_270033600-700  LinJ.27:1153377:1155197:+   \n",
       "5287  LinJ.27   LINF_270033600-700  LinJ.27:1153377:1155197:+   \n",
       "6000  LinJ.29       LINF_290033800  LinJ.29:1181710:1184611:+   \n",
       "6001  LinJ.29       LINF_290033800  LinJ.29:1181710:1184611:+   \n",
       "6820  LinJ.31  LINF_310033100-3000  LinJ.31:1273465:1275764:-   \n",
       "6821  LinJ.31  LINF_310033100-3000  LinJ.31:1273465:1275764:-   \n",
       "6844  LinJ.31  LINF_310035100-5000  LinJ.31:1343999:1346787:-   \n",
       "6845  LinJ.31  LINF_310035100-5000  LinJ.31:1343999:1346787:-   \n",
       "7181  LinJ.32   LINF_320029000-100    LinJ.32:889966:894771:-   \n",
       "7182  LinJ.32   LINF_320029000-100    LinJ.32:889966:894771:-   \n",
       "7195  LinJ.32   LINF_320030400-500    LinJ.32:933016:935642:-   \n",
       "7196  LinJ.32   LINF_320030400-500    LinJ.32:933016:935642:-   \n",
       "7275  LinJ.32   LINF_320037300-400  LinJ.32:1177409:1179063:+   \n",
       "7276  LinJ.32   LINF_320037300-400  LinJ.32:1177409:1179063:+   \n",
       "7568  LinJ.33   LINF_330021700-800    LinJ.33:553571:558357:+   \n",
       "7569  LinJ.33   LINF_330021700-800    LinJ.33:553571:558357:+   \n",
       "\n",
       "               Transcript_ID           Transcript_coord    CDS/miscRNA_ID  \\\n",
       "777    LINF_06T0017800-900_1    LinJ.06:487685:490191:+    LINF_060017800   \n",
       "778    LINF_06T0017800-900_2    LinJ.06:487685:490191:+    LINF_060017900   \n",
       "841        LINF_07T0008800_1    LinJ.07:153281:154334:-  LINF_070008800_1   \n",
       "842        LINF_07T0008800_2    LinJ.07:158854:160411:-  LINF_070008800_2   \n",
       "1376   LINF_10T0011400-500_1    LinJ.10:263964:266643:+    LINF_100011400   \n",
       "1377   LINF_10T0011400-500_2    LinJ.10:263964:266643:+    LINF_100011500   \n",
       "1430       LINF_10T0016800_1    LinJ.10:468436:468904:-                ND   \n",
       "1431       LINF_10T0016800_2    LinJ.10:469155:469642:-    LINF_100016800   \n",
       "1748   LINF_12T0014100-200_1    LinJ.12:440484:443559:+    LINF_120014100   \n",
       "1749   LINF_12T0014100-200_2    LinJ.12:440484:443559:+    LINF_120014200   \n",
       "1752   LINF_12T0014500-600_1    LinJ.12:450370:452072:+    LINF_120014500   \n",
       "1753   LINF_12T0014500-600_2    LinJ.12:450370:452072:+    LINF_120014600   \n",
       "3774   LINF_22T0018100-200_1    LinJ.22:557968:563169:+    LINF_220018100   \n",
       "3775   LINF_22T0018100-200_2    LinJ.22:557968:563169:+    LINF_220018200   \n",
       "4286       LINF_24T0023900_1    LinJ.24:689866:690901:-  LINF_240023900_1   \n",
       "4287       LINF_24T0023900_2    LinJ.24:692650:693427:-  LINF_240023900_2   \n",
       "4469   LINF_25T0015300-400_1    LinJ.25:347501:349146:+    LINF_250015300   \n",
       "4470   LINF_25T0015300-400_2    LinJ.25:347501:349146:+    LINF_250015400   \n",
       "4495   LINF_25T0017700-600_1    LinJ.25:451512:453695:-    LINF_250017600   \n",
       "4496   LINF_25T0017700-600_2    LinJ.25:451512:453695:-    LINF_250017700   \n",
       "4917   LINF_26T0029800-900_1    LinJ.26:953645:955999:+    LINF_260029800   \n",
       "4918   LINF_26T0029800-900_2    LinJ.26:953645:955999:+    LINF_260029900   \n",
       "5286   LINF_27T0033600-700_1  LinJ.27:1153377:1155197:+    LINF_270033600   \n",
       "5287   LINF_27T0033600-700_2  LinJ.27:1153377:1155197:+    LINF_270033700   \n",
       "6000       LINF_29T0033800_1  LinJ.29:1181710:1182263:+  LINF_290033800_1   \n",
       "6001       LINF_29T0033800_2  LinJ.29:1183113:1184611:+  LINF_290033800_2   \n",
       "6820  LINF_31T0033100-3000_1  LinJ.31:1273465:1275764:-    LINF_310033000   \n",
       "6821  LINF_31T0033100-3000_2  LinJ.31:1273465:1275764:-    LINF_310033100   \n",
       "6844  LINF_31T0035100-5000_1  LinJ.31:1343999:1346787:-    LINF_310035000   \n",
       "6845  LINF_31T0035100-5000_2  LinJ.31:1343999:1346787:-    LINF_310035100   \n",
       "7181   LINF_32T0029000-100_1    LinJ.32:889966:894771:-    LINF_320029000   \n",
       "7182   LINF_32T0029000-100_2    LinJ.32:889966:894771:-    LINF_320029100   \n",
       "7195   LINF_32T0030400-500_1    LinJ.32:933016:935642:-    LINF_320030400   \n",
       "7196   LINF_32T0030400-500_2    LinJ.32:933016:935642:-    LINF_320030500   \n",
       "7275   LINF_32T0037300-400_1  LinJ.32:1177409:1179063:+    LINF_320037300   \n",
       "7276   LINF_32T0037300-400_2  LinJ.32:1177409:1179063:+    LINF_320037400   \n",
       "7568   LINF_33T0021700-800_1    LinJ.33:553571:558357:+    LINF_330021700   \n",
       "7569   LINF_33T0021700-800_2    LinJ.33:553571:558357:+    LINF_330021800   \n",
       "\n",
       "     Pseudogen          CDS/miscRNA_coord  \\\n",
       "777         NO    LinJ.06:487818:488159:+   \n",
       "778         NO    LinJ.06:488373:488633:+   \n",
       "841         NO    LinJ.07:153994:154334:-   \n",
       "842         NO    LinJ.07:158854:160240:-   \n",
       "1376        NO    LinJ.10:264569:264949:+   \n",
       "1377        NO    LinJ.10:264981:265235:+   \n",
       "1430        NO                         ND   \n",
       "1431        NO    LinJ.10:469169:469561:-   \n",
       "1748        NO    LinJ.12:440748:441215:+   \n",
       "1749        NO    LinJ.12:441320:441805:+   \n",
       "1752        NO    LinJ.12:450545:451012:+   \n",
       "1753        NO    LinJ.12:451117:451827:+   \n",
       "3774        NO    LinJ.22:558160:559113:+   \n",
       "3775        NO    LinJ.22:559255:563064:+   \n",
       "4286        NO    LinJ.24:690626:690901:-   \n",
       "4287        NO    LinJ.24:692650:693288:-   \n",
       "4469        NO    LinJ.25:347718:348536:+   \n",
       "4470        NO    LinJ.25:348604:349041:+   \n",
       "4495        NO    LinJ.25:451989:452591:-   \n",
       "4496        NO    LinJ.25:452600:453340:-   \n",
       "4917        NO    LinJ.26:953976:954413:+   \n",
       "4918        NO    LinJ.26:954522:954851:+   \n",
       "5286        NO  LinJ.27:1153523:1154887:+   \n",
       "5287        NO  LinJ.27:1153982:1154887:+   \n",
       "6000        NO  LinJ.29:1182001:1182263:+   \n",
       "6001        NO  LinJ.29:1183113:1184352:+   \n",
       "6820        NO  LinJ.31:1274043:1274432:-   \n",
       "6821        NO  LinJ.31:1275097:1275579:-   \n",
       "6844        NO  LinJ.31:1344181:1344771:-   \n",
       "6845        NO  LinJ.31:1344807:1345379:-   \n",
       "7181        NO    LinJ.32:890313:890891:-   \n",
       "7182        NO    LinJ.32:890951:893725:-   \n",
       "7195        NO    LinJ.32:934389:934805:-   \n",
       "7196        NO    LinJ.32:934940:935455:-   \n",
       "7275        NO  LinJ.32:1177678:1178130:+   \n",
       "7276        NO  LinJ.32:1178292:1178738:+   \n",
       "7568        NO    LinJ.33:555988:556314:+   \n",
       "7569        NO    LinJ.33:557286:557831:+   \n",
       "\n",
       "                                               Function  \n",
       "777                                hypothetical_protein  \n",
       "778                                hypothetical_protein  \n",
       "841                  ATP-dependent RNA helicase (DBP2B)  \n",
       "842                  ATP-dependent RNA helicase (DBP2B)  \n",
       "1376                   hypothetical_protein_-_conserved  \n",
       "1377                               hypothetical_protein  \n",
       "1430                                         histone H3  \n",
       "1431                                         histone H3  \n",
       "1748                   hypothetical_protein_-_conserved  \n",
       "1749                   hypothetical_protein_-_conserved  \n",
       "1752                   hypothetical_protein_-_conserved  \n",
       "1753                   hypothetical_protein_-_conserved  \n",
       "3774                               hypothetical_protein  \n",
       "3775               WD_domain_-_G-beta_repeat_-_putative  \n",
       "4286                                 nPABP-like protein  \n",
       "4287                                 nPABP-like protein  \n",
       "4469                   hypothetical_protein_-_conserved  \n",
       "4470                               hypothetical_protein  \n",
       "4495                               hypothetical_protein  \n",
       "4496       alpha-tubulin_N-acetyltransferase_-_putative  \n",
       "4917   Lys-63-specific_deubiquitinase_BRCC36_-_putative  \n",
       "4918                               hypothetical_protein  \n",
       "5286                   hypothetical_protein_-_conserved  \n",
       "5287                   hypothetical_protein_-_conserved  \n",
       "6000  polynucleotide adenylyltransferase|poly(A) pol...  \n",
       "6001  polynucleotide adenylyltransferase|poly(A) pol...  \n",
       "6820                        uridine_kinase-like_protein  \n",
       "6821                     ferredoxin_2fe-2s-like_protein  \n",
       "6844                               hypothetical_protein  \n",
       "6845                   hypothetical_protein_-_conserved  \n",
       "7181                               hypothetical_protein  \n",
       "7182           membrane_associated_protein-like_protein  \n",
       "7195                               hypothetical_protein  \n",
       "7196                   hypothetical_protein_-_conserved  \n",
       "7275        Domain_of_unknown_function_DUF21_-_putative  \n",
       "7276                               hypothetical_protein  \n",
       "7568                               hypothetical_protein  \n",
       "7569                               hypothetical_protein  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>777</th>\n",
       "      <td>LinJ.06</td>\n",
       "      <td>LINF_060017800-900</td>\n",
       "      <td>LinJ.06:487685:490191:+</td>\n",
       "      <td>LINF_06T0017800-900_1</td>\n",
       "      <td>LinJ.06:487685:490191:+</td>\n",
       "      <td>LINF_060017800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.06:487818:488159:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>778</th>\n",
       "      <td>LinJ.06</td>\n",
       "      <td>LINF_060017800-900</td>\n",
       "      <td>LinJ.06:487685:490191:+</td>\n",
       "      <td>LINF_06T0017800-900_2</td>\n",
       "      <td>LinJ.06:487685:490191:+</td>\n",
       "      <td>LINF_060017900</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.06:488373:488633:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>841</th>\n",
       "      <td>LinJ.07</td>\n",
       "      <td>LINF_070008800</td>\n",
       "      <td>LinJ.07:153281:160411:-</td>\n",
       "      <td>LINF_07T0008800_1</td>\n",
       "      <td>LinJ.07:153281:154334:-</td>\n",
       "      <td>LINF_070008800_1</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.07:153994:154334:-</td>\n",
       "      <td>ATP-dependent RNA helicase (DBP2B)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>842</th>\n",
       "      <td>LinJ.07</td>\n",
       "      <td>LINF_070008800</td>\n",
       "      <td>LinJ.07:153281:160411:-</td>\n",
       "      <td>LINF_07T0008800_2</td>\n",
       "      <td>LinJ.07:158854:160411:-</td>\n",
       "      <td>LINF_070008800_2</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.07:158854:160240:-</td>\n",
       "      <td>ATP-dependent RNA helicase (DBP2B)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1376</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100011400-500</td>\n",
       "      <td>LinJ.10:263964:266643:+</td>\n",
       "      <td>LINF_10T0011400-500_1</td>\n",
       "      <td>LinJ.10:263964:266643:+</td>\n",
       "      <td>LINF_100011400</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.10:264569:264949:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1377</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100011400-500</td>\n",
       "      <td>LinJ.10:263964:266643:+</td>\n",
       "      <td>LINF_10T0011400-500_2</td>\n",
       "      <td>LinJ.10:263964:266643:+</td>\n",
       "      <td>LINF_100011500</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.10:264981:265235:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1430</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>LinJ.10:468436:469642:-</td>\n",
       "      <td>LINF_10T0016800_1</td>\n",
       "      <td>LinJ.10:468436:468904:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>histone H3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1431</th>\n",
       "      <td>LinJ.10</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>LinJ.10:468436:469642:-</td>\n",
       "      <td>LINF_10T0016800_2</td>\n",
       "      <td>LinJ.10:469155:469642:-</td>\n",
       "      <td>LINF_100016800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.10:469169:469561:-</td>\n",
       "      <td>histone H3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1748</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>LINF_120014100-200</td>\n",
       "      <td>LinJ.12:440484:443559:+</td>\n",
       "      <td>LINF_12T0014100-200_1</td>\n",
       "      <td>LinJ.12:440484:443559:+</td>\n",
       "      <td>LINF_120014100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.12:440748:441215:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1749</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>LINF_120014100-200</td>\n",
       "      <td>LinJ.12:440484:443559:+</td>\n",
       "      <td>LINF_12T0014100-200_2</td>\n",
       "      <td>LinJ.12:440484:443559:+</td>\n",
       "      <td>LINF_120014200</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.12:441320:441805:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1752</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>LINF_120014500-600</td>\n",
       "      <td>LinJ.12:450370:452072:+</td>\n",
       "      <td>LINF_12T0014500-600_1</td>\n",
       "      <td>LinJ.12:450370:452072:+</td>\n",
       "      <td>LINF_120014500</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.12:450545:451012:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1753</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>LINF_120014500-600</td>\n",
       "      <td>LinJ.12:450370:452072:+</td>\n",
       "      <td>LINF_12T0014500-600_2</td>\n",
       "      <td>LinJ.12:450370:452072:+</td>\n",
       "      <td>LINF_120014600</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.12:451117:451827:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3774</th>\n",
       "      <td>LinJ.22</td>\n",
       "      <td>LINF_220018100-200</td>\n",
       "      <td>LinJ.22:557968:563169:+</td>\n",
       "      <td>LINF_22T0018100-200_1</td>\n",
       "      <td>LinJ.22:557968:563169:+</td>\n",
       "      <td>LINF_220018100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.22:558160:559113:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3775</th>\n",
       "      <td>LinJ.22</td>\n",
       "      <td>LINF_220018100-200</td>\n",
       "      <td>LinJ.22:557968:563169:+</td>\n",
       "      <td>LINF_22T0018100-200_2</td>\n",
       "      <td>LinJ.22:557968:563169:+</td>\n",
       "      <td>LINF_220018200</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.22:559255:563064:+</td>\n",
       "      <td>WD_domain_-_G-beta_repeat_-_putative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4286</th>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>LINF_240023900</td>\n",
       "      <td>LinJ.24:689866:693427:-</td>\n",
       "      <td>LINF_24T0023900_1</td>\n",
       "      <td>LinJ.24:689866:690901:-</td>\n",
       "      <td>LINF_240023900_1</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.24:690626:690901:-</td>\n",
       "      <td>nPABP-like protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4287</th>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>LINF_240023900</td>\n",
       "      <td>LinJ.24:689866:693427:-</td>\n",
       "      <td>LINF_24T0023900_2</td>\n",
       "      <td>LinJ.24:692650:693427:-</td>\n",
       "      <td>LINF_240023900_2</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.24:692650:693288:-</td>\n",
       "      <td>nPABP-like protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4469</th>\n",
       "      <td>LinJ.25</td>\n",
       "      <td>LINF_250015300-400</td>\n",
       "      <td>LinJ.25:347501:349146:+</td>\n",
       "      <td>LINF_25T0015300-400_1</td>\n",
       "      <td>LinJ.25:347501:349146:+</td>\n",
       "      <td>LINF_250015300</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.25:347718:348536:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4470</th>\n",
       "      <td>LinJ.25</td>\n",
       "      <td>LINF_250015300-400</td>\n",
       "      <td>LinJ.25:347501:349146:+</td>\n",
       "      <td>LINF_25T0015300-400_2</td>\n",
       "      <td>LinJ.25:347501:349146:+</td>\n",
       "      <td>LINF_250015400</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.25:348604:349041:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4495</th>\n",
       "      <td>LinJ.25</td>\n",
       "      <td>LINF_250017700-600</td>\n",
       "      <td>LinJ.25:451512:453695:-</td>\n",
       "      <td>LINF_25T0017700-600_1</td>\n",
       "      <td>LinJ.25:451512:453695:-</td>\n",
       "      <td>LINF_250017600</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.25:451989:452591:-</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4496</th>\n",
       "      <td>LinJ.25</td>\n",
       "      <td>LINF_250017700-600</td>\n",
       "      <td>LinJ.25:451512:453695:-</td>\n",
       "      <td>LINF_25T0017700-600_2</td>\n",
       "      <td>LinJ.25:451512:453695:-</td>\n",
       "      <td>LINF_250017700</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.25:452600:453340:-</td>\n",
       "      <td>alpha-tubulin_N-acetyltransferase_-_putative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4917</th>\n",
       "      <td>LinJ.26</td>\n",
       "      <td>LINF_260029800-900</td>\n",
       "      <td>LinJ.26:953645:955999:+</td>\n",
       "      <td>LINF_26T0029800-900_1</td>\n",
       "      <td>LinJ.26:953645:955999:+</td>\n",
       "      <td>LINF_260029800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.26:953976:954413:+</td>\n",
       "      <td>Lys-63-specific_deubiquitinase_BRCC36_-_putative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4918</th>\n",
       "      <td>LinJ.26</td>\n",
       "      <td>LINF_260029800-900</td>\n",
       "      <td>LinJ.26:953645:955999:+</td>\n",
       "      <td>LINF_26T0029800-900_2</td>\n",
       "      <td>LinJ.26:953645:955999:+</td>\n",
       "      <td>LINF_260029900</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.26:954522:954851:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5286</th>\n",
       "      <td>LinJ.27</td>\n",
       "      <td>LINF_270033600-700</td>\n",
       "      <td>LinJ.27:1153377:1155197:+</td>\n",
       "      <td>LINF_27T0033600-700_1</td>\n",
       "      <td>LinJ.27:1153377:1155197:+</td>\n",
       "      <td>LINF_270033600</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.27:1153523:1154887:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5287</th>\n",
       "      <td>LinJ.27</td>\n",
       "      <td>LINF_270033600-700</td>\n",
       "      <td>LinJ.27:1153377:1155197:+</td>\n",
       "      <td>LINF_27T0033600-700_2</td>\n",
       "      <td>LinJ.27:1153377:1155197:+</td>\n",
       "      <td>LINF_270033700</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.27:1153982:1154887:+</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6000</th>\n",
       "      <td>LinJ.29</td>\n",
       "      <td>LINF_290033800</td>\n",
       "      <td>LinJ.29:1181710:1184611:+</td>\n",
       "      <td>LINF_29T0033800_1</td>\n",
       "      <td>LinJ.29:1181710:1182263:+</td>\n",
       "      <td>LINF_290033800_1</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.29:1182001:1182263:+</td>\n",
       "      <td>polynucleotide adenylyltransferase|poly(A) pol...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6001</th>\n",
       "      <td>LinJ.29</td>\n",
       "      <td>LINF_290033800</td>\n",
       "      <td>LinJ.29:1181710:1184611:+</td>\n",
       "      <td>LINF_29T0033800_2</td>\n",
       "      <td>LinJ.29:1183113:1184611:+</td>\n",
       "      <td>LINF_290033800_2</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.29:1183113:1184352:+</td>\n",
       "      <td>polynucleotide adenylyltransferase|poly(A) pol...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6820</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>LINF_310033100-3000</td>\n",
       "      <td>LinJ.31:1273465:1275764:-</td>\n",
       "      <td>LINF_31T0033100-3000_1</td>\n",
       "      <td>LinJ.31:1273465:1275764:-</td>\n",
       "      <td>LINF_310033000</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.31:1274043:1274432:-</td>\n",
       "      <td>uridine_kinase-like_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6821</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>LINF_310033100-3000</td>\n",
       "      <td>LinJ.31:1273465:1275764:-</td>\n",
       "      <td>LINF_31T0033100-3000_2</td>\n",
       "      <td>LinJ.31:1273465:1275764:-</td>\n",
       "      <td>LINF_310033100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.31:1275097:1275579:-</td>\n",
       "      <td>ferredoxin_2fe-2s-like_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6844</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>LINF_310035100-5000</td>\n",
       "      <td>LinJ.31:1343999:1346787:-</td>\n",
       "      <td>LINF_31T0035100-5000_1</td>\n",
       "      <td>LinJ.31:1343999:1346787:-</td>\n",
       "      <td>LINF_310035000</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.31:1344181:1344771:-</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6845</th>\n",
       "      <td>LinJ.31</td>\n",
       "      <td>LINF_310035100-5000</td>\n",
       "      <td>LinJ.31:1343999:1346787:-</td>\n",
       "      <td>LINF_31T0035100-5000_2</td>\n",
       "      <td>LinJ.31:1343999:1346787:-</td>\n",
       "      <td>LINF_310035100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.31:1344807:1345379:-</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7181</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320029000-100</td>\n",
       "      <td>LinJ.32:889966:894771:-</td>\n",
       "      <td>LINF_32T0029000-100_1</td>\n",
       "      <td>LinJ.32:889966:894771:-</td>\n",
       "      <td>LINF_320029000</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:890313:890891:-</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7182</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320029000-100</td>\n",
       "      <td>LinJ.32:889966:894771:-</td>\n",
       "      <td>LINF_32T0029000-100_2</td>\n",
       "      <td>LinJ.32:889966:894771:-</td>\n",
       "      <td>LINF_320029100</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:890951:893725:-</td>\n",
       "      <td>membrane_associated_protein-like_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7195</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320030400-500</td>\n",
       "      <td>LinJ.32:933016:935642:-</td>\n",
       "      <td>LINF_32T0030400-500_1</td>\n",
       "      <td>LinJ.32:933016:935642:-</td>\n",
       "      <td>LINF_320030400</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:934389:934805:-</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7196</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320030400-500</td>\n",
       "      <td>LinJ.32:933016:935642:-</td>\n",
       "      <td>LINF_32T0030400-500_2</td>\n",
       "      <td>LinJ.32:933016:935642:-</td>\n",
       "      <td>LINF_320030500</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:934940:935455:-</td>\n",
       "      <td>hypothetical_protein_-_conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7275</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320037300-400</td>\n",
       "      <td>LinJ.32:1177409:1179063:+</td>\n",
       "      <td>LINF_32T0037300-400_1</td>\n",
       "      <td>LinJ.32:1177409:1179063:+</td>\n",
       "      <td>LINF_320037300</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:1177678:1178130:+</td>\n",
       "      <td>Domain_of_unknown_function_DUF21_-_putative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7276</th>\n",
       "      <td>LinJ.32</td>\n",
       "      <td>LINF_320037300-400</td>\n",
       "      <td>LinJ.32:1177409:1179063:+</td>\n",
       "      <td>LINF_32T0037300-400_2</td>\n",
       "      <td>LinJ.32:1177409:1179063:+</td>\n",
       "      <td>LINF_320037400</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.32:1178292:1178738:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7568</th>\n",
       "      <td>LinJ.33</td>\n",
       "      <td>LINF_330021700-800</td>\n",
       "      <td>LinJ.33:553571:558357:+</td>\n",
       "      <td>LINF_33T0021700-800_1</td>\n",
       "      <td>LinJ.33:553571:558357:+</td>\n",
       "      <td>LINF_330021700</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.33:555988:556314:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7569</th>\n",
       "      <td>LinJ.33</td>\n",
       "      <td>LINF_330021700-800</td>\n",
       "      <td>LinJ.33:553571:558357:+</td>\n",
       "      <td>LINF_33T0021700-800_2</td>\n",
       "      <td>LinJ.33:553571:558357:+</td>\n",
       "      <td>LINF_330021800</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.33:557286:557831:+</td>\n",
       "      <td>hypothetical_protein</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 55
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "## Doubts with @Javi and @JMR 18/07/24 "
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "### Doubts resolved only with @Javi"
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Change CDS coord in LINF_230015300\n",
    "\n",
    "* New CDS coord: LinJ.23:367028:369478:-\n"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:25:56.839642Z",
     "start_time": "2024-09-27T08:25:56.833601Z"
    }
   },
   "source": [
    "# Check the element \"LINF_230015300\"\n",
    "df_filtered[df_filtered['Gene_ID'] == \"LINF_230015300\"]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID               Gene_coord    Transcript_ID  \\\n",
       "3949  LinJ.23  LINF_230015300  LinJ.23:366240:369541:-  LINF_23T0015300   \n",
       "\n",
       "             Transcript_coord  CDS/miscRNA_ID Pseudogen  \\\n",
       "3949  LinJ.23:366240:369541:-  LINF_230015300        NO   \n",
       "\n",
       "            CDS/miscRNA_coord                          Function  \n",
       "3949  LinJ.23:367028:369919:-  hypothetical protein - conserved  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3949</th>\n",
       "      <td>LinJ.23</td>\n",
       "      <td>LINF_230015300</td>\n",
       "      <td>LinJ.23:366240:369541:-</td>\n",
       "      <td>LINF_23T0015300</td>\n",
       "      <td>LinJ.23:366240:369541:-</td>\n",
       "      <td>LINF_230015300</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.23:367028:369919:-</td>\n",
       "      <td>hypothetical protein - conserved</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 56
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:25:59.952951Z",
     "start_time": "2024-09-27T08:25:59.945403Z"
    }
   },
   "source": [
    "# Change \"CDS/miscRNA_coord\" to \"LinJ.23:367028:369478:-\"\n",
    "df_filtered.loc[df_filtered['Gene_ID'] == \"LINF_230015300\", 'CDS/miscRNA_coord'] = \"LinJ.23:367028:369478:-\"\n",
    "\n",
    "# Check the element \"LINF_230015300\"\n",
    "df_filtered[df_filtered['Gene_ID'] == \"LINF_230015300\"]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID               Gene_coord    Transcript_ID  \\\n",
       "3949  LinJ.23  LINF_230015300  LinJ.23:366240:369541:-  LINF_23T0015300   \n",
       "\n",
       "             Transcript_coord  CDS/miscRNA_ID Pseudogen  \\\n",
       "3949  LinJ.23:366240:369541:-  LINF_230015300        NO   \n",
       "\n",
       "            CDS/miscRNA_coord                          Function  \n",
       "3949  LinJ.23:367028:369478:-  hypothetical protein - conserved  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3949</th>\n",
       "      <td>LinJ.23</td>\n",
       "      <td>LINF_230015300</td>\n",
       "      <td>LinJ.23:366240:369541:-</td>\n",
       "      <td>LINF_23T0015300</td>\n",
       "      <td>LinJ.23:366240:369541:-</td>\n",
       "      <td>LINF_230015300</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.23:367028:369478:-</td>\n",
       "      <td>hypothetical protein - conserved</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 57
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "#### Element LINF_290024900. Removing CDS"
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:26:45.026341Z",
     "start_time": "2024-09-27T08:26:45.020731Z"
    }
   },
   "source": [
    "# Check the element \"LINF_290024900\"\n",
    "df_filtered[df_filtered['Gene_ID'] == \"LINF_290024900\"]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID               Gene_coord    Transcript_ID  \\\n",
       "5906  LinJ.29  LINF_290024900  LinJ.29:850028:850959:-  LINF_29T0024900   \n",
       "\n",
       "             Transcript_coord  CDS/miscRNA_ID Pseudogen  \\\n",
       "5906  LinJ.29:850028:850959:-  LINF_290024900        NO   \n",
       "\n",
       "            CDS/miscRNA_coord                        Function  \n",
       "5906  LinJ.29:849730:850788:-  paraflagellar rod protein-like  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5906</th>\n",
       "      <td>LinJ.29</td>\n",
       "      <td>LINF_290024900</td>\n",
       "      <td>LinJ.29:850028:850959:-</td>\n",
       "      <td>LINF_29T0024900</td>\n",
       "      <td>LinJ.29:850028:850959:-</td>\n",
       "      <td>LINF_290024900</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.29:849730:850788:-</td>\n",
       "      <td>paraflagellar rod protein-like</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 58
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:26:46.630640Z",
     "start_time": "2024-09-27T08:26:46.623504Z"
    }
   },
   "source": [
    "# Removing column \"CDS/miscRNA_ID\" and \"CDS/miscRNA_coord\" for the element \"LINF_290024900\" and replace it for \"ND\"\n",
    "df_filtered.loc[df_filtered['Gene_ID'] == \"LINF_290024900\", ['CDS/miscRNA_ID', 'CDS/miscRNA_coord']] = \"ND\"\n",
    "\n",
    "# Check the element \"LINF_290024900\"\n",
    "df_filtered[df_filtered['Gene_ID'] == \"LINF_290024900\"]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID               Gene_coord    Transcript_ID  \\\n",
       "5906  LinJ.29  LINF_290024900  LinJ.29:850028:850959:-  LINF_29T0024900   \n",
       "\n",
       "             Transcript_coord CDS/miscRNA_ID Pseudogen CDS/miscRNA_coord  \\\n",
       "5906  LinJ.29:850028:850959:-             ND        NO                ND   \n",
       "\n",
       "                            Function  \n",
       "5906  paraflagellar rod protein-like  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5906</th>\n",
       "      <td>LinJ.29</td>\n",
       "      <td>LINF_290024900</td>\n",
       "      <td>LinJ.29:850028:850959:-</td>\n",
       "      <td>LINF_29T0024900</td>\n",
       "      <td>LinJ.29:850028:850959:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>paraflagellar rod protein-like</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 59
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Element LINF_350034200 and LINF_350034300\n",
    "\n",
    "* LINF_350034200 CDS ==> Change CDS to LinJ.35:1106590:1108413:-\n",
    "* LINF_350034300 ==> remove CDS"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:26:59.360722Z",
     "start_time": "2024-09-27T08:26:59.355090Z"
    }
   },
   "source": [
    "# Check the elements \"LINF_350034200\" and \"LINF_350034300\"\n",
    "df_filtered[df_filtered['Gene_ID'].isin([\"LINF_350034200\", \"LINF_350034300\"])]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID                 Gene_coord    Transcript_ID  \\\n",
       "8710  LinJ.35  LINF_350034200  LinJ.35:1106085:1108479:-  LINF_35T0034200   \n",
       "8711  LinJ.35  LINF_350034300  LinJ.35:1108768:1111622:-  LINF_35T0034300   \n",
       "\n",
       "               Transcript_coord  CDS/miscRNA_ID Pseudogen  \\\n",
       "8710  LinJ.35:1106085:1108479:-  LINF_350034200        NO   \n",
       "8711  LinJ.35:1108768:1111622:-  LINF_350034300        NO   \n",
       "\n",
       "              CDS/miscRNA_coord                                     Function  \n",
       "8710  LinJ.35:1109345:1111528:-  mt-LAF5|mitoribosomal LSU assembly factor 5  \n",
       "8711  LinJ.35:1109345:1111843:-             Hypothetical protein - conserved  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8710</th>\n",
       "      <td>LinJ.35</td>\n",
       "      <td>LINF_350034200</td>\n",
       "      <td>LinJ.35:1106085:1108479:-</td>\n",
       "      <td>LINF_35T0034200</td>\n",
       "      <td>LinJ.35:1106085:1108479:-</td>\n",
       "      <td>LINF_350034200</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.35:1109345:1111528:-</td>\n",
       "      <td>mt-LAF5|mitoribosomal LSU assembly factor 5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8711</th>\n",
       "      <td>LinJ.35</td>\n",
       "      <td>LINF_350034300</td>\n",
       "      <td>LinJ.35:1108768:1111622:-</td>\n",
       "      <td>LINF_35T0034300</td>\n",
       "      <td>LinJ.35:1108768:1111622:-</td>\n",
       "      <td>LINF_350034300</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.35:1109345:1111843:-</td>\n",
       "      <td>Hypothetical protein - conserved</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 60
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:27:01.589434Z",
     "start_time": "2024-09-27T08:27:01.579196Z"
    }
   },
   "source": [
    "# Change \"CDS/miscRNA_coord\" to \"LinJ.35:1106590:1108413:-\" in \"LINF_350034200\"\n",
    "df_filtered.loc[df_filtered['Gene_ID'] == \"LINF_350034200\", 'CDS/miscRNA_coord'] = \"LinJ.35:1106590:1108413:-\"\n",
    "\n",
    "# Remove \"CDS/miscRNA_coord\" and \"CDS/miscRNA_ID\" in \"LINF_350034300\" to \"ND\"\n",
    "df_filtered.loc[df_filtered['Gene_ID'] == \"LINF_350034300\", [\"CDS/miscRNA_ID\", \"CDS/miscRNA_coord\"]] = \"ND\"\n",
    "\n",
    "# Check the elements \"LINF_350034200\" and \"LINF_350034300\"\n",
    "df_filtered[df_filtered['Gene_ID'].isin([\"LINF_350034200\", \"LINF_350034300\"])]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID                 Gene_coord    Transcript_ID  \\\n",
       "8710  LinJ.35  LINF_350034200  LinJ.35:1106085:1108479:-  LINF_35T0034200   \n",
       "8711  LinJ.35  LINF_350034300  LinJ.35:1108768:1111622:-  LINF_35T0034300   \n",
       "\n",
       "               Transcript_coord  CDS/miscRNA_ID Pseudogen  \\\n",
       "8710  LinJ.35:1106085:1108479:-  LINF_350034200        NO   \n",
       "8711  LinJ.35:1108768:1111622:-              ND        NO   \n",
       "\n",
       "              CDS/miscRNA_coord                                     Function  \n",
       "8710  LinJ.35:1106590:1108413:-  mt-LAF5|mitoribosomal LSU assembly factor 5  \n",
       "8711                         ND             Hypothetical protein - conserved  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8710</th>\n",
       "      <td>LinJ.35</td>\n",
       "      <td>LINF_350034200</td>\n",
       "      <td>LinJ.35:1106085:1108479:-</td>\n",
       "      <td>LINF_35T0034200</td>\n",
       "      <td>LinJ.35:1106085:1108479:-</td>\n",
       "      <td>LINF_350034200</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.35:1106590:1108413:-</td>\n",
       "      <td>mt-LAF5|mitoribosomal LSU assembly factor 5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8711</th>\n",
       "      <td>LinJ.35</td>\n",
       "      <td>LINF_350034300</td>\n",
       "      <td>LinJ.35:1108768:1111622:-</td>\n",
       "      <td>LINF_35T0034300</td>\n",
       "      <td>LinJ.35:1108768:1111622:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>Hypothetical protein - conserved</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 61
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "### Doubts resolved with @Javi and @JMR"
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The question to @JMR about the main table was:\n",
    "\n",
    "***\n",
    "<div style=\"color:orange\";>\n",
    "Hola buenas José,\n",
    "\n",
    "Revisando la tabla maestra hemos observado los siguientes casos:\n",
    "\n",
    "* *Pseudogenes*: las \"CDS/misRNA\" coordenadas se salen de las coordenadas de \"Gene_coord\". Hemos pensado en eliminar el CDS.\n",
    "    * **LINF_110011400**\n",
    "      * Gene_coord: <u>202887</u> : 204220\n",
    "      * CDS/miscRNA_coord: <u>202513</u> : 204017\n",
    "  \n",
    "    * **LINF_140012250**\n",
    "      * Gene_coord: 248820 : <w>251321</w>\n",
    "      * CDS/miscRNA_coord: 250151 : <w>251383</w>\n",
    "\n",
    "* *Histone H4*:\n",
    "  * **LINF_150005000** -→ las coordenadas \"CDS/misRNA\" se escapan de las coordenadas de \"Gene_coord\".\n",
    "    * Hemos observado que el tamaño de la secuencia es de 211 nucleótidos, en comparación al resto de H4 que son de 303 nucleótidos.\n",
    "    * Gene_coord: 1108 : <u>1687</u>\n",
    "    * miscRNA_coord: 1505 : <u>1715</u>\n",
    "\n",
    "* **LINF_360079700** =→ Las coordenadas del transcrito \"Transcript_coord\" se sobresalen de las coordenadas de \"Gene_coord\".\n",
    "  * Gene_coord: <u>2681995</u> : <u>2683375</u>\n",
    "  * Transcript_coord: <u>2681395</u> : <u>2683433</u>\n",
    "\n",
    "¡Espero que tengáis un buen día!\n",
    "\n",
    "R. Pacheco\n",
    "</div>\n",
    "\n",
    "***"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "#### Removing CDS from pseudogenes"
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "And the answer was:\n",
    "  * **LINF_110011400**: pseudogene and remove CDS\n",
    "  * **LINF_140012250** pseudogenes and remove CDS"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:27:34.022923Z",
     "start_time": "2024-09-27T08:27:34.016957Z"
    }
   },
   "source": [
    "# Check the elements and the \"CDS/miscRNA_ID\" values\n",
    "df_filtered[df_filtered['Gene_ID'].isin([\"LINF_110011400\", \"LINF_140012250\"])]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID               Gene_coord    Transcript_ID  \\\n",
       "1551  LinJ.11  LINF_110011400  LinJ.11:202887:204220:+  LINF_11T0011400   \n",
       "2067  LinJ.14  LINF_140012250  LinJ.14:248820:251321:-  LINF_14T0012250   \n",
       "\n",
       "             Transcript_coord  CDS/miscRNA_ID Pseudogen  \\\n",
       "1551  LinJ.11:202887:204220:+  LINF_110011400       YES   \n",
       "2067  LinJ.14:248820:251321:-  LINF_140012250       YES   \n",
       "\n",
       "            CDS/miscRNA_coord                             Function  \n",
       "1551  LinJ.11:202513:204017:+  argonaute like protein (pseudogene)  \n",
       "2067  LinJ.14:250151:251383:-                  fatty acid elongase  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1551</th>\n",
       "      <td>LinJ.11</td>\n",
       "      <td>LINF_110011400</td>\n",
       "      <td>LinJ.11:202887:204220:+</td>\n",
       "      <td>LINF_11T0011400</td>\n",
       "      <td>LinJ.11:202887:204220:+</td>\n",
       "      <td>LINF_110011400</td>\n",
       "      <td>YES</td>\n",
       "      <td>LinJ.11:202513:204017:+</td>\n",
       "      <td>argonaute like protein (pseudogene)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2067</th>\n",
       "      <td>LinJ.14</td>\n",
       "      <td>LINF_140012250</td>\n",
       "      <td>LinJ.14:248820:251321:-</td>\n",
       "      <td>LINF_14T0012250</td>\n",
       "      <td>LinJ.14:248820:251321:-</td>\n",
       "      <td>LINF_140012250</td>\n",
       "      <td>YES</td>\n",
       "      <td>LinJ.14:250151:251383:-</td>\n",
       "      <td>fatty acid elongase</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 62
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:27:36.626828Z",
     "start_time": "2024-09-27T08:27:36.620322Z"
    }
   },
   "source": [
    "# Replacing with \"ND\"\n",
    "df_filtered.loc[df_filtered['Gene_ID'].isin([\"LINF_110011400\", \"LINF_140012250\"]), 'CDS/miscRNA_coord'] = \"ND\"\n",
    "\n",
    "# Check the elements\n",
    "df_filtered[df_filtered['Gene_ID'].isin([\"LINF_110011400\", \"LINF_140012250\"])]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID               Gene_coord    Transcript_ID  \\\n",
       "1551  LinJ.11  LINF_110011400  LinJ.11:202887:204220:+  LINF_11T0011400   \n",
       "2067  LinJ.14  LINF_140012250  LinJ.14:248820:251321:-  LINF_14T0012250   \n",
       "\n",
       "             Transcript_coord  CDS/miscRNA_ID Pseudogen CDS/miscRNA_coord  \\\n",
       "1551  LinJ.11:202887:204220:+  LINF_110011400       YES                ND   \n",
       "2067  LinJ.14:248820:251321:-  LINF_140012250       YES                ND   \n",
       "\n",
       "                                 Function  \n",
       "1551  argonaute like protein (pseudogene)  \n",
       "2067                  fatty acid elongase  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1551</th>\n",
       "      <td>LinJ.11</td>\n",
       "      <td>LINF_110011400</td>\n",
       "      <td>LinJ.11:202887:204220:+</td>\n",
       "      <td>LINF_11T0011400</td>\n",
       "      <td>LinJ.11:202887:204220:+</td>\n",
       "      <td>LINF_110011400</td>\n",
       "      <td>YES</td>\n",
       "      <td>ND</td>\n",
       "      <td>argonaute like protein (pseudogene)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2067</th>\n",
       "      <td>LinJ.14</td>\n",
       "      <td>LINF_140012250</td>\n",
       "      <td>LinJ.14:248820:251321:-</td>\n",
       "      <td>LINF_14T0012250</td>\n",
       "      <td>LinJ.14:248820:251321:-</td>\n",
       "      <td>LINF_140012250</td>\n",
       "      <td>YES</td>\n",
       "      <td>ND</td>\n",
       "      <td>fatty acid elongase</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 63
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Repairing Histone H4 LINF_150005000\n",
    "\n",
    "* New CDS coordinates: LinJ.15:1308:1505:-\n",
    "* New CDS sequence: ATGGCGCGCCGCGGTGGCGTGAAGCGCATCTCGAGCGAGGTCTACGAAGAGGTGCGCCGCGTGCTGAAGGCCTACGTGGAGGACATTGTGCGCTGCAGCACGGCCTACACCGAGTACGCGCGCAAGAAGACGGTGACGGCGTGCGATGTTGTGAACGCGCTGCGCAAGCAAGGCCACATCCTCTACGGCTATGCGTAA"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:28:08.511624Z",
     "start_time": "2024-09-27T08:28:08.504132Z"
    }
   },
   "source": [
    "# Check the element \"LINF_150005000\"\n",
    "df_filtered[df_filtered['Gene_ID'] == \"LINF_150005000\"]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID           Gene_coord    Transcript_ID  \\\n",
       "2174  LinJ.15  LINF_150005000  LinJ.15:1108:1687:-  LINF_15T0005000   \n",
       "\n",
       "         Transcript_coord  CDS/miscRNA_ID Pseudogen    CDS/miscRNA_coord  \\\n",
       "2174  LinJ.15:1108:1687:-  LINF_150005000        NO  LinJ.15:1505:1715:-   \n",
       "\n",
       "        Function  \n",
       "2174  histone H4  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2174</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150005000</td>\n",
       "      <td>LinJ.15:1108:1687:-</td>\n",
       "      <td>LINF_15T0005000</td>\n",
       "      <td>LinJ.15:1108:1687:-</td>\n",
       "      <td>LINF_150005000</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.15:1505:1715:-</td>\n",
       "      <td>histone H4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 64
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:28:11.194085Z",
     "start_time": "2024-09-27T08:28:11.187365Z"
    }
   },
   "source": [
    "# Change the coordinates from \"LinJ.15:1505:1715:-\" to \"LinJ.15:1308:1505:-\"\n",
    "df_filtered.loc[df_filtered['Gene_ID'] == \"LINF_150005000\", 'CDS/miscRNA_coord'] = \"inJ.15:1308:1505:-\"\n",
    "\n",
    "# Check the change\n",
    "df_filtered[df_filtered['Gene_ID'] == \"LINF_150005000\"]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID           Gene_coord    Transcript_ID  \\\n",
       "2174  LinJ.15  LINF_150005000  LinJ.15:1108:1687:-  LINF_15T0005000   \n",
       "\n",
       "         Transcript_coord  CDS/miscRNA_ID Pseudogen   CDS/miscRNA_coord  \\\n",
       "2174  LinJ.15:1108:1687:-  LINF_150005000        NO  inJ.15:1308:1505:-   \n",
       "\n",
       "        Function  \n",
       "2174  histone H4  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2174</th>\n",
       "      <td>LinJ.15</td>\n",
       "      <td>LINF_150005000</td>\n",
       "      <td>LinJ.15:1108:1687:-</td>\n",
       "      <td>LINF_15T0005000</td>\n",
       "      <td>LinJ.15:1108:1687:-</td>\n",
       "      <td>LINF_150005000</td>\n",
       "      <td>NO</td>\n",
       "      <td>inJ.15:1308:1505:-</td>\n",
       "      <td>histone H4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 65
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Transcript LINF_360079700 coordinates outside gene coordinates\n",
    "\n",
    "* New gene coordinates = LinJ.36:2681395:2683433:-\n",
    "* New CDS coordinates = LinJ.36:2682378:2682647:-  (I think it was already like this).\n"
   ]
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:28:21.706030Z",
     "start_time": "2024-09-27T08:28:21.700046Z"
    }
   },
   "source": [
    "# Check the element \"LINF_360079700\"\n",
    "df_filtered[df_filtered['Gene_ID'] == \"LINF_360079700\"]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID                 Gene_coord    Transcript_ID  \\\n",
       "9840  LinJ.36  LINF_360079700  LinJ.36:2681995:2683375:-  LINF_36T0079700   \n",
       "\n",
       "               Transcript_coord  CDS/miscRNA_ID Pseudogen  \\\n",
       "9840  LinJ.36:2681395:2683433:-  LINF_360079700        SI   \n",
       "\n",
       "              CDS/miscRNA_coord Function  \n",
       "9840  LinJ.36:2682378:2682647:-     SET8  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9840</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>LINF_360079700</td>\n",
       "      <td>LinJ.36:2681995:2683375:-</td>\n",
       "      <td>LINF_36T0079700</td>\n",
       "      <td>LinJ.36:2681395:2683433:-</td>\n",
       "      <td>LINF_360079700</td>\n",
       "      <td>SI</td>\n",
       "      <td>LinJ.36:2682378:2682647:-</td>\n",
       "      <td>SET8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 66
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:28:28.069618Z",
     "start_time": "2024-09-27T08:28:28.061442Z"
    }
   },
   "source": [
    "# Let's change the gene coordinates from \"LinJ.36:2681995:2683375:-\" to \"LinJ.36:2681395:2683433:-\"\n",
    "df_filtered.loc[df_filtered['Gene_ID'] == \"LINF_360079700\", 'Gene_coord'] = \"LinJ.36:2681395:2683433:-\"\n",
    "\n",
    "# Check the change\n",
    "df_filtered[df_filtered['Gene_ID'] == \"LINF_360079700\"]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom         Gene_ID                 Gene_coord    Transcript_ID  \\\n",
       "9840  LinJ.36  LINF_360079700  LinJ.36:2681395:2683433:-  LINF_36T0079700   \n",
       "\n",
       "               Transcript_coord  CDS/miscRNA_ID Pseudogen  \\\n",
       "9840  LinJ.36:2681395:2683433:-  LINF_360079700        SI   \n",
       "\n",
       "              CDS/miscRNA_coord Function  \n",
       "9840  LinJ.36:2682378:2682647:-     SET8  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9840</th>\n",
       "      <td>LinJ.36</td>\n",
       "      <td>LINF_360079700</td>\n",
       "      <td>LinJ.36:2681395:2683433:-</td>\n",
       "      <td>LINF_36T0079700</td>\n",
       "      <td>LinJ.36:2681395:2683433:-</td>\n",
       "      <td>LINF_360079700</td>\n",
       "      <td>SI</td>\n",
       "      <td>LinJ.36:2682378:2682647:-</td>\n",
       "      <td>SET8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 67
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": "# Creating GTF"
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:31:25.098808Z",
     "start_time": "2024-09-27T08:31:25.096455Z"
    }
   },
   "source": [
    "# LINF_220021000  # Removed before because it didn't have Gene_coord\n",
    "javi_ignore = [\"LINF_360065300\"]  # #Could be bi-cistronic, need more experimental data\n",
    "cds_extended_unusual_ignore = [\n",
    "    # \"LINF_110011400\",  # Already fixed\n",
    "    # \"LINF_140012250\",  # Already fixed\n",
    "    # \"LINF_150005000\",   # Already fixed\n",
    "    # \"LINF_230015300\",  # Already fixed\n",
    "    # \"LINF_290024900\",  # Already fixed\n",
    "    # \"LINF_350034200\",  # Already fixed\n",
    "    # \"LINF_350034300\"  # Already fixed\n",
    "    ]\n",
    "transcript_extended_unusual = [\n",
    "    # \"LINF_360079700\"  # Already fixed\n",
    "    ]\n",
    "\n",
    "# Sum every string\n",
    "ignore_gene_ids = javi_ignore + cds_extended_unusual_ignore + transcript_extended_unusual"
   ],
   "outputs": [],
   "execution_count": 68
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:31:29.687912Z",
     "start_time": "2024-09-27T08:31:29.681897Z"
    }
   },
   "source": [
    "df_filtered[df_filtered['Gene_ID'] == \"LINF_010007050\"]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "      Chrom         Gene_ID             Gene_coord    Transcript_ID  \\\n",
       "21  LinJ.01  LINF_010007050  LinJ.01:54441:56024:-  LINF_01T0007050   \n",
       "\n",
       "         Transcript_coord CDS/miscRNA_ID Pseudogen CDS/miscRNA_coord Function  \n",
       "21  LinJ.01:54441:56024:-             ND        NO                ND  unknown  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>LINF_010007050</td>\n",
       "      <td>LinJ.01:54441:56024:-</td>\n",
       "      <td>LINF_01T0007050</td>\n",
       "      <td>LinJ.01:54441:56024:-</td>\n",
       "      <td>ND</td>\n",
       "      <td>NO</td>\n",
       "      <td>ND</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 69
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "## Cleaning Function column"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T08:57:16.494663Z",
     "start_time": "2024-09-27T08:57:16.489584Z"
    }
   },
   "cell_type": "code",
   "source": "df_filtered[\"Function\"].value_counts()",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Function\n",
       "hypothetical_protein_-_conserved                                                                                    1704\n",
       "unknown                                                                                                             1101\n",
       "Protein of unknown function - conserved                                                                              527\n",
       "protein of unknown function - conserved                                                                              241\n",
       "Hypothetical protein - conserved                                                                                     185\n",
       "                                                                                                                    ... \n",
       "argininosuccinate synthase|Citrulline-aspartate ligase                                                                 1\n",
       "Pteridine reductase 1|PTR1                                                                                             1\n",
       "Checkpoint protein HUS1                                                                                                1\n",
       "Tryptophanyl-tRNA synthetase|mt-TrpRS                                                                                  1\n",
       "Putative_methyltransferase/Methyltransferase_small_domain/Methyltransferase_domain_containing_protein_-_putative       1\n",
       "Name: count, Length: 4477, dtype: int64"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 72
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "As we can see there are a lot of names that are the same, but are divided because of case or other small characters differences"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T10:02:00.769377Z",
     "start_time": "2024-09-27T10:02:00.730362Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Let's create a function with replace mapping to start replacing all of them.\n",
    "with open(\"./data/tabla_maestra_function_corrector.json\") as json_file:\n",
    "    replace_map = json.load(json_file)\n",
    "\n",
    "def repairing_functions_typos(replacing_map, data_frame, column_name):\n",
    "    # Inverting dictionary\n",
    "    inverse_replace_map = {value: key for key in replacing_map for value in replacing_map[key]}\n",
    "    \n",
    "    # Let's copy the data_frame\n",
    "    data_frame_cp = data_frame.copy()\n",
    "    \n",
    "    # Let's replace \"_\" for \" \" in the column \"column_name\" of `data_frame_cp` \n",
    "    data_frame_cp[column_name] = data_frame_cp[column_name].str.replace(\"_\", \" \")\n",
    "    \n",
    "    # Applying the mapping\n",
    "    data_frame_cp[column_name] = data_frame_cp[column_name].map(inverse_replace_map).fillna(data_frame_cp[column_name])\n",
    "    \n",
    "    return data_frame_cp\n",
    "\n",
    "repaired_function_df = repairing_functions_typos(replace_map, df_filtered, \"Function\")\n",
    "repaired_function_df.to_csv(\"./data/tabla_maestra_function_corrector.csv\", index=False, header=0, sep=\",\")  ## To keep correcting later on\n",
    "repaired_function_df['Function'].value_counts()"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Function\n",
       "hypothetical protein - conserved                                                 2023\n",
       "unknown                                                                          1101\n",
       "protein of unknown function - conserved                                           768\n",
       "hypothetical protein                                                              187\n",
       "protein kinase                                                                     70\n",
       "                                                                                 ... \n",
       "Protein of unknown function - conserved (L1p/L10e family)                           1\n",
       "DNA repair and recombination helicase protein PIF6|ATP-dependent DNA helicase       1\n",
       "thymidine kinase                                                                    1\n",
       "Flagellum attachment zone protein 6|FAZ6                                            1\n",
       "Domain of unknown function (DUF4499)                                                1\n",
       "Name: count, Length: 4438, dtype: int64"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 104
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T10:07:19.004926Z",
     "start_time": "2024-09-27T10:07:19.001605Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Let's change df_filtered to the data in repaired_functions_df\n",
    "df_filtered = repaired_function_df.copy()"
   ],
   "outputs": [],
   "execution_count": 118
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "## Creating GTF"
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T10:07:19.248322Z",
     "start_time": "2024-09-27T10:07:19.244165Z"
    }
   },
   "source": [
    "def cds_for_gtf(gtf_row, gtf_transcript_id, gtf_transcript=False):\n",
    "    gtf_feature = \"CDS\"\n",
    "    gtf_start = gtf_row['CDS/miscRNA_coord'].split(':')[1]\n",
    "    gtf_end = gtf_row['CDS/miscRNA_coord'].split(':')[2]\n",
    "    if gtf_transcript:\n",
    "        gtf_attribute = f'parent_id \"{gtf_transcript_id}\"; transcript_id \"{gtf_transcript_id}\"; '\n",
    "    else:\n",
    "        gtf_attribute = f'parent_id \"{gtf_row[\"Gene_ID\"]}\"; '\n",
    "    return gtf_feature, gtf_start, gtf_end, gtf_attribute\n",
    "\n",
    "def left_utr(gtf_row, gtf_transcript_id, gtf_transcript=False):\n",
    "    gtf_gene_start = int(gtf_row['Gene_coord'].split(':')[1])\n",
    "    # gtf_gene_end = int(gtf_row['Gene_coord'].split(':')[2])\n",
    "    gtf_sense = gtf_row['Gene_coord'].split(':')[3]\n",
    "\n",
    "    gtf_cds_start = int(gtf_row['CDS/miscRNA_coord'].split(':')[1])\n",
    "    # gtf_cds_end = int(gtf_row['CDS/miscRNA_coord'].split(':')[2])\n",
    "\n",
    "    if gtf_gene_start != gtf_cds_start:\n",
    "        gtf_feature = '3utr' if gtf_sense == '-' else '5utr'  # if the strand is \"minus\" the left UTR is the 3'UTR, since the transcript fo 3'utr <-----\n",
    "        gtf_start = str(gtf_gene_start)\n",
    "        gtf_end = str(gtf_cds_start - 1)\n",
    "        gtf_strand = gtf_sense\n",
    "        gtf_attribute = (\n",
    "            (f'parent_id \"{gtf_transcript_id}\"; ' if gtf_transcript else f'parent_id \"{gtf_row[\"Gene_ID\"]}\"; ')\n",
    "        )\n",
    "        return gtf_feature, gtf_start, gtf_end, gtf_strand, gtf_attribute\n",
    "    else:\n",
    "        return None, None, None, None, None\n",
    "\n",
    "\n",
    "def right_utr(gtf_row, gtf_transcript_id, gtf_transcript=False):\n",
    "    # utr_gene_start = int(gtf_row['Gene_coord'].split(':')[1])\n",
    "    utr_gene_end = int(gtf_row['Gene_coord'].split(':')[2])\n",
    "    # utr_cds_start = int(gtf_row['CDS/miscRNA_coord'].split(':')[1])\n",
    "    utr_cds_end = int(gtf_row['CDS/miscRNA_coord'].split(':')[2])\n",
    "    utr_sense = gtf_row['Gene_coord'].split(':')[3]\n",
    "\n",
    "    if utr_gene_end != utr_cds_end:\n",
    "        utr_feature = '5utr' if utr_sense == '-' else '3utr'\n",
    "        utr_start = str(utr_cds_end + 1)\n",
    "        utr_end = str(utr_gene_end)\n",
    "        utr_strand = utr_sense\n",
    "        utr_attribute = (\n",
    "            (f'parent_id \"{gtf_transcript_id}\"; ' if gtf_transcript else f'parent_id \"{gtf_row[\"Gene_ID\"]}\"; ')\n",
    "        )\n",
    "        return utr_feature, utr_start, utr_end, utr_strand, utr_attribute\n",
    "    else:\n",
    "        return None, None, None, None, None"
   ],
   "outputs": [],
   "execution_count": 119
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T10:07:20.035633Z",
     "start_time": "2024-09-27T10:07:19.297354Z"
    }
   },
   "source": [
    "# Let's index the dataframe and add the columns to a new list\n",
    "pre_gtf = []\n",
    "for index, row in df_filtered.iterrows():\n",
    "    if row['Gene_ID'] in ignore_gene_ids:  # Because of the gene_id, we skip this row\n",
    "        print(\"=\" * 100)\n",
    "        print(f\"Skipping row {index} with Gene_ID '{row['Gene_ID']}' because it is in the ignore list\")\n",
    "        continue\n",
    "    # GENE ID\n",
    "    # noinspection SpellCheckingInspection\n",
    "    seqname = row['Chrom']\n",
    "    source = 'CBM'\n",
    "    feature = 'gene'\n",
    "    start = row['Gene_coord'].split(':')[1]\n",
    "    end = row['Gene_coord'].split(':')[2]\n",
    "    score = '.'\n",
    "    strand = row['Gene_coord'].split(':')[3]\n",
    "    frame = '.'\n",
    "    attribute = (\n",
    "        f'gene_id \"{row['Gene_ID']}\"; ' + f'gene_name \"{row[\"Gene_ID\"]}\"; ' +   # gene_id needed for the gtf; gene_name for the display in IGV\n",
    "        (f'pseudogen \"unknown\"; ' if row['Pseudogen'] in [\"YES\", \"SI\"] else \"\") +  # The original table have \"YES\" and \"SI\" for pseudogen :/\n",
    "        (f'biotype \"protein_coding\"; ' if row['CDS/miscRNA_ID'] != \"ND\" else \"\") +  # biotype is protein_coding if there is a CDS/miscRNA_ID\n",
    "        (f'notes \"{row['Function'].replace(\" \",\"_\").replace(\"\\t\", \"\")}\"; ' if row['Function'] != \"unknown\" else \"\") # Replace space with underscore, there can't be spaces inside the quotes\n",
    "    )\n",
    "\n",
    "    pre_gtf.append([seqname, source, feature, start, end, score, strand, frame, attribute])\n",
    "\n",
    "    # TRANSCRIPT ID\n",
    "    transcript = False\n",
    "    if row['Transcript_ID'] != \"ND\":  # Elements that have a transcript and a CDS/miscRNA_ID\n",
    "        transcript = True\n",
    "        feature = 'transcript'\n",
    "        start = row['Transcript_coord'].split(':')[1]\n",
    "        end = row['Transcript_coord'].split(':')[2]\n",
    "\n",
    "        transcript_id = row['Transcript_ID']\n",
    "        attribute = (\n",
    "            f'parent_id \"{row['Gene_ID']}\"; transcript_id \"{transcript_id}\"; transcript_name \"{transcript_id}\"; ' +  # transcript_id groups exon, CDS and other elements of the transcript. It is needed for the gtf display!!!!\n",
    "            (f'biotype \"protein_coding\"; ' if row['CDS/miscRNA_ID'] != \"ND\" else \"\") + # biotype is protein_coding if there is a CDS/miscRNA_ID\n",
    "            (f'notes \"{row['Function'].replace(\" \",\"_\").replace(\"\\t\", \"\")}\"; ' if row['Function'] != \"unknown\" else \"\") # Replace space with underscore, there can't be spaces inside the quotes\n",
    "        )\n",
    "        pre_gtf.append([seqname, source, feature, start, end, score, strand, frame, attribute])\n",
    "\n",
    "        # CDS annotation\n",
    "        if row['CDS/miscRNA_ID'] != \"ND\" and row['CDS/miscRNA_coord'] != \"ND\":  # Where the cds exists\n",
    "            feature, start, end, attribute = cds_for_gtf(row, transcript_id, transcript)  # transcript is TRUE\n",
    "            pre_gtf.append([seqname, source, feature, start, end, score, strand, frame, attribute]) \n",
    "\n",
    "            # UTR annotation\n",
    "            feature, start, end, strand, attribute = left_utr(row, transcript_id, transcript)  # transcript is TRUE\n",
    "            if feature is not None:\n",
    "                pre_gtf.append([seqname, source, feature, start, end, score, strand, frame, attribute])\n",
    "\n",
    "            # UTR annotation\n",
    "            feature, start, end, strand, attribute = right_utr(row, transcript_id, transcript)  # transcript is TRUE\n",
    "            if feature is not None:\n",
    "                pre_gtf.append([seqname, source, feature, start, end, score, strand, frame, attribute])\n",
    "\n",
    "    elif row['CDS/miscRNA_ID'] != \"ND\" and row['CDS/miscRNA_coord'] != \"ND\":  # but Transcript_ID is ND\n",
    "        feature, start, end, attribute = cds_for_gtf(row, transcript_id, transcript)  # transcript is FALSE\n",
    "        pre_gtf.append([seqname, source, feature, start, end, score, strand, frame, attribute]) \n",
    "\n",
    "        feature, start, end, strand, attribute = left_utr(row, transcript_id, transcript)  # transcript is FALSE\n",
    "        if feature is not None:\n",
    "            pre_gtf.append([seqname, source, feature, start, end, score, strand, frame, attribute])\n",
    "\n",
    "        feature, start, end, strand, attribute = right_utr(row, transcript_id, transcript)  # transcript is FALSE\n",
    "        if feature is not None:\n",
    "            pre_gtf.append([seqname, source, feature, start, end, score, strand, frame, attribute])"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "====================================================================================================\n",
      "Skipping row 9677 with Gene_ID 'LINF_360065300' because it is in the ignore list\n"
     ]
    }
   ],
   "execution_count": 120
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T10:07:20.038982Z",
     "start_time": "2024-09-27T10:07:20.036936Z"
    }
   },
   "source": [
    "# Create a list with the names of the columns for the GTF file:\n",
    "# noinspection SpellCheckingInspection\n",
    "gtf_columns = ['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attribute']  # type: ignore"
   ],
   "outputs": [],
   "execution_count": 121
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T10:07:20.059426Z",
     "start_time": "2024-09-27T10:07:20.039556Z"
    }
   },
   "source": [
    "# Checking columns data types\n",
    "gtf_df = pd.DataFrame(pre_gtf, columns=gtf_columns)\n",
    "print(gtf_df.shape)\n",
    "print(gtf_df.dtypes)\n",
    "gtf_df.head()"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(45376, 9)\n",
      "seqname      object\n",
      "source       object\n",
      "feature      object\n",
      "start        object\n",
      "end          object\n",
      "score        object\n",
      "strand       object\n",
      "frame        object\n",
      "attribute    object\n",
      "dtype: object\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "   seqname source     feature start   end score strand frame  \\\n",
       "0  LinJ.01    CBM        gene  1520  5066     .      -     .   \n",
       "1  LinJ.01    CBM  transcript  1520  5066     .      -     .   \n",
       "2  LinJ.01    CBM         CDS  3710  4711     .      -     .   \n",
       "3  LinJ.01    CBM        3utr  1520  3709     .      -     .   \n",
       "4  LinJ.01    CBM        5utr  4712  5066     .      -     .   \n",
       "\n",
       "                                           attribute  \n",
       "0  gene_id \"LINF_010005000\"; gene_name \"LINF_0100...  \n",
       "1  parent_id \"LINF_010005000\"; transcript_id \"LIN...  \n",
       "2  parent_id \"LINF_01T0005000\"; transcript_id \"LI...  \n",
       "3                      parent_id \"LINF_01T0005000\";   \n",
       "4                      parent_id \"LINF_01T0005000\";   "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqname</th>\n",
       "      <th>source</th>\n",
       "      <th>feature</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>frame</th>\n",
       "      <th>attribute</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>1520</td>\n",
       "      <td>5066</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_010005000\"; gene_name \"LINF_0100...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>CBM</td>\n",
       "      <td>transcript</td>\n",
       "      <td>1520</td>\n",
       "      <td>5066</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>parent_id \"LINF_010005000\"; transcript_id \"LIN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>CBM</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3710</td>\n",
       "      <td>4711</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>parent_id \"LINF_01T0005000\"; transcript_id \"LI...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>CBM</td>\n",
       "      <td>3utr</td>\n",
       "      <td>1520</td>\n",
       "      <td>3709</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>parent_id \"LINF_01T0005000\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>CBM</td>\n",
       "      <td>5utr</td>\n",
       "      <td>4712</td>\n",
       "      <td>5066</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>parent_id \"LINF_01T0005000\";</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 122
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T10:07:20.065460Z",
     "start_time": "2024-09-27T10:07:20.060294Z"
    }
   },
   "source": [
    "# Checking two-types values for feature and strand\n",
    "print(gtf_df.value_counts(\"feature\"))\n",
    "print(gtf_df.value_counts(\"strand\"))"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "feature\n",
      "gene          9869\n",
      "transcript    9660\n",
      "CDS           8744\n",
      "3utr          8554\n",
      "5utr          8549\n",
      "Name: count, dtype: int64\n",
      "strand\n",
      "+    24190\n",
      "-    21186\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "execution_count": 123
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T10:07:20.105176Z",
     "start_time": "2024-09-27T10:07:20.066035Z"
    }
   },
   "source": [
    "# Changing data types for start and end columns to integers\n",
    "gtf_df[['start', 'end']] = gtf_df[['start', 'end']].apply(pd.to_numeric)\n",
    "gtf_df.dtypes"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "seqname      object\n",
       "source       object\n",
       "feature      object\n",
       "start         int64\n",
       "end           int64\n",
       "score        object\n",
       "strand       object\n",
       "frame        object\n",
       "attribute    object\n",
       "dtype: object"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 124
  },
  {
   "cell_type": "code",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    },
    "ExecuteTime": {
     "end_time": "2024-09-27T10:07:20.135698Z",
     "start_time": "2024-09-27T10:07:20.105711Z"
    }
   },
   "source": [
    "# Get duplicated rows in the data frame\n",
    "gtf_df[gtf_df.duplicated(keep=False)]\n"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "       seqname source feature    start      end score strand frame  \\\n",
       "3453   LinJ.06    CBM    gene   487685   490191     .      +     .   \n",
       "3458   LinJ.06    CBM    gene   487685   490191     .      +     .   \n",
       "3734   LinJ.07    CBM    gene   153281   160411     .      -     .   \n",
       "3739   LinJ.07    CBM    gene   153281   160411     .      -     .   \n",
       "7882   LinJ.12    CBM    gene   440484   443559     .      +     .   \n",
       "7887   LinJ.12    CBM    gene   440484   443559     .      +     .   \n",
       "7902   LinJ.12    CBM    gene   450370   452072     .      +     .   \n",
       "7907   LinJ.12    CBM    gene   450370   452072     .      +     .   \n",
       "19603  LinJ.24    CBM    gene   689866   693427     .      -     .   \n",
       "19608  LinJ.24    CBM    gene   689866   693427     .      -     .   \n",
       "24266  LinJ.27    CBM    gene  1153377  1155197     .      +     .   \n",
       "24271  LinJ.27    CBM    gene  1153377  1155197     .      +     .   \n",
       "27569  LinJ.29    CBM    gene  1181710  1184611     .      +     .   \n",
       "27574  LinJ.29    CBM    gene  1181710  1184611     .      +     .   \n",
       "34821  LinJ.33    CBM    gene   553571   558357     .      +     .   \n",
       "34826  LinJ.33    CBM    gene   553571   558357     .      +     .   \n",
       "\n",
       "                                               attribute  \n",
       "3453   gene_id \"LINF_060017800-900\"; gene_name \"LINF_...  \n",
       "3458   gene_id \"LINF_060017800-900\"; gene_name \"LINF_...  \n",
       "3734   gene_id \"LINF_070008800\"; gene_name \"LINF_0700...  \n",
       "3739   gene_id \"LINF_070008800\"; gene_name \"LINF_0700...  \n",
       "7882   gene_id \"LINF_120014100-200\"; gene_name \"LINF_...  \n",
       "7887   gene_id \"LINF_120014100-200\"; gene_name \"LINF_...  \n",
       "7902   gene_id \"LINF_120014500-600\"; gene_name \"LINF_...  \n",
       "7907   gene_id \"LINF_120014500-600\"; gene_name \"LINF_...  \n",
       "19603  gene_id \"LINF_240023900\"; gene_name \"LINF_2400...  \n",
       "19608  gene_id \"LINF_240023900\"; gene_name \"LINF_2400...  \n",
       "24266  gene_id \"LINF_270033600-700\"; gene_name \"LINF_...  \n",
       "24271  gene_id \"LINF_270033600-700\"; gene_name \"LINF_...  \n",
       "27569  gene_id \"LINF_290033800\"; gene_name \"LINF_2900...  \n",
       "27574  gene_id \"LINF_290033800\"; gene_name \"LINF_2900...  \n",
       "34821  gene_id \"LINF_330021700-800\"; gene_name \"LINF_...  \n",
       "34826  gene_id \"LINF_330021700-800\"; gene_name \"LINF_...  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqname</th>\n",
       "      <th>source</th>\n",
       "      <th>feature</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>frame</th>\n",
       "      <th>attribute</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3453</th>\n",
       "      <td>LinJ.06</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>487685</td>\n",
       "      <td>490191</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_060017800-900\"; gene_name \"LINF_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3458</th>\n",
       "      <td>LinJ.06</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>487685</td>\n",
       "      <td>490191</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_060017800-900\"; gene_name \"LINF_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3734</th>\n",
       "      <td>LinJ.07</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>153281</td>\n",
       "      <td>160411</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_070008800\"; gene_name \"LINF_0700...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3739</th>\n",
       "      <td>LinJ.07</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>153281</td>\n",
       "      <td>160411</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_070008800\"; gene_name \"LINF_0700...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7882</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>440484</td>\n",
       "      <td>443559</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_120014100-200\"; gene_name \"LINF_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7887</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>440484</td>\n",
       "      <td>443559</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_120014100-200\"; gene_name \"LINF_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7902</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>450370</td>\n",
       "      <td>452072</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_120014500-600\"; gene_name \"LINF_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7907</th>\n",
       "      <td>LinJ.12</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>450370</td>\n",
       "      <td>452072</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_120014500-600\"; gene_name \"LINF_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19603</th>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>689866</td>\n",
       "      <td>693427</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_240023900\"; gene_name \"LINF_2400...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19608</th>\n",
       "      <td>LinJ.24</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>689866</td>\n",
       "      <td>693427</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_240023900\"; gene_name \"LINF_2400...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24266</th>\n",
       "      <td>LinJ.27</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>1153377</td>\n",
       "      <td>1155197</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_270033600-700\"; gene_name \"LINF_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24271</th>\n",
       "      <td>LinJ.27</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>1153377</td>\n",
       "      <td>1155197</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_270033600-700\"; gene_name \"LINF_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27569</th>\n",
       "      <td>LinJ.29</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>1181710</td>\n",
       "      <td>1184611</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_290033800\"; gene_name \"LINF_2900...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27574</th>\n",
       "      <td>LinJ.29</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>1181710</td>\n",
       "      <td>1184611</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_290033800\"; gene_name \"LINF_2900...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34821</th>\n",
       "      <td>LinJ.33</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>553571</td>\n",
       "      <td>558357</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_330021700-800\"; gene_name \"LINF_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34826</th>\n",
       "      <td>LinJ.33</td>\n",
       "      <td>CBM</td>\n",
       "      <td>gene</td>\n",
       "      <td>553571</td>\n",
       "      <td>558357</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"LINF_330021700-800\"; gene_name \"LINF_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 125
  },
  {
   "cell_type": "code",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    },
    "ExecuteTime": {
     "end_time": "2024-09-27T10:07:20.146264Z",
     "start_time": "2024-09-27T10:07:20.136808Z"
    }
   },
   "source": [
    "# This one was duplicated in the original data frame, each one with different \"CDS/miscRNA_coord\" values\n",
    "df_filtered[df_filtered['Gene_ID'] == \"LINF_270033600-700\"]"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Chrom             Gene_ID                 Gene_coord  \\\n",
       "5286  LinJ.27  LINF_270033600-700  LinJ.27:1153377:1155197:+   \n",
       "5287  LinJ.27  LINF_270033600-700  LinJ.27:1153377:1155197:+   \n",
       "\n",
       "              Transcript_ID           Transcript_coord  CDS/miscRNA_ID  \\\n",
       "5286  LINF_27T0033600-700_1  LinJ.27:1153377:1155197:+  LINF_270033600   \n",
       "5287  LINF_27T0033600-700_2  LinJ.27:1153377:1155197:+  LINF_270033700   \n",
       "\n",
       "     Pseudogen          CDS/miscRNA_coord                          Function  \n",
       "5286        NO  LinJ.27:1153523:1154887:+  hypothetical protein - conserved  \n",
       "5287        NO  LinJ.27:1153982:1154887:+  hypothetical protein - conserved  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_coord</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Transcript_coord</th>\n",
       "      <th>CDS/miscRNA_ID</th>\n",
       "      <th>Pseudogen</th>\n",
       "      <th>CDS/miscRNA_coord</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5286</th>\n",
       "      <td>LinJ.27</td>\n",
       "      <td>LINF_270033600-700</td>\n",
       "      <td>LinJ.27:1153377:1155197:+</td>\n",
       "      <td>LINF_27T0033600-700_1</td>\n",
       "      <td>LinJ.27:1153377:1155197:+</td>\n",
       "      <td>LINF_270033600</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.27:1153523:1154887:+</td>\n",
       "      <td>hypothetical protein - conserved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5287</th>\n",
       "      <td>LinJ.27</td>\n",
       "      <td>LINF_270033600-700</td>\n",
       "      <td>LinJ.27:1153377:1155197:+</td>\n",
       "      <td>LINF_27T0033600-700_2</td>\n",
       "      <td>LinJ.27:1153377:1155197:+</td>\n",
       "      <td>LINF_270033700</td>\n",
       "      <td>NO</td>\n",
       "      <td>LinJ.27:1153982:1154887:+</td>\n",
       "      <td>hypothetical protein - conserved</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 126
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T10:07:20.180272Z",
     "start_time": "2024-09-27T10:07:20.146832Z"
    }
   },
   "source": [
    "# Remove duplicated rows\n",
    "gtf_df = gtf_df.drop_duplicates().copy()"
   ],
   "outputs": [],
   "execution_count": 127
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-09-27T10:07:20.279884Z",
     "start_time": "2024-09-27T10:07:20.181569Z"
    }
   },
   "source": [
    "# Exporting table\n",
    "import csv\n",
    "gtf_df.to_csv(\"./data/20240703111001_LINF-Tabla_maestra_v3-20244_RP_v0.8.gtf\", sep=\"\\t\", index=False, header=False, quoting=csv.QUOTE_NONE, escapechar='\\\\')"
   ],
   "outputs": [],
   "execution_count": 128
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
