{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Needed libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import subprocess\n",
    "import os\n",
    "\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Needed functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "def blastn_dic(path_input):\n",
    "    os.system(\"makeblastdb -in \" + path_input + \" -dbtype nucl -parse_seqids\")\n",
    "    print(\"\\nBlast Dictionary created in\", path_input)\n",
    "    \n",
    "def blastn_blaster(query, db, perc_indentity):\n",
    "    cmd = \"blastn -word_size 11 \" \\\n",
    "    + \" -query \" + query \\\n",
    "    + \" -db \"  + db \\\n",
    "    + \" -perc_identity \" + str(perc_indentity) \\\n",
    "    + \" -outfmt '10 qseqid sseqid pident length qlen slen mismatch gapopen qstart qend sstart send evalue bitscore qcovhsp sstrand sseq'\"\n",
    "    data = subprocess.check_output(cmd, shell=True, universal_newlines=True)\n",
    "    data = pd.DataFrame([x.split(\",\") for x in data.split(\"\\n\") if x])\n",
    "\n",
    "    data.columns = [\"qseqid\", \"sseqid\", \"pident\", \"length\", \"qlen\", \"slen\", \"mismatch\", \"gapopen\", \"qstart\", \"qend\", \"sstart\", \"send\", \"evalue\", \"bitscore\", \"qcovhsp\", \"sstrand\", \"sseq\"]\n",
    "\n",
    "    return data\n",
    "\n",
    "# Get fasta f# Define fasta_creator file from the first chromosome\n",
    "def fasta_creator(data, fasta_output_path):\n",
    "    matrix = []\n",
    "    for index, sequence in data.iterrows():\n",
    "        rec = SeqRecord(Seq(sequence[5]),  # In the 5 position is the seq\n",
    "                        id=f\"Seq_{index}_{sequence[0]}\",\n",
    "                        description=\"Leishmania infantum\"\n",
    "                        )\n",
    "        matrix.append(rec)\n",
    "    SeqIO.write(matrix, fasta_output_path, \"fasta\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Import data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1951, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>173</td>\n",
       "      <td>1</td>\n",
       "      <td>173</td>\n",
       "      <td>plus</td>\n",
       "      <td>ACACCAGTACACCAGTACACCAGTACACCAGTACACCAGTACACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>699</td>\n",
       "      <td>24093</td>\n",
       "      <td>24791</td>\n",
       "      <td>plus</td>\n",
       "      <td>GGGGGAGGCGGGGGAGGCGGGGGGCACGCACCTCCATGCGTGGCAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>845</td>\n",
       "      <td>35316</td>\n",
       "      <td>36160</td>\n",
       "      <td>plus</td>\n",
       "      <td>GAGGTGGAGGCCGCTCTGCCCCCCCCGCCGCCGAGTGCTGCAGGCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>892</td>\n",
       "      <td>39698</td>\n",
       "      <td>40589</td>\n",
       "      <td>plus</td>\n",
       "      <td>CTCACCCTCATCCCACCCCTCCTCGTCCATCGACGGGAGTGGGCGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LinJ.01</td>\n",
       "      <td>888</td>\n",
       "      <td>54885</td>\n",
       "      <td>55772</td>\n",
       "      <td>plus</td>\n",
       "      <td>TGTTTGGTCTTCCGCGTGTCCGTTTTCGCTGCCGCACACTGCGAGG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         0    1      2      3     4  \\\n",
       "0  LinJ.01  173      1    173  plus   \n",
       "1  LinJ.01  699  24093  24791  plus   \n",
       "2  LinJ.01  845  35316  36160  plus   \n",
       "3  LinJ.01  892  39698  40589  plus   \n",
       "4  LinJ.01  888  54885  55772  plus   \n",
       "\n",
       "                                                   5  \n",
       "0  ACACCAGTACACCAGTACACCAGTACACCAGTACACCAGTACACCA...  \n",
       "1  GGGGGAGGCGGGGGAGGCGGGGGGCACGCACCTCCATGCGTGGCAT...  \n",
       "2  GAGGTGGAGGCCGCTCTGCCCCCCCCGCCGCCGAGTGCTGCAGGCA...  \n",
       "3  CTCACCCTCATCCCACCCCTCCTCGTCCATCGACGGGAGTGGGCGG...  \n",
       "4  TGTTTGGTCTTCCGCGTGTCCGTTTTCGCTGCCGCACACTGCGAGG...  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_csv(\"./dict/positives_testing_elements.csv\", header=None)\n",
    "print(data.shape)\n",
    "data.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Group by chromosomes in column 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_grouped = data.groupby(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "def save_sequences_to_csv_pandas(data, filename):\n",
    "    # Convert the list of lists to a DataFrame\n",
    "    df = pd.DataFrame(data)\n",
    "    # Save the DataFrame to a CSV file\n",
    "    df.to_csv(filename, header=False, index=False)\n",
    "\n",
    "def find_maximal_sets(sequences):\n",
    "    # Identify maximal sets that are not subsets of any other set in the list\n",
    "    maximal_sets = []\n",
    "    for i, seq in enumerate(sequences):\n",
    "        is_subset = False\n",
    "        for j, other_seq in enumerate(sequences):\n",
    "            # print(f\"For index {i} and {j}, sequences are {seq} and {other_seq}\")\n",
    "            if i != j and set(seq).issubset(set(other_seq)):\n",
    "                # print(\"\\tChecking if it's a subset\")\n",
    "                is_subset = True\n",
    "                break\n",
    "        if not is_subset:\n",
    "            # print(f\"\\tAdding {seq} to maximal_sets\")\n",
    "            maximal_sets.append(seq)\n",
    "    return maximal_sets\n",
    "\n",
    "def count_sequences(sequences):\n",
    "    from collections import Counter\n",
    "    # Flatten the list of sequences\n",
    "    flat_list = [seq for sublist in sequences for seq in sublist]\n",
    "    # Count each sequence using Counter\n",
    "    sequence_counts = Counter(flat_list)\n",
    "    # Identify sequences that appear more than once\n",
    "    multiple_occurrences = {seq: count for seq, count in sequence_counts.items() if count > 1}\n",
    "    return multiple_occurrences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chromosome LinJ.01 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:36\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.01/positives_LinJ.01.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.01/positives_LinJ.01.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.01/positives_LinJ.01.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 12 sequences in 0.00178313 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.01/positives_LinJ.01.fasta\n",
      "Chromosome LinJ.02 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:37\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.02/positives_LinJ.02.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.02/positives_LinJ.02.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.02/positives_LinJ.02.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 17 sequences in 0.000966072 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.02/positives_LinJ.02.fasta\n",
      "Chromosome LinJ.03 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:37\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.03/positives_LinJ.03.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.03/positives_LinJ.03.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.03/positives_LinJ.03.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 8 sequences in 0.000744104 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.03/positives_LinJ.03.fasta\n",
      "Chromosome LinJ.04 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:37\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.04/positives_LinJ.04.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.04/positives_LinJ.04.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.04/positives_LinJ.04.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 10 sequences in 0.00114799 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.04/positives_LinJ.04.fasta\n",
      "Chromosome LinJ.05 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:37\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.05/positives_LinJ.05.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.05/positives_LinJ.05.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.05/positives_LinJ.05.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 19 sequences in 0.00220704 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.05/positives_LinJ.05.fasta\n",
      "Chromosome LinJ.06 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:37\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.06/positives_LinJ.06.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.06/positives_LinJ.06.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.06/positives_LinJ.06.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 23 sequences in 0.00219202 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.06/positives_LinJ.06.fasta\n",
      "Chromosome LinJ.07 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:37\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.07/positives_LinJ.07.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.07/positives_LinJ.07.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.07/positives_LinJ.07.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 32 sequences in 0.000953913 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.07/positives_LinJ.07.fasta\n",
      "Chromosome LinJ.08 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:38\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.08/positives_LinJ.08.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.08/positives_LinJ.08.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.08/positives_LinJ.08.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 31 sequences in 0.00233078 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.08/positives_LinJ.08.fasta\n",
      "Chromosome LinJ.09 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:38\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.09/positives_LinJ.09.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.09/positives_LinJ.09.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.09/positives_LinJ.09.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 29 sequences in 0.002424 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.09/positives_LinJ.09.fasta\n",
      "Chromosome LinJ.10 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:38\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.10/positives_LinJ.10.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.10/positives_LinJ.10.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.10/positives_LinJ.10.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 33 sequences in 0.00258493 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.10/positives_LinJ.10.fasta\n",
      "Chromosome LinJ.11 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:38\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.11/positives_LinJ.11.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.11/positives_LinJ.11.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.11/positives_LinJ.11.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 39 sequences in 0.00121713 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.11/positives_LinJ.11.fasta\n",
      "Chromosome LinJ.12 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:39\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.12/positives_LinJ.12.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.12/positives_LinJ.12.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.12/positives_LinJ.12.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 40 sequences in 0.00139594 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.12/positives_LinJ.12.fasta\n",
      "Chromosome LinJ.13 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:39\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.13/positives_LinJ.13.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.13/positives_LinJ.13.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.13/positives_LinJ.13.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 30 sequences in 0.001086 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.13/positives_LinJ.13.fasta\n",
      "Chromosome LinJ.14 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:39\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.14/positives_LinJ.14.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.14/positives_LinJ.14.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.14/positives_LinJ.14.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 40 sequences in 0.00148201 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.14/positives_LinJ.14.fasta\n",
      "Chromosome LinJ.15 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:39\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.15/positives_LinJ.15.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.15/positives_LinJ.15.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.15/positives_LinJ.15.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 25 sequences in 0.00112605 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.15/positives_LinJ.15.fasta\n",
      "Chromosome LinJ.16 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:40\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.16/positives_LinJ.16.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.16/positives_LinJ.16.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.16/positives_LinJ.16.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 33 sequences in 0.00162697 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.16/positives_LinJ.16.fasta\n",
      "Chromosome LinJ.17 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:40\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.17/positives_LinJ.17.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.17/positives_LinJ.17.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.17/positives_LinJ.17.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 42 sequences in 0.0030179 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.17/positives_LinJ.17.fasta\n",
      "Chromosome LinJ.18 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:40\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.18/positives_LinJ.18.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.18/positives_LinJ.18.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.18/positives_LinJ.18.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 57 sequences in 0.00203705 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.18/positives_LinJ.18.fasta\n",
      "Chromosome LinJ.19 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:41\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.19/positives_LinJ.19.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.19/positives_LinJ.19.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.19/positives_LinJ.19.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 65 sequences in 0.00209498 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.19/positives_LinJ.19.fasta\n",
      "Chromosome LinJ.20 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:41\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.20/positives_LinJ.20.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.20/positives_LinJ.20.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.20/positives_LinJ.20.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 43 sequences in 0.00155711 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.20/positives_LinJ.20.fasta\n",
      "Chromosome LinJ.21 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:41\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.21/positives_LinJ.21.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.21/positives_LinJ.21.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.21/positives_LinJ.21.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 40 sequences in 0.00105619 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.21/positives_LinJ.21.fasta\n",
      "Chromosome LinJ.22 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:42\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.22/positives_LinJ.22.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.22/positives_LinJ.22.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.22/positives_LinJ.22.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 52 sequences in 0.00347114 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.22/positives_LinJ.22.fasta\n",
      "Chromosome LinJ.23 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:42\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.23/positives_LinJ.23.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.23/positives_LinJ.23.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.23/positives_LinJ.23.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 68 sequences in 0.00236106 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.23/positives_LinJ.23.fasta\n",
      "Chromosome LinJ.24 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:43\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.24/positives_LinJ.24.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.24/positives_LinJ.24.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.24/positives_LinJ.24.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 52 sequences in 0.00374699 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.24/positives_LinJ.24.fasta\n",
      "Chromosome LinJ.25 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:43\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.25/positives_LinJ.25.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.25/positives_LinJ.25.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.25/positives_LinJ.25.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 39 sequences in 0.0012958 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.25/positives_LinJ.25.fasta\n",
      "Chromosome LinJ.26 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:43\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.26/positives_LinJ.26.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.26/positives_LinJ.26.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.26/positives_LinJ.26.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 48 sequences in 0.00120497 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.26/positives_LinJ.26.fasta\n",
      "Chromosome LinJ.27 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:44\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.27/positives_LinJ.27.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.27/positives_LinJ.27.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.27/positives_LinJ.27.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 67 sequences in 0.00275803 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.27/positives_LinJ.27.fasta\n",
      "Chromosome LinJ.28 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:44\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.28/positives_LinJ.28.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.28/positives_LinJ.28.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.28/positives_LinJ.28.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 49 sequences in 0.00166392 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.28/positives_LinJ.28.fasta\n",
      "Chromosome LinJ.29 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:44\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.29/positives_LinJ.29.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.29/positives_LinJ.29.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.29/positives_LinJ.29.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 116 sequences in 0.00450897 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.29/positives_LinJ.29.fasta\n",
      "Chromosome LinJ.30 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:46\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.30/positives_LinJ.30.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.30/positives_LinJ.30.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.30/positives_LinJ.30.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 92 sequences in 0.00217104 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.30/positives_LinJ.30.fasta\n",
      "Chromosome LinJ.31 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:46\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.31/positives_LinJ.31.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.31/positives_LinJ.31.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.31/positives_LinJ.31.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 104 sequences in 0.0040009 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.31/positives_LinJ.31.fasta\n",
      "Chromosome LinJ.32 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:47\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.32/positives_LinJ.32.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.32/positives_LinJ.32.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.32/positives_LinJ.32.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 83 sequences in 0.00184298 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.32/positives_LinJ.32.fasta\n",
      "Chromosome LinJ.33 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:48\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.33/positives_LinJ.33.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.33/positives_LinJ.33.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.33/positives_LinJ.33.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 100 sequences in 0.00207496 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.33/positives_LinJ.33.fasta\n",
      "Chromosome LinJ.34 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:49\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.34/positives_LinJ.34.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.34/positives_LinJ.34.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.34/positives_LinJ.34.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 146 sequences in 0.00308204 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.34/positives_LinJ.34.fasta\n",
      "Chromosome LinJ.35 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:50\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.35/positives_LinJ.35.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.35/positives_LinJ.35.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.35/positives_LinJ.35.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 120 sequences in 0.00250816 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.35/positives_LinJ.35.fasta\n",
      "Chromosome LinJ.36 folder created\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/03/2024 15:19:52\n",
      "New DB name:   /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.36/positives_LinJ.36.fasta\n",
      "New DB title:  ./dict/chromosomes/LinJ.36/positives_LinJ.36.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/rfpacheco/Documents/CBM_Work/Testing_Leishmania_project/9.subfamilies_global/dict/chromosomes/LinJ.36/positives_LinJ.36.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 147 sequences in 0.00472093 seconds.\n",
      "\n",
      "\n",
      "\n",
      "Blast Dictionary created in ./dict/chromosomes/LinJ.36/positives_LinJ.36.fasta\n"
     ]
    }
   ],
   "source": [
    "repeated_chr = []\n",
    "for index, group in data_grouped:\n",
    "    cmd = [\"mkdir\", \"-p\", f\"./dict/chromosomes/{index}\"]\n",
    "    subprocess.run(cmd)  # creating a folder for each chromosome\n",
    "    print(f\"Chromosome {index} folder created\")\n",
    "\n",
    "    fasta_creator(group, f\"./dict/chromosomes/{index}/positives_{index}.fasta\")  # creating a for each chromosome\n",
    "    blastn_dic(f\"./dict/chromosomes/{index}/positives_{index}.fasta\")  # creating a BLASTN dict for each chromosome\n",
    "\n",
    "    data = blastn_blaster(f\"./dict/chromosomes/{index}/positives_{index}.fasta\", f\"./dict/chromosomes/{index}/positives_{index}.fasta\", 85)  # BLASTN for each chromosome\n",
    "    data = data[data[\"length\"].astype(int) > 100]  # filtering by length\n",
    "    # data = data.query(\"qseqid != sseqid\")  # filtering by qseqid != sseqid\n",
    "\n",
    "    dict = {}\n",
    "    for seq in data[\"qseqid\"].unique():\n",
    "        values = data[data[\"qseqid\"] == seq].loc[:, [\"sseqid\"]].values.flatten().tolist()\n",
    "        values = sorted(values)\n",
    "        dict[seq] = values\n",
    "    dict = {key: sorted(value, key=lambda x: int(x.split('_')[1])) for key, value in dict.items()}   # Sort the values by the number of the sequence\n",
    "\n",
    "    dataset = []\n",
    "    for key, value in dict.items():\n",
    "        if value in dataset:\n",
    "            continue  # If it's in the dataset, it doesn't do anything.\n",
    "        else:\n",
    "            dataset.append(value)  # If it's not in the dataset, it appends it.\n",
    "    \n",
    "    dataset = find_maximal_sets(dataset)  # Find the maximal sets\n",
    "\n",
    "    counters = count_sequences(dataset)  # Count the sequences if there is >1\n",
    "    if len(counters) > 0:\n",
    "        repeated_chr.append(index)\n",
    "        dataset.append(\"Repeated values:\")\n",
    "        combined_list = [item for pair in zip(counters.keys(), counters.values()) for item in pair]\n",
    "        dataset.append(combined_list)  # Append the sequences that appear more than once\n",
    "\n",
    "    \n",
    "    save_sequences_to_csv_pandas(dataset, f\"./dict/chromosomes/{index}/families_{index}.csv\")  # saving the dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['LinJ.18',\n",
       " 'LinJ.19',\n",
       " 'LinJ.21',\n",
       " 'LinJ.22',\n",
       " 'LinJ.23',\n",
       " 'LinJ.24',\n",
       " 'LinJ.29',\n",
       " 'LinJ.30',\n",
       " 'LinJ.31',\n",
       " 'LinJ.33',\n",
       " 'LinJ.34',\n",
       " 'LinJ.35',\n",
       " 'LinJ.36']"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(repeated_chr))\n",
    "repeated_chr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "# test1 = [807, 808, 809]\n",
    "test1 = [10, 20, 30]\n",
    "test2 = [807, 809, 810]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    False\n",
       "1    False\n",
       "2    False\n",
       "dtype: bool"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test1_series = pd.Series(test1)\n",
    "test2_series = pd.Series(test2)\n",
    "\n",
    "test1_series.apply(lambda x: x in test2)\n",
    "# test1.apply(lambda x: x in test2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test1_series.apply(lambda x: x in test2)[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test1_series.apply(lambda x: x in test2).value_counts().iloc[0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False\n"
     ]
    }
   ],
   "source": [
    "try :\n",
    "    test1_series.apply(lambda x: x in test2).value_counts().loc[True]\n",
    "    print(\"True\")\n",
    "except:\n",
    "    print(\"False\")  \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0    807\n",
      "1    808\n",
      "2    809\n",
      "dtype: int64\n",
      "0    807\n",
      "1    809\n",
      "2    810\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print(test1_series, test2_series, sep=\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "dict_test = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'test1': [807, 808, 809]}"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dict_test[\"test1\"] = test1\n",
    "dict_test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "dict_test[\"test1\"].append(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'test1': [807, 808, 809, 10]}"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dict_test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "dict_test2 = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'test1': [1231]}"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dict_test2[\"test1\"] = []\n",
    "dict_test2[\"test1\"].append(1231)\n",
    "dict_test2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1231]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dict_test2[\"test1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n"
     ]
    }
   ],
   "source": [
    "is_subset = False\n",
    "if not is_subset:\n",
    "    print(\"1\")\n",
    "else:\n",
    "    print(\"2\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
